// @scalar/components/dist/components/ScalarButton/ScalarButton.vue.js
import { defineComponent as f, computed as g, useAttrs as c, openBlock as a, createElementBlock as s, mergeProps as v, unref as i, normalizeClass as p, renderSlot as l, createCommentVNode as m, createVNode as b } from "vue";
import { cx as y } from "../../cva.js";
import { variants as h } from "./variants.js";
import k from "../ScalarLoading/ScalarLoading.vue.js";
const z = ["ariaDisabled", "type"], B = {
  key: 3,
  class: "centered-x absolute"
}, W = /* @__PURE__ */ f({
  inheritAttrs: !1,
  __name: "ScalarButton",
  props: {
    disabled: { type: Boolean },
    fullWidth: { type: Boolean, default: !1 },
    loading: {},
    size: { default: "md" },
    variant: { default: "solid" },
    type: { default: "button" }
  },
  setup(L) {
    const o = g(() => {
      const { class: e, ...t } = c();
      return { class: e || "", rest: t };
    });
    return (e, t) => {
      var n, d, r, u;
      return a(), s("button", v(o.value.rest, {
        ariaDisabled: e.disabled || void 0,
        class: i(y)(
          i(h)({ fullWidth: e.fullWidth, disabled: e.disabled, size: e.size, variant: e.variant }),
          { relative: (n = e.loading) == null ? void 0 : n.isLoading },
          `${o.value.class}`
        ),
        type: e.type
      }), [
        e.$slots.icon ? (a(), s("div", {
          key: 0,
          class: p(["mr-2 h-4 w-4", { invisible: (d = e.loading) == null ? void 0 : d.isLoading }])
        }, [
          l(e.$slots, "icon")
        ], 2)) : m("", !0),
        e.loading ? (a(), s("span", {
          key: 1,
          class: p({ invisible: (r = e.loading) == null ? void 0 : r.isLoading })
        }, [
          l(e.$slots, "default")
        ], 2)) : l(e.$slots, "default", { key: 2 }),
        (u = e.loading) != null && u.isLoading ? (a(), s("div", B, [
          b(i(k), {
            loadingState: e.loading,
            size: "12px"
          }, null, 8, ["loadingState"])
        ])) : m("", !0)
      ], 16, z);
    };
  }
});
export {
  W as default
};
