// @scalar/components/dist/components/ScalarButton/ScalarButton.vue.js
import { defineComponent as f, createBlock as r, openBlock as t, resolveDynamicComponent as d, mergeProps as z, unref as a, withCtx as g, createElementBlock as s, createCommentVNode as c, renderSlot as n, normalizeClass as o, createVNode as x } from "vue";
import { cva as l, useBindCx as y } from "@scalar/use-hooks/useBindCx";
import { BUTTON_VARIANT_STYLES as h } from "./constants.js";
import k from "../ScalarLoading/ScalarLoading.vue.js";
const A = {
  key: 3,
  class: "centered"
}, B = {}, T = /* @__PURE__ */ f({
  ...B,
  inheritAttrs: !1,
  __name: "ScalarButton",
  props: {
    is: { default: "button" },
    variant: { default: "solid" },
    size: { default: "md" },
    disabled: { type: Boolean },
    icon: { type: [Object, Function] },
    loader: {}
  },
  setup(e) {
    const m = l({
      base: "scalar-button flex cursor-pointer items-center justify-center rounded font-medium -outline-offset-1",
      variants: {
        disabled: { true: "bg-b-2 text-color-3 shadow-none" },
        size: {
          xs: "px-2 py-1 text-xs leading-5",
          sm: "px-3.5 py-2 text-sm leading-5",
          md: "px-5 py-3 text-sm leading-5"
        },
        variant: h
      },
      compoundVariants: [
        {
          disabled: !0,
          variant: ["solid", "outlined", "ghost", "gradient", "danger"],
          class: "bg-b-2 text-c-3 shadow-none hover:bg-b-[_] cursor-not-allowed active:bg-b-[_] hover:text-c-[_] active:text-c-[_]"
        },
        {
          disabled: !0,
          variant: ["gradient"],
          class: "to-b-1.5 bg-linear-to-b hover:bg-linear-to-b dark:hover:bg-linear-to-t"
        }
      ]
    }), u = l({
      base: "shrink-0",
      variants: {
        size: {
          xs: "size-2.75 -ml-0.25 mr-1",
          sm: "size-3.25 -ml-0.5 mr-1.5",
          md: "size-3.5 -ml-0.5 mr-1.5"
        }
      }
    }), b = l({
      variants: {
        size: {
          xs: "size-4",
          sm: "size-5",
          md: "size-6"
        }
      }
    }), { cx: v } = y();
    return (i, V) => (t(), r(d(e.is), z(
      {
        "aria-disabled": e.disabled || void 0,
        type: e.is === "button" ? "button" : void 0
      },
      a(v)(a(m)({ disabled: e.disabled, size: e.size, variant: e.variant }), {
        relative: e.loader?.isActive
      })
    ), {
      default: g(() => [
        i.$slots.icon || e.icon ? (t(), s("div", {
          key: 0,
          class: o([a(u)({ size: e.size }), { invisible: e.loader?.isActive }])
        }, [
          n(i.$slots, "icon", {}, () => [
            (t(), r(d(e.icon), { class: "size-full" }))
          ])
        ], 2)) : c("", !0),
        e.loader ? (t(), s("span", {
          key: 1,
          class: o({ invisible: e.loader?.isActive })
        }, [
          n(i.$slots, "default")
        ], 2)) : n(i.$slots, "default", { key: 2 }),
        e.loader?.isActive ? (t(), s("div", A, [
          x(a(k), {
            class: o(a(b)({ size: e.size })),
            loader: e.loader
          }, null, 8, ["class", "loader"])
        ])) : c("", !0)
      ]),
      _: 3
    }, 16, ["aria-disabled", "type"]));
  }
});
export {
  T as default
};
