// @scalar/api-reference/dist/components/Content/Content.vue2.js
import { defineComponent as O, computed as m, openBlock as r, createElementBlock as n, Fragment as c, createElementVNode as k, renderSlot as f, createVNode as l, unref as o, createBlock as s, withCtx as p, normalizeClass as T, renderList as b, resolveDynamicComponent as L, createCommentVNode as C, createStaticVNode as E } from "vue";
import R from "./Tag/TagAccordion.vue.js";
import W from "./Tag/Tag.vue.js";
import D from "./Operation/OperationAccordion.vue.js";
import F from "./Operation/Operation.vue.js";
import j from "./Lazy/Loading.vue.js";
/* empty css                  */
import q from "./Introduction/Introduction.vue.js";
import G from "../../features/BaseUrl/BaseUrl.vue.js";
import H from "./Authentication/Authentication.vue.js";
import J from "./ClientLibraries/ClientLibraries.vue.js";
import $ from "./Lazy/Lazy.vue.js";
import K from "./Webhooks/Webhooks.vue.js";
import P from "./Models/ModelsAccordion.vue.js";
import Q from "./Models/Models.vue.js";
import { useSidebar as X } from "../../hooks/useSidebar.js";
import { hasModels as Y } from "../../helpers/hasModels.js";
import { getModels as w } from "../../helpers/getModels.js";
import { useNavState as Z } from "../../hooks/useNavState.js";
const _ = /* @__PURE__ */ E('<div class="section-flare" data-v-8d751fa0><div class="section-flare-item" data-v-8d751fa0></div><div class="section-flare-item" data-v-8d751fa0></div><div class="section-flare-item" data-v-8d751fa0></div><div class="section-flare-item" data-v-8d751fa0></div><div class="section-flare-item" data-v-8d751fa0></div><div class="section-flare-item" data-v-8d751fa0></div><div class="section-flare-item" data-v-8d751fa0></div><div class="section-flare-item" data-v-8d751fa0></div></div>', 1), x = { class: "narrow-references-container" }, be = /* @__PURE__ */ O({
  __name: "Content",
  props: {
    parsedSpec: {},
    layout: {},
    baseServerURL: {},
    servers: {},
    proxy: {}
  },
  setup(z) {
    const i = z, { getOperationId: u, getTagId: d, hash: I } = Z(), { hideModels: M, collapsedSidebarItems: N } = X(), V = m(
      () => i.layout === "accordion" ? R : W
    ), A = m(
      () => i.layout === "accordion" ? D : F
    ), U = m(
      () => i.layout === "accordion" ? "after" : "aside"
    ), g = i.layout !== "accordion" && !I.value.startsWith("model");
    return (e, ee) => {
      var v, y, S, h;
      return r(), n(c, null, [
        _,
        k("div", x, [
          f(e.$slots, "start", {}, void 0, !0),
          l(o(j), {
            layout: e.layout,
            parsedSpec: e.parsedSpec
          }, null, 8, ["layout", "parsedSpec"]),
          (y = (v = e.parsedSpec) == null ? void 0 : v.info) != null && y.title || (h = (S = e.parsedSpec) == null ? void 0 : S.info) != null && h.description ? (r(), s(o(q), {
            key: 0,
            info: e.parsedSpec.info,
            parsedSpec: e.parsedSpec
          }, {
            [U.value]: p(() => [
              k("div", {
                class: T(["introduction-card", { "introduction-card-row": e.layout === "accordion" }])
              }, [
                l(o(G), {
                  class: "introduction-card-item",
                  defaultServerUrl: e.baseServerURL,
                  servers: i.servers,
                  specification: e.parsedSpec
                }, null, 8, ["defaultServerUrl", "servers", "specification"]),
                l(o(H), {
                  class: "introduction-card-item",
                  parsedSpec: e.parsedSpec,
                  proxy: e.proxy
                }, null, 8, ["parsedSpec", "proxy"]),
                l(o(J), { class: "introduction-card-item" })
              ], 2)
            ]),
            _: 2
          }, 1032, ["info", "parsedSpec"])) : f(e.$slots, "empty-state", { key: 1 }, void 0, !0),
          (r(!0), n(c, null, b(e.parsedSpec.tags, (a) => (r(), s(o($), {
            id: o(d)(a),
            key: o(d)(a),
            isLazy: o(g) && !o(N)[o(d)(a)]
          }, {
            default: p(() => [
              (r(), s(L(V.value), {
                id: o(d)(a),
                spec: e.parsedSpec,
                tag: a
              }, {
                default: p(() => [
                  (r(!0), n(c, null, b(a.operations, (t, B) => (r(), s(o($), {
                    id: o(u)(t, a),
                    key: `${t.httpVerb}-${t.operationId}`,
                    isLazy: B > 0
                  }, {
                    default: p(() => [
                      (r(), s(L(A.value), {
                        id: o(u)(t, a),
                        operation: t,
                        tag: a
                      }, null, 8, ["id", "operation", "tag"]))
                    ]),
                    _: 2
                  }, 1032, ["id", "isLazy"]))), 128))
                ]),
                _: 2
              }, 1032, ["id", "spec", "tag"]))
            ]),
            _: 2
          }, 1032, ["id", "isLazy"]))), 128)),
          e.parsedSpec.webhooks ? (r(), s(o(K), {
            key: 2,
            webhooks: e.parsedSpec.webhooks
          }, null, 8, ["webhooks"])) : C("", !0),
          o(Y)(e.parsedSpec) && !o(M) ? (r(), n(c, { key: 3 }, [
            e.layout === "accordion" ? (r(), s(o(P), {
              key: 0,
              schemas: o(w)(e.parsedSpec)
            }, null, 8, ["schemas"])) : (r(), s(o(Q), {
              key: 1,
              schemas: o(w)(e.parsedSpec)
            }, null, 8, ["schemas"]))
          ], 64)) : C("", !0),
          f(e.$slots, "end", {}, void 0, !0)
        ])
      ], 64);
    };
  }
});
export {
  be as default
};
