// @scalar/api-client/dist/components/CodeInput/CodeInput.vue2.js
import { defineComponent as E, useAttrs as q, ref as y, toRef as o, watch as I, computed as M, openBlock as t, createElementBlock as s, Fragment as N, createBlock as g, mergeProps as T, unref as A, renderSlot as v, createCommentVNode as B } from "vue";
import { useCodeMirror as U, colorPicker as W } from "@scalar/use-codemirror";
import { nanoid as j } from "nanoid";
import V from "../DataTable/DataTableInputSelect.vue.js";
import { dropdownPlugin as z } from "./codeDropdownWidget.js";
import { pillPlugin as D, backspaceCommand as G } from "./codeVariableWidget.js";
import { useWorkspace as H } from "../../store/workspace.js";
const J = ["id"], K = {
  key: 3,
  class: "absolute centered-y right-7 text-orange text-xs"
}, L = {
  key: 4,
  class: "required absolute centered-y right-0 pt-px pr-2 text-xxs text-c-3 bg-b-1 shadow-[-8px_0_4px_var(--scalar-background-1)] opacity-100 duration-150 transition-opacity peer-has-[.cm-focused]:opacity-0 pointer-events-none"
}, Q = {
  inheritAttrs: !1
}, ae = /* @__PURE__ */ E({
  ...Q,
  __name: "CodeInput",
  props: {
    colorPicker: { type: Boolean, default: !1 },
    modelValue: {},
    error: { type: Boolean },
    emitOnBlur: { type: Boolean, default: !0 },
    lineNumbers: { type: Boolean },
    lint: { type: Boolean },
    disableTabIndent: { type: Boolean, default: !1 },
    language: {},
    handleFieldSubmit: {},
    handleFieldChange: {},
    placeholder: {},
    required: { type: Boolean },
    disableEnter: { type: Boolean, default: !1 },
    disableCloseBrackets: { type: Boolean, default: !1 },
    enum: {},
    type: {},
    nullable: { type: Boolean, default: !1 },
    withVariables: { type: Boolean, default: !0 },
    withServers: { type: Boolean, default: !1 }
  },
  emits: ["submit", "update:modelValue"],
  setup(k, { expose: w, emit: C }) {
    var b;
    const e = k, r = C, d = q(), _ = d.id || `id-${j()}`, p = y(!1), {
      activeWorkspace: x,
      activeParsedEnvironments: c,
      isReadOnly: F,
      environments: m,
      router: S
    } = H();
    function P(l) {
      return l === e.modelValue ? null : e.handleFieldChange ? e.handleFieldChange(l) : r("update:modelValue", l);
    }
    function O(l) {
      return e.handleFieldSubmit ? e.handleFieldSubmit(l) : r("submit", l);
    }
    function R(l) {
      p.value = !1, e.emitOnBlur && e.modelValue && O(l);
    }
    const a = [];
    e.colorPicker && a.push(W), e.withVariables && !x.value.isReadOnly && a.push(
      z({
        withServers: e.withServers,
        activeParsedEnvironments: c,
        environments: m,
        router: S
      })
    ), a.push(
      D({ environments: m, activeParsedEnvironments: c, isReadOnly: F })
    ), a.push(G);
    const f = y(null), { codeMirror: n } = U({
      content: o(
        () => e.modelValue !== void 0 ? String(e.modelValue) : ""
      ),
      onChange: P,
      onFocus: () => p.value = !0,
      onBlur: (l) => R(l),
      codeMirrorRef: f,
      disableTabIndent: o(() => e.disableTabIndent),
      disableEnter: o(() => e.disableEnter),
      disableCloseBrackets: o(() => e.disableCloseBrackets),
      lineNumbers: o(() => e.lineNumbers),
      language: o(() => e.language),
      lint: o(() => e.lint),
      extensions: a,
      placeholder: o(() => e.placeholder)
    });
    (b = n.value) == null || b.focus(), I(n, () => {
      n.value && Object.prototype.hasOwnProperty.call(d, "autofocus") && n.value.focus();
    });
    const $ = M(() => {
      var l;
      return e.type === "boolean" || (l = e.type) != null && l.includes("boolean") || e.nullable ? ["true", "false", "null"] : ["true", "false"];
    });
    return w({
      focus: () => {
        var l;
        (l = n.value) == null || l.focus();
      }
    }), (l, u) => {
      var h;
      return t(), s(N, null, [
        e.enum && e.enum.length ? (t(), g(V, {
          key: 0,
          modelValue: e.modelValue,
          value: e.enum,
          "onUpdate:modelValue": u[0] || (u[0] = (i) => r("update:modelValue", i))
        }, null, 8, ["modelValue", "value"])) : e.type === "boolean" || (h = e.type) != null && h.includes("boolean") ? (t(), g(V, {
          key: 1,
          modelValue: e.modelValue,
          value: $.value,
          "onUpdate:modelValue": u[1] || (u[1] = (i) => r("update:modelValue", i))
        }, null, 8, ["modelValue", "value"])) : (t(), s("div", T({
          key: 2,
          id: A(_)
        }, l.$attrs, {
          ref_key: "codeMirrorRef",
          ref: f,
          class: ["peer font-code w-full whitespace-nowrap text-xs leading-[1.44] relative", {
            "flow-code-input--error": l.error
          }]
        }), null, 16, J)),
        l.$slots.warning ? (t(), s("div", K, [
          v(l.$slots, "warning", {}, void 0, !0)
        ])) : B("", !0),
        v(l.$slots, "icon", {}, void 0, !0),
        l.required ? (t(), s("div", L, " Required ")) : B("", !0)
      ], 64);
    };
  }
});
export {
  ae as default
};
