// @scalar/api-client/dist/components/CodeInput/codeVariableWidget.js
var u = Object.defineProperty;
var h = (n, e, t) => e in n ? u(n, e, { enumerable: !0, configurable: !0, writable: !0, value: t }) : n[e] = t;
var r = (n, e, t) => h(n, typeof e != "symbol" ? e + "" : e, t);
import { ScalarButton as f, ScalarIcon as g, ScalarTooltip as b } from "@scalar/components";
import { ViewPlugin as x, RangeSetBuilder as w, Decoration as y, EditorView as N, WidgetType as E } from "@scalar/use-codemirror";
import { defineComponent as C, h as a, createApp as S } from "vue";
const k = (n, e) => "_scalarEnvId" in n ? `bg-${e[n._scalarEnvId].color}` : "bg-grey";
class d extends E {
  constructor(t, i, s, o) {
    super();
    r(this, "app");
    r(this, "environments");
    r(this, "activeParsedEnvironments");
    r(this, "isReadOnly");
    this.variableName = t, this.variableName = t, this.environments = i, this.activeParsedEnvironments = s, this.isReadOnly = o;
  }
  toDOM() {
    const t = document.createElement("span");
    t.className = "cm-pill", t.textContent = `${this.variableName}`;
    const i = C({
      props: { variableName: { type: String, default: null } },
      render: () => {
        const s = this.activeParsedEnvironments.value.find(
          (c) => c.key === this.variableName
        );
        s && (t.className += ` ${k(s, this.environments)}`);
        const o = s ? a("div", { class: "p-2" }, s.value) : a("div", { class: "divide-y divide-1/2 grid" }, [
          a("span", { class: "p-2" }, "Variable not found"),
          !this.isReadOnly && a("div", { class: "p-1" }, [
            a(
              f,
              {
                class: "gap-1.5 justify-start font-normal px-1 py-1.5 h-auto transition-colors rounded no-underline text-xxs w-full hover:bg-b-2",
                variant: "ghost",
                onClick: () => {
                  window.location.href = "/environment";
                }
              },
              [
                a(g, { class: "w-2", icon: "Add", size: "xs" }),
                "Add variable"
              ]
            )
          ])
        ]);
        return a(
          b,
          {
            align: "start",
            class: "bg-b-1 w-full",
            delay: 0,
            side: "bottom",
            sideOffset: 6
          },
          {
            trigger: () => a("span", `${this.variableName}`),
            content: () => a(
              "div",
              {
                class: "w-content shadow-lg rounded bg-b-1 text-xxs leading-5 text-c-1"
              },
              o
            )
          }
        );
      }
    });
    return this.app = S(i, { variableName: this.variableName }), this.app.mount(t), t;
  }
  destroy() {
    this.app && this.app.unmount();
  }
  eq(t) {
    return t instanceof d && t.variableName === this.variableName;
  }
  ignoreEvent() {
    return !1;
  }
}
const $ = (n) => x.fromClass(
  class {
    constructor(e) {
      r(this, "decorations");
      this.decorations = this.buildDecorations(e);
    }
    update(e) {
      (e.docChanged || e.viewportChanged) && (this.decorations = this.buildDecorations(e.view));
    }
    buildDecorations(e) {
      const t = new w();
      for (const { from: i, to: s } of e.visibleRanges) {
        const o = e.state.doc.sliceString(i, s), c = /{{(.*?)}}/g;
        let l;
        for (; (l = c.exec(o)) !== null; ) {
          const m = i + l.index, v = m + l[0].length, p = l[1];
          t.add(
            m,
            v,
            y.widget({
              widget: new d(
                p,
                n.environments,
                n.activeParsedEnvironments,
                n.isReadOnly
              ),
              side: 1
            })
          );
        }
      }
      return t.finish();
    }
  },
  {
    decorations: (e) => e.decorations
  }
), A = N.domEventHandlers({
  keydown(n, e) {
    if (n.key === "Backspace") {
      const { state: t } = e, { from: i, to: s } = t.selection.main;
      if (i === s && i > 0 && t.doc.sliceString(i - 2, i) === "}}")
        return e.dispatch({
          changes: { from: i - 2, to: s },
          selection: { anchor: i - 2 }
        }), n.preventDefault(), !0;
    }
    return !1;
  }
});
export {
  A as backspaceCommand,
  $ as pillPlugin
};
