// @scalar/api-client/dist/components/CodeInput/codeDropdownWidget.js
var g = Object.defineProperty;
var f = (s, t, e) => t in s ? g(s, t, { enumerable: !0, configurable: !0, writable: !0, value: e }) : s[t] = e;
var o = (s, t, e) => f(s, typeof t != "symbol" ? t + "" : t, e);
import { ViewPlugin as p, Decoration as h, StateEffect as v, WidgetType as y } from "@scalar/use-codemirror";
import { createApp as T, h as w, Teleport as S } from "vue";
import q from "../../views/Environment/EnvironmentVariableDropdown.vue.js";
class u extends y {
  constructor(e, i, d, n, c, r) {
    super();
    o(this, "queryTerm");
    o(this, "onSelect");
    o(this, "dropdown", null);
    o(this, "withServers");
    o(this, "activeParsedEnvironments");
    o(this, "environments");
    o(this, "router");
    this.queryTerm = e, this.onSelect = i, this.withServers = r, this.activeParsedEnvironments = d, this.environments = n, this.router = c;
  }
  updateQueryTerm(e) {
    this.queryTerm !== e && (this.queryTerm = e);
  }
  getQueryTerm() {
    return this.queryTerm;
  }
  toDOM(e) {
    const i = document.createElement("span");
    return i.style.position = "fixed", setTimeout(() => {
      var c;
      const d = e.state.selection.main.head, n = e.coordsAtPos(d - this.queryTerm.length - 2);
      if (n) {
        const r = (c = document.querySelector(".scalar-client")) == null ? void 0 : c.getBoundingClientRect();
        r && (this.dropdown = T({
          render: () => w(S, { to: ".scalar-client" }, [
            w(q, {
              query: this.queryTerm,
              onSelect: this.onSelect,
              withServers: this.withServers,
              activeParsedEnvironments: this.activeParsedEnvironments,
              environments: this.environments,
              router: this.router,
              style: {
                position: "absolute",
                left: `${n.left - r.left}px`,
                top: `calc(${n.bottom - r.top}px + 6px)`
              }
            })
          ])
        })), this.dropdown.use(this.router), this.dropdown.mount(i);
      }
    }, 0), i;
  }
  destroy() {
    this.dropdown && (this.dropdown.unmount(), this.dropdown = null);
  }
  eq(e) {
    return e instanceof u && this.queryTerm === e.queryTerm;
  }
}
const C = (s) => p.fromClass(
  class {
    constructor(t) {
      o(this, "decorations");
      o(this, "widget", null);
      o(this, "view");
      this.view = t, this.decorations = h.none, this.handleDropdownSelect = this.handleDropdownSelect.bind(this);
    }
    handleDropdownSelect(t) {
      var m;
      if (!this.widget) return;
      const { state: e, dispatch: i } = this.view, d = e.selection.main.head, n = Math.max(0, d - this.widget.getQueryTerm().length - 2), c = d, r = `{{${t}}}`;
      i({
        changes: { from: n, to: c, insert: r }
      }), (m = this.widget) == null || m.destroy(), this.widget = null, this.decorations = h.none, this.view.dispatch({
        effects: v.appendConfig.of([])
      });
      const a = n + r.length, l = "";
      this.widget = new u(
        l,
        this.handleDropdownSelect,
        s.activeParsedEnvironments,
        s.environments,
        s.router,
        s.withServers
      ), this.decorations = h.set([
        h.widget({
          widget: this.widget,
          side: 1
        }).range(a)
      ]), this.view.dispatch({
        selection: { anchor: a }
      }), this.view.focus();
    }
    update(t) {
      var r, a;
      const e = t.state.selection.main.head, i = t.state.doc.sliceString(0, e), d = [], n = i.lastIndexOf("{{"), c = i.lastIndexOf("}}");
      if (n > c) {
        const l = i.slice(n + 2);
        !this.widget || this.widget.getQueryTerm() !== l ? ((r = this.widget) == null || r.destroy(), this.widget = new u(
          l,
          this.handleDropdownSelect,
          s.activeParsedEnvironments,
          s.environments,
          s.router
        )) : this.widget.updateQueryTerm(l), d.push(
          h.widget({
            widget: this.widget,
            side: 1
          }).range(e)
        );
      } else
        (a = this.widget) == null || a.destroy(), this.widget = null;
      this.decorations = h.set(d);
    }
  },
  {
    decorations: (t) => t.decorations
  }
);
export {
  C as dropdownPlugin
};
