// @scalar/api-client/dist/components/CodeInput/CodeInput.vue2.js
import { defineComponent as W, useAttrs as j, ref as d, computed as p, toRef as s, watch as H, createElementBlock as i, openBlock as a, Fragment as Q, createBlock as b, createCommentVNode as f, normalizeClass as G, unref as w, createElementVNode as V, toDisplayString as J, mergeProps as X, withKeys as y, withModifiers as q, createTextVNode as x, renderSlot as R } from "vue";
import { useCodeMirror as Y, useDropdown as Z, colorPicker as _ } from "@scalar/use-codemirror";
import { nanoid as ee } from "nanoid";
import te from "../../views/Environment/EnvironmentVariableDropdown.vue.js";
import B from "../DataTable/DataTableInputSelect.vue.js";
import { pillPlugin as oe, backspaceCommand as le } from "./codeVariableWidget.js";
import { useLayout as ne } from "../../hooks/useLayout.js";
const ae = { class: "whitespace-nowrap" }, re = ["id"], se = {
  key: 0,
  class: "z-context text-c-2 absolute right-1.5 bottom-1 hidden font-sans group-has-[:focus-visible]/input:block",
  role: "alert"
}, ue = {
  key: 5,
  class: "centered-y text-orange absolute right-7 text-xs"
}, ie = {
  key: 6,
  class: "centered-y absolute right-0 flex h-full items-center p-1.5 group-has-[.cm-focused]:z-1"
}, de = {
  key: 7,
  class: "required centered-y text-xxs text-c-3 group-[.error]:text-red bg-b-1 pointer-events-none absolute right-0 mr-0.5 pt-px pr-2 opacity-100 shadow-[-8px_0_4px_var(--scalar-background-1)] transition-opacity duration-150 group-[.alert]:bg-transparent group-[.alert]:shadow-none group-[.error]:bg-transparent group-[.error]:shadow-none peer-has-[.cm-focused]:opacity-0"
}, pe = {
  inheritAttrs: !1
}, he = /* @__PURE__ */ W({
  ...pe,
  __name: "CodeInput",
  props: {
    colorPicker: { type: Boolean, default: !1 },
    disabled: { type: Boolean, default: !1 },
    modelValue: {},
    error: { type: Boolean },
    emitOnBlur: { type: Boolean, default: !0 },
    extensions: { default: () => [] },
    lineNumbers: { type: Boolean },
    lint: { type: Boolean },
    disableTabIndent: { type: Boolean, default: !1 },
    language: {},
    handleFieldSubmit: {},
    handleFieldChange: {},
    placeholder: {},
    required: { type: Boolean },
    disableEnter: { type: Boolean, default: !1 },
    disableCloseBrackets: { type: Boolean, default: !1 },
    enum: {},
    examples: {},
    type: {},
    nullable: { type: Boolean, default: !1 },
    withVariables: { type: Boolean, default: !0 },
    importCurl: { type: Boolean },
    default: {},
    environment: {},
    envVariables: {},
    lineWrapping: { type: Boolean, default: !1 }
  },
  emits: ["submit", "update:modelValue", "curl", "blur"],
  setup(l, { expose: A, emit: I }) {
    const e = l, u = I, k = j(), M = k.id || `id-${ee()}`, C = d(!1), g = d(!1), S = d(""), D = d({ left: 0, top: 0 }), m = d(null), { layout: v } = ne();
    function P(t) {
      return t === e.modelValue ? null : e.importCurl && t.trim().toLowerCase().startsWith("curl") ? (u("curl", t), r.value?.dispatch({
        changes: {
          from: 0,
          to: r.value.state.doc.length,
          insert: String(e.modelValue)
        }
      }), null) : e.handleFieldChange ? e.handleFieldChange(t) : u("update:modelValue", t);
    }
    function h(t) {
      return e.handleFieldSubmit ? e.handleFieldSubmit(t) : u("submit", t);
    }
    function E(t) {
      C.value = !1, e.emitOnBlur && e.modelValue && h(t), u("blur", t);
    }
    const F = [...e.extensions];
    e.colorPicker && F.push(_);
    const N = p(
      () => oe({
        environment: e.environment,
        envVariables: e.envVariables,
        isReadOnly: v === "modal"
      })
    ), O = p(() => [
      ...F,
      N.value,
      le
    ]), T = d(null), { codeMirror: r } = Y({
      content: s(
        () => e.modelValue !== void 0 ? String(e.modelValue) : ""
      ),
      onChange: (t) => {
        P(t), z();
      },
      onFocus: () => C.value = !0,
      onBlur: (t) => E(t),
      codeMirrorRef: T,
      disableTabIndent: s(() => e.disableTabIndent),
      disableEnter: s(() => e.disableEnter),
      disableCloseBrackets: s(() => e.disableCloseBrackets),
      lineNumbers: s(() => e.lineNumbers),
      language: s(() => e.language),
      lint: s(() => e.lint),
      extensions: O,
      placeholder: s(() => e.placeholder)
    });
    r.value?.focus(), H(r, () => {
      r.value && Object.hasOwn(k, "autofocus") && r.value.focus();
    });
    const { handleDropdownSelect: K, updateDropdownVisibility: z } = Z({
      codeMirror: r,
      query: S,
      showDropdown: g,
      dropdownPosition: D
    }), $ = p(
      () => e.nullable ? ["true", "false", "null"] : ["true", "false"]
    ), c = (t, o) => {
      g.value ? t === "down" ? (o.preventDefault(), m.value?.handleArrowKey("down")) : t === "up" ? (o.preventDefault(), m.value?.handleArrowKey("up")) : t === "enter" && (o.preventDefault(), m.value?.handleSelect()) : t === "escape" ? e.disableTabIndent || o.stopPropagation() : t === "enter" && o.target instanceof HTMLDivElement && h(o.target.textContent ?? "");
    }, L = p(() => Array.isArray(e.type) ? (
      // Find the first type, that's not 'null'
      e.type.find((t) => t !== "null") ?? "string"
    ) : (
      // If it's not an array, just return the type
      e.type
    )), U = p(
      () => g.value && e.withVariables && v !== "modal" && e.environment
    );
    return A({
      /** Expose focus method */
      focus: () => {
        r.value?.focus();
      },
      // Expose these methods for testing
      handleChange: P,
      handleSubmit: h,
      handleBlur: E,
      booleanOptions: $,
      codeMirror: r,
      modelValue: e.modelValue
    }), (t, o) => (a(), i(Q, null, [
      l.disabled ? (a(), i("div", {
        key: 0,
        class: G(["text-c-2 flex cursor-default items-center justify-center", w(v) === "modal" ? "font-code pr-2 pl-1 text-base" : "px-2"]),
        "data-testid": "code-input-disabled"
      }, [
        V("span", ae, J(l.modelValue), 1)
      ], 2)) : e.enum && e.enum.length ? (a(), b(B, {
        key: 1,
        default: e.default,
        modelValue: l.modelValue,
        type: L.value,
        value: e.enum,
        "onUpdate:modelValue": o[0] || (o[0] = (n) => u("update:modelValue", n))
      }, null, 8, ["default", "modelValue", "type", "value"])) : l.type === "boolean" || l.type?.includes("boolean") ? (a(), b(B, {
        key: 2,
        default: e.default,
        modelValue: l.modelValue,
        value: $.value,
        "onUpdate:modelValue": o[1] || (o[1] = (n) => u("update:modelValue", n))
      }, null, 8, ["default", "modelValue", "value"])) : e.examples && e.examples.length ? (a(), b(B, {
        key: 3,
        default: e.default,
        modelValue: e.modelValue,
        value: e.examples,
        "onUpdate:modelValue": o[2] || (o[2] = (n) => u("update:modelValue", n))
      }, null, 8, ["default", "modelValue", "value"])) : (a(), i("div", X({
        key: 4,
        id: w(M)
      }, t.$attrs, {
        ref_key: "codeMirrorRef",
        ref: T,
        class: ["group/input group-[.alert]:outline-orange group-[.error]:outline-red font-code peer relative w-full overflow-hidden text-xs leading-[1.44] whitespace-nowrap -outline-offset-1 has-[:focus-visible]:rounded-[4px] has-[:focus-visible]:outline", {
          "line-wrapping has-[:focus-visible]:bg-b-1 has-[:focus-visible]:absolute has-[:focus-visible]:z-1": l.lineWrapping,
          "flow-code-input--error": l.error
        }],
        onKeydown: [
          o[3] || (o[3] = y(q((n) => c("down", n), ["stop"]), ["down"])),
          o[4] || (o[4] = y((n) => c("enter", n), ["enter"])),
          o[5] || (o[5] = y((n) => c("escape", n), ["escape"])),
          o[6] || (o[6] = y(q((n) => c("up", n), ["stop"]), ["up"]))
        ]
      }), [
        l.disableTabIndent ? f("", !0) : (a(), i("div", se, [...o[7] || (o[7] = [
          x(" Press ", -1),
          V("kbd", { class: "-mx-0.25 rounded border px-0.5 font-mono" }, "Esc", -1),
          x(" then ", -1),
          V("kbd", { class: "-mx-0.25 rounded border px-0.5 font-mono" }, "Tab", -1),
          x(" to exit ", -1)
        ])]))
      ], 16, re)),
      t.$slots.warning ? (a(), i("div", ue, [
        R(t.$slots, "warning", {}, void 0, !0)
      ])) : f("", !0),
      t.$slots.icon ? (a(), i("div", ie, [
        R(t.$slots, "icon", {}, void 0, !0)
      ])) : f("", !0),
      l.required ? (a(), i("div", de, " Required ")) : f("", !0),
      U.value ? (a(), b(te, {
        key: 8,
        ref_key: "dropdownRef",
        ref: m,
        dropdownPosition: D.value,
        envVariables: l.envVariables,
        environment: l.environment,
        query: S.value,
        onSelect: w(K)
      }, null, 8, ["dropdownPosition", "envVariables", "environment", "query", "onSelect"])) : f("", !0)
    ], 64));
  }
});
export {
  he as default
};
