// @scalar/components/dist/components/ScalarSearchInput/ScalarSearchInput.vue.js
import { defineComponent as d, computed as i, useAttrs as m, openBlock as o, createElementBlock as p, mergeProps as f, unref as a, createElementVNode as r, renderSlot as g, createTextVNode as h, createBlock as s, createCommentVNode as b } from "vue";
import { cx as V } from "../../cva.js";
import k from "../ScalarLoading/ScalarLoading.vue.js";
import S from "../ScalarIconButton/ScalarIconButton.vue.js";
const v = { class: "sr-only" }, N = ["value"], I = /* @__PURE__ */ d({
  inheritAttrs: !1,
  __name: "ScalarSearchInput",
  props: {
    loading: {},
    modelValue: {}
  },
  emits: ["update:modelValue"],
  setup(_, { emit: c }) {
    const l = c;
    function u(e) {
      l("update:modelValue", e.target.value);
    }
    const n = i(() => {
      const { class: e, ...t } = m();
      return { className: e || "", rest: t };
    });
    return (e, t) => (o(), p("label", f(n.value.rest, {
      class: a(V)(
        "flex rounded border bg-b-1 text-sm font-medium focus-within:border-c-1",
        n.value.className
      )
    }), [
      r("span", v, [
        g(e.$slots, "label", {}, () => [
          h("Enter search")
        ])
      ]),
      r("input", {
        autocapitalize: "off",
        autocomplete: "off",
        autocorrect: "off",
        class: "flex-1 rounded border-none bg-transparent p-3 outline-none",
        placeholder: "Search...",
        spellcheck: "false",
        type: "text",
        value: e.modelValue,
        onInput: u
      }, null, 40, N),
      e.loading && e.loading.isLoading ? (o(), s(a(k), {
        key: 0,
        class: "mr-3 self-center",
        loadingState: e.loading,
        size: "20px"
      }, null, 8, ["loadingState"])) : e.modelValue ? (o(), s(a(S), {
        key: 1,
        class: "p-2.5",
        icon: "Close",
        label: "Clear Search",
        thickness: "1.5",
        onClick: t[0] || (t[0] = (C) => l("update:modelValue", ""))
      })) : b("", !0)
    ], 16));
  }
});
export {
  I as default
};
