// @scalar/api-reference/dist/components/Content/Operation/ContentTypeSelect.vue2.js
import { defineComponent as v, computed as g, ref as C, openBlock as o, createElementBlock as l, normalizeClass as f, createElementVNode as _, toDisplayString as d, Fragment as B, renderList as h, createCommentVNode as T } from "vue";
const q = ["value"], V = ["value"], k = /* @__PURE__ */ v({
  __name: "ContentTypeSelect",
  props: {
    requestBody: {},
    defaultValue: {}
  },
  emits: ["selectContentType"],
  setup(p, { emit: i }) {
    const e = p, m = i, y = (t) => {
      const n = t.target.value;
      a.value = n, m("selectContentType", { contentType: n });
    }, s = g(() => {
      var t;
      return (t = e.requestBody) != null && t.content ? Object.keys(e.requestBody.content) : [];
    }), a = C(
      e.defaultValue || s.value[0]
    );
    return (t, c) => {
      var n;
      return o(), l("div", {
        class: f(["content-type-select", { "content-type-no-select": s.value.length <= 1 }])
      }, [
        _("span", null, d(a.value), 1),
        e != null && e.requestBody && s.value.length > 1 ? (o(), l("select", {
          key: 0,
          value: a.value,
          onChange: c[0] || (c[0] = (r) => y(r))
        }, [
          (o(!0), l(B, null, h((n = e.requestBody) == null ? void 0 : n.content, (r, u) => (o(), l("option", {
            key: u,
            value: u
          }, d(u), 9, V))), 128))
        ], 40, q)) : T("", !0)
      ], 2);
    };
  }
});
export {
  k as default
};
