// @scalar/api-reference/dist/components/Content/Operation/EndpointDetailsCard.vue2.js
import { defineComponent as c, openBlock as l, createElementBlock as u, createVNode as e, unref as a, withCtx as t, createTextVNode as r } from "vue";
import s from "./Parameters.vue.js";
import _ from "./RequestBody.vue.js";
import { useOperation as f } from "../../../hooks/useOperation.js";
import { useResponses as y } from "../../../hooks/useResponses.js";
const B = { class: "endpoint-details-card" }, k = /* @__PURE__ */ c({
  __name: "EndpointDetailsCard",
  props: {
    operation: {}
  },
  setup(p) {
    const i = p, { parameterMap: o } = f(i), { responses: d } = y(i.operation);
    return (m, h) => {
      var n;
      return l(), u("div", B, [
        e(s, {
          class: "endpoint-details-card-item",
          parameters: a(o).path
        }, {
          title: t(() => [
            r("Path Parameters")
          ]),
          _: 1
        }, 8, ["parameters"]),
        e(s, {
          class: "endpoint-details-card-item",
          parameters: a(o).query
        }, {
          title: t(() => [
            r("Query Parameters")
          ]),
          _: 1
        }, 8, ["parameters"]),
        e(s, {
          class: "endpoint-details-card-item",
          parameters: a(o).header
        }, {
          title: t(() => [
            r("Headers")
          ]),
          _: 1
        }, 8, ["parameters"]),
        e(_, {
          class: "endpoint-details-card-item",
          requestBody: (n = m.operation.information) == null ? void 0 : n.requestBody
        }, {
          title: t(() => [
            r("Body")
          ]),
          _: 1
        }, 8, ["requestBody"]),
        e(s, {
          class: "endpoint-details-card-item",
          parameters: a(d)
        }, {
          title: t(() => [
            r("Responses")
          ]),
          _: 1
        }, 8, ["parameters"])
      ]);
    };
  }
});
export {
  k as default
};
