// @scalar/api-reference/dist/components/Content/Operation/Operation.vue2.js
import { defineComponent as S, computed as _, openBlock as i, createBlock as p, unref as o, withCtx as t, createVNode as r, createTextVNode as s, createCommentVNode as d, createElementVNode as u, normalizeClass as k, toDisplayString as y } from "vue";
import E from "./EndpointDetails.vue.js";
import g from "./EndpointPath.vue.js";
import v from "./TestRequestButton.vue.js";
import B from "../../Section/Section.vue.js";
import N from "../../Section/SectionContent.vue.js";
import R from "../../Section/SectionColumns.vue.js";
import f from "../../Section/SectionColumn.vue.js";
import V from "../../Badge/Badge.vue.js";
import b from "../../Section/SectionHeader.vue.js";
import q from "../../Anchor/Anchor.vue.js";
import x from "../../../features/ExampleRequest/ExampleRequest.vue.js";
import T from "./PathResponses/PathResponses.vue.js";
import { useHttpClientStore as D } from "../../../stores/useHttpClientStore.js";
const H = { class: "examples" }, Q = /* @__PURE__ */ S({
  __name: "Operation",
  props: {
    id: {},
    operation: {}
  },
  setup(c) {
    const m = c, { availableTargets: h } = D(), C = _(() => {
      var n;
      const e = ["x-custom-examples", "x-codeSamples", "x-code-samples"];
      for (const a of e)
        if ((n = m.operation.information) != null && n[a])
          return m.operation.information[a];
      return null;
    });
    return (e, n) => (i(), p(o(B), {
      id: e.id,
      label: e.operation.name
    }, {
      default: t(() => [
        r(o(N), null, {
          default: t(() => [
            r(o(R), null, {
              default: t(() => [
                r(o(f), null, {
                  default: t(() => {
                    var a, l;
                    return [
                      (a = e.operation.information) != null && a.deprecated ? (i(), p(o(V), { key: 0 }, {
                        default: t(() => [
                          s(" Deprecated ")
                        ]),
                        _: 1
                      })) : d("", !0),
                      u("div", {
                        class: k((l = e.operation.information) != null && l.deprecated ? "deprecated" : "")
                      }, [
                        r(o(b), { level: 3 }, {
                          default: t(() => [
                            r(o(q), {
                              id: e.id ?? ""
                            }, {
                              default: t(() => [
                                s(y(e.operation.name), 1)
                              ]),
                              _: 1
                            }, 8, ["id"])
                          ]),
                          _: 1
                        })
                      ], 2),
                      r(E, { operation: e.operation }, null, 8, ["operation"])
                    ];
                  }),
                  _: 1
                }),
                r(o(f), null, {
                  default: t(() => [
                    u("div", H, [
                      o(h).length ? (i(), p(o(x), {
                        key: 0,
                        customExamples: C.value,
                        operation: e.operation
                      }, {
                        header: t(() => {
                          var a;
                          return [
                            r(g, {
                              class: "example-path",
                              deprecated: (a = e.operation.information) == null ? void 0 : a.deprecated,
                              path: e.operation.path
                            }, null, 8, ["deprecated", "path"])
                          ];
                        }),
                        footer: t(() => [
                          r(v, { operation: e.operation }, null, 8, ["operation"])
                        ]),
                        _: 1
                      }, 8, ["customExamples", "operation"])) : d("", !0),
                      r(o(T), {
                        operation: e.operation,
                        style: { "margin-top": "12px" }
                      }, null, 8, ["operation"])
                    ])
                  ]),
                  _: 1
                })
              ]),
              _: 1
            })
          ]),
          _: 1
        })
      ]),
      _: 1
    }, 8, ["id", "label"]));
  }
});
export {
  Q as default
};
