// @scalar/api-reference/dist/components/Content/Operation/OperationAccordion.vue2.js
import { defineComponent as l, openBlock as r, createBlock as p, unref as e, createSlots as c, withCtx as i, createElementVNode as n, createVNode as t, toDisplayString as m, withModifiers as h } from "vue";
import { ScalarIcon as f, ScalarIconButton as u, ScalarMarkdown as v } from "@scalar/components";
import y from "./EndpointDetailsCard.vue.js";
import _ from "./EndpointPath.vue.js";
import b from "./TestRequestButton.vue.js";
import k from "../../Section/SectionAccordion.vue.js";
import C from "../../HttpMethod/HttpMethod.vue.js";
import S from "../../Anchor/Anchor.vue.js";
import w from "./PathResponses/PathResponses.vue.js";
import B from "../../../features/ExampleRequest/ExampleRequest.vue.js";
import { useClipboard as E } from "../../../hooks/useClipboard.js";
const R = { class: "endpoint-header" }, g = { class: "endpoint-details" }, A = { class: "endpoint-label" }, I = { class: "endpoint-label-path" }, P = { class: "endpoint-label-name" }, V = { class: "endpoint-content" }, H = /* @__PURE__ */ l({
  __name: "OperationAccordion",
  props: {
    id: {},
    operation: {}
  },
  setup(q) {
    const { copyToClipboard: d } = E();
    return (o, s) => (r(), p(e(k), {
      id: o.id,
      class: "reference-endpoint",
      transparent: ""
    }, c({
      title: i(() => [
        n("h3", R, [
          n("div", g, [
            t(e(C), {
              class: "endpoint-type",
              method: o.operation.httpVerb,
              short: ""
            }, null, 8, ["method"]),
            t(e(S), {
              id: o.id ?? "",
              class: "endpoint-anchor"
            }, {
              default: i(() => {
                var a;
                return [
                  n("div", A, [
                    n("div", I, [
                      t(_, {
                        deprecated: (a = o.operation.information) == null ? void 0 : a.deprecated,
                        path: o.operation.path
                      }, null, 8, ["deprecated", "path"])
                    ]),
                    n("div", P, m(o.operation.name), 1)
                  ])
                ];
              }),
              _: 1
            }, 8, ["id"])
          ])
        ])
      ]),
      actions: i(({ active: a }) => [
        a ? (r(), p(b, {
          key: 0,
          operation: o.operation
        }, null, 8, ["operation"])) : (r(), p(e(f), {
          key: 1,
          class: "endpoint-try-hint",
          icon: "Play",
          thickness: "1.75px"
        })),
        t(e(u), {
          class: "endpoint-copy",
          icon: "Clipboard",
          label: "Copy endpoint URL",
          size: "xs",
          variant: "ghost",
          onClick: s[0] || (s[0] = h((D) => e(d)(o.operation.path), ["stop"]))
        })
      ]),
      default: i(() => [
        n("div", V, [
          t(y, { operation: o.operation }, null, 8, ["operation"]),
          t(e(w), { operation: o.operation }, null, 8, ["operation"]),
          t(e(B), { operation: o.operation }, null, 8, ["operation"])
        ])
      ]),
      _: 2
    }, [
      o.operation.description ? {
        name: "description",
        fn: i(() => [
          t(e(v), {
            value: o.operation.description,
            withImages: ""
          }, null, 8, ["value"])
        ]),
        key: "0"
      } : void 0
    ]), 1032, ["id"]));
  }
});
export {
  H as default
};
