// @scalar/api-reference/dist/components/Content/Operation/Parameters.vue2.js
import { defineComponent as o, openBlock as t, createElementBlock as l, createElementVNode as s, renderSlot as n, Fragment as m, renderList as p, createBlock as d, createCommentVNode as c } from "vue";
import i from "./ParameterItem.vue.js";
const h = {
  key: 0,
  class: "parameters"
}, u = { class: "parameters-title" }, f = { class: "parameter-list" }, I = /* @__PURE__ */ o({
  __name: "Parameters",
  props: {
    parameters: {},
    showChildren: { type: Boolean, default: !1 },
    collapsableItems: { type: Boolean, default: !1 }
  },
  setup(_) {
    return (e, C) => {
      var r;
      return (r = e.parameters) != null && r.length ? (t(), l("div", h, [
        s("div", u, [
          n(e.$slots, "title", {}, void 0, !0)
        ]),
        s("ul", f, [
          (t(!0), l(m, null, p(e.parameters, (a) => (t(), d(i, {
            key: a.name,
            collapsableItems: e.collapsableItems,
            parameter: a,
            showChildren: e.showChildren
          }, null, 8, ["collapsableItems", "parameter", "showChildren"]))), 128))
        ])
      ])) : c("", !0);
    };
  }
});
export {
  I as default
};
