// @scalar/api-reference/dist/components/Content/Operation/ParameterItem.vue2.js
import { defineComponent as B, ref as u, computed as d, openBlock as l, createElementBlock as s, normalizeClass as g, createVNode as v, unref as f, createElementVNode as y, toDisplayString as h, createBlock as b, createCommentVNode as m } from "vue";
import { ScalarIcon as q } from "@scalar/components";
import w from "./ContentTypeSelect.vue.js";
import I from "../Schema/SchemaProperty.vue.js";
const S = { class: "parameter-item" }, V = { class: "parameter-item-name" }, T = { class: "parameter-item-type" }, j = {
  key: 1,
  class: "parameter-item-container parameter-item-container-markdown"
}, P = /* @__PURE__ */ B({
  __name: "ParameterItem",
  props: {
    parameter: {},
    showChildren: { type: Boolean, default: !1 },
    collapsableItems: { type: Boolean, default: !1 }
  },
  setup(C) {
    const t = C, r = u(!1), k = d(() => t.parameter.content ? Object.keys(t.parameter.content) : []), n = u(
      k.value[0]
    );
    t.parameter.content && "application/json" in t.parameter.content && (n.value = "application/json");
    const a = d(() => t.collapsableItems && t.parameter.content);
    return (e, o) => {
      var p, i;
      return l(), s("li", S, [
        a.value ? (l(), s("div", {
          key: 0,
          class: g(["flex parameter-item-trigger", { "parameter-item-trigger-open": r.value }]),
          onClick: o[1] || (o[1] = (c) => r.value = !r.value)
        }, [
          v(f(q), {
            class: "parameter-item-icon",
            icon: r.value ? "ChevronDown" : "ChevronRight",
            size: "md",
            thickness: "1.75"
          }, null, 8, ["icon"]),
          y("span", V, h(e.parameter.name), 1),
          y("span", T, h(e.parameter.description), 1),
          a.value && t.parameter.content ? (l(), b(w, {
            key: 0,
            class: "parameter-item-content-type",
            defaultValue: n.value,
            requestBody: t.parameter,
            onSelectContentType: o[0] || (o[0] = ({ contentType: c }) => n.value = c)
          }, null, 8, ["defaultValue", "requestBody"])) : m("", !0)
        ], 2)) : m("", !0),
        a.value && r.value || !a.value ? (l(), s("div", j, [
          v(f(I), {
            compact: "",
            description: a.value ? "" : e.parameter.description,
            level: 0,
            name: a.value ? "" : e.parameter.name,
            noncollapsible: e.showChildren,
            required: e.parameter.required,
            value: e.parameter.content ? (i = (p = e.parameter.content) == null ? void 0 : p[n.value]) == null ? void 0 : i.schema : e.parameter.schema
          }, null, 8, ["description", "name", "noncollapsible", "required", "value"])
        ])) : m("", !0)
      ]);
    };
  }
});
export {
  P as default
};
