// @scalar/api-reference/dist/components/Content/Operation/RequestBody.vue2.js
import { defineComponent as b, computed as k, ref as g, openBlock as o, createElementBlock as n, createElementVNode as a, renderSlot as p, normalizeClass as S, toDisplayString as B, withDirectives as C, Fragment as V, renderList as N, vModelSelect as j, createCommentVNode as u, createVNode as _, unref as f } from "vue";
import { ScalarMarkdown as w } from "@scalar/components";
import D from "../Schema/Schema.vue.js";
const E = { key: 0 }, M = { class: "request-body-title" }, T = ["value"], x = {
  key: 0,
  class: "request-body-description"
}, z = {
  key: 0,
  class: "request-body-schema"
}, R = /* @__PURE__ */ b({
  __name: "RequestBody",
  props: {
    requestBody: {}
  },
  setup(h) {
    var c;
    const e = h, s = k(() => {
      var l;
      return (l = e.requestBody) != null && l.content ? Object.keys(e.requestBody.content) : [];
    }), t = g("application/json");
    return (c = e.requestBody) != null && c.content && s.value.length > 0 && (t.value = s.value[0]), (l, d) => {
      var i, v, y, m;
      return e != null && e.requestBody ? (o(), n("div", E, [
        a("div", M, [
          p(l.$slots, "title", {}, void 0, !0),
          a("div", {
            class: S(["request-body-title-select", { "request-body-title-no-select": s.value.length <= 1 }])
          }, [
            a("span", null, B(t.value), 1),
            e != null && e.requestBody && s.value.length > 1 ? C((o(), n("select", {
              key: 0,
              "onUpdate:modelValue": d[0] || (d[0] = (q) => t.value = q)
            }, [
              (o(!0), n(V, null, N((i = e.requestBody) == null ? void 0 : i.content, (q, r) => (o(), n("option", {
                key: r,
                value: r
              }, B(r), 9, T))), 128))
            ], 512)), [
              [j, t.value]
            ]) : u("", !0)
          ], 2),
          e != null && e.requestBody.description ? (o(), n("div", x, [
            _(f(w), {
              value: e.requestBody.description
            }, null, 8, ["value"])
          ])) : u("", !0)
        ]),
        (v = e == null ? void 0 : e.requestBody.content) != null && v[t.value] ? (o(), n("div", z, [
          _(f(D), {
            compact: "",
            noncollapsible: "",
            value: (m = (y = e == null ? void 0 : e.requestBody.content) == null ? void 0 : y[t.value]) == null ? void 0 : m.schema
          }, null, 8, ["value"])
        ])) : u("", !0)
      ])) : u("", !0);
    };
  }
});
export {
  R as default
};
