// @scalar/api-reference/dist/components/Content/Operation/TestRequestButton.vue2.js
import { defineComponent as n, openBlock as r, createElementBlock as s, withModifiers as i, unref as p, createVNode as a, pushScopeId as c, popScopeId as m, createElementVNode as d } from "vue";
import { ScalarIcon as u } from "@scalar/components";
import { apiClientBus as l } from "../../api-client-bus.js";
const h = (e) => (c("data-v-1dbecca0"), e = e(), m(), e), f = ["method"], _ = /* @__PURE__ */ h(() => /* @__PURE__ */ d("span", null, "Test Request", -1)), V = /* @__PURE__ */ n({
  __name: "TestRequestButton",
  props: {
    operation: {}
  },
  setup(e) {
    return (t, o) => (r(), s("button", {
      class: "show-api-client-button",
      method: t.operation.httpVerb,
      type: "button",
      onClick: o[0] || (o[0] = i((b) => p(l).emit({
        open: {
          path: t.operation.path,
          method: t.operation.httpVerb
        }
      }), ["stop"]))
    }, [
      a(p(u), {
        icon: "Play",
        size: "sm"
      }),
      _
    ], 8, f));
  }
});
export {
  V as default
};
