// @scalar/api-reference/dist/components/ApiClientModal.vue.js
import { defineComponent as S, ref as l, onMounted as U, watch as _, onBeforeUnmount as h, openBlock as x, createElementBlock as y } from "vue";
import { modalStateBus as B, apiClientBus as A } from "./api-client-bus.js";
import { getUrlFromServerState as C } from "../legacy/helpers/getUrlFromServerState.js";
import { useAuthenticationStore as k } from "../legacy/stores/useAuthenticationStore.js";
import { useServerStore as M } from "../legacy/stores/useServerStore.js";
const b = /* @__PURE__ */ S({
  __name: "ApiClientModal",
  props: {
    proxyUrl: {},
    spec: {},
    servers: {}
  },
  setup(u) {
    const t = u, o = l(null), p = l(null), { authentication: c } = k(), { server: i } = M();
    return U(async () => {
      if (!o.value) return;
      const { createApiClientModal: r } = await import("@scalar/api-client"), { app: n, open: m, updateAuth: f, updateServerUrl: d, modalState: v, updateSpec: a } = await r(o.value, {
        spec: t.spec ?? {},
        proxyUrl: t.proxyUrl,
        servers: t.servers
      });
      B.emit(v), A.on((e) => {
        if (e.open) {
          f(c);
          const s = C(i);
          s && d(s), m(e.open);
        }
        e.updateSpec && a(e.updateSpec);
      }), _(
        () => t.spec,
        (e) => e && a(e),
        { deep: !0 }
      ), p.value = n;
    }), h(() => {
      var r;
      return (r = p.value) == null ? void 0 : r.unmount();
    }), (r, n) => (x(), y("div", {
      ref_key: "el",
      ref: o
    }, null, 512));
  }
});
export {
  b as default
};
