// @scalar/api-reference/dist/components/ApiReferenceLayout.vue2.js
import { defineComponent as Z, ref as h, onMounted as B, onBeforeMount as D, computed as b, watch as _, onServerPrefetch as x, useSSRContext as ee, getCurrentInstance as oe, provide as F, openBlock as v, createElementBlock as S, Fragment as M, createElementVNode as i, normalizeClass as te, unref as u, normalizeStyle as re, renderSlot as l, normalizeProps as d, guardReactiveProps as c, createVNode as f, withCtx as g, createCommentVNode as w, withDirectives as ae, vShow as ne, createSlots as se } from "vue";
import { provideUseId as ie } from "@headlessui/vue";
import { addScalarClassesToHeadless as le } from "@scalar/components";
import { defaultStateFactory as P } from "@scalar/oas-utils/helpers";
import { getThemeStyles as de } from "@scalar/themes";
import { useToasts as ce, ScalarToasts as ue } from "@scalar/use-toasts";
import { useMediaQuery as pe, useResizeObserver as fe, useDebounceFn as ge } from "@vueuse/core";
import me from "./ApiClientModal.vue.js";
/* empty css                    */
import ve from "./GettingStarted.vue.js";
import { downloadSpecBus as Se, downloadSpecFile as he } from "../helpers/specDownloads.js";
import { useDeprecationWarnings as be } from "../hooks/useDeprecationWarnings.js";
import we from "./Sidebar/Sidebar.vue.js";
import ye from "./Content/Content.vue.js";
import { useSidebar as Te } from "../hooks/useSidebar.js";
import { GLOBAL_SECURITY_SYMBOL as Re, HIDE_DOWNLOAD_BUTTON_SYMBOL as $e } from "../helpers/provideSymbols.js";
import { useNavState as Ce } from "../hooks/useNavState.js";
import { sleep as Le } from "../helpers/sleep.js";
const Ee = ["innerHTML"], Ie = { class: "references-header" }, Oe = {
  key: 0,
  class: "references-navigation t-doc__sidebar"
}, ke = { class: "references-navigation-list" }, Be = { class: "references-editor" }, De = { class: "references-editor-textarea" }, Fe = { class: "references-rendered" }, Me = {
  key: 0,
  class: "references-footer"
}, xe = /* @__PURE__ */ Z({
  inheritAttrs: !1,
  __name: "ApiReferenceLayout",
  props: {
    configuration: {},
    parsedSpec: {},
    rawSpec: {}
  },
  emits: ["changeTheme", "updateContent", "loadSwaggerFile", "linkSwaggerFile", "toggleDarkMode"],
  setup(U) {
    const t = U, { initializeToasts: A, toast: H } = ce();
    A((e) => H(e));
    const N = pe("(min-width: 1150px)"), z = h("100dvh"), p = h(null);
    fe(p, (e) => {
      z.value = e[0].contentRect.height + "px";
    });
    const y = h(0);
    B(() => {
      var a, r, n;
      const e = (r = (a = p.value) == null ? void 0 : a.parentElement) == null ? void 0 : r.getBoundingClientRect(), o = (n = p.value) == null ? void 0 : n.getBoundingClientRect();
      if (e && o) {
        const k = o.top - e.top;
        y.value = k < 2 ? 0 : k;
      }
    });
    const {
      breadcrumb: Y,
      collapsedSidebarItems: m,
      isSidebarOpen: V,
      setCollapsedSidebarItem: T,
      hideModels: G,
      defaultOpenAllTags: W,
      setParsedSpec: j,
      scrollToOperation: K
    } = Te(), {
      getPathRoutingId: R,
      getSectionId: Q,
      getTagId: q,
      hash: $,
      isIntersectionEnabled: C,
      pathRouting: L,
      updateHash: E
    } = Ce();
    L.value = t.configuration.pathRouting, D(() => E());
    const I = async (e) => {
      var o;
      C.value = !1, E(), e ? K(e) : (o = p.value) == null || o.scrollTo(0, 0), await Le(100), C.value = !0;
    };
    D(() => le()), B(() => {
      Se.on(({ specTitle: e }) => {
        he(t.rawSpec, e);
      }), window.onhashchange = () => I(decodeURIComponent(window.location.hash.replace(/^#/, ""))), window.onpopstate = () => L.value && I(R(window.location.pathname));
    });
    const J = b(
      () => N.value || !t.configuration.isEditable
    ), O = ge((e) => {
      if ((e.target.scrollTop ?? 0) < 50) {
        const a = t.configuration.pathRouting ? t.configuration.pathRouting.basePath : window.location.pathname;
        window.history.replaceState({}, "", a + window.location.search), $.value = "";
      }
    }), s = b(() => ({
      breadcrumb: Y.value,
      spec: t.parsedSpec
    }));
    _(() => t.parsedSpec, j, { deep: !0 }), x(() => {
      var o, a;
      const e = ee();
      if (e && (e.payload || (e.payload = { data: P() }), (o = e.payload).data || (o.data = P()), t.configuration.pathRouting)) {
        const r = R(e.url);
        $.value = r, e.payload.data.hash = r;
        for (const n in m)
          Object.hasOwn(m, n) && delete m[n];
        if (r)
          T(Q(r), !0);
        else {
          const n = (a = t.parsedSpec.tags) == null ? void 0 : a[0];
          n && T(q(n), !0);
        }
        e.payload.data["useSidebarContent-collapsedSidebarItems"] = m;
      }
    }), ie(() => {
      var r;
      const e = oe(), o = "scalar-instance-id";
      if (!e) return o;
      let a = e.uid;
      return typeof window > "u" ? (e.attrs || (e.attrs = {}), e.attrs[o] = a) : (r = e.vnode.el) != null && r.getAttribute && (a = e.vnode.el.getAttribute(o)), `${o}-${a}`;
    }), F(Re, () => t.parsedSpec.security), F(
      $e,
      () => t.configuration.hideDownloadButton
    ), G.value = t.configuration.hideModels ?? !1, W.value = t.configuration.defaultOpenAllTags ?? !1, be(t.configuration);
    const X = b(
      () => `<style>
  ${de(t.configuration.theme, {
        fonts: t.configuration.withDefaultFonts
      })}</style>`
    );
    return (e, o) => {
      var a;
      return v(), S(M, null, [
        i("div", { innerHTML: X.value }, null, 8, Ee),
        i("div", {
          ref_key: "documentEl",
          ref: p,
          class: te(["scalar-app scalar-api-reference references-layout", [
            {
              "references-editable": e.configuration.isEditable,
              "references-sidebar": e.configuration.showSidebar,
              "references-sidebar-mobile-open": u(V),
              "references-classic": e.configuration.layout === "classic"
            },
            e.$attrs.class
          ]]),
          style: re({
            "--scalar-y-offset": `var(--scalar-custom-header-height, ${y.value}px)`
          }),
          onScrollPassive: o[4] || (o[4] = //@ts-ignore
          (...r) => u(O) && u(O)(...r))
        }, [
          i("div", Ie, [
            l(e.$slots, "header", d(c(s.value)), void 0, !0)
          ]),
          e.configuration.showSidebar ? (v(), S("aside", Oe, [
            i("div", ke, [
              f(u(we), {
                parsedSpec: e.parsedSpec,
                tagsSorter: e.configuration.tagsSorter
              }, {
                "sidebar-start": g(() => [
                  l(e.$slots, "sidebar-start", d(c(s.value)), void 0, !0)
                ]),
                "sidebar-end": g(() => [
                  l(e.$slots, "sidebar-end", d(c(s.value)), void 0, !0)
                ]),
                _: 3
              }, 8, ["parsedSpec", "tagsSorter"])
            ])
          ])) : w("", !0),
          ae(i("div", Be, [
            i("div", De, [
              l(e.$slots, "editor", d(c(s.value)), void 0, !0)
            ])
          ], 512), [
            [ne, e.configuration.isEditable]
          ]),
          J.value ? (v(), S(M, { key: 1 }, [
            i("div", Fe, [
              f(u(ye), {
                baseServerURL: e.configuration.baseServerURL,
                layout: e.configuration.layout === "classic" ? "accordion" : "default",
                parsedSpec: e.parsedSpec,
                proxy: e.configuration.proxy,
                servers: e.configuration.servers
              }, se({
                start: g(() => [
                  l(e.$slots, "content-start", d(c(s.value)), void 0, !0)
                ]),
                end: g(() => [
                  l(e.$slots, "content-end", d(c(s.value)), void 0, !0)
                ]),
                _: 2
              }, [
                (a = e.configuration) != null && a.isEditable ? {
                  name: "empty-state",
                  fn: g(() => {
                    var r;
                    return [
                      f(ve, {
                        theme: ((r = e.configuration) == null ? void 0 : r.theme) || "default",
                        onChangeTheme: o[0] || (o[0] = (n) => e.$emit("changeTheme", n)),
                        onLinkSwaggerFile: o[1] || (o[1] = (n) => e.$emit("linkSwaggerFile")),
                        onLoadSwaggerFile: o[2] || (o[2] = (n) => e.$emit("loadSwaggerFile")),
                        onUpdateContent: o[3] || (o[3] = (n) => e.$emit("updateContent", n))
                      }, null, 8, ["theme"])
                    ];
                  }),
                  key: "0"
                } : void 0
              ]), 1032, ["baseServerURL", "layout", "parsedSpec", "proxy", "servers"])
            ]),
            e.$slots.footer ? (v(), S("div", Me, [
              l(e.$slots, "footer", d(c(s.value)), void 0, !0)
            ])) : w("", !0)
          ], 64)) : w("", !0),
          f(me, {
            proxyUrl: e.configuration.proxy,
            servers: e.configuration.servers,
            spec: e.configuration.spec
          }, null, 8, ["proxyUrl", "servers", "spec"])
        ], 38),
        f(u(ue))
      ], 64);
    };
  }
});
export {
  xe as default
};
