// @scalar/api-reference/dist/components/ApiReference.vue.js
import { defineComponent as T, watch as u, computed as g, toRef as C, openBlock as v, createElementBlock as $, Fragment as b, createBlock as A, resolveDynamicComponent as B, withCtx as c, createTextVNode as E, toDisplayString as H, createCommentVNode as N, createVNode as R, unref as a, renderSlot as S } from "vue";
import { migrateThemeVariables as V } from "@scalar/themes";
import { useSeoMeta as _ } from "@unhead/vue";
import { useDarkModeState as F } from "../hooks/useDarkModeState.js";
import { useReactiveSpec as U } from "../hooks/useReactiveSpec.js";
import j from "./Layouts/Layouts.vue.js";
import { useHttpClientStore as q } from "../stores/useHttpClientStore.js";
import { useAuthenticationStore as z } from "../legacy/stores/useAuthenticationStore.js";
const W = /* @__PURE__ */ T({
  __name: "ApiReference",
  props: {
    configuration: {}
  },
  emits: ["updateContent"],
  setup(k) {
    var m, l, d;
    const r = k, { toggleDarkMode: p, isDark: f } = F(
      (m = r.configuration) == null ? void 0 : m.darkMode,
      (l = r.configuration) == null ? void 0 : l.forceDarkModeState
    );
    u(
      () => {
        var e;
        return (e = r.configuration) == null ? void 0 : e.darkMode;
      },
      (e) => {
        e !== f.value && p();
      }
    );
    const i = g(() => {
      var e, t;
      if ((e = r.configuration) != null && e.customCss)
        return V((t = r.configuration) == null ? void 0 : t.customCss);
    });
    u(i, () => console.log(i.value));
    const o = g(() => {
      var e;
      return {
        spec: {
          content: void 0,
          url: void 0,
          ...(e = r.configuration) == null ? void 0 : e.spec
        },
        proxy: void 0,
        theme: "default",
        showSidebar: !0,
        isEditable: !1,
        ...r.configuration,
        customCss: i.value
      };
    });
    (d = o.value) != null && d.metaData && _(o.value.metaData);
    function s(e, t) {
      u(
        () => {
          var n;
          return (n = o.value) == null ? void 0 : n[e];
        },
        (n) => {
          typeof n < "u" && t(n);
        },
        { immediate: !0 }
      );
    }
    const { setAuthentication: D } = z();
    s("authentication", D);
    const { setExcludedClients: y, setDefaultHttpClient: M } = q();
    s("defaultHttpClient", M), s("hiddenClients", y);
    const { parsedSpec: h, rawSpec: w } = U({
      proxy: C(() => o.value.proxy || ""),
      specConfig: C(() => o.value.spec || {})
    });
    return (e, t) => {
      var n;
      return v(), $(b, null, [
        (n = o.value) != null && n.customCss ? (v(), A(B("style"), { key: 0 }, {
          default: c(() => [
            E(H(o.value.customCss), 1)
          ]),
          _: 1
        })) : N("", !0),
        R(a(j), {
          configuration: o.value,
          isDark: a(f),
          parsedSpec: a(h),
          rawSpec: a(w),
          onToggleDarkMode: t[0] || (t[0] = () => a(p)()),
          onUpdateContent: t[1] || (t[1] = (x) => e.$emit("updateContent", x))
        }, {
          footer: c(() => [
            S(e.$slots, "footer")
          ]),
          "content-end": c(() => [
            S(e.$slots, "footer")
          ]),
          _: 3
        }, 8, ["configuration", "isDark", "parsedSpec", "rawSpec"])
      ], 64);
    };
  }
});
export {
  W as default
};
