// @scalar/api-client/dist/v2/features/modal/Modal.vue2.js
import { defineComponent as W, ref as w, useId as D, watch as I, nextTick as N, onBeforeMount as E, onBeforeUnmount as F, computed as d, toValue as U, withDirectives as k, createElementBlock as u, openBlock as m, createElementVNode as s, unref as o, createVNode as l, withCtx as V, vShow as b } from "vue";
import { addScalarClassesToHeadless as A, ScalarTeleportRoot as $ } from "@scalar/components";
import { ScalarToasts as z } from "@scalar/use-toasts";
import { useFocusTrap as H } from "@vueuse/integrations/useFocusTrap";
import { mergeAuthConfig as L } from "../../blocks/scalar-auth-selector-block/helpers/merge-auth-config.js";
import { initializeModalEvents as M } from "./modal-events.js";
import O from "../operation/Operation.vue.js";
import { getActiveEnvironment as R } from "../../helpers/get-active-environment.js";
import { useGlobalHotKeys as j } from "../../hooks/use-global-hot-keys.js";
import { useScrollLock as q } from "../../hooks/use-scroll-lock.js";
import G from "../../components/sidebar/SidebarToggle.vue.js";
import K from "../../components/sidebar/Sidebar.vue.js";
const P = { class: "scalar scalar-app" }, J = { class: "scalar-container" }, Q = ["id"], X = {
  key: 0,
  class: "relative flex flex-1"
}, Y = {
  key: 1,
  class: "flex h-full w-full items-center justify-center"
}, Z = {}, me = /* @__PURE__ */ W({
  ...Z,
  __name: "Modal",
  props: {
    workspaceStore: {},
    document: {},
    path: {},
    eventBus: {},
    method: {},
    exampleName: {},
    modalState: {},
    sidebarState: {},
    plugins: {},
    options: {}
  },
  setup(e, { expose: x }) {
    typeof window < "u" && (window.dataDumpWorkspace = () => e.workspaceStore);
    const r = {
      label: "default",
      id: "default"
    }, n = w(!1);
    M({
      eventBus: e.eventBus,
      isSidebarOpen: n,
      sidebarState: e.sidebarState,
      modalState: e.modalState,
      store: e.workspaceStore
    }), j(e.eventBus, "modal");
    const v = w(null), f = D(), { activate: y, deactivate: B } = H(v, {
      allowOutsideClick: !0,
      fallbackFocus: `#${f}`
    }), S = () => {
      B(), e.eventBus.emit("operation:cancel:request"), e.workspaceStore.revertDocumentChanges(
        e.document.value?.["x-scalar-navigation"]?.name ?? ""
      );
    }, g = q(() => typeof window < "u" ? window.document.body : null);
    I(
      () => e.modalState.open,
      (a) => {
        g.value = a, a ? y({ checkCanFocusTrap: () => N() }) : S();
      }
    ), E(() => A()), F(() => S());
    const p = 288, c = d(
      () => e.workspaceStore?.workspace?.["x-scalar-sidebar-width"] ?? p
    ), T = (a) => e.workspaceStore?.update("x-scalar-sidebar-width", a), h = d(
      () => R(e.workspaceStore, e.document.value)
    ), C = d(
      () => L(
        e.document.value?.components?.securitySchemes,
        U(e.options)?.authentication?.securitySchemes
      )
    );
    return x({
      sidebarWidth: c,
      environment: h
    }), (a, t) => k((m(), u("div", P, [
      s("div", J, [
        s("div", {
          id: o(f),
          ref_key: "client",
          ref: v,
          "aria-label": "API Client",
          "aria-modal": "true",
          class: "scalar-app-layout scalar-client flex",
          role: "dialog",
          tabindex: "-1"
        }, [
          l(o($), null, {
            default: V(() => [
              l(o(z)),
              e.document.value && e.path?.value && e.method?.value ? (m(), u("main", X, [
                l(o(G), {
                  modelValue: n.value,
                  "onUpdate:modelValue": t[0] || (t[0] = (i) => n.value = i),
                  class: "absolute top-2 left-3 z-[10001]"
                }, null, 8, ["modelValue"]),
                k(l(o(K), {
                  sidebarWidth: c.value,
                  "onUpdate:sidebarWidth": [
                    t[1] || (t[1] = (i) => c.value = i),
                    T
                  ],
                  activeWorkspace: r,
                  class: "z-[10000] h-full max-md:absolute! max-md:w-full!",
                  documents: [e.document.value],
                  eventBus: e.eventBus,
                  isDroppable: () => !1,
                  layout: "modal",
                  sidebarState: e.sidebarState.state,
                  workspaces: [],
                  onSelectItem: e.sidebarState.handleSelectItem
                }, null, 8, ["sidebarWidth", "documents", "eventBus", "sidebarState", "onSelectItem"]), [
                  [b, n.value]
                ]),
                l(O, {
                  activeWorkspace: r,
                  class: "flex-1",
                  document: e.document.value,
                  documentSlug: e.document.value["x-scalar-navigation"]?.id ?? "",
                  environment: h.value,
                  eventBus: e.eventBus,
                  exampleName: e.exampleName?.value,
                  layout: "modal",
                  method: e.method?.value,
                  options: e.options,
                  path: e.path?.value,
                  plugins: e.plugins,
                  securitySchemes: C.value,
                  workspaceStore: e.workspaceStore
                }, null, 8, ["document", "documentSlug", "environment", "eventBus", "exampleName", "method", "options", "path", "plugins", "securitySchemes", "workspaceStore"])
              ])) : (m(), u("div", Y, [...t[3] || (t[3] = [
                s("span", { class: "text-c-3" }, "No document selected", -1)
              ])]))
            ]),
            _: 1
          })
        ], 8, Q),
        s("div", {
          class: "scalar-app-exit",
          onClick: t[2] || (t[2] = (i) => e.modalState.hide())
        })
      ])
    ], 512)), [
      [b, e.modalState.open]
    ]);
  }
});
export {
  me as default
};
