// @scalar/agent-chat/dist/views/Settings/Auth.vue.js
import { defineComponent as a, computed as n, watch as i, shallowRef as l, createElementBlock as y, openBlock as m, createBlock as d, createCommentVNode as S, unref as h } from "vue";
import { AuthSelector as f } from "@scalar/api-client/v2/blocks/scalar-auth-selector-block";
import { getSecurityRequirements as v, getSelectedSecurity as x } from "@scalar/api-client/v2/features/operation";
import { useFocusWithin as R } from "@vueuse/core";
import { authStorage as k, restoreAuthSecretsFromStorage as A } from "../../helpers.js";
import { useState as B } from "../../state/state.js";
const N = /* @__PURE__ */ a({
  __name: "Auth",
  props: {
    options: {},
    name: {},
    document: {},
    eventBus: {},
    securitySchemes: {},
    selectedServer: {},
    environment: {}
  },
  setup(e) {
    const { workspaceStore: r } = B(), s = n(
      () => v(e.document?.security)
    ), c = n(
      () => x(
        e.document?.["x-scalar-selected-security"],
        void 0,
        s.value
      )
    );
    i(
      [() => c, () => e.securitySchemes],
      () => {
        const t = k();
        t.setSchemas(e.name, e.securitySchemes), c.value && t.setSelectedSchemes(e.name, {
          "x-scalar-selected-security": c.value
        }), A({ documentName: e.name, workspaceStore: r });
      },
      { deep: !0 }
    );
    const u = l(), { focused: o } = R(u);
    return i(o, (t) => {
      t && r.update("x-scalar-active-document", e.name);
    }), (t, g) => (m(), y("div", {
      ref_key: "focusRef",
      ref: u,
      tabindex: "0"
    }, [
      Object.keys(e.securitySchemes).length ? (m(), d(h(f), {
        key: 0,
        environment: e.environment,
        eventBus: e.eventBus,
        isReadOnly: "",
        isStatic: "",
        layout: "reference",
        meta: { type: "document" },
        persistAuth: e.options.persistAuth,
        proxyUrl: e.options.proxyUrl ?? "",
        securityRequirements: s.value,
        securitySchemes: e.securitySchemes,
        selectedSecurity: c.value,
        server: e.selectedServer,
        title: "Authentication"
      }, null, 8, ["environment", "eventBus", "persistAuth", "proxyUrl", "securityRequirements", "securitySchemes", "selectedSecurity", "server"])) : S("", !0)
    ], 512));
  }
});
export {
  N as default
};
