// @scalar/agent-chat/dist/views/Settings/DocSettings.vue2.js
import { defineComponent as d, computed as t, createElementBlock as a, openBlock as S, createElementVNode as s, createVNode as m, unref as n } from "vue";
import { getSelectedServer as f } from "@scalar/api-client/v2/features/operation";
import { getActiveEnvironment as g, getServers as B } from "@scalar/api-client/v2/helpers";
import h from "../../components/ServerSelector.vue.js";
import { useState as y } from "../../state/state.js";
import k from "./Auth.vue.js";
const p = { class: "docSettings" }, C = /* @__PURE__ */ d({
  __name: "DocSettings",
  props: {
    document: {},
    name: {}
  },
  setup(e) {
    const { workspaceStore: u, config: i, eventBus: r } = y(), v = t(
      () => g(u, e.document)
    ), c = t(() => {
      const o = B(e.document.servers, {
        documentUrl: e.document["x-scalar-original-source-url"]
      });
      return f(e.document, o);
    }), l = t(
      () => e.document.components?.securitySchemes ?? {}
    );
    return (o, x) => (S(), a("div", p, [
      s("div", null, [
        m(k, {
          document: e.document,
          environment: v.value,
          eventBus: n(r),
          name: e.name,
          options: n(i),
          securitySchemes: l.value,
          selectedServer: c.value
        }, null, 8, ["document", "environment", "eventBus", "name", "options", "securitySchemes", "selectedServer"])
      ]),
      s("div", null, [
        m(h, {
          eventBus: n(r),
          selectedServer: c.value,
          servers: e.document.servers ?? []
        }, null, 8, ["eventBus", "selectedServer", "servers"])
      ])
    ]));
  }
});
export {
  C as default
};
