// @scalar/agent-chat/dist/views/Settings/Settings.vue2.js
import { defineComponent as k, createBlock as i, openBlock as t, unref as e, withCtx as g, createElementVNode as r, createVNode as d, createElementBlock as n, Fragment as y, renderList as f, createCommentVNode as v, normalizeClass as x, createTextVNode as S, toDisplayString as _, isRef as C } from "vue";
import { ScalarModal as w, ScalarColorModeToggle as D, ScalarTextInput as h } from "@scalar/components";
import { ScalarIconCaretDown as N, ScalarIconCaretRight as U } from "@scalar/icons";
import { useState as V } from "../../state/state.js";
import b from "./DocSettings.vue.js";
const I = { class: "settingsHeading" }, T = { class: "documentList" }, B = ["onClick"], L = { key: 0 }, M = {
  key: 1,
  class: "noDocuments"
}, R = { class: "proxyUrlContainer" }, F = /* @__PURE__ */ k({
  __name: "Settings",
  props: {
    modalState: {}
  },
  setup(u) {
    const { workspaceStore: o, proxyUrl: c } = V();
    function p(m) {
      o.update("x-scalar-active-document", m);
    }
    return (m, s) => (t(), i(e(w), {
      class: "settingsModal",
      state: u.modalState
    }, {
      default: g(() => [
        r("div", I, [
          s[1] || (s[1] = r("h1", null, "Settings", -1)),
          d(e(D), { class: "colorToggle ml-auto" })
        ]),
        r("div", T, [
          Object.entries(e(o).workspace.documents).length ? (t(!0), n(y, { key: 0 }, f(Object.entries(
            e(o).workspace.documents
          ), ([a, l]) => (t(), n("div", {
            key: a,
            class: "document"
          }, [
            r("button", {
              class: x(["documentName", {
                documentNameActive: e(o).workspace.activeDocument === l
              }]),
              type: "button",
              onClick: (j) => p(a)
            }, [
              S(" @" + _(a) + " ", 1),
              e(o).workspace.activeDocument === l ? (t(), i(e(N), { key: 0 })) : (t(), i(e(U), { key: 1 }))
            ], 10, B),
            e(o).workspace.activeDocument === l ? (t(), n("div", L, [
              d(b, {
                document: l,
                name: a
              }, null, 8, ["document", "name"])
            ])) : v("", !0)
          ]))), 128)) : (t(), n("div", M, " No APIs selected. Use + to add context. "))
        ]),
        r("div", R, [
          s[2] || (s[2] = r("label", { for: "proxyUrl" }, "Proxy URL", -1)),
          d(e(h), {
            id: "proxyUrl",
            modelValue: e(c),
            "onUpdate:modelValue": s[0] || (s[0] = (a) => C(c) ? c.value = a : null),
            placeholder: "https://proxy.scalar.com"
          }, null, 8, ["modelValue"])
        ])
      ]),
      _: 1
    }, 8, ["state"]));
  }
});
export {
  F as default
};
