// @scalar/oas-utils/dist/entities/workspace/workspace.js
import { themeIds } from '@scalar/themes';
import { z } from 'zod';
import { KEYDOWN_KEYS, HOTKEY_EVENT_NAMES } from './consts/hot-keys.js';
import { nanoidSchema } from './shared/utility.js';

const modifier = z
    .enum(['Meta', 'Control', 'Shift', 'Alt', 'default'])
    .optional()
    .default('default');
const modifiers = z.array(modifier).optional().default(['default']);
const hotKeys = z.record(z.enum(KEYDOWN_KEYS), z.object({
    modifiers: modifiers.optional(),
    event: z.enum(HOTKEY_EVENT_NAMES),
}));
const hotKeyConfigSchema = z
    .object({
    modifiers,
    hotKeys: hotKeys.optional(),
})
    .optional();
const workspaceSchema = z.object({
    uid: nanoidSchema,
    name: z.string().default('Default Workspace'),
    /** Workspace description */
    description: z.string().default('Basic Scalar Workspace'),
    /** Controls read only mode for most entitites, but not things like params */
    isReadOnly: z.boolean().default(false),
    /** List of all collection uids in a given workspace */
    collectionUids: z.array(z.string()).default([]),
    /** List of all environment uids in a given workspace */
    environmentUids: z.array(z.string()).default([]),
    /** Customize hotkeys */
    hotKeyConfig: hotKeyConfigSchema,
    /** List of all cookie uids in a given workspace */
    cookieUids: z.array(z.string()).default([]),
    /** Workspace level proxy for all requests to be sent through */
    proxyUrl: z.string().optional(),
    /** Workspace level theme, we might move this to user level later */
    themeId: z.enum(themeIds).optional().default('default'),
});
const createWorkspace = (payload) => workspaceSchema.parse(payload);

export { createWorkspace };
