// @scalar/agent-chat/dist/components/ResponseBody/ResponseBody.vue.js
import { defineComponent as d, watch as l, createElementBlock as m, openBlock as a, Fragment as r, createBlock as n, createCommentVNode as i, withCtx as f, createTextVNode as g } from "vue";
import y from "./ResponseBodyInfo.vue.js";
import c from "./ResponseBodyPreview.vue.js";
import s from "./ResponseBodyRaw.vue.js";
const k = /* @__PURE__ */ d({
  __name: "ResponseBody",
  props: {
    data: {},
    responseBody: {},
    mediaConfig: {},
    display: {}
  },
  setup(e) {
    return l(
      () => e.display,
      (o) => console.log(o)
    ), (o, t) => (a(), m(r, null, [
      e.mediaConfig?.raw && e.display === "raw" && e.mediaConfig.language ? (a(), n(s, {
        key: e.responseBody.dataUrl,
        content: e.data,
        language: e.mediaConfig.language
      }, null, 8, ["content", "language"])) : i("", !0),
      e.mediaConfig?.preview && e.display === "preview" ? (a(), n(c, {
        key: e.responseBody.dataUrl,
        alpha: e.mediaConfig.alpha,
        mode: e.mediaConfig.preview,
        src: e.responseBody.dataUrl,
        type: e.responseBody.mimeType?.essence ?? ""
      }, null, 8, ["alpha", "mode", "src", "type"])) : i("", !0),
      !e.mediaConfig?.raw && !e.mediaConfig?.preview ? (a(), n(y, { key: 2 }, {
        default: f(() => [...t[0] || (t[0] = [
          g(" Binary file ", -1)
        ])]),
        _: 1
      })) : i("", !0)
    ], 64));
  }
});
export {
  k as default
};
