// @scalar/components/dist/components/ScalarThemeSwatches/ScalarThemeSwatches.vue.js
import { defineComponent as o, createElementBlock as a, openBlock as s, mergeProps as c, unref as e, Fragment as g, renderList as d, normalizeStyle as i } from "vue";
import { useBindCx as m } from "@scalar/use-hooks/useBindCx";
import { useThemeSwatches as u, THEME_CSS_VARS as b } from "./useThemeSwatches.js";
const h = {}, x = /* @__PURE__ */ o({
  ...h,
  inheritAttrs: !1,
  __name: "ScalarThemeSwatches",
  props: {
    css: {}
  },
  setup(l) {
    const { colors: r } = u(() => l.css), { cx: n } = m();
    return (k, _) => (s(), a("div", c(
      {
        style: {
          "--bg-light": e(r).light["--scalar-background-1"],
          "--bg-dark": e(r).dark["--scalar-background-1"]
        }
      },
      e(n)(
        "flex *:size-3 overflow-hidden rounded",
        "bg-(--bg-light) dark:bg-(--bg-dark)"
      )
    ), [
      (s(!0), a(g, null, d(e(b), (t) => (s(), a("div", {
        key: t,
        class: "bg-(--bg-light) dark:bg-(--bg-dark)",
        style: i({ "--bg-light": e(r).light[t], "--bg-dark": e(r).dark[t] })
      }, null, 4))), 128))
    ], 16));
  }
});
export {
  x as default
};
