// @scalar/api-client/dist/v2/components/sidebar/SidebarMenu.vue.js
import { defineComponent as p, createBlock as u, openBlock as k, unref as o, withCtx as t, createVNode as r, renderSlot as m, createTextVNode as d } from "vue";
import { ScalarMenu as S, ScalarMenuSection as f, ScalarMenuWorkspacePicker as w, ScalarMenuLink as M, ScalarMenuResources as C, ScalarMenuSupport as V } from "@scalar/components";
import { ScalarIconGear as W } from "@scalar/icons";
import { RouterLink as v } from "vue-router";
const L = /* @__PURE__ */ p({
  __name: "SidebarMenu",
  props: {
    activeWorkspace: {},
    workspaces: {}
  },
  emits: ["create:workspace", "select:workspace"],
  setup(a, { emit: i }) {
    const n = i;
    return (c, e) => (k(), u(o(S), null, {
      products: t(() => [...e[2] || (e[2] = [])]),
      sections: t(({ close: l }) => [
        r(o(f), null, {
          default: t(() => [
            m(c.$slots, "sidebarMenuActions", {}, () => [
              r(o(w), {
                modelValue: a.activeWorkspace.id,
                workspaceOptions: a.workspaces,
                onCreateWorkspace: e[0] || (e[0] = (s) => n("create:workspace")),
                "onUpdate:modelValue": e[1] || (e[1] = (s) => n("select:workspace", s))
              }, null, 8, ["modelValue", "workspaceOptions"]),
              r(o(M), {
                is: o(v),
                icon: o(W),
                to: "/settings",
                onClick: l
              }, {
                default: t(() => [...e[3] || (e[3] = [
                  d(" Settings ", -1)
                ])]),
                _: 1
              }, 8, ["is", "icon", "onClick"])
            ])
          ]),
          _: 2
        }, 1024),
        r(o(C)),
        r(o(V))
      ]),
      _: 3
    }));
  }
});
export {
  L as default
};
