// @scalar/api-client/dist/v2/components/sidebar/Sidebar.vue.js
import { defineComponent as h, mergeModels as y, useSlots as D, ref as V, useModel as B, computed as M, createBlock as d, openBlock as l, unref as r, withCtx as s, createVNode as C, createSlots as E, renderSlot as i, createElementBlock as v, createCommentVNode as n, Fragment as R, normalizeProps as f, guardReactiveProps as b, createElementVNode as w, isRef as z } from "vue";
import { ScalarIconButton as F, ScalarSidebarSearchInput as N } from "@scalar/components";
import { ScalarIconFileDashed as q, ScalarIconMagnifyingGlass as A } from "@scalar/icons";
import { ScalarSidebar as U } from "@scalar/sidebar";
import j from "./SidebarMenu.vue.js";
import { useSearchIndex as G } from "../../features/search/hooks/use-search-index.js";
import H from "../resize/Resize.vue.js";
const J = { class: "bg-sidebar-b-1 z-1 flex flex-col gap-1.5 px-3 pb-1.5" }, K = { class: "flex items-center justify-between" }, L = { key: 1 }, _ = /* @__PURE__ */ h({
  __name: "Sidebar",
  props: /* @__PURE__ */ y({
    sidebarState: {},
    layout: {},
    activeWorkspace: {},
    workspaces: {},
    documents: {},
    isDroppable: { type: [Boolean, Function] }
  }, {
    sidebarWidth: {
      required: !0,
      default: 288
    },
    sidebarWidthModifiers: {}
  }),
  emits: /* @__PURE__ */ y(["selectItem", "select:workspace", "create:workspace", "reorder"], ["update:sidebarWidth"]),
  setup(t, { emit: g }) {
    const c = g, p = D(), u = V(!1), k = B(t, "sidebarWidth"), S = (o) => o.type === "example" && o.title === "draft", { query: m, results: x } = G(() => t.documents), I = M(() => x.value ?? t.sidebarState.items.value), W = (o) => {
      c("selectItem", o), m.value = "", u.value = !1;
    };
    return (o, a) => (l(), d(r(H), {
      width: k.value,
      "onUpdate:width": a[5] || (a[5] = (e) => k.value = e),
      class: "flex flex-col"
    }, {
      default: s(() => [
        C(r(U), {
          class: "flex w-auto flex-1 pt-2",
          indent: 15,
          isDraggable: t.layout !== "modal",
          isDroppable: t.isDroppable,
          isExpanded: t.sidebarState.isExpanded,
          isSelected: t.sidebarState.isSelected,
          items: I.value,
          layout: "client",
          onReorder: a[4] || (a[4] = (e, $) => c("reorder", e, $)),
          onSelectItem: W
        }, E({
          header: s(() => [
            w("div", J, [
              w("div", K, [
                t.layout === "desktop" ? (l(), d(j, {
                  key: 0,
                  activeWorkspace: t.activeWorkspace,
                  workspaces: t.workspaces,
                  "onCreate:workspace": a[0] || (a[0] = (e) => c("create:workspace")),
                  "onSelect:workspace": a[1] || (a[1] = (e) => c("select:workspace", e))
                }, {
                  sidebarMenuActions: s(() => [
                    i(o.$slots, "sidebarMenuActions")
                  ]),
                  _: 3
                }, 8, ["activeWorkspace", "workspaces"])) : t.layout === "modal" ? (l(), v("div", L)) : n("", !0),
                t.layout !== "web" ? (l(), d(r(F), {
                  key: 2,
                  icon: r(A),
                  label: "Search",
                  onClick: a[2] || (a[2] = (e) => u.value = !u.value)
                }, null, 8, ["icon"])) : n("", !0)
              ]),
              u.value || t.layout === "web" ? (l(), d(r(N), {
                key: 0,
                modelValue: r(m),
                "onUpdate:modelValue": a[3] || (a[3] = (e) => z(m) ? m.value = e : null),
                autofocus: t.layout !== "web"
              }, null, 8, ["modelValue", "autofocus"])) : n("", !0)
            ])
          ]),
          decorator: s((e) => [
            i(o.$slots, "decorator", f(b(e)))
          ]),
          icon: s((e) => [
            p.icon || S(e.item) ? (l(), v(R, { key: 0 }, [
              S(e.item) ? (l(), d(r(q), { key: 0 })) : n("", !0),
              i(o.$slots, "icon", f(b(e)))
            ], 64)) : n("", !0)
          ]),
          before: s(() => [
            i(o.$slots, "workspaceButton")
          ]),
          footer: s(() => [
            i(o.$slots, "footer")
          ]),
          _: 2
        }, [
          p.empty ? {
            name: "empty",
            fn: s((e) => [
              i(o.$slots, "empty", f(b(e)))
            ]),
            key: "0"
          } : void 0
        ]), 1032, ["isDraggable", "isDroppable", "isExpanded", "isSelected", "items"])
      ]),
      _: 3
    }, 8, ["width"]));
  }
});
export {
  _ as default
};
