// @scalar/api-client/dist/views/Request/RequestSection/RequestAuth/RequestAuth.vue2.js
import { defineComponent as z, useId as K, ref as x, computed as h, createBlock as V, openBlock as u, withCtx as s, createVNode as m, renderSlot as j, unref as i, createElementVNode as r, createElementBlock as y, Fragment as g, createTextVNode as k, toDisplayString as p, createCommentVNode as D, withModifiers as Y, normalizeClass as F } from "vue";
import { useModal as H, ScalarComboboxMultiselect as J, ScalarButton as P, ScalarListboxCheckbox as W, ScalarIconButton as G } from "@scalar/components";
import { safeLocalStorage as Q, CLIENT_LS_KEYS as X } from "@scalar/helpers/object/local-storage";
import { ScalarIconCaretDown as Z, ScalarIconTrash as _ } from "@scalar/icons";
import { isDefined as ee } from "@scalar/oas-utils/helpers";
import te from "../../../../components/ViewLayout/ViewLayoutCollapse.vue.js";
import { useWorkspace as oe } from "../../../../store/store.js";
import le from "./DeleteRequestAuthModal.vue.js";
import ne from "./RequestAuthDataTable.vue.js";
import { getSecurityRequirements as ie, formatComplexScheme as se, formatScheme as re, getSchemeOptions as ae } from "../../libs/auth.js";
const ce = ["id"], ue = { class: "flex flex-1" }, me = { class: "min-w-0 flex-1 truncate" }, Ce = /* @__PURE__ */ z({
  __name: "RequestAuth",
  props: {
    collection: {},
    isReadOnly: { type: Boolean, default: !1 },
    environment: {},
    envVariables: {},
    layout: {},
    operation: {},
    persistAuth: { type: Boolean, default: !1 },
    selectedSecuritySchemeUids: {},
    server: {},
    title: {},
    workspace: {}
  },
  emits: ["authorized", "activeSchemes"],
  setup(l, { emit: E }) {
    const C = E, {
      securitySchemes: d,
      securitySchemeMutators: $,
      requestMutators: B,
      collectionMutators: I
    } = oe(), A = K(), v = x(null), f = H(), S = x(
      null
    ), R = x(!1), q = h(() => {
      const o = ie(l.operation, l.collection);
      return { filteredRequirements: o.filter((t) => Object.keys(t).length), requirements: o };
    }), b = h(() => {
      const { filteredRequirements: o, requirements: e } = q.value;
      if (!e.length)
        return null;
      const n = !e.some(
        (N) => Object.keys(N).length > 1
      ) && o.length < e.length;
      return { icon: n ? "Unlock" : "Lock", text: n ? "Optional" : "Required" };
    }), a = h(
      () => l.selectedSecuritySchemeUids.map((o) => {
        if (Array.isArray(o))
          return se(o, d);
        const e = d[o ?? ""];
        if (e)
          return re(e);
      }).filter(ee)
    );
    function M(o) {
      const e = o.find((n) => n.payload), t = o.filter((n) => !n.payload).map(({ id: n }) => {
        const c = n.split(",");
        return c.length > 1 ? c : n;
      });
      if (e?.payload) {
        const n = $.add(
          e.payload,
          l.collection?.uid
        );
        n && t.push(n.uid);
      }
      w(t);
    }
    const w = (o) => {
      if (l.collection.useCollectionSecurity) {
        if (I.edit(l.collection.uid, "selectedSecuritySchemeUids", o), !l.persistAuth)
          return;
        const e = o.map((t) => Array.isArray(t) ? t.map((n) => d[n]?.nameKey) : d[t]?.nameKey);
        Q().setItem(
          X.SELECTED_SECURITY_SCHEMES,
          JSON.stringify(e)
        );
      } else l.operation?.uid && B.edit(l.operation.uid, "selectedSecuritySchemeUids", o);
    };
    function O({ id: o, label: e }) {
      S.value = { id: o, label: e }, f.show();
    }
    const T = (o) => {
      if (!o)
        return;
      const e = l.selectedSecuritySchemeUids.filter((t) => {
        const n = o.split(",");
        return n.length > 1 && Array.isArray(t) && n.length === t.length ? t.every((c) => !n.includes(c)) : t !== o;
      });
      w(e), v.value?.$el.focus(), f.hide();
    }, U = h(
      () => ae(
        q.value.filteredRequirements,
        l.collection?.securitySchemes ?? [],
        d,
        l.isReadOnly
      )
    ), L = (o) => {
      R.value && o.stopPropagation(), v.value?.$el.click();
    };
    return (o, e) => (u(), V(te, {
      class: "group/params relative",
      itemCount: a.value.length,
      layout: l.layout,
      "onUpdate:modelValue": e[4] || (e[4] = (t) => R.value = t)
    }, {
      title: s(() => [
        r("div", {
          id: i(A),
          class: "inline-flex items-center gap-0.5 leading-[20px]"
        }, [
          r("span", null, p(l.title), 1),
          b.value ? (u(), y("span", {
            key: 0,
            class: F(["text-c-3 hover:bg-b-3 hover:text-c-1 -mr-1 cursor-pointer rounded px-1 py-0.5 text-xs leading-[normal]", { "text-c-1": b.value.text === "Required" }]),
            onClick: L
          }, p(b.value.text), 3)) : D("", !0)
        ], 8, ce)
      ]),
      actions: s(() => [
        r("div", ue, [
          m(i(J), {
            class: "w-72 text-xs",
            modelValue: a.value,
            multiple: "",
            options: U.value,
            placement: "bottom-end",
            teleport: "",
            onDelete: O,
            "onUpdate:modelValue": M
          }, {
            option: s(({ option: t, selected: n }) => [
              m(i(W), {
                multiselect: "",
                selected: n
              }, null, 8, ["selected"]),
              r("div", me, p(t.label), 1),
              t.isDeletable ?? !l.isReadOnly ? (u(), V(i(G), {
                key: 0,
                class: "-m-0.5 shrink-0 p-0.5 opacity-0 group-hover/item:opacity-100",
                icon: i(_),
                label: `Delete ${t.label}`,
                size: "xs",
                onClick: Y((c) => O(t), ["stop"])
              }, null, 8, ["icon", "label", "onClick"])) : D("", !0)
            ]),
            default: s(() => [
              m(i(P), {
                ref_key: "comboboxButtonRef",
                ref: v,
                "aria-describedby": i(A),
                class: "group/combobox-button hover:text-c-1 text-c-2 flex h-fit w-full items-center gap-1 px-0.75 py-0.25 text-base font-normal transition-transform",
                variant: "ghost"
              }, {
                default: s(() => [
                  a.value.length === 1 ? (u(), y(g, { key: 0 }, [
                    e[5] || (e[5] = r("span", { class: "sr-only" }, "Selected Auth Type:", -1)),
                    k(" " + p(a.value[0]?.label), 1)
                  ], 64)) : a.value.length > 1 ? (u(), y(g, { key: 1 }, [
                    e[6] || (e[6] = k(" Multiple ", -1)),
                    e[7] || (e[7] = r("span", { class: "sr-only" }, "Auth Types Selected", -1))
                  ], 64)) : (u(), y(g, { key: 2 }, [
                    e[8] || (e[8] = r("span", { class: "sr-only" }, "Select", -1)),
                    e[9] || (e[9] = k(" Auth Type ", -1))
                  ], 64)),
                  m(i(Z), {
                    class: "size-3 shrink-0 transition-transform duration-100 group-aria-expanded/combobox-button:rotate-180",
                    weight: "bold"
                  })
                ]),
                _: 1
              }, 8, ["aria-describedby"])
            ]),
            _: 1
          }, 8, ["modelValue", "options"])
        ])
      ]),
      default: s(() => [
        m(ne, {
          collection: l.collection,
          envVariables: l.envVariables,
          environment: l.environment,
          layout: l.layout,
          persistAuth: l.persistAuth,
          selectedSchemeOptions: a.value,
          server: l.server,
          workspace: l.workspace,
          onActiveSchemes: e[0] || (e[0] = (t) => C("activeSchemes", t)),
          onAuthorized: e[1] || (e[1] = (t) => C("authorized"))
        }, {
          "oauth-actions": s(() => [
            j(o.$slots, "oauth-actions", {}, void 0, !0)
          ]),
          _: 3
        }, 8, ["collection", "envVariables", "environment", "layout", "persistAuth", "selectedSchemeOptions", "server", "workspace"]),
        m(le, {
          scheme: S.value,
          state: i(f),
          onClose: e[2] || (e[2] = (t) => i(f).hide()),
          onDelete: e[3] || (e[3] = (t) => T(S.value?.id))
        }, null, 8, ["scheme", "state"])
      ]),
      _: 3
    }, 8, ["itemCount", "layout"]));
  }
});
export {
  Ce as default
};
