// @scalar/api-client/dist/views/Request/RequestSection/RequestAuth/RequestAuthDataTable.vue2.js
import { defineComponent as A, ref as h, computed as v, watch as f, createElementBlock as l, openBlock as n, withModifiers as z, createCommentVNode as b, createBlock as C, createVNode as x, normalizeClass as a, Fragment as V, renderList as $, createElementVNode as y, toDisplayString as O, unref as u, withCtx as p, renderSlot as B } from "vue";
import { useModal as M } from "@scalar/components";
import { isDefined as N } from "@scalar/helpers/array/is-defined";
import D from "./DeleteRequestAuthModal.vue.js";
import E from "./RequestAuthTab.vue.js";
import { useWorkspace as U } from "../../../../store/store.js";
import j from "../../../../components/DataTable/DataTable.vue.js";
const q = ["onClick"], F = { class: "relative z-10 font-medium whitespace-nowrap" }, I = {
  key: 0,
  class: "absolute inset-x-1 bottom-[var(--scalar-border-width)] left-1/2 z-1 h-px w-full -translate-x-1/2 bg-current"
}, P = /* @__PURE__ */ A({
  __name: "RequestAuthDataTable",
  props: {
    collection: {},
    environment: {},
    envVariables: {},
    layout: { default: "client" },
    persistAuth: { type: Boolean, default: !1 },
    selectedSchemeOptions: { default: () => [] },
    server: {},
    workspace: {}
  },
  emits: ["authorized", "activeSchemes"],
  setup(e, { emit: S }) {
    const m = S, { securitySchemes: k } = U(), d = M(), g = h(
      null
    ), o = h(0), i = v(() => {
      if (!e.selectedSchemeOptions || e.selectedSchemeOptions.length === 0)
        return [];
      const r = e.selectedSchemeOptions[o.value];
      if (!r)
        return [];
      const t = r.id.split(",").filter(Boolean);
      return t.length > 1 ? t : [r.id];
    });
    f(
      i,
      (r) => {
        m(
          "activeSchemes",
          r.map((t) => k[t]).filter(N)
        );
      },
      { immediate: !0 }
    );
    const w = v(() => i.value.length > 0);
    return f(
      () => e.selectedSchemeOptions,
      (r) => {
        (!r || !r[o.value]) && (o.value = Math.max(0, o.value - 1));
      }
    ), (r, t) => (n(), l("form", {
      onSubmit: t[2] || (t[2] = z(() => {
      }, ["prevent"]))
    }, [
      e.selectedSchemeOptions.length > 1 ? (n(), l("div", {
        key: 0,
        class: a(["box-content flex flex-wrap gap-x-2.5 overflow-hidden border border-b-0 px-3", e.layout === "client" && "border-x-0"])
      }, [
        (n(!0), l(V, null, $(e.selectedSchemeOptions, (s, c) => (n(), l("div", {
          key: s.id,
          class: a(["relative z-1 -mb-[var(--scalar-border-width)] flex h-8 cursor-pointer", [o.value === c ? "text-c-1" : "text-c-3"]])
        }, [
          y("button", {
            class: "floating-bg relative cursor-pointer border-b-[1px] border-transparent py-1 text-sm font-medium",
            type: "button",
            onClick: (L) => o.value = c
          }, [
            y("span", F, O(s.label), 1)
          ], 8, q),
          o.value === c ? (n(), l("div", I)) : b("", !0)
        ], 2))), 128))
      ], 2)) : b("", !0),
      w.value ? (n(), C(u(j), {
        key: 1,
        class: a(["flex-1", e.layout === "reference" && "bg-b-1 rounded-b-lg border border-t-0"]),
        columns: [""],
        presentational: ""
      }, {
        default: p(() => [
          x(E, {
            collection: e.collection,
            envVariables: e.envVariables,
            environment: e.environment,
            layout: e.layout,
            persistAuth: e.persistAuth,
            securitySchemeUids: i.value,
            server: e.server,
            workspace: e.workspace,
            onAuthorized: t[0] || (t[0] = (s) => m("authorized"))
          }, {
            "oauth-actions": p(() => [
              B(r.$slots, "oauth-actions", {}, void 0, !0)
            ]),
            _: 3
          }, 8, ["collection", "envVariables", "environment", "layout", "persistAuth", "securitySchemeUids", "server", "workspace"])
        ]),
        _: 3
      }, 8, ["class"])) : (n(), l("div", {
        key: 2,
        class: a([
          "text-c-3 bg-b-1 flex min-h-16 items-center justify-center border-t px-4 text-sm",
          e.layout === "reference" && "min-h-[calc(4rem+0.5px)] rounded-b-lg border"
        ])
      }, " No authentication selected ", 2)),
      x(D, {
        scheme: g.value,
        state: u(d),
        onClose: t[1] || (t[1] = (s) => u(d).hide())
      }, null, 8, ["scheme", "state"])
    ], 32));
  }
});
export {
  P as default
};
