// @scalar/api-client/dist/views/Request/RequestSection/RequestAuth/RequestAuthTab.vue.js
import { defineComponent as N, computed as P, ref as j, onMounted as L, createElementBlock as i, openBlock as a, Fragment as p, renderList as C, createBlock as k, createCommentVNode as b, unref as d, withCtx as l, createVNode as u, createElementVNode as h, toDisplayString as A, mergeProps as y, createTextVNode as x, normalizeClass as T, renderSlot as M, capitalize as g } from "vue";
import { ScalarMarkdownSummary as R } from "@scalar/components";
import { useWorkspace as Q } from "../../../../store/store.js";
import { updateScheme as q } from "../helpers/update-scheme.js";
import { restoreAuthFromLocalStorage as D } from "./helpers/restore-auth-from-local-storage.js";
import E from "./OAuth2.vue.js";
import V from "./RequestAuthDataTableInput.vue.js";
import c from "../../../../components/DataTable/DataTableRow.vue.js";
import F from "../../../../components/DataTable/DataTableCell.vue.js";
const O = { class: "bg-b-1 text-c-2 outline-b-3 top-0 z-1 h-full w-full overflow-hidden px-3 py-1.25 text-ellipsis group-hover/auth:absolute group-hover/auth:h-auto group-hover/auth:border-b *:first:line-clamp-1 *:first:text-ellipsis group-hover/auth:*:first:line-clamp-none" }, _ = {
  key: 0,
  class: "flex min-h-8 border-t text-base"
}, H = { class: "flex h-8 max-w-full gap-2.5 overflow-x-auto px-3" }, W = ["onClick"], G = { class: "relative z-10" }, le = /* @__PURE__ */ N({
  __name: "RequestAuthTab",
  props: {
    collection: {},
    environment: {},
    envVariables: {},
    layout: {},
    persistAuth: { type: Boolean, default: !1 },
    securitySchemeUids: {},
    server: {},
    workspace: {}
  },
  emits: ["authorized"],
  setup(r, { emit: z }) {
    const B = z, $ = Q(), { securitySchemes: I } = $, U = P(
      () => r.securitySchemeUids.map((o) => ({
        scheme: I[o]
      }))
    ), s = j(""), S = (o) => {
      const t = o.description ? `: ${o.description}` : "", e = `${g(o.nameKey)}${t || `: ${o.type}`}`;
      if (o.type === "apiKey")
        return `${g(o.nameKey)}${t || `: ${o.in}`}`;
      if (o.type === "oauth2") {
        const n = Object.values(o.flows ?? {})[0];
        return `${g(o.nameKey)}: ${s.value ? s.value : n?.type ?? ""}${t}`;
      }
      return o.type === "http" ? `${g(o.nameKey)}: ${o.scheme}${t}` : `${e}${t}`;
    }, v = (o, t, e) => {
      q(o, t, e, $, r.persistAuth);
    };
    L(() => {
      r.persistAuth && D($, r.collection.uid);
    });
    const m = {
      environment: r.environment,
      envVariables: r.envVariables,
      workspace: r.workspace
    };
    return (o, t) => (a(!0), i(p, null, C(U.value, ({ scheme: e }) => (a(), i(p, {
      key: e?.uid
    }, [
      U.value.length > 1 && e ? (a(), k(d(c), { key: 0 }, {
        default: l(() => [
          u(d(F), {
            "aria-label": S(e),
            class: "text-c-2 group/auth flex items-center leading-[22px] whitespace-nowrap outline-none hover:whitespace-normal"
          }, {
            default: l(() => [
              h("p", O, A(S(e)), 1)
            ]),
            _: 2
          }, 1032, ["aria-label"])
        ]),
        _: 2
      }, 1024)) : b("", !0),
      e?.description && U.value.length <= 1 ? (a(), k(d(c), { key: 1 }, {
        default: l(() => [
          u(d(F), {
            "aria-label": e.description,
            class: "max-h-[auto]"
          }, {
            default: l(() => [
              u(d(R), {
                class: "auth-description bg-b-1 text-c-2 min-w-0 flex-1 px-3 py-1.25",
                value: e.description
              }, null, 8, ["value"])
            ]),
            _: 2
          }, 1032, ["aria-label"])
        ]),
        _: 2
      }, 1024)) : b("", !0),
      e?.type === "http" ? (a(), i(p, { key: 2 }, [
        e.scheme === "bearer" ? (a(), k(d(c), { key: 0 }, {
          default: l(() => [
            u(V, y({ ref_for: !0 }, m, {
              containerClass: r.layout === "reference" && "border-t",
              modelValue: e.token,
              placeholder: "Token",
              type: "password",
              "onUpdate:modelValue": (n) => v(e.uid, "token", n)
            }), {
              default: l(() => [...t[1] || (t[1] = [
                x(" Bearer Token ", -1)
              ])]),
              _: 1
            }, 16, ["containerClass", "modelValue", "onUpdate:modelValue"])
          ]),
          _: 2
        }, 1024)) : e?.scheme === "basic" ? (a(), i(p, { key: 1 }, [
          u(d(c), null, {
            default: l(() => [
              u(V, y({ ref_for: !0 }, m, {
                class: "text-c-2",
                modelValue: e.username,
                placeholder: "janedoe",
                required: "",
                "onUpdate:modelValue": (n) => v(e.uid, "username", n)
              }), {
                default: l(() => [...t[2] || (t[2] = [
                  x(" Username ", -1)
                ])]),
                _: 1
              }, 16, ["modelValue", "onUpdate:modelValue"])
            ]),
            _: 2
          }, 1024),
          u(d(c), null, {
            default: l(() => [
              u(V, y({ ref_for: !0 }, m, {
                modelValue: e.password,
                placeholder: "********",
                type: "password",
                "onUpdate:modelValue": (n) => v(e.uid, "password", n)
              }), {
                default: l(() => [...t[3] || (t[3] = [
                  x(" Password ", -1)
                ])]),
                _: 1
              }, 16, ["modelValue", "onUpdate:modelValue"])
            ]),
            _: 2
          }, 1024)
        ], 64)) : b("", !0)
      ], 64)) : e?.type === "apiKey" ? (a(), i(p, { key: 3 }, [
        u(d(c), null, {
          default: l(() => [
            u(V, y({ ref_for: !0 }, m, {
              containerClass: r.layout === "reference" && "border-t",
              modelValue: e.name,
              placeholder: "api-key",
              "onUpdate:modelValue": (n) => v(e.uid, "name", n)
            }), {
              default: l(() => [...t[4] || (t[4] = [
                x(" Name ", -1)
              ])]),
              _: 1
            }, 16, ["containerClass", "modelValue", "onUpdate:modelValue"])
          ]),
          _: 2
        }, 1024),
        u(d(c), null, {
          default: l(() => [
            u(V, y({ ref_for: !0 }, m, {
              modelValue: e.value,
              placeholder: "QUxMIFlPVVIgQkFTRSBBUkUgQkVMT05HIFRPIFVT",
              type: "password",
              "onUpdate:modelValue": (n) => v(e.uid, "value", n)
            }), {
              default: l(() => [...t[5] || (t[5] = [
                x(" Value ", -1)
              ])]),
              _: 1
            }, 16, ["modelValue", "onUpdate:modelValue"])
          ]),
          _: 2
        }, 1024)
      ], 64)) : e?.type === "oauth2" ? (a(), i(p, { key: 4 }, [
        u(d(c), null, {
          default: l(() => [
            Object.keys(e.flows).length > 1 ? (a(), i("div", _, [
              h("div", H, [
                (a(!0), i(p, null, C(e?.flows, (n, f, w) => (a(), i("button", {
                  key: f,
                  class: T(["floating-bg text-c-3 relative cursor-pointer border-b-[1px] border-transparent py-1 text-base font-medium", {
                    "!text-c-1 !rounded-none border-b-[1px] !border-current": r.layout !== "reference" && (s.value === f || w === 0 && !s.value),
                    "!text-c-1 !rounded-none border-b-[1px] !border-current opacity-100": r.layout === "reference" && (s.value === f || w === 0 && !s.value)
                  }]),
                  type: "button",
                  onClick: (K) => s.value = f
                }, [
                  h("span", G, A(f), 1)
                ], 10, W))), 128))
              ])
            ])) : b("", !0)
          ]),
          _: 2
        }, 1024),
        (a(!0), i(p, null, C(e?.flows, (n, f, w) => (a(), i(p, { key: f }, [
          s.value === f || w === 0 && !s.value ? (a(), k(E, y({
            key: 0,
            ref_for: !0
          }, m, {
            collection: r.collection,
            flow: n,
            persistAuth: r.persistAuth,
            scheme: e,
            server: r.server,
            workspace: r.workspace,
            onAuthorized: t[0] || (t[0] = (K) => B("authorized"))
          }), {
            "oauth-actions": l(() => [
              M(o.$slots, "oauth-actions")
            ]),
            _: 3
          }, 16, ["collection", "flow", "persistAuth", "scheme", "server", "workspace"])) : b("", !0)
        ], 64))), 128))
      ], 64)) : e?.type === "openIdConnect" ? (a(), i("div", {
        key: 5,
        class: T(["text-c-3 bg-b-1 flex min-h-[calc(4rem+1px)] items-center justify-center border-t border-b-0 px-4 text-base", { "rounded-b-lg": r.layout === "reference" }])
      }, " Coming soon ", 2)) : b("", !0)
    ], 64))), 128));
  }
});
export {
  le as default
};
