// @scalar/api-client/dist/views/Request/Request.vue2.js
import { defineComponent as _, computed as P, createElementBlock as f, createBlock as u, unref as e, openBlock as o, normalizeClass as k, createElementVNode as W, createVNode as n, isRef as H, withCtx as y, createCommentVNode as N } from "vue";
import S from "../../components/EmptyState.vue.js";
import g from "../../components/ViewLayout/ViewLayout.vue.js";
import B from "../../components/ViewLayout/ViewLayoutContent.vue.js";
import { useSidebar as I } from "../../hooks/useSidebar.js";
import { importCurlCommand as L } from "../../libs/importers/curl.js";
import { useActiveEntities as z } from "../../store/active-entities.js";
import A from "./RequestSection/RequestSection.vue.js";
import D from "./RequestSubpageHeader.vue.js";
import M from "./ResponseSection/ResponseSection.vue.js";
import { useWorkspace as x } from "../../store/store.js";
import { useLayout as O } from "../../hooks/useLayout.js";
const T = { class: "flex h-full" }, $ = {
  key: 0,
  class: "flex h-full flex-1 flex-col"
}, re = /* @__PURE__ */ _({
  __name: "Request",
  props: {
    invalidParams: {},
    selectedSecuritySchemeUids: {},
    requestResult: {}
  },
  emits: ["newTab"],
  setup(c) {
    const { events: b } = x(), { isSidebarOpen: r } = I(), R = x(), { layout: q } = O(), {
      activeCollection: t,
      activeExample: m,
      activeRequest: a,
      activeWorkspace: s,
      activeServer: d,
      activeEnvVariables: p,
      activeEnvironment: v,
      activeWorkspaceRequests: h
    } = z(), { modalState: C, requestHistory: V } = R, w = P(
      () => [...V].reverse().find((l) => l.request.uid === m.value?.uid)
    );
    function E(l) {
      b.commandPalette.emit({
        commandName: "Import from cURL",
        metaData: {
          parsedCurl: L(l),
          collectionUid: t.value?.uid
        }
      });
    }
    return (l, i) => e(t) && e(s) ? (o(), f("div", {
      key: 0,
      class: k(["bg-b-1 relative z-0 flex h-full flex-1 flex-col overflow-hidden pt-0", {
        "!mr-0 !mb-0 !border-0": e(q) === "modal"
      }])
    }, [
      W("div", T, [
        e(a) ? (o(), f("div", $, [
          n(D, {
            modelValue: e(r),
            "onUpdate:modelValue": i[0] || (i[0] = (U) => H(r) ? r.value = U : null),
            collection: e(t),
            envVariables: e(p),
            environment: e(v),
            operation: e(a),
            server: e(d),
            workspace: e(s),
            onHideModal: i[1] || (i[1] = () => e(C).hide()),
            onImportCurl: E
          }, null, 8, ["modelValue", "collection", "envVariables", "environment", "operation", "server", "workspace"]),
          n(g, null, {
            default: y(() => [
              e(m) ? (o(), u(B, {
                key: 0,
                class: k(["flex-1", [e(r) ? "sidebar-active-hide-layout" : ""]])
              }, {
                default: y(() => [
                  n(A, {
                    collection: e(t),
                    envVariables: e(p),
                    environment: e(v),
                    example: e(m),
                    invalidParams: c.invalidParams,
                    operation: e(a),
                    selectedSecuritySchemeUids: c.selectedSecuritySchemeUids,
                    server: e(d),
                    workspace: e(s)
                  }, null, 8, ["collection", "envVariables", "environment", "example", "invalidParams", "operation", "selectedSecuritySchemeUids", "server", "workspace"]),
                  n(M, {
                    collection: e(t),
                    numWorkspaceRequests: e(h).length,
                    operation: e(a),
                    requestResult: c.requestResult,
                    response: w.value?.response,
                    workspace: e(s)
                  }, null, 8, ["collection", "numWorkspaceRequests", "operation", "requestResult", "response", "workspace"])
                ]),
                _: 1
              }, 8, ["class"])) : N("", !0)
            ]),
            _: 1
          })
        ])) : (o(), u(S, { key: 1 }))
      ])
    ], 2)) : (o(), u(S, { key: 1 }));
  }
});
export {
  re as default
};
