// @scalar/api-client/dist/views/Request/RequestSidebarItem.vue2.js
import { defineComponent as fe, useCssVars as ce, computed as n, ref as H, resolveComponent as ve, openBlock as m, createElementBlock as M, Fragment as G, createElementVNode as l, normalizeClass as v, unref as i, createVNode as o, withCtx as d, createBlock as b, toDisplayString as J, createCommentVNode as y, createTextVNode as D, renderSlot as be, withDirectives as ge, renderList as he, vShow as Ue, withKeys as we, withModifiers as ye, pushScopeId as De, popScopeId as ke } from "vue";
import qe from "../../components/Sidebar/Actions/DeleteSidebarListElement.vue.js";
import { PathId as I } from "../../router.js";
import { useWorkspace as Re } from "../../store/workspace.js";
import { useModal as Q, ScalarContextMenu as X, ScalarIcon as $e, ScalarModal as Y, ScalarTextField as xe, ScalarButton as Z } from "@scalar/components";
import { Draggable as Se } from "@scalar/draggable";
import { useRouter as Ce, RouterLink as Me } from "vue-router";
import q from "./RequestSidebarItemMenu.vue.js";
import Ie from "../../components/HttpMethod/HttpMethod.vue.js";
import { useSidebar as Te } from "../../hooks/useSidebar.js";
const Ee = (g) => (De("data-v-17dc8b59"), g = g(), ke(), g), ze = { class: "flex flex-row gap-1 items-center" }, Ne = { class: "relative" }, Ve = { class: "flex items-start" }, Oe = { class: "z-10 flex h-5 items-center justify-center max-w-[14px]" }, Be = { class: "flex flex-1 flex-row justify-between editable-sidebar-hover" }, Fe = { class: "relative flex h-fit" }, je = /* @__PURE__ */ Ee(() => /* @__PURE__ */ l("span", null, " ", -1)), Ke = { key: 2 }, Le = { class: "flex gap-3" }, _ = "hover:bg-sidebar-active-b indent-padding-left", _e = /* @__PURE__ */ fe({
  __name: "RequestSidebarItem",
  props: {
    isDraggable: { type: Boolean, default: !1 },
    isDroppable: { type: [Boolean, Function], default: !1 },
    parentUids: {},
    item: {}
  },
  emits: ["onDragEnd"],
  setup(g) {
    ce((e) => ({
      "4035a823": se.value,
      d307928a: le.value
    }));
    const t = g, {
      activeRequest: T,
      activeRouterParams: R,
      activeWorkspace: h,
      collections: E,
      folders: ee,
      isReadOnly: a,
      requests: z,
      requestExamples: $,
      collectionMutators: N,
      folderMutators: V,
      requestMutators: O,
      requestExampleMutators: B
    } = Re(), { replace: F } = Ce(), { collapsedSidebarFolders: j, toggleSidebarFolder: te } = Te(), ie = n(() => "childUids" in t.item), K = n(() => "spec" in t.item), L = n(() => "summary" in t.item), se = n(() => t.parentUids.length ? a.value ? `${(t.parentUids.length - 1) * 12}px` : `${t.parentUids.length * 12}px` : "12px"), le = n(() => t.parentUids.length ? a.value ? `${(t.parentUids.length - 1) * 12}px` : `${t.parentUids.length * 12}px` : "0px"), x = (e) => {
      var s;
      return "spec" in e ? (s = e.spec.info) == null ? void 0 : s.title : "summary" in e ? e.summary || e.path : "name" in e ? e.name : "";
    }, re = n(() => ("requestUid" in t.item ? z[t.item.requestUid] : t.item).method), ae = n(
      () => {
        var e;
        return j[t.item.uid] || ((e = T.value) == null ? void 0 : e.uid) === t.item.uid && t.item.childUids.length > 1;
      }
    ), ne = () => "requestUid" in t.item ? `/workspace/${h.value.uid}/request/${t.item.requestUid}/examples/${t.item.uid}` : `/workspace/${h.value.uid}/request/${t.item.uid}`, oe = n(
      () => R.value[I.Request] === "default" && T.value.uid === t.item.uid
    ), S = H(null), W = n(() => {
      let e = 0.5, s = 0.5;
      if (!S.value) return { ceiling: e, floor: s };
      const { draggingItem: w } = S.value;
      return !E[w == null ? void 0 : w.id] && K.value ? (e = 1, s = 0) : ie.value && !L.value && !K.value && (e = 0.8, s = 0.2), { ceiling: e, floor: s };
    }), de = (e, s) => !(h.value.isReadOnly || $[s.id] || E[e.id]), f = H(""), k = Q(), U = Q(), P = () => {
      "summary" in t.item ? O.edit(t.item.uid, "summary", f.value) : "requestUid" in t.item ? B.edit(t.item.uid, "name", f.value) : "spec" in t.item ? N.edit(t.item.uid, "spec.info.title", f.value) : V.edit(t.item.uid, "name", f.value), k.hide();
    }, C = () => {
      f.value = x(t.item) || "", k.show();
    }, ue = () => {
      "requestUid" in t.item ? (B.delete(t.item), R.value[I.Examples] === t.item.uid && F(`/workspace/${h.value}/request/default`)) : "summary" in t.item ? (O.delete(
        t.item,
        t.parentUids[t.parentUids.length - 1]
      ), R.value[I.Request] === t.item.uid && F(`/workspace/${h.value.uid}/request/default`)) : "spec" in t.item ? N.delete(t.item) : "name" in t.item && V.delete(
        t.item,
        t.parentUids[t.parentUids.length - 1]
      );
    }, me = n(() => {
      var e;
      return "summary" in t.item ? t.item.summary || "" : "name" in t.item ? t.item.name || "" : "spec" in t.item && ((e = t.item.spec.info) == null ? void 0 : e.title) || "";
    }), c = n(() => "requestUid" in t.item ? "Example" : "summary" in t.item ? "Request" : "spec" in t.item ? "Collection" : "Folder");
    return (e, s) => {
      const w = ve("RequestSidebarItem", !0);
      return m(), M(G, null, [
        l("div", {
          class: v(["relative flex flex-row", [
            i(a) && e.parentUids.length > 1 || !i(a) && e.parentUids.length ? "before:bg-b-3 before:absolute before:left-[calc(.75rem_+_.5px)] before:top-0 before:z-10 before:h-[calc(100%_+_.5px)] last:before:h-full before:w-[.5px] mb-[.5px] last:mb-0 indent-border-line-offset" : ""
          ]])
        }, [
          o(i(Se), {
            id: e.item.uid,
            ref_key: "draggableRef",
            ref: S,
            ceiling: W.value.ceiling,
            class: "flex flex-1 flex-col gap-[.5px] text-sm",
            floor: W.value.floor,
            isDraggable: e.isDraggable,
            isDroppable: e.isDroppable,
            parentIds: e.parentUids,
            onOnDragEnd: s[5] || (s[5] = (...p) => e.$emit("onDragEnd", ...p))
          }, {
            default: d(() => {
              var p, A;
              return [
                "summary" in e.item || "requestUid" in e.item ? (m(), b(i(Me), {
                  key: 0,
                  class: "no-underline",
                  to: ne()
                }, {
                  default: d(({ isExactActive: r }) => [
                    o(i(X), { disabled: i(a) }, {
                      trigger: d(() => [
                        l("div", {
                          class: v(["group relative flex min-h-8 cursor-pointer flex-row items-start justify-between gap-2 py-1.5 pr-2 rounded editable-sidebar-hover w-full", [
                            _,
                            r || oe.value ? "bg-sidebar-active-b text-sidebar-active-c transition-none" : "text-sidebar-c-2"
                          ]]),
                          tabindex: "0"
                        }, [
                          l("span", {
                            class: v(["z-10 font-medium w-full pl-2 word-break-break-word", {
                              "editable-sidebar-hover-item": !i(a)
                            }])
                          }, J(x(e.item)), 3),
                          l("div", ze, [
                            l("div", Ne, [
                              i(a) ? y("", !0) : (m(), b(q, {
                                key: 0,
                                item: e.item,
                                parentUids: e.parentUids,
                                resourceTitle: c.value,
                                onDelete: s[0] || (s[0] = (u) => i(U).show()),
                                onRename: C
                              }, null, 8, ["item", "parentUids", "resourceTitle"]))
                            ]),
                            l("span", Ve, [
                              D("   "),
                              o(i(Ie), {
                                class: "font-bold",
                                method: re.value
                              }, null, 8, ["method"])
                            ])
                          ])
                        ], 2)
                      ]),
                      content: d(() => [
                        o(q, {
                          item: e.item,
                          parentUids: e.parentUids,
                          resourceTitle: c.value,
                          static: "",
                          onDelete: s[1] || (s[1] = (u) => i(U).show()),
                          onRename: C
                        }, null, 8, ["item", "parentUids", "resourceTitle"])
                      ]),
                      _: 2
                    }, 1032, ["disabled"])
                  ]),
                  _: 1
                }, 8, ["to"])) : !i(a) || e.parentUids.length ? (m(), b(i(X), {
                  key: 1,
                  disabled: i(a) || ((A = (p = e.item.spec) == null ? void 0 : p.info) == null ? void 0 : A.title) === "Drafts"
                }, {
                  trigger: d(() => {
                    var r, u;
                    return [
                      l("button", {
                        class: v(["hover:bg-b-2 group relative flex w-full flex-row justify-start gap-1.5 rounded p-1.5 z-[1]", _]),
                        type: "button",
                        onClick: s[2] || (s[2] = (pe) => i(te)(e.item.uid))
                      }, [
                        l("span", Oe, [
                          be(e.$slots, "leftIcon", {}, () => [
                            l("div", {
                              class: v({
                                "rotate-90": i(j)[e.item.uid]
                              })
                            }, [
                              o(i($e), {
                                class: "text-c-3 text-sm",
                                icon: "ChevronRight",
                                size: "sm",
                                thickness: "2.5"
                              })
                            ], 2)
                          ], !0),
                          D("   ")
                        ]),
                        l("div", Be, [
                          l("span", {
                            class: v(["z-10 font-medium text-left w-full word-break-break-word", {
                              "editable-sidebar-hover-item": !i(a)
                            }])
                          }, J(x(e.item)), 3),
                          l("div", Fe, [
                            !i(a) && ((u = (r = e.item.spec) == null ? void 0 : r.info) == null ? void 0 : u.title) !== "Drafts" ? (m(), b(q, {
                              key: 0,
                              item: e.item,
                              parentUids: e.parentUids,
                              resourceTitle: c.value
                            }, null, 8, ["item", "parentUids", "resourceTitle"])) : y("", !0),
                            je
                          ])
                        ])
                      ])
                    ];
                  }),
                  content: d(() => {
                    var r, u;
                    return [
                      !i(a) && ((u = (r = e.item.spec) == null ? void 0 : r.info) == null ? void 0 : u.title) !== "Drafts" ? (m(), b(q, {
                        key: 0,
                        item: e.item,
                        parentUids: e.parentUids,
                        resourceTitle: c.value,
                        static: "",
                        onDelete: s[3] || (s[3] = (pe) => i(U).show()),
                        onRename: C
                      }, null, 8, ["item", "parentUids", "resourceTitle"])) : y("", !0)
                    ];
                  }),
                  default: d(() => [
                    D(" > ")
                  ]),
                  _: 3
                }, 8, ["disabled"])) : y("", !0),
                "childUids" in e.item ? ge((m(), M("div", Ke, [
                  (m(!0), M(G, null, he(L.value ? e.item.childUids.slice(1) : e.item.childUids, (r) => (m(), b(w, {
                    key: r,
                    isDraggable: !i($)[r],
                    isDroppable: de,
                    item: i(ee)[r] || i(z)[r] || i($)[r],
                    parentUids: [...e.parentUids, e.item.uid],
                    onOnDragEnd: s[4] || (s[4] = (...u) => e.$emit("onDragEnd", ...u))
                  }, null, 8, ["isDraggable", "item", "parentUids"]))), 128))
                ], 512)), [
                  [Ue, ae.value]
                ]) : y("", !0)
              ];
            }),
            _: 3
          }, 8, ["id", "ceiling", "floor", "isDraggable", "isDroppable", "parentIds"])
        ], 2),
        o(i(Y), {
          size: "sm",
          state: i(U),
          title: `Delete ${c.value}`
        }, {
          default: d(() => [
            o(qe, {
              variableName: me.value,
              warningMessage: "Warning: Deleting this will delete all items inside of this",
              onClose: s[6] || (s[6] = (p) => i(U).hide()),
              onDelete: ue
            }, null, 8, ["variableName"])
          ]),
          _: 1
        }, 8, ["state", "title"]),
        o(i(Y), {
          state: i(k),
          title: `Rename ${c.value}`
        }, {
          default: d(() => [
            o(i(xe), {
              modelValue: f.value,
              "onUpdate:modelValue": s[7] || (s[7] = (p) => f.value = p),
              label: c.value,
              labelShadowColor: "var(--scalar-background-1)",
              onKeydown: we(ye(P, ["prevent"]), ["enter"])
            }, null, 8, ["modelValue", "label", "onKeydown"]),
            l("div", Le, [
              o(i(Z), {
                class: "flex-1",
                variant: "outlined",
                onClick: s[8] || (s[8] = (p) => i(k).hide())
              }, {
                default: d(() => [
                  D(" Cancel ")
                ]),
                _: 1
              }),
              o(i(Z), {
                class: "flex-1",
                type: "submit",
                onClick: P
              }, {
                default: d(() => [
                  D(" Save ")
                ]),
                _: 1
              })
            ])
          ]),
          _: 1
        }, 8, ["state", "title"])
      ], 64);
    };
  }
});
export {
  _e as default
};
