// @scalar/api-client/dist/views/Request/RequestSidebarItemMenu.vue2.js
import { defineComponent as L, ref as P, watch as F, onMounted as K, onBeforeUnmount as j, computed as G, createElementBlock as w, openBlock as l, Fragment as C, createBlock as u, createCommentVNode as r, createVNode as i, unref as t, withCtx as m, withKeys as H, createElementVNode as I, createTextVNode as W, toDisplayString as J } from "vue";
import { useModal as $, ScalarFloating as O, ScalarDropdownMenu as Q, ScalarDropdownButton as k, ScalarIcon as v, ScalarModal as q } from "@scalar/components";
import { useRouter as X } from "vue-router";
import b from "../../components/Sidebar/Actions/DeleteSidebarListElement.vue.js";
import Y from "../../components/Sidebar/Actions/EditSidebarListCollection.vue.js";
import Z from "../../components/Sidebar/Actions/EditSidebarListElement.vue.js";
import { PathId as d } from "../../routes.js";
import { useActiveEntities as _ } from "../../store/active-entities.js";
import { createInitialRequest as ee } from "../../store/requests.js";
import { useWorkspace as te } from "../../store/store.js";
const ce = /* @__PURE__ */ L({
  __name: "RequestSidebarItemMenu",
  props: {
    menuItem: {}
  },
  emits: ["closeMenu", "toggleWatchMode", "clearDrafts"],
  setup(n, { emit: z }) {
    const s = n, M = z, { replace: g } = X(), {
      activeRouterParams: E,
      activeWorkspaceCollections: y,
      activeWorkspaceRequests: N
    } = _(), { events: S, requestMutators: T } = te(), c = $(), x = $(), p = $(), A = () => S.commandPalette.emit({
      commandName: "Add Example",
      metaData: {
        itemUid: s.menuItem.item?.entity.uid
      }
    }), R = (a, e) => {
      s.menuItem.item?.edit(a, e), c.hide();
    }, B = () => {
      if (s.menuItem.item?.delete(), !N.value.length) {
        const { request: a } = ee(), e = y.value.find(
          (o) => o.info?.title === "Drafts"
        );
        e && (T.add(a, e.uid), g({
          name: "request",
          params: {
            [d.Request]: a.uid
          }
        }));
      }
      if (E.value[d.Request] === s.menuItem.item?.entity.uid && g({
        name: "request",
        params: {
          [d.Request]: "default"
        }
      }), E.value[d.Examples] === s.menuItem.item?.entity.uid && g({
        name: "request",
        params: {
          [d.Request]: "default"
        }
      }), y.value[0]) {
        const a = y.value[0].requests[0];
        g({
          name: "request",
          params: {
            [d.Request]: a
          }
        });
      }
      x.hide();
    }, f = P(null);
    F([() => s.menuItem.open, f], ([a]) => {
      a && f.value?.$parent?.$el && f.value.$parent.$el.focus();
    });
    const h = () => s.menuItem.open && M("closeMenu");
    K(() => window.addEventListener("click", h)), j(() => window.removeEventListener("click", h));
    const U = () => {
      M("toggleWatchMode", s.menuItem.item);
    }, V = () => {
      M("clearDrafts"), p.hide();
    }, D = G(() => s.menuItem.item?.title === "Drafts");
    return (a, e) => (l(), w(C, null, [
      n.menuItem.targetRef && n.menuItem.open ? (l(), u(t(O), {
        key: 0,
        placement: "right-start",
        target: n.menuItem.targetRef,
        teleport: ""
      }, {
        floating: m(() => [
          i(t(Q), {
            onKeydown: e[3] || (e[3] = H((o) => a.$emit("closeMenu"), ["escape"]))
          }, {
            default: m(() => [
              n.menuItem.item?.entity.type === "request" ? (l(), u(t(k), {
                key: 0,
                class: "flex items-center gap-2",
                onClick: A
              }, {
                default: m(() => [
                  i(t(v), {
                    class: "inline-flex",
                    icon: "Example",
                    size: "md",
                    thickness: "1.5"
                  }),
                  e[8] || (e[8] = I("span", null, "Add Example", -1))
                ]),
                _: 1
              })) : r("", !0),
              D.value ? r("", !0) : (l(), u(t(k), {
                key: 1,
                ref_key: "menuRef",
                ref: f,
                class: "flex items-center gap-2",
                onClick: e[0] || (e[0] = (o) => t(c).show())
              }, {
                default: m(() => [
                  i(t(v), {
                    class: "inline-flex",
                    icon: "Edit",
                    size: "md",
                    thickness: "1.5"
                  }),
                  I("span", null, [
                    n.menuItem.item?.entity.type === "collection" ? (l(), w(C, { key: 0 }, [
                      W(" Edit ")
                    ], 64)) : (l(), w(C, { key: 1 }, [
                      W(" Rename ")
                    ], 64))
                  ])
                ]),
                _: 1
              }, 512)),
              n.menuItem.item?.documentUrl ? (l(), u(t(k), {
                key: 2,
                ref_key: "menuRef",
                ref: f,
                class: "flex items-center gap-2",
                onClick: U
              }, {
                default: m(() => [
                  i(t(v), {
                    class: "inline-flex",
                    icon: n.menuItem.item?.watchMode ? "Unwatch" : "Watch",
                    size: "md",
                    thickness: "1.5"
                  }, null, 8, ["icon"]),
                  I("span", null, J(n.menuItem.item?.watchMode ? "Disable Watch Mode" : "Enable Watch Mode"), 1)
                ]),
                _: 1
              }, 512)) : r("", !0),
              D.value ? r("", !0) : (l(), u(t(k), {
                key: 3,
                class: "flex items-center gap-2",
                onClick: e[1] || (e[1] = (o) => t(x).show())
              }, {
                default: m(() => [
                  i(t(v), {
                    class: "inline-flex",
                    icon: "Delete",
                    size: "md",
                    thickness: "1.5"
                  }),
                  e[9] || (e[9] = I("span", null, "Delete", -1))
                ]),
                _: 1
              })),
              D.value ? (l(), u(t(k), {
                key: 4,
                class: "flex items-center gap-2",
                onClick: e[2] || (e[2] = (o) => t(p).show())
              }, {
                default: m(() => [
                  i(t(v), {
                    class: "inline-flex",
                    icon: "Delete",
                    size: "md",
                    thickness: "1.5"
                  }),
                  e[10] || (e[10] = I("span", null, "Clear Drafts", -1))
                ]),
                _: 1
              })) : r("", !0)
            ]),
            _: 1
          })
        ]),
        _: 1
      }, 8, ["target"])) : r("", !0),
      i(t(q), {
        size: "xxs",
        state: t(x),
        title: `Delete ${n.menuItem.item?.resourceTitle}`
      }, {
        default: m(() => [
          i(b, {
            variableName: n.menuItem.item?.title ?? "",
            warningMessage: n.menuItem.item?.warning,
            onClose: e[4] || (e[4] = (o) => t(x).hide()),
            onDelete: B
          }, null, 8, ["variableName", "warningMessage"])
        ]),
        _: 1
      }, 8, ["state", "title"]),
      i(t(q), {
        size: "xxs",
        state: t(c),
        title: `Edit ${n.menuItem.item?.resourceTitle}`
      }, {
        default: m(() => [
          n.menuItem.item?.resourceTitle === "Collection" ? (l(), u(Y, {
            key: 0,
            icon: n.menuItem.item?.icon || "interface-content-folder",
            name: n.menuItem.item?.title,
            onClose: e[5] || (e[5] = (o) => t(c).hide()),
            onEdit: R
          }, null, 8, ["icon", "name"])) : (l(), u(Z, {
            key: 1,
            name: n.menuItem.item?.title ?? "",
            onClose: e[6] || (e[6] = (o) => t(c).hide()),
            onEdit: R
          }, null, 8, ["name"]))
        ]),
        _: 1
      }, 8, ["state", "title"]),
      i(t(q), {
        size: "xxs",
        state: t(p),
        title: "Clear Drafts"
      }, {
        default: m(() => [
          i(b, {
            variableName: "All Drafts",
            warningMessage: "This action will clear all drafts. This cannot be undone.",
            onClose: e[7] || (e[7] = (o) => t(p).hide()),
            onDelete: V
          })
        ]),
        _: 1
      }, 8, ["state"])
    ], 64));
  }
});
export {
  ce as default
};
