// @scalar/api-client/dist/views/Request/Request.vue2.js
import { defineComponent as se, ref as ie, watch as P, onMounted as ae, onBeforeUnmount as le, openBlock as h, createElementBlock as k, Fragment as F, createElementVNode as v, normalizeClass as b, unref as s, createVNode as c, createCommentVNode as U, withCtx as p, withDirectives as ne, createSlots as de, withModifiers as N, renderList as ce, createBlock as R, createTextVNode as ue, vShow as me, pushScopeId as fe, popScopeId as pe } from "vue";
import ve from "../../components/AddressBar/AddressBar.vue.js";
import he from "../../components/Search/SearchButton.vue.js";
import be from "../../components/Search/SearchModal.vue.js";
import xe from "../../components/Sidebar/SidebarButton.vue.js";
import ge from "../../components/Sidebar/SidebarToggle.vue.js";
import we from "../../components/ViewLayout/ViewLayout.vue.js";
import ye from "../../components/ViewLayout/ViewLayoutContent.vue.js";
import { useWorkspace as Se } from "../../store/workspace.js";
import _e from "./RequestSection/RequestSection.vue.js";
/* empty css                                   */
import ke from "./ResponseSection/ResponseSection.vue.js";
import { useModal as Ue, ScalarIcon as C } from "@scalar/components";
import { useToasts as Re } from "@scalar/use-toasts";
import { useMediaQuery as Ce } from "@vueuse/core";
import Be from "./RequestSidebarItem.vue.js";
import De from "../../components/Sidebar/Sidebar.vue.js";
/* empty css                                    */
import Oe from "./components/WorkspaceDropdown.vue.js";
import { useSidebar as qe } from "../../hooks/useSidebar.js";
import { sendRequest as Me } from "../../libs/sendRequest.js";
import { executeRequestBus as W } from "../../libs/event-busses/execute-request-bus.js";
import { hotKeyBus as z } from "../../libs/event-busses/hot-keys-bus.js";
import { commandPaletteBus as j } from "../../libs/event-busses/command-palette-bus.js";
import { requestStatusBus as S } from "../../libs/event-busses/request-status-bus.js";
const Ie = (w) => (fe("data-v-b7bd40a0"), w = w(), pe(), w), Ve = { class: "lg:min-h-header flex items-center w-full justify-center p-1 flex-wrap t-app__top-container border-b-1/2" }, $e = { class: "flex flex-row items-center gap-1 lg:px-1 lg:mb-0 mb-0.5 lg:flex-1 w-6/12" }, Ee = /* @__PURE__ */ Ie(() => /* @__PURE__ */ v("a", {
  class: "text-c-2 text-sm font-medium gitbook-show ml-.5 hover:text-c-1 border p-1 rounded hover:bg-b-3",
  href: "https://scalar.com/",
  target: "_blank"
}, " Powered by Scalar.com ", -1)), Pe = { class: "flex flex-row items-center gap-1 lg:px-1 lg:mb-0 mb-0.5 lg:flex-1 justify-end w-6/12" }, dt = /* @__PURE__ */ se({
  __name: "Request",
  setup(w) {
    var E;
    const {
      activeExample: y,
      activeRequest: f,
      activeSecuritySchemes: L,
      activeWorkspace: l,
      activeWorkspaceCollections: T,
      activeWorkspaceServers: A,
      activeWorkspaceRequests: K,
      collectionMutators: B,
      collections: m,
      cookies: H,
      environments: J,
      findRequestFolders: Q,
      folders: x,
      folderMutators: D,
      modalState: G,
      requestMutators: X,
      workspaceMutators: O
    } = Se(), { collapsedSidebarFolders: Y, setCollapsedSidebarFolder: Z } = qe(), { toast: q } = Re(), M = Ue(), ee = Ce("(max-width: 780px)"), u = ie(!((E = l.value) != null && E.isReadOnly));
    P(ee, (o) => u.value = !o), P(
      f,
      (o) => {
        o && Q(o.uid).forEach(
          (e) => Z(e, !0)
        );
      },
      { immediate: !0 }
    );
    const I = async () => {
      var i;
      if (!f.value || !y.value) {
        console.warn(
          "There is no request active at the moment. Please select one then try again."
        );
        return;
      }
      let o = y.value.url;
      const e = Object.values(J).reduce(
        (r, a) => {
          try {
            return { ...r, ...JSON.parse(a.raw) };
          } catch {
            return r;
          }
        },
        {}
      ), t = /\{\{(.*?)\}\}/g;
      o = o.replace(t, (r, a) => (A.value.forEach((n) => {
        if (n.url === a)
          return a;
      }), e[a] || a)), S.emit("start");
      try {
        const { request: r, response: a, error: n } = await Me(
          f.value,
          y.value,
          o,
          L.value,
          (i = l.value) == null ? void 0 : i.proxyUrl,
          H
        );
        r && a ? (X.edit(f.value.uid, "history", [
          ...f.value.history,
          {
            request: r,
            response: a,
            timestamp: Date.now()
          }
        ]), S.emit("stop")) : (q((n == null ? void 0 : n.message) ?? "Send Request Failed", "error"), S.emit("abort"));
      } catch (r) {
        console.error(r), q(`Oops! 
${r}`, "error"), S.emit("abort");
      }
    }, V = (o) => {
      o.toggleSidebar && (u.value = !u.value), o.openCommandPalette && (o.openCommandPalette.preventDefault(), j.emit());
    };
    ae(() => {
      W.on(I), z.on(V);
    }), le(() => {
      W.off(I), z.off(V);
    });
    const $ = (o, e) => {
      m[o] ? B.edit(o, "childUids", e) : x[o] && D.edit(o, "childUids", e);
    }, te = (o, e) => {
      if (!o || !e) return;
      const { id: t, parentId: i } = o, { id: r, parentId: a, offset: n } = e;
      if (i ? m[i] ? B.edit(
        i,
        "childUids",
        m[i].childUids.filter(
          (d) => d !== t
        )
      ) : x[i] && D.edit(
        i,
        "childUids",
        x[i].childUids.filter((d) => d !== t)
      ) : O.edit(
        l.value.uid,
        "collectionUids",
        l.value.collectionUids.filter((d) => d !== t)
      ), n === 2) {
        const d = m[r] || x[r];
        $(r, [...d.childUids, t]);
      } else if (a) {
        const g = [...(m[a] || x[a]).childUids], _ = g.findIndex((re) => r === re) ?? 0;
        g.splice(_ + n, 0, t), $(a, g);
      } else {
        const d = [...l.value.collectionUids], g = d.findIndex((_) => r === _) ?? 0;
        d.splice(g + n, 0, t), O.edit(
          l.value.uid,
          "collectionUids",
          d
        );
      }
    }, oe = (o, e) => {
      var t, i, r;
      return !(l.value.isReadOnly || !m[o.id] && e.offset !== 2 || m[o.id] && ((r = (i = (t = m[e.id]) == null ? void 0 : t.spec) == null ? void 0 : i.info) == null ? void 0 : r.title) === "Drafts");
    };
    return (o, e) => (h(), k(F, null, [
      v("div", {
        class: b(["flex flex-1 flex-col rounded pt-0 h-full bg-b-1 relative border-1/2 rounded mr-1.5 mb-1.5 overflow-hidden", {
          "!mr-0 !mb-0 !border-0": s(l).isReadOnly
        }])
      }, [
        v("div", Ve, [
          v("div", $e, [
            c(ge, {
              modelValue: u.value,
              "onUpdate:modelValue": e[0] || (e[0] = (t) => u.value = t),
              class: "gitbook-hidden"
            }, null, 8, ["modelValue"]),
            Ee
          ]),
          c(ve),
          v("div", Pe, [
            s(l).isReadOnly ? (h(), k("button", {
              key: 0,
              class: "text-c-3 hover:bg-b-2 active:text-c-1 p-2 rounded",
              type: "button",
              onClick: e[1] || (e[1] = (t) => s(G).hide())
            }, [
              c(s(C), {
                icon: "Close",
                size: "lg",
                thickness: "1.75"
              })
            ])) : U("", !0)
          ])
        ]),
        c(we, null, {
          default: p(() => [
            ne(c(s(De), {
              class: b([u.value ? "sidebar-active-width" : ""]),
              showSideBar: u.value,
              "onUpdate:showSideBar": e[5] || (e[5] = (t) => u.value = t)
            }, de({
              content: p(() => [
                c(he, {
                  onOpenSearchModal: e[2] || (e[2] = (t) => s(M).show())
                }),
                v("div", {
                  class: b(["custom-scroll flex flex-1 flex-col overflow-visible px-3 pb-3 pt-0", {
                    "pb-14": !s(l).isReadOnly
                  }]),
                  onDragenter: e[3] || (e[3] = N(() => {
                  }, ["prevent"])),
                  onDragover: e[4] || (e[4] = N(() => {
                  }, ["prevent"]))
                }, [
                  (h(!0), k(F, null, ce(s(T), (t) => {
                    var i, r;
                    return h(), R(Be, {
                      key: t.uid,
                      isDraggable: !s(l).isReadOnly && ((r = (i = t.spec) == null ? void 0 : i.info) == null ? void 0 : r.title) !== "Drafts",
                      isDroppable: oe,
                      item: t,
                      parentUids: [],
                      onOnDragEnd: te
                    }, {
                      leftIcon: p(() => [
                        c(s(C), {
                          class: "text-sidebar-c-2 text-sm group-hover:hidden",
                          icon: "CodeFolder",
                          size: "sm",
                          thickness: "2"
                        }),
                        v("div", {
                          class: b({
                            "rotate-90": s(Y)[t.uid]
                          })
                        }, [
                          c(s(C), {
                            class: "text-c-3 hidden text-sm group-hover:block",
                            icon: "ChevronRight",
                            size: "sm"
                          })
                        ], 2)
                      ]),
                      _: 2
                    }, 1032, ["isDraggable", "item"]);
                  }), 128))
                ], 34)
              ]),
              button: p(() => [
                s(l).isReadOnly ? U("", !0) : (h(), R(xe, {
                  key: 0,
                  class: b({
                    "empty-sidebar-item": s(K).length === 1
                  }),
                  click: s(j).emit
                }, {
                  title: p(() => [
                    ue("Add Item")
                  ]),
                  _: 1
                }, 8, ["class", "click"]))
              ]),
              _: 2
            }, [
              s(l).isReadOnly ? void 0 : {
                name: "header",
                fn: p(() => [
                  c(s(Oe))
                ]),
                key: "0"
              }
            ]), 1032, ["class", "showSideBar"]), [
              [me, u.value]
            ]),
            s(y) ? (h(), R(ye, {
              key: 0,
              class: b(["flex-1", [u.value ? "sidebar-active-hide-layout" : ""]])
            }, {
              default: p(() => {
                var t, i, r, a, n;
                return [
                  c(_e),
                  c(ke, {
                    response: (n = (a = (t = s(f)) == null ? void 0 : t.history) == null ? void 0 : a[((r = (i = s(f)) == null ? void 0 : i.history) == null ? void 0 : r.length) - 1]) == null ? void 0 : n.response
                  }, null, 8, ["response"])
                ];
              }),
              _: 1
            }, 8, ["class"])) : U("", !0)
          ]),
          _: 1
        })
      ], 2),
      c(be, { modalState: s(M) }, null, 8, ["modalState"])
    ], 64));
  }
});
export {
  dt as default
};
