// @scalar/api-client/dist/views/Request/RequestSidebarItemMenu.vue2.js
import { defineComponent as _, computed as g, openBlock as u, createBlock as f, unref as e, withCtx as o, createVNode as t, createCommentVNode as C, pushScopeId as S, popScopeId as v, createElementVNode as c } from "vue";
import { ScalarDropdown as y, ScalarDropdownItem as r, ScalarIcon as i, ScalarButton as z } from "@scalar/components";
import { commandPaletteBus as E } from "../../libs/event-busses/command-palette-bus.js";
const m = (a) => (S("data-v-5686fba5"), a = a(), v(), a), I = /* @__PURE__ */ m(() => /* @__PURE__ */ c("span", null, "Add Example", -1)), b = /* @__PURE__ */ m(() => /* @__PURE__ */ c("span", null, "Rename", -1)), B = /* @__PURE__ */ m(() => /* @__PURE__ */ c("span", null, "Delete", -1)), N = /* @__PURE__ */ _({
  __name: "RequestSidebarItemMenu",
  props: {
    item: {},
    resourceTitle: {},
    static: { type: Boolean, default: !1 }
  },
  emits: ["delete", "rename"],
  setup(a, { emit: x }) {
    const l = a, d = x, h = () => E.emit({
      commandName: "Add Example",
      metaData: l.item.uid
    }), k = g(() => "summary" in l.item);
    return (p, s) => (u(), f(e(y), {
      static: p.static,
      teleport: !p.static
    }, {
      items: o(() => [
        k.value ? (u(), f(e(r), {
          key: 0,
          class: "flex !gap-2",
          onClick: h
        }, {
          default: o(() => [
            t(e(i), {
              class: "inline-flex",
              icon: "Add",
              size: "sm",
              thickness: "1.5"
            }),
            I
          ]),
          _: 1
        })) : C("", !0),
        t(e(r), {
          class: "flex !gap-2",
          onClick: s[1] || (s[1] = (n) => d("rename"))
        }, {
          default: o(() => [
            t(e(i), {
              class: "inline-flex",
              icon: "Edit",
              size: "sm",
              thickness: "1.5"
            }),
            b
          ]),
          _: 1
        }),
        t(e(r), {
          class: "flex !gap-2",
          onClick: s[2] || (s[2] = (n) => d("delete"))
        }, {
          default: o(() => [
            t(e(i), {
              class: "inline-flex",
              icon: "Trash",
              size: "sm",
              thickness: "1.5"
            }),
            B
          ]),
          _: 1
        })
      ]),
      default: o(() => [
        t(e(z), {
          class: "px-0.5 py-0 z-10 hover:bg-b-3 hidden group-hover:flex ui-open:flex absolute -translate-y-1/2 right-0 aspect-square inset-y-2/4 h-fit",
          size: "sm",
          variant: "ghost",
          onClick: s[0] || (s[0] = (n) => {
            (l.resourceTitle === "Collection" || l.resourceTitle === "Folder") && n.stopPropagation();
          })
        }, {
          default: o(() => [
            t(e(i), {
              icon: "Ellipses",
              size: "sm"
            })
          ]),
          _: 1
        })
      ]),
      _: 1
    }, 8, ["static", "teleport"]));
  }
});
export {
  N as default
};
