// @scalar/api-client/dist/v2/blocks/scalar-address-bar-block/components/AddressBar.vue2.js
import { defineComponent as P, useId as K, computed as x, ref as v, useTemplateRef as E, onMounted as Q, onBeforeUnmount as j, createElementBlock as $, openBlock as b, unref as s, createElementVNode as l, normalizeClass as V, createVNode as a, createCommentVNode as q, normalizeStyle as G, createBlock as J, withCtx as T, createTextVNode as g, toDisplayString as f } from "vue";
import { ScalarButton as z, ScalarWrappingText as X, ScalarIcon as Y } from "@scalar/components";
import { REQUEST_METHODS as Z } from "@scalar/helpers/http/http-info";
import { ScalarIconCopy as _, ScalarIconWarningCircle as ee } from "@scalar/icons";
import { useClipboard as te } from "@scalar/use-hooks/useClipboard";
import { getResolvedUrl as ne } from "../../operation-block/helpers/get-resolved-url.js";
import { useLoadingAnimation as oe } from "../hooks/use-loading-animation.js";
import le from "./AddressBarHistory.vue.js";
import se from "../../../../components/HttpMethod/HttpMethod.vue.js";
import ae from "../../../components/server/ServerDropdown.vue.js";
import re from "../../../components/code-input/CodeInput.vue.js";
const ie = ["id"], de = { class: "flex gap-1" }, ue = { class: "scroll-timeline-x scroll-timeline-x-hidden relative flex w-full bg-blend-normal" }, me = {
  key: 0,
  class: "absolute inset-x-0 top-[calc(100%+4px)] flex flex-col items-center rounded px-6"
}, ve = { class: "text-c-danger bg-b-danger border-c-danger flex items-center gap-1 rounded border p-1" }, fe = { class: "min-w-0 flex-1" }, ce = {
  "aria-hidden": "true",
  class: "inline-flex items-center gap-1"
}, he = { class: "sr-only" }, Re = /* @__PURE__ */ P({
  __name: "AddressBar",
  props: {
    path: {},
    method: {},
    server: {},
    servers: {},
    history: {},
    layout: {},
    eventBus: {},
    environment: {}
  },
  emits: ["execute", "update:servers", "select:history:item"],
  setup(e, { expose: D, emit: I }) {
    const d = I, c = K(), { percentage: L, startLoading: y, stopLoading: h, isLoading: A } = oe(), M = x(() => ({
      backgroundColor: `color-mix(in srgb, transparent 90%, ${Z[e.method].colorVar})`,
      transform: `translate3d(-${L.value}%,0,0)`
    })), i = v(null), r = v(null), B = x(() => r.value || i.value), p = (n, t, o) => {
      const W = u.value?.cursorPosition();
      e.eventBus.emit(
        "operation:update:pathMethod",
        {
          meta: { method: e.method, path: e.path },
          payload: { method: n, path: t },
          callback: (m) => {
            (m === "success" || m === "no-change") && (r.value = null, i.value = null), m === "success" ? e.eventBus.emit("ui:focus:address-bar", { position: W }) : m === "conflict" && (n !== e.method && (r.value = n), t !== e.path && (i.value = t));
          }
        },
        o
      );
    }, O = (n) => p(n, i.value ?? e.path), H = (n) => {
      const t = n.startsWith("/") ? n : `/${n}`;
      p(r.value ?? e.method, t, {
        debounceKey: `operation:update:pathMethod-${e.path}-${e.method}`
      });
    }, C = E("sendButtonRef"), u = E("addressBarRef"), w = () => C.value?.$el?.focus(), k = (n) => {
      if (u.value?.isFocused && e.layout !== "desktop")
        return;
      const t = n && "position" in n ? n.position : "end";
      u.value?.focus(t), n && "event" in n && n.event.preventDefault();
    };
    Q(() => {
      e.eventBus.on("ui:focus:address-bar", k), e.eventBus.on("ui:focus:send-button", w), e.eventBus.on("hooks:on:request:sent", y), e.eventBus.on("hooks:on:request:complete", h);
    }), j(() => {
      e.eventBus.off("ui:focus:address-bar", k), e.eventBus.off("ui:focus:send-button", w), e.eventBus.off("hooks:on:request:sent", y), e.eventBus.off("hooks:on:request:complete", h), h();
    });
    const { copyToClipboard: N } = te(), F = async () => {
      await N(
        ne({ environment: e.environment, server: e.server, path: e.path, pathVariables: {} })
      );
    }, S = v(!1), U = v(!1), R = x(
      () => S.value || U.value
    );
    return D({
      methodConflict: r,
      pathConflict: i
    }), (n, t) => (b(), $("div", {
      id: s(c),
      class: "scalar-address-bar order-last flex h-(--scalar-address-bar-height) w-full [--scalar-address-bar-height:32px] lg:order-0 lg:w-auto"
    }, [
      l("div", {
        class: V(["address-bar-bg-states text-xxs group relative order-last flex w-full max-w-[calc(100dvw-24px)] flex-1 flex-row items-stretch rounded-lg p-0.75 lg:order-none lg:max-w-[580px] lg:min-w-[580px] xl:max-w-[720px] xl:min-w-[720px]", {
          "outline-c-danger outline": B.value,
          "rounded-b-none": R.value
        }])
      }, [
        l("div", {
          class: V(["pointer-events-none absolute top-0 left-0 block h-full w-full overflow-hidden rounded-lg border", {
            "rounded-b-none": R.value
          }])
        }, [
          l("div", {
            class: "absolute top-0 left-0 h-full w-full",
            style: G(M.value)
          }, null, 4)
        ], 2),
        l("div", de, [
          a(s(se), {
            isEditable: e.layout !== "modal",
            isSquare: "",
            method: r.value ?? e.method,
            teleport: "",
            onChange: O
          }, null, 8, ["isEditable", "method"])
        ]),
        l("div", ue, [
          e.servers.length ? (b(), J(s(ae), {
            key: 0,
            layout: e.layout,
            server: e.server,
            servers: e.servers,
            target: s(c),
            "onUpdate:open": t[0] || (t[0] = (o) => S.value = o),
            "onUpdate:selectedServer": t[1] || (t[1] = (o) => e.eventBus.emit("server:update:selected", o)),
            "onUpdate:servers": t[2] || (t[2] = (o) => d("update:servers")),
            "onUpdate:variable": t[3] || (t[3] = (o) => e.eventBus.emit("server:update:variables", o))
          }, null, 8, ["layout", "server", "servers", "target"])) : q("", !0),
          t[8] || (t[8] = l("div", { class: "fade-left" }, null, -1)),
          a(s(re), {
            ref_key: "addressBarRef",
            ref: u,
            alwaysEmitChange: "",
            "aria-label": "Path",
            class: "min-w-fit outline-none",
            disableCloseBrackets: "",
            disabled: e.layout === "modal",
            disableEnter: "",
            disableTabIndent: "",
            emitOnBlur: !1,
            environment: e.environment,
            importCurl: "",
            layout: e.layout,
            modelValue: e.path,
            placeholder: e.server ? "" : "Enter a URL",
            server: "",
            onSubmit: t[4] || (t[4] = (o) => d("execute")),
            "onUpdate:modelValue": H
          }, null, 8, ["disabled", "environment", "layout", "modelValue", "placeholder"]),
          t[9] || (t[9] = l("div", { class: "fade-right" }, null, -1))
        ]),
        a(s(z), {
          class: "hover:bg-b-3 mx-1",
          size: "xs",
          variant: "ghost",
          onClick: F
        }, {
          default: T(() => [
            a(s(_)),
            t[10] || (t[10] = l("span", { class: "sr-only" }, "Copy URL", -1))
          ]),
          _: 1
        }),
        a(le, {
          history: e.history,
          target: s(c),
          "onSelect:history:item": t[5] || (t[5] = (o) => d("select:history:item", o)),
          "onUpdate:open": t[6] || (t[6] = (o) => U.value = o)
        }, null, 8, ["history", "target"]),
        B.value ? (b(), $("div", me, [
          l("div", ve, [
            a(s(ee), { size: "sm" }),
            l("div", fe, [
              t[11] || (t[11] = g(" A ", -1)),
              l("em", null, f(r.value?.toUpperCase() ?? e.method.toUpperCase()), 1),
              t[12] || (t[12] = g(" request to ", -1)),
              a(s(X), {
                text: i.value ?? e.path
              }, null, 8, ["text"]),
              t[13] || (t[13] = g(" already exists in this document ", -1))
            ])
          ])
        ])) : q("", !0),
        a(s(z), {
          ref_key: "sendButtonRef",
          ref: C,
          class: "relative h-auto shrink-0 overflow-hidden py-1 pr-2.5 pl-2 font-bold",
          disabled: s(A),
          onClick: t[7] || (t[7] = (o) => d("execute"))
        }, {
          default: T(() => [
            l("span", ce, [
              a(s(Y), {
                class: "relative shrink-0 fill-current",
                icon: "Play",
                size: "xs"
              }),
              t[14] || (t[14] = l("span", { class: "text-xxs hidden lg:flex" }, "Send", -1))
            ]),
            l("span", he, " Send " + f(e.method) + " request to " + f(e.server?.url ?? "") + f(e.path), 1)
          ]),
          _: 1
        }, 8, ["disabled"])
      ], 2)
    ], 8, ie));
  }
});
export {
  Re as default
};
