// @scalar/api-client/dist/v2/features/collection/components/Authentication.vue2.js
import { defineComponent as u, computed as s, createElementBlock as m, openBlock as d, createElementVNode as n, createVNode as c, unref as i, normalizeClass as a } from "vue";
import { ScalarToggle as v } from "@scalar/components";
import y from "../../../blocks/scalar-auth-selector-block/components/AuthSelector.vue.js";
const f = { class: "flex flex-col gap-4" }, p = { class: "flex flex-col gap-2" }, h = { class: "flex h-8 items-center justify-between" }, b = /* @__PURE__ */ u({
  __name: "Authentication",
  props: {
    documentSlug: {},
    document: {},
    eventBus: {},
    layout: {},
    path: {},
    method: {},
    exampleName: {},
    environment: {},
    securitySchemes: {},
    workspaceStore: {},
    activeWorkspace: {},
    plugins: {},
    collectionType: {}
  },
  setup(e) {
    const l = s(
      () => e.document?.["x-scalar-set-operation-security"] ?? !1
    ), r = s(
      () => e.document?.servers?.find(
        ({ url: o }) => o === e.document?.["x-scalar-selected-server"]
      ) ?? null
    );
    return (o, t) => (d(), m("div", f, [
      n("div", p, [
        n("div", h, [
          t[1] || (t[1] = n("h3", { class: "font-bold" }, "Authentication", -1)),
          c(i(v), {
            class: "w-4",
            modelValue: l.value,
            "onUpdate:modelValue": t[0] || (t[0] = () => e.eventBus.emit("document:toggle:security"))
          }, null, 8, ["modelValue"])
        ]),
        t[2] || (t[2] = n("p", { class: "pr-6 text-sm" }, " If enabled, all selected authentication will apply to all operations in this document. You can override this by disabling the toggle and authentication will then be applied at the operation level. ", -1))
      ]),
      n("div", {
        class: a(!l.value && "cursor-not-allowed")
      }, [
        c(i(y), {
          class: a([
            "scalar-collection-auth !border-none",
            !l.value && "pointer-events-none opacity-50 mix-blend-luminosity"
          ]),
          environment: e.environment,
          eventBus: e.eventBus,
          isStatic: "",
          meta: { type: "document" },
          proxyUrl: e.workspaceStore.workspace["x-scalar-active-proxy"] ?? "",
          securityRequirements: e.document?.security ?? [],
          securitySchemes: e.securitySchemes,
          selectedSecurity: e.document?.["x-scalar-selected-security"],
          server: r.value,
          title: "Authentication"
        }, null, 8, ["class", "environment", "eventBus", "proxyUrl", "securityRequirements", "securitySchemes", "selectedSecurity", "server"])
      ], 2)
    ]));
  }
});
export {
  b as default
};
