// @scalar/api-client/dist/v2/features/collection/components/Cookies.vue.js
import { defineComponent as s, computed as c, createElementBlock as n, openBlock as l, createStaticVNode as a, createVNode as i, unref as r } from "vue";
import u from "../../global-cookies/components/CookiesTable.vue.js";
const d = { class: "flex flex-col gap-4" }, f = /* @__PURE__ */ s({
  __name: "Cookies",
  props: {
    documentSlug: {},
    document: {},
    eventBus: {},
    layout: {},
    path: {},
    method: {},
    exampleName: {},
    environment: {},
    securitySchemes: {},
    workspaceStore: {},
    activeWorkspace: {},
    plugins: {},
    collectionType: {}
  },
  setup(e) {
    const t = c(() => (e.collectionType === "document" ? e.document["x-scalar-cookies"] : e.workspaceStore.workspace["x-scalar-cookies"]) ?? []);
    return (m, o) => (l(), n("div", d, [
      o[0] || (o[0] = a('<div class="flex items-start justify-between gap-2"><div class="flex flex-col gap-2"><div class="flex h-8 items-center"><h3 class="font-bold">Cookies</h3></div><p class="text-c-2 mb-4 text-sm"> Manage your collection&#39;s cookies here.<br>Cookies allow you to store and send key-value data with your API requests—often used for things like session tokens, authentication, and saving user preferences.<br></p></div></div>', 1)),
      i(r(u), {
        cookies: t.value,
        eventBus: e.eventBus,
        collectionType: e.collectionType
      }, null, 8, ["cookies", "eventBus", "collectionType"])
    ]));
  }
});
export {
  f as default
};
