// @scalar/api-client/dist/v2/features/collection/components/Environment.vue.js
import { defineComponent as o, computed as c, createElementBlock as s, openBlock as l, createStaticVNode as a, createVNode as i, unref as r } from "vue";
import m from "../../environments/EnvironmentsList.vue.js";
const v = { class: "flex flex-col gap-4" }, f = /* @__PURE__ */ o({
  __name: "Environment",
  props: {
    documentSlug: {},
    document: {},
    eventBus: {},
    layout: {},
    path: {},
    method: {},
    exampleName: {},
    environment: {},
    securitySchemes: {},
    workspaceStore: {},
    activeWorkspace: {},
    plugins: {},
    collectionType: {}
  },
  setup(e) {
    const n = c(
      () => (e.collectionType === "document" ? e.document["x-scalar-environments"] : e.workspaceStore.workspace["x-scalar-environments"]) ?? {}
    );
    return (d, t) => (l(), s("div", v, [
      t[0] || (t[0] = a('<div class="flex items-start justify-between gap-2"><div class="flex flex-col gap-2"><div class="flex h-8 items-center"><h3 class="font-bold">Environment Variables</h3></div><p class="text-c-2 mb-4 text-sm"> Set environment variables at your collection level. Use <code class="font-code text-c-2"> {{ variable }} </code> to add / search among the selected environment&#39;s variables in your request inputs. </p></div></div>', 1)),
      i(r(m), {
        collectionType: e.collectionType,
        environments: n.value,
        eventBus: e.eventBus
      }, null, 8, ["collectionType", "environments", "eventBus"])
    ]));
  }
});
export {
  f as default
};
