// @scalar/api-client/dist/v2/features/collection/components/Overview.vue2.js
import { defineComponent as k, computed as x, ref as y, useTemplateRef as w, createElementBlock as i, openBlock as t, createElementVNode as o, createBlock as m, createCommentVNode as h, unref as n, withCtx as v, createVNode as r, Fragment as f, nextTick as B } from "vue";
import { ScalarButton as b, ScalarMarkdown as C } from "@scalar/components";
import { ScalarIconPencil as g } from "@scalar/icons";
import I from "../../../components/code-input/CodeInput.vue.js";
const S = { class: "flex flex-col gap-2" }, V = { class: "flex items-center justify-between gap-2 pl-1.5" }, z = { class: "has-[:focus-visible]:bg-b-1 group rounded-lg" }, N = {
  key: 1,
  class: "text-c-3 flex items-center justify-center rounded-lg border p-4"
}, j = /* @__PURE__ */ k({
  __name: "Overview",
  props: {
    documentSlug: {},
    document: {},
    eventBus: {},
    layout: {},
    path: {},
    method: {},
    exampleName: {},
    environment: {},
    securitySchemes: {},
    workspaceStore: {},
    activeWorkspace: {},
    plugins: {},
    collectionType: {}
  },
  setup(d) {
    const u = x(
      () => d.document?.info?.description ?? ""
    ), a = y("preview"), p = w("codeInputRef"), l = async (c) => {
      a.value = c, c === "edit" && (await B(), p.value?.focus());
    };
    return (c, e) => (t(), i("div", S, [
      o("div", V, [
        e[6] || (e[6] = o("h3", { class: "font-bold" }, "Description", -1)),
        a.value === "preview" ? (t(), m(n(b), {
          key: 0,
          class: "text-c-2 hover:text-c-1 flex items-center gap-2",
          size: "sm",
          type: "button",
          variant: "outlined",
          onClick: e[0] || (e[0] = (s) => l("edit"))
        }, {
          default: v(() => [
            r(n(g), {
              size: "sm",
              thickness: "1.5"
            }),
            e[5] || (e[5] = o("span", null, "Edit", -1))
          ]),
          _: 1
        })) : h("", !0)
      ]),
      o("div", z, [
        a.value === "preview" ? (t(), i(f, { key: 0 }, [
          u.value.trim().length ? (t(), i(f, { key: 0 }, [
            r(n(C), {
              class: "flex-1 rounded border border-transparent p-1.5 hover:border-(--scalar-background-3)",
              value: u.value,
              withImages: "",
              onDblclick: e[1] || (e[1] = (s) => l("edit"))
            }, null, 8, ["value"]),
            e[7] || (e[7] = o("div", { class: "brightness-lifted bg-b-1 absolute inset-0 -z-1 hidden rounded group-hover:block group-has-[:focus-visible]:hidden" }, null, -1))
          ], 64)) : (t(), i("div", N, [
            r(n(b), {
              class: "hover:bg-b-2 hover:text-c-1 text-c-2 flex items-center gap-2",
              size: "sm",
              variant: "ghost",
              onClick: e[2] || (e[2] = (s) => l("edit"))
            }, {
              default: v(() => [
                r(n(g), {
                  size: "sm",
                  thickness: "1.5"
                }),
                e[8] || (e[8] = o("span", null, "Write a description", -1))
              ]),
              _: 1
            })
          ]))
        ], 64)) : (t(), m(n(I), {
          key: 1,
          ref_key: "codeInputRef",
          ref: p,
          class: "border px-0.5 py-0",
          environment: void 0,
          layout: d.layout,
          modelValue: u.value,
          onBlur: e[3] || (e[3] = (s) => l("preview")),
          "onUpdate:modelValue": e[4] || (e[4] = (s) => d.eventBus.emit("document:update:info", { description: s }))
        }, null, 8, ["layout", "modelValue"]))
      ])
    ]));
  }
});
export {
  j as default
};
