// @scalar/api-client/dist/v2/features/collection/components/Settings.vue.js
import { defineComponent as i, createBlock as o, openBlock as c, unref as a } from "vue";
import { useRouter as s } from "vue-router";
import h from "../../settings/DocumentSettings.vue.js";
import k from "../../settings/CollectionSettings.vue.js";
const y = /* @__PURE__ */ i({
  __name: "Settings",
  props: {
    documentSlug: {},
    document: {},
    eventBus: {},
    layout: {},
    path: {},
    method: {},
    exampleName: {},
    environment: {},
    securitySchemes: {},
    workspaceStore: {},
    activeWorkspace: {},
    plugins: {},
    collectionType: {}
  },
  setup(e) {
    const n = (t) => {
      e.eventBus.emit("document:update:watch-mode", t);
    }, r = (t) => {
      e.eventBus.emit("workspace:update:theme", t);
    }, d = (t) => {
      e.eventBus.emit("workspace:update:active-proxy", t);
    }, m = (t) => {
      e.eventBus.emit("workspace:update:color-mode", t);
    }, l = s(), u = () => {
      e.workspaceStore.deleteDocument(e.documentSlug), l.push({
        name: "workspace.environment",
        params: {
          workspaceSlug: e.activeWorkspace.id
        }
      });
    };
    return (t, p) => e.collectionType === "document" ? (c(), o(a(h), {
      key: 0,
      documentUrl: e.document?.["x-scalar-original-source-url"],
      isDraftDocument: e.documentSlug === "drafts",
      title: e.document?.info.title ?? "",
      watchMode: e.document?.["x-scalar-watch-mode"] ?? !0,
      "onDelete:document": u,
      "onUpdate:watchMode": n
    }, null, 8, ["documentUrl", "isDraftDocument", "title", "watchMode"])) : (c(), o(a(k), {
      key: 1,
      activeProxyUrl: e.workspaceStore.workspace["x-scalar-active-proxy"],
      activeThemeId: e.workspaceStore.workspace["x-scalar-theme"] ?? "default",
      colorMode: e.workspaceStore.workspace["x-scalar-color-mode"] ?? "system",
      "onUpdate:colorMode": m,
      "onUpdate:proxyUrl": d,
      "onUpdate:themeId": r
    }, null, 8, ["activeProxyUrl", "activeThemeId", "colorMode"]));
  }
});
export {
  y as default
};
