// @scalar/api-client/dist/views/Request/components/OAuthScopesInput.vue.js
import { defineComponent as S, computed as f, openBlock as m, createBlock as h, unref as o, withCtx as s, createElementVNode as l, createVNode as c, toDisplayString as p, createElementBlock as g, Fragment as b, renderList as C } from "vue";
import { Disclosure as F, DisclosureButton as k, DisclosurePanel as D } from "@headlessui/vue";
import { ScalarIcon as y } from "@scalar/components";
import _ from "../../../components/DataTable/DataTableCell.vue.js";
import V from "../../../components/DataTable/DataTableRow.vue.js";
import B from "../../../components/DataTable/DataTableCheckbox.vue.js";
const O = { class: "flex h-full w-full" }, $ = /* @__PURE__ */ l("div", { class: "text-c-2 min-w-[120px] items-center pl-2 h-full border-r-1/2" }, [
  /* @__PURE__ */ l("span", { class: "h-8 flex items-center" }, " Scopes ")
], -1), j = { class: "flex-1 text-c-3" }, E = {
  class: "grid auto-rows-auto border-t-1/2",
  style: { gridTemplateColumns: "1fr auto" }
}, I = { class: "font-code text-xs" }, P = /* @__PURE__ */ S({
  __name: "OAuthScopesInput",
  props: {
    activeFlow: {},
    updateScheme: { type: Function }
  },
  setup(v) {
    const n = v, x = f(
      () => {
        var e;
        return Object.entries(((e = n.activeFlow) == null ? void 0 : e.scopes) ?? {}).map(([a, t]) => ({
          id: a,
          label: a,
          description: t
        }));
      }
    ), r = f(() => {
      var e;
      return ((e = n.activeFlow) == null ? void 0 : e.selectedScopes) || [];
    });
    function w(e, a) {
      a ? n.updateScheme("flow.selectedScopes", [...r.value, e]) : n.updateScheme(
        "flow.selectedScopes",
        r.value.filter((t) => t !== e)
      );
    }
    return (e, a) => (m(), h(o(_), { class: "items-center min-h-8 h-auto" }, {
      default: s(() => [
        l("div", O, [
          $,
          c(o(F), {
            as: "div",
            class: "flex flex-col w-full"
          }, {
            default: s(() => [
              c(o(k), { class: "group/scopes-accordion flex items-center text-left min-h-8 gap-1.5 h-auto pl-2 hover:bg-b-2 pr-2.5 cursor-pointer" }, {
                default: s(({ open: t }) => {
                  var i, u, d;
                  return [
                    l("div", j, " Selected " + p(((u = (i = e.activeFlow) == null ? void 0 : i.selectedScopes) == null ? void 0 : u.length) || 0) + " / " + p(Object.keys(((d = e.activeFlow) == null ? void 0 : d.scopes) ?? {}).length || 0), 1),
                    c(o(y), {
                      class: "text-c-3 group-hover/scopes-accordion:text-c-2",
                      icon: t ? "ChevronDown" : "ChevronRight",
                      size: "xs"
                    }, null, 8, ["icon"])
                  ];
                }),
                _: 1
              }),
              c(o(D), { as: "template" }, {
                default: s(() => [
                  l("table", E, [
                    (m(!0), g(b, null, C(x.value, ({ id: t, label: i, description: u }) => (m(), h(o(V), {
                      key: t,
                      class: "text-c-2",
                      onClick: (d) => w(t, !r.value.includes(t))
                    }, {
                      default: s(() => [
                        c(o(_), { class: "w-full px-2 py-1.5 hover:bg-b-2 cursor-pointer" }, {
                          default: s(() => [
                            l("span", null, [
                              l("span", I, p(i), 1),
                              l("span", null, " – " + p(u), 1)
                            ])
                          ]),
                          _: 2
                        }, 1024),
                        c(o(B), {
                          modelValue: r.value.includes(t),
                          "onUpdate:modelValue": () => {
                          }
                        }, null, 8, ["modelValue"])
                      ]),
                      _: 2
                    }, 1032, ["onClick"]))), 128))
                  ])
                ]),
                _: 1
              })
            ]),
            _: 1
          })
        ])
      ]),
      _: 1
    }));
  }
});
export {
  P as default
};
