// @scalar/api-client/dist/views/Request/components/OAuth2.vue.js
import { defineComponent as S, openBlock as u, createElementBlock as c, Fragment as h, createVNode as t, unref as d, withCtx as o, createTextVNode as s, createElementVNode as V, createBlock as f, createCommentVNode as m } from "vue";
import { useWorkspace as $ } from "../../../store/workspace.js";
import n from "../RequestSection/RequestAuthDataTableInput.vue.js";
import { useLoadingState as v, ScalarButton as k } from "@scalar/components";
import C from "./OAuthScopesInput.vue.js";
import i from "../../../components/DataTable/DataTableRow.vue.js";
import { authorizeOauth2 as T } from "../libs/oauth2.js";
const F = { class: "h-8 flex items-center justify-self-end" }, I = { class: "h-8 flex items-center justify-self-end" }, E = /* @__PURE__ */ S({
  __name: "OAuth2",
  props: {
    scheme: {}
  },
  setup(y) {
    const w = y, p = v(), { isReadOnly: U, securitySchemeMutators: g } = $(), r = (e, l) => g.edit(w.scheme.uid, e, l), z = async () => {
      if (p.isLoading) return;
      p.startLoading();
      const e = await T(w.scheme).finally(
        () => p.stopLoading()
      );
      e && r("flow.token", e);
    };
    return (e, l) => e.scheme.flow.token ? (u(), c(h, { key: 0 }, [
      t(d(i), null, {
        default: o(() => [
          t(n, {
            id: "oauth2-access-token",
            class: "border-r-transparent",
            modelValue: e.scheme.flow.token,
            placeholder: "QUxMIFlPVVIgQkFTRSBBUkUgQkVMT05HIFRPIFVT",
            type: "password",
            "onUpdate:modelValue": l[0] || (l[0] = (a) => r("flow.token", a))
          }, {
            default: o(() => [
              s(" Access Token ")
            ]),
            _: 1
          }, 8, ["modelValue"])
        ]),
        _: 1
      }),
      t(d(i), { class: "min-w-full" }, {
        default: o(() => [
          V("div", F, [
            t(d(k), {
              class: "p-0 py-0.5 px-2 mr-1",
              loading: d(p),
              size: "sm",
              variant: "outlined",
              onClick: l[1] || (l[1] = (a) => r("flow.token", ""))
            }, {
              default: o(() => [
                s(" Clear ")
              ]),
              _: 1
            }, 8, ["loading"])
          ])
        ]),
        _: 1
      })
    ], 64)) : (u(), c(h, { key: 1 }, [
      d(U) ? m("", !0) : (u(), f(d(i), { key: 0 }, {
        default: o(() => [
          "authorizationUrl" in e.scheme.flow ? (u(), f(n, {
            key: 0,
            id: `oauth2-authorization-url-${e.scheme.uid}`,
            modelValue: e.scheme.flow.authorizationUrl,
            placeholder: "https://galaxy.scalar.com/authorize",
            "onUpdate:modelValue": l[2] || (l[2] = (a) => r("flow.authorizationUrl", a))
          }, {
            default: o(() => [
              s(" Authorization Url ")
            ]),
            _: 1
          }, 8, ["id", "modelValue"])) : m("", !0),
          "tokenUrl" in e.scheme.flow ? (u(), f(n, {
            key: 1,
            id: `oauth2-token-url-${e.scheme.uid}`,
            modelValue: e.scheme.flow.tokenUrl,
            placeholder: "https://galaxy.scalar.com/token",
            "onUpdate:modelValue": l[3] || (l[3] = (a) => r("flow.tokenUrl", a))
          }, {
            default: o(() => [
              s(" Token Url ")
            ]),
            _: 1
          }, 8, ["id", "modelValue"])) : m("", !0)
        ]),
        _: 1
      })),
      "redirectUri" in e.scheme.flow ? (u(), f(d(i), { key: 1 }, {
        default: o(() => [
          t(n, {
            id: `oauth2-redirect-uri-${e.scheme.uid}`,
            modelValue: e.scheme.flow.redirectUri,
            placeholder: "https://galaxy.scalar.com/callback",
            "onUpdate:modelValue": l[4] || (l[4] = (a) => r("flow.redirectUri", a))
          }, {
            default: o(() => [
              s(" Redirect Url ")
            ]),
            _: 1
          }, 8, ["id", "modelValue"])
        ]),
        _: 1
      })) : m("", !0),
      e.scheme.flow.type === "password" ? (u(), c(h, { key: 2 }, [
        t(d(i), null, {
          default: o(() => [
            t(n, {
              id: `oauth2-password-username-${e.scheme.uid}`,
              class: "text-c-2",
              modelValue: e.scheme.flow.value,
              placeholder: "ScalarEnjoyer01",
              "onUpdate:modelValue": l[5] || (l[5] = (a) => r("flow.value", a))
            }, {
              default: o(() => [
                s(" Username ")
              ]),
              _: 1
            }, 8, ["id", "modelValue"])
          ]),
          _: 1
        }),
        t(d(i), null, {
          default: o(() => [
            t(n, {
              id: `oauth2-password-password-${e.scheme.uid}`,
              modelValue: e.scheme.flow.secondValue,
              placeholder: "xxxxxx",
              type: "password",
              "onUpdate:modelValue": l[6] || (l[6] = (a) => r("flow.secondValue", a))
            }, {
              default: o(() => [
                s(" Password ")
              ]),
              _: 1
            }, 8, ["id", "modelValue"])
          ]),
          _: 1
        })
      ], 64)) : m("", !0),
      t(d(i), null, {
        default: o(() => [
          t(n, {
            id: `oauth2-client-id-${e.scheme.uid}`,
            modelValue: e.scheme.clientId,
            placeholder: "12345",
            "onUpdate:modelValue": l[7] || (l[7] = (a) => r("clientId", a))
          }, {
            default: o(() => [
              s(" Client ID ")
            ]),
            _: 1
          }, 8, ["id", "modelValue"])
        ]),
        _: 1
      }),
      "clientSecret" in e.scheme.flow ? (u(), f(d(i), { key: 3 }, {
        default: o(() => [
          t(n, {
            id: `oauth2-client-secret-${e.scheme.uid}`,
            modelValue: e.scheme.flow.clientSecret,
            placeholder: "XYZ123",
            type: "password",
            "onUpdate:modelValue": l[8] || (l[8] = (a) => r("flow.clientSecret", a))
          }, {
            default: o(() => [
              s(" Client Secret ")
            ]),
            _: 1
          }, 8, ["id", "modelValue"])
        ]),
        _: 1
      })) : m("", !0),
      e.scheme.flow.scopes ? (u(), f(d(i), { key: 4 }, {
        default: o(() => [
          t(C, {
            activeFlow: e.scheme.flow,
            updateScheme: r
          }, null, 8, ["activeFlow"])
        ]),
        _: 1
      })) : m("", !0),
      t(d(i), { class: "min-w-full" }, {
        default: o(() => [
          V("div", I, [
            t(d(k), {
              class: "p-0 py-0.5 px-2 mr-1",
              loading: d(p),
              size: "sm",
              variant: "outlined",
              onClick: z
            }, {
              default: o(() => [
                s(" Authorize ")
              ]),
              _: 1
            }, 8, ["loading"])
          ])
        ]),
        _: 1
      })
    ], 64));
  }
});
export {
  E as default
};
