// @scalar/api-client/dist/views/Request/components/WorkspaceDropdown.vue.js
import { defineComponent as K, computed as O, ref as W, openBlock as p, createElementBlock as D, Fragment as S, createElementVNode as a, createVNode as t, unref as e, withCtx as s, renderList as U, createBlock as x, withModifiers as w, normalizeClass as A, createTextVNode as k, toDisplayString as z, withKeys as F } from "vue";
import I from "../../../components/Sidebar/Actions/DeleteSidebarListElement.vue.js";
import { useWorkspace as L } from "../../../store/workspace.js";
import { useModal as $, ScalarDropdown as N, ScalarContextMenu as P, ScalarDropdownItem as m, ScalarIcon as c, ScalarTooltip as q, ScalarDropdownDivider as G, ScalarButton as _, ScalarModal as V, ScalarTextField as H } from "@scalar/components";
import { useRouter as J } from "vue-router";
import { commandPaletteBus as Q } from "../../../libs/event-busses/command-palette-bus.js";
const X = { class: "xl:min-h-header xl:py-2.5 py-1 flex items-center border-b-1/2 px-2.5 text-sm" }, Y = { class: "font-medium m-0 text-sm flex gap-1.5 items-center" }, Z = /* @__PURE__ */ a("span", null, "Rename", -1), ee = /* @__PURE__ */ a("span", null, "Delete", -1), te = /* @__PURE__ */ a("div", { class: "grid gap-1.5 pointer-events-none min-w-48 w-content shadow-lg rounded bg-b-1 z-100 p-2 text-xxs leading-5 z-10 text-c-1" }, [
  /* @__PURE__ */ a("div", { class: "flex items-center text-c-2" }, [
    /* @__PURE__ */ a("span", null, "Only workspace cannot be deleted.")
  ])
], -1), se = /* @__PURE__ */ a("span", null, "Delete", -1), ae = { class: "flex items-center justify-center h-4 w-4" }, ne = /* @__PURE__ */ a("span", null, "Create new workspace", -1), le = { class: "flex gap-3" }, me = /* @__PURE__ */ K({
  __name: "WorkspaceDropdown",
  setup(oe) {
    const { activeWorkspace: f, workspaces: d, workspaceMutators: g } = L(), { push: b } = J(), j = (n) => {
      n !== f.value.uid && b(`/workspace/${n}`);
    }, y = O(() => Object.keys(d).length === 1), B = () => Q.emit({ commandName: "Create Workspace" }), o = W(""), r = W(""), v = $(), h = $(), R = (n) => {
      o.value = d[n].name, r.value = n, v.show();
    }, C = () => {
      o.value.trim() && (g.rename(r.value, o.value.trim()), v.hide());
    }, T = (n) => {
      o.value = d[n].name, r.value = n, h.show();
    }, E = async () => {
      if (!y.value) {
        const n = f.value.uid === r.value, l = { ...d };
        if (delete l[r.value], g.delete(r.value), n) {
          const i = Object.keys(l)[0];
          await b(`/workspace/${i}/`);
        }
      }
      h.hide();
    };
    return (n, l) => (p(), D(S, null, [
      a("div", X, [
        t(e(N), null, {
          items: s(() => [
            (p(!0), D(S, null, U(e(d), (i, u) => (p(), x(e(P), { key: u }, {
              trigger: s(() => [
                t(e(m), {
                  class: "flex gap-1.5 group/item items-center whitespace-nowrap text-ellipsis overflow-hidden w-full",
                  onClick: w((M) => j(u), ["stop"])
                }, {
                  default: s(() => [
                    a("div", {
                      class: A([
                        "flex items-center justify-center rounded-full p-[3px] w-4 h-4 group-hover/item:shadow-border",
                        e(f).uid === u ? "bg-blue text-b-1" : "text-transparent"
                      ])
                    }, [
                      t(e(c), {
                        class: "size-2.5",
                        icon: "Checkmark",
                        thickness: "3.5"
                      })
                    ], 2),
                    k(" " + z(i.name), 1)
                  ]),
                  _: 2
                }, 1032, ["onClick"])
              ]),
              content: s(() => [
                t(e(N), { static: "" }, {
                  items: s(() => [
                    t(e(m), {
                      class: "flex !gap-2",
                      onMousedown: (M) => R(u)
                    }, {
                      default: s(() => [
                        t(e(c), {
                          class: "inline-flex",
                          icon: "Edit",
                          size: "sm",
                          thickness: "1.5"
                        }),
                        Z
                      ]),
                      _: 2
                    }, 1032, ["onMousedown"]),
                    y.value ? (p(), x(e(q), {
                      key: 0,
                      class: "z-10",
                      side: "bottom"
                    }, {
                      trigger: s(() => [
                        t(e(m), {
                          class: "flex w-full",
                          disabled: "",
                          onMousedown: l[0] || (l[0] = w(() => {
                          }, ["prevent"]))
                        }, {
                          default: s(() => [
                            t(e(c), {
                              class: "inline-flex",
                              icon: "Trash",
                              size: "sm",
                              thickness: "1.5"
                            }),
                            ee
                          ]),
                          _: 1
                        })
                      ]),
                      content: s(() => [
                        te
                      ]),
                      _: 1
                    })) : (p(), x(e(m), {
                      key: 1,
                      class: "flex !gap-2",
                      onMousedown: w((M) => T(u), ["prevent"])
                    }, {
                      default: s(() => [
                        t(e(c), {
                          class: "inline-flex",
                          icon: "Trash",
                          size: "sm",
                          thickness: "1.5"
                        }),
                        se
                      ]),
                      _: 2
                    }, 1032, ["onMousedown"]))
                  ]),
                  _: 2
                }, 1024)
              ]),
              _: 2
            }, 1024))), 128)),
            t(e(G)),
            t(e(m), {
              class: "flex items-center gap-1.5",
              onClick: B
            }, {
              default: s(() => [
                a("div", ae, [
                  t(e(c), {
                    class: "h-2.5",
                    icon: "Add",
                    thickness: "3"
                  })
                ]),
                ne
              ]),
              _: 1
            })
          ]),
          default: s(() => [
            t(e(_), {
              class: "font-normal h-full justify-start py-1.5 px-1.5 text-c-1 hover:bg-b-2 w-fit",
              fullWidth: "",
              variant: "ghost"
            }, {
              default: s(() => [
                a("h2", Y, [
                  k(z(e(f).name) + " ", 1),
                  t(e(c), {
                    class: "size-2.5",
                    icon: "ChevronDown",
                    thickness: "3.5"
                  })
                ])
              ]),
              _: 1
            })
          ]),
          _: 1
        })
      ]),
      t(e(V), {
        size: "sm",
        state: e(h),
        title: `Delete ${o.value}`
      }, {
        default: s(() => [
          t(I, {
            variableName: o.value,
            warningMessage: "Warning: Deleting this will delete all items inside of this",
            onClose: l[1] || (l[1] = (i) => e(h).hide()),
            onDelete: E
          }, null, 8, ["variableName"])
        ]),
        _: 1
      }, 8, ["state", "title"]),
      t(e(V), {
        state: e(v),
        title: "Rename Workspace"
      }, {
        default: s(() => [
          t(e(H), {
            modelValue: o.value,
            "onUpdate:modelValue": l[2] || (l[2] = (i) => o.value = i),
            label: "Workspace",
            labelShadowColor: "var(--scalar-background-1)",
            onKeydown: F(w(C, ["prevent"]), ["enter"])
          }, null, 8, ["modelValue", "onKeydown"]),
          a("div", le, [
            t(e(_), {
              class: "flex-1",
              variant: "outlined",
              onClick: l[3] || (l[3] = (i) => e(v).hide())
            }, {
              default: s(() => [
                k(" Cancel ")
              ]),
              _: 1
            }),
            t(e(_), {
              class: "flex-1",
              type: "submit",
              onClick: C
            }, {
              default: s(() => [
                k(" Save ")
              ]),
              _: 1
            })
          ])
        ]),
        _: 1
      }, 8, ["state"])
    ], 64));
  }
});
export {
  me as default
};
