// @scalar/api-reference/dist/features/info-object/License.vue.js
import { defineComponent, createElementBlock, openBlock, createVNode, createElementVNode, unref, toDisplayString, Fragment } from "vue";
import { ScalarIconGavel } from "@scalar/icons";
const _hoisted_1 = { class: "group flex h-fit items-center last:border-r-0 xl:border-r xl:first:ml-auto" };
const _hoisted_2 = ["href"];
const _hoisted_3 = { class: "ml-1 empty:hidden" };
const _hoisted_4 = { class: "ml-1 empty:hidden" };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "License",
  props: {
    value: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", _hoisted_1, [
        __props.value?.url ? (openBlock(), createElementBlock("a", {
          key: 0,
          class: "text-c-1 hover:bg-b-2 mr-2 flex min-h-7 min-w-7 items-center rounded-lg border px-2 py-1 no-underline group-last:mr-0 xl:border-none",
          href: __props.value.url,
          target: "_blank",
          rel: "noopener noreferrer"
        }, [
          createVNode(unref(ScalarIconGavel), {
            weight: "bold",
            class: "size-3 text-current"
          }),
          createElementVNode("span", _hoisted_3, toDisplayString(__props.value?.name || __props.value.identifier || __props.value.url), 1)
        ], 8, _hoisted_2)) : (openBlock(), createElementBlock(Fragment, { key: 1 }, [
          createVNode(unref(ScalarIconGavel), {
            weight: "bold",
            class: "size-3 text-current"
          }),
          createElementVNode("span", _hoisted_4, toDisplayString(__props.value?.name), 1)
        ], 64))
      ]);
    };
  }
});
export {
  _sfc_main as default
};
