// @scalar/api-reference/dist/features/info-object/TermsOfService.vue.js
import { defineComponent, createElementBlock, createCommentVNode, openBlock, createElementVNode, createVNode, unref } from "vue";
import { ScalarIconScroll } from "@scalar/icons";
const _hoisted_1 = {
  key: 0,
  class: "group flex items-center last:border-r-0 xl:border-r xl:first:ml-auto"
};
const _hoisted_2 = ["href"];
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "TermsOfService",
  props: {
    value: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return __props.value ? (openBlock(), createElementBlock("div", _hoisted_1, [
        createElementVNode("a", {
          class: "text-c-1 hover:bg-b-2 mr-2 flex min-h-7 min-w-7 items-center rounded-lg border px-2 py-1 no-underline group-last:mr-0 xl:border-none",
          href: __props.value,
          target: "_blank",
          rel: "noopener noreferrer"
        }, [
          createVNode(unref(ScalarIconScroll), {
            weight: "bold",
            class: "size-3 text-current"
          }),
          _cache[0] || (_cache[0] = createElementVNode("span", { class: "ml-1 empty:hidden" }, "Terms of Service", -1))
        ], 8, _hoisted_2)
      ])) : createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as default
};
