// @scalar/api-reference/dist/components/Content/Models/Models.vue2.js
import { defineComponent as S, computed as p, openBlock as i, createBlock as d, unref as e, withCtx as l, createVNode as a, createElementBlock as v, Fragment as k, renderList as y, createTextVNode as b, toDisplayString as g, createCommentVNode as c, pushScopeId as C, popScopeId as I, createElementVNode as M } from "vue";
import N from "../../ShowMoreButton.vue.js";
import w from "../../Section/SectionContainer.vue.js";
import f from "../Lazy/Lazy.vue.js";
import B from "../../Section/Section.vue.js";
import V from "../../Section/SectionContent.vue.js";
import L from "../../Section/SectionHeader.vue.js";
import j from "../../Anchor/Anchor.vue.js";
import z from "../Schema/Schema.vue.js";
import { useSidebar as A } from "../../../hooks/useSidebar.js";
import { useNavState as E } from "../../../hooks/useNavState.js";
const O = (r) => (C("data-v-f19478dd"), r = r(), I(), r), D = /* @__PURE__ */ O(() => /* @__PURE__ */ M("div", { id: "models" }, null, -1)), U = /* @__PURE__ */ S({
  __name: "Models",
  props: {
    schemas: {}
  },
  setup(r) {
    const m = r, { collapsedSidebarItems: h } = A(), { getModelId: s } = E(), n = p(
      () => Object.keys(m.schemas ?? {}).length <= 3 || h[s()]
    ), u = p(() => {
      const o = Object.keys(m.schemas ?? {});
      return n.value ? o : o.slice(0, 3);
    });
    return (o, F) => o.schemas ? (i(), d(e(w), { key: 0 }, {
      default: l(() => [
        a(e(f), {
          id: "models",
          isLazy: !1
        }, {
          default: l(() => [
            D
          ]),
          _: 1
        }),
        (i(!0), v(k, null, y(u.value, (t, _) => (i(), d(e(f), {
          id: e(s)(t),
          key: t,
          isLazy: ""
        }, {
          default: l(() => [
            a(e(B), {
              id: e(s)(t),
              label: t
            }, {
              default: l(() => [
                o.schemas[t] ? (i(), d(e(V), { key: 0 }, {
                  default: l(() => [
                    a(e(L), { level: 2 }, {
                      default: l(() => [
                        a(e(j), {
                          id: e(s)(t)
                        }, {
                          default: l(() => [
                            b(g(o.schemas[t].title ?? t), 1)
                          ]),
                          _: 2
                        }, 1032, ["id"])
                      ]),
                      _: 2
                    }, 1024),
                    a(e(z), {
                      name: t,
                      noncollapsible: "",
                      value: o.schemas[t]
                    }, null, 8, ["name", "value"]),
                    !n.value && _ === u.value.length - 1 ? (i(), d(N, {
                      key: 0,
                      id: e(s)(),
                      class: "something-special"
                    }, null, 8, ["id"])) : c("", !0)
                  ]),
                  _: 2
                }, 1024)) : c("", !0)
              ]),
              _: 2
            }, 1032, ["id", "label"])
          ]),
          _: 2
        }, 1032, ["id"]))), 128))
      ]),
      _: 1
    })) : c("", !0);
  }
});
export {
  U as default
};
