// @scalar/api-reference/dist/components/Content/Models/ModelsAccordion.vue2.js
import { defineComponent as A, computed as C, openBlock as t, createBlock as s, unref as e, withCtx as o, createVNode as i, createTextVNode as N, createElementBlock as c, Fragment as v, renderList as k, createCommentVNode as g } from "vue";
import h from "../../Section/SectionContainerAccordion.vue.js";
import y from "../../Section/SectionHeader.vue.js";
import B from "../../Section/SectionAccordion.vue.js";
import M from "../../Anchor/Anchor.vue.js";
import V from "../Schema/SchemaHeading.vue.js";
import S from "../Schema/SchemaProperty.vue.js";
import { useNavState as j } from "../../../hooks/useNavState.js";
const H = {
  key: 0,
  class: "properties"
}, O = { key: 1 }, D = /* @__PURE__ */ A({
  __name: "ModelsAccordion",
  props: {
    schemas: {}
  },
  setup(b) {
    const d = b, q = C(() => d.schemas ? Object.entries(d.schemas).map(([a, m]) => ({
      name: a,
      schema: m
    })) : []), { getModelId: u } = j();
    return (a, m) => d.schemas ? (t(), s(e(h), {
      key: 0,
      class: "reference-models"
    }, {
      title: o(() => [
        i(e(y), { level: 2 }, {
          default: o(() => [
            N("Models")
          ]),
          _: 1
        })
      ]),
      default: o(() => [
        (t(!0), c(v, null, k(q.value, ({ name: l, schema: r }) => (t(), s(e(B), {
          id: e(u)(l),
          key: l,
          label: l
        }, {
          title: o(() => [
            i(e(M), {
              id: e(u)(l),
              class: "reference-models-anchor"
            }, {
              default: o(() => [
                i(e(V), {
                  class: "reference-models-label",
                  name: l,
                  value: r
                }, null, 8, ["name", "value"])
              ]),
              _: 2
            }, 1032, ["id"])
          ]),
          default: o(() => [
            r != null && r.properties ? (t(), c("div", H, [
              (t(!0), c(v, null, k(Object.entries(r.properties), ([n, x]) => {
                var p, f, _;
                return t(), s(e(S), {
                  key: n,
                  name: n,
                  required: ((p = r.required) == null ? void 0 : p.includes(n)) || ((_ = (f = r.properties) == null ? void 0 : f[n]) == null ? void 0 : _.required) === !0,
                  value: x
                }, null, 8, ["name", "required", "value"]);
              }), 128))
            ])) : (t(), c("div", O, [
              i(e(S), { value: r }, null, 8, ["value"])
            ]))
          ]),
          _: 2
        }, 1032, ["id", "label"]))), 128))
      ]),
      _: 1
    })) : g("", !0);
  }
});
export {
  D as default
};
