// @scalar/components/dist/components/ScalarContextMenu/ScalarContextMenu.vue.js
import { defineComponent as s, openBlock as f, createBlock as i, unref as e, withCtx as t, createVNode as o, renderSlot as a } from "vue";
import { ContextMenuRoot as r, ContextMenuTrigger as u, ContextMenuPortal as c, ContextMenuContent as p, ContextMenuItem as m } from "radix-vue";
const x = /* @__PURE__ */ s({
  __name: "ScalarContextMenu",
  props: {
    align: { default: "center" },
    side: { default: "bottom" },
    sideOffset: {},
    disabled: { type: Boolean, default: !1 }
  },
  setup(d) {
    const n = d;
    return (l, C) => (f(), i(e(r), null, {
      default: t(() => [
        o(e(u), {
          disabled: n.disabled
        }, {
          default: t(() => [
            a(l.$slots, "trigger")
          ]),
          _: 3
        }, 8, ["disabled"]),
        o(e(c), null, {
          default: t(() => [
            o(e(p), {
              align: n.align,
              side: n.side,
              sideOffset: n.sideOffset
            }, {
              default: t(() => [
                o(e(m), null, {
                  default: t(() => [
                    a(l.$slots, "content")
                  ]),
                  _: 3
                })
              ]),
              _: 3
            }, 8, ["align", "side", "sideOffset"])
          ]),
          _: 3
        })
      ]),
      _: 3
    }));
  }
});
export {
  x as default
};
