// @scalar/api-client/dist/v2/features/environments/components/EnvironmentDeleteModal.vue.js
import { defineComponent as m, createBlock as u, openBlock as f, unref as n, withCtx as o, createElementVNode as l, createVNode as i, createTextVNode as s } from "vue";
import { ScalarModal as x, ScalarButton as r } from "@scalar/components";
const c = { class: "flex justify-between" }, k = /* @__PURE__ */ m({
  __name: "EnvironmentDeleteModal",
  props: {
    state: {},
    name: { default: "unknown" }
  },
  emits: ["cancel", "submit"],
  setup(t, { emit: d }) {
    const a = d;
    return (b, e) => (f(), u(n(x), {
      bodyClass: "border-t-0 rounded-t-lg flex flex-col gap-4",
      size: "xxs",
      state: t.state,
      title: `Delete ${t.name}`
    }, {
      default: o(() => [
        e[4] || (e[4] = l("p", { class: "text-c-2 text-sm leading-normal text-pretty" }, " Are you sure you want to delete this environment? This action cannot be undone. ", -1)),
        l("div", c, [
          i(n(r), {
            size: "sm",
            variant: "outlined",
            onClick: e[0] || (e[0] = () => {
              a("cancel"), t.state.hide();
            })
          }, {
            default: o(() => [...e[2] || (e[2] = [
              s(" Cancel ", -1)
            ])]),
            _: 1
          }),
          i(n(r), {
            size: "sm",
            variant: "danger",
            onClick: e[1] || (e[1] = () => {
              a("submit"), t.state.hide();
            })
          }, {
            default: o(() => [...e[3] || (e[3] = [
              s(" Delete Environment ", -1)
            ])]),
            _: 1
          })
        ])
      ]),
      _: 1
    }, 8, ["state", "title"]));
  }
});
export {
  k as default
};
