// @scalar/api-client/dist/v2/features/environments/components/Environment.vue.js
import { defineComponent as m, createElementBlock as a, openBlock as d, createElementVNode as n, createVNode as i, normalizeStyle as v, toDisplayString as u, unref as o } from "vue";
import { ScalarIconButton as r } from "@scalar/components";
import { ScalarIconNotePencil as f, ScalarIconTrash as p } from "@scalar/icons";
import x from "./EnvironmentVariablesTable.vue.js";
const y = { class: "group rounded-lg border" }, b = { class: "bg-b-2 flex justify-between rounded-t-lg px-1 py-1 text-sm" }, g = { class: "flex items-center gap-1" }, h = { class: "flex h-6 w-6 items-center justify-center p-1" }, N = { class: "px-1 py-0.5 text-sm" }, B = { class: "hidden flex-row items-center gap-1 group-hover:flex" }, C = /* @__PURE__ */ m({
  __name: "Environment",
  props: {
    environment: {},
    environmentName: {},
    eventBus: {},
    collectionType: {}
  },
  emits: ["edit", "delete"],
  setup(e, { emit: s }) {
    const l = s;
    return (E, t) => (d(), a("div", y, [
      n("div", b, [
        n("div", g, [
          n("div", h, [
            n("span", {
              class: "h-2.5 w-2.5 rounded-full",
              style: v({ backgroundColor: e.environment.color })
            }, null, 4)
          ]),
          n("span", N, u(e.environmentName), 1)
        ]),
        n("div", B, [
          i(o(r), {
            icon: o(f),
            label: "Edit Environment",
            size: "sm",
            onClick: t[0] || (t[0] = (c) => l("edit"))
          }, null, 8, ["icon"]),
          i(o(r), {
            icon: o(p),
            label: "Delete Environment",
            size: "sm",
            onClick: t[1] || (t[1] = (c) => l("delete"))
          }, null, 8, ["icon"])
        ])
      ]),
      i(x, {
        collectionType: e.collectionType,
        environment: e.environment,
        environmentName: e.environmentName,
        eventBus: e.eventBus
      }, null, 8, ["collectionType", "environment", "environmentName", "eventBus"])
    ]));
  }
});
export {
  C as default
};
