// @scalar/api-client/dist/v2/features/environments/components/EnvironmentVariablesTable.vue2.js
import { defineComponent as h, computed as y, createBlock as s, openBlock as u, unref as a, withCtx as o, createVNode as r, createElementBlock as T, createTextVNode as v, Fragment as C, renderList as k, createCommentVNode as B } from "vue";
import { ScalarButton as w } from "@scalar/components";
import { ScalarIconTrash as x } from "@scalar/icons";
import I from "../../../components/data-table/DataTable.vue.js";
import b from "../../../components/data-table/DataTableRow.vue.js";
import c from "../../../components/data-table/DataTableHeader.vue.js";
import d from "../../../components/data-table/DataTableCell.vue.js";
import V from "../../../components/code-input/CodeInput.vue.js";
const W = /* @__PURE__ */ h({
  __name: "EnvironmentVariablesTable",
  props: {
    environment: {},
    environmentName: {},
    eventBus: {},
    collectionType: {}
  },
  setup(e) {
    const p = ["1fr", "1fr", "36px"], N = y(() => {
      const i = e.environment.variables.map((t) => ({
        name: t.name,
        value: typeof t.value == "string" ? t.value : t.value.default
      })), n = e.environment.variables.at(-1), l = n && (n.name !== "" || n.value !== "");
      return !n || l ? [...i, { name: "", value: "" }] : i;
    }), f = (i, n, l) => {
      const t = l >= e.environment.variables.length;
      if (!(!i && t)) {
        if (l >= e.environment.variables.length) {
          e.eventBus.emit("environment:upsert:environment-variable", {
            environmentName: e.environmentName,
            variable: { name: i, value: n },
            collectionType: e.collectionType
          });
          return;
        }
        e.eventBus.emit("environment:upsert:environment-variable", {
          environmentName: e.environmentName,
          variable: { name: i, value: n },
          index: l,
          collectionType: e.collectionType
        });
      }
    }, g = (i) => e.eventBus.emit("environment:delete:environment-variable", {
      environmentName: e.environmentName,
      index: i,
      collectionType: e.collectionType
    });
    return (i, n) => (u(), s(a(I), {
      class: "group/table data-table h-min flex-1 rounded",
      columns: p
    }, {
      default: o(() => [
        r(a(b), { class: "sr-only !block" }, {
          default: o(() => [
            r(a(c), null, {
              default: o(() => [...n[0] || (n[0] = [
                v("Name", -1)
              ])]),
              _: 1
            }),
            r(a(c), null, {
              default: o(() => [...n[1] || (n[1] = [
                v("Value", -1)
              ])]),
              _: 1
            }),
            r(a(c), null, {
              default: o(() => [...n[2] || (n[2] = [
                v("Actions", -1)
              ])]),
              _: 1
            })
          ]),
          _: 1
        }),
        (u(!0), T(C, null, k(N.value, (l, t) => (u(), s(a(b), {
          key: t,
          class: "group/row"
        }, {
          default: o(() => [
            r(a(d), null, {
              default: o(() => [
                r(a(V), {
                  "aria-label": "Environment Variable Name",
                  disableCloseBrackets: "",
                  disableTabIndent: "",
                  environment: void 0,
                  lineWrapping: "",
                  modelValue: l.name,
                  placeholder: "Name",
                  "onUpdate:modelValue": (m) => f(m, l.value, t)
                }, null, 8, ["modelValue", "onUpdate:modelValue"])
              ]),
              _: 2
            }, 1024),
            r(a(d), null, {
              default: o(() => [
                r(a(V), {
                  "aria-label": "Environment Variable Value",
                  disableTabIndent: "",
                  environment: void 0,
                  lineWrapping: "",
                  modelValue: l.value,
                  placeholder: "Value",
                  "onUpdate:modelValue": (m) => f(l.name, m, t)
                }, null, 8, ["modelValue", "onUpdate:modelValue"])
              ]),
              _: 2
            }, 1024),
            r(a(d), { class: "flex items-center justify-center" }, {
              default: o(() => [
                t < e.environment.variables.length ? (u(), s(a(w), {
                  key: 0,
                  class: "text-c-2 hover:text-c-1 hover:bg-b-2 hidden h-fit rounded p-1 group-focus-within:flex group-hover/row:flex",
                  size: "sm",
                  variant: "ghost",
                  onClick: (m) => g(t)
                }, {
                  default: o(() => [
                    r(a(x), { class: "size-3.5" })
                  ]),
                  _: 1
                }, 8, ["onClick"])) : B("", !0)
              ]),
              _: 2
            }, 1024)
          ]),
          _: 2
        }, 1024))), 128))
      ]),
      _: 1
    }));
  }
});
export {
  W as default
};
