// @scalar/api-client/dist/views/Environment/EnvironmentColors.vue.js
import { defineComponent as E, ref as f, computed as I, watch as S, createElementBlock as v, openBlock as t, createCommentVNode as n, Fragment as w, normalizeStyle as m, normalizeClass as r, createBlock as x, unref as C, createElementVNode as a, renderList as j, withDirectives as D, vModelText as V, createVNode as $, nextTick as N } from "vue";
import { ScalarIcon as k } from "@scalar/components";
const R = ["onClick"], T = ["placeholder"], U = /* @__PURE__ */ E({
  __name: "EnvironmentColors",
  props: {
    activeColor: {},
    selector: { type: Boolean, default: !1 }
  },
  emits: ["select"],
  setup(c, { emit: F }) {
    const e = c, z = F, l = f(""), h = f(null), u = f(!1), i = f(!1), b = [
      { color: "#FFFFFF" },
      { color: "#EF0006" },
      { color: "#EDBE20" },
      { color: "#069061" },
      { color: "#FB892C" },
      { color: "#0082D0" },
      { color: "#5203D1" },
      { color: "#FFC0CB" }
    ], y = I(() => e.activeColor && !b.some((s) => s.color === e.activeColor) || l.value ? `background-color: ${e.activeColor || l.value};` : "background: linear-gradient(to right, rgb(235, 87, 87), rgb(242, 201, 76), rgb(76, 183, 130), rgb(78, 167, 252), rgb(250, 96, 122));"), p = () => {
      u.value = !u.value, e.selector && (i.value = !1), N(() => {
        h.value && h.value.focus();
      });
    };
    S(l, (s) => {
      s && !s.startsWith("#") && (l.value = `#${s}`), u.value = !0;
    });
    const B = () => {
      e.selector && (i.value = !i.value);
    }, g = (s) => {
      z("select", s), e.selector && (i.value = !1);
    };
    return (s, d) => (t(), v("div", null, [
      u.value ? n("", !0) : (t(), v(w, { key: 0 }, [
        e.selector && !i.value ? (t(), v("div", {
          key: 0,
          class: r(["flex cursor-pointer items-center justify-center rounded-full", e.selector ? "h-4 w-4" : "h-5 w-5"]),
          style: m({ backgroundColor: c.activeColor }),
          onClick: B
        }, [
          c.activeColor ? (t(), x(C(k), {
            key: 0,
            class: r(["text-c-btn", e.selector && "p-0.5"]),
            icon: "Checkmark",
            size: "xs"
          }, null, 8, ["class"])) : n("", !0)
        ], 6)) : n("", !0),
        i.value || !e.selector ? (t(), v("div", {
          key: 1,
          class: r(["color-selector flex flex-row items-center justify-between gap-1.5 space-x-1", e.selector ? "h-4" : "min-h-10 min-w-[296px]"])
        }, [
          (t(), v(w, null, j(b, (o) => a("div", {
            key: o.color,
            class: r(["flex cursor-pointer items-center justify-center rounded-full", e.selector ? "h-4 w-4" : "h-5 w-5"]),
            style: m({ backgroundColor: o.color }),
            onClick: (L) => g(o.color)
          }, [
            c.activeColor === o.color && !l.value ? (t(), x(C(k), {
              key: 0,
              class: r(["text-c-btn", e.selector && "p-0.5"]),
              icon: "Checkmark",
              size: "xs"
            }, null, 8, ["class"])) : n("", !0)
          ], 14, R)), 64)),
          d[2] || (d[2] = a("hr", { class: "border-ghost h-5 w-0.5 border-l" }, null, -1)),
          a("label", {
            class: r(["z-10 flex cursor-pointer flex-row items-center justify-center gap-2 rounded-full", e.selector ? "h-4 w-4" : "h-5 w-5"]),
            style: m(y.value),
            onClick: p
          }, [
            !u.value && (c.activeColor === l.value || c.activeColor && !b.some((o) => o.color === c.activeColor)) ? (t(), x(C(k), {
              key: 0,
              class: "text-c-btn",
              icon: "Checkmark",
              size: "xs"
            })) : n("", !0)
          ], 6)
        ], 2)) : n("", !0)
      ], 64)),
      u.value ? (t(), v("div", {
        key: 1,
        class: r(["color-selector flex flex-1 items-center gap-2 rounded", e.selector ? "h-4" : "min-h-10"])
      }, [
        a("span", {
          class: r(["absolute rounded-full border border-dashed", e.selector ? "h-4 w-4" : "h-5 w-5"])
        }, null, 2),
        a("span", {
          class: r(["z-[1] rounded-full", e.selector ? "h-4 w-4" : "h-5 w-5"]),
          style: m(y.value)
        }, null, 6),
        D(a("input", {
          ref_key: "customColorInputRef",
          ref: h,
          "onUpdate:modelValue": d[0] || (d[0] = (o) => l.value = o),
          class: "w-full flex-1 border-transparent text-sm outline-none",
          placeholder: c.activeColor || "#000000",
          type: "text",
          onInput: d[1] || (d[1] = (o) => g(l.value))
        }, null, 40, T), [
          [V, l.value]
        ]),
        a("button", {
          class: "text-c-3 hover:bg-b-2 rounded-lg p-1.5",
          type: "button",
          onClick: p
        }, [
          $(C(k), {
            icon: "Checkmark",
            size: "xs"
          })
        ])
      ], 2)) : n("", !0)
    ]));
  }
});
export {
  U as default
};
