// @scalar/api-client/dist/views/Environment/EnvironmentVariableDropdown.vue.js
import { defineComponent as b, ref as y, computed as E, openBlock as r, createElementBlock as o, Fragment as C, renderList as V, createElementVNode as a, normalizeClass as B, createTextVNode as i, toDisplayString as u, createBlock as I, unref as d, withCtx as z, createVNode as N } from "vue";
import { ScalarButton as P, ScalarIcon as S } from "@scalar/components";
import $ from "fuse.js";
const j = ["open"], q = { key: 0 }, A = ["onClick"], D = { class: "flex items-center gap-1.5 whitespace-nowrap" }, F = { class: "w-20 overflow-hidden text-ellipsis text-right" }, R = /* @__PURE__ */ b({
  __name: "EnvironmentVariableDropdown",
  props: {
    query: {},
    activeParsedEnvironments: {},
    environments: {},
    router: {}
  },
  emits: ["select"],
  setup(p, { emit: v }) {
    const n = p, m = v, c = y(!0), f = () => {
      const e = _.value.params.workspace;
      h(`/workspace/${e}/environment/default`), c.value = !1;
    }, { push: h, currentRoute: _ } = n.router, g = new $(n.activeParsedEnvironments.value, {
      keys: ["key", "value"]
    }), l = E(() => {
      const e = n.query;
      if (!e)
        return n.activeParsedEnvironments.value.slice(-4);
      const s = g.search(e);
      return s.length > 0 ? s.map((t) => t.item) : [];
    }), k = (e) => {
      m("select", e);
    }, w = (e) => "_scalarEnvId" in e ? `bg-${n.environments[e._scalarEnvId].color}` : "bg-grey";
    return (e, s) => (r(), o("dialog", {
      id: "env-dialog",
      class: "z-10 w-60 rounded border bg-b-1 p-1",
      open: c.value,
      tabindex: "0"
    }, [
      l.value.length ? (r(), o("ul", q, [
        (r(!0), o(C, null, V(l.value, (t, x) => (r(), o("li", {
          key: x,
          class: "font-code text-3xs hover:bg-b-2 flex cursor-pointer items-center justify-between gap-1.5 rounded p-1.5 transition-colors duration-150",
          onClick: (T) => k(t.key)
        }, [
          a("div", D, [
            a("span", {
              class: B(["h-2.5 w-2.5 min-w-2.5 rounded-full", w(t)])
            }, null, 2),
            i(" " + u(t.key), 1)
          ]),
          a("span", F, u(t.value), 1)
        ], 8, A))), 128))
      ])) : (r(), I(d(P), {
        key: 1,
        class: "font-code text-3xs hover:bg-b-2 flex h-7 w-full justify-start gap-2 px-2 transition-colors duration-150",
        variant: "secondary",
        onClick: f
      }, {
        default: z(() => [
          N(d(S), {
            class: "w-2",
            icon: "Add",
            size: "xs"
          }),
          i(" Add variable ")
        ]),
        _: 1
      }))
    ], 8, j));
  }
});
export {
  R as default
};
