// @scalar/api-client/dist/views/Environment/EnvironmentColors.vue.js
import { defineComponent as c, openBlock as o, createElementBlock as l, Fragment as n, renderList as t, createElementVNode as a, normalizeClass as g, createBlock as i, unref as b, createCommentVNode as C } from "vue";
import { ScalarIcon as m } from "@scalar/components";
const p = { class: "flex flex-row gap-1.5 space-x-1" }, u = ["onClick"], v = /* @__PURE__ */ c({
  __name: "EnvironmentColors",
  props: {
    activeColor: {}
  },
  emits: ["select"],
  setup(d) {
    const s = [
      { color: "grey", bgClass: "bg-grey" },
      { color: "red", bgClass: "bg-red" },
      { color: "yellow", bgClass: "bg-yellow" },
      { color: "green", bgClass: "bg-green" },
      { color: "orange", bgClass: "bg-orange" },
      { color: "blue", bgClass: "bg-blue" },
      { color: "text-c-2", bgClass: "bg-c-2" },
      { color: "purple", bgClass: "bg-purple" },
      { color: "pink", bgClass: "bg-pink" }
    ];
    return (r, f) => (o(), l("div", p, [
      (o(), l(n, null, t(s, (e) => a("div", {
        key: e.color,
        class: g(["flex h-5 w-5 cursor-pointer items-center justify-center rounded-full", e.bgClass]),
        onClick: (k) => r.$emit("select", e.color)
      }, [
        r.activeColor === e.color ? (o(), i(b(m), {
          key: 0,
          icon: "Checkmark",
          size: "xs"
        })) : C("", !0)
      ], 10, u)), 64))
    ]));
  }
});
export {
  v as default
};
