// @scalar/api-client/dist/views/Environment/Environment.vue2.js
import { defineComponent as D, ref as c, onMounted as N, openBlock as r, createBlock as f, withCtx as o, createVNode as a, createElementVNode as _, createElementBlock as d, Fragment as C, renderList as S, unref as u, createTextVNode as $, createSlots as h, createCommentVNode as I, toDisplayString as L, withKeys as B, nextTick as R } from "vue";
import A from "../../components/CodeInput/CodeInput.vue.js";
import K from "../../components/Sidebar/Sidebar.vue.js";
/* empty css                                    */
import M from "../../components/Sidebar/SidebarButton.vue.js";
import T from "../../components/Sidebar/SidebarList.vue.js";
import U from "../../components/Sidebar/SidebarListElement.vue.js";
import F from "../../components/SubpageHeader.vue.js";
import H from "../../components/ViewLayout/ViewLayout.vue.js";
import J from "../../components/ViewLayout/ViewLayoutContent.vue.js";
import O from "../../components/ViewLayout/ViewLayoutSection.vue.js";
import { useWorkspace as W } from "../../store/workspace.js";
import { nanoid as j } from "nanoid";
import { useRouter as q } from "vue-router";
import z from "./EnvironmentColors.vue.js";
const G = { class: "flex-1" }, P = ["value"], Q = { class: "colors ml-auto" }, de = /* @__PURE__ */ D({
  __name: "Environment",
  setup(X) {
    const v = q(), { environments: l, environmentMutators: s } = W(), e = c(null), p = c(null), m = c(!1);
    function y() {
      const t = {
        name: "New Environment",
        uid: j(),
        color: "grey",
        raw: JSON.stringify({ exampleKey: "exampleValue" }, null, 2),
        parsed: [],
        isDefault: !1
      };
      s.add(t), e.value = t.uid, v.push(e.value);
    }
    function E(t) {
      e.value && s.edit(e.value, "raw", t);
    }
    const k = (t) => {
      s.delete(t), e.value === t && (e.value = null);
    }, g = (t) => {
      e.value && (l[e.value].color = t);
    }, b = () => {
      v.currentRoute.value.params.environment === "default" && (e.value = l.default.uid);
    }, x = () => {
      e.value && !l[e.value].isDefault && (m.value = !0, R(() => {
        var t;
        (t = p.value) == null || t.focus();
      }));
    }, V = (t) => {
      const n = t.target.value;
      e.value && !l[e.value].isDefault && s.edit(e.value, "name", n);
    };
    return N(b), (t, i) => (r(), f(F, null, {
      default: o(() => [
        a(H, null, {
          default: o(() => [
            a(K, { title: "Environment" }, {
              content: o(() => [
                _("div", G, [
                  a(T, null, {
                    default: o(() => [
                      (r(!0), d(C, null, S(u(l), (n) => (r(), f(U, {
                        key: n.uid,
                        class: "text-xs",
                        variable: {
                          name: n.name,
                          uid: n.uid,
                          color: n.color,
                          isDefault: n.isDefault
                        },
                        warningMessage: "Are you sure you want to delete this environment?",
                        onClick: (w) => e.value = n.uid,
                        onDelete: (w) => k(n.uid)
                      }, null, 8, ["variable", "onClick", "onDelete"]))), 128))
                    ]),
                    _: 1
                  })
                ])
              ]),
              button: o(() => [
                a(M, { click: y }, {
                  title: o(() => [
                    $("Add Environment Variable")
                  ]),
                  _: 1
                })
              ]),
              _: 1
            }),
            a(J, { class: "flex-1" }, {
              default: o(() => [
                a(O, null, h({
                  default: o(() => [
                    e.value ? (r(), f(A, {
                      key: 0,
                      class: "px-2 py-2.5",
                      lineNumbers: "",
                      modelValue: u(l)[e.value].raw,
                      "onUpdate:modelValue": E
                    }, null, 8, ["modelValue"])) : I("", !0)
                  ]),
                  _: 2
                }, [
                  e.value ? {
                    name: "title",
                    fn: o(() => [
                      !m.value || u(l)[e.value].isDefault ? (r(), d("span", {
                        key: 0,
                        onDblclick: x
                      }, L(u(l)[e.value].name), 33)) : (r(), d("input", {
                        key: 1,
                        ref_key: "nameInputRef",
                        ref: p,
                        class: "ring-1 ring-offset-4 ring-b-outline rounded",
                        spellcheck: "false",
                        type: "text",
                        value: u(l)[e.value].name,
                        onBlur: i[0] || (i[0] = (n) => m.value = !1),
                        onInput: V,
                        onKeyup: i[1] || (i[1] = B((n) => m.value = !1, ["enter"]))
                      }, null, 40, P)),
                      _("div", Q, [
                        a(z, {
                          activeColor: u(l)[e.value].color,
                          onSelect: g
                        }, null, 8, ["activeColor"])
                      ])
                    ]),
                    key: "0"
                  } : void 0
                ]), 1024)
              ]),
              _: 1
            })
          ]),
          _: 1
        })
      ]),
      _: 1
    }));
  }
});
export {
  de as default
};
