// @scalar/api-client/dist/views/Environment/EnvironmentVariableDropdown.vue.js
import { defineComponent as B, ref as h, computed as w, onMounted as R, createBlock as b, createCommentVNode as y, openBlock as n, unref as l, withCtx as k, createElementVNode as c, normalizeStyle as g, createElementBlock as u, Fragment as D, renderList as T, normalizeClass as j, createTextVNode as V, toDisplayString as C, createVNode as q } from "vue";
import { ScalarTeleport as F, ScalarButton as G } from "@scalar/components";
import { ScalarIconGlobe as K, ScalarIconPlus as O } from "@scalar/icons";
import { onClickOutside as L } from "@vueuse/core";
import M from "fuse.js";
import { useRouter as Q } from "vue-router";
import { getEnvColor as W } from "../../libs/env-helpers.js";
import { PathId as S } from "../../routes.js";
import { useActiveEntities as $ } from "../../store/active-entities.js";
import { parseEnvVariables as _ } from "../../libs/environment-parser.js";
const H = {
  key: 0,
  class: "gap-1/2 flex flex-col"
}, J = ["onClick"], U = { class: "flex items-center gap-2 whitespace-nowrap" }, X = { class: "w-20 overflow-hidden text-right text-ellipsis whitespace-nowrap" }, de = /* @__PURE__ */ B({
  __name: "EnvironmentVariableDropdown",
  props: {
    query: {},
    environment: {},
    envVariables: {},
    dropdownPosition: {}
  },
  emits: ["select"],
  setup(d, { expose: E, emit: z }) {
    const s = d, I = z, f = h(!0), p = h(null), r = h(0), m = Q(), { activeCollection: v } = $(), N = () => {
      if (!m)
        return;
      const { currentRoute: t, push: o } = m, e = t.value.params.workspace;
      !v.value || v.value.info?.title === "Drafts" ? o({
        name: "environment.default",
        params: {
          [S.Workspace]: e
        }
      }) : o({
        name: "collection.environment",
        params: {
          [S.Collection]: v.value.uid
        }
      }), f.value = !1;
    }, P = new M(_(s.envVariables), {
      keys: ["key", "value"]
    }), i = w(() => {
      const t = s.query;
      if (!t)
        return _(s.envVariables).slice(-4).filter(({ key: e, value: a }) => e !== "" || a !== "");
      const o = P.search(t, { limit: 10 });
      return o.length > 0 ? o.map((e) => e.item).filter(({ key: e, value: a }) => e !== "" || a !== "") : [];
    }), x = (t) => {
      I("select", t);
    };
    E({
      handleArrowKey: (t) => {
        const o = t === "up" ? -1 : 1, e = i.value.length;
        e !== 0 && (r.value = (r.value + o + e) % e);
      },
      handleSelect: () => {
        if (r.value >= 0) {
          const t = i.value[r.value];
          t && x(t.key);
        }
      }
    }), R(() => {
      r.value = 0;
    });
    const A = w(() => ({
      left: (s.dropdownPosition?.left ?? 0) + "px",
      // Add a 5px offset from the editor
      top: (s.dropdownPosition?.top ?? 0) + 5 + "px"
    }));
    return L(
      p,
      () => {
        f.value = !1;
      },
      { ignore: [p] }
    ), (t, o) => f.value ? (n(), b(l(F), {
      key: 0,
      class: "scalar-client"
    }, {
      default: k(() => [
        c("div", {
          ref_key: "dropdownRef",
          ref: p,
          class: "custom-scroll fixed top-0 left-0 flex max-h-[60svh] w-56 flex-col rounded border p-0.75",
          style: g(A.value)
        }, [
          i.value.length ? (n(), u("ul", H, [
            (n(!0), u(D, null, T(i.value, (e, a) => (n(), u("li", {
              key: e.key,
              class: j(["font-code text-xxs hover:bg-b-2 flex h-8 cursor-pointer items-center justify-between gap-1.5 rounded p-1.5 transition-colors duration-150", { "bg-b-2": a === r.value }]),
              onClick: (ee) => x(e.key)
            }, [
              c("div", U, [
                e.source === "collection" && d.environment.name !== "No Environment" ? (n(), u("span", {
                  key: 0,
                  class: "h-2.25 w-2.25 min-w-2.25 rounded-full",
                  style: g({
                    backgroundColor: l(W)(d.environment)
                  })
                }, null, 4)) : (n(), b(l(K), {
                  key: 1,
                  class: "-ml-0.25 size-2.5",
                  icon: "Globe"
                })),
                V(" " + C(e.key), 1)
              ]),
              c("span", X, C(e.value), 1)
            ], 10, J))), 128))
          ])) : l(m) ? (n(), b(l(G), {
            key: 1,
            class: "font-code text-xxs bg-b-inherit hover:bg-b-2 flex h-8 w-full justify-start gap-2 px-1.5 transition-colors duration-150",
            variant: "outlined",
            onClick: N
          }, {
            default: k(() => [
              q(l(O), { class: "size-3" }),
              o[0] || (o[0] = V(" Add Variable ", -1))
            ]),
            _: 1
          })) : y("", !0),
          o[1] || (o[1] = c("div", { class: "bg-b-1 brightness-lifted absolute inset-0 -z-1 rounded shadow-lg" }, null, -1))
        ], 4)
      ]),
      _: 1
    })) : y("", !0);
  }
});
export {
  de as default
};
