// @scalar/api-client/dist/v2/features/app/app-state.js
import { isDefined as A } from "@scalar/helpers/array/is-defined";
import { sortByOrder as Y } from "@scalar/helpers/array/sort-by-order";
import { isHttpMethod as j } from "@scalar/helpers/http/is-http-method";
import { createSidebarState as q, generateReverseIndex as z } from "@scalar/sidebar";
import { createWorkspaceStore as C } from "@scalar/workspace-store/client";
import { createWorkspaceEventBus as J } from "@scalar/workspace-store/events";
import { generateUniqueValue as M } from "@scalar/workspace-store/helpers/generate-unique-value";
import { getParentEntry as m } from "@scalar/workspace-store/navigation";
import { createWorkspaceStorePersistence as G } from "@scalar/workspace-store/persistence";
import { persistencePlugin as Q } from "@scalar/workspace-store/plugins/client";
import { extensions as X } from "@scalar/workspace-store/schemas/extensions";
import { shallowRef as E, computed as i, ref as p, watch as Z } from "vue";
import { getActiveEnvironment as ee } from "../../helpers/get-active-environment.js";
import { getTabDetails as te } from "../../helpers/get-tab-details.js";
import { slugify as ae } from "../../helpers/slugify.js";
import { workspaceStorage as ne } from "../../helpers/storage.js";
import { initializeAppEventHandlers as re } from "./app-events.js";
const oe = 1e3, h = J({
  debug: !1
}), v = p(void 0), k = p(void 0), w = p(void 0), B = p(void 0), O = p(void 0), $ = p(void 0), f = p(!1), l = E(null);
Z(l, (e) => {
  e && e.afterEach((t) => ge(t));
});
const d = i(() => l.value?.currentRoute.value ?? null);
function c(e, t) {
  const a = t?.params[e];
  if (typeof a == "string")
    return e === "method" ? a && j(a) ? a : void 0 : decodeURIComponent(a);
}
const W = E(null), I = p([]), o = E(null), T = i(() => o.value?.workspace.documents[w.value ?? ""] || null), se = i(() => ee(o.value, T.value)), { workspace: x } = await G(), g = (e, t) => `${e}/${t}`;
I.value = await x.getAll().then(
  (e) => e.map(({ teamUid: t, namespace: a, slug: n, name: u }) => ({
    id: g(a, n),
    teamUid: t,
    namespace: a,
    slug: n,
    label: u
  }))
);
const ce = async ({ namespace: e, slug: t }) => C({
  plugins: [await Q({ workspaceId: `${e}-${t}`, debounceDelay: oe })]
}), ue = async (e, t) => {
  const a = await x.getItem({ namespace: e, slug: t });
  if (!a)
    return {
      success: !1
    };
  const n = await ce({ namespace: e, slug: t });
  return n.loadWorkspace(a.workspace), W.value = { id: g(a.namespace, a.slug), label: a.name }, o.value = n, {
    success: !0,
    workspace: n.workspace
  };
}, le = async ({
  name: e,
  teamUid: t,
  namespace: a,
  slug: n
}) => {
  const u = C();
  await u.addDocument({
    name: "drafts",
    document: {
      openapi: "3.1.0",
      info: {
        title: "Drafts",
        version: "1.0.0"
      },
      paths: {
        "/": {
          get: {}
        }
      },
      "x-scalar-original-document-hash": "drafts",
      "x-scalar-icon": "interface-edit-tool-pencil"
    }
  });
  const s = await x.setItem(
    { namespace: a, slug: n },
    {
      name: e,
      teamUid: t,
      workspace: u.exportWorkspace()
    }
  );
  return I.value.push({
    id: g(s.namespace, s.slug),
    teamUid: s.teamUid,
    namespace: s.namespace,
    slug: s.slug,
    label: s.name
  }), s;
}, L = async (e, t) => {
  await l.value?.push({
    name: "workspace.environment",
    params: { namespace: e, workspaceSlug: t }
  });
}, N = async ({
  key: e,
  name: t
}) => {
  o.value = null;
  const a = await M({
    defaultValue: e?.slug ?? t,
    // Use the provided id if it exists, otherwise use the name
    validation: async (s) => !await x.has({ namespace: e?.namespace ?? "local", slug: s }),
    maxRetries: 100,
    transformation: ae
  });
  if (!a)
    return;
  const n = {
    teamUid: e?.teamUid,
    namespace: e?.namespace,
    slug: a,
    name: t
  }, u = await le(n);
  return await L(u.namespace, u.slug), u;
}, ie = async (e, t) => {
  o.value = null, f.value = !0;
  const a = await ue(e, t);
  if (a.success) {
    const u = a.workspace["x-scalar-active-tab"] ?? 0, s = a.workspace["x-scalar-tabs"], D = s?.[u];
    D && await l.value?.replace(D.path), s && u >= s.length && h.emit("tabs:update:tabs", {
      "x-scalar-active-tab": 0
    }), s || h.emit("tabs:update:tabs", {
      "x-scalar-tabs": [U(d.value)],
      "x-scalar-active-tab": 0
    }), f.value = !1;
    return;
  }
  const n = await N({
    name: "Default Workspace",
    key: { slug: "default" }
  });
  if (f.value = !1, !n)
    return console.error("Failed to create the default workspace, something went wrong, can not load the workspace");
  r.reset();
}, P = i(() => {
  const e = o.value;
  if (!e)
    return [];
  const t = e.workspace["x-scalar-order"] ?? Object.keys(e.workspace.documents);
  return Y(Object.keys(e.workspace.documents), t, (a) => a).map((a) => e.workspace.documents[a]?.["x-scalar-navigation"]).filter(A);
}), r = q(P), S = ({
  document: e,
  path: t,
  method: a,
  example: n
}) => JSON.stringify([e, t, a, n].filter(A)), R = i(
  () => z({
    items: P.value,
    nestedKey: "children",
    filter: (e) => e.type === "document" || e.type === "operation" || e.type === "example",
    getId: (e) => {
      const t = m("document", e), a = m("operation", e);
      return S({
        document: t?.name ?? "",
        path: a?.path,
        method: a?.method,
        example: e.type === "example" ? e.name : void 0
      });
    }
  })
), b = (e) => {
  const t = R.value.get(S(e));
  return t || R.value.get(
    S({
      document: e.document,
      path: e.path,
      method: e.method
    })
  );
}, _ = (e) => {
  const t = r.getEntryById(e);
  if (!t) {
    console.warn(`Could not find sidebar entry with id ${e} to select`);
    return;
  }
  if (t.type === "document") {
    if (r.selectedItem.value === e) {
      r.setExpanded(e, !r.isExpanded(e));
      return;
    }
    return r.setSelected(e), r.setExpanded(e, !0), l.value?.push({
      name: "document.overview",
      params: { documentSlug: t.name }
    });
  }
  if (t.type === "operation") {
    if (r.isSelected(e)) {
      r.setExpanded(e, !r.isExpanded(e));
      return;
    }
    const a = t.children?.find((n) => n.type === "example");
    return a ? (r.setSelected(a.id), r.setExpanded(a.id, !0)) : r.setSelected(e), l.value?.push({
      name: "example",
      params: {
        documentSlug: m("document", t)?.name,
        pathEncoded: encodeURIComponent(t.path),
        method: t.method,
        exampleName: a?.name ?? "default"
      }
    });
  }
  if (t.type === "example") {
    r.setSelected(e);
    const a = m("operation", t);
    return l.value?.push({
      name: "example",
      params: {
        documentSlug: m("document", t)?.name,
        pathEncoded: encodeURIComponent(a?.path ?? ""),
        method: a?.method,
        exampleName: t.name
      }
    });
  }
  if (t.type === "text")
    return l.value?.push({
      name: "document.overview",
      params: {
        documentSlug: m("document", t)?.name
      }
    });
  r.setExpanded(e, !r.isExpanded(e));
}, pe = async () => {
  if (!o.value)
    return;
  const e = o.value.workspace["x-scalar-active-tab"] ?? 0, t = o.value.workspace["x-scalar-tabs"]?.[e];
  t && await l.value?.replace(t.path);
}, F = (e) => {
  e && o.value?.buildSidebar(e);
}, me = (e) => {
  const t = T.value?.["x-scalar-navigation"]?.name;
  if (!t)
    return;
  const a = b({
    document: t,
    path: e.path,
    method: e.method,
    example: e.exampleKey
  });
  (!a || a.type !== "example") && (F(t), d.value && V(d.value));
}, de = 288, ve = i(() => o.value?.workspace?.["x-scalar-sidebar-width"] ?? de), fe = (e) => o.value?.update("x-scalar-sidebar-width", e), y = p(!0), he = "x-scalar-tabs", we = "x-scalar-active-tab", U = (e) => {
  const t = c("method", e), a = c("pathEncoded", e), n = c("documentSlug", e), u = c("workspaceSlug", e);
  return {
    ...te({
      workspace: u,
      document: n,
      path: a,
      method: t,
      getEntryByLocation: b
    }),
    path: d.value?.path ?? ""
  };
}, K = i(() => o.value?.workspace[he] ?? [U(d.value)]), xe = i(() => o.value?.workspace[we] ?? 0), H = async (e) => {
  const t = K.value[e];
  if (!t) {
    console.warn(`Cannot copy URL: tab at index ${e} does not exist`);
    return;
  }
  const a = `${window.location.origin}${t.path}`;
  try {
    await navigator.clipboard.writeText(a);
  } catch (n) {
    console.error("Failed to copy URL to clipboard:", n);
  }
}, ge = (e) => {
  const t = c("workspaceSlug", e), a = c("documentSlug", e);
  if (v.value = c("namespace", e), k.value = t, w.value = a, B.value = c("method", e), O.value = c("pathEncoded", e), $.value = c("exampleName", e), !(!v.value || !t)) {
    if (e.path !== "" && ne.setCurrentPath(e.path), g(v.value, t) !== W.value?.id)
      return ie(v.value, t);
    if (a && a !== o.value?.workspace[X.workspace.activeDocument])
      return o?.value?.update("x-scalar-active-document", a);
    be(e), V(e);
  }
}, be = (e) => {
  const t = o.value?.workspace["x-scalar-tabs"] ?? [], a = o.value?.workspace["x-scalar-active-tab"] ?? 0, n = t[a];
  !n || n.path === e.path || (t[a] = U(e));
}, V = (e) => {
  const t = c("documentSlug", e);
  if (!t) {
    r.setSelected(null);
    return;
  }
  const a = b({
    document: t,
    path: c("pathEncoded", e),
    method: c("method", e),
    example: c("exampleName", e)
  });
  a && (r.setSelected(a.id), r.setExpanded(a.id, !0));
};
re({
  eventBus: h,
  router: l,
  store: o,
  navigateToCurrentTab: pe,
  rebuildSidebar: F,
  onAfterExampleCreation: me,
  onSelectSidebarItem: _,
  onCopyTabUrl: (e) => H(e),
  onToggleSidebar: () => y.value = !y.value
});
function Pe(e) {
  return e && (l.value = e), {
    /** Active workspace store */
    store: o,
    sidebar: {
      state: r,
      width: ve,
      isOpen: y,
      handleSelectItem: _,
      handleSidebarWidthUpdate: fe,
      getEntryByLocation: b
    },
    tabs: {
      state: K,
      activeTabIndex: xe,
      copyTabUrl: H
    },
    workspace: {
      create: N,
      workspaceList: I,
      activeWorkspace: W,
      navigateToWorkspace: L,
      isOpen: i(() => !!(k.value && !w.value))
    },
    eventBus: h,
    router: l,
    currentRoute: d,
    loading: f,
    activeEntities: {
      namespace: v,
      workspaceSlug: k,
      documentSlug: w,
      path: O,
      method: B,
      exampleName: $
    },
    environment: se,
    document: T
  };
}
export {
  Pe as useAppState
};
