// @scalar/api-client/dist/v2/features/app/app-events.js
import { computed as v } from "vue";
import { initializeWorkspaceEventHandlers as E } from "../../workspace-events.js";
function N({
  eventBus: m,
  store: r,
  router: o,
  rebuildSidebar: p,
  navigateToCurrentTab: c,
  onSelectSidebarItem: d,
  onAfterExampleCreation: s,
  onCopyTabUrl: i,
  onToggleSidebar: f
}) {
  const u = v(() => o.value?.currentRoute?.value), l = ({
    documentName: e,
    path: a,
    method: t,
    exampleName: n
  }) => !(e !== void 0 && e !== u.value?.params.documentSlug || a !== void 0 && encodeURIComponent(a) !== u.value?.params.pathEncoded || t !== void 0 && t !== u.value?.params.method || n !== void 0 && n !== u.value?.params.exampleName);
  E({
    eventBus: m,
    store: r,
    hooks: {
      //------------------------------------------------------------------------------------
      // Document Related Hooks
      //------------------------------------------------------------------------------------
      "document:delete:document": {
        onAfterExecute: async (e) => {
          u?.value?.params.documentSlug === e.name && await o.value?.push({
            name: "workspace.environment"
          });
        }
      },
      //------------------------------------------------------------------------------------
      // Operation Related Hooks
      //------------------------------------------------------------------------------------
      "operation:update:pathMethod": {
        onBeforeExecute: (e) => ({
          ...e,
          callback: async (a) => {
            a === "success" && (await o.value?.replace({
              name: "example",
              params: {
                method: e.payload.method,
                pathEncoded: encodeURIComponent(e.payload.path),
                exampleName: u.value?.params.exampleName
              }
            }), p(r.value?.workspace.activeDocument?.["x-scalar-navigation"]?.name)), e.callback(a);
          }
        })
      },
      "operation:upsert:parameter": {
        onAfterExecute: (e) => s(e.meta)
      },
      "operation:update:extra-parameters": {
        onAfterExecute: (e) => s(e.meta)
      },
      "operation:reload:history": {
        onAfterExecute: (e) => s({ ...e.meta, exampleKey: "draft" })
      },
      "operation:delete:operation": {
        onAfterExecute: async (e) => {
          p(e.documentName);
          const {
            documentName: a,
            meta: { path: t, method: n }
          } = e;
          l({
            documentName: a,
            path: t,
            method: n
          }) && await o.value?.replace({
            name: "document.overview",
            params: {
              documentSlug: a
            }
          });
        }
      },
      "operation:delete:example": {
        onAfterExecute: async (e) => {
          p(e.documentName);
          const {
            documentName: a,
            meta: { path: t, method: n, exampleKey: x }
          } = e;
          l({
            documentName: a,
            path: t,
            method: n,
            exampleName: x
          }) && await o.value?.replace({
            name: "example",
            params: {
              pathEncoded: encodeURIComponent(t),
              method: n,
              documentSlug: a,
              exampleName: "default"
            }
          });
        }
      },
      //------------------------------------------------------------------------------------
      // Operation Request Body Related Hooks
      //------------------------------------------------------------------------------------
      "operation:update:requestBody:value": {
        onAfterExecute: (e) => s(e.meta)
      },
      "operation:update:requestBody:formValue": {
        onAfterExecute: (e) => s(e.meta)
      },
      //------------------------------------------------------------------------------------
      // Tag Related Event Hooks
      //------------------------------------------------------------------------------------
      "tag:create:tag": {
        onAfterExecute: (e) => p(e.documentName)
      },
      "tag:delete:tag": {
        onAfterExecute: (e) => p(e.documentName)
      },
      //------------------------------------------------------------------------------------
      // Tabs Related Event Hooks
      //------------------------------------------------------------------------------------
      "tabs:add:tab": {
        onAfterExecute: c
      },
      "tabs:close:tab": {
        onAfterExecute: c
      },
      "tabs:focus:tab": {
        onAfterExecute: c
      },
      "tabs:focus:tab-last": {
        onAfterExecute: c
      },
      "tabs:navigate:previous": {
        onAfterExecute: c
      },
      "tabs:navigate:next": {
        onAfterExecute: c
      },
      "tabs:update:tabs": {
        onAfterExecute: c
      }
    }
  }), m.on("scroll-to:nav-item", ({ id: e }) => d(e)), m.on("ui:toggle:sidebar", f), m.on("ui:route:page", ({ name: e }) => o.value?.push({ name: e })), m.on("ui:route:example", async ({ exampleName: e, callback: a }) => {
    const t = await o.value?.replace({
      name: "example",
      params: {
        exampleName: e
      }
    });
    a(t ? "error" : "success");
  }), m.on("tabs:copy:url", (e) => i(e.index));
}
export {
  N as initializeAppEventHandlers
};
