// @scalar/api-client/dist/v2/features/app/App.vue.js
import { defineComponent as S, computed as m, createElementBlock as b, createBlock as l, unref as t, openBlock as c, Fragment as W, createElementVNode as p, createVNode as n, withCtx as d, renderSlot as T, normalizeProps as h, guardReactiveProps as y } from "vue";
import { useModal as g, ScalarTeleportRoot as C } from "@scalar/components";
import { getThemeStyles as B } from "@scalar/themes";
import { ScalarToasts as x } from "@scalar/use-toasts";
import { extensions as E } from "@scalar/workspace-store/schemas/extensions";
import { RouterView as O } from "vue-router";
import A from "./components/CreateWorkspaceModal.vue.js";
import M from "./components/SplashScreen.vue.js";
import { useDocumentWatcher as U } from "./hooks/use-document-watcher.js";
import I from "../command-palette/TheCommandPalette.vue.js";
import { useColorMode as _ } from "../../hooks/use-color-mode.js";
import { useGlobalHotKeys as D } from "../../hooks/use-global-hot-keys.js";
import L from "./components/AppSidebar.vue.js";
import N from "./components/DesktopTabs.vue.js";
import $ from "./components/WebTopNav.vue.js";
const P = ["innerHTML"], F = { class: "flex min-h-0 flex-1" }, H = { class: "bg-b-1 flex-1" }, V = {}, re = /* @__PURE__ */ S({
  ...V,
  __name: "App",
  props: {
    layout: {},
    plugins: { default: () => [] },
    getAppState: { type: Function },
    getCommandPaletteState: { type: Function }
  },
  setup(o, { expose: v }) {
    v({
      openCreateWorkspace: () => i.show()
    });
    const e = o.getAppState(), k = o.getCommandPaletteState();
    typeof window < "u" && (window.dataDumpWorkspace = () => e.store.value, window.dumpAppState = () => e), _({ workspaceStore: e.store }), D(e.eventBus, o.layout), U({
      documentName: () => e.store.value?.workspace[E.workspace.activeDocument],
      store: e.store,
      initialTimeout: 5e3
    });
    const w = m(() => {
      if (e.store.value === null)
        return "";
      const s = e.store.value.workspace["x-scalar-theme"];
      return s ? `<style>${B(s)}</style>` : "";
    }), u = (s) => {
      if (!s)
        return;
      const a = e.workspace.workspaceList.value?.find(
        (r) => r.id === s
      );
      a && e.workspace.navigateToWorkspace(a.namespace, a.slug);
    }, i = g(), f = m(() => ({
      documentSlug: e.activeEntities.documentSlug.value ?? "",
      document: e.store.value?.workspace.activeDocument ?? null,
      environment: e.environment.value,
      eventBus: e.eventBus,
      exampleName: e.activeEntities.exampleName.value,
      layout: o.layout,
      method: e.activeEntities.method.value,
      path: e.activeEntities.path.value,
      workspaceStore: e.store.value,
      activeWorkspace: e.workspace.activeWorkspace.value,
      plugins: o.plugins,
      securitySchemes: e.document.value?.components?.securitySchemes ?? {}
    }));
    return (s, a) => t(e).store.value !== null && t(e).workspace.activeWorkspace.value !== null && !t(e).loading.value ? (c(), b(W, { key: 0 }, [
      p("div", { innerHTML: w.value }, null, 8, P),
      n(t(C), null, {
        default: d(() => [
          n(t(x)),
          o.layout === "desktop" ? (c(), l(N, {
            key: 0,
            activeTabIndex: t(e).tabs.activeTabIndex.value,
            eventBus: t(e).eventBus,
            tabs: t(e).tabs.state.value
          }, null, 8, ["activeTabIndex", "eventBus", "tabs"])) : (c(), l($, {
            key: 1,
            activeWorkspace: t(e).workspace.activeWorkspace.value,
            workspaces: t(e).workspace.workspaceList.value,
            "onCreate:workspace": a[0] || (a[0] = (r) => t(i).show()),
            "onSelect:workspace": u
          }, null, 8, ["activeWorkspace", "workspaces"])),
          p("main", F, [
            n(L, {
              isSidebarOpen: t(e).sidebar.isOpen.value,
              "onUpdate:isSidebarOpen": a[1] || (a[1] = (r) => t(e).sidebar.isOpen.value = r),
              activeWorkspace: t(e).workspace.activeWorkspace.value,
              eventBus: t(e).eventBus,
              isWorkspaceOpen: t(e).workspace.isOpen.value,
              layout: o.layout,
              sidebarState: t(e).sidebar.state,
              sidebarWidth: t(e).sidebar.width.value,
              store: t(e).store.value,
              workspaces: t(e).workspace.workspaceList.value,
              "onClick:workspace": t(e).workspace.navigateToWorkspace,
              "onCreate:workspace": a[2] || (a[2] = (r) => t(i).show()),
              "onSelect:workspace": u,
              onSelectItem: t(e).sidebar.handleSelectItem,
              "onUpdate:sidebarWidth": t(e).sidebar.handleSidebarWidthUpdate
            }, {
              sidebarMenuActions: d(() => [
                T(s.$slots, "sidebar-menu-actions")
              ]),
              _: 3
            }, 8, ["isSidebarOpen", "activeWorkspace", "eventBus", "isWorkspaceOpen", "layout", "sidebarState", "sidebarWidth", "store", "workspaces", "onClick:workspace", "onSelectItem", "onUpdate:sidebarWidth"]),
            n(A, {
              state: t(i),
              "onCreate:workspace": a[3] || (a[3] = (r) => t(e).workspace.create(r))
            }, null, 8, ["state"]),
            n(I, {
              eventBus: t(e).eventBus,
              paletteState: t(k),
              workspaceStore: t(e).store.value
            }, null, 8, ["eventBus", "paletteState", "workspaceStore"]),
            p("div", H, [
              n(t(O), h(y(f.value)), null, 16)
            ])
          ])
        ]),
        _: 3
      })
    ], 64)) : (c(), l(M, { key: 1 }));
  }
});
export {
  re as default
};
