// @scalar/components/dist/components/ScalarVirtualText/ScalarVirtualText.vue.js
import { defineComponent as b, ref as o, computed as t, onMounted as w, onBeforeUnmount as E, watchEffect as k, createElementBlock as v, openBlock as f, normalizeClass as h, createElementVNode as $, normalizeStyle as m, Fragment as z, renderList as L, toDisplayString as R } from "vue";
const I = /* @__PURE__ */ b({
  __name: "ScalarVirtualText",
  props: {
    text: {},
    lineHeight: { default: 20 },
    containerClass: { default: "" },
    contentClass: { default: "" },
    lineClass: { default: "" }
  },
  setup(l) {
    const e = l, n = o(null), s = o(null), r = o(0), d = o(0), a = t(() => e.text.split(`
`)), x = t(() => a.value.length * e.lineHeight), i = t(
      () => Math.floor(r.value / e.lineHeight)
    ), p = t(
      () => Math.min(
        Math.ceil(
          (r.value + d.value) / e.lineHeight
        ),
        a.value.length
      )
    ), H = t(() => {
      const g = Math.max(0, i.value - 10), u = Math.min(a.value.length, p.value + 10);
      return a.value.slice(g, u);
    }), y = t(() => ({
      height: `${x.value}px`,
      transform: `translateY(${Math.max(0, i.value - 10) * e.lineHeight}px)`
    })), C = () => n.value && (r.value = n.value.scrollTop), c = () => n.value && (d.value = n.value.clientHeight);
    return w(() => {
      c(), window.addEventListener("resize", c);
    }), E(() => {
      window.removeEventListener("resize", c);
    }), k(() => {
      s.value && (s.value.style.transform = `translateY(${Math.max(0, i.value - 10) * e.lineHeight}px)`);
    }), (M, g) => (f(), v("div", {
      ref_key: "containerRef",
      ref: n,
      class: h(["scalar-virtual-text overflow-auto", l.containerClass]),
      onScroll: C
    }, [
      $("code", {
        ref_key: "contentRef",
        ref: s,
        class: h(["scalar-virtual-text-content", l.contentClass]),
        style: m(y.value)
      }, [
        (f(!0), v(z, null, L(H.value, (u, S) => (f(), v("div", {
          key: i.value + S,
          class: h(["scalar-virtual-text-line", l.lineClass]),
          style: m({
            height: `${e.lineHeight}px`,
            lineHeight: `${e.lineHeight}px`
          })
        }, R(u), 7))), 128))
      ], 6)
    ], 34));
  }
});
export {
  I as default
};
