// @scalar/components/dist/components/ScalarCombobox/ScalarComboboxPopover.vue.js
import { defineComponent as m, ref as u, openBlock as c, createBlock as v, unref as o, withCtx as t, createVNode as n, mergeProps as y, renderSlot as a, createElementVNode as w } from "vue";
import { Popover as _, PopoverPanel as b, PopoverButton as h } from "@headlessui/vue";
import B from "../ScalarFloating/ScalarFloating.vue.js";
const g = /* @__PURE__ */ w("div", { class: "absolute inset-0 -z-1 rounded bg-b-1 shadow-md brightness-lifted" }, null, -1), O = /* @__PURE__ */ m({
  inheritAttrs: !1,
  __name: "ScalarComboboxPopover",
  props: {
    placement: {},
    resize: { type: Boolean },
    isOpen: { type: Boolean },
    teleport: { type: [Boolean, String] }
  },
  setup(k, { expose: p }) {
    const l = u(null), i = (e) => {
      var r;
      ["ArrowUp", "ArrowDown"].includes(e.key) && (e.preventDefault(), (r = e.target) == null || r.dispatchEvent(new KeyboardEvent("keydown", { key: "Enter" })));
    };
    return p({ popoverButtonRef: l }), (e, r) => (c(), v(o(_), { as: "template" }, {
      default: t(({ open: s }) => [
        n(o(B), {
          isOpen: s ?? e.isOpen,
          placement: e.placement ?? "bottom-start",
          resize: e.resize,
          teleport: e.teleport
        }, {
          floating: t(({ width: d }) => [
            n(o(b), y({
              class: "relative flex w-40 flex-col rounded border text-sm",
              focus: "",
              style: { width: d }
            }, e.$attrs), {
              default: t(({ close: f }) => [
                a(e.$slots, "popover", {
                  close: f,
                  open: s
                }),
                g
              ]),
              _: 2
            }, 1040, ["style"])
          ]),
          default: t(() => [
            n(o(h), {
              ref_key: "popoverButtonRef",
              ref: l,
              as: "template",
              onKeydown: i
            }, {
              default: t(() => [
                a(e.$slots, "default")
              ]),
              _: 3
            }, 512)
          ]),
          _: 2
        }, 1032, ["isOpen", "placement", "resize", "teleport"])
      ]),
      _: 3
    }));
  }
});
export {
  O as default
};
