// @scalar/components/dist/components/ScalarCombobox/ScalarComboboxOption.vue.js
import { defineComponent as o, createElementBlock as n, openBlock as r, normalizeClass as l, unref as t, renderSlot as s } from "vue";
import { cva as i, cx as d } from "@scalar/use-hooks/useBindCx";
const m = ["aria-selected"], b = /* @__PURE__ */ o({
  __name: "ScalarComboboxOption",
  props: {
    active: { type: Boolean },
    selected: { type: Boolean }
  },
  setup(e) {
    const a = i({
      base: [
        // Group
        "group/item",
        // Layout
        "flex min-w-0 items-center gap-1.5 rounded px-2 py-1.5 text-left",
        // Text / background style
        "truncate bg-transparent text-c-1",
        // Interaction
        "cursor-pointer hover:bg-b-2"
      ],
      variants: {
        selected: { true: "text-c-1" },
        active: { true: "bg-b-2" }
      }
    });
    return (c, u) => (r(), n("li", {
      "aria-selected": e.selected,
      class: l(t(d)(t(a)({ active: e.active, selected: e.selected }))),
      role: "option",
      tabindex: "-1"
    }, [
      s(c.$slots, "default", {
        active: e.active,
        selected: e.selected
      })
    ], 10, m));
  }
});
export {
  b as default
};
