// @scalar/components/dist/components/ScalarCombobox/ScalarComboboxMultiselect.vue.js
import { defineComponent as $, mergeModels as f, useModel as b, ref as y, createBlock as p, openBlock as u, withCtx as l, createCommentVNode as k, createSlots as V, renderSlot as d, normalizeProps as n, guardReactiveProps as m } from "vue";
import w from "./ScalarComboboxOptions.vue.js";
import z from "./ScalarComboboxPopover.vue.js";
const P = /* @__PURE__ */ $({
  __name: "ScalarComboboxMultiselect",
  props: /* @__PURE__ */ f({
    options: {},
    placeholder: {},
    placement: {},
    offset: { type: [Number, Object, Function] },
    resize: { type: Boolean },
    target: {},
    middleware: {},
    teleport: { type: [Boolean, String] }
  }, {
    modelValue: { default: [] },
    modelModifiers: {}
  }),
  emits: /* @__PURE__ */ f(["add"], ["update:modelValue"]),
  setup(e, { expose: c, emit: g }) {
    const v = g, s = b(e, "modelValue"), i = y(null);
    return c({ comboboxPopoverRef: i }), (t, a) => (u(), p(z, {
      ref_key: "comboboxPopoverRef",
      ref: i,
      middleware: e.middleware,
      offset: e.offset,
      placement: e.placement ?? "bottom-start",
      resize: e.resize,
      target: e.target,
      teleport: e.teleport
    }, {
      default: l(({ open: r }) => [
        d(t.$slots, "default", { open: r })
      ]),
      popover: l(({ open: r }) => [
        e.options?.length ? (u(), p(w, {
          key: 0,
          modelValue: s.value,
          "onUpdate:modelValue": a[0] || (a[0] = (o) => s.value = o),
          multiselect: "",
          open: r,
          options: e.options,
          placeholder: e.placeholder,
          onAdd: a[1] || (a[1] = (o) => v("add"))
        }, V({ _: 2 }, [
          t.$slots.option ? {
            name: "option",
            fn: l((o) => [
              d(t.$slots, "option", n(m(o)))
            ]),
            key: "0"
          } : void 0,
          t.$slots.group ? {
            name: "group",
            fn: l((o) => [
              d(t.$slots, "group", n(m(o)))
            ]),
            key: "1"
          } : void 0,
          t.$slots.add ? {
            name: "add",
            fn: l((o) => [
              d(t.$slots, "add", n(m(o)))
            ]),
            key: "2"
          } : void 0
        ]), 1032, ["modelValue", "open", "options", "placeholder"])) : k("", !0)
      ]),
      _: 3
    }, 8, ["middleware", "offset", "placement", "resize", "target", "teleport"]));
  }
});
export {
  P as default
};
