// @scalar/components/dist/components/ScalarCombobox/ScalarComboboxOption.vue.js
import { defineComponent as a, openBlock as n, createElementBlock as c, normalizeClass as r, unref as t, createElementVNode as o, createVNode as l, renderSlot as i } from "vue";
import { cva as p, cx as d } from "../../cva.js";
import u from "../ScalarIcon/ScalarIcon.vue.js";
const m = { class: "inline-block min-w-0 flex-1 truncate text-c-1" }, g = /* @__PURE__ */ a({
  __name: "ScalarComboboxOption",
  props: {
    active: { type: Boolean },
    selected: { type: Boolean },
    style: {}
  },
  setup(b) {
    const s = p({
      base: [
        // Layout
        "group",
        "flex min-w-0 items-center gap-1.5 rounded px-2 py-1.5 text-left",
        // Text / background style
        "truncate bg-transparent text-c-1",
        // Interaction
        "cursor-pointer hover:bg-b-2"
      ],
      variants: {
        selected: { true: "text-c-1" },
        active: { true: "bg-b-2" }
      }
    });
    return (e, f) => (n(), c("li", {
      class: r(t(d)(t(s)({ active: e.active, selected: e.selected })))
    }, [
      o("div", {
        class: r(["flex size-4 items-center justify-center p-0.75 group-hover:shadow-border", [
          e.selected ? "bg-blue text-b-1" : "text-transparent",
          e.style === "checkbox" ? "rounded" : "rounded-full"
        ]])
      }, [
        l(t(u), {
          class: "relative top-[0.5px] size-2.5",
          icon: "Checkmark",
          thickness: "2.5"
        })
      ], 2),
      o("span", m, [
        i(e.$slots, "default")
      ])
    ], 2));
  }
});
export {
  g as default
};
