// @scalar/components/dist/components/ScalarCombobox/ScalarCombobox.vue.js
import { defineComponent as f, mergeModels as s, useModel as c, createBlock as g, openBlock as $, withCtx as l, createVNode as v, createSlots as V, renderSlot as a, normalizeProps as n, guardReactiveProps as m } from "vue";
import h from "./ScalarComboboxOptions.vue.js";
import y from "./ScalarComboboxPopover.vue.js";
const B = /* @__PURE__ */ f({
  __name: "ScalarCombobox",
  props: /* @__PURE__ */ s({
    options: {},
    placeholder: {},
    placement: {},
    offset: { type: [Number, Object, Function] },
    resize: { type: Boolean },
    target: {},
    middleware: {},
    teleport: { type: [Boolean, String] }
  }, {
    modelValue: {},
    modelModifiers: {}
  }),
  emits: /* @__PURE__ */ s(["add"], ["update:modelValue"]),
  setup(e, { emit: p }) {
    const u = p, d = c(e, "modelValue");
    return (o, b) => ($(), g(y, {
      middleware: e.middleware,
      offset: e.offset,
      placement: e.placement ?? "bottom-start",
      resize: e.resize,
      target: e.target,
      teleport: e.teleport
    }, {
      default: l(({ open: r }) => [
        a(o.$slots, "default", { open: r })
      ]),
      popover: l(({ open: r, close: i }) => [
        v(h, {
          modelValue: d.value ? [d.value] : [],
          open: r,
          options: e.options,
          placeholder: e.placeholder,
          onAdd: () => (i(), u("add")),
          "onUpdate:modelValue": (t) => (i(), d.value = t[0])
        }, V({ _: 2 }, [
          o.$slots.option ? {
            name: "option",
            fn: l((t) => [
              a(o.$slots, "option", n(m(t)))
            ]),
            key: "0"
          } : void 0,
          o.$slots.group ? {
            name: "group",
            fn: l((t) => [
              a(o.$slots, "group", n(m(t)))
            ]),
            key: "1"
          } : void 0,
          o.$slots.add ? {
            name: "add",
            fn: l((t) => [
              a(o.$slots, "add", n(m(t)))
            ]),
            key: "2"
          } : void 0
        ]), 1032, ["modelValue", "open", "options", "placeholder", "onAdd", "onUpdate:modelValue"])
      ]),
      _: 3
    }, 8, ["middleware", "offset", "placement", "resize", "target", "teleport"]));
  }
});
export {
  B as default
};
