// @scalar/components/dist/components/ScalarCombobox/ScalarComboboxMultiselect.vue.js
import { defineComponent as n, ref as m, openBlock as s, createBlock as i, withCtx as t, createVNode as d, renderSlot as u } from "vue";
import f from "./ScalarComboboxOptions.vue.js";
import c from "./ScalarComboboxPopover.vue.js";
const y = /* @__PURE__ */ n({
  __name: "ScalarComboboxMultiselect",
  props: {
    options: {},
    modelValue: {},
    placeholder: {},
    placement: {},
    resize: { type: Boolean },
    isOpen: { type: Boolean },
    teleport: { type: [Boolean, String] }
  },
  emits: ["update:modelValue"],
  setup(V, { expose: p }) {
    const o = m(null);
    return p({ comboboxPopoverRef: o }), (e, l) => (s(), i(c, {
      ref_key: "comboboxPopoverRef",
      ref: o,
      isOpen: e.isOpen,
      placement: e.placement ?? "bottom-start",
      resize: e.resize,
      teleport: e.teleport
    }, {
      popover: t(({ open: r }) => [
        d(f, {
          modelValue: e.modelValue,
          multiselect: "",
          open: r,
          options: e.options,
          placeholder: e.placeholder,
          "onUpdate:modelValue": l[0] || (l[0] = (a) => e.$emit("update:modelValue", a))
        }, null, 8, ["modelValue", "open", "options", "placeholder"])
      ]),
      default: t(() => [
        u(e.$slots, "default")
      ]),
      _: 3
    }, 8, ["isOpen", "placement", "resize", "teleport"]));
  }
});
export {
  y as default
};
