// @scalar/components/dist/components/ScalarCombobox/ScalarCombobox.vue.js
import { defineComponent as a, openBlock as r, createBlock as n, withCtx as o, createVNode as m, renderSlot as s } from "vue";
import d from "./ScalarComboboxOptions.vue.js";
import i from "./ScalarComboboxPopover.vue.js";
const B = /* @__PURE__ */ a({
  __name: "ScalarCombobox",
  props: {
    options: {},
    modelValue: {},
    placeholder: {},
    placement: {},
    resize: { type: Boolean },
    isOpen: { type: Boolean },
    teleport: { type: [Boolean, String] }
  },
  emits: ["update:modelValue"],
  setup(u) {
    return (e, c) => (r(), n(i, {
      isOpen: e.isOpen,
      placement: e.placement ?? "bottom-start",
      resize: e.resize,
      teleport: e.teleport
    }, {
      popover: o(({ open: l, close: p }) => [
        m(d, {
          modelValue: e.modelValue ? [e.modelValue] : [],
          open: l,
          options: e.options,
          placeholder: e.placeholder,
          "onUpdate:modelValue": (t) => (p(), e.$emit("update:modelValue", t[0]))
        }, null, 8, ["modelValue", "open", "options", "placeholder", "onUpdate:modelValue"])
      ]),
      default: o(() => [
        s(e.$slots, "default")
      ]),
      _: 3
    }, 8, ["isOpen", "placement", "resize", "teleport"]));
  }
});
export {
  B as default
};
