// @scalar/components/dist/components/ScalarCombobox/ScalarComboboxOptions.vue.js
import { defineComponent as E, computed as f, ref as S, watch as M, openBlock as i, createElementBlock as u, Fragment as p, createElementVNode as b, createVNode as K, unref as w, withDirectives as $, withKeys as h, withModifiers as x, vModelText as T, renderList as B, toDisplayString as I, createCommentVNode as _, createBlock as q, normalizeStyle as F, withCtx as G, createTextVNode as O, vShow as U } from "vue";
import { nanoid as j } from "nanoid";
import H from "./ScalarComboboxOption.vue.js";
import { isGroups as N } from "./types.js";
import J from "../ScalarIcon/ScalarIcon.vue.js";
const P = { class: "relative flex" }, Q = ["aria-controls", "placeholder"], R = ["id"], W = {
  key: 0,
  class: "min-w-0 truncate px-2 py-1.5 text-left text-c-2"
}, le = /* @__PURE__ */ E({
  inheritAttrs: !1,
  __name: "ScalarComboboxOptions",
  props: {
    options: {},
    modelValue: {},
    placeholder: {},
    open: { type: Boolean },
    multiselect: { type: Boolean }
  },
  emits: ["update:modelValue"],
  setup(L, { emit: z }) {
    var g;
    const o = L, A = z, k = j(), y = f(
      () => N(o.options) ? o.options.flatMap((t) => t.options) : o.options
    ), D = f(
      () => N(o.options) ? o.options : [{ label: "", options: o.options }]
    ), s = S(""), r = S(((g = o.modelValue) == null ? void 0 : g[0]) ?? y.value[0]);
    M(
      () => o.open,
      () => s.value = ""
    ), M(
      () => s.value,
      () => r.value = d.value[0]
    );
    const d = f(
      () => s.value === "" ? y.value : y.value.filter((t) => t.label.toLowerCase().includes(s.value.toLowerCase()))
    ), a = f({
      get: () => o.modelValue ?? [],
      set: (t) => t && A("update:modelValue", t)
    });
    function V(t) {
      o.multiselect ? a.value.some((e) => e.id === t.id) ? a.value = a.value.filter((e) => e.id !== t.id) : a.value = [...a.value, t] : a.value = [t];
    }
    function C(t) {
      const e = d.value, c = e.findIndex((l) => l.id === r.value.id) + t;
      c < 0 || c > e.length - 1 || (r.value = e[c]);
    }
    return (t, e) => (i(), u(p, null, [
      b("div", P, [
        K(w(J), {
          class: "pointer-events-none absolute left-3 top-1/2 -translate-y-1/2 text-c-3",
          icon: "Search",
          size: "sm"
        }),
        $(b("input", {
          "onUpdate:modelValue": e[0] || (e[0] = (n) => s.value = n),
          "aria-autocomplete": "list",
          "aria-controls": w(k),
          class: "min-w-0 flex-1 rounded-none border-0 py-2.5 pl-8 pr-3 leading-none text-c-1 outline-none",
          "data-1p-ignore": "",
          placeholder: t.placeholder,
          role: "combobox",
          tabindex: "0",
          type: "text",
          onKeydown: [
            e[1] || (e[1] = h(x((n) => C(1), ["prevent"]), ["down"])),
            e[2] || (e[2] = h(x((n) => V(r.value), ["prevent"]), ["enter"])),
            e[3] || (e[3] = h(x((n) => C(-1), ["prevent"]), ["up"]))
          ]
        }, null, 40, Q), [
          [T, s.value]
        ])
      ]),
      $(b("ul", {
        id: w(k),
        class: "border-t p-0.75"
      }, [
        (i(!0), u(p, null, B(D.value, (n, c) => (i(), u(p, { key: c }, [
          n.label && // Only show the group label if there are some results
          n.options.some((l) => d.value.some((m) => m.id === l.id)) ? (i(), u("div", W, I(n.label), 1)) : _("", !0),
          (i(!0), u(p, null, B(d.value, (l) => {
            var m;
            return i(), u(p, {
              key: l.id
            }, [
              n.options.some((v) => v.id === l.id) ? (i(), q(H, {
                key: 0,
                active: ((m = r.value) == null ? void 0 : m.id) === l.id,
                selected: a.value.some((v) => v.id === l.id),
                style: F(t.multiselect ? "checkbox" : "radio"),
                onClick: (v) => V(l),
                onMousedown: e[4] || (e[4] = x(() => {
                }, ["prevent"])),
                onMouseenter: (v) => r.value = l
              }, {
                default: G(() => [
                  O(I(l.label), 1)
                ]),
                _: 2
              }, 1032, ["active", "selected", "style", "onClick", "onMouseenter"])) : _("", !0)
            ], 64);
          }), 128))
        ], 64))), 128))
      ], 8, R), [
        [U, d.value.length]
      ])
    ], 64));
  }
});
export {
  le as default
};
