// @scalar/api-client/dist/v2/blocks/scalar-auth-selector-block/helpers/convert-security-scheme-secrets.js
import { objectEntries as l } from "@scalar/helpers/object/object-entries";
import { coerceValue as a } from "@scalar/workspace-store/schemas/typebox-coerce";
import { SecuritySchemeObjectSchema as n } from "@scalar/workspace-store/schemas/v3.1/strict/openapi-document";
const S = {
  clientSecret: "x-scalar-secret-client-secret",
  password: "x-scalar-secret-password",
  token: "x-scalar-secret-token",
  username: "x-scalar-secret-username",
  value: "x-scalar-secret-token",
  "x-scalar-client-id": "x-scalar-secret-client-id",
  "x-scalar-redirect-uri": "x-scalar-secret-redirect-uri"
}, i = (e) => {
  const r = { ...e };
  for (const [t, s] of l(S))
    e[t] && (r[s] ||= e[t]);
  return r;
}, E = (e) => {
  if (e.type === "oauth2") {
    const r = /* @__PURE__ */ new Set(), t = l(e.flows ?? {}).reduce(
      (o, [d, c]) => (o[d] = c && i(c), c?.selectedScopes?.forEach((u) => r.add(u)), o),
      {}
    ), s = r.size > 0 && !e["x-default-scopes"]?.length ? { "x-default-scopes": Array.from(r) } : {};
    return a(n, {
      ...e,
      flows: t,
      ...s
    });
  }
  return a(n, i(e));
};
export {
  E as convertSecuritySchemeSecrets
};
