// @scalar/api-client/dist/v2/blocks/response-block/ResponseBlock.vue2.js
import { defineComponent as T, computed as i, ref as $, useId as S, createBlock as r, openBlock as o, withCtx as d, createElementVNode as c, normalizeClass as h, createElementBlock as p, createVNode as R, Fragment as y, createCommentVNode as u, createTextVNode as A, renderList as g, unref as I, withDirectives as E, resolveDynamicComponent as O, vShow as z } from "vue";
import { ScalarErrorBoundary as N } from "@scalar/components";
import { isDefined as P } from "@scalar/helpers/array/is-defined";
import D from "../../../components/SectionFilter.vue.js";
import L from "../../../components/ViewLayout/ViewLayoutSection.vue.js";
import V from "./components/Headers.vue.js";
import j from "./components/ResponseBody.vue.js";
import w from "./components/ResponseBodyStreaming.vue.js";
import F from "./components/ResponseBodyVirtual.vue.js";
/* empty css                                    */
import U from "./components/ResponseCookies.vue.js";
import K from "./components/ResponseEmpty.vue.js";
import M from "./components/ResponseLoadingOverlay.vue.js";
import Z from "./components/ResponseMetaInformation.vue.js";
import { textMediaTypes as G } from "./helpers/media-types.js";
import { parseSetCookie as J } from "./helpers/parse-set-cookie.js";
const Q = { class: "flex h-8 flex-1 items-center" }, W = ["id", "role"], b = 2e5, ve = /* @__PURE__ */ T({
  __name: "ResponseBlock",
  props: {
    response: {},
    request: {},
    layout: {},
    totalPerformedRequests: {},
    appVersion: {},
    plugins: {},
    eventBus: {}
  },
  setup(e, { expose: x }) {
    const m = i(() => {
      const t = e.response?.headers;
      return t ? Object.keys(t).map((s) => ({
        name: s,
        value: t[s] ?? ""
      })) : [];
    }), k = i(
      () => e.response?.cookieHeaderKeys.map((t) => J(t)).filter(P) ?? []
    ), C = ["Cookies", "Headers", "Body"], n = $("All"), v = i(() => ["All", ...C]), a = i(
      () => Object.fromEntries(
        v.value.map((t) => [t, S()])
      )
    ), B = i(() => {
      if (!e.response || !("size" in e.response))
        return !1;
      const t = e.response.headers?.["content-type"] || e.response.headers?.["Content-Type"];
      return !t || (e.response.size ?? 0) <= b || t.includes("text/html") ? !1 : G.some((l) => t.includes(l)) && (e.response.size ?? 0) > b;
    }), q = i(
      () => e.request?.headers ? [...e.request.headers].map((t) => ({
        name: t[0],
        value: t[1],
        required: !1
      })) : []
    ), f = (t) => n.value === "All" || n.value === t;
    return x({
      responseHeaders: m,
      responseCookies: k,
      requestHeaders: q,
      shouldVirtualize: B,
      activeFilter: n,
      filters: v
    }), (t, s) => (o(), r(L, { "aria-label": "Response" }, {
      title: d(() => [
        c("div", Q, [
          c("div", {
            "aria-live": "polite",
            class: h(["flex items-center", { "animate-response-heading": e.response }])
          }, [
            s[4] || (s[4] = c("span", { class: "response-heading pointer-events-none absolute" }, " Response ", -1)),
            e.response ? (o(), r(Z, {
              key: 0,
              class: "animate-response-children",
              eventBus: e.eventBus,
              response: e.response
            }, null, 8, ["eventBus", "response"])) : u("", !0)
          ], 2),
          R(D, {
            modelValue: n.value,
            "onUpdate:modelValue": s[0] || (s[0] = (l) => n.value = l),
            filterIds: a.value,
            filters: v.value
          }, null, 8, ["modelValue", "filterIds", "filters"])
        ])
      ]),
      default: d(() => [
        c("div", {
          id: a.value.All,
          class: h(["custom-scroll response-section-content relative grid h-full justify-stretch", {
            "content-start": e.response
          }]),
          role: n.value === "All" && e.response ? "tabpanel" : "none"
        }, [
          e.response ? (o(), p(y, { key: 1 }, [
            f("Cookies") ? (o(), r(U, {
              key: 0,
              id: a.value.Cookies,
              class: "response-section-content-cookies",
              cookies: k.value,
              role: n.value === "All" ? "none" : "tabpanel"
            }, null, 8, ["id", "cookies", "role"])) : u("", !0),
            f("Headers") ? (o(), r(V, {
              key: 1,
              id: a.value.Headers,
              class: "response-section-content-headers",
              headers: q.value,
              role: n.value === "All" ? "none" : "tabpanel"
            }, {
              title: d(() => [...s[5] || (s[5] = [
                A("Request Headers", -1)
              ])]),
              _: 1
            }, 8, ["id", "headers", "role"])) : u("", !0),
            f("Headers") ? (o(), r(V, {
              key: 2,
              id: a.value.Headers,
              class: "response-section-content-headers",
              headers: m.value,
              role: n.value === "All" ? "none" : "tabpanel"
            }, {
              title: d(() => [...s[6] || (s[6] = [
                A("Response Headers", -1)
              ])]),
              _: 1
            }, 8, ["id", "headers", "role"])) : u("", !0),
            (o(!0), p(y, null, g(e.plugins, (l, H) => (o(), r(I(N), { key: H }, {
              default: d(() => [
                l.components?.response && e.request && e.response ? E((o(), r(O(l.components.response), {
                  key: 0,
                  request: e.request,
                  response: e.response
                }, null, 8, ["request", "response"])), [
                  [z, n.value === "All"]
                ]) : u("", !0)
              ]),
              _: 2
            }, 1024))), 128)),
            n.value === "All" || n.value === "Body" ? (o(), p(y, { key: 3 }, [
              "reader" in e.response ? (o(), r(w, {
                key: 0,
                id: a.value.Body,
                class: "response-section-content-body",
                reader: e.response.reader
              }, null, 8, ["id", "reader"])) : B.value && typeof e.response?.data == "string" ? (o(), r(F, {
                key: 1,
                id: a.value.Body,
                content: e.response.data,
                data: e.response?.data,
                headers: m.value,
                role: n.value === "All" ? "none" : "tabpanel"
              }, null, 8, ["id", "content", "data", "headers", "role"])) : (o(), r(j, {
                key: 2,
                id: a.value.Body,
                active: !0,
                class: "response-section-content-body",
                data: e.response?.data,
                headers: m.value,
                layout: "client",
                role: n.value === "All" ? "none" : "tabpanel",
                title: "Body"
              }, null, 8, ["id", "data", "headers", "role"]))
            ], 64)) : u("", !0)
          ], 64)) : (o(), r(K, {
            key: 0,
            appVersion: e.appVersion,
            layout: e.layout,
            totalPerformedRequests: e.totalPerformedRequests,
            onAddRequest: s[1] || (s[1] = (l) => e.eventBus.emit("ui:open:command-palette", {
              action: "create-request",
              payload: void 0
            })),
            onOpenCommandPalette: s[2] || (s[2] = (l) => e.eventBus.emit("ui:open:command-palette")),
            onSendRequest: s[3] || (s[3] = (l) => e.eventBus.emit("operation:send:request:hotkey"))
          }, null, 8, ["appVersion", "layout", "totalPerformedRequests"])),
          R(M, { eventBus: e.eventBus }, null, 8, ["eventBus"])
        ], 10, W)
      ]),
      _: 1
    }));
  }
});
export {
  ve as default
};
