// @scalar/agent-chat/dist/components/ApprovalSection.vue2.js
import { defineComponent as s, createElementBlock as p, openBlock as a, createElementVNode as t, createVNode as i, createTextVNode as c, unref as l } from "vue";
import { ScalarIconInfo as u } from "@scalar/icons";
const m = { class: "approvalSection" }, v = { class: "approvalText flex items-center gap-1.5" }, d = { class: "approveContainer" }, B = /* @__PURE__ */ s({
  __name: "ApprovalSection",
  emits: ["approve", "reject"],
  setup(f, { emit: n }) {
    const o = n;
    return (_, e) => (a(), p("div", m, [
      t("strong", v, [
        i(l(u), {
          class: "text-blue size-4",
          weight: "bold"
        }),
        e[2] || (e[2] = c(" One or more requests require approval. ", -1))
      ]),
      t("div", d, [
        t("button", {
          type: "button",
          class: "actionButton rejectButton",
          onClick: e[0] || (e[0] = (r) => o("reject"))
        }, " Reject "),
        t("button", {
          type: "button",
          class: "actionButton approveButton",
          onClick: e[1] || (e[1] = (r) => o("approve"))
        }, " Approve ")
      ])
    ]));
  }
});
export {
  B as default
};
