// @scalar/agent-chat/dist/components/FreeMessagesInfoSection.vue2.js
import { defineComponent as u, ref as d, withDirectives as f, createElementBlock as p, openBlock as m, createElementVNode as e, createVNode as r, createTextVNode as g, unref as i, vShow as _ } from "vue";
import { ScalarIconInfo as w, ScalarIconX as b } from "@scalar/icons";
import { useState as v } from "../state/state.js";
const h = { class: "freeMessagesInfoSection" }, x = { class: "infoText flex items-center gap-1.5" }, I = { class: "actionsContainer" }, k = /* @__PURE__ */ u({
  __name: "FreeMessagesInfoSection",
  setup(S) {
    const t = d(!1), { dashboardUrl: o, mode: s, uploadedTmpDocumentUrl: n } = v();
    function c() {
      s === "full" && window.location.replace(o), s === "preview" && window.location.replace(
        n.value ? `${o}/register?flow=oss-agent&docUrl=${n.value}` : o
      );
    }
    function l() {
      t.value = !0;
    }
    return (y, a) => f((m(), p("div", h, [
      e("strong", x, [
        r(i(w), {
          class: "text-blue size-4",
          weight: "bold"
        }),
        a[0] || (a[0] = g(" You get 10 free messages, you can upgrade at anytime to get more for your API. ", -1))
      ]),
      e("div", I, [
        e("button", {
          class: "actionButton upgradeButton",
          type: "button",
          onClick: c
        }, " Upgrade "),
        e("button", {
          "aria-label": "Close",
          class: "closeButton",
          type: "button",
          onClick: l
        }, [
          r(i(b), {
            class: "size-4",
            weight: "bold"
          })
        ])
      ])
    ], 512)), [
      [_, !t.value]
    ]);
  }
});
export {
  k as default
};
