// @scalar/agent-chat/dist/components/PaymentSection.vue2.js
import { defineComponent as d, createElementBlock as p, openBlock as i, createElementVNode as t, createVNode as r, createTextVNode as l, unref as v, createStaticVNode as c } from "vue";
import { ScalarIconInfo as m } from "@scalar/icons";
import { useState as b } from "../state/state.js";
const f = { class: "paymentSection" }, u = { class: "approvalText flex items-center gap-1.5" }, x = /* @__PURE__ */ d({
  __name: "PaymentSection",
  setup(y) {
    const { dashboardUrl: e, mode: n, uploadedTmpDocumentUrl: s } = b();
    function o() {
      n === "full" && window.location.replace(e), n === "preview" && window.location.replace(
        s.value ? `${e}/register?flow=oss-agent&docUrl=${s.value}` : e
      );
    }
    return (I, a) => (i(), p("div", f, [
      t("strong", u, [
        r(v(m), {
          class: "text-blue size-4",
          weight: "bold"
        }),
        a[0] || (a[0] = l(" You've used up all your free messages, upgrade for $24/mo. ", -1))
      ]),
      t("div", { class: "paymentContainer" }, [
        t("button", {
          class: "actionButton approveButton",
          type: "button",
          onClick: o
        }, " Upgrade "),
        a[1] || (a[1] = c('<div class="paymentInfo" data-v-59a0be07><h3 data-v-59a0be07>$24 <span data-v-59a0be07>/ month</span></h3><div class="paymentInfoSection" data-v-59a0be07><div class="paymentInfoItem" data-v-59a0be07><span data-v-59a0be07>Members in your workspace</span><span data-v-59a0be07>1</span></div><div class="paymentInfoItem" data-v-59a0be07><span data-v-59a0be07>x $24 / month / member</span><span data-v-59a0be07>$24.00</span></div></div><div class="paymentInfoSection" data-v-59a0be07><div class="paymentInfoItem" data-v-59a0be07><span data-v-59a0be07>Messages</span><span data-v-59a0be07>1,000</span></div><div class="paymentInfoItem" data-v-59a0be07><span data-v-59a0be07>Additional Messages</span><span data-v-59a0be07>+ $0.02 Message</span></div></div></div>', 1))
      ])
    ]));
  }
});
export {
  x as default
};
