// @scalar/agent-chat/dist/components/RequestPreview.vue2.js
import { defineComponent as w, computed as i, ref as R, createElementBlock as n, openBlock as t, normalizeClass as S, createBlock as o, createCommentVNode as f, createElementVNode as l, createVNode as d, unref as a } from "vue";
import { ScalarCodeBlock as h } from "@scalar/components";
import { ScalarIconCaretDown as g, ScalarIconCaretRight as q } from "@scalar/icons";
import T from "./AutosendPaused.vue.js";
import b from "./RequestApproved.vue.js";
import V from "./RequestFailed.vue.js";
import N from "./RequestRejected.vue.js";
import $ from "./RequestSuccess.vue.js";
import { getMediaTypeConfig as A } from "./ResponseBody/helpers/media-types.js";
import { processResponseBody as E } from "./ResponseBody/helpers/process-response-body.js";
import j from "./ResponseBody/ResponseBody.vue.js";
import D from "./ResponseBody/ResponseBodyToggle.vue.js";
import F from "./SendingRequest.vue.js";
const I = {
  key: 1,
  class: "autosendContainer"
}, O = { class: "requestContent" }, P = { class: "requestContentInner" }, x = {
  key: 0,
  class: "code"
}, H = {
  key: 1,
  class: "code"
}, M = { class: "requestHeaderContainer" }, te = /* @__PURE__ */ w({
  __name: "RequestPreview",
  props: {
    request: {},
    response: {},
    state: {}
  },
  setup(e) {
    const c = i(() => {
      if (e.response?.structuredContent.success)
        return {
          data: e.response.structuredContent.data.responseBody,
          headers: e.response.structuredContent.data.headers
        };
      if (e.response?.structuredContent.error?.code === "REQUEST_NOT_OK")
        return {
          data: e.response.structuredContent.error.detail.responseBody,
          headers: e.response.structuredContent.error.detail.headers
        };
    }), r = R(!1), v = i(() => e.state === "requestFailed" || e.state === "requiresApproval" ? !0 : r.value), C = i(
      () => E({
        data: c.value?.data,
        headers: c.value?.headers
      })
    ), u = i(
      () => A(C.value.mimeType?.essence ?? "")
    ), y = R();
    function B(k) {
      y.value = k;
    }
    const p = i(() => y.value ? y.value : u.value?.raw && !u.value.preview ? "raw" : u.value?.preview ? "preview" : "raw");
    return (k, s) => (t(), n("div", {
      class: S(["requestPreview", {
        open: v.value,
        succeeded: e.state === "requestSucceeded"
      }])
    }, [
      e.state === "approved" ? (t(), o(b, { key: 0 })) : e.state === "requiresApproval" ? (t(), n("div", I, [
        d(T)
      ])) : e.state === "sendingRequest" ? (t(), n("button", {
        key: 2,
        class: "toggleButton",
        type: "button",
        onClick: s[0] || (s[0] = (m) => r.value = !r.value)
      }, [
        d(F),
        v.value ? (t(), o(a(g), { key: 0 })) : (t(), o(a(q), { key: 1 }))
      ])) : e.state === "requestSucceeded" ? (t(), n("button", {
        key: 3,
        class: "toggleButton",
        type: "button",
        onClick: s[1] || (s[1] = (m) => r.value = !r.value)
      }, [
        d($),
        v.value ? (t(), o(a(g), { key: 0 })) : (t(), o(a(q), { key: 1 }))
      ])) : e.state === "rejected" ? (t(), n("button", {
        key: 4,
        class: "toggleButton",
        type: "button",
        onClick: s[2] || (s[2] = (m) => r.value = !r.value)
      }, [
        d(N),
        v.value ? (t(), o(a(g), { key: 0 })) : (t(), o(a(q), { key: 1 }))
      ])) : e.state === "requestFailed" ? (t(), o(V, { key: 5 })) : f("", !0),
      l("div", O, [
        l("div", P, [
          e.request ? (t(), n("div", x, [
            s[4] || (s[4] = l("div", { class: "requestHeaderContainer" }, [
              l("h1", null, "Request")
            ], -1)),
            d(a(h), {
              class: "codeBlock",
              content: JSON.stringify(e.request, null, 2),
              lang: "json"
            }, null, 8, ["content"])
          ])) : f("", !0),
          c.value ? (t(), n("div", H, [
            l("div", M, [
              s[5] || (s[5] = l("h1", null, "Response", -1)),
              u.value?.raw && u.value.preview ? (t(), o(D, {
                key: 0,
                modelValue: p.value,
                "onUpdate:modelValue": s[3] || (s[3] = (m) => p.value = m),
                onToggle: B
              }, null, 8, ["modelValue"])) : f("", !0)
            ]),
            d(j, {
              data: c.value.data,
              display: p.value,
              mediaConfig: u.value,
              responseBody: C.value
            }, null, 8, ["data", "display", "mediaConfig", "responseBody"])
          ])) : f("", !0)
        ])
      ])
    ], 2));
  }
});
export {
  te as default
};
