// @scalar/agent-chat/dist/components/SearchPopover.vue2.js
import { defineComponent as d, computed as i, createBlock as f, openBlock as s, unref as o, withCtx as c, renderSlot as h, createVNode as m, createElementBlock as r, Fragment as _, renderList as g, createCommentVNode as y, createElementVNode as v, toDisplayString as k } from "vue";
import { ScalarPopover as I, ScalarTextInput as S } from "@scalar/components";
import { ScalarIconMagnifyingGlass as V } from "@scalar/icons";
import { useSearch as x } from "../hooks/use-search.js";
import { useState as C } from "../state/state.js";
const N = ["onClick"], P = ["src"], b = {
  key: 1,
  class: "searchResultsEmpty"
}, w = /* @__PURE__ */ d({
  __name: "SearchPopover",
  setup(A) {
    const n = C(), l = x(), u = i(
      () => l.results.value.filter(
        (e) => !n.registryDocuments.value.some(
          (a) => a.namespace === e.namespace && a.slug === e.slug
        )
      ).map((e) => ({
        ...e,
        label: e.title,
        id: e.id
      }))
    );
    return (e, a) => (s(), f(o(I), {
      offset: 0,
      placement: "top-start",
      resize: "",
      style: { width: "220px" }
    }, {
      popover: c(({ close: p }) => [
        m(o(S), {
          autofocus: "",
          class: "searchInput",
          modelValue: o(l).query.value,
          placeholder: "Add an API",
          "onUpdate:modelValue": a[0] || (a[0] = (t) => o(l).query.value = t ?? "")
        }, {
          prefix: c(() => [
            m(o(V), { class: "searchIcon" })
          ]),
          _: 1
        }, 8, ["modelValue"]),
        u.value.length ? (s(!0), r(_, { key: 0 }, g(u.value, (t) => (s(), r("button", {
          key: t.id,
          class: "searchItem",
          type: "button",
          onClick: () => {
            o(n).addDocument(t), p();
          }
        }, [
          t.logoUrl ? (s(), r("img", {
            key: 0,
            class: "searchItemLogo",
            src: t.logoUrl
          }, null, 8, P)) : y("", !0),
          v("span", null, k(t.title), 1)
        ], 8, N))), 128)) : (s(), r("span", b, " No APIs found "))
      ]),
      default: c(() => [
        h(e.$slots, "default", {}, void 0, !0)
      ]),
      _: 3
    }));
  }
});
export {
  w as default
};
