// @scalar/agent-chat/dist/components/Selector.vue.js
import { defineComponent as g, computed as s, createBlock as S, createElementBlock as w, openBlock as u, unref as n, withCtx as i, createVNode as d, createElementVNode as a, toDisplayString as c } from "vue";
import { ScalarListbox as b, ScalarButton as h } from "@scalar/components";
import { ScalarIconCaretDown as y } from "@scalar/icons";
const V = { class: "overflow-x-auto" }, k = {
  key: 1,
  class: "text-c-1 flex h-auto w-full items-center gap-0.75 rounded-b-lg px-3 py-1.5 text-base leading-[20px] whitespace-nowrap"
}, B = { class: "overflow-x-auto" }, E = /* @__PURE__ */ g({
  __name: "Selector",
  props: {
    selectedServer: {},
    servers: {},
    target: {}
  },
  emits: ["update:modelValue"],
  setup(e, { expose: p, emit: m }) {
    const v = m, o = s(
      () => e.servers.map((r) => ({
        id: r.url,
        label: r.url
      }))
    ), l = s(
      () => e.selectedServer?.url?.replace(/\/$/, "") || ""
    ), f = s(
      () => o.value.find((r) => r.id === e.selectedServer?.url)
    );
    return p({
      servers: e.servers,
      serverUrlWithoutTrailingSlash: l,
      serverOptions: o,
      selectedServer: e.selectedServer
    }), (r, t) => o.value.length > 1 ? (u(), S(n(b), {
      key: 0,
      ref: "elem",
      class: "group",
      modelValue: f.value,
      options: o.value,
      placement: "bottom-start",
      resize: "",
      target: e.target,
      "onUpdate:modelValue": t[0] || (t[0] = (x) => v("update:modelValue", x.id))
    }, {
      default: i(() => [
        d(n(h), {
          class: "bg-b-1 text-c-1 h-auto w-full justify-start gap-1.5 overflow-x-auto rounded-t-none rounded-b-lg px-3 py-1.5 text-base font-normal whitespace-nowrap -outline-offset-1",
          variant: "ghost"
        }, {
          default: i(() => [
            t[1] || (t[1] = a("span", { class: "sr-only" }, "Server:", -1)),
            a("span", V, c(l.value || "Select a server"), 1),
            d(n(y), {
              class: "text-c-2 ui-open:rotate-180 mt-0.25 size-3 transition-transform duration-100",
              weight: "bold"
            })
          ]),
          _: 1
        })
      ]),
      _: 1
    }, 8, ["modelValue", "options", "target"])) : (u(), w("div", k, [
      t[2] || (t[2] = a("span", { class: "sr-only" }, "Server:", -1)),
      a("span", B, c(l.value), 1)
    ]));
  }
});
export {
  E as default
};
