// @scalar/agent-chat/dist/components/ServerSelector.vue.js
import { defineComponent as m, useId as b, createElementBlock as S, openBlock as d, Fragment as f, createElementVNode as n, createVNode as g, createBlock as c, createCommentVNode as o, normalizeClass as x, unref as l } from "vue";
import { ServerVariablesForm as k } from "@scalar/api-client/components/Server";
import { ScalarMarkdown as B } from "@scalar/components";
import V from "./Selector.vue.js";
const y = ["id"], C = {}, I = /* @__PURE__ */ m({
  ...C,
  __name: "ServerSelector",
  props: {
    eventBus: {},
    selectedServer: {},
    servers: {}
  },
  setup(e) {
    const s = b(), v = (r) => {
      e.eventBus.emit("server:update:selected", {
        url: e.selectedServer?.url === r ? "" : r
      });
    }, i = (r, t) => {
      const a = e.servers.findIndex((u) => u.url === e.selectedServer?.url);
      a !== -1 && e.eventBus.emit("server:update:variables", {
        index: a,
        key: r,
        value: t
      });
    };
    return (r, t) => (d(), S(f, null, [
      t[0] || (t[0] = n("label", { class: "bg-b-2 flex h-8 items-center rounded-t-lg border border-b-0 px-3 py-2.5 font-medium" }, " Server ", -1)),
      n("div", {
        id: l(s),
        class: x(["border", {
          "rounded-b-lg": !e.selectedServer?.description && !e.selectedServer?.variables
        }])
      }, [
        e.servers.length ? (d(), c(V, {
          key: 0,
          selectedServer: e.selectedServer,
          servers: e.servers,
          target: l(s),
          "onUpdate:modelValue": v
        }, null, 8, ["selectedServer", "servers", "target"])) : o("", !0)
      ], 10, y),
      g(l(k), {
        layout: "reference",
        variables: e.selectedServer?.variables,
        "onUpdate:variable": i
      }, null, 8, ["variables"]),
      e.selectedServer?.description ? (d(), c(l(B), {
        key: 0,
        class: "text-c-3 rounded-b-lg border border-t-0 px-3 py-1.5",
        value: e.selectedServer.description
      }, null, 8, ["value"])) : o("", !0)
    ], 64));
  }
});
export {
  I as default
};
