// @scalar/agent-chat/dist/components/UploadSection.vue2.js
import { defineComponent as d, createElementBlock as t, openBlock as o, normalizeClass as i, createCommentVNode as n, createVNode as s, unref as a, createTextVNode as r, toDisplayString as u } from "vue";
import { useLoadingState as p, ScalarLoading as g } from "@scalar/components";
import { ScalarIconCheck as m, ScalarIconXCircle as S } from "@scalar/icons";
const x = {
  key: 0,
  class: "flex items-center gap-1.5"
}, y = {
  key: 0,
  class: "uploadText"
}, f = {
  key: 1,
  class: "uploadText"
}, k = {
  key: 1,
  class: "uploadText flex items-center gap-1.5"
}, h = {
  key: 2,
  class: "uploadText flex items-center gap-1.5"
}, V = /* @__PURE__ */ d({
  __name: "UploadSection",
  props: {
    uploadState: {}
  },
  setup(e) {
    const c = p();
    return (C, l) => (o(), t("div", {
      class: i(["uploadSection", {
        error: e.uploadState.type === "error",
        done: e.uploadState.type === "done"
      }])
    }, [
      e.uploadState.type === "uploading" || e.uploadState.type === "processing" ? (o(), t("div", x, [
        s(a(g), {
          class: "text-blue",
          loader: {
            ...a(c),
            isLoading: !0
          },
          size: "lg"
        }, null, 8, ["loader"]),
        e.uploadState.type === "processing" ? (o(), t("strong", y, " Processing document... ")) : (o(), t("strong", f, " Uploading document... "))
      ])) : n("", !0),
      e.uploadState.type === "done" ? (o(), t("strong", k, [
        s(a(m), { class: "icon text-green" }),
        l[0] || (l[0] = r(" Document processed ", -1))
      ])) : n("", !0),
      e.uploadState.type === "error" ? (o(), t("strong", h, [
        s(a(S), { class: "icon text-red" }),
        r(" " + u(e.uploadState.error), 1)
      ])) : n("", !0)
    ], 2));
  }
});
export {
  V as default
};
