// @scalar/api-client/dist/v2/blocks/scalar-auth-selector-block/components/AuthSelector.vue.js
import { defineComponent as U, useId as $, ref as p, computed as x, createBlock as S, openBlock as c, unref as s, withCtx as r, createVNode as m, createCommentVNode as b, createElementBlock as v, Fragment as g, createElementVNode as u, createTextVNode as k, toDisplayString as y, withModifiers as q, normalizeClass as M } from "vue";
import { useModal as N, ScalarComboboxMultiselect as T, ScalarButton as j, ScalarListboxCheckbox as z, ScalarIconButton as E } from "@scalar/components";
import { ScalarIconCaretDown as L, ScalarIconTrash as P } from "@scalar/icons";
import { getResolvedRef as F } from "@scalar/workspace-store/helpers/get-resolved-ref";
import { unpackProxyObject as G } from "@scalar/workspace-store/helpers/unpack-proxy";
import H from "./DeleteRequestAuthModal.vue.js";
import { isAuthOptional as J } from "../helpers/is-auth-optional.js";
import { getSecuritySchemeOptions as K, formatComplexScheme as Q, formatScheme as W } from "../helpers/security-scheme.js";
import X from "./RequestAuthDataTable.vue.js";
import Y from "../../../components/layout/CollapsibleSection.vue.js";
const Z = ["id"], _ = { class: "min-w-0 flex-1 truncate" }, ue = /* @__PURE__ */ U({
  __name: "AuthSelector",
  props: {
    environment: {},
    eventBus: {},
    isStatic: { type: Boolean, default: !1 },
    meta: {},
    proxyUrl: {},
    securityRequirements: {},
    securitySchemes: {},
    selectedSecurity: {},
    server: {},
    title: {}
  },
  setup(e, { expose: I }) {
    const C = $(), B = p(null), O = p(!1), d = N(), o = p(null), h = x(() => {
      if (!e.securityRequirements?.length)
        return null;
      const l = J(e.securityRequirements);
      return {
        icon: l ? "Unlock" : "Lock",
        text: l ? "Optional" : "Required"
      };
    }), R = x(
      () => K(
        e.securityRequirements ?? [],
        e.securitySchemes ?? {},
        e.selectedSecurity?.selectedSchemes ?? []
      )
    ), i = x(() => {
      const l = e.selectedSecurity?.selectedSchemes;
      return l?.length ? l.flatMap((t) => {
        const n = Object.keys(t);
        if (n.length === 0)
          return [];
        if (n.length > 1)
          return Q(t);
        const a = n[0];
        if (!a)
          return [];
        const f = F(e.securitySchemes?.[a]);
        return f ? W({
          name: a,
          type: f.type,
          value: t
        }) : [];
      }) : [];
    }), w = (l) => {
      O.value && l.stopPropagation(), B.value?.$el.click();
    }, A = (l) => {
      const t = l.filter((a) => a.payload === void 0).map((a) => G(a.value, { depth: 2 })), n = l.filter((a) => a.payload !== void 0).map((a) => ({
        name: a.label,
        scheme: a.payload
      }));
      e.eventBus.emit("auth:update:selected-security-schemes", {
        selectedRequirements: t,
        newSchemes: n,
        meta: e.meta
      });
    }, D = (l) => {
      o.value = { label: l.label, payload: l.value }, d.show();
    }, V = () => {
      o.value && (e.eventBus.emit("auth:delete:security-scheme", {
        names: Object.keys(o.value.payload)
      }), o.value = null, d.hide());
    };
    return I({
      authIndicator: h,
      selectedSchemeOptions: i,
      schemeOptions: R
    }), (l, t) => (c(), S(s(Y), {
      class: "group/params relative",
      isStatic: e.isStatic,
      itemCount: i.value.length,
      "onUpdate:modelValue": t[1] || (t[1] = (n) => O.value = n)
    }, {
      title: r(() => [
        u("div", {
          id: s(C),
          class: "inline-flex items-center gap-0.5 leading-[20px]"
        }, [
          u("span", null, y(e.title), 1),
          h.value ? (c(), v("span", {
            key: 0,
            class: M(["text-c-3 hover:bg-b-3 hover:text-c-1 -mr-1 cursor-pointer rounded px-1 py-0.5 text-xs leading-[normal]", { "text-c-1": h.value.text === "Required" }]),
            "data-testid": "auth-indicator",
            onClick: w
          }, y(h.value.text), 3)) : b("", !0)
        ], 8, Z)
      ]),
      actions: r(() => [
        m(s(T), {
          class: "w-72 text-xs",
          modelValue: i.value,
          multiple: "",
          options: R.value,
          placement: "bottom-end",
          teleport: "",
          onDelete: D,
          "onUpdate:modelValue": A
        }, {
          option: r(({ option: n, selected: a }) => [
            m(s(z), {
              multiselect: "",
              selected: a
            }, null, 8, ["selected"]),
            u("div", _, y(n.label), 1),
            n.isDeletable ? (c(), S(s(E), {
              key: 0,
              class: "-m-0.5 shrink-0 p-0.5 opacity-0 group-hover/item:opacity-100",
              icon: s(P),
              label: `Delete ${n.label}`,
              size: "xs",
              onClick: q((f) => D(n), ["stop"])
            }, null, 8, ["icon", "label", "onClick"])) : b("", !0)
          ]),
          default: r(() => [
            m(s(j), {
              ref_key: "comboboxButtonRef",
              ref: B,
              "aria-describedby": s(C),
              class: "group/combobox-button hover:text-c-1 text-c-2 flex h-fit w-full items-center gap-1 px-0.75 py-0.25 text-base font-normal",
              variant: "ghost"
            }, {
              default: r(() => [
                i.value.length === 1 ? (c(), v(g, { key: 0 }, [
                  t[2] || (t[2] = u("span", { class: "sr-only" }, "Selected Auth Type:", -1)),
                  k(" " + y(i.value[0]?.label), 1)
                ], 64)) : i.value.length > 1 ? (c(), v(g, { key: 1 }, [
                  t[3] || (t[3] = k(" Multiple ", -1)),
                  t[4] || (t[4] = u("span", { class: "sr-only" }, "Auth Types Selected", -1))
                ], 64)) : (c(), v(g, { key: 2 }, [
                  t[5] || (t[5] = u("span", { class: "sr-only" }, "Select", -1)),
                  t[6] || (t[6] = k(" Auth Type ", -1))
                ], 64)),
                m(s(L), {
                  class: "size-3 shrink-0 transition-transform duration-100 group-aria-expanded/combobox-button:rotate-180",
                  weight: "bold"
                })
              ]),
              _: 1
            }, 8, ["aria-describedby"])
          ]),
          _: 1
        }, 8, ["modelValue", "options"])
      ]),
      default: r(() => [
        m(X, {
          activeAuthIndex: e.selectedSecurity?.selectedIndex ?? 0,
          environment: e.environment,
          eventBus: e.eventBus,
          isStatic: e.isStatic,
          meta: e.meta,
          proxyUrl: e.proxyUrl,
          securitySchemes: e.securitySchemes,
          selectedSchemeOptions: i.value,
          server: e.server
        }, null, 8, ["activeAuthIndex", "environment", "eventBus", "isStatic", "meta", "proxyUrl", "securitySchemes", "selectedSchemeOptions", "server"]),
        o.value ? (c(), S(H, {
          key: 0,
          label: o.value.label,
          scheme: o.value,
          state: s(d),
          onClose: t[0] || (t[0] = (n) => s(d).hide()),
          onDelete: V
        }, null, 8, ["label", "scheme", "state"])) : b("", !0)
      ]),
      _: 1
    }, 8, ["isStatic", "itemCount"]));
  }
});
export {
  ue as default
};
