// @scalar/api-client/dist/v2/blocks/scalar-auth-selector-block/components/DeleteRequestAuthModal.vue.js
import { defineComponent as m, createBlock as f, openBlock as p, unref as o, withCtx as n, createElementVNode as s, toDisplayString as a, createVNode as c, createTextVNode as i } from "vue";
import { ScalarModal as x, ScalarButton as r } from "@scalar/components";
const h = { class: "text-c-2 mb-4 text-sm leading-normal" }, b = { class: "flex justify-between gap-2" }, _ = /* @__PURE__ */ m({
  __name: "DeleteRequestAuthModal",
  props: {
    state: {},
    label: {}
  },
  emits: ["close", "delete"],
  setup(t, { emit: u }) {
    const l = u, d = () => {
      l("delete");
    };
    return (y, e) => (p(), f(o(x), {
      size: "xxs",
      state: t.state,
      title: "Delete Security Scheme"
    }, {
      default: n(() => [
        s("p", h, " This cannot be undone. You're about to delete the " + a(t.label) + " security scheme from the collection. ", 1),
        s("div", b, [
          c(o(r), {
            class: "flex h-8 cursor-pointer items-center gap-1.5 px-3 shadow-none focus:outline-none",
            type: "button",
            variant: "outlined",
            onClick: e[0] || (e[0] = (S) => l("close"))
          }, {
            default: n(() => [...e[1] || (e[1] = [
              i(" Cancel ", -1)
            ])]),
            _: 1
          }),
          c(o(r), {
            class: "flex h-8 cursor-pointer items-center gap-1.5 px-3 shadow-none focus:outline-none",
            type: "submit",
            onClick: d
          }, {
            default: n(() => [
              i(" Delete " + a(t.label), 1)
            ]),
            _: 1
          })
        ])
      ]),
      _: 1
    }, 8, ["state"]));
  }
});
export {
  _ as default
};
