// @scalar/api-client/dist/v2/blocks/scalar-auth-selector-block/components/OAuth2.vue.js
import { defineComponent as b, computed as w, watch as B, createElementBlock as x, openBlock as c, Fragment as y, createVNode as r, unref as o, withCtx as t, createTextVNode as s, createElementVNode as U, createBlock as f, createCommentVNode as v } from "vue";
import { useLoadingState as L, ScalarButton as S } from "@scalar/components";
import { pkceOptions as P } from "@scalar/oas-utils/entities/spec";
import { useToasts as R } from "@scalar/use-toasts";
import F from "./OAuthScopesInput.vue.js";
/* empty css                      */
import { authorizeOauth2 as O } from "../helpers/oauth.js";
import m from "./RequestAuthDataTableInput.vue.js";
import u from "../../../../components/DataTable/DataTableRow.vue.js";
const A = { class: "flex h-8 items-center justify-end gap-2 border-t" }, I = { class: "flex h-8 w-full items-center justify-end border-t" }, X = /* @__PURE__ */ b({
  __name: "OAuth2",
  props: {
    environment: {},
    flows: {},
    type: {},
    selectedScopes: {},
    server: {},
    proxyUrl: {}
  },
  emits: ["update:securityScheme", "update:selectedScopes"],
  setup(a, { emit: g }) {
    const V = g, p = L(), { toast: C } = R(), l = w(() => a.flows[a.type]), k = w(
      () => a.selectedScopes.filter((i) => i in (l.value.scopes ?? {}))
    ), d = (i) => V("update:securityScheme", {
      type: "oauth2",
      flows: {
        [a.type]: i
      }
    });
    B(
      () => l.value["x-scalar-secret-redirect-uri"],
      (i) => {
        i || typeof window > "u" || !("x-scalar-secret-redirect-uri" in l.value) || d({
          "x-scalar-secret-redirect-uri": window.location.origin + window.location.pathname
        });
      },
      { immediate: !0 }
    );
    const T = async () => {
      if (p.isLoading)
        return;
      p.start();
      const [i, e] = await O(
        a.flows,
        a.type,
        k.value,
        a.server,
        a.proxyUrl
      );
      await p.clear(), e ? d({ "x-scalar-secret-token": e }) : (console.error(i), C(i?.message ?? "Failed to authorize", "error"));
    }, z = (i) => d({
      "x-scalar-credentials-location": i === "body" ? "body" : "header"
    });
    return (i, e) => l.value["x-scalar-secret-token"] ? (c(), x(y, { key: 0 }, [
      r(o(u), null, {
        default: t(() => [
          r(m, {
            class: "border-r-transparent",
            environment: a.environment,
            modelValue: l.value["x-scalar-secret-token"],
            placeholder: "QUxMIFlPVVIgQkFTRSBBUkUgQkVMT05HIFRPIFVT",
            type: "password",
            "onUpdate:modelValue": e[0] || (e[0] = (n) => d({ "x-scalar-secret-token": n }))
          }, {
            default: t(() => [...e[12] || (e[12] = [
              s(" Access Token ", -1)
            ])]),
            _: 1
          }, 8, ["environment", "modelValue"])
        ]),
        _: 1
      }),
      r(o(u), { class: "min-w-full" }, {
        default: t(() => [
          U("div", A, [
            r(o(S), {
              class: "mr-1 p-0 px-2 py-0.5",
              loader: o(p),
              size: "sm",
              variant: "outlined",
              onClick: e[1] || (e[1] = () => d({ "x-scalar-secret-token": "" }))
            }, {
              default: t(() => [...e[13] || (e[13] = [
                s(" Clear ", -1)
              ])]),
              _: 1
            }, 8, ["loader"])
          ])
        ]),
        _: 1
      })
    ], 64)) : (c(), x(y, { key: 1 }, [
      r(o(u), null, {
        default: t(() => [
          "authorizationUrl" in l.value ? (c(), f(m, {
            key: 0,
            containerClass: "border-r-0",
            environment: a.environment,
            modelValue: l.value.authorizationUrl,
            placeholder: "https://galaxy.scalar.com/authorize",
            "onUpdate:modelValue": e[2] || (e[2] = (n) => d({ authorizationUrl: n }))
          }, {
            default: t(() => [...e[14] || (e[14] = [
              s(" Auth URL ", -1)
            ])]),
            _: 1
          }, 8, ["environment", "modelValue"])) : v("", !0),
          "tokenUrl" in l.value ? (c(), f(m, {
            key: 1,
            environment: a.environment,
            modelValue: l.value.tokenUrl,
            placeholder: "https://galaxy.scalar.com/token",
            "onUpdate:modelValue": e[3] || (e[3] = (n) => d({ tokenUrl: n }))
          }, {
            default: t(() => [...e[15] || (e[15] = [
              s(" Token URL ", -1)
            ])]),
            _: 1
          }, 8, ["environment", "modelValue"])) : v("", !0)
        ]),
        _: 1
      }),
      "x-scalar-secret-redirect-uri" in l.value ? (c(), f(o(u), { key: 0 }, {
        default: t(() => [
          r(m, {
            environment: a.environment,
            modelValue: l.value["x-scalar-secret-redirect-uri"],
            placeholder: "https://galaxy.scalar.com/callback",
            "onUpdate:modelValue": e[4] || (e[4] = (n) => d({ "x-scalar-secret-redirect-uri": n }))
          }, {
            default: t(() => [...e[16] || (e[16] = [
              s(" Redirect URL ", -1)
            ])]),
            _: 1
          }, 8, ["environment", "modelValue"])
        ]),
        _: 1
      })) : v("", !0),
      "x-scalar-secret-username" in l.value && "x-scalar-secret-password" in l.value ? (c(), x(y, { key: 1 }, [
        r(o(u), null, {
          default: t(() => [
            r(m, {
              class: "text-c-2",
              environment: a.environment,
              modelValue: l.value["x-scalar-secret-username"],
              placeholder: "janedoe",
              "onUpdate:modelValue": e[5] || (e[5] = (n) => d({ "x-scalar-secret-username": n }))
            }, {
              default: t(() => [...e[17] || (e[17] = [
                s(" Username ", -1)
              ])]),
              _: 1
            }, 8, ["environment", "modelValue"])
          ]),
          _: 1
        }),
        r(o(u), null, {
          default: t(() => [
            r(m, {
              environment: a.environment,
              modelValue: l.value["x-scalar-secret-password"],
              placeholder: "********",
              type: "password",
              "onUpdate:modelValue": e[6] || (e[6] = (n) => d({ "x-scalar-secret-password": n }))
            }, {
              default: t(() => [...e[18] || (e[18] = [
                s(" Password ", -1)
              ])]),
              _: 1
            }, 8, ["environment", "modelValue"])
          ]),
          _: 1
        })
      ], 64)) : v("", !0),
      r(o(u), null, {
        default: t(() => [
          r(m, {
            environment: a.environment,
            modelValue: l.value["x-scalar-secret-client-id"],
            placeholder: "12345",
            "onUpdate:modelValue": e[7] || (e[7] = (n) => d({ "x-scalar-secret-client-id": n }))
          }, {
            default: t(() => [...e[19] || (e[19] = [
              s(" Client ID ", -1)
            ])]),
            _: 1
          }, 8, ["environment", "modelValue"])
        ]),
        _: 1
      }),
      "x-scalar-secret-client-secret" in l.value ? (c(), f(o(u), { key: 2 }, {
        default: t(() => [
          r(m, {
            environment: a.environment,
            modelValue: l.value["x-scalar-secret-client-secret"],
            placeholder: "XYZ123",
            type: "password",
            "onUpdate:modelValue": e[8] || (e[8] = (n) => d({ "x-scalar-secret-client-secret": n }))
          }, {
            default: t(() => [...e[20] || (e[20] = [
              s(" Client Secret ", -1)
            ])]),
            _: 1
          }, 8, ["environment", "modelValue"])
        ]),
        _: 1
      })) : v("", !0),
      "x-usePkce" in l.value ? (c(), f(o(u), { key: 3 }, {
        default: t(() => [
          r(m, {
            enum: o(P),
            environment: a.environment,
            modelValue: l.value["x-usePkce"],
            readOnly: "",
            "onUpdate:modelValue": e[9] || (e[9] = (n) => d({
              "x-usePkce": n
            }))
          }, {
            default: t(() => [...e[21] || (e[21] = [
              s(" Use PKCE ", -1)
            ])]),
            _: 1
          }, 8, ["enum", "environment", "modelValue"])
        ]),
        _: 1
      })) : v("", !0),
      a.type !== "implicit" ? (c(), f(o(u), { key: 4 }, {
        default: t(() => [
          r(m, {
            enum: ["header", "body"],
            environment: a.environment,
            modelValue: l.value["x-scalar-credentials-location"] || "header",
            placeholder: "header",
            readOnly: "",
            "onUpdate:modelValue": e[10] || (e[10] = (n) => z(n))
          }, {
            default: t(() => [...e[22] || (e[22] = [
              s(" Credentials Location ", -1)
            ])]),
            _: 1
          }, 8, ["environment", "modelValue"])
        ]),
        _: 1
      })) : v("", !0),
      r(o(u), null, {
        default: t(() => [
          r(F, {
            flow: l.value,
            flowType: a.type,
            selectedScopes: k.value,
            "onUpdate:selectedScopes": e[11] || (e[11] = (n) => V("update:selectedScopes", n))
          }, null, 8, ["flow", "flowType", "selectedScopes"])
        ]),
        _: 1
      }),
      r(o(u), { class: "min-w-full" }, {
        default: t(() => [
          U("div", I, [
            r(o(S), {
              class: "mr-0.75 p-0 px-2 py-0.5",
              loader: o(p),
              size: "sm",
              variant: "outlined",
              onClick: T
            }, {
              default: t(() => [...e[23] || (e[23] = [
                s(" Authorize ", -1)
              ])]),
              _: 1
            }, 8, ["loader"])
          ])
        ]),
        _: 1
      })
    ], 64));
  }
});
export {
  X as default
};
