// @scalar/api-client/dist/v2/blocks/scalar-auth-selector-block/components/RequestAuthTab.vue.js
import { defineComponent as A, computed as N, ref as P, createElementBlock as u, openBlock as l, Fragment as c, renderList as g, createBlock as y, createCommentVNode as f, unref as d, withCtx as r, createVNode as i, createElementVNode as V, toDisplayString as R, createTextVNode as v, normalizeClass as z, capitalize as Q } from "vue";
import { ScalarMarkdownSummary as _ } from "@scalar/components";
import { getResolvedRef as q } from "@scalar/workspace-store/helpers/get-resolved-ref";
import D from "./OAuth2.vue.js";
import b from "./RequestAuthDataTableInput.vue.js";
import m from "../../../components/data-table/DataTableRow.vue.js";
import I from "../../../components/data-table/DataTableCell.vue.js";
const E = { class: "bg-b-1 text-c-2 outline-b-3 top-0 z-1 h-full w-full overflow-hidden px-3 py-1.25 text-ellipsis group-hover/auth:absolute group-hover/auth:h-auto group-hover/auth:border-b *:first:line-clamp-1 *:first:text-ellipsis group-hover/auth:*:first:line-clamp-none" }, H = { class: "flex min-h-8 border-t text-base" }, L = { class: "flex h-8 max-w-full gap-2.5 overflow-x-auto px-3" }, G = ["onClick"], J = { class: "relative z-10" }, le = /* @__PURE__ */ A({
  __name: "RequestAuthTab",
  props: {
    environment: {},
    isStatic: { type: Boolean },
    proxyUrl: {},
    selectedSecuritySchemas: {},
    securitySchemes: {},
    server: {}
  },
  emits: ["update:securityScheme", "update:selectedScopes"],
  setup(n, { emit: K }) {
    const x = K, $ = N(
      () => Object.entries(n.selectedSecuritySchemas).map(([o, t = []]) => ({
        scheme: q(n.securitySchemes[o]),
        name: o,
        scopes: t
      }))
    ), w = P(""), C = N(() => $.value.length > 1), F = (o, t) => {
      const e = Q(o), a = t.description ? `: ${t.description}` : "";
      switch (t.type) {
        case "apiKey":
          return `${e}${a || `: ${t.in}`}`;
        case "oauth2": {
          const k = Object.keys(t.flows ?? {})[0], s = w.value || k;
          return `${e}: ${s}${a}`;
        }
        case "http":
          return `${e}: ${t.scheme}${a}`;
        default:
          return `${e}${a || `: ${t.type}`}`;
      }
    }, T = (o, t) => w.value === o || t === 0 && !w.value, B = () => n.isStatic && "border-t", h = (o, t, e) => x(
      "update:securityScheme",
      {
        type: "http",
        [o]: t
      },
      e
    ), j = (o, t, e) => x(
      "update:securityScheme",
      {
        type: "apiKey",
        [o]: t
      },
      e
    ), M = (o, t) => {
      x("update:selectedScopes", {
        id: Object.keys(n.selectedSecuritySchemas),
        name: o,
        ...t
      });
    }, O = (o, t) => {
      const e = "floating-bg text-c-3 relative cursor-pointer border-b border-transparent py-1 text-base font-medium";
      return T(o, t) ? `${e} !text-c-1 !rounded-none border-b !border-current ${n.isStatic ? "opacity-100" : ""}` : e;
    };
    return (o, t) => (l(!0), u(c, null, g($.value, ({ scheme: e, name: a, scopes: k }) => (l(), u(c, { key: a }, [
      C.value && e ? (l(), y(d(m), { key: 0 }, {
        default: r(() => [
          i(d(I), {
            "aria-label": F(a, e),
            class: "text-c-2 group/auth flex items-center leading-[22px] whitespace-nowrap outline-none hover:whitespace-normal"
          }, {
            default: r(() => [
              V("p", E, R(F(a, e)), 1)
            ]),
            _: 2
          }, 1032, ["aria-label"])
        ]),
        _: 2
      }, 1024)) : f("", !0),
      e?.description && !C.value ? (l(), y(d(m), { key: 1 }, {
        default: r(() => [
          i(d(I), {
            "aria-label": e.description,
            class: "max-h-[auto]"
          }, {
            default: r(() => [
              i(d(_), {
                class: "auth-description bg-b-1 text-c-2 min-w-0 flex-1 px-3 py-1.25",
                value: e.description
              }, null, 8, ["value"])
            ]),
            _: 2
          }, 1032, ["aria-label"])
        ]),
        _: 2
      }, 1024)) : f("", !0),
      e?.type === "http" ? (l(), u(c, { key: 2 }, [
        e.scheme === "bearer" ? (l(), y(d(m), { key: 0 }, {
          default: r(() => [
            i(b, {
              containerClass: B(),
              environment: n.environment,
              modelValue: e["x-scalar-secret-token"],
              placeholder: "Token",
              type: "password",
              "onUpdate:modelValue": (s) => h("x-scalar-secret-token", s, a)
            }, {
              default: r(() => [...t[0] || (t[0] = [
                v(" Bearer Token ", -1)
              ])]),
              _: 1
            }, 8, ["containerClass", "environment", "modelValue", "onUpdate:modelValue"])
          ]),
          _: 2
        }, 1024)) : e?.scheme === "basic" ? (l(), u(c, { key: 1 }, [
          i(d(m), null, {
            default: r(() => [
              i(b, {
                class: "text-c-2",
                environment: n.environment,
                modelValue: e["x-scalar-secret-username"],
                placeholder: "janedoe",
                required: "",
                "onUpdate:modelValue": (s) => h("x-scalar-secret-username", s, a)
              }, {
                default: r(() => [...t[1] || (t[1] = [
                  v(" Username ", -1)
                ])]),
                _: 1
              }, 8, ["environment", "modelValue", "onUpdate:modelValue"])
            ]),
            _: 2
          }, 1024),
          i(d(m), null, {
            default: r(() => [
              i(b, {
                environment: n.environment,
                modelValue: e["x-scalar-secret-password"],
                placeholder: "********",
                type: "password",
                "onUpdate:modelValue": (s) => h("x-scalar-secret-password", s, a)
              }, {
                default: r(() => [...t[2] || (t[2] = [
                  v(" Password ", -1)
                ])]),
                _: 1
              }, 8, ["environment", "modelValue", "onUpdate:modelValue"])
            ]),
            _: 2
          }, 1024)
        ], 64)) : f("", !0)
      ], 64)) : e?.type === "apiKey" ? (l(), u(c, { key: 3 }, [
        i(d(m), null, {
          default: r(() => [
            i(b, {
              containerClass: B(),
              environment: n.environment,
              modelValue: e.name,
              placeholder: "api-key",
              "onUpdate:modelValue": (s) => j("name", s, a)
            }, {
              default: r(() => [...t[3] || (t[3] = [
                v(" Name ", -1)
              ])]),
              _: 1
            }, 8, ["containerClass", "environment", "modelValue", "onUpdate:modelValue"])
          ]),
          _: 2
        }, 1024),
        i(d(m), null, {
          default: r(() => [
            i(b, {
              environment: n.environment,
              modelValue: e["x-scalar-secret-token"],
              placeholder: "QUxMIFlPVVIgQkFTRSBBUkUgQkVMT05HIFRPIFVT",
              type: "password",
              "onUpdate:modelValue": (s) => j("x-scalar-secret-token", s, a)
            }, {
              default: r(() => [...t[4] || (t[4] = [
                v(" Value ", -1)
              ])]),
              _: 1
            }, 8, ["environment", "modelValue", "onUpdate:modelValue"])
          ]),
          _: 2
        }, 1024)
      ], 64)) : e?.type === "oauth2" ? (l(), u(c, { key: 4 }, [
        Object.keys(e.flows).length > 1 ? (l(), y(d(m), { key: 0 }, {
          default: r(() => [
            V("div", H, [
              V("div", L, [
                (l(!0), u(c, null, g(e.flows, (s, p, U) => (l(), u("button", {
                  key: p,
                  class: z(O(p, U)),
                  type: "button",
                  onClick: (S) => w.value = p
                }, [
                  V("span", J, R(p), 1)
                ], 10, G))), 128))
              ])
            ])
          ]),
          _: 2
        }, 1024)) : f("", !0),
        (l(!0), u(c, null, g(e.flows, (s, p, U) => (l(), u(c, { key: p }, [
          T(p, U) ? (l(), y(D, {
            key: 0,
            environment: n.environment,
            flows: e.flows,
            proxyUrl: n.proxyUrl,
            selectedScopes: k,
            server: n.server,
            type: p,
            "onUpdate:securityScheme": (S) => x("update:securityScheme", S, a),
            "onUpdate:selectedScopes": (S) => M(a, S)
          }, null, 8, ["environment", "flows", "proxyUrl", "selectedScopes", "server", "type", "onUpdate:securityScheme", "onUpdate:selectedScopes"])) : f("", !0)
        ], 64))), 128))
      ], 64)) : e?.type === "openIdConnect" ? (l(), u("div", {
        key: 5,
        class: z(["text-c-3 bg-b-1 flex min-h-[calc(4rem+1px)] items-center justify-center border-t border-b-0 px-4 text-base", { "rounded-b-lg": n.isStatic }])
      }, " Coming soon ", 2)) : f("", !0)
    ], 64))), 128));
  }
});
export {
  le as default
};
