// @scalar/api-client/dist/v2/blocks/scalar-auth-selector-block/components/RequestAuthDataTable.vue.js
import { defineComponent as f, computed as d, createElementBlock as c, openBlock as a, withModifiers as y, createCommentVNode as m, createBlock as g, normalizeClass as s, Fragment as p, renderList as k, createElementVNode as u, toDisplayString as w, unref as B, withCtx as U, createVNode as C } from "vue";
import A from "./RequestAuthTab.vue.js";
import z from "../../../components/data-table/DataTable.vue.js";
const N = ["onClick"], O = { class: "relative z-10 font-medium whitespace-nowrap" }, T = {
  key: 0,
  class: "absolute inset-x-1 bottom-[var(--scalar-border-width)] left-1/2 z-1 h-px w-full -translate-x-1/2 bg-current"
}, E = /* @__PURE__ */ f({
  __name: "RequestAuthDataTable",
  props: {
    environment: {},
    isStatic: { type: Boolean },
    selectedSchemeOptions: {},
    activeAuthIndex: {},
    proxyUrl: {},
    securitySchemes: { default: () => ({}) },
    server: {},
    eventBus: {},
    meta: {}
  },
  setup(e, { expose: h }) {
    const r = d(
      () => e.selectedSchemeOptions[e.activeAuthIndex]
    ), b = d(() => e.selectedSchemeOptions.length > 1), v = (t) => e.eventBus.emit("auth:update:active-index", {
      index: t,
      meta: e.meta
    }), x = (t, n) => e.eventBus.emit("auth:update:security-scheme", {
      payload: t,
      name: n
    }), S = (t) => e.eventBus.emit("auth:update:selected-scopes", {
      ...t,
      meta: e.meta
    }), o = (t) => e.activeAuthIndex === t;
    return h({
      activeScheme: r
    }), (t, n) => (a(), c("form", {
      onSubmit: n[0] || (n[0] = y(() => {
      }, ["prevent"]))
    }, [
      b.value ? (a(), c("div", {
        key: 0,
        class: s(["box-content flex flex-wrap gap-x-2.5 overflow-hidden border border-b-0 px-3", { "border-x-0": !e.isStatic }]),
        "data-testid": "auth-tabs"
      }, [
        (a(!0), c(p, null, k(e.selectedSchemeOptions, (l, i) => (a(), c("div", {
          key: l.id,
          class: "relative z-1 -mb-[var(--scalar-border-width)] flex h-8"
        }, [
          u("button", {
            class: s(["floating-bg relative cursor-pointer border-b border-transparent py-1 text-sm font-medium transition-colors", o(i) ? "text-c-1" : "text-c-3"]),
            type: "button",
            onClick: (I) => v(i)
          }, [
            u("span", O, w(l.label), 1)
          ], 10, N),
          o(i) ? (a(), c("div", T)) : m("", !0)
        ]))), 128))
      ], 2)) : m("", !0),
      r.value ? (a(), g(B(z), {
        key: 1,
        class: s(["flex-1", { "bg-b-1 rounded-b-lg border border-t-0": e.isStatic }]),
        columns: [""],
        presentational: ""
      }, {
        default: U(() => [
          C(A, {
            environment: e.environment,
            isStatic: e.isStatic,
            proxyUrl: e.proxyUrl,
            securitySchemes: e.securitySchemes,
            selectedSecuritySchemas: r.value.value,
            server: e.server,
            "onUpdate:securityScheme": x,
            "onUpdate:selectedScopes": S
          }, null, 8, ["environment", "isStatic", "proxyUrl", "securitySchemes", "selectedSecuritySchemas", "server"])
        ]),
        _: 1
      }, 8, ["class"])) : (a(), c("div", {
        key: 2,
        class: s(["bg-b-1 text-c-3 flex min-h-16 items-center justify-center border-t px-4 text-sm", { "min-h-[calc(4rem+0.5px)] rounded-b-lg border": e.isStatic }])
      }, " No authentication selected ", 2))
    ], 32));
  }
});
export {
  E as default
};
