// @scalar/api-client/dist/v2/blocks/scalar-auth-selector-block/components/RequestAuthDataTableInput.vue.js
import { defineComponent as f, mergeModels as d, useModel as p, useId as V, createBlock as y, openBlock as c, unref as a, mergeProps as v, withCtx as i, renderSlot as s, createElementVNode as C } from "vue";
import B from "../../../components/data-table/DataTableInput.vue.js";
const b = ["for"], O = /* @__PURE__ */ f({
  __name: "RequestAuthDataTableInput",
  props: /* @__PURE__ */ d({
    containerClass: {},
    environment: {},
    readOnly: { type: Boolean, default: !1 },
    required: { type: Boolean, default: !1 },
    type: {}
  }, {
    modelValue: { default: "", required: !0 },
    modelModifiers: {}
  }),
  emits: /* @__PURE__ */ d(["inputFocus", "inputBlur", "selectVariable"], ["update:modelValue"]),
  setup(t, { emit: m }) {
    const l = m, u = p(t, "modelValue"), r = V();
    return (o, e) => (c(), y(a(B), v({ id: a(r) }, o.$attrs, {
      modelValue: u.value,
      "onUpdate:modelValue": e[0] || (e[0] = (n) => u.value = n),
      canAddCustomEnumValue: !t.readOnly,
      containerClass: t.containerClass,
      environment: t.environment,
      lineWrapping: "",
      readOnly: t.readOnly,
      required: t.required,
      type: t.type,
      onInputBlur: e[1] || (e[1] = (n) => l("inputBlur")),
      onInputFocus: e[2] || (e[2] = (n) => l("inputFocus")),
      onSelectVariable: e[3] || (e[3] = (n) => l("selectVariable", n))
    }), {
      default: i(() => [
        C("label", { for: a(r) }, [
          s(o.$slots, "default")
        ], 8, b)
      ]),
      icon: i(() => [
        s(o.$slots, "icon")
      ]),
      _: 3
    }, 16, ["id", "modelValue", "canAddCustomEnumValue", "containerClass", "environment", "readOnly", "required", "type"]));
  }
});
export {
  O as default
};
