// @scalar/api-reference/dist/legacy/helpers/getRequestFromAuthentication.js
import { encodeStringAsBase64 as k } from "./encodeStringAsBase64.js";
function i(e) {
  return !(!e || Array.isArray(e) && !e.length || (e ?? []).some(
    (r) => !Object.keys(r).length
  ));
}
function K(e, r) {
  var o, c;
  const n = [], p = [], l = [];
  if (!e.customSecurity && (!e.preferredSecurityScheme || !i(r)))
    return { headers: n, queryString: p, cookies: l };
  const m = (r == null ? void 0 : r.some(
    (t) => e.preferredSecurityScheme && Object.keys(t).includes(
      e.preferredSecurityScheme
    )
  )) || e.customSecurity ? e.preferredSecurityScheme : Object.keys((r == null ? void 0 : r[0]) ?? {}).pop(), s = (o = e.securitySchemes) == null ? void 0 : o[m ?? ""];
  if (s) {
    if ("type" in s && s.type === "apiKey") {
      if ("in" in s && s.in === "header") {
        const t = (c = e.apiKey.token) != null && c.length ? e.apiKey.token : "YOUR_TOKEN";
        n.push({
          name: "name" in s ? s.name : "",
          value: t
        });
      } else if ("in" in s && s.in === "cookie") {
        const t = e.apiKey.token.length ? e.apiKey.token : "YOUR_TOKEN";
        l.push({
          name: s.name,
          value: t
        });
      } else if ("in" in s && s.in === "query") {
        const t = e.apiKey.token.length ? e.apiKey.token : "YOUR_TOKEN";
        p.push({
          name: s.name,
          value: t
        });
      }
    } else if ("type" in s && (s.type === "http" || s.type === "basic")) {
      if ("type" in s && (s.type === "basic" || s.type === "http" && s.scheme === "basic")) {
        const { username: t, password: y } = e.http.basic, h = f(t, y);
        n.push({
          name: "Authorization",
          value: `Basic ${h}`.trim()
        });
      } else if ("type" in s && s.type === "http" && s.scheme === "bearer") {
        const t = e.http.bearer.token.length ? e.http.bearer.token : "YOUR_SECRET_TOKEN";
        n.push({
          name: "Authorization",
          value: `Bearer ${t}`
        });
      }
    } else if ("type" in s && s.type.toLowerCase() === "oauth2") {
      const t = e.oAuth2.accessToken || "YOUR_SECRET_TOKEN";
      n.push({
        name: "Authorization",
        value: `Bearer ${t}`
      });
    }
  }
  return { headers: n, queryString: p, cookies: l };
}
function f(e, r) {
  return e != null && e.length || r != null && r.length ? k(`${e}:${r}`) : "";
}
function S(e) {
  return [
    e.apiKey.token,
    e.http.bearer.token,
    e.oAuth2.accessToken,
    // The basic auth token is the base64 encoded username and password
    f(
      e.http.basic.username,
      e.http.basic.password
    ),
    // The plain text password shouldn’t appear anyway, but just in case
    e.http.basic.password
  ].filter(Boolean);
}
export {
  f as getBase64Token,
  K as getRequestFromAuthentication,
  S as getSecretCredentialsFromAuthentication
};
