// @scalar/api-client/dist/components/SideNav/SideNavLink.vue.js
import { defineComponent as a, resolveComponent as s, openBlock as i, createBlock as c, unref as t, withCtx as p, createElementVNode as o, normalizeClass as l, createVNode as m, toDisplayString as u, renderSlot as d } from "vue";
import { useWorkspace as f } from "../../store/workspace.js";
import { ScalarIcon as v } from "@scalar/components";
import { useRouter as _ } from "vue-router";
const k = { class: "no-underline font-medium text-[11px] hidden scalar-app-show capitalize" }, x = { class: "sr-only" }, b = /* @__PURE__ */ a({
  __name: "SideNavLink",
  props: {
    icon: {},
    name: {},
    prettyName: {},
    active: { type: Boolean }
  },
  setup(g) {
    const { activeWorkspace: n } = f();
    return _(), (e, h) => {
      const r = s("router-link");
      return i(), c(r, {
        activeClass: "active-link",
        class: "flex flex-col items-center gap-1 group no-underline",
        to: `/workspace/${t(n).uid}/${e.name}/default`
      }, {
        default: p(() => [
          o("div", {
            class: l(["min-w-[37px] max-w-[42px] group-hover:bg-b-3 active:text-c-1 flex items-center justify-center rounded-lg p-[7px] scalar-app-nav-padding text-c-3", {
              "bg-b-3 transition-none group-hover:cursor-auto !text-c-1": e.active
            }])
          }, [
            m(t(v), {
              icon: e.icon,
              thickness: "1.5"
            }, null, 8, ["icon"])
          ], 2),
          o("div", k, u(e.prettyName), 1),
          o("span", x, [
            d(e.$slots, "default")
          ])
        ]),
        _: 3
      }, 8, ["to"]);
    };
  }
});
export {
  b as default
};
