// @scalar/api-client/dist/components/SideNav/SideNav.vue2.js
import { defineComponent as _, openBlock as r, createElementBlock as a, createVNode as e, createElementVNode as t, Fragment as f, renderList as d, unref as n, withCtx as p, createTextVNode as u, toDisplayString as g } from "vue";
import x from "../DarkModeToggle/DarkModeIconToggle.vue.js";
import { ROUTES as h } from "../../constants.js";
import { useRouter as v } from "vue-router";
import N from "./SideHelp.vue.js";
import y from "./SideNavLink.vue.js";
import S from "./WorkspaceProfileIcon.vue.js";
/* empty css                          */
const $ = {
  "aria-label": "Side Navigation",
  class: "text-c-2 w-15 flex flex-col items-center px-2 py-2 scalar-sidenav relative drag-region",
  role: "navigation"
}, k = { class: "mt-3.5 flex flex-col gap-1.5" }, E = { class: "mt-auto flex flex-col py-1.5" }, R = { class: "flex items-center no-drag-region" }, V = { class: "flex items-center no-drag-region" }, j = /* @__PURE__ */ _({
  __name: "SideNav",
  setup(w) {
    const { currentRoute: s } = v();
    return (B, C) => (r(), a("nav", $, [
      e(S, { class: "no-drag-region" }),
      t("ul", k, [
        (r(!0), a(f, null, d(n(h), ({ icon: c, name: o, prettyName: l }, m) => {
          var i;
          return r(), a("li", {
            key: m,
            class: "no-drag-region"
          }, [
            e(y, {
              active: (i = n(s).name) == null ? void 0 : i.startsWith(o),
              icon: c,
              name: o,
              prettyName: l
            }, {
              default: p(() => [
                u(g(o), 1)
              ]),
              _: 2
            }, 1032, ["active", "icon", "name", "prettyName"])
          ]);
        }), 128))
      ]),
      t("ul", E, [
        t("li", R, [
          e(N)
        ]),
        t("li", V, [
          e(x)
        ])
      ])
    ]));
  }
});
export {
  j as default
};
