// @scalar/api-client/dist/components/SideNav/SideHelp.vue.js
import { defineComponent as l, openBlock as o, createBlock as i, unref as s, withCtx as c, createElementVNode as e, createVNode as t } from "vue";
import { ScalarDropdown as r, ScalarDropdownItem as a, ScalarIcon as n } from "@scalar/components";
const d = {
  class: "min-w-[37px] max-w-[42px] hover:bg-b-3 flex items-center justify-center rounded-lg p-[7px] text-c-3 focus:text-c-1 scalar-app-nav-padding",
  type: "button"
}, _ = /* @__PURE__ */ e("span", { class: "px-2.5 py-1.5 text-xs text-c-2 font-medium" }, "Support", -1), p = {
  class: "no-underline block",
  href: "https://discord.gg/scalar",
  target: "_blank"
}, u = { class: "flex items-center justify-center" }, f = /* @__PURE__ */ e("span", null, "Discord", -1), h = {
  class: "no-underline block",
  href: "https://github.com/scalar/scalar",
  target: "_blank"
}, m = { class: "flex items-center justify-center" }, x = /* @__PURE__ */ e("span", null, "GitHub", -1), g = {
  class: "no-underline block",
  href: "mailto:marc@scalar.com",
  target: "_blank"
}, b = { class: "flex items-center justify-center" }, k = /* @__PURE__ */ e("span", null, "Email", -1), w = /* @__PURE__ */ e("span", { class: "px-2.5 py-1.5 text-xs text-c-2 font-medium" }, "Product", -1), y = {
  class: "no-underline block",
  href: "https://scalar.com/changelog",
  target: "_blank"
}, v = { class: "flex items-center justify-center" }, j = /* @__PURE__ */ e("span", null, "Changelog", -1), z = {
  class: "no-underline block",
  href: "https://github.com/scalar/scalar/issues/2669",
  target: "_blank"
}, S = { class: "flex items-center justify-center" }, C = /* @__PURE__ */ e("span", null, "Roadmap", -1), I = /* @__PURE__ */ l({
  __name: "SideHelp",
  setup(D) {
    return (H, E) => (o(), i(s(r), {
      class: "max-w-[150px]",
      placement: "top-end"
    }, {
      items: c(() => [
        _,
        e("a", p, [
          t(s(a), { class: "flex items-center gap-1.5 w-full" }, {
            default: c(() => [
              e("div", u, [
                t(s(n), {
                  icon: "DiscordLine",
                  size: "xs",
                  thickness: "1.75"
                })
              ]),
              f
            ]),
            _: 1
          })
        ]),
        e("a", h, [
          t(s(a), { class: "flex items-center gap-1.5 w-full" }, {
            default: c(() => [
              e("div", m, [
                t(s(n), {
                  icon: "GitHubLine",
                  size: "xs",
                  thickness: "1.75"
                })
              ]),
              x
            ]),
            _: 1
          })
        ]),
        e("a", g, [
          t(s(a), { class: "flex items-center gap-1.5 w-full" }, {
            default: c(() => [
              e("div", b, [
                t(s(n), {
                  icon: "Email",
                  size: "xs"
                })
              ]),
              k
            ]),
            _: 1
          })
        ]),
        w,
        e("a", y, [
          t(s(a), { class: "flex items-center gap-1.5 w-full" }, {
            default: c(() => [
              e("div", v, [
                t(s(n), {
                  icon: "Changelog",
                  size: "xs",
                  thickness: "1.75"
                })
              ]),
              j
            ]),
            _: 1
          })
        ]),
        e("a", z, [
          t(s(a), { class: "flex items-center gap-1.5 w-full" }, {
            default: c(() => [
              e("div", S, [
                t(s(n), {
                  icon: "Roadmap",
                  size: "xs",
                  thickness: "1.75"
                })
              ]),
              C
            ]),
            _: 1
          })
        ])
      ]),
      default: c(() => [
        e("button", d, [
          t(s(n), {
            icon: "Help",
            thickness: "1.5"
          })
        ])
      ]),
      _: 1
    }));
  }
});
export {
  I as default
};
