// @scalar/api-reference/dist/components/Content/Tag/Endpoints.vue2.js
import { defineComponent as v, openBlock as n, createBlock as s, unref as e, withCtx as o, createVNode as a, createTextVNode as c, toDisplayString as p, createElementVNode as u, createElementBlock as f, Fragment as y, renderList as N, normalizeClass as V, createCommentVNode as b } from "vue";
import { ScalarMarkdown as E } from "@scalar/components";
import I from "../../Section/Section.vue.js";
import w from "../../Section/SectionContent.vue.js";
import B from "../../Section/SectionColumns.vue.js";
import g from "../../Section/SectionColumn.vue.js";
import H from "../../Section/SectionHeader.vue.js";
import T from "../../Anchor/Anchor.vue.js";
import O from "../../Card/Card.vue.js";
import $ from "../../Card/CardHeader.vue.js";
import z from "../../Card/CardContent.vue.js";
import A from "../../HttpMethod/HttpMethod.vue.js";
import { useSidebar as D } from "../../../hooks/useSidebar.js";
import { useNavState as F } from "../../../hooks/useNavState.js";
const L = { class: "endpoints" }, M = ["onClick"], te = /* @__PURE__ */ v({
  __name: "Endpoints",
  props: {
    id: {},
    tag: {}
  },
  setup(C) {
    const _ = C, { getOperationId: l, getTagId: h } = F(), { scrollToOperation: k } = D(), S = async (t) => {
      const d = l(t, _.tag);
      k(d);
    };
    return (t, d) => (n(), s(e(I), {
      id: t.id,
      label: t.tag.name.toUpperCase()
    }, {
      default: o(() => [
        a(e(w), null, {
          default: o(() => [
            a(e(B), null, {
              default: o(() => [
                a(e(g), null, {
                  default: o(() => [
                    a(e(H), { level: 2 }, {
                      default: o(() => [
                        a(e(T), {
                          id: e(h)(t.tag)
                        }, {
                          default: o(() => [
                            c(p(t.tag["x-displayName"] ?? t.tag.name), 1)
                          ]),
                          _: 1
                        }, 8, ["id"])
                      ]),
                      _: 1
                    }),
                    a(e(E), {
                      value: t.tag.description,
                      withImages: ""
                    }, null, 8, ["value"])
                  ]),
                  _: 1
                }),
                a(e(g), null, {
                  default: o(() => {
                    var i;
                    return [
                      ((i = t.tag.operations) == null ? void 0 : i.length) > 0 ? (n(), s(e(O), {
                        key: 0,
                        class: "scalar-card-sticky"
                      }, {
                        default: o(() => [
                          a(e($), { muted: "" }, {
                            default: o(() => [
                              c("Endpoints")
                            ]),
                            _: 1
                          }),
                          a(e(z), {
                            class: "custom-scroll",
                            muted: ""
                          }, {
                            default: o(() => [
                              u("div", L, [
                                (n(!0), f(y, null, N(t.tag.operations, (r) => {
                                  var m;
                                  return n(), f("a", {
                                    key: e(l)(r, t.tag),
                                    class: "endpoint",
                                    onClick: (U) => S(r)
                                  }, [
                                    a(e(A), {
                                      method: r.httpVerb
                                    }, null, 8, ["method"]),
                                    u("span", {
                                      class: V({ deprecated: (m = r.information) == null ? void 0 : m.deprecated })
                                    }, p(r.path), 3)
                                  ], 8, M);
                                }), 128))
                              ])
                            ]),
                            _: 1
                          })
                        ]),
                        _: 1
                      })) : b("", !0)
                    ];
                  }),
                  _: 1
                })
              ]),
              _: 1
            })
          ]),
          _: 1
        })
      ]),
      _: 1
    }, 8, ["id", "label"]));
  }
});
export {
  te as default
};
