// @scalar/api-reference/dist/components/Content/Tag/Tag.vue2.js
import { defineComponent as c, ref as f, computed as u, openBlock as r, createBlock as a, unref as i, withCtx as g, createCommentVNode as h, renderSlot as k } from "vue";
import C from "../../ShowMoreButton.vue.js";
import S from "./Endpoints.vue.js";
import y from "../../Section/SectionContainer.vue.js";
import { useSidebar as T } from "../../../hooks/useSidebar.js";
import { useNavState as v } from "../../../hooks/useNavState.js";
const _ = /* @__PURE__ */ c({
  __name: "Tag",
  props: {
    id: {},
    tag: {},
    spec: {}
  },
  setup(s) {
    const o = s, d = f(null), { collapsedSidebarItems: l } = T(), { getTagId: m } = v(), p = u(
      () => {
        var e, n, t;
        return ((e = o.spec.tags) == null ? void 0 : e.length) !== 1 || ((n = o.tag) == null ? void 0 : n.name) !== "default" || ((t = o.tag) == null ? void 0 : t.description) !== "";
      }
    );
    return (e, n) => (r(), a(i(y), {
      ref_key: "sectionContainerRef",
      ref: d,
      class: "tag-section-container"
    }, {
      default: g(() => {
        var t;
        return [
          p.value ? (r(), a(S, {
            key: 0,
            id: e.id,
            tag: e.tag
          }, null, 8, ["id", "tag"])) : h("", !0),
          !i(l)[i(m)(e.tag)] && ((t = e.tag.operations) == null ? void 0 : t.length) > 1 ? (r(), a(C, {
            key: 1,
            id: e.id ?? ""
          }, null, 8, ["id"])) : k(e.$slots, "default", { key: 2 }, void 0, !0)
        ];
      }),
      _: 3
    }, 512));
  }
});
export {
  _ as default
};
