// @scalar/api-client/dist/v2/features/modal/helpers/create-api-client-modal.js
import { useModal as P } from "@scalar/components";
import { createWorkspaceEventBus as A } from "@scalar/workspace-store/events";
import { reactive as C, computed as t, createApp as I } from "vue";
import { resolveRouteParameters as R } from "./resolve-route-parameters.js";
import { useModalSidebar as S } from "../hooks/use-modal-sidebar.js";
import j from "../Modal.vue.js";
const T = ({
  el: f,
  eventBus: v = A({
    debug: !1
  }),
  mountOnInitialize: h = !0,
  plugins: g = [],
  workspaceStore: a,
  options: x = {}
}) => {
  const n = {
    path: "default",
    method: "default",
    example: "default",
    documentSlug: a.workspace["x-scalar-active-document"] || "default"
  }, m = C({ ...n }), r = (e) => {
    Object.assign(m, n, e);
  }, o = t(() => R(a, m)), s = t(() => o.value.documentSlug), u = t(() => o.value.path), d = t(() => o.value.method), i = t(() => o.value.example), M = t(() => a.workspace.documents[s.value ?? ""] ?? null), b = S({
    workspaceStore: a,
    documentSlug: s,
    path: u,
    method: d,
    exampleName: i,
    route: r
  }), c = P(), l = I(j, {
    document: M,
    eventBus: v,
    exampleName: i,
    method: d,
    modalState: c,
    path: u,
    plugins: g,
    sidebarState: b,
    workspaceStore: a,
    options: x
  });
  l.config.idPrefix = "scalar-client";
  const p = (e = f) => {
    if (!e) {
      console.error(
        "[@scalar/api-client] Could not create the API client Modal.",
        "Invalid HTML element provided.",
        "Read more: https://github.com/scalar/scalar/tree/main/packages/api-client"
      );
      return;
    }
    l.mount(e);
  };
  return h && p(), {
    /** The Vue app instance for the modal. Use with caution. */
    app: l,
    /** Open the modal and optionally navigate to a specific route. */
    open: (e) => {
      c.open = !0, e && r(e);
    },
    /** Mount the modal to a given element. */
    mount: p,
    /** Navigate to the specified path, method, and example. */
    route: r,
    /** Controls the visibility of the modal. */
    modalState: c
  };
};
export {
  T as createApiClientModal
};
