// @scalar/api-client/dist/components/TopNav/TopNavItem.vue2.js
import { defineComponent as m, openBlock as i, createBlock as s, unref as n, withCtx as a, createElementVNode as o, normalizeClass as p, createVNode as l, toDisplayString as k, createCommentVNode as f } from "vue";
import v from "../ScalarHotkey.vue.js";
import { ScalarTooltip as y, ScalarIcon as c } from "@scalar/components";
const d = { class: "nav-item-icon-copy flex flex-1 items-center justify-center gap-1.5" }, u = { class: "nav-item-copy text-xs" }, g = /* @__PURE__ */ m({
  __name: "TopNavItem",
  props: {
    hotkey: {},
    active: { type: Boolean },
    label: {},
    icon: {}
  },
  emits: ["click", "close"],
  setup(h) {
    return (e, t) => (i(), s(n(y), {
      class: "scalar-client",
      delay: 500,
      resize: "",
      sideOffset: 4
    }, {
      trigger: a(() => [
        o("div", {
          class: p(["nav-item webkit-app-no-drag", { "nav-item__active": e.active }]),
          onClick: t[1] || (t[1] = (r) => e.$emit("click"))
        }, [
          o("div", d, [
            l(n(c), {
              icon: e.icon,
              size: "xs",
              thickness: "2.5"
            }, null, 8, ["icon"]),
            o("span", u, k(e.label), 1)
          ]),
          o("button", {
            class: "nav-item-close",
            type: "button",
            onClick: t[0] || (t[0] = (r) => e.$emit("close"))
          }, [
            l(n(c), {
              icon: "Close",
              thickness: "1.75"
            })
          ])
        ], 2)
      ]),
      content: a(() => [
        e.hotkey ? (i(), s(v, {
          key: 0,
          hotkey: e.hotkey
        }, null, 8, ["hotkey"])) : f("", !0)
      ]),
      _: 1
    }));
  }
});
export {
  g as default
};
