// @scalar/api-client/dist/components/TopNav/TopNav.vue2.js
import { defineComponent as T, reactive as x, ref as S, watch as C, computed as j, onMounted as B, onBeforeUnmount as R, openBlock as u, createElementBlock as p, createElementVNode as s, createVNode as b, unref as _, toDisplayString as w, Fragment as A, renderList as L, createBlock as M, pushScopeId as z, popScopeId as E } from "vue";
import { ROUTES as V } from "../../constants.js";
import { ScalarIcon as g } from "@scalar/components";
import { capitalize as q } from "@scalar/oas-utils/helpers";
import K from "./TopNavItem.vue.js";
import { useWorkspace as U } from "../../store/workspace.js";
import { hotKeyBus as k } from "../../libs/event-busses/hot-keys-bus.js";
const $ = (i) => (z("data-v-91b3e5a4"), i = i(), E(), i), D = { class: "flex h-10 t-app__top-nav" }, F = /* @__PURE__ */ $(() => /* @__PURE__ */ s("div", { class: "t-app__top-nav-draggable" }, null, -1)), H = { class: "flex h-10 flex-1 items-center justify-center gap-1.5 text-sm font-medium relative" }, O = {
  key: 0,
  class: "flex items-center gap-1 w-full justify-center"
}, Z = /* @__PURE__ */ T({
  __name: "TopNav",
  setup(i) {
    const { activeRequest: N, router: n } = U(), t = x([{ label: "", path: "", icon: "Add" }]), a = S(0);
    function m() {
      var o;
      const e = V.find((r) => n.currentRoute.value.name == r.name);
      e && ((e == null ? void 0 : e.name) === "request" ? t[a.value] = {
        label: ((o = N.value) == null ? void 0 : o.summary) || "",
        path: n.currentRoute.value.path,
        icon: e.icon
      } : t[a.value] = {
        label: q(e == null ? void 0 : e.name) || "",
        path: n.currentRoute.value.path,
        icon: e.icon
      });
    }
    function v() {
      n.push(t[a.value].path);
    }
    function d() {
      t.push({ label: "", path: "", icon: "Add" }), a.value = t.length - 1, m();
    }
    function l(e) {
      a.value = e, v();
    }
    C(
      () => n.currentRoute.value.path,
      () => {
        m();
      },
      { immediate: !0 }
    );
    function f(e) {
      t.splice(e, 1), a.value = Math.min(
        a.value,
        t.length - 1
      ), v();
    }
    const I = j(() => a.value), h = (e) => {
      if (e.addTopNav && d(), e.closeTopNav && f(a.value), e.navigateTopNavLeft && l(Math.max(a.value - 1, 0)), e.navigateTopNavRight && l(Math.min(a.value + 1, t.length - 1)), e.jumpToTab) {
        const o = Number(e.jumpToTab.key) - 1;
        o >= 0 && o < t.length && l(o);
      }
      e.jumpToLastTab && l(t.length - 1);
    };
    return B(() => k.on(h)), R(() => k.off(h)), (e, o) => (u(), p("nav", D, [
      F,
      s("div", H, [
        t.length === 1 ? (u(), p("div", O, [
          b(_(g), {
            icon: t[0].icon,
            size: "xs",
            thickness: "2.5"
          }, null, 8, ["icon"]),
          s("div", null, w(t[0].label), 1)
        ])) : (u(!0), p(A, { key: 1 }, L(t, (r, c) => (u(), M(K, {
          key: c,
          active: c === I.value,
          hotkey: (c + 1).toString(),
          icon: r.icon,
          label: r.label,
          onClick: (y) => l(c),
          onClose: (y) => f(c)
        }, null, 8, ["active", "hotkey", "icon", "label", "onClick", "onClose"]))), 128)),
        s("button", {
          class: "text-c-3 hover:bg-b-3 p-1.5 rounded-lg webkit-app-no-drag",
          type: "button",
          onClick: d
        }, [
          b(_(g), {
            icon: "Add",
            size: "xs",
            thickness: "2.5"
          })
        ])
      ])
    ]));
  }
});
export {
  Z as default
};
