// @scalar/components/dist/components/ScalarListbox/ScalarListbox.vue.js
import { defineComponent as b, openBlock as i, createBlock as u, unref as t, normalizeClass as n, withCtx as l, createVNode as a, mergeProps as y, createElementBlock as g, Fragment as v, renderList as h, createElementVNode as o, toDisplayString as x, renderSlot as B } from "vue";
import { Listbox as V, ListboxOptions as k, ListboxOption as z, ListboxButton as w } from "@headlessui/vue";
import { cva as L, cx as O } from "../../cva.js";
import S from "../ScalarFloating/ScalarFloating.vue.js";
import _ from "../ScalarIcon/ScalarIcon.vue.js";
const C = { class: "inline-block min-w-0 flex-1 truncate text-c-1" }, $ = /* @__PURE__ */ o("div", { class: "absolute inset-0 -z-1 rounded bg-b-1 shadow-md brightness-lifted" }, null, -1), F = /* @__PURE__ */ b({
  inheritAttrs: !1,
  __name: "ScalarListbox",
  props: {
    multiple: { type: Boolean, default: !1 },
    options: {},
    modelValue: {},
    fullWidth: { type: Boolean },
    id: {},
    placement: {},
    resize: { type: Boolean },
    isOpen: { type: Boolean },
    teleport: { type: [Boolean, String] }
  },
  emits: ["update:modelValue"],
  setup(j) {
    const m = L({
      base: [
        // Layout
        "group",
        "flex min-w-0 items-center gap-1.5 rounded px-2 py-1.5 text-left",
        "first-of-type:mt-0.75 last-of-type:mb-0.75",
        // Text / background style
        "truncate bg-transparent text-c-1",
        // Interaction
        "cursor-pointer hover:bg-b-2"
      ],
      variants: {
        selected: { true: "text-c-1" },
        active: { true: "bg-b-2" },
        disabled: { true: "pointer-events-none opacity-50" }
      }
    });
    return (e, d) => (i(), u(t(V), {
      as: "div",
      class: n([{ "w-full": e.fullWidth }, "text-sm"]),
      modelValue: e.modelValue,
      multiple: e.multiple,
      "onUpdate:modelValue": d[0] || (d[0] = (r) => e.$emit("update:modelValue", r))
    }, {
      default: l(({ open: r }) => [
        a(t(S), {
          isOpen: r ?? e.isOpen,
          placement: e.placement ?? "bottom-start",
          resize: e.resize,
          teleport: e.teleport
        }, {
          floating: l(({ width: f }) => [
            a(t(k), y({
              class: "relative flex w-40 flex-col rounded border p-0.75",
              style: { width: f }
            }, e.$attrs), {
              default: l(() => [
                (i(!0), g(v, null, h(e.options, (s) => (i(), u(t(z), {
                  key: s.id,
                  as: "template",
                  disabled: s.disabled,
                  value: s
                }, {
                  default: l(({ active: c, selected: p }) => [
                    o("li", {
                      class: n(
                        t(O)(t(m)({ active: c, selected: p, disabled: s.disabled }))
                      )
                    }, [
                      o("div", {
                        class: n(["flex size-4 items-center justify-center rounded-full p-[3px] group-hover:shadow-border", p ? "bg-blue text-b-1" : "text-transparent"])
                      }, [
                        a(t(_), {
                          class: "relative top-[0.5px] size-2.5",
                          icon: "Checkmark",
                          thickness: "2.5"
                        })
                      ], 2),
                      o("span", C, x(s.label), 1)
                    ], 2)
                  ]),
                  _: 2
                }, 1032, ["disabled", "value"]))), 128)),
                $
              ]),
              _: 2
            }, 1040, ["style"])
          ]),
          default: l(() => [
            a(t(w), {
              id: e.id,
              as: "template",
              class: "justify-start"
            }, {
              default: l(() => [
                B(e.$slots, "default")
              ]),
              _: 3
            }, 8, ["id"])
          ]),
          _: 2
        }, 1032, ["isOpen", "placement", "resize", "teleport"])
      ]),
      _: 3
    }, 8, ["class", "modelValue", "multiple"]));
  }
});
export {
  F as default
};
