// @scalar/components/dist/components/ScalarListbox/ScalarListboxItem.vue.js
import { defineComponent as c, createBlock as r, openBlock as s, unref as t, withCtx as m, createElementVNode as i, normalizeClass as a, createVNode as u, toDisplayString as d } from "vue";
import { ListboxOption as p } from "@headlessui/vue";
import { cva as b, cx as f } from "@scalar/use-hooks/useBindCx";
import x from "./ScalarListboxCheckbox.vue.js";
const B = /* @__PURE__ */ c({
  __name: "ScalarListboxItem",
  props: {
    option: {},
    multiselect: { type: Boolean }
  },
  setup(e) {
    const n = b({
      base: [
        // Layout
        "group/item",
        "flex min-w-0 items-center gap-1.5 rounded px-2 py-1.5 text-left",
        // Text / background style
        "truncate bg-transparent text-c-1",
        // Interaction
        "cursor-pointer hover:bg-b-2"
      ],
      variants: {
        selected: { true: "text-c-1" },
        active: { true: "bg-b-2" },
        disabled: { true: "pointer-events-none opacity-50" }
      }
    });
    return (v, g) => (s(), r(t(p), {
      as: "template",
      disabled: e.option.disabled,
      value: e.option
    }, {
      default: m(({ active: l, selected: o }) => [
        i("li", {
          class: a(t(f)(t(n)({ active: l, selected: o, disabled: e.option.disabled })))
        }, [
          u(x, {
            multiselect: e.multiselect,
            selected: o
          }, null, 8, ["multiselect", "selected"]),
          i("span", {
            class: a(["inline-block min-w-0 flex-1 truncate", e.option.color ? e.option.color : "text-c-1"])
          }, d(e.option.label), 3)
        ], 2)
      ]),
      _: 1
    }, 8, ["disabled", "value"]));
  }
});
export {
  B as default
};
