// @scalar/api-reference/dist/components/AgentScalar/AgentScalarDrawer.vue2.js
import { defineComponent, defineAsyncComponent, createElementBlock, openBlock, Fragment, withDirectives, createElementVNode, unref, normalizeClass, createVNode, vShow } from "vue";
import { ScalarIconX } from "@scalar/icons";
import "@scalar/agent-chat/style.css";
import { useAgentContext } from "../../hooks/use-agent.js";
const _hoisted_1 = {
  class: "app-exit-button zoomed:static zoomed:p-1 fixed top-2 right-2 rounded-full p-2",
  type: "button"
};
const _hoisted_2 = { class: "agent-scalar" };
const _hoisted_3 = { class: "agent-scalar-container custom-scroll custom-scroll-self-contain-overflow" };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "AgentScalarDrawer",
  props: {
    agentScalarConfiguration: {},
    workspaceStore: {},
    eventBus: {}
  },
  setup(__props) {
    const agentContext = useAgentContext();
    const AgentScalarChatInterface = defineAsyncComponent(
      async () => import("./AgentScalarChatInterface.vue.js")
    );
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(Fragment, null, [
        withDirectives(createElementVNode("div", {
          class: normalizeClass(["scalar-app-exit", unref(agentContext)?.showAgent.value ? "scalar-app-exit-animation" : ""]),
          onClick: _cache[0] || (_cache[0] = ($event) => unref(agentContext)?.closeAgent())
        }, [
          createElementVNode("button", _hoisted_1, [
            createVNode(unref(ScalarIconX), { weight: "bold" }),
            _cache[1] || (_cache[1] = createElementVNode("span", { class: "sr-only" }, "Close Client", -1))
          ])
        ], 2), [
          [vShow, unref(agentContext)?.showAgent.value]
        ]),
        withDirectives(createElementVNode("div", _hoisted_2, [
          createElementVNode("div", _hoisted_3, [
            createVNode(unref(AgentScalarChatInterface), {
              agentScalarConfiguration: __props.agentScalarConfiguration,
              prefilledMessage: unref(agentContext)?.prefilledMessage,
              workspaceStore: __props.workspaceStore
            }, null, 8, ["agentScalarConfiguration", "prefilledMessage", "workspaceStore"])
          ])
        ], 512), [
          [vShow, unref(agentContext)?.showAgent.value]
        ])
      ], 64);
    };
  }
});
export {
  _sfc_main as default
};
