// @scalar/api-reference/dist/components/AgentScalar/AgentScalarChatInterface.vue2.js
import { defineComponent, createBlock, openBlock, unref } from "vue";
import { Chat } from "@scalar/agent-chat";
import { REGISTRY_URL, DASHBOARD_URL, API_BASE_URL } from "../../consts/urls.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "AgentScalarChatInterface",
  props: {
    agentScalarConfiguration: {},
    workspaceStore: {},
    prefilledMessage: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createBlock(unref(Chat), {
        baseUrl: unref(API_BASE_URL),
        dashboardUrl: unref(DASHBOARD_URL),
        getActiveDocumentJson: () => __props.workspaceStore.exportActiveDocument("json"),
        getAgentKey: __props.agentScalarConfiguration?.key ? () => __props.agentScalarConfiguration?.key ?? "" : void 0,
        mode: __props.agentScalarConfiguration?.key ? "full" : "preview",
        prefilledMessage: __props.prefilledMessage,
        registryDocuments: [],
        registryUrl: unref(REGISTRY_URL)
      }, null, 8, ["baseUrl", "dashboardUrl", "getActiveDocumentJson", "getAgentKey", "mode", "prefilledMessage", "registryUrl"]);
    };
  }
});
export {
  _sfc_main as default
};
