// @scalar/api-reference/dist/components/Content/Webhooks/Webhooks.vue2.js
import { defineComponent as y, computed as v, openBlock as r, createBlock as s, unref as o, withCtx as n, createElementBlock as u, Fragment as d, renderList as S, createVNode as a, createTextVNode as g, toDisplayString as C, createCommentVNode as c, pushScopeId as I, popScopeId as N, createElementVNode as B } from "vue";
import { ScalarMarkdown as j } from "@scalar/components";
import E from "./Webhook.vue.js";
import O from "../../Section/SectionContainer.vue.js";
import W from "../../Section/Section.vue.js";
import A from "../../Section/SectionContent.vue.js";
import D from "../../Section/SectionHeader.vue.js";
import F from "../../Anchor/Anchor.vue.js";
import { useNavState as H } from "../../../hooks/useNavState.js";
const K = (l) => (I("data-v-38408a09"), l = l(), N(), l), L = /* @__PURE__ */ K(() => /* @__PURE__ */ B("div", { id: "webhooks" }, null, -1)), U = /* @__PURE__ */ y({
  __name: "Webhooks",
  props: {
    webhooks: {}
  },
  setup(l) {
    const _ = l, k = v(() => Object.keys(_.webhooks ?? {})), { getWebhookId: m } = H();
    return (i, M) => k.value.length ? (r(), s(o(O), { key: 0 }, {
      default: n(() => [
        L,
        (r(!0), u(d, null, S(k.value, (e) => {
          var p;
          return r(), u(d, { key: e }, [
            (r(!0), u(d, null, S(Object.keys(
              (p = i.webhooks) == null ? void 0 : p[e]
            ), (t) => (r(), s(o(W), {
              id: o(m)(e, t),
              key: t,
              label: e
            }, {
              default: n(() => {
                var f;
                return [
                  (f = i.webhooks) != null && f[e] ? (r(), s(o(A), { key: 0 }, {
                    default: n(() => {
                      var h, w;
                      return [
                        a(o(D), { level: 2 }, {
                          default: n(() => [
                            a(o(F), {
                              id: o(m)(e, t)
                            }, {
                              default: n(() => {
                                var b;
                                return [
                                  g(C((b = i.webhooks[e][t]) == null ? void 0 : b.name), 1)
                                ];
                              }),
                              _: 2
                            }, 1032, ["id"])
                          ]),
                          _: 2
                        }, 1024),
                        (h = i.webhooks[e][t]) != null && h.description ? (r(), s(o(j), {
                          key: 0,
                          value: (w = i.webhooks[e][t]) == null ? void 0 : w.description,
                          withImages: ""
                        }, null, 8, ["value"])) : c("", !0),
                        a(E, {
                          webhook: i.webhooks[e][t]
                        }, null, 8, ["webhook"])
                      ];
                    }),
                    _: 2
                  }, 1024)) : c("", !0)
                ];
              }),
              _: 2
            }, 1032, ["id", "label"]))), 128))
          ], 64);
        }), 128))
      ]),
      _: 1
    })) : c("", !0);
  }
});
export {
  U as default
};
