// @scalar/api-reference/dist/features/Search/SearchModal.vue2.js
import { defineComponent as F, toRef as g, ref as B, watch as T, openBlock as u, createBlock as v, unref as t, withCtx as l, createElementVNode as d, createVNode as h, isRef as U, createTextVNode as i, toDisplayString as n, createElementBlock as q, Fragment as E, renderList as L, createSlots as M, pushScopeId as D, popScopeId as H } from "vue";
import { ScalarModal as O, ScalarSearchInput as P, ScalarSearchResultList as $, ScalarSearchResultItem as K } from "@scalar/components";
import W from "../../components/Sidebar/SidebarHttpBadge.vue.js";
import { useKeyboardNavigation as Y } from "./useKeyboardNavigation.js";
import { useSearchIndex as j } from "./useSearchIndex.js";
import { useSidebar as z } from "../../hooks/useSidebar.js";
const A = (r) => (D("data-v-6ee7fa31"), r = r(), H(), r), G = /* @__PURE__ */ A(() => /* @__PURE__ */ d("div", { class: "ref-search-meta" }, [
  /* @__PURE__ */ d("span", null, "↑↓ Navigate"),
  /* @__PURE__ */ d("span", null, "⏎ Select")
], -1)), te = /* @__PURE__ */ F({
  __name: "SearchModal",
  props: {
    parsedSpec: {},
    modalState: {}
  },
  setup(r) {
    const s = r, R = g(s, "parsedSpec"), k = g(s.modalState.open), {
      resetSearch: I,
      fuseSearch: _,
      selectedSearchResult: m,
      searchResultsWithPlaceholderResults: p,
      searchText: c
    } = j({
      specification: R
    });
    Y({
      selectedSearchResult: m,
      active: k,
      searchResultsWithPlaceholderResults: p,
      onSearchResultClick: f
    });
    const w = {
      heading: "DocsPage",
      model: "Brackets",
      req: "Terminal",
      tag: "CodeFolder",
      webhook: "Terminal"
    }, C = B(null);
    T(
      () => s.modalState.open,
      (o) => {
        o && I();
      }
    );
    const { setCollapsedSidebarItem: V } = z(), b = /#(tag\/[^/]*)/;
    function f(o) {
      let a = "models";
      const e = o.item.href.match(b);
      e != null && e.length && e.length > 1 && (a = e[1]), V(a, !0), s.modalState.hide();
    }
    function N(o) {
      const a = new URL(window.location.href);
      return a.hash = o, a.toString();
    }
    return (o, a) => (u(), v(t(O), {
      state: o.modalState,
      variant: "search"
    }, {
      default: l(() => [
        d("div", {
          ref_key: "searchModalRef",
          ref: C,
          class: "ref-search-container"
        }, [
          h(t(P), {
            modelValue: t(c),
            "onUpdate:modelValue": a[0] || (a[0] = (e) => U(c) ? c.value = e : null),
            onInput: t(_)
          }, null, 8, ["modelValue", "onInput"])
        ], 512),
        h(t($), {
          class: "ref-search-results custom-scroll",
          noResults: !t(p).length
        }, {
          query: l(() => [
            i(n(t(c)), 1)
          ]),
          default: l(() => [
            (u(!0), q(E, null, L(t(p), (e, S) => (u(), v(t(K), {
              id: e.item.href,
              key: e.refIndex,
              active: t(m) === S,
              href: N(e.item.href),
              icon: w[e.item.type],
              onClick: (x) => f(e),
              onFocus: (x) => m.value = S
            }, M({
              default: l(() => [
                i(n(e.item.title) + " ", 1)
              ]),
              _: 2
            }, [
              (e.item.httpVerb || e.item.path) && e.item.path !== e.item.title ? {
                name: "description",
                fn: l(() => [
                  i(n(e.item.path), 1)
                ]),
                key: "0"
              } : e.item.description ? {
                name: "description",
                fn: l(() => [
                  i(n(e.item.description), 1)
                ]),
                key: "1"
              } : void 0,
              e.item.type === "req" ? {
                name: "addon",
                fn: l(() => [
                  h(W, {
                    method: e.item.httpVerb ?? "get"
                  }, null, 8, ["method"])
                ]),
                key: "2"
              } : void 0
            ]), 1032, ["id", "active", "href", "icon", "onClick", "onFocus"]))), 128))
          ]),
          _: 1
        }, 8, ["noResults"]),
        G
      ]),
      _: 1
    }, 8, ["state"]));
  }
});
export {
  te as default
};
