// @scalar/api-reference/dist/features/Search/SearchButton.vue2.js
import { defineComponent as m, ref as u, onMounted as f, openBlock as S, createElementBlock as _, Fragment as y, createElementVNode as a, normalizeClass as b, unref as t, createVNode as i, toDisplayString as l, pushScopeId as k, popScopeId as v } from "vue";
import { useModal as K, ScalarIcon as w } from "@scalar/components";
import { isMacOS as r } from "@scalar/use-tooltip";
import { useMagicKeys as M, whenever as g } from "@vueuse/core";
import { modalStateBus as B } from "../../components/api-client-bus.js";
import C from "./Search.vue.js";
const H = (o) => (k("data-v-f812b614"), o = o(), v(), o), I = { class: "sidebar-search-input" }, $ = /* @__PURE__ */ H(() => /* @__PURE__ */ a("span", { class: "sidebar-search-placeholder" }, "Search", -1)), E = { class: "sidebar-search-shortcut" }, z = { class: "sidebar-search-key" }, j = /* @__PURE__ */ m({
  __name: "SearchButton",
  props: {
    spec: {},
    searchHotKey: { default: "k" }
  },
  setup(o) {
    const c = o, s = K(), n = u(null), d = M({
      passive: !1,
      onEventFired(e) {
        !r() && e.ctrlKey && e.key === c.searchHotKey && (e.preventDefault(), e.stopPropagation());
      }
    });
    return f(() => {
      B.on((e) => {
        n.value = e;
      });
    }), g(
      d[`${r() ? "meta" : "control"}_${c.searchHotKey}`],
      () => {
        var e;
        (e = n.value) != null && e.open || (s.open ? s.hide() : s.show());
      }
    ), (e, p) => (S(), _(y, null, [
      a("button", {
        class: b(["sidebar-search", e.$attrs.class]),
        type: "button",
        onClick: p[0] || (p[0] = //@ts-ignore
        (...h) => t(s).show && t(s).show(...h))
      }, [
        i(t(w), {
          class: "search-icon",
          icon: "Search",
          size: "sm",
          thickness: "2.5"
        }),
        a("div", I, [
          $,
          a("span", E, [
            a("span", z, l(t(r)() ? "⌘" : "⌃") + l(e.searchHotKey), 1)
          ])
        ])
      ], 2),
      i(C, {
        modalState: t(s),
        parsedSpec: e.spec
      }, null, 8, ["modalState", "parsedSpec"])
    ], 64));
  }
});
export {
  j as default
};
