// @scalar/api-reference/dist/features/Search/useSearchIndex.js
import V from "fuse.js";
import { ref as c, computed as A, watch as B } from "vue";
import { extractRequestBody as P } from "../../helpers/specHelpers.js";
import { useSidebar as z } from "../../hooks/useSidebar.js";
import { useOperation as G } from "../../hooks/useOperation.js";
import { useNavState as J } from "../../hooks/useNavState.js";
import { getHeadingsFromMarkdown as K } from "../../helpers/getHeadingsFromMarkdown.js";
import { getModels as Q } from "../../helpers/getModels.js";
function ae({
  specification: a
}) {
  const { getHeadingId: R, getWebhookId: T, getModelId: j, getOperationId: L, getTagId: W } = J(), { hideModels: q } = z(), o = c([]), u = c([]), h = c(0), s = c(""), d = new V(o.value, {
    keys: ["title", "description", "body"]
  }), C = () => {
    h.value = 0, u.value = d.search(s.value);
  };
  function F() {
    s.value = "", h.value = 0, u.value = [];
  }
  const H = A(
    () => s.value.length === 0 ? o.value.slice(0, 25).map((r) => ({
      item: r
    })) : u.value.slice(0, 25)
  );
  return B(
    a.value,
    async () => {
      var y, g, I, w, M, k, $, D, x;
      const f = performance.now();
      if (o.value = [], !((g = (y = a.value) == null ? void 0 : y.tags) != null && g.length) && !((w = (I = a.value) == null ? void 0 : I.webhooks) != null && w.length)) {
        d.setCollection([]);
        return;
      }
      const r = [], v = K(
        ((k = (M = a.value) == null ? void 0 : M.info) == null ? void 0 : k.description) ?? ""
      );
      v.length && (v.forEach((e) => {
        r.push({
          type: "heading",
          title: `Info > ${e.value}`,
          description: "",
          href: `#${R(e)}`,
          tag: e.slug,
          body: ""
        });
      }), o.value = o.value.concat(r)), (D = ($ = a.value) == null ? void 0 : $.tags) == null || D.forEach((e) => {
        const E = {
          title: e["x-displayName"] ?? e.name,
          href: `#${W(e)}`,
          description: e.description,
          type: "tag",
          tag: e.name,
          body: ""
        };
        o.value.push(E), e.operations && e.operations.forEach((t) => {
          const { parameterMap: n } = G({ operation: t }), S = P(t) || n.value;
          let p = null;
          typeof S != "boolean" && (p = S);
          const O = {
            type: "req",
            title: t.name ?? t.path,
            href: `#${L(t, e)}`,
            operationId: t.operationId,
            description: t.description ?? "",
            httpVerb: t.httpVerb,
            path: t.path,
            tag: e.name,
            operation: t
          };
          p && (O.body = p), o.value.push(O);
        });
      });
      const l = (x = a.value) == null ? void 0 : x.webhooks, m = [];
      l && Object.keys(l).forEach((e) => {
        Object.keys(
          l[e]
        ).forEach((t) => {
          var n;
          m.push({
            type: "webhook",
            title: `Webhook: ${(n = l[e][t]) == null ? void 0 : n.name}`,
            href: `#${T(e, t)}`,
            description: e,
            httpVerb: t,
            tag: e,
            body: ""
          });
        }), o.value = o.value.concat(m);
      });
      const i = q.value ? {} : Q(a.value), b = [];
      i && (Object.keys(i).forEach((e) => {
        b.push({
          type: "model",
          title: "Model",
          href: `#${j(e)}`,
          description: i[e].title ?? e,
          tag: e,
          body: ""
        });
      }), o.value = o.value.concat(b)), d.setCollection(o.value);
      const N = performance.now();
      console.log(`create-search-index: ${Math.round(N - f)} ms`);
    },
    { immediate: !0 }
  ), {
    resetSearch: F,
    fuseSearch: C,
    selectedSearchResult: h,
    searchResultsWithPlaceholderResults: H,
    searchText: s
  };
}
export {
  ae as useSearchIndex
};
