// @scalar/components/dist/components/ScalarFileUpload/ScalarFileUploadInputCompact.vue.js
import { defineComponent as m, createElementBlock as n, openBlock as l, mergeProps as d, unref as s, createElementVNode as i, renderSlot as r, createVNode as u, toDisplayString as a, createCommentVNode as f } from "vue";
import { ScalarIconUploadSimple as x } from "@scalar/icons";
import { useBindCx as b } from "@scalar/use-hooks/useBindCx";
const k = { class: "flex items-center gap-1 whitespace-nowrap text-c-2" }, y = {
  key: 0,
  class: "text-c-3 text-xs"
}, g = {}, h = /* @__PURE__ */ m({
  ...g,
  inheritAttrs: !1,
  __name: "ScalarFileUploadInputCompact",
  props: {
    multiple: { type: Boolean },
    extensions: {}
  },
  emits: ["click"],
  setup(e) {
    const { cx: c } = b();
    return (t, o) => (l(), n("button", d(
      { type: "button" },
      s(c)(
        "flex items-center justify-center gap-1 rounded flex-col px-4 py-3 hover:bg-b-2"
      ),
      {
        onClick: o[0] || (o[0] = (p) => t.$emit("click", p))
      }
    ), [
      i("div", k, [
        r(t.$slots, "default", {}, () => [
          u(s(x), { class: "size-3.5" }),
          i("span", null, " Upload " + a(e.multiple ? "files" : "file"), 1)
        ])
      ]),
      r(t.$slots, "sublabel", {}, () => [
        e.extensions ? (l(), n("div", y, a(e.extensions.join(", ")), 1)) : f("", !0)
      ])
    ], 16));
  }
});
export {
  h as default
};
