// @scalar/components/dist/components/ScalarFileUpload/ScalarFileUploadInput.vue.js
import { defineComponent as f, createElementBlock as n, openBlock as l, normalizeProps as c, guardReactiveProps as u, unref as r, createElementVNode as t, renderSlot as a, createVNode as x, createTextVNode as i, toDisplayString as p, withCtx as k, createCommentVNode as y } from "vue";
import { useBindCx as v } from "@scalar/use-hooks/useBindCx";
import B from "../ScalarButton/ScalarButton.vue.js";
const C = { class: "flex items-center gap-1.5 whitespace-nowrap" }, g = { class: "text-c-2" }, w = {
  key: 0,
  class: "text-c-3 text-xs"
}, N = {}, $ = /* @__PURE__ */ f({
  ...N,
  inheritAttrs: !1,
  __name: "ScalarFileUploadInput",
  props: {
    multiple: { type: Boolean },
    extensions: {}
  },
  emits: ["click"],
  setup(s) {
    const { cx: d } = v();
    return (o, e) => (l(), n("div", c(u(r(d)("flex items-center justify-center gap-2 flex-col px-6 py-4"))), [
      t("div", C, [
        t("span", g, [
          a(o.$slots, "default", {}, () => [
            i("Drop " + p(s.multiple ? "files" : "file") + " here to upload", 1)
          ]),
          e[1] || (e[1] = t("span", { class: "text-c-3" }, " or", -1))
        ]),
        x(r(B), {
          size: "xs",
          tabindex: "-1",
          variant: "outlined",
          onClick: e[0] || (e[0] = (m) => o.$emit("click", m))
        }, {
          default: k(() => [...e[2] || (e[2] = [
            i(" Browse files ", -1)
          ])]),
          _: 1
        })
      ]),
      a(o.$slots, "sublabel", {}, () => [
        s.extensions ? (l(), n("div", w, [
          e[3] || (e[3] = t("span", { class: "font-medium" }, "Supported file types:", -1)),
          i(" " + p(s.extensions.join(", ")), 1)
        ])) : y("", !0)
      ])
    ], 16));
  }
});
export {
  $ as default
};
