// @scalar/api-reference/dist/components/DarkModeToggle/DarkModeToggle.vue2.js
import { defineComponent as n, openBlock as o, createElementBlock as t, createElementVNode as s, createVNode as d, unref as c, pushScopeId as i, popScopeId as p } from "vue";
import { ScalarIcon as l } from "@scalar/components";
const k = (e) => (i("data-v-bfb18750"), e = e(), p(), e), m = { class: "darklight-reference" }, g = { key: 0 }, _ = { key: 1 }, h = /* @__PURE__ */ k(() => /* @__PURE__ */ s("a", {
  class: "darklight-reference-promo",
  href: "https://www.scalar.com",
  target: "_blank"
}, " Powered by scalar.com ", -1)), D = /* @__PURE__ */ n({
  __name: "DarkModeToggle",
  props: {
    isDarkMode: { type: Boolean }
  },
  emits: ["toggleDarkMode"],
  setup(e) {
    return (r, a) => (o(), t("div", m, [
      s("button", {
        class: "darklight",
        type: "button",
        onClick: a[0] || (a[0] = (f) => r.$emit("toggleDarkMode"))
      }, [
        d(c(l), {
          icon: "LightDarkModeToggle",
          size: "md"
        }),
        r.isDarkMode ? (o(), t("span", g, "Light Mode")) : (o(), t("span", _, "Dark Mode"))
      ]),
      h
    ]));
  }
});
export {
  D as default
};
