// @scalar/api-client/dist/v2/components/data-table/DataTableInputSelect.vue.js
import { defineComponent as T, computed as p, ref as V, watch as C, nextTick as E, createElementBlock as m, openBlock as o, createBlock as f, withDirectives as K, unref as a, withCtx as n, createVNode as s, createElementVNode as i, toDisplayString as x, withKeys as M, vModelText as R, createCommentVNode as _, Fragment as B, renderList as U, normalizeClass as $ } from "vue";
import { ScalarComboboxMultiselect as F, ScalarButton as D, ScalarIcon as v, ScalarDropdown as L, ScalarDropdownItem as j, ScalarDropdownDivider as W } from "@scalar/components";
const q = { class: "group-[.alert]:outline-orange group-[.error]:outline-red w-full pr-10 -outline-offset-1 has-[:focus-visible]:rounded-[4px] has-[:focus-visible]:outline" }, G = { class: "text-c-1 whitespace-nowrap" }, H = { class: "text-c-1 overflow-hidden text-ellipsis" }, J = { class: "overflow-hidden text-ellipsis" }, P = { class: "flex h-4 w-4 items-center justify-center" }, Z = /* @__PURE__ */ T({
  __name: "DataTableInputSelect",
  props: {
    modelValue: { type: [String, Number, Boolean, Array, Object] },
    value: {},
    default: { type: [String, Number, Boolean, Array, Object] },
    canAddCustomValue: { type: Boolean, default: !0 },
    type: {}
  },
  emits: ["update:modelValue"],
  setup(h, { emit: z }) {
    const u = h, d = z, y = p(() => u.value ?? []), c = V(!1), r = V(""), g = V(null);
    C(r, (e) => {
      d("update:modelValue", e);
    });
    const S = (e) => {
      d("update:modelValue", e), c.value = !1;
    }, A = () => {
      r.value.trim() && S(r.value);
    }, N = () => {
      r.value.trim() || d("update:modelValue", ""), c.value = !1;
    }, O = (e) => u.modelValue.toString() === e;
    C(c, (e) => {
      e && E(() => {
        g.value?.focus();
      });
    });
    const k = p(() => u.modelValue !== void 0 ? u.modelValue : u.default), b = p(
      () => y.value.map((e) => {
        const t = e.toString();
        return { id: t, label: t, value: t };
      })
    ), w = p(() => {
      const e = new Set(u.modelValue.toString().split(","));
      return b.value.filter((t) => e.has(t.id));
    }), I = (e) => {
      const t = e.map((l) => l.value);
      d("update:modelValue", t.join(","));
    };
    return (e, t) => (o(), m("div", q, [
      h.type === "array" ? (o(), f(a(F), {
        key: 0,
        modelValue: w.value,
        options: b.value,
        "onUpdate:modelValue": I
      }, {
        default: n(() => [
          s(a(D), {
            class: "custom-scroll h-full justify-start gap-1.5 px-2 py-1.5 pr-6 font-normal outline-none",
            fullWidth: "",
            variant: "ghost"
          }, {
            default: n(() => [
              i("span", G, x(w.value.length > 0 ? w.value.map((l) => l.label).join(", ") : "Select a value"), 1),
              s(a(v), {
                class: "min-w-4",
                icon: "ChevronDown",
                size: "md"
              })
            ]),
            _: 1
          })
        ]),
        _: 1
      }, 8, ["modelValue", "options"])) : c.value ? K((o(), m("input", {
        key: 1,
        ref_key: "inputRef",
        ref: g,
        "onUpdate:modelValue": t[0] || (t[0] = (l) => r.value = l),
        class: "text-c-1 w-full min-w-0 border-none px-2 py-1.5 outline-none",
        placeholder: "Value",
        type: "text",
        onBlur: N,
        onKeyup: M(A, ["enter"])
      }, null, 544)), [
        [R, r.value]
      ]) : (o(), f(a(L), {
        key: 2,
        resize: "",
        value: k.value
      }, {
        items: n(() => [
          (o(!0), m(B, null, U(y.value, (l) => (o(), f(a(j), {
            key: l,
            class: "group/item flex items-center gap-1.5 overflow-hidden text-ellipsis whitespace-nowrap",
            value: l,
            onClick: (Q) => S(l)
          }, {
            default: n(() => [
              i("div", {
                class: $([
                  "flex h-4 w-4 items-center justify-center rounded-full p-[3px]",
                  O(l) ? "bg-c-accent text-b-1" : "shadow-border text-transparent"
                ])
              }, [
                s(a(v), {
                  class: "size-2.5",
                  icon: "Checkmark",
                  thickness: "3"
                })
              ], 2),
              i("span", J, x(l), 1)
            ]),
            _: 2
          }, 1032, ["value", "onClick"]))), 128)),
          h.canAddCustomValue ? (o(), m(B, { key: 0 }, [
            y.value.length ? (o(), f(a(W), { key: 0 })) : _("", !0),
            s(a(j), {
              class: "flex items-center gap-1.5",
              onClick: t[1] || (t[1] = (l) => c.value = !0)
            }, {
              default: n(() => [
                i("div", P, [
                  s(a(v), {
                    icon: "Add",
                    size: "sm"
                  })
                ]),
                t[2] || (t[2] = i("span", null, "Add value", -1))
              ]),
              _: 1
            })
          ], 64)) : _("", !0)
        ]),
        default: n(() => [
          s(a(D), {
            class: "size-full justify-start gap-1.5 overflow-auto px-2 py-1.5 font-normal whitespace-nowrap outline-none",
            variant: "ghost"
          }, {
            default: n(() => [
              i("span", H, x(k.value ?? "Select a value"), 1),
              s(a(v), {
                icon: "ChevronDown",
                size: "md"
              })
            ]),
            _: 1
          })
        ]),
        _: 1
      }, 8, ["value"]))
    ]));
  }
});
export {
  Z as default
};
