// @scalar/api-client/dist/v2/features/command-palette/hooks/use-command-palette-state.js
import { useModal as k } from "@scalar/components";
import { ScalarIconArrowSquareIn as y, ScalarIconPackage as _, ScalarIconFolder as P, ScalarIconArrowUpRight as A, ScalarIconPuzzlePiece as R, ScalarIconBracketsCurly as b, ScalarIconCookie as q, ScalarIconSlidersHorizontal as L } from "@scalar/icons";
import d from "fuse.js";
import { ref as p, shallowRef as $, computed as o, toValue as a } from "vue";
import x from "../components/CommandPaletteDocument.vue.js";
import z from "../components/CommandPaletteExample.vue.js";
import F from "../components/CommandPaletteImport.vue.js";
import U from "../components/CommandPaletteImportCurl.vue.js";
import E from "../components/CommandPaletteRequest.vue.js";
import O from "../components/CommandPaletteTag.vue.js";
const W = (m = B, l = Q) => {
  const c = k(), n = p(""), s = $(null), i = p(null), f = o(() => c.open), v = o(
    () => new d(a(m), {
      keys: ["name"],
      threshold: 0.1
    })
  ), C = o(
    () => new d(a(l), {
      keys: ["name"],
      threshold: 0.1
    })
  ), S = o(() => {
    const e = n.value.toLowerCase().trim();
    return (e ? v.value.search(e).map((t) => t.item) : a(m)).filter((t) => !t.hidden);
  }), w = o(() => {
    const e = n.value.toLowerCase().trim();
    return e ? C.value.search(e).map((r) => r.item) : a(l);
  }), g = (e, ...r) => {
    e && (s.value = a(m).find((t) => t.id === e) ?? null, i.value = r[0] ?? null), c.show();
  }, h = () => {
    c.hide(), u();
  }, I = (e) => {
    n.value = e;
  }, u = () => {
    n.value = "", s.value = null, i.value = null;
  };
  return {
    isOpen: f,
    activeCommand: s,
    activeCommandProps: i,
    filterQuery: n,
    filteredCommands: o(() => [
      {
        label: "",
        commands: S.value
      },
      {
        label: "Pages",
        commands: w.value
      }
    ]),
    open: g,
    close: h,
    setFilterQuery: I,
    reset: u
  };
}, Q = [
  {
    type: "route",
    id: "environment",
    name: "Environment",
    icon: b,
    to: {
      name: "workspace.environment"
    }
  },
  {
    type: "route",
    id: "cookies",
    name: "Cookies",
    icon: q,
    to: {
      name: "workspace.cookies"
    }
  },
  {
    type: "route",
    id: "settings",
    name: "Settings",
    icon: L,
    to: {
      name: "workspace.settings"
    }
  }
], B = [
  {
    id: "import-from-openapi-swagger-postman-curl",
    name: "Import from OpenAPI/Swagger/Postman/cURL",
    component: F,
    icon: y
  },
  {
    id: "create-document",
    name: "Create Document",
    component: x,
    icon: _
  },
  {
    id: "add-tag",
    name: "Add Tag",
    component: O,
    icon: P
  },
  {
    id: "create-request",
    name: "Create Request",
    component: E,
    icon: A
  },
  {
    id: "add-example",
    name: "Add Example",
    component: z,
    icon: R
  },
  {
    id: "import-curl-command",
    name: "Import cURL Command",
    hidden: !0,
    component: U
  }
];
export {
  B as baseClientActions,
  Q as baseRoutes,
  W as useCommandPaletteState
};
