// @scalar/use-codemirror/dist/components/CodeMirror/CodeMirror.vue2.js
import { defineComponent as i, ref as s, toRef as o, openBlock as c, createElementBlock as m } from "vue";
import { useCodeMirror as u } from "../../hooks/useCodeMirror.js";
const h = /* @__PURE__ */ i({
  __name: "CodeMirror",
  props: {
    content: {},
    readOnly: { type: Boolean },
    language: {},
    withVariables: { type: Boolean },
    lineNumbers: { type: Boolean },
    withoutTheme: { type: Boolean },
    disableEnter: { type: Boolean, default: !1 }
  },
  emits: ["change"],
  setup(t, { emit: a }) {
    const e = t, l = a, r = s(null);
    return u({
      content: o(() => e.content),
      readOnly: o(() => e.readOnly),
      language: o(() => e.language),
      withVariables: o(() => e.withVariables),
      lineNumbers: o(() => e.lineNumbers),
      withoutTheme: o(() => e.withoutTheme),
      disableEnter: o(() => e.disableEnter),
      onChange: (n) => l("change", n || ""),
      codeMirrorRef: r,
      classes: ["codemirror"]
    }), (n, d) => (c(), m("div", {
      ref_key: "codeMirrorRef",
      ref: r,
      class: "codemirror-container"
    }, null, 512));
  }
});
export {
  h as default
};
