// @scalar/api-client/dist/v2/blocks/request-block/RequestBlock.vue2.js
import { defineComponent as j, computed as n, ref as z, useId as s, watch as X, createBlock as C, openBlock as d, withCtx as q, createElementVNode as K, withDirectives as c, createElementBlock as x, createVNode as m, unref as v, vShow as y, mergeProps as B, toHandlers as b, Fragment as J, renderList as W, createCommentVNode as E, resolveDynamicComponent as Y, toDisplayString as Z } from "vue";
import { ScalarErrorBoundary as _ } from "@scalar/components";
import { canMethodHaveBody as w } from "@scalar/helpers/http/can-method-have-body";
import { REGEX as ee } from "@scalar/helpers/regex/regex-helpers";
import { getResolvedRef as R } from "@scalar/workspace-store/helpers/get-resolved-ref";
import { unpackProxyObject as te } from "@scalar/workspace-store/helpers/unpack-proxy";
import ae from "../../../components/SectionFilter.vue.js";
import le from "../../../components/ViewLayout/ViewLayoutSection.vue.js";
import { filterGlobalCookie as ne } from "../operation-block/helpers/filter-global-cookies.js";
import { getExample as oe } from "../operation-block/helpers/get-example.js";
import { getResolvedUrl as re } from "../operation-block/helpers/get-resolved-url.js";
import ie from "./components/RequestBody.vue.js";
import ue from "./components/RequestCodeSnippet.vue.js";
import k from "./components/RequestParams.vue.js";
import { createParameterHandlers as S } from "./helpers/create-parameter-handlers.js";
import { getDefaultHeaders as se } from "./helpers/get-default-headers.js";
import { getParameterSchema as me } from "./helpers/get-parameter-schema.js";
import { groupBy as D } from "./helpers/group-by.js";
import { isParamDisabled as de } from "./helpers/is-param-disabled.js";
import ce from "../scalar-auth-selector-block/components/AuthSelector.vue.js";
const ve = { class: "group pointer-events-none flex flex-1 items-center gap-1 lg:pr-24" }, ye = ["for"], he = ["id", "placeholder", "value"], fe = {
  key: 2,
  class: "text-c-1 flex h-8 items-center"
}, xe = ["id", "role"], Ne = /* @__PURE__ */ j({
  __name: "RequestBlock",
  props: {
    selectedSecurity: {},
    authMeta: { default: () => ({ type: "document" }) },
    clientOptions: {},
    environment: {},
    eventBus: {},
    exampleKey: {},
    layout: {},
    method: {},
    operation: {},
    path: {},
    plugins: {},
    proxyUrl: {},
    securityRequirements: {},
    securitySchemes: {},
    selectedClient: {},
    selectedSecuritySchemes: {},
    server: {},
    globalCookies: {}
  },
  setup(e) {
    const r = n(() => ({
      method: e.method,
      path: e.path,
      exampleKey: e.exampleKey
    })), o = n(
      () => D(
        e.operation.parameters?.map((t) => R(t)) ?? [],
        "in",
        (t) => {
          const a = oe(t, e.exampleKey, void 0);
          return {
            name: t.name,
            value: a?.value ?? "",
            description: t.description,
            schema: me(t),
            isRequired: t.required,
            isDisabled: de(t, a),
            originalParameter: t
          };
        }
      )
    ), I = n(
      () => D(
        o.value.header?.map((t) => ({
          ...t,
          name: t.name.toLowerCase()
        })) ?? [],
        "name"
      )
    ), N = n(
      () => se({ method: e.method, operation: e.operation, exampleKey: e.exampleKey })
    ), V = n(() => {
      const t = e.operation["x-scalar-disable-parameters"]?.["default-headers"]?.[e.exampleKey] ?? {};
      return N.value.map((a) => {
        const l = I.value[a.name.toLowerCase()]?.[0];
        return {
          name: a.name,
          value: a.defaultValue,
          schema: void 0,
          isOverridden: l && !l?.isDisabled,
          isReadonly: !0,
          isDisabled: t[a.name.toLowerCase()] ?? !1
        };
      });
    }), A = n(() => [
      ...V.value,
      ...o.value.header ?? []
    ]), P = n(() => {
      const t = re({
        environment: e.environment,
        server: e.server,
        path: e.path,
        pathVariables: {}
      }), a = e.operation["x-scalar-disable-parameters"]?.["global-cookies"]?.[e.exampleKey] ?? {};
      return e.globalCookies?.filter(
        (l) => ne({
          cookie: l,
          url: t,
          // Do not filter global cookies for the default cookies section (it's already filtered in the buildRequestCookieHeader function)
          // This is because we still want to show them on the UI
          disabledGlobalCookies: {}
        })
      ).map((l) => ({
        name: l.name,
        value: l.value,
        globalRoute: l.location === "document" ? "document.cookies" : "workspace.cookies",
        isReadonly: !0,
        isDisabled: a[l.name.toLowerCase()] ?? !1
      })) ?? [];
    }), H = n(() => [
      ...P.value ?? [],
      ...o.value.cookie ?? []
    ]), i = z("All"), T = [
      "Auth",
      "Variables",
      "Cookies",
      "Headers",
      "Query",
      "Body"
    ], $ = {
      All: s(),
      Auth: s(),
      Variables: s(),
      Cookies: s(),
      Headers: s(),
      Query: s(),
      Body: s()
    }, O = n(() => {
      const t = /* @__PURE__ */ new Set(["All", ...T]);
      return o.value.path?.length || t.delete("Variables"), w(e.method) || t.delete("Body"), U.value && t.delete("Auth"), [...t];
    }), u = n(
      () => Object.fromEntries(
        O.value.map((t) => [t, $[t]])
      )
    ), U = n(
      () => e.layout === "modal" && !e.operation.security && !Object.keys(e.securitySchemes ?? {}).length
    ), L = n(() => e.operation.summary ? e.operation.summary : e.path.replace(ee.PROTOCOL, "") || "Request Name"), h = (t) => i.value === "All" || i.value === t;
    X(
      () => e.method,
      (t) => {
        i.value === "Body" && !w(t) && (i.value = "All");
      }
    );
    const F = (t) => {
      const a = t.target.value;
      e.eventBus.emit("operation:update:summary", {
        meta: r.value,
        payload: { summary: a }
      });
    }, f = n(() => ({
      path: S("path", e.eventBus, r.value, {
        context: o.value.path ?? []
      }),
      cookie: S("cookie", e.eventBus, r.value, {
        context: H.value ?? [],
        globalParameters: P.value.length
      }),
      header: S("header", e.eventBus, r.value, {
        context: A.value,
        defaultParameters: V.value.length
      }),
      query: S("query", e.eventBus, r.value, {
        context: o.value.query ?? []
      })
    })), G = (t) => e.eventBus.emit("operation:update:requestBody:contentType", {
      payload: { contentType: t.value },
      meta: r.value
    }), Q = ({
      payload: t,
      contentType: a
    }) => {
      const l = typeof t == "string" ? `update:requestBody:value-${a}` : void 0;
      e.eventBus.emit(
        "operation:update:requestBody:value",
        {
          payload: t,
          contentType: a,
          meta: r.value
        },
        {
          debounceKey: l
        }
      );
    }, M = ({
      payload: t,
      contentType: a
    }) => {
      const l = `update:requestBody:${a}-form-value`;
      e.eventBus.emit(
        "operation:update:requestBody:formValue",
        {
          payload: t.map((g) => te(g, { depth: 1 })),
          contentType: a,
          meta: r.value
        },
        {
          debounceKey: l
        }
      );
    }, p = s();
    return (t, a) => (d(), C(le, {
      "aria-label": `Request: ${e.operation.summary}`
    }, {
      title: q(() => [
        K("div", ve, [
          e.layout !== "modal" ? (d(), x("label", {
            key: 0,
            class: "pointer-events-auto absolute top-0 left-0 h-full w-full cursor-text opacity-0",
            for: v(p)
          }, null, 8, ye)) : E("", !0),
          e.layout !== "modal" ? (d(), x("input", {
            key: 1,
            id: v(p),
            class: "text-c-1 group-hover-input pointer-events-auto relative z-10 -ml-0.5 h-8 w-full rounded pl-1.25 has-[:focus-visible]:outline md:-ml-1.25",
            placeholder: L.value,
            value: e.operation.summary,
            onInput: F
          }, null, 40, he)) : (d(), x("span", fe, Z(e.operation.summary), 1))
        ]),
        m(ae, {
          modelValue: i.value,
          "onUpdate:modelValue": a[0] || (a[0] = (l) => i.value = l),
          filterIds: u.value,
          filters: O.value
        }, null, 8, ["modelValue", "filterIds", "filters"])
      ]),
      default: q(() => [
        K("div", {
          id: u.value.All,
          class: "request-section-content custom-scroll relative flex flex-1 flex-col",
          role: i.value === "All" ? "tabpanel" : "none"
        }, [
          c(m(v(ce), {
            id: u.value.Auth,
            environment: e.environment,
            eventBus: e.eventBus,
            meta: e.authMeta,
            proxyUrl: e.proxyUrl,
            securityRequirements: e.securityRequirements,
            securitySchemes: e.securitySchemes,
            selectedSecurity: e.selectedSecurity,
            selectedSecuritySchemes: e.selectedSecuritySchemes,
            server: e.server,
            title: "Authentication"
          }, null, 8, ["id", "environment", "eventBus", "meta", "proxyUrl", "securityRequirements", "securitySchemes", "selectedSecurity", "selectedSecuritySchemes", "server"]), [
            [y, h("Auth") && !U.value]
          ]),
          c(m(k, B({
            id: u.value.Variables,
            environment: e.environment,
            eventBus: e.eventBus,
            exampleKey: e.exampleKey,
            rows: o.value.path ?? [],
            showAddRowPlaceholder: !1,
            title: "Variables"
          }, b(f.value.path)), null, 16, ["id", "environment", "eventBus", "exampleKey", "rows"]), [
            [y, h("Variables") && o.value.path?.length]
          ]),
          c(m(k, B({
            id: u.value.Cookies,
            environment: e.environment,
            eventBus: e.eventBus,
            exampleKey: e.exampleKey,
            rows: H.value ?? [],
            showAddRowPlaceholder: !0,
            title: "Cookies"
          }, b(f.value.cookie)), null, 16, ["id", "environment", "eventBus", "exampleKey", "rows"]), [
            [y, h("Cookies")]
          ]),
          c(m(k, B({
            id: u.value.Headers,
            environment: e.environment,
            eventBus: e.eventBus,
            exampleKey: e.exampleKey,
            rows: A.value ?? [],
            title: "Headers"
          }, b(f.value.header)), null, 16, ["id", "environment", "eventBus", "exampleKey", "rows"]), [
            [y, h("Headers")]
          ]),
          c(m(k, B({
            id: u.value.Query,
            environment: e.environment,
            eventBus: e.eventBus,
            exampleKey: e.exampleKey,
            rows: o.value.query ?? [],
            title: "Query Parameters"
          }, b(f.value.query)), null, 16, ["id", "environment", "eventBus", "exampleKey", "rows"]), [
            [y, h("Query")]
          ]),
          c(m(ie, {
            id: u.value.Body,
            environment: e.environment,
            exampleKey: e.exampleKey,
            requestBody: v(R)(e.operation.requestBody),
            title: "Request Body",
            "onUpdate:contentType": G,
            "onUpdate:formValue": M,
            "onUpdate:value": Q
          }, null, 8, ["id", "environment", "exampleKey", "requestBody"]), [
            [y, h("Body") && v(w)(e.method)]
          ]),
          (d(!0), x(J, null, W(e.plugins, (l, g) => (d(), C(v(_), { key: g }, {
            default: q(() => [
              l?.components?.request ? (d(), C(Y(l.components.request), {
                key: 0,
                operation: e.operation,
                selectedExample: e.exampleKey
              }, null, 8, ["operation", "selectedExample"])) : E("", !0)
            ]),
            _: 2
          }, 1024))), 128)),
          a[1] || (a[1] = K("div", { class: "flex flex-grow" }, null, -1)),
          c(m(ue, {
            clientOptions: e.clientOptions,
            eventBus: e.eventBus,
            globalCookies: e.globalCookies,
            integration: "client",
            method: e.method,
            operation: e.operation,
            path: e.path,
            securitySchemes: e.selectedSecuritySchemes,
            selectedClient: e.selectedClient,
            selectedContentType: v(R)(e.operation.requestBody)?.["x-scalar-selected-content-type"]?.[e.exampleKey],
            selectedServer: e.server ?? void 0
          }, null, 8, ["clientOptions", "eventBus", "globalCookies", "method", "operation", "path", "securitySchemes", "selectedClient", "selectedContentType", "selectedServer"]), [
            [y, i.value === "All"]
          ])
        ], 8, xe)
      ]),
      _: 1
    }, 8, ["aria-label"]));
  }
});
export {
  Ne as default
};
