// @scalar/api-client/dist/v2/blocks/operation-code-sample/components/OperationCodeSample.vue2.js
import { defineComponent as $, computed as o, ref as b, watch as D, useId as I, createBlock as u, createCommentVNode as d, openBlock as s, unref as a, withCtx as n, createVNode as r, createSlots as N, createElementVNode as g, createElementBlock as x, renderSlot as y, createTextVNode as W, toDisplayString as R } from "vue";
import { ScalarCard as S, ScalarCardHeader as z, ScalarCombobox as U, ScalarButton as A, ScalarCardSection as T, ScalarCodeBlock as F, ScalarVirtualText as M, ScalarCardFooter as K } from "@scalar/components";
import { freezeElement as Z } from "@scalar/helpers/dom/freeze-element";
import { ScalarIconCaretDown as G } from "@scalar/icons";
import { getResolvedRef as J } from "@scalar/workspace-store/helpers/get-resolved-ref";
import { operationToHar as P } from "../helpers/operation-to-har/operation-to-har.js";
import V from "./HttpMethod.vue.js";
import { findClient as q } from "../helpers/find-client.js";
import { getClients as Q } from "../helpers/get-clients.js";
import { getCustomCodeSamples as X } from "../helpers/get-custom-code-samples.js";
import { getSecrets as Y } from "../helpers/get-secrets.js";
import { generateCodeSnippet as _ } from "../helpers/generate-code-snippet.js";
import ee from "./ExamplePicker.vue.js";
const te = ["innerHTML"], le = ["id"], ae = {
  key: 0,
  class: "request-card-footer-addon"
}, oe = { class: "request-header" }, ne = {}, be = /* @__PURE__ */ $({
  ...ne,
  __name: "OperationCodeSample",
  props: {
    integration: {},
    clientOptions: {},
    selectedClient: {},
    selectedServer: { default: null },
    selectedContentType: {},
    selectedExample: {},
    eventBus: {},
    securitySchemes: { default: () => [] },
    method: {},
    path: {},
    operation: {},
    fallback: { type: Boolean },
    generateLabel: { type: Function },
    isWebhook: { type: Boolean },
    globalCookies: {}
  },
  setup(e) {
    const i = o(() => {
      const t = J(e.operation.requestBody)?.content ?? {}, l = e.selectedContentType || Object.keys(t)[0];
      return l ? t[l]?.examples ?? {} : {};
    }), m = b(
      e.selectedExample ?? Object.keys(i.value)[0] ?? ""
    ), C = o(() => X(e.operation)), f = o(
      () => Q(C.value, e.clientOptions)
    ), c = b(
      q(f.value, e.selectedClient)
    );
    D(
      () => e.selectedClient,
      (t) => {
        const l = q(f.value, t);
        l && (c.value = l);
      }
    );
    const k = o(() => {
      if (!e.isWebhook) return null;
      try {
        return P({
          operation: e.operation,
          method: e.method,
          path: e.path,
          example: m.value
        });
      } catch (t) {
        return console.error("[webhookHar]", t), null;
      }
    }), h = o(() => e.isWebhook ? k.value?.postData?.text ?? "" : _({
      includeDefaultHeaders: e.integration === "client",
      clientId: c.value?.id,
      customCodeSamples: C.value,
      operation: e.operation,
      method: e.method,
      path: e.path,
      contentType: e.selectedContentType,
      server: e.selectedServer,
      securitySchemes: e.securitySchemes,
      example: m.value,
      globalCookies: e.globalCookies
    })), B = o(() => e.isWebhook ? w.value : c.value?.lang), w = o(() => {
      if (!k.value?.postData) return "json";
      const t = k.value.postData.mimeType;
      return t?.includes("json") ? "json" : t?.includes("xml") ? "xml" : t?.includes("yaml") || t?.includes("yml") ? "yaml" : t?.includes("text/plain") ? "text" : "json";
    }), H = o(() => Y(e.securitySchemes)), p = b(null), O = (t) => {
      if (p.value) {
        const l = Z(p.value.$el);
        setTimeout(() => {
          l();
        }, 300);
      }
      c.value = t, t && !t.id.startsWith("custom") && e.eventBus.emit("workspace:update:selected-client", t.id);
    }, j = 2e4, E = o(
      () => (h.value.length ?? 0) > j
    ), L = I();
    return (t, l) => h.value ? (s(), u(a(S), {
      key: 0,
      ref_key: "elem",
      ref: p,
      class: "request-card dark-mode"
    }, {
      default: n(() => [
        r(a(z), { class: "pr-2.5" }, N({
          default: n(() => [
            l[2] || (l[2] = g("span", { class: "sr-only" }, "Request Example for", -1)),
            r(V, {
              as: "span",
              class: "request-method",
              method: e.method
            }, null, 8, ["method"]),
            e.generateLabel ? (s(), x("span", {
              key: 0,
              innerHTML: e.generateLabel()
            }, null, 8, te)) : d("", !0),
            y(t.$slots, "header", {}, void 0, !0)
          ]),
          _: 2
        }, [
          !e.isWebhook && f.value.length ? {
            name: "actions",
            fn: n(() => [
              r(a(U), {
                class: "max-h-80",
                modelValue: c.value,
                options: f.value,
                placement: "bottom-end",
                teleport: "",
                "onUpdate:modelValue": l[0] || (l[0] = (v) => O(v))
              }, {
                default: n(() => [
                  r(a(A), {
                    class: "text-c-2 hover:text-c-1 flex h-full w-fit gap-1.5 px-0.5 py-0 text-base font-normal",
                    "data-testid": "client-picker",
                    variant: "ghost"
                  }, {
                    default: n(() => [
                      W(R(c.value?.title) + " ", 1),
                      r(a(G), {
                        class: "ui-open:rotate-180 mt-0.25 size-3 transition-transform duration-100",
                        weight: "bold"
                      })
                    ]),
                    _: 1
                  })
                ]),
                _: 1
              }, 8, ["modelValue", "options"])
            ]),
            key: "0"
          } : void 0
        ]), 1024),
        r(a(T), { class: "request-editor-section custom-scroll p-0" }, {
          default: n(() => [
            g("div", {
              id: `${a(L)}-example`,
              class: "code-snippet"
            }, [
              E.value ? (s(), u(a(M), {
                key: 1,
                containerClass: "custom-scroll scalar-code-block border rounded-b flex flex-1 max-h-screen",
                contentClass: "language-plaintext whitespace-pre font-code text-base",
                lineHeight: 20,
                text: h.value
              }, null, 8, ["text"])) : (s(), u(a(F), {
                key: 0,
                class: "bg-b-2 !min-h-full -outline-offset-2",
                content: h.value,
                hideCredentials: H.value,
                lang: B.value,
                lineNumbers: ""
              }, null, 8, ["content", "hideCredentials", "lang"]))
            ], 8, le)
          ]),
          _: 1
        }),
        Object.keys(i.value).length > 1 || t.$slots.footer ? (s(), u(a(K), {
          key: 0,
          class: "request-card-footer bg-b-3"
        }, {
          default: n(() => [
            Object.keys(i.value).length > 1 ? (s(), x("div", ae, [
              Object.keys(i.value).length ? (s(), u(ee, {
                key: 0,
                modelValue: m.value,
                "onUpdate:modelValue": l[1] || (l[1] = (v) => m.value = v),
                examples: i.value
              }, null, 8, ["modelValue", "examples"])) : d("", !0)
            ])) : d("", !0),
            y(t.$slots, "footer", {}, void 0, !0)
          ]),
          _: 3
        })) : d("", !0)
      ]),
      _: 3
    }, 512)) : e.fallback ? (s(), u(a(S), {
      key: 1,
      class: "request-card dark-mode"
    }, {
      default: n(() => [
        r(a(T), { class: "request-card-simple" }, {
          default: n(() => [
            g("div", oe, [
              r(V, {
                as: "span",
                class: "request-method",
                method: e.method
              }, null, 8, ["method"]),
              y(t.$slots, "header", {}, void 0, !0)
            ]),
            y(t.$slots, "footer", {}, void 0, !0)
          ]),
          _: 3
        })
      ]),
      _: 3
    })) : d("", !0);
  }
});
export {
  be as default
};
