// @scalar/components/dist/components/ScalarColorModeToggle/ScalarColorModeToggle.vue.js
import { defineComponent as u, computed as c, createBlock as r, openBlock as t, unref as o, isRef as s } from "vue";
import { useColorMode as f } from "@scalar/use-hooks/useColorMode";
import p from "./ScalarColorModeToggleButton.vue.js";
import _ from "./ScalarColorModeToggleIcon.vue.js";
const k = {}, S = /* @__PURE__ */ u({
  ...k,
  __name: "ScalarColorModeToggle",
  props: {
    variant: { default: "switch" }
  },
  setup(n) {
    const { isDarkMode: e, toggleColorMode: i, darkLightMode: d } = f(), l = c(
      () => e.value ? "Set light mode" : "Set dark mode"
    );
    return (g, a) => n.variant === "switch" ? (t(), r(p, {
      key: 0,
      modelValue: o(e),
      "onUpdate:modelValue": a[0] || (a[0] = (m) => s(e) ? e.value = m : null),
      "aria-label": l.value
    }, null, 8, ["modelValue", "aria-label"])) : (t(), r(_, {
      key: 1,
      "aria-label": l.value,
      mode: o(d),
      onClick: o(i)
    }, null, 8, ["aria-label", "mode", "onClick"]));
  }
});
export {
  S as default
};
