// @scalar/components/dist/components/ScalarMarkdown/ScalarMarkdown.vue.js
import { defineComponent as s, useTemplateRef as f, computed as u, createElementBlock as h, openBlock as d, mergeProps as g, unref as w } from "vue";
import { htmlFromMarkdown as x } from "@scalar/code-highlight";
import { useBindCx as T } from "@scalar/use-hooks/useBindCx";
const v = ["innerHTML"], B = /* @__PURE__ */ s({
  inheritAttrs: !1,
  __name: "ScalarMarkdown",
  props: {
    value: {},
    withImages: { type: Boolean, default: !1 },
    withAnchors: { type: Boolean, default: !1 },
    transform: { type: Function },
    transformType: {},
    clamp: {},
    anchorPrefix: {}
  },
  setup(e, { expose: n }) {
    const { cx: a } = T(), m = f("div");
    n({ el: m });
    const o = (t) => {
      if (!e.withAnchors)
        return e.transform?.(t) || t;
      const r = (t.children?.[0]?.value || "").toLowerCase().replace(/\s+/g, "-"), l = e.anchorPrefix ? `${e.anchorPrefix}/description/${r}` : r;
      return t.data = {
        hProperties: {
          id: l
        }
      }, e.transform ? e.transform(t) : t;
    }, i = u(() => x(e.value ?? "", {
      removeTags: e.withImages ? [] : ["img", "picture"],
      transform: e.withAnchors && e.transformType === "heading" ? o : e.transform,
      transformType: e.transformType
    }));
    return (t, c) => (d(), h("div", g({ ref: "div" }, w(a)("markdown", { "line-clamp-(--markdown-clamp)": !!e.clamp }), {
      style: { "--markdown-clamp": e.clamp },
      innerHTML: i.value
    }), null, 16, v));
  }
});
export {
  B as default
};
