// @scalar/components/dist/components/ScalarMarkdown/ScalarMarkdownSummary.vue.js
import { defineComponent as p, mergeModels as f, createPropsRestProxy as v, useId as h, useModel as y, useTemplateRef as w, ref as x, onMounted as k, createElementBlock as u, openBlock as c, normalizeProps as g, guardReactiveProps as B, unref as t, createVNode as M, createCommentVNode as P, mergeProps as V, normalizeClass as b, renderSlot as C, createTextVNode as S, toDisplayString as T } from "vue";
import { useBindCx as R } from "@scalar/use-hooks/useBindCx";
import { useResizeObserver as z } from "@vueuse/core";
import N from "./ScalarMarkdown.vue.js";
/* empty css                    */
const $ = ["aria-controls", "aria-expanded"], F = /* @__PURE__ */ p({
  inheritAttrs: !1,
  __name: "ScalarMarkdownSummary",
  props: /* @__PURE__ */ f({
    value: {},
    withImages: { type: Boolean },
    withAnchors: { type: Boolean },
    transform: { type: Function },
    transformType: {},
    clamp: { default: 1 },
    anchorPrefix: {},
    controlled: { type: Boolean }
  }, {
    modelValue: { type: Boolean, default: !1 },
    modelModifiers: {}
  }),
  emits: ["update:modelValue"],
  setup(a) {
    const m = v(a, ["clamp"]), l = h(), e = y(a, "modelValue"), r = w("scalar-markdown"), n = x(!1);
    z(() => r.value?.el, s);
    function s() {
      const o = r.value?.el;
      o && (n.value = o.scrollHeight > o.clientHeight || o.scrollWidth > o.clientWidth);
    }
    k(s);
    const { cx: d } = R();
    return (o, i) => (c(), u("div", g(B(
      t(d)(
        "group/summary flex gap-0.5",
        e.value ? "flex-col" : "flex-row items-baseline"
      )
    )), [
      M(N, V(m, {
        id: t(l),
        ref: "scalar-markdown",
        clamp: e.value ? void 0 : a.clamp,
        class: { "markdown-summary truncate": !e.value }
      }), null, 16, ["id", "clamp", "class"]),
      !a.controlled && (n.value || e.value) ? (c(), u("button", {
        key: 0,
        "aria-controls": t(l),
        "aria-expanded": e.value,
        class: b(["whitespace-nowrap font-medium hover:underline", { "self-end": e.value }]),
        type: "button",
        onClick: i[0] || (i[0] = (A) => e.value = !e.value)
      }, [
        C(o.$slots, "button", { open: e.value }, () => [
          S(T(e.value ? " Show Less" : "More"), 1)
        ])
      ], 10, $)) : P("", !0)
    ], 16));
  }
});
export {
  F as default
};
