// @scalar/api-client/dist/v2/blocks/operation-block/components/Header.vue2.js
import { defineComponent as c, createElementBlock as s, openBlock as l, createElementVNode as o, createVNode as u, unref as i, createBlock as v, createCommentVNode as m } from "vue";
import { ScalarIcon as a } from "@scalar/components";
import f from "../../scalar-address-bar-block/components/AddressBar.vue.js";
import y from "../../../../components/OpenApiClientButton.vue.js";
const g = { class: "lg:min-h-header t-app__top-container flex w-full flex-wrap items-center justify-center p-2 pt-2 lg:p-1 lg:pt-1" }, x = { class: "mb-2 flex w-1/2 flex-row items-center justify-end gap-1 lg:mb-0 lg:flex-1 lg:px-2.5" }, w = /* @__PURE__ */ c({
  __name: "Header",
  props: {
    path: {},
    method: {},
    layout: {},
    hideClientButton: { type: Boolean, default: !1 },
    integration: {},
    documentUrl: {},
    source: {},
    server: {},
    servers: {},
    history: {},
    eventBus: {},
    environment: {}
  },
  emits: ["execute", "update:servers", "select:history:item"],
  setup(e, { emit: d }) {
    const r = d;
    return (b, t) => (l(), s("div", g, [
      t[7] || (t[7] = o("div", { class: "mb-2 flex w-1/2 flex-row items-center gap-1 lg:mb-0 lg:flex-1 lg:px-1" }, [
        o("div", { class: "size-8" })
      ], -1)),
      u(i(f), {
        environment: e.environment,
        eventBus: e.eventBus,
        history: e.history,
        layout: e.layout,
        method: e.method,
        path: e.path,
        server: e.server,
        servers: e.servers,
        onExecute: t[0] || (t[0] = (n) => r("execute")),
        "onSelect:history:item": t[1] || (t[1] = (n) => r("select:history:item", n)),
        "onUpdate:servers": t[2] || (t[2] = (n) => r("update:servers"))
      }, null, 8, ["environment", "eventBus", "history", "layout", "method", "path", "server", "servers"]),
      o("div", x, [
        e.layout === "modal" && e.documentUrl && !e.hideClientButton ? (l(), v(i(y), {
          key: 0,
          buttonSource: "modal",
          class: "!w-fit lg:-mr-1",
          integration: e.integration ?? null,
          source: e.source ?? "api-reference",
          url: e.documentUrl
        }, null, 8, ["integration", "source", "url"])) : m("", !0),
        e.layout === "modal" && e.source !== "gitbook" ? (l(), s("button", {
          key: 1,
          class: "app-exit-button zoomed:static zoomed:p-1 fixed top-2 right-2 rounded-full p-2",
          type: "button",
          onClick: t[3] || (t[3] = (n) => e.eventBus.emit("ui:close:client-modal"))
        }, [
          u(i(a), {
            icon: "Close",
            size: "lg",
            thickness: "2"
          }),
          t[5] || (t[5] = o("span", { class: "sr-only" }, "Close Client", -1))
        ])) : m("", !0),
        e.layout === "modal" && e.source === "gitbook" ? (l(), s("button", {
          key: 2,
          class: "text-c-1 hover:bg-b-2 active:text-c-1 -mr-1.5 rounded p-2",
          type: "button",
          onClick: t[4] || (t[4] = (n) => e.eventBus.emit("ui:close:client-modal"))
        }, [
          u(i(a), {
            icon: "Close",
            size: "md",
            thickness: "1.75"
          }),
          t[6] || (t[6] = o("span", { class: "sr-only" }, "Close Client", -1))
        ])) : m("", !0)
      ])
    ]));
  }
});
export {
  w as default
};
