// @scalar/api-client/dist/libs/create-client.js
import { createWorkspaceStore as j } from "../store/workspace.js";
import { createWorkspace as q } from "@scalar/oas-utils/entities/workspace";
import { objectMerge as O } from "@scalar/oas-utils/helpers";
import { getNestedValue as P } from "@scalar/object-utils/nested";
import { createApp as W } from "vue";
const T = ({
  el: b,
  appComponent: U,
  configuration: t = {},
  isReadOnly: i = !1,
  persistData: y = !0,
  mountOnInitialize: A = !0,
  router: p
}) => {
  const a = j(p, y);
  a.workspaceMutators.rawAdd(
    q({
      uid: "default",
      name: "Workspace",
      isReadOnly: i,
      proxyUrl: t == null ? void 0 : t.proxyUrl
    })
  );
  const d = W(U);
  d.use(p), d.provide("workspace", a);
  const {
    activeCollection: S,
    activeWorkspace: o,
    importSpecFile: c,
    importSpecFromUrl: I,
    modalState: m,
    requests: u,
    securitySchemeMutators: x,
    securitySchemes: C,
    serverMutators: M,
    workspaceMutators: h
  } = a, k = (e = b) => {
    if (!e) {
      console.error(
        "[@scalar/api-client-modal] Could not create the API client.",
        "Invalid HTML element provided.",
        "Read more: https://github.com/scalar/scalar/tree/main/packages/api-client"
      );
      return;
    }
    d.mount(e);
  };
  return o.value && (A && k(), t != null && t.proxyUrl && h.edit(
    o.value.uid,
    "proxyUrl",
    t == null ? void 0 : t.proxyUrl
  ), t != null && t.themeId && h.edit(
    o.value.uid,
    "themeId",
    t == null ? void 0 : t.themeId
  )), {
    /** The vue app instance for the modal, be careful with this */
    app: d,
    /** Update the API client config */
    updateConfig(e, l = !0) {
      l ? Object.assign(t ?? {}, e) : O(t ?? {}, e), e.spec && c(e.spec);
    },
    /**
     * TODO this is just temporary for the modal, we'll put in a proper solution later
     * Here we update the currently selected serverUrl
     */
    updateServerUrl: (e) => {
      var l;
      return M.edit(
        ((l = S.value) == null ? void 0 : l.selectedServerUid) ?? "",
        "url",
        e
      );
    },
    /**
     * Update the security schemes
     * maps the references useAuthenticationStore to the client auth
     */
    updateAuth: (e) => {
      Object.values(C).forEach((s) => {
        const r = (v, w = "value") => v.length && !P(s, w).length && x.edit(s.uid, w, v);
        switch (s.type) {
          case "apiKey":
            r(e.apiKey.token);
            break;
          case "http":
            s.scheme === "bearer" ? r(e.http.bearer.token) : s.scheme === "basic" && (r(e.http.basic.username), r(e.http.basic.password, "secondValue"));
            break;
          case "oauth2":
            r(e.oAuth2.clientId, "clientId"), (s.flow.type === "implicit" || s.flow.type === "password") && (r(e.oAuth2.accessToken, "flow.token"), r(e.oAuth2.scopes, "flow.selectedScopes"), s.flow.type === "password" && (r(e.oAuth2.username, "flow.value"), r(e.oAuth2.password, "flow.secondValue")));
            break;
        }
      });
    },
    /** Update the spec file, this will re-parse it and clear your store */
    updateSpec: async (e) => {
      e != null && e.url ? await I(e.url, t == null ? void 0 : t.proxyUrl) : e != null && e.content ? await c(e == null ? void 0 : e.content) : console.error(
        "[@scalar/api-client-modal] Could not create the API client.",
        "Please provide an OpenAPI document: { spec: { url: '…' } }",
        "Read more: https://github.com/scalar/scalar/tree/main/packages/api-client"
      );
    },
    /** Route to a method + path */
    route: (e) => {
      const l = Object.values(u).find(
        ({ path: s, method: r }) => s === e.path && r.toUpperCase() === e.method.toUpperCase()
      );
      l && p.push(`/workspace/default/request/${l.uid}`);
    },
    /** Open the API client modal and optionally route to a request */
    open: (e) => {
      if (e) {
        const l = Object.values(u).find(
          ({ path: s, method: r }) => e ? (
            // The given operation
            s === e.path && r.toUpperCase() === e.method.toUpperCase()
          ) : (
            // Or the first request
            !0
          )
        );
        l && p.push(`/workspace/default/request/${l.uid}`);
      }
      m.open = !0;
    },
    /** Mount the references to a given element */
    mount: k,
    /** State for controlling the modal */
    modalState: m,
    /* The workspace store */
    store: a
  };
};
export {
  T as createApiClient
};
