// @scalar/themes/dist/index.js
import a from "./fonts.css.js";
import s from "./presets/alternate.css2.js";
import u from "./presets/bluePlanet.css2.js";
import i from "./presets/deepSpace.css2.js";
import o from "./presets/default.css2.js";
import f from "./presets/kepler.css2.js";
import c from "./presets/mars.css2.js";
import d from "./presets/moon.css2.js";
import h from "./presets/purple.css2.js";
import T from "./presets/saturn.css2.js";
import b from "./presets/solarized.css2.js";
import S from "./variables.css.js";
import { migrateThemeVariables as w } from "./utilities/legacy.js";
const $ = [
  "alternate",
  "default",
  "moon",
  "purple",
  "solarized",
  "bluePlanet",
  "deepSpace",
  "saturn",
  "kepler",
  "mars",
  "none"
], A = {
  default: "Default",
  alternate: "Alternate",
  moon: "Moon",
  purple: "Purple",
  solarized: "Solarized",
  bluePlanet: "Blue Planet",
  saturn: "Saturn",
  kepler: "Kepler-11e",
  mars: "Mars",
  deepSpace: "Deep Space",
  none: ""
}, m = {
  alternate: s,
  default: o,
  moon: d,
  purple: h,
  solarized: b,
  bluePlanet: u,
  deepSpace: i,
  saturn: T,
  kepler: f,
  mars: c
}, I = () => a, K = Object.keys(m), y = (e) => e === "none" ? "" : m[e || "default"] ?? o, L = (e, l) => {
  const { variables: n = !0, fonts: p = !0, layer: r = "scalar-theme" } = l ?? {}, t = [
    y(e),
    n ? S : "",
    p ? a : ""
  ].join("");
  return r ? `@layer ${r} {
${t}}` : t;
};
export {
  K as availableThemes,
  I as getDefaultFonts,
  y as getThemeById,
  L as getThemeStyles,
  w as migrateThemeVariables,
  m as presets,
  $ as themeIds,
  A as themeLabels
};
