// @scalar/api-client/dist/v2/blocks/response-block/components/Headers.vue.js
import { defineComponent as h, createBlock as m, openBlock as t, unref as u, withCtx as e, createElementBlock as l, createVNode as a, createTextVNode as n, Fragment as i, renderList as y, toDisplayString as c, renderSlot as g } from "vue";
import k from "../../../../components/DataTable/DataTable.vue.js";
import p from "../../../../components/DataTable/DataTableRow.vue.js";
import _ from "../../../../components/DataTable/DataTableText.vue.js";
import w from "../../../../components/HelpfulLink.vue.js";
import { httpHeaders as b } from "../../../../data/httpHeaders.js";
import C from "../../../components/layout/CollapsibleSection.vue.js";
import x from "../../../../components/DataTable/DataTableHeader.vue.js";
const $ = {
  key: 0,
  class: "max-h-[calc(100%-32px)] overflow-y-auto"
}, v = {
  key: 1,
  class: "text-c-3 bg-b-1 flex min-h-[64px] items-center justify-center border-t px-4 text-sm"
}, T = /* @__PURE__ */ h({
  __name: "Headers",
  props: {
    headers: {}
  },
  setup(s) {
    const d = (f) => b.find(
      (r) => r.name.toLowerCase() === f.toLowerCase()
    );
    return (f, r) => (t(), m(u(C), {
      class: "overflow-auto",
      defaultOpen: !1,
      itemCount: s.headers.length
    }, {
      title: e(() => [
        g(f.$slots, "title")
      ]),
      default: e(() => [
        s.headers.length ? (t(), l("div", $, [
          a(k, {
            columns: ["minmax(auto, min-content)", "minmax(50%, 1fr)"],
            scroll: ""
          }, {
            default: e(() => [
              a(p, { class: "sr-only !block" }, {
                default: e(() => [
                  a(u(x), null, {
                    default: e(() => [...r[0] || (r[0] = [
                      n("Header Key", -1)
                    ])]),
                    _: 1
                  }),
                  a(u(x), null, {
                    default: e(() => [...r[1] || (r[1] = [
                      n("Header Value", -1)
                    ])]),
                    _: 1
                  })
                ]),
                _: 1
              }),
              (t(!0), l(i, null, y(s.headers, (o) => (t(), m(p, {
                key: o.name,
                class: "group/row text-c-1"
              }, {
                default: e(() => [
                  a(_, { class: "bg-b-1 sticky left-0 z-1 max-w-full group-first/row:border-t-0" }, {
                    default: e(() => [
                      typeof d(o.name)?.url == "string" ? (t(), m(w, {
                        key: 0,
                        class: "decoration-c-3",
                        href: d(o.name).url
                      }, {
                        default: e(() => [
                          n(c(o.name), 1)
                        ]),
                        _: 2
                      }, 1032, ["href"])) : (t(), l(i, { key: 1 }, [
                        n(c(o.name), 1)
                      ], 64))
                    ]),
                    _: 2
                  }, 1024),
                  a(_, {
                    class: "z-0 group-first/row:border-t-0",
                    text: o.value
                  }, null, 8, ["text"])
                ]),
                _: 2
              }, 1024))), 128))
            ]),
            _: 1
          })
        ])) : (t(), l("div", v, " No headers "))
      ]),
      _: 3
    }, 8, ["itemCount"]));
  }
});
export {
  T as default
};
