// @scalar/api-client/dist/v2/blocks/response-block/components/ResponseCookies.vue.js
import { defineComponent as c, createBlock as r, openBlock as e, unref as f, withCtx as t, createElementBlock as s, Fragment as i, renderList as u, createVNode as a, createTextVNode as k } from "vue";
import x from "../../../../components/DataTable/DataTable.vue.js";
import d from "../../../../components/DataTable/DataTableRow.vue.js";
import m from "../../../../components/DataTable/DataTableText.vue.js";
import p from "../../../components/layout/CollapsibleSection.vue.js";
const _ = {
  key: 1,
  class: "text-c-3 bg-b-1 flex min-h-[64px] items-center justify-center border-t px-4 text-sm"
}, N = /* @__PURE__ */ c({
  __name: "ResponseCookies",
  props: {
    cookies: {}
  },
  setup(o) {
    return (C, l) => (e(), r(f(p), {
      defaultOpen: !1,
      itemCount: o.cookies.length
    }, {
      title: t(() => [...l[0] || (l[0] = [
        k("Cookies", -1)
      ])]),
      default: t(() => [
        (e(), s(i, { key: 0 }, [
          o.cookies.length ? (e(), r(x, {
            key: 0,
            class: "flex-1",
            columns: ["", ""]
          }, {
            default: t(() => [
              (e(!0), s(i, null, u(o.cookies, (n) => (e(), r(d, {
                key: n.name
              }, {
                default: t(() => [
                  a(m, {
                    text: n.name
                  }, null, 8, ["text"]),
                  a(m, {
                    text: n.value
                  }, null, 8, ["text"])
                ]),
                _: 2
              }, 1024))), 128))
            ]),
            _: 1
          })) : (e(), s("div", _, " No cookies "))
        ], 64))
      ]),
      _: 1
    }, 8, ["itemCount"]));
  }
});
export {
  N as default
};
