// @scalar/api-client/dist/v2/blocks/response-block/components/ResponseEmpty.vue2.js
import { defineComponent as p, createElementBlock as a, openBlock as s, createElementVNode as o, createCommentVNode as r, normalizeClass as f, createVNode as l, createTextVNode as n, toDisplayString as y, unref as i } from "vue";
import { ScalarHotkey as d } from "@scalar/components";
import x from "../../../../assets/computer.ascii.js";
import k from "../../../../components/EmptyState.vue.js";
import b from "../../../../components/ScalarAsciiArt.vue.js";
const v = { class: "flex-center relative flex flex-1 flex-col gap-6 p-2 capitalize" }, g = {
  key: 0,
  class: "scalar-version-number"
}, R = { class: "text-c-3 right-4 mt-auto flex w-full flex-col items-end gap-2 text-sm" }, P = /* @__PURE__ */ p({
  __name: "ResponseEmpty",
  props: {
    layout: {},
    totalPerformedRequests: {},
    appVersion: {}
  },
  emits: ["addRequest", "sendRequest", "openCommandPalette"],
  setup(t, { emit: c }) {
    const m = c;
    return (q, e) => (s(), a("div", v, [
      o("div", {
        class: f(["flex h-[calc(100%_-_50px)] flex-col items-center justify-center", {
          "hidden opacity-0": t.totalPerformedRequests < 1 && t.layout !== "modal"
        }])
      }, [
        t.layout !== "modal" ? (s(), a("div", g, [
          n(" Scalar App V" + y(t.appVersion) + " Beta ", 1),
          e[3] || (e[3] = o("div", { class: "mt-2" }, [
            o("a", {
              href: "https://github.com/scalar/scalar/issues/2669",
              target: "_blank"
            }, " Roadmap ")
          ], -1))
        ])) : r("", !0),
        e[4] || (e[4] = o("a", {
          class: "gitbook-show scalar-version-number",
          href: "https://www.scalar.com",
          target: "_blank"
        }, " Powered By Scalar.com ", -1)),
        l(b, {
          art: i(x),
          class: "text-c-3"
        }, null, 8, ["art"])
      ], 2),
      t.layout !== "modal" ? (s(), a("div", {
        key: 0,
        class: f(["hidden h-[calc(100%_-_50px)] items-center justify-center pb-5", {
          "!flex opacity-100": t.totalPerformedRequests == 0
        }])
      }, [
        l(k)
      ], 2)) : r("", !0),
      o("div", R, [
        t.layout !== "modal" ? (s(), a("button", {
          key: 0,
          class: "flex items-center gap-1.5",
          type: "button",
          onClick: e[0] || (e[0] = (u) => m("openCommandPalette"))
        }, [
          e[5] || (e[5] = n(" Get Started ", -1)),
          l(i(d), {
            hotkey: "p",
            modifier: ["default", "Shift"]
          })
        ])) : r("", !0),
        t.layout === "desktop" ? (s(), a("button", {
          key: 1,
          class: "flex items-center gap-1.5",
          type: "button",
          onClick: e[1] || (e[1] = (u) => m("addRequest"))
        }, [
          e[6] || (e[6] = n(" New Request ", -1)),
          l(i(d), { hotkey: "N" })
        ])) : r("", !0),
        o("button", {
          class: "flex items-center gap-1.5",
          type: "button",
          onClick: e[2] || (e[2] = (u) => m("sendRequest"))
        }, [
          e[7] || (e[7] = n(" Send Request ", -1)),
          l(i(d), { hotkey: "↵" })
        ])
      ])
    ]));
  }
});
export {
  P as default
};
