// @scalar/api-client/dist/v2/blocks/response-block/components/ResponseMetaInformation.vue.js
import { defineComponent as C, ref as v, onMounted as g, onBeforeUnmount as S, computed as x, createElementBlock as a, openBlock as o, toDisplayString as n, unref as c, Fragment as p, createElementVNode as l, createCommentVNode as k, createTextVNode as i, createBlock as w, withCtx as I, normalizeStyle as y } from "vue";
import { httpStatusCodes as b } from "@scalar/oas-utils/helpers";
import h from "pretty-ms";
import q from "../../../../components/HelpfulLink.vue.js";
import { getContentLength as B } from "../helpers/get-content-length.js";
const M = { class: "text-c-1 flex gap-1.5" }, N = { key: 0 }, V = { key: 0 }, z = { key: 1 }, T = /* @__PURE__ */ C({
  __name: "ResponseMetaInformation",
  props: {
    response: {},
    eventBus: {}
  },
  setup(e) {
    const r = v(), u = v(0), m = () => {
      clearInterval(r.value), r.value = void 0, u.value = 0;
    }, f = () => {
      r.value = setInterval(() => u.value += 1e3, 1e3);
    };
    g(() => {
      e.eventBus.on("hooks:on:request:sent", f), e.eventBus.on("hooks:on:request:complete", m);
    }), S(() => {
      e.eventBus.off("hooks:on:request:sent", f), e.eventBus.off("hooks:on:request:complete", m), m();
    });
    const t = x(() => {
      const d = e.response?.status;
      if (d)
        return b[d] ?? void 0;
    });
    return (d, s) => (o(), a("div", M, [
      r.value && u.value ? (o(), a("span", N, n(c(h)(u.value)), 1)) : (o(), a(p, { key: 1 }, [
        l("span", null, [
          s[0] || (s[0] = l("span", { class: "sr-only" }, "Response Information, Duration:", -1)),
          i(" " + n(c(h)(e.response.duration)), 1)
        ]),
        c(B)(e.response) ? (o(), a("span", V, [
          s[1] || (s[1] = l("span", { class: "sr-only" }, ", Size:", -1)),
          i(" " + n(c(B)(e.response)), 1)
        ])) : k("", !0),
        t.value ? (o(), a(p, { key: 1 }, [
          s[2] || (s[2] = l("span", { class: "sr-only" }, ", Status:", -1)),
          t.value.url ? (o(), w(q, {
            key: 0,
            class: "flex items-center gap-1.5",
            href: t.value.url
          }, {
            default: I(() => [
              i(n(e.response.status) + " " + n(t.value.name) + " ", 1),
              l("span", {
                class: "block h-1.5 w-1.5 rounded-full",
                style: y({ backgroundColor: t.value.color })
              }, null, 4)
            ]),
            _: 1
          }, 8, ["href"])) : (o(), a("span", z, [
            i(n(e.response.status) + " " + n(t.value.name) + " ", 1),
            l("span", {
              class: "block h-1.5 w-1.5 rounded-full",
              style: y({ backgroundColor: t.value.color })
            }, null, 4)
          ]))
        ], 64)) : k("", !0)
      ], 64))
    ]));
  }
});
export {
  T as default
};
