// @scalar/api-client/dist/components/ContextBar.vue2.js
import { defineComponent as i, openBlock as n, createElementBlock as o, createElementVNode as s, Fragment as c, renderList as r, normalizeClass as a, toDisplayString as l } from "vue";
const p = { class: "fade-request-section-content sticky top-0 z-10 pointer-events-none" }, d = { class: "request-section-content request-section-content-filter bg-b-1 border-1/2 text-c-3 bg-b-1 mb-2.5 hidden xl:flex justify-center rounded p-[2.25px] text-xs gap-0.5 pointer-events-auto" }, u = ["isActive", "onClick"], _ = /* @__PURE__ */ i({
  __name: "ContextBar",
  props: {
    sections: {},
    activeSection: {}
  },
  emits: ["setActiveSection"],
  setup(m) {
    return (t, v) => (n(), o("div", p, [
      s("div", d, [
        (n(!0), o(c, null, r(t.sections, (e) => (n(), o("button", {
          key: e,
          class: a(["hover:bg-b-2 w-full rounded px-1 py-1 text-center font-medium", [
            t.activeSection === e ? "bg-b-2 text-c-1  pointer-events-none" : ""
          ]]),
          isActive: t.activeSection === e,
          type: "button",
          onClick: (b) => t.$emit("setActiveSection", e)
        }, l(e), 11, u))), 128))
      ])
    ]));
  }
});
export {
  _ as default
};
