// @scalar/api-client/dist/components/ScalarAsciiArt.vue2.js
import { defineComponent as d, computed as g, openBlock as a, createElementBlock as i, normalizeClass as h, Fragment as f, renderList as x, normalizeStyle as c, createElementVNode as y, toDisplayString as v } from "vue";
const r = 500, l = 100, k = /* @__PURE__ */ d({
  __name: "ScalarAsciiArt",
  props: {
    art: {},
    animate: { type: Boolean }
  },
  setup(m) {
    const p = m, e = g(() => p.art.split(`
`)), u = (t, o) => ({
      animationDuration: `${t * l}ms, ${r}ms`,
      animationTimingFunction: `steps(${t}), step-end`,
      animationDelay: `${o * l}ms, 0ms`,
      animationIterationCount: `1, ${(e.value.length + e.value[e.value.length - 1].length + 5) * l / r}`
    });
    return (t, o) => (a(), i("div", {
      class: h(["ascii-art font-code text-c-3 flex flex-col items-start text-[6px] leading-[7px]", { "ascii-art-animate": t.animate }])
    }, [
      (a(!0), i(f, null, x(e.value, (n, s) => (a(), i("span", {
        key: s,
        class: "inline-block",
        style: c({ width: `calc(${n.length + 1}ch)` })
      }, [
        y("span", {
          class: "inline-block whitespace-pre overflow-hidden",
          style: c(u(n.length, s))
        }, v(n), 5)
      ], 4))), 128))
    ], 2));
  }
});
export {
  k as default
};
