// @scalar/api-client/dist/components/ScalarHotkey.vue.js
import { defineComponent as l, computed as t, openBlock as d, createElementBlock as y, mergeProps as k, toDisplayString as u } from "vue";
import { isMacOS as r } from "@scalar/use-tooltip";
import { useMagicKeys as f, whenever as v } from "@vueuse/core";
import { keyMap as h } from "../hooks/useKeyMap.js";
const P = /* @__PURE__ */ l({
  __name: "ScalarHotkey",
  props: {
    hotkey: {}
  },
  emits: ["hotkeyPressed"],
  setup(n, { emit: a }) {
    const o = n, i = a, p = t(() => r() ? "⌘" : "⌃"), s = t(() => h().get(o.hotkey)), c = t(() => `${p.value}${o.hotkey}`), m = f({
      passive: !1,
      onEventFired(e) {
        !r() && e.ctrlKey && e.key === s.value && (e.preventDefault(), e.stopPropagation());
      }
    });
    return v(m[`${r() ? "meta" : "control"}_${o.hotkey}`], () => {
      i("hotkeyPressed", s.value || "");
    }), (e, b) => (d(), y("div", k(e.$attrs, { class: "bg-c-btn border-b-3 inline-block overflow-hidden rounded border-1/2 text-xxs rounded-b px-1 font-medium uppercase" }), u(c.value), 17));
  }
});
export {
  P as default
};
