// @scalar/api-client/dist/components/SectionFilter.vue2.js
import { defineComponent as x, mergeModels as h, useModel as g, ref as p, createElementBlock as u, openBlock as r, withKeys as f, createElementVNode as i, Fragment as b, renderList as k, createBlock as y, withCtx as V, createTextVNode as w, toDisplayString as m, createVNode as S, unref as q, nextTick as B } from "vue";
import { ScalarIcon as C } from "@scalar/components";
import I from "./SectionFilterButton.vue.js";
const $ = { class: "request-section-content request-section-content-filter fade-request-section-content text-c-3 pointer-events-auto relative hidden w-full justify-end gap-[1.5px] rounded py-1.75 text-xs xl:flex" }, F = { class: "filter-button context-bar-group-hover:text-c-1 absolute -right-[30px] flex items-center" }, M = { class: "context-bar-group-hover:hidden mr-1.5" }, L = /* @__PURE__ */ x({
  __name: "SectionFilter",
  props: /* @__PURE__ */ h({
    filters: { default: () => [] },
    filterIds: {}
  }, {
    modelValue: {},
    modelModifiers: {}
  }),
  emits: ["update:modelValue"],
  setup(t) {
    const o = g(t, "modelValue"), l = p(), c = (a) => {
      const n = a === "prev" ? -1 : 1, e = o.value ? t.filters.indexOf(o.value) : 0, s = t.filters.length, v = (e + n + s) % s;
      o.value = t.filters[v], B(() => {
        if (l.value) {
          const d = l.value.querySelector(
            'button[aria-selected="true"]'
          );
          d && d.focus();
        }
      });
    };
    return (a, n) => (r(), u("div", {
      ref_key: "tablist",
      ref: l,
      class: "filter-hover context-bar-group ml-auto hidden lg:flex",
      role: "tablist",
      onKeydown: [
        n[0] || (n[0] = f((e) => c("prev"), ["left"])),
        n[1] || (n[1] = f((e) => c("next"), ["right"]))
      ]
    }, [
      i("div", $, [
        (r(!0), u(b, null, k(t.filters, (e) => (r(), y(I, {
          key: e,
          class: "filter-hover-item",
          controls: t.filterIds?.[e],
          role: "tab",
          selected: o.value === e,
          onClick: (s) => o.value = e
        }, {
          default: V(() => [
            w(m(e), 1)
          ]),
          _: 2
        }, 1032, ["controls", "selected", "onClick"]))), 128)),
        i("div", F, [
          i("span", M, m(o.value), 1),
          S(q(C), {
            icon: "FilterList",
            size: "md",
            thickness: "2"
          })
        ])
      ])
    ], 544));
  }
});
export {
  L as default
};
