// @scalar/api-client/dist/components/SubpageHeader.vue2.js
import { defineComponent as a, resolveComponent as d, openBlock as l, createElementBlock as i, createElementVNode as r, createVNode as t, unref as o, withCtx as p, renderSlot as u, pushScopeId as m, popScopeId as f } from "vue";
import { ScalarIcon as _ } from "@scalar/components";
import { useRoute as h } from "vue-router";
const b = (e) => (m("data-v-8a69ee40"), e = e(), f(), e), v = { class: "flex flex-1 flex-col rounded pt-0 h-full bg-b-1 relative border-1/2 rounded mr-1.5 mb-1.5 overflow-hidden" }, x = { class: "lg:min-h-header items-center w-full p-1 t-app__top-container flex items-center border-b-1/2" }, k = /* @__PURE__ */ b(() => /* @__PURE__ */ r("span", { class: "leading-none" }, "Back To Requests", -1)), I = /* @__PURE__ */ a({
  __name: "SubpageHeader",
  setup(e) {
    const n = h();
    return (s, g) => {
      const c = d("router-link");
      return l(), i("div", v, [
        r("div", x, [
          t(c, {
            class: "text-c-2 text-sm font-medium ml-1 flex items-center p-1.5 hover:bg-b-2 rounded cursor-pointer gap-0.5 active:text-c-1 no-underline dark:hover:bg-b-2",
            to: `/workspace/${o(n).params.workspace}/request/default`
          }, {
            default: p(() => [
              t(o(_), {
                icon: "ChevronLeft",
                size: "md",
                thickness: "1.75"
              }),
              k
            ]),
            _: 1
          }, 8, ["to"])
        ]),
        u(s.$slots, "default", {}, void 0, !0)
      ]);
    };
  }
});
export {
  I as default
};
