// @scalar/api-client/dist/views/Request/RequestSection/RequestTable.vue2.js
import { defineComponent as z, createBlock as c, openBlock as d, withCtx as t, createVNode as n, createElementBlock as f, createTextVNode as v, toDisplayString as p, Fragment as g, renderList as q, normalizeClass as k, unref as u, createCommentVNode as y, createElementVNode as m } from "vue";
import { ScalarTooltip as D, ScalarIcon as B, ScalarButton as R } from "@scalar/components";
import { ScalarIconTrash as E } from "@scalar/icons";
import { RouterLink as G } from "vue-router";
import C from "../../../components/CodeInput/CodeInput.vue.js";
import N from "../../../components/DataTable/DataTable.vue.js";
import V from "../../../components/DataTable/DataTableCell.vue.js";
import P from "../../../components/DataTable/DataTableCheckbox.vue.js";
import h from "../../../components/DataTable/DataTableHeader.vue.js";
import $ from "../../../components/DataTable/DataTableRow.vue.js";
import { parameterIsInvalid as j, hasItemProperties as w } from "../libs/request.js";
import K from "./RequestTableTooltip.vue.js";
const O = { class: "text-c-2 filemask flex w-full max-w-[100%] items-end justify-end overflow-hidden p-1" }, A = ["onClick"], L = {
  key: 1,
  class: "p-0.5"
}, oe = /* @__PURE__ */ z({
  __name: "RequestTable",
  props: {
    items: {},
    hasCheckboxDisabled: { type: Boolean, default: !1 },
    showUploadButton: { type: Boolean, default: !1 },
    isGlobal: { type: Boolean, default: !1 },
    isReadOnly: { type: Boolean, default: !1 },
    environment: {},
    envVariables: {},
    workspace: {},
    invalidParams: {},
    label: {}
  },
  emits: ["updateRow", "toggleRow", "addRow", "deleteRow", "inputFocus", "inputBlur", "uploadFile", "removeFile"],
  setup(l, { emit: F }) {
    const b = l, s = F, U = ["", "", "36px"], x = (r, o, e) => {
      s("updateRow", r, o, e);
    }, S = (r) => {
      s("uploadFile", r);
    }, T = (r) => Array.isArray(r.default) && r.default.length === 1 ? r.default[0] : r.default, I = (r) => !!(r.key || r.value);
    return (r, o) => (d(), c(N, {
      class: "group/table flex-1",
      columns: U
    }, {
      default: t(() => [
        n($, { class: "sr-only !block" }, {
          default: t(() => [
            n(h, null, {
              default: t(() => [
                v(p(l.label) + " Enabled", 1)
              ]),
              _: 1
            }),
            n(h, null, {
              default: t(() => [
                v(p(l.label) + " Key", 1)
              ]),
              _: 1
            }),
            n(h, null, {
              default: t(() => [
                v(p(l.label) + " Value", 1)
              ]),
              _: 1
            })
          ]),
          _: 1
        }),
        (d(!0), f(g, null, q(l.items, (e, i) => (d(), c($, {
          id: e.key,
          key: i,
          class: k({
            alert: u(j)(e).value,
            error: l.invalidParams && l.invalidParams.has(e.key)
          })
        }, {
          default: t(() => [
            l.isGlobal ? (d(), c(u(G), {
              key: 0,
              class: "text-c-2 flex items-center justify-center border-t !border-r",
              to: e.route ?? {}
            }, {
              default: t(() => [
                o[4] || (o[4] = m("span", { class: "sr-only" }, "Global", -1)),
                n(u(D), {
                  content: "Global cookies are shared across the whole workspace.",
                  placement: "top"
                }, {
                  default: t(() => [
                    n(u(B), {
                      tabindex: "0",
                      class: "text-c-1",
                      icon: "Globe",
                      size: "xs"
                    })
                  ]),
                  _: 1
                })
              ]),
              _: 1
            }, 8, ["to"])) : (d(), c(P, {
              key: 1,
              class: "!border-r",
              disabled: b.hasCheckboxDisabled,
              modelValue: e.enabled,
              "onUpdate:modelValue": (a) => s("toggleRow", i, a)
            }, null, 8, ["disabled", "modelValue", "onUpdate:modelValue"])),
            n(V, null, {
              default: t(() => [
                n(C, {
                  "aria-label": `${l.label} Key`,
                  disableCloseBrackets: "",
                  disabled: b.isReadOnly,
                  disableEnter: "",
                  disableTabIndent: "",
                  lineWrapping: "",
                  envVariables: l.envVariables,
                  environment: l.environment,
                  modelValue: e.key,
                  placeholder: "Key",
                  required: !!e.required,
                  workspace: l.workspace,
                  onBlur: o[0] || (o[0] = (a) => s("inputBlur")),
                  onFocus: o[1] || (o[1] = (a) => s("inputFocus")),
                  onSelectVariable: (a) => x(i, "key", a),
                  "onUpdate:modelValue": (a) => s("updateRow", i, "key", a)
                }, null, 8, ["aria-label", "disabled", "envVariables", "environment", "modelValue", "required", "workspace", "onSelectVariable", "onUpdate:modelValue"])
              ]),
              _: 2
            }, 1024),
            n(V, null, {
              default: t(() => [
                n(C, {
                  "aria-label": `${l.label} Value`,
                  class: k(
                    u(w)(e) ? "pr-6 group-hover:pr-10 group-has-[.cm-focused]:pr-10" : "group-hover:pr-6 group-has-[.cm-focused]:pr-6"
                  ),
                  default: e.default,
                  disableCloseBrackets: "",
                  disabled: b.isReadOnly,
                  disableEnter: "",
                  disableTabIndent: "",
                  lineWrapping: "",
                  enum: e.enum ?? [],
                  envVariables: l.envVariables,
                  environment: l.environment,
                  examples: e.examples ?? [],
                  max: e.maximum,
                  min: e.minimum,
                  modelValue: e.value,
                  nullable: !!e.nullable,
                  placeholder: "Value",
                  type: e.type,
                  workspace: l.workspace,
                  onBlur: o[2] || (o[2] = (a) => s("inputBlur")),
                  onFocus: o[3] || (o[3] = (a) => s("inputFocus")),
                  onSelectVariable: (a) => x(i, "value", a),
                  "onUpdate:modelValue": (a) => s("updateRow", i, "value", a)
                }, {
                  icon: t(() => [
                    I(e) && !e.required ? (d(), c(u(R), {
                      key: 0,
                      class: k([{
                        "-mr-0.5": u(w)(e)
                      }, "text-c-2 hover:text-c-1 hover:bg-b-2 z-context hidden h-fit rounded p-1 group-hover:flex group-has-[.cm-focused]:flex"]),
                      size: "sm",
                      variant: "ghost",
                      onClick: (a) => s("deleteRow", i)
                    }, {
                      default: t(() => [
                        n(u(E), { class: "size-3.5" })
                      ]),
                      _: 1
                    }, 8, ["class", "onClick"])) : y("", !0),
                    u(w)(e) ? (d(), c(K, {
                      key: 1,
                      item: { ...e, default: T(e) }
                    }, null, 8, ["item"])) : y("", !0)
                  ]),
                  _: 2
                }, 1032, ["aria-label", "class", "default", "disabled", "enum", "envVariables", "environment", "examples", "max", "min", "modelValue", "nullable", "type", "workspace", "onSelectVariable", "onUpdate:modelValue"])
              ]),
              _: 2
            }, 1024),
            l.showUploadButton ? (d(), c(V, {
              key: 2,
              class: "group/upload flex items-center justify-center whitespace-nowrap"
            }, {
              default: t(() => [
                e.file ? (d(), f(g, { key: 0 }, [
                  m("div", O, [
                    m("span", null, p(e.file?.name), 1)
                  ]),
                  m("button", {
                    class: "bg-b-2 centered-x centered-y absolute hidden w-[calc(100%_-_8px)] rounded p-0.5 text-center text-xs font-medium group-hover/upload:block",
                    type: "button",
                    onClick: (a) => s("removeFile", i)
                  }, " Delete ", 8, A)
                ], 64)) : (d(), f("div", L, [
                  n(u(R), {
                    class: "bg-b-2 hover:bg-b-3 text-c-2 h-fit border-0 py-px shadow-none",
                    size: "sm",
                    variant: "outlined",
                    onClick: (a) => S(i)
                  }, {
                    default: t(() => [
                      o[5] || (o[5] = m("span", null, "Upload File", -1)),
                      n(u(B), {
                        class: "ml-1",
                        icon: "Upload",
                        size: "xs",
                        thickness: "2.5"
                      })
                    ]),
                    _: 1
                  }, 8, ["onClick"])
                ]))
              ]),
              _: 2
            }, 1024)) : y("", !0)
          ]),
          _: 2
        }, 1032, ["id", "class"]))), 128))
      ]),
      _: 1
    }));
  }
});
export {
  oe as default
};
