// @scalar/api-client/dist/views/Request/RequestSection/RequestBody.vue.js
import { defineComponent as Q, ref as L, computed as _, watch as N, openBlock as i, createBlock as v, withCtx as d, createTextVNode as V, toDisplayString as k, createElementBlock as y, Fragment as j, createVNode as m, unref as c, createElementVNode as b, nextTick as Z } from "vue";
import I from "../../../components/CodeInput/CodeInput.vue.js";
import ee from "../../../components/DataTable/DataTable.vue.js";
import te from "../../../components/DataTable/DataTableHeader.vue.js";
import F from "../../../components/DataTable/DataTableRow.vue.js";
import ae from "../../../components/ViewLayout/ViewLayoutCollapse.vue.js";
import { useWorkspace as oe } from "../../../store/workspace.js";
import { ScalarListbox as ne, ScalarButton as D, ScalarIcon as q } from "@scalar/components";
import { createRequestExampleParameter as S } from "@scalar/oas-utils/entities/workspace/spec";
import W from "./RequestTable.vue.js";
import { useFileDialog as A } from "../../../hooks/useFileDialog.js";
const le = {
  key: 0,
  class: "text-c-3 flex min-h-14 w-full items-center justify-center rounded border border-dashed text-center text-base"
}, re = /* @__PURE__ */ b("span", null, "No Body", -1), ue = [
  re
], se = {
  key: 0,
  class: "text-c-3 flex min-h-10 w-full items-center justify-center p-2 text-sm"
}, ie = /* @__PURE__ */ b("span", null, "No Body", -1), de = [
  ie
], ce = {
  key: 1,
  class: "flex items-center justify-center p-1.5 overflow-hidden"
}, me = { class: "text-c-2 text-xs w-full border rounded p-1 max-w-full overflow-hidden whitespace-nowrap" }, pe = /* @__PURE__ */ b("span", null, "Upload File", -1), Re = /* @__PURE__ */ Q({
  __name: "RequestBody",
  props: {
    title: {},
    body: {},
    formData: {}
  },
  setup(fe) {
    const M = {
      multipartForm: "Multipart Form",
      formUrlEncoded: "Form URL Encoded",
      binaryFile: "Binary File",
      json: "JSON",
      xml: "XML",
      yaml: "YAML",
      edn: "EDN",
      other: "Other",
      none: "None"
    }, { activeRequest: p, activeExample: t, requestExampleMutators: r } = oe(), u = L("none"), h = L(null), O = {
      json: "json",
      xml: "xml",
      yaml: "yaml",
      edn: "edn",
      other: "html"
    }, z = _(() => {
      const e = u.value;
      return O[e] ?? "plaintext";
    });
    function R() {
      console.log("deleteRow");
    }
    const C = (e, a, o) => {
      if (!p.value || !t.value) return;
      const s = f.value;
      if (s.length > e) {
        const n = [...s];
        n[e] = { ...n[e], [a]: o }, (n[e].key !== "" || n[e].value !== "") && (n[e].enabled = !0), n[e].key === "" && n[e].value === "" && n.splice(e, 1), r.edit(
          t.value.uid,
          "body.formData.value",
          n
        );
      } else {
        const n = [S({ [a]: o })];
        r.edit(
          t.value.uid,
          "body.formData.value",
          n
        ), Z(() => {
          var P;
          if (!h.value) return;
          (P = h.value.querySelectorAll("input")[a === "key" ? 0 : 1]) == null || P.focus();
        });
      }
    }, f = _(
      () => {
        var e;
        return ((e = t.value) == null ? void 0 : e.body.formData.value) ?? [];
      }
    );
    function $() {
      f.value.length === 0 && w();
    }
    const w = () => {
      if (!p.value || !t.value) return;
      const e = S({
        enabled: !1
      }), a = [...f.value, e];
      r.edit(
        t.value.uid,
        "body.formData.value",
        a
      );
    }, H = (e) => {
      !p.value || !t.value || r.edit(
        t.value.uid,
        "body.raw.value",
        e
      );
    }, J = (e) => e === "multipartForm" ? "multipart/form-data" : e === "formUrlEncoded" ? "application/x-www-form-urlencoded" : e === "binaryFile" ? "application/octet-stream" : e !== "none" ? `application/${e}` : "", X = (e) => e === "multipartForm" || e === "formUrlEncoded" ? "formData" : e === "binaryFile" ? "binary" : "raw", Y = (e) => {
      const a = J(e), o = X(e);
      r.edit(
        t.value.uid,
        "body.activeBody",
        o
      );
      const s = [
        ...t.value.parameters.headers
      ].findIndex(
        (l) => l.key.toLowerCase() === "Content-Type".toLowerCase()
      ), n = [...t.value.parameters.headers][s];
      if ((n == null ? void 0 : n.value) !== a) {
        if (n && a) {
          const l = [...t.value.parameters.headers];
          l[s].value = a, r.edit(
            t.value.uid,
            "parameters.headers",
            l
          );
        } else if (a) {
          const l = [
            ...t.value.parameters.headers.filter(
              (E) => E.key.toLowerCase() !== "Content-Type".toLowerCase()
            )
          ];
          l.push({
            key: "Content-Type",
            value: a,
            enabled: !0
          }), r.edit(
            t.value.uid,
            "parameters.headers",
            l
          );
        }
      }
    }, T = async (e) => {
      const { open: a } = A({
        onChange: async (o) => {
          const s = o == null ? void 0 : o[0];
          if (s && p.value && t.value) {
            const l = [...f.value];
            l[e] = {
              ...l[e],
              file: s
            }, r.edit(
              t.value.uid,
              "body.formData.value",
              l
            );
          }
        },
        multiple: !1,
        accept: "*/*"
      });
      a();
    };
    function G() {
      !p.value || !t.value || r.edit(t.value.uid, "body.binary", void 0);
    }
    function U(e) {
      if (!p.value || !t.value) return;
      const o = [...f.value];
      o[e] = {
        ...o[e],
        file: void 0
      }, r.edit(
        t.value.uid,
        "body.formData.value",
        o
      );
    }
    function K() {
      const { open: e } = A({
        onChange: async (a) => {
          const o = a == null ? void 0 : a[0];
          o && p.value && t.value && r.edit(
            t.value.uid,
            "body.binary",
            o
          );
        },
        multiple: !1,
        accept: "*/*"
      });
      e();
    }
    const B = Object.entries(M).map(
      ([e, a]) => ({
        id: e,
        label: a,
        value: e
      })
    ), x = _({
      get: () => B.find((e) => e.id === u.value),
      set: (e) => {
        e != null && e.id && (u.value = e.id);
      }
    }), g = _(() => t.value ? t.value.body.activeBody === "formData" && t.value.body.formData && t.value.body.formData.value.length > 0 ? "multipartForm" : t.value.body.activeBody === "raw" ? t.value.body.raw.encoding : u.value : "none");
    return g.value !== "none" && (u.value = g.value), N(
      g,
      (e) => {
        e && (u.value = e);
      },
      { immediate: !0 }
    ), N(
      u,
      (e) => {
        (e === "multipartForm" || e === "formUrlEncoded") && $(), Y(e);
      },
      { immediate: !0 }
    ), (e, a) => (i(), v(ae, null, {
      title: d(() => [
        V(k(e.title), 1)
      ]),
      default: d(() => [
        e.body && e.body.length === 0 && e.formData && e.formData.length === 0 ? (i(), y("div", le, ue)) : e.formData && e.formData.length > 0 ? (i(), y(j, { key: 1 }, [], 64)) : (i(), v(ee, {
          key: 2,
          columns: [""]
        }, {
          default: d(() => [
            m(F, null, {
              default: d(() => [
                m(te, { class: "relative col-span-full flex h-8 cursor-pointer items-center px-[2.25px] py-[2.25px]" }, {
                  default: d(() => [
                    m(c(ne), {
                      modelValue: x.value,
                      "onUpdate:modelValue": a[0] || (a[0] = (o) => x.value = o),
                      class: "text-xxs w-full",
                      fullWidth: "",
                      options: c(B),
                      teleport: ""
                    }, {
                      default: d(() => [
                        m(c(D), {
                          class: "flex gap-1.5 h-auto px-1.5 text-c-2 font-normal hover:text-c-1",
                          fullWidth: "",
                          variant: "ghost"
                        }, {
                          default: d(() => {
                            var o;
                            return [
                              b("span", null, k((o = x.value) == null ? void 0 : o.label), 1),
                              m(c(q), {
                                icon: "ChevronDown",
                                size: "xs",
                                thickness: "2.5"
                              })
                            ];
                          }),
                          _: 1
                        })
                      ]),
                      _: 1
                    }, 8, ["modelValue", "options"])
                  ]),
                  _: 1
                })
              ]),
              _: 1
            }),
            m(F, null, {
              default: d(() => {
                var o, s, n;
                return [
                  u.value === "none" ? (i(), y("div", se, de)) : u.value === "binaryFile" ? (i(), y("div", ce, [
                    (o = c(t)) != null && o.body.binary ? (i(), y(j, { key: 0 }, [
                      b("span", me, k((s = c(t)) == null ? void 0 : s.body.binary.name), 1),
                      m(c(D), {
                        class: "bg-b-2 hover:bg-b-3 border-0 text-c-2 ml-1 shadow-none",
                        size: "sm",
                        variant: "outlined",
                        onClick: G
                      }, {
                        default: d(() => [
                          V(" Delete ")
                        ]),
                        _: 1
                      })
                    ], 64)) : (i(), v(c(D), {
                      key: 1,
                      class: "bg-b-2 hover:bg-b-3 border-0 text-c-2 shadow-none",
                      size: "sm",
                      variant: "outlined",
                      onClick: K
                    }, {
                      default: d(() => [
                        pe,
                        m(c(q), {
                          class: "ml-1",
                          icon: "UploadSimple",
                          size: "xs",
                          thickness: "2.5"
                        })
                      ]),
                      _: 1
                    }))
                  ])) : u.value == "multipartForm" ? (i(), v(W, {
                    key: 2,
                    ref_key: "tableWrapperRef",
                    ref: h,
                    class: "!m-0 rounded-t-none shadow-none border-l-0 border-r-0 border-t-0 border-b-0",
                    columns: ["32px", "", "", "61px"],
                    items: f.value,
                    showUploadButton: "",
                    onAddRow: w,
                    onDeleteRow: R,
                    onRemoveFile: U,
                    onUpdateRow: C,
                    onUploadFile: T
                  }, null, 8, ["items"])) : u.value == "formUrlEncoded" ? (i(), v(W, {
                    key: 3,
                    ref_key: "tableWrapperRef",
                    ref: h,
                    class: "!m-0 rounded-t-none border-t-0 shadow-none border-l-0 border-r-0 border-t-0 border-b-0",
                    columns: ["32px", "", "", "61px"],
                    items: f.value,
                    showUploadButton: "",
                    onAddRow: w,
                    onDeleteRow: R,
                    onRemoveFile: U,
                    onUpdateRow: C,
                    onUploadFile: T
                  }, null, 8, ["items"])) : (i(), v(I, {
                    key: 4,
                    content: "",
                    language: z.value,
                    lineNumbers: "",
                    lint: "",
                    modelValue: ((n = c(t)) == null ? void 0 : n.body.raw.value) ?? "",
                    "onUpdate:modelValue": H
                  }, null, 8, ["language", "modelValue"]))
                ];
              }),
              _: 1
            }),
            m(F)
          ]),
          _: 1
        }))
      ]),
      _: 1
    }));
  }
});
export {
  Re as default
};
