// @scalar/api-client/dist/views/Request/RequestSection/RequestAuthDataTableInput.vue.js
import { defineComponent as d, openBlock as i, createBlock as p, mergeProps as m, withCtx as a, createElementVNode as f, renderSlot as r } from "vue";
import V from "../../../components/DataTable/DataTableInput.vue.js";
const y = ["for"], C = /* @__PURE__ */ d({
  __name: "RequestAuthDataTableInput",
  props: {
    id: {},
    type: {},
    containerClass: {},
    required: { type: Boolean, default: !1 },
    modelValue: {},
    readOnly: { type: Boolean, default: !1 }
  },
  emits: ["update:modelValue", "inputFocus", "inputBlur", "selectVariable"],
  setup(u, { emit: s }) {
    const t = u, o = s;
    return (n, e) => (i(), p(V, m({
      id: t.id,
      containerClass: t.containerClass,
      modelValue: t.modelValue,
      readOnly: t.readOnly,
      required: t.required,
      type: t.type
    }, n.$attrs, {
      onInputBlur: e[0] || (e[0] = (l) => o("inputBlur")),
      onInputFocus: e[1] || (e[1] = (l) => o("inputFocus")),
      onSelectVariable: e[2] || (e[2] = (l) => o("selectVariable", l)),
      "onUpdate:modelValue": e[3] || (e[3] = (l) => o("update:modelValue", l))
    }), {
      default: a(() => [
        f("label", {
          for: t.id
        }, [
          r(n.$slots, "default")
        ], 8, y)
      ]),
      icon: a(() => [
        r(n.$slots, "icon")
      ]),
      _: 3
    }, 16, ["id", "containerClass", "modelValue", "readOnly", "required", "type"]));
  }
});
export {
  C as default
};
