// @scalar/api-client/dist/views/Request/RequestSection/RequestTableTooltip.vue.js
import { defineComponent as m, openBlock as t, createBlock as c, unref as i, withCtx as a, createElementVNode as n, createVNode as l, createElementBlock as o, toDisplayString as s, createCommentVNode as r } from "vue";
import { ScalarTooltip as f, ScalarIcon as p } from "@scalar/components";
const d = { class: "pl-1 pr-1.5 py-[9px]" }, b = { class: "grid gap-1.5 pointer-events-none min-w-48 w-content shadow-lg rounded bg-b-1 z-100 p-2 text-xxs leading-5 z-10 text-c-1" }, h = { class: "flex items-center text-c-2" }, u = { key: 0 }, g = {
  key: 1,
  class: "before:content-['路'] before:block before:mx-[0.5ch] flex"
}, x = {
  key: 2,
  class: "before:content-['路'] before:block before:mx-[0.5ch] flex whitespace-nowrap"
}, _ = {
  key: 3,
  class: "before:content-['路'] before:block before:mx-[0.5ch] flex whitespace-nowrap"
}, k = {
  key: 4,
  class: "before:content-['路'] before:block before:mx-[0.5ch] flex whitespace-nowrap"
}, y = {
  key: 0,
  class: "leading-snug text-pretty text-sm",
  style: { maxWidth: "16rem" }
}, B = /* @__PURE__ */ m({
  __name: "RequestTableTooltip",
  props: {
    item: {}
  },
  setup(w) {
    return (e, v) => (t(), c(i(f), {
      align: "start",
      class: "w-full",
      delay: 0,
      side: "left",
      triggerClass: "before:absolute before:content-[''] before:bg-gradient-to-r before:from-transparent before:to-b-1 before:min-h-full before:right-[23px] before:w-3 absolute h-8 right-0"
    }, {
      trigger: a(() => [
        n("div", d, [
          l(i(p), {
            class: "text-c-3 group-hover/info:text-c-1",
            icon: "Info",
            size: "sm",
            thickness: "1.5"
          })
        ])
      ]),
      content: a(() => [
        n("div", b, [
          n("div", h, [
            e.item.type ? (t(), o("span", u, s(e.item.type), 1)) : r("", !0),
            e.item.format ? (t(), o("span", g, s(e.item.format), 1)) : r("", !0),
            e.item.minimum ? (t(), o("span", x, "min: " + s(e.item.minimum), 1)) : r("", !0),
            e.item.maximum ? (t(), o("span", _, "max: " + s(e.item.maximum), 1)) : r("", !0),
            e.item.default ? (t(), o("span", k, "default: " + s(e.item.default), 1)) : r("", !0)
          ]),
          e.item.description ? (t(), o("span", y, s(e.item.description), 1)) : r("", !0)
        ])
      ]),
      _: 1
    }));
  }
});
export {
  B as default
};
