// @scalar/api-client/dist/views/Request/RequestSection/RequestParams.vue.js
import { defineComponent as K, computed as d, onMounted as P, ref as $, watch as x, openBlock as E, createBlock as T, withCtx as i, createTextVNode as y, toDisplayString as A, createElementVNode as R, createVNode as g, unref as B, withModifiers as M, nextTick as h } from "vue";
import N from "../../../components/ViewLayout/ViewLayoutCollapse.vue.js";
import { useWorkspace as S } from "../../../store/workspace.js";
import V from "./RequestTable.vue.js";
import { ScalarButton as W } from "@scalar/components";
import { createRequestExampleParameter as _ } from "@scalar/oas-utils/entities/workspace/spec";
const z = { class: "text-c-2 flex whitespace-nowrap opacity-0 group-hover/params:opacity-100 request-meta-buttons" }, O = /* @__PURE__ */ K({
  __name: "RequestParams",
  props: {
    title: {},
    paramKey: {}
  },
  setup(k) {
    const s = k, { activeRequest: n, activeExample: t, requestExampleMutators: u } = S(), l = d(
      () => {
        var e;
        return ((e = t.value) == null ? void 0 : e.parameters[s.paramKey]) ?? [];
      }
    );
    P(() => {
      c();
    });
    const o = () => {
      if (!n.value || !t.value) return;
      const e = _({ enabled: !1 }), r = [...l.value, e];
      u.edit(
        t.value.uid,
        `parameters.${s.paramKey}`,
        r
      );
    }, p = $(null), b = (e, r, m) => {
      if (!n.value || !t.value) return;
      const f = l.value;
      if (f.length > e) {
        const a = [...f];
        a[e] = { ...a[e], [r]: m }, (a[e].key !== "" || a[e].value !== "") && (a[e].enabled = !0), a[e].key === "" && a[e].value === "" && a.splice(e, 1), u.edit(
          t.value.uid,
          `parameters.${s.paramKey}`,
          a
        );
      } else {
        const a = [_({ [r]: m })];
        u.edit(
          t.value.uid,
          `parameters.${s.paramKey}`,
          a
        ), h(() => {
          var v;
          if (!p.value) return;
          (v = p.value.querySelectorAll("input")[r === "key" ? 0 : 1]) == null || v.focus();
        });
      }
    }, q = (e, r) => n.value && t.value && u.edit(
      t.value.uid,
      `parameters.${s.paramKey}.${e}.enabled`,
      r
    ), w = () => {
      !n.value || !t.value || (u.edit(
        t.value.uid,
        `parameters.${s.paramKey}`,
        []
      ), h(() => {
        o();
      }));
    };
    function c() {
      if (l.value.length === 0)
        o();
      else if (l.value.length >= 1) {
        const e = l.value[l.value.length - 1];
        e.key !== "" && e.value !== "" && o();
      }
    }
    const C = d(
      () => l.value.filter((e) => e.key || e.value).length
    );
    return x(
      () => t.value,
      (e, r) => {
        e !== r && c();
      },
      { immediate: !0 }
    ), (e, r) => (E(), T(N, {
      class: "group/params",
      itemCount: C.value
    }, {
      title: i(() => [
        y(A(e.title), 1)
      ]),
      actions: i(() => [
        R("div", z, [
          g(B(W), {
            class: "px-1 transition-none",
            size: "sm",
            variant: "ghost",
            onClick: M(w, ["stop"])
          }, {
            default: i(() => [
              y(" Clear ")
            ]),
            _: 1
          })
        ])
      ]),
      default: i(() => [
        R("div", {
          ref_key: "tableWrapperRef",
          ref: p
        }, [
          g(V, {
            class: "flex-1",
            columns: ["32px", "", ""],
            items: l.value,
            onAddRow: o,
            onToggleRow: q,
            onUpdateRow: b
          }, null, 8, ["items"])
        ], 512)
      ]),
      _: 1
    }, 8, ["itemCount"]));
  }
});
export {
  O as default
};
