// @scalar/api-client/dist/views/Request/RequestSection/RequestAuth.vue2.js
import { defineComponent as K, ref as L, computed as C, openBlock as i, createBlock as b, withCtx as o, createElementVNode as h, toDisplayString as g, createVNode as u, unref as n, createElementBlock as c, Fragment as v, renderList as q, createTextVNode as f, withModifiers as z, createCommentVNode as A, capitalize as E, pushScopeId as Q, popScopeId as G } from "vue";
import J from "../../../components/DataTable/DataTableCell.vue.js";
import X from "../../../components/ViewLayout/ViewLayoutCollapse.vue.js";
import { useWorkspace as Y } from "../../../store/workspace.js";
import V from "./RequestAuthDataTableInput.vue.js";
import { ScalarComboboxMultiselect as Z, ScalarButton as ee, ScalarIcon as F } from "@scalar/components";
import { camelToTitleWords as te } from "@scalar/oas-utils/helpers";
import { ADD_AUTH_OPTIONS as le, ADD_AUTH_DICT as ae } from "../consts/new-auth-options.js";
import oe from "../../../components/DataTable/DataTable.vue.js";
import _ from "../../../components/DataTable/DataTableRow.vue.js";
import ue from "../../../components/DataTable/DataTableHeader.vue.js";
import ne from "../components/OAuth2.vue.js";
const D = (S) => (Q("data-v-afbf407f"), S = S(), G(), S), ie = { class: "flex gap-1" }, se = /* @__PURE__ */ D(() => /* @__PURE__ */ h("div", { class: "text-c-2 h-8 flex min-w-[120px] items-center border-r-1/2 pr-0 pl-2" }, " Auth Type ", -1)), re = {
  key: 0,
  class: "flex relative scroll-timeline-x w-full"
}, de = /* @__PURE__ */ D(() => /* @__PURE__ */ h("div", { class: "fade-left" }, null, -1)), ce = { class: "flex flex-1 gap-0.75 mr-1.5 items-center" }, pe = /* @__PURE__ */ D(() => /* @__PURE__ */ h("div", { class: "fade-right" }, null, -1)), me = {
  key: 1,
  class: "pl-2"
}, $ = "*sclrMulti*", Te = /* @__PURE__ */ K({
  __name: "RequestAuth",
  props: {
    title: {}
  },
  setup(S) {
    const {
      activeCollection: T,
      activeRequest: k,
      activeSecurityRequirements: O,
      activeSecuritySchemes: R,
      isReadOnly: w,
      requestMutators: P,
      securitySchemeMutators: N,
      securitySchemes: U
    } = Y(), B = L(null), I = (t) => {
      switch (t == null ? void 0 : t.type) {
        case "apiKey":
          return `${E(t.in)}`;
        case "http":
          return `${E(t.scheme)} Authentication`;
        case "oauth2":
          return te(t.flow.type);
        case "openIdConnect":
          return "Open ID Connect";
        default:
          return "None";
      }
    }, j = C(
      () => {
        var e;
        const t = (e = T.value) == null ? void 0 : e.securitySchemeDict, l = O.value.flatMap((d) => {
          const a = Object.keys(d);
          if (a.length === 0 && w.value)
            return { id: "none", label: "None" };
          if (!t) return [];
          if (a.length > 1) {
            const p = a.map((r) => {
              const x = t[r];
              return U[x];
            }), s = p.map((r) => r.nameKey).join(" & ");
            return {
              id: $ + p.map((r) => r.uid).join($),
              label: s
            };
          }
          return a.flatMap((p) => {
            const s = t[p], r = U[s], x = I(r);
            return {
              id: s,
              label: x
            };
          });
        });
        if (w.value)
          return l;
        {
          const d = k.value.securitySchemeUids.map((s) => {
            const r = U[s], x = I(r);
            return {
              id: s,
              label: "name" in r && r.name.length ? `${x} (${r.name})` : x
            };
          }), a = k.value.selectedSecuritySchemeUids, p = d.filter(
            (s) => a.includes(s.id)
          );
          return [
            { label: "Select auth", options: [...l, ...p] },
            {
              label: "Add new auth",
              options: le
            }
          ];
        }
      }
    ), M = (t) => {
      if (t.startsWith($)) {
        const l = t.split($);
        return l.shift(), l;
      } else
        return [t];
    }, m = C({
      // Grab the selected OR first security scheme
      get: () => {
        var l;
        const t = new Set(
          ((l = k.value) == null ? void 0 : l.selectedSecuritySchemeUids) || []
        );
        return j.value.flatMap(
          (e) => "options" in e ? e.options : e
        ).filter(
          (e) => M(e.id).some((d) => t.has(d))
        );
      },
      // Update the selected auth per this request
      set: (t) => {
        var e, d, a, p;
        const l = t.find(
          (s) => ae[s.id]
        );
        l != null && l.payload && T.value ? ((p = (a = (d = (e = B.value) == null ? void 0 : e.comboboxPopoverRef) == null ? void 0 : d.popoverButtonRef) == null ? void 0 : a.el) == null || p.click(), N.add(
          l.payload,
          T.value.uid,
          k.value,
          !0
        )) : P.edit(
          k.value.uid,
          "selectedSecuritySchemeUids",
          t.flatMap((s) => M(s.id))
        );
      }
    }), W = C(
      () => m.value.flatMap(({ id: t }) => M(t)).map((t) => {
        const l = U[t];
        return `${I(l)} (${l.nameKey})`;
      })
    ), y = (t, l, e) => N.edit(t.uid ?? "", l, e), H = (t) => {
      const l = m.value.filter(
        (e) => e.id !== t
      );
      m.value = l;
    };
    return (t, l) => (i(), b(X, {
      class: "group/params",
      itemCount: m.value.length
    }, {
      title: o(() => [
        h("div", ie, g(t.title), 1)
      ]),
      default: o(() => [
        h("form", null, [
          u(n(oe), {
            class: "flex-1",
            columns: [""]
          }, {
            default: o(() => [
              u(n(_), null, {
                default: o(() => [
                  u(n(ue), { class: "relative col-span-full cursor-pointer py-[0px] px-[0px] flex items-center" }, {
                    default: o(() => [
                      u(n(Z), {
                        ref_key: "comboboxRef",
                        ref: B,
                        modelValue: m.value,
                        "onUpdate:modelValue": l[0] || (l[0] = (e) => m.value = e),
                        class: "text-xs w-full",
                        fullWidth: "",
                        multiple: "",
                        options: j.value,
                        style: { "margin-left": "120px" },
                        teleport: ""
                      }, {
                        default: o(() => [
                          u(n(ee), {
                            class: "h-auto py-0 px-0 text-c-2 hover:text-c-1 font-normal justify-start",
                            fullWidth: "",
                            variant: "ghost"
                          }, {
                            default: o(() => [
                              se,
                              m.value.length ? (i(), c("div", re, [
                                de,
                                h("div", ce, [
                                  (i(!0), c(v, null, q(m.value, (e) => (i(), c("span", {
                                    key: e.id,
                                    class: "cm-pill flex items-center mx-0 h-fit"
                                  }, [
                                    f(g(e.label) + " ", 1),
                                    u(n(F), {
                                      class: "ml-1 cursor-pointer text-c-3 hover:text-c-1",
                                      icon: "Close",
                                      size: "xs",
                                      onClick: z((d) => H(e.id), ["stop"])
                                    }, null, 8, ["onClick"])
                                  ]))), 128))
                                ]),
                                pe
                              ])) : (i(), c("div", me, " None ")),
                              u(n(F), {
                                class: "min-w-3 ml-auto mr-2.5",
                                icon: "ChevronDown",
                                size: "xs"
                              })
                            ]),
                            _: 1
                          })
                        ]),
                        _: 1
                      }, 8, ["modelValue", "options"])
                    ]),
                    _: 1
                  })
                ]),
                _: 1
              }),
              (i(!0), c(v, null, q(n(R), (e, d) => (i(), c(v, {
                key: e.uid
              }, [
                n(R).length > 1 ? (i(), b(n(_), {
                  key: 0,
                  class: "group/delete"
                }, {
                  default: o(() => [
                    u(J, { class: "text-c-2 pl-2 text-xs font-medium flex items-center bg-b-2" }, {
                      default: o(() => [
                        f(g(W.value[d]), 1)
                      ]),
                      _: 2
                    }, 1024)
                  ]),
                  _: 2
                }, 1024)) : A("", !0),
                e.type === "http" ? (i(), c(v, { key: 1 }, [
                  e.scheme === "bearer" ? (i(), b(n(_), { key: 0 }, {
                    default: o(() => [
                      u(V, {
                        id: `http-bearer-token-${e.uid}`,
                        modelValue: e.value,
                        placeholder: "Token",
                        type: "password",
                        "onUpdate:modelValue": (a) => y(e, "value", a)
                      }, {
                        default: o(() => [
                          f(" Bearer Token ")
                        ]),
                        _: 2
                      }, 1032, ["id", "modelValue", "onUpdate:modelValue"])
                    ]),
                    _: 2
                  }, 1024)) : e.scheme === "basic" ? (i(), c(v, { key: 1 }, [
                    u(n(_), null, {
                      default: o(() => [
                        u(V, {
                          id: `http-basic-username-${e.uid}`,
                          class: "text-c-2",
                          modelValue: e.value,
                          placeholder: "ScalarEnjoyer01",
                          required: "",
                          "onUpdate:modelValue": (a) => y(e, "value", a)
                        }, {
                          default: o(() => [
                            f(" Username ")
                          ]),
                          _: 2
                        }, 1032, ["id", "modelValue", "onUpdate:modelValue"])
                      ]),
                      _: 2
                    }, 1024),
                    u(n(_), null, {
                      default: o(() => [
                        u(V, {
                          id: `http-basic-password-${e.uid}`,
                          modelValue: e.secondValue,
                          placeholder: "xxxxxx",
                          type: "password",
                          "onUpdate:modelValue": (a) => y(e, "secondValue", a)
                        }, {
                          default: o(() => [
                            f(" Password ")
                          ]),
                          _: 2
                        }, 1032, ["id", "modelValue", "onUpdate:modelValue"])
                      ]),
                      _: 2
                    }, 1024)
                  ], 64)) : A("", !0)
                ], 64)) : e.type === "apiKey" ? (i(), c(v, { key: 2 }, [
                  n(w) ? (i(), b(n(_), { key: 1 }, {
                    default: o(() => [
                      u(V, {
                        id: `api-key-value-${e.uid}`,
                        modelValue: e.value,
                        placeholder: "Value",
                        "onUpdate:modelValue": (a) => y(e, "value", a)
                      }, {
                        default: o(() => [
                          f(g(e.name), 1)
                        ]),
                        _: 2
                      }, 1032, ["id", "modelValue", "onUpdate:modelValue"])
                    ]),
                    _: 2
                  }, 1024)) : (i(), c(v, { key: 0 }, [
                    u(n(_), null, {
                      default: o(() => [
                        u(V, {
                          id: `api-key-name-${e.uid}`,
                          modelValue: e.name,
                          placeholder: "api-key",
                          "onUpdate:modelValue": (a) => y(e, "name", a)
                        }, {
                          default: o(() => [
                            f(" Name ")
                          ]),
                          _: 2
                        }, 1032, ["id", "modelValue", "onUpdate:modelValue"])
                      ]),
                      _: 2
                    }, 1024),
                    u(n(_), null, {
                      default: o(() => [
                        u(V, {
                          id: `api-key-value-add-${e.uid}`,
                          modelValue: e.value,
                          placeholder: "QUxMIFlPVVIgQkFTRSBBUkUgQkVMT05HIFRPIFVT",
                          "onUpdate:modelValue": (a) => y(e, "value", a)
                        }, {
                          default: o(() => [
                            f(" Value ")
                          ]),
                          _: 2
                        }, 1032, ["id", "modelValue", "onUpdate:modelValue"])
                      ]),
                      _: 2
                    }, 1024)
                  ], 64))
                ], 64)) : e.type === "oauth2" ? (i(), b(n(ne), {
                  key: 3,
                  scheme: e
                }, null, 8, ["scheme"])) : A("", !0)
              ], 64))), 128))
            ]),
            _: 1
          })
        ])
      ]),
      _: 1
    }, 8, ["itemCount"]));
  }
});
export {
  Te as default
};
