// @scalar/api-client/dist/views/Request/RequestSection/RequestParams.vue.js
import { defineComponent as T, computed as o, onMounted as O, nextTick as d, ref as S, watch as N, createBlock as v, openBlock as f, withCtx as r, createElementVNode as p, createCommentVNode as h, createVNode as k, normalizeClass as A, unref as g, withModifiers as B, createTextVNode as P, toDisplayString as R } from "vue";
import { ScalarTooltip as M, ScalarButton as W } from "@scalar/components";
import { requestExampleParametersSchema as x } from "@scalar/oas-utils/entities/spec";
import z from "../../../components/ViewLayout/ViewLayoutCollapse.vue.js";
import w from "./RequestTable.vue.js";
import { useWorkspace as D } from "../../../store/store.js";
const G = { class: "text-c-2 request-meta-buttons flex whitespace-nowrap opacity-0 group-hover/params:opacity-100 has-[:focus-visible]:opacity-100" }, U = { class: "sr-only" }, Z = /* @__PURE__ */ T({
  __name: "RequestParams",
  props: {
    example: {},
    environment: {},
    envVariables: {},
    workspace: {},
    title: {},
    label: {},
    paramKey: {},
    readOnlyEntries: { default: () => [] },
    invalidParams: {}
  },
  setup(a) {
    const { requestExampleMutators: i } = D(), n = o(() => a.example.parameters[a.paramKey] ?? []);
    O(() => {
      d(() => {
        y();
      });
    });
    const m = () => {
      const e = x.parse({ enabled: !1 }), t = [...n.value, e];
      i.edit(a.example.uid, `parameters.${a.paramKey}`, t);
    }, u = S(null), V = (e, t, s) => {
      const c = n.value;
      if (c.length > e) {
        const l = [...c];
        if (!l[e])
          return;
        l[e] = { ...l[e], [t]: s }, (l[e].key !== "" || l[e].value !== "") && (l[e].enabled = !0), l[e].key === "" && l[e].value === "" && l.splice(e, 1), i.edit(
          a.example.uid,
          `parameters.${a.paramKey}`,
          l
        );
      } else {
        const l = [x.parse({ [t]: s })];
        i.edit(a.example.uid, `parameters.${a.paramKey}`, l), d(() => {
          if (!u.value)
            return;
          u.value.querySelectorAll("input")[t === "key" ? 0 : 1]?.focus();
        });
      }
      e === c.length - 1 && m();
    }, C = (e, t) => i.edit(
      a.example.uid,
      `parameters.${a.paramKey}.${e}.enabled`,
      t
    ), K = () => {
      const e = n.value.filter((t) => t.required);
      i.edit(
        a.example.uid,
        `parameters.${a.paramKey}`,
        e
      ), d(() => m());
    }, $ = (e) => {
      const t = n.value;
      if (t.length > e) {
        const s = [...t];
        s.splice(e, 1), i.edit(
          a.example.uid,
          `parameters.${a.paramKey}`,
          s
        );
      }
    };
    function y() {
      if (n.value.length === 0)
        m();
      else if (n.value.length >= 1) {
        const e = n.value[n.value.length - 1];
        e && e.key !== "" && e.value !== "" && m();
      }
    }
    const q = o(
      () => n.value.filter((e) => e.key || e.value).length
    ), E = o(() => n.value.length > 1);
    N(
      () => a.example,
      (e, t) => {
        e !== t && y();
      },
      { immediate: !0 }
    );
    const b = o(() => (a.readOnlyEntries ?? []).length > 0);
    return (e, t) => (f(), v(z, {
      class: "group/params",
      itemCount: q.value
    }, {
      title: r(() => [
        P(R(a.title), 1)
      ]),
      actions: r(() => [
        p("div", G, [
          E.value ? (f(), v(g(M), {
            key: 0,
            content: "Clear optional parameters",
            placement: "left"
          }, {
            default: r(() => [
              k(g(W), {
                class: "pr-0.75 pl-1 transition-none",
                size: "sm",
                variant: "ghost",
                onClick: B(K, ["stop"])
              }, {
                default: r(() => [
                  t[0] || (t[0] = P(" Clear ", -1)),
                  p("span", U, "All " + R(a.title), 1)
                ]),
                _: 1
              })
            ]),
            _: 1
          })) : h("", !0)
        ])
      ]),
      default: r(() => [
        p("div", {
          ref_key: "tableWrapperRef",
          ref: u
        }, [
          b.value ? (f(), v(w, {
            key: 0,
            class: A(["flex-1", { "bg-c-3/5": b.value }]),
            columns: ["32px", "", ""],
            envVariables: a.envVariables,
            environment: a.environment,
            invalidParams: a.invalidParams,
            isGlobal: "",
            isReadOnly: "",
            items: a.readOnlyEntries,
            label: a.label,
            workspace: a.workspace
          }, null, 8, ["class", "envVariables", "environment", "invalidParams", "items", "label", "workspace"])) : h("", !0),
          k(w, {
            class: "flex-1",
            columns: ["32px", "", ""],
            envVariables: a.envVariables,
            environment: a.environment,
            invalidParams: a.invalidParams,
            items: n.value,
            label: a.label,
            workspace: a.workspace,
            onToggleRow: C,
            onUpdateRow: V,
            onDeleteRow: $
          }, null, 8, ["envVariables", "environment", "invalidParams", "items", "label", "workspace"])
        ], 512)
      ]),
      _: 1
    }, 8, ["itemCount"]));
  }
});
export {
  Z as default
};
