// @scalar/api-client/dist/views/Request/RequestSection/RequestPathParams.vue.js
import { defineComponent as K, computed as R, watch as E, openBlock as d, createBlock as f, withCtx as y, createTextVNode as g, toDisplayString as q, createElementBlock as C, createElementVNode as w } from "vue";
import P from "../../../components/ViewLayout/ViewLayoutCollapse.vue.js";
import { useWorkspace as M } from "../../../store/workspace.js";
import V from "./RequestTable.vue.js";
const B = {
  key: 1,
  class: "text-c-3 px-4 text-sm border rounded min-h-12 justify-center flex items-center bg-b-1 mx-1"
}, N = /* @__PURE__ */ w("code", { class: "bg-b-2 ml-1 px-1 rounded" }, "/endpoint/{my_path_variable}", -1), H = /* @__PURE__ */ K({
  __name: "RequestPathParams",
  props: {
    title: {},
    paramKey: {}
  },
  setup(_) {
    const s = _, {
      activeRequest: o,
      activeExample: r,
      requestMutators: h,
      requestExampleMutators: v
    } = M(), u = R(
      () => {
        var e;
        return ((e = r.value) == null ? void 0 : e.parameters[s.paramKey]) ?? [];
      }
    ), x = (e, n, a) => {
      var p, t;
      if (!o.value || !r.value) return;
      const c = r.value.parameters[s.paramKey], l = (p = c[e]) == null ? void 0 : p.key;
      if (n === "key") {
        if ((t = c[e]) != null && t.required)
          return;
        if (a) {
          const i = encodeURIComponent(l), m = encodeURIComponent(a), $ = new RegExp(`(?<=/):${i}(?=[/?#]|$)`, "g"), k = o.value.path.replace(
            $,
            `:${m}`
          );
          h.edit(o.value.uid, "path", k);
        } else {
          c.splice(e, 1);
          const i = new RegExp(`/:${encodeURIComponent(l)}(?=[/?#]|$)`, "g"), m = o.value.path.replace(i, "");
          h.edit(o.value.uid, "path", m);
        }
      }
      v.edit(
        r.value.uid,
        `parameters.${s.paramKey}.${e}.${n}`,
        a
      );
    }, b = (e) => {
      var p;
      if (!r.value) return;
      const n = ((p = e.match(new RegExp("(?<!{){([^{}]+)}(?!})", "g"))) == null ? void 0 : p.map((t) => t.slice(1, -1))) || [], a = r.value.parameters[s.paramKey], c = new Map(a.map((t) => [t.key, t])), l = n.map(
        (t) => c.get(t) || { key: t, value: "", enabled: !0 }
      );
      a.forEach((t) => {
        !n.includes(t.key) && (t.value || t.required) && l.push(t);
      }), a.splice(0, a.length, ...l), v.edit(
        r.value.uid,
        `parameters.${s.paramKey}`,
        a
      );
    };
    return E(
      () => {
        var e;
        return (e = r.value) == null ? void 0 : e.url;
      },
      (e) => {
        e && b(e);
      }
    ), (e, n) => (d(), f(P, {
      itemCount: u.value.length
    }, {
      title: y(() => [
        g(q(e.title), 1)
      ]),
      default: y(() => [
        u.value.length ? (d(), f(V, {
          key: 0,
          class: "flex-1",
          isEnabledHidden: "",
          items: u.value,
          onUpdateRow: x
        }, null, 8, ["items"])) : (d(), C("div", B, [
          g(" You can use variables in your path: "),
          N
        ]))
      ]),
      _: 1
    }, 8, ["itemCount"]));
  }
});
export {
  H as default
};
