// @scalar/api-client/dist/views/Request/RequestSection/RequestSection.vue.js
import { defineComponent as k, computed as h, ref as R, watch as B, openBlock as y, createBlock as b, withCtx as _, createVNode as l, unref as o, createElementVNode as m, createTextVNode as C, createElementBlock as E, createCommentVNode as N, withDirectives as a, vShow as s } from "vue";
import g from "../../../components/ContextBar.vue.js";
import w from "../../../components/ViewLayout/ViewLayoutSection.vue.js";
import { useWorkspace as H } from "../../../store/workspace.js";
import O from "./RequestAuth.vue.js";
import P from "./RequestBody.vue.js";
import d from "./RequestParams.vue.js";
import T from "./RequestPathParams.vue.js";
import { ScalarIcon as V } from "@scalar/components";
const K = { class: "flex-1 flex items-center pointer-events-none" }, $ = {
  key: 0,
  class: "absolute w-full h-full top-0 left-0 pointer-events-auto opacity-0 cursor-text",
  for: "requestname"
}, Q = ["disabled", "value"], D = { class: "request-section-content custom-scroll flex flex-1 flex-col px-2 xl:px-5 py-2.5" }, X = /* @__PURE__ */ k({
  __name: "RequestSection",
  setup(I) {
    const {
      activeRequest: i,
      activeSecuritySchemes: x,
      activeExample: v,
      activeSecurityRequirements: p,
      isReadOnly: r,
      requestMutators: A
    } = H(), c = ["POST", "PUT", "PATCH", "DELETE"], S = h(() => {
      const t = [
        "All",
        "Auth",
        "Request",
        "Cookies",
        "Headers",
        "Query",
        "Body"
      ];
      return v.value.parameters.path.length || t.splice(t.indexOf("Request"), 1), c.includes(i.value.method) || t.splice(t.indexOf("Body"), 1), f.value && t.splice(t.indexOf("Auth"), 1), t;
    }), f = h(
      () => r.value && (p.value.length === 0 || JSON.stringify(p.value) === "[{}]")
    ), e = R("All");
    B(i, (t) => {
      e.value === "Body" && !c.includes(t.method) && (e.value = "All");
    });
    const q = (t) => {
      const n = t.target;
      A.edit(i.value.uid, "summary", n.value);
    };
    return (t, n) => (y(), b(w, null, {
      title: _(() => {
        var u;
        return [
          l(o(V), {
            class: "text-c-3 mr-2 pointer-events-none",
            icon: "ExternalLink",
            size: "sm",
            thickness: "2.5"
          }),
          m("div", K, [
            C(" Request "),
            o(r) ? N("", !0) : (y(), E("label", $)),
            m("input", {
              id: "requestname",
              class: "pl-1 outline-none border-0 text-c-2 rounded pointer-events-auto relative w-full",
              disabled: o(r),
              placeholder: "Request Name",
              value: (u = o(i)) == null ? void 0 : u.summary,
              onInput: q
            }, null, 40, Q)
          ])
        ];
      }),
      default: _(() => [
        m("div", D, [
          l(g, {
            activeSection: e.value,
            sections: S.value,
            onSetActiveSection: n[0] || (n[0] = (u) => e.value = u)
          }, null, 8, ["activeSection", "sections"]),
          a(l(O, {
            index: 0,
            securityScheme: o(x)[0],
            title: "Authentication"
          }, null, 8, ["securityScheme"]), [
            [
              s,
              !f.value && (e.value === "All" || e.value === "Auth")
            ]
          ]),
          a(l(T, {
            paramKey: "path",
            title: "Path Variables"
          }, null, 512), [
            [
              s,
              (e.value === "All" || e.value === "Request") && o(v).parameters.path.length > 0
            ]
          ]),
          a(l(d, {
            paramKey: "cookies",
            title: "Cookies"
          }, null, 512), [
            [s, e.value === "All" || e.value === "Cookies"]
          ]),
          a(l(d, {
            paramKey: "headers",
            title: "Headers"
          }, null, 512), [
            [s, e.value === "All" || e.value === "Headers"]
          ]),
          a(l(d, {
            paramKey: "query",
            title: "Query Parameters"
          }, null, 512), [
            [s, e.value === "All" || e.value === "Query"]
          ]),
          a(l(P, {
            body: "foo",
            title: "Body"
          }, null, 512), [
            [
              s,
              (e.value === "All" || e.value === "Body") && c.includes(o(i).method)
            ]
          ])
        ])
      ]),
      _: 1
    }));
  }
});
export {
  X as default
};
