// @scalar/api-client/dist/views/Request/RequestSection/RequestTable.vue2.js
import { defineComponent as F, openBlock as u, createBlock as r, unref as m, withCtx as d, createElementBlock as w, Fragment as V, renderList as C, createCommentVNode as c, createVNode as i, normalizeClass as I, createSlots as U, createTextVNode as x, toDisplayString as k, createElementVNode as p, computed as E, pushScopeId as T, popScopeId as N } from "vue";
import g from "../../../components/CodeInput/CodeInput.vue.js";
import q from "../../../components/DataTable/DataTable.vue.js";
import f from "../../../components/DataTable/DataTableCell.vue.js";
import z from "../../../components/DataTable/DataTableCheckbox.vue.js";
import D from "../../../components/DataTable/DataTableRow.vue.js";
import { ScalarButton as H, ScalarIcon as A } from "@scalar/components";
import M from "./RequestTableTooltip.vue.js";
const O = (s) => (T("data-v-0c169f89"), s = s(), N(), s), j = { class: "text-c-2 max-w-[100%] overflow-hidden filemask flex items-end justify-end" }, K = ["onClick"], L = /* @__PURE__ */ O(() => /* @__PURE__ */ p("span", null, "File", -1)), ee = /* @__PURE__ */ F({
  __name: "RequestTable",
  props: {
    items: {},
    isEnabledHidden: { type: Boolean, default: !1 },
    showUploadButton: { type: Boolean, default: !1 }
  },
  emits: ["updateRow", "toggleRow", "addRow", "deleteRow", "inputFocus", "inputBlur", "uploadFile", "removeFile"],
  setup(s, { emit: h }) {
    const B = s, a = h, _ = B.isEnabledHidden ? ["", ""] : ["", "", "36px"], b = (e, n, l) => {
      a("updateRow", e, n, l);
    }, $ = (e) => {
      a("uploadFile", e);
    }, v = (e) => !!(e.description || e.type || e.default || e.format), y = (e) => E(() => {
      if (e.type === "integer" && e.value !== void 0) {
        const n = Number(e.value);
        if (e.minimum !== void 0 && n < e.minimum)
          return `Min is ${e.minimum}`;
        if (e.maximum !== void 0 && n > e.maximum)
          return `Max is ${e.maximum}`;
      }
      return !1;
    }), R = (e) => Array.isArray(e.default) && e.default.length === 1 ? e.default[0] : e.default;
    return (e, n) => (u(), r(q, {
      class: "flex-1",
      columns: m(_)
    }, {
      default: d(() => [
        (u(!0), w(V, null, C(e.items, (l, t) => (u(), r(D, { key: t }, {
          default: d(() => [
            e.isEnabledHidden ? c("", !0) : (u(), r(z, {
              key: 0,
              modelValue: l.enabled,
              "onUpdate:modelValue": (o) => a("toggleRow", t, o)
            }, null, 8, ["modelValue", "onUpdate:modelValue"])),
            i(f, null, {
              default: d(() => [
                i(g, {
                  disableCloseBrackets: "",
                  disableEnter: "",
                  disableTabIndent: "",
                  modelValue: l.key,
                  placeholder: "Key",
                  required: l.required,
                  onBlur: n[0] || (n[0] = (o) => a("inputBlur")),
                  onFocus: n[1] || (n[1] = (o) => a("inputFocus")),
                  onInput: (o) => e.items && t === e.items.length - 1 && a("addRow"),
                  onSelectVariable: (o) => b(t, "key", o),
                  "onUpdate:modelValue": (o) => a("updateRow", t, "key", o)
                }, null, 8, ["modelValue", "required", "onInput", "onSelectVariable", "onUpdate:modelValue"])
              ]),
              _: 2
            }, 1024),
            i(f, null, {
              default: d(() => [
                i(g, {
                  class: I({
                    "pr-6": v(l)
                  }),
                  disableCloseBrackets: "",
                  disableEnter: "",
                  disableTabIndent: "",
                  enum: l.enum,
                  max: l.maximum,
                  min: l.minimum,
                  modelValue: l.value,
                  nullable: l.nullable,
                  placeholder: "Value",
                  type: l.type,
                  onBlur: n[2] || (n[2] = (o) => a("inputBlur")),
                  onFocus: n[3] || (n[3] = (o) => a("inputFocus")),
                  onInput: (o) => e.items && t === e.items.length - 1 && a("addRow"),
                  onSelectVariable: (o) => b(t, "value", o),
                  "onUpdate:modelValue": (o) => a("updateRow", t, "value", o)
                }, U({
                  icon: d(() => [
                    v(l) ? (u(), r(M, {
                      key: 0,
                      item: { ...l, default: R(l) }
                    }, null, 8, ["item"])) : c("", !0)
                  ]),
                  _: 2
                }, [
                  y(l).value ? {
                    name: "warning",
                    fn: d(() => [
                      x(k(y(l).value), 1)
                    ]),
                    key: "0"
                  } : void 0
                ]), 1032, ["class", "enum", "max", "min", "modelValue", "nullable", "type", "onInput", "onSelectVariable", "onUpdate:modelValue"])
              ]),
              _: 2
            }, 1024),
            e.showUploadButton ? (u(), r(f, {
              key: 1,
              class: "group/upload p-1 overflow-hidden relative text-ellipsis whitespace-nowrap"
            }, {
              default: d(() => {
                var o;
                return [
                  l.file ? (u(), w(V, { key: 0 }, [
                    p("div", j, [
                      p("span", null, k((o = l.file) == null ? void 0 : o.name), 1)
                    ]),
                    p("button", {
                      class: "absolute bg-b-2 font-medium centered-x centered-y hidden rounded text-center p-0.5 w-[calc(100%_-_8px)] group-hover/upload:block text-xs",
                      type: "button",
                      onClick: (S) => a("removeFile", t)
                    }, " Delete ", 8, K)
                  ], 64)) : (u(), r(m(H), {
                    key: 1,
                    class: "bg-b-2 hover:bg-b-3 border-0 py-px text-c-2 shadow-none",
                    size: "sm",
                    variant: "outlined",
                    onClick: (S) => $(t)
                  }, {
                    default: d(() => [
                      L,
                      i(m(A), {
                        class: "ml-1",
                        icon: "UploadSimple",
                        size: "xs",
                        thickness: "2.5"
                      })
                    ]),
                    _: 2
                  }, 1032, ["onClick"]))
                ];
              }),
              _: 2
            }, 1024)) : c("", !0)
          ]),
          _: 2
        }, 1024))), 128))
      ]),
      _: 1
    }, 8, ["columns"]));
  }
});
export {
  ee as default
};
