// @scalar/api-client/dist/views/Servers/ServerForm.vue.js
import { defineComponent as U, computed as l, openBlock as u, createElementBlock as p, Fragment as y, createVNode as O, createBlock as S, createCommentVNode as i } from "vue";
import m from "../../components/Form/Form.vue.js";
import { useWorkspace as V } from "../../store/workspace.js";
import { useRoute as b } from "vue-router";
const j = { class: "w-full" }, L = /* @__PURE__ */ U({
  __name: "ServerForm",
  setup(B) {
    const { activeCollection: o, servers: v, serverMutators: n } = V(), f = [
      { label: "URL", key: "url", placeholder: "https://void.scalar.com/api" },
      {
        label: "Label",
        key: "description",
        placeholder: "Production"
      }
    ], c = b(), a = l(
      () => {
        var e, t;
        return v[o.value && c.params.server === "default" ? (e = o.value) == null ? void 0 : e.spec.serverUids[0] : ((t = o.value) == null ? void 0 : t.spec.serverUids.find(
          (r) => r === c.params.server
        )) ?? ""];
      }
    ), _ = (e, t) => {
      o.value && n.edit(a.value.uid, e, t);
    }, k = (e, t) => {
      o.value && n.edit(a.value.uid, `variables.${e}.value`, t);
    }, h = l(
      () => {
        var e;
        return Object.entries(((e = a.value) == null ? void 0 : e.variables) ?? {}).map(
          ([t, r]) => {
            var s;
            return {
              key: t,
              label: t,
              placeholder: r.default ?? ((s = r == null ? void 0 : r.enum) == null ? void 0 : s[0]) ?? ""
            };
          }
        );
      }
    ), d = l(
      () => Object.entries(a.value.variables ?? {}).reduce((e, [t, r]) => {
        var s;
        return e[t] = r.value ?? r.default ?? ((s = r == null ? void 0 : r.enum) == null ? void 0 : s[0]) ?? "", e;
      }, {})
    );
    return (e, t) => (u(), p("div", j, [
      a.value ? (u(), p(y, { key: 0 }, [
        O(m, {
          data: a.value,
          onUpdate: _,
          options: f,
          title: "Server"
        }, null, 8, ["data"]),
        Object.keys(d.value).length ? (u(), S(m, {
          key: 0,
          data: d.value,
          onUpdate: k,
          options: h.value,
          title: "Variables"
        }, null, 8, ["data", "options"])) : i("", !0)
      ], 64)) : i("", !0)
    ]));
  }
});
export {
  L as default
};
