// @scalar/api-client/dist/views/Servers/Servers.vue2.js
import { defineComponent as d, openBlock as l, createBlock as s, withCtx as e, createVNode as r, unref as n, createElementVNode as p, createElementBlock as _, Fragment as v, renderList as S, createTextVNode as x } from "vue";
import b from "../../components/Sidebar/SidebarButton.vue.js";
import h from "../../components/Sidebar/SidebarList.vue.js";
import k from "../../components/Sidebar/SidebarListElement.vue.js";
import L from "../../components/SubpageHeader.vue.js";
import V from "../../components/ViewLayout/ViewLayout.vue.js";
import w from "../../components/ViewLayout/ViewLayoutContent.vue.js";
import { useWorkspace as C } from "../../store/workspace.js";
import { createServer as $ } from "@scalar/oas-utils/entities/workspace/server";
import { useRouter as y } from "vue-router";
import B from "./ServerForm.vue.js";
import E from "../../components/Sidebar/Sidebar.vue.js";
/* empty css                                    */
const N = { class: "flex-1" }, K = /* @__PURE__ */ d({
  __name: "Servers",
  setup(g) {
    const { activeCollection: t, servers: m, serverMutators: c } = C(), { push: u } = y(), f = () => {
      if (!t.value) return;
      const o = $({ url: "http://localhost" });
      c.add(o, t.value.uid), u(`/servers/${o.uid}`);
    };
    return (o, H) => (l(), s(L, null, {
      default: e(() => [
        r(V, null, {
          default: e(() => [
            r(n(E), { title: "Servers" }, {
              content: e(() => [
                p("div", N, [
                  r(h, null, {
                    default: e(() => {
                      var i;
                      return [
                        (l(!0), _(v, null, S((i = n(t)) == null ? void 0 : i.spec.serverUids, (a) => (l(), s(k, {
                          key: a,
                          class: "text-xs",
                          variable: { name: n(m)[a].url, uid: a }
                        }, null, 8, ["variable"]))), 128))
                      ];
                    }),
                    _: 1
                  })
                ])
              ]),
              button: e(() => [
                r(b, { click: f }, {
                  title: e(() => [
                    x("Add Server")
                  ]),
                  _: 1
                })
              ]),
              _: 1
            }),
            r(w, { class: "flex-1" }, {
              default: e(() => [
                r(B)
              ]),
              _: 1
            })
          ]),
          _: 1
        })
      ]),
      _: 1
    }));
  }
});
export {
  K as default
};
