// @scalar/components/dist/components/ScalarHeader/ScalarHeaderButton.vue.js
import { defineComponent as a, createBlock as i, openBlock as s, resolveDynamicComponent as u, mergeProps as c, unref as t, withCtx as l, renderSlot as d } from "vue";
import { cva as f, useBindCx as p } from "@scalar/use-hooks/useBindCx";
const m = {}, x = /* @__PURE__ */ a({
  ...m,
  inheritAttrs: !1,
  __name: "ScalarHeaderButton",
  props: {
    is: { default: "button" },
    active: { type: Boolean }
  },
  setup(e) {
    const n = f({
      base: "group/button flex items-center rounded  px-2.5 py-1.5 font-medium no-underline leading-3 ",
      variants: {
        active: {
          true: "bg-b-3 cursor-default",
          false: "bg-transparent hover:bg-b-3 cursor-pointer"
        }
      }
    }), { cx: o } = p();
    return (r, b) => (s(), i(u(e.is), c({
      type: e.is === "button" ? "button" : void 0
    }, t(o)(t(n)({ active: e.active }))), {
      default: l(() => [
        d(r.$slots, "default")
      ]),
      _: 3
    }, 16, ["type"]));
  }
});
export {
  x as default
};
