// @scalar/api-client/dist/v2/blocks/request-block/helpers/get-default-headers.js
import { canMethodHaveBody as p } from "@scalar/helpers/http/can-method-have-body";
import { getResolvedRef as n } from "@scalar/workspace-store/helpers/get-resolved-ref";
import { isElectron as u } from "../../../../libs/electron.js";
const i = "2.22.3", m = "application/json", h = "*/*", c = (r, t, o) => ({
  name: r,
  defaultValue: t,
  isOverridden: o.has(r.toLowerCase())
}), y = ({
  method: r,
  operation: t,
  exampleKey: o,
  hideDisabledHeaders: f = !1
}) => {
  const a = new Set(
    t.parameters?.filter((e) => n(e).in === "header").map((e) => n(e).name.toLowerCase()) ?? []
  ), l = t["x-scalar-disable-parameters"]?.["default-headers"]?.[o] ?? {}, s = [], d = n(t.requestBody);
  if (p(r)) {
    const e = d?.["x-scalar-selected-content-type"]?.[o] ?? Object.keys(d?.content ?? {})[0] ?? m;
    e !== "none" && s.push(c("Content-Type", e, a));
  }
  return s.push(c("Accept", h, a)), u() && i && s.push(c("User-Agent", `Scalar/${i}`, a)), f ? s.filter((e) => l[e.name.toLowerCase()] !== !0) : s;
};
export {
  y as getDefaultHeaders
};
