// @scalar/api-client/dist/views/Request/ResponseSection/ResponseEmpty.vue2.js
import { defineComponent as u, openBlock as c, createElementBlock as a, createElementVNode as e, unref as s, createTextVNode as o, createCommentVNode as n, createVNode as r, pushScopeId as _, popScopeId as f } from "vue";
import h from "../../../assets/computer.ascii.js";
import x from "../../../components/ScalarAsciiArt.vue.js";
import i from "../../../components/ScalarHotkey.vue.js";
import { useWorkspace as k } from "../../../store/workspace.js";
import { executeRequestBus as v } from "../../../libs/event-busses/execute-request-bus.js";
const p = (t) => (_("data-v-84b69d7e"), t = t(), f(), t), y = { class: "relative col-1 flex-center gap-6 p-2 capitalize" }, b = { class: "flex h-[calc(100%_-_50px)] flex-col items-center justify-center" }, R = {
  key: 0,
  class: "scalar-version-number"
}, g = /* @__PURE__ */ p(() => /* @__PURE__ */ e("b", null, "Alpha", -1)), N = /* @__PURE__ */ p(() => /* @__PURE__ */ e("div", { class: "mt-1" }, [
  /* @__PURE__ */ e("a", {
    href: "https://github.com/scalar/scalar/issues/2669",
    target: "_blank"
  }, "Roadmap")
], -1)), S = { class: "text-c-3 right-4 mt-auto flex w-full flex-col items-end gap-2 text-sm" }, V = {
  key: 0,
  class: "flex items-center gap-1.5",
  type: "button"
}, W = /* @__PURE__ */ u({
  __name: "ResponseEmpty",
  setup(t) {
    const { isReadOnly: m, activeWorkspace: d } = k();
    return (C, l) => (c(), a("div", y, [
      e("div", b, [
        s(d).isReadOnly ? n("", !0) : (c(), a("div", R, [
          o(" Scalar V1.0.6 "),
          g,
          o(" Release "),
          N
        ])),
        r(x, { art: s(h) }, null, 8, ["art"])
      ]),
      e("div", S, [
        e("button", {
          class: "flex items-center gap-1.5",
          type: "button",
          onClick: l[0] || (l[0] = (q) => s(v).emit())
        }, [
          o(" Send Request "),
          r(i, { hotkey: "↵" })
        ]),
        s(m) ? n("", !0) : (c(), a("button", V, [
          o(" New Request "),
          r(i, { hotkey: "N" })
        ]))
      ])
    ]));
  }
});
export {
  W as default
};
