// @scalar/api-client/dist/views/Request/ResponseSection/ResponseBodyDownload.vue.js
import { defineComponent as s, computed as p, openBlock as d, createElementBlock as a, withModifiers as i, createVNode as l, unref as c, createElementVNode as m } from "vue";
import { ScalarIcon as f } from "@scalar/components";
import { mediaTypes as u } from "../consts/mediaTypes.js";
const x = ["download", "href"], h = /* @__PURE__ */ m("span", null, "Download", -1), k = /* @__PURE__ */ s({
  __name: "ResponseBodyDownload",
  props: {
    href: {},
    type: {}
  },
  setup(n) {
    const t = n, r = p(
      () => {
        var e;
        return ((e = u[t.type ?? ""]) == null ? void 0 : e.extension) ?? ".unknown";
      }
    );
    return (e, o) => (d(), a("a", {
      class: "flex gap-1 text-c-3 text-xxs no-underline items-center hover:bg-b-3 rounded py-0.5 px-1.5",
      download: `response${r.value}`,
      href: e.href,
      onClick: o[0] || (o[0] = i(() => {
      }, ["stop"]))
    }, [
      l(c(f), {
        icon: "Download",
        size: "xs"
      }),
      h
    ], 8, x));
  }
});
export {
  k as default
};
