// @scalar/api-client/dist/views/Request/ResponseSection/ResponseSection.vue.js
import { defineComponent as _, computed as m, ref as h, openBlock as r, createBlock as n, withCtx as p, createVNode as u, unref as x, createElementVNode as d, createTextVNode as y, createCommentVNode as l, createElementBlock as C, Fragment as A } from "vue";
import B from "../../../components/ContextBar.vue.js";
import S from "../../../components/ViewLayout/ViewLayoutSection.vue.js";
import $ from "./ResponseBody.vue.js";
import E from "./ResponseEmpty.vue.js";
import N from "./ResponseMetaInformation.vue.js";
import { ScalarIcon as V } from "@scalar/components";
import g from "./ResponseCookies.vue.js";
import H from "./ResponseHeaders.vue.js";
const R = { class: "flex items-center flex-1" }, b = { class: "custom-scroll flex flex-1 flex-col px-2 xl:px-6 py-2.5" }, T = /* @__PURE__ */ _({
  __name: "ResponseSection",
  props: {
    response: {}
  },
  setup(f) {
    const c = f, i = m(() => {
      var s;
      const e = (s = c.response) == null ? void 0 : s.headers;
      return e ? Object.keys(e).map((o) => ({
        name: o,
        value: e[o],
        required: !1
      })).filter(
        (o) => ![
          "rest-api-client-content-length",
          "X-API-Client-Content-Length"
        ].includes(o.name)
      ) : [];
    }), v = m(() => {
      var s, o;
      const e = (o = (s = c.response) == null ? void 0 : s.headers) == null ? void 0 : o.cookies;
      return e ? Object.keys(e).map((a) => ({
        name: a,
        value: e[a],
        required: !1
      })) : [];
    }), k = ["All", "Body", "Headers", "Cookies"], t = h("All");
    return (e, s) => (r(), n(S, null, {
      title: p(() => [
        u(x(V), {
          class: "text-c-3 mr-2 rotate-180",
          icon: "ExternalLink",
          size: "sm",
          thickness: "2.5"
        }),
        d("div", R, [
          y(" Response "),
          e.response ? (r(), n(N, {
            key: 0,
            response: e.response
          }, null, 8, ["response"])) : l("", !0)
        ])
      ]),
      default: p(() => {
        var o;
        return [
          d("div", b, [
            e.response ? (r(), C(A, { key: 1 }, [
              u(B, {
                activeSection: t.value,
                sections: k,
                onSetActiveSection: s[0] || (s[0] = (a) => t.value = a)
              }, null, 8, ["activeSection"]),
              t.value === "All" || t.value === "Cookies" ? (r(), n(g, {
                key: 0,
                cookies: v.value
              }, null, 8, ["cookies"])) : l("", !0),
              t.value === "All" || t.value === "Headers" ? (r(), n(H, {
                key: 1,
                headers: i.value
              }, null, 8, ["headers"])) : l("", !0),
              t.value === "All" || t.value === "Body" ? (r(), n($, {
                key: 2,
                active: !0,
                data: (o = c.response) == null ? void 0 : o.data,
                headers: i.value,
                title: "Body"
              }, null, 8, ["data", "headers"])) : l("", !0)
            ], 64)) : (r(), n(E, { key: 0 }))
          ])
        ];
      }),
      _: 1
    }));
  }
});
export {
  T as default
};
