// @scalar/api-client/dist/views/Request/ResponseSection/ResponseBody.vue.js
import { defineComponent as U, ref as x, computed as l, openBlock as r, createBlock as u, createSlots as L, withCtx as p, createTextVNode as w, toDisplayString as g, createElementBlock as V, createElementVNode as b, createCommentVNode as i, createVNode as j } from "vue";
import T from "../../../components/ViewLayout/ViewLayoutCollapse.vue.js";
import C from "whatwg-mimetype";
import N from "./ResponseBodyDownload.vue.js";
import O from "./ResponseBodyInfo.vue.js";
import S from "./ResponseBodyPreview.vue.js";
import E from "./ResponseBodyRaw.vue.js";
import $ from "./ResponseBodyToggle.vue.js";
import { mediaTypes as I } from "../consts/mediaTypes.js";
const M = {
  key: 0,
  class: "mx-1 border-1/2 flex flex-col rounded bg-b-1"
}, P = { class: "flex justify-between items-center border-b-1/2 p-1.5" }, D = { class: "text-xxs leading-3 font-code" }, W = /* @__PURE__ */ U({
  __name: "ResponseBody",
  props: {
    title: {},
    data: {},
    headers: {}
  },
  setup(h) {
    const a = h, _ = (e) => e instanceof Blob, d = x(!0), m = l(
      () => {
        var e;
        return !!((e = t.value) != null && e.raw && t.value.preview);
      }
    ), B = l(() => d.value || !m.value), R = l(() => !d.value || !m.value), o = l(() => {
      var s;
      const e = ((s = a.headers.find((c) => c.name.toLowerCase() === "content-type")) == null ? void 0 : s.value) ?? "";
      return new C(e);
    }), t = l(() => I[o.value.essence]), n = l(() => _(a.data) ? URL.createObjectURL(a.data) : typeof a.data == "string" ? URL.createObjectURL(
      new Blob([a.data], { type: o.value.toString() })
    ) : typeof a.data == "object" && Object.keys(a.data).length ? URL.createObjectURL(
      new Blob([JSON.stringify(a.data)], {
        type: o.value.toString()
      })
    ) : "");
    return (e, s) => (r(), u(T, null, L({
      title: p(() => [
        w(g(e.title), 1)
      ]),
      default: p(() => {
        var c, v, f, y;
        return [
          e.data ? (r(), V("div", M, [
            b("div", P, [
              b("span", D, g(o.value.essence), 1),
              m.value ? (r(), u($, {
                key: 0,
                modelValue: d.value,
                "onUpdate:modelValue": s[0] || (s[0] = (k) => d.value = k)
              }, null, 8, ["modelValue"])) : i("", !0)
            ]),
            (c = t.value) != null && c.raw && R.value ? (r(), u(E, {
              key: n.value,
              data: e.data,
              language: t.value.language
            }, null, 8, ["data", "language"])) : i("", !0),
            (v = t.value) != null && v.preview && B.value ? (r(), u(S, {
              key: n.value,
              alpha: t.value.alpha,
              mode: t.value.preview,
              src: n.value,
              type: o.value.essence
            }, null, 8, ["alpha", "mode", "src", "type"])) : i("", !0),
            !((f = t.value) != null && f.raw) && !((y = t.value) != null && y.preview) ? (r(), u(O, { key: 2 }, {
              default: p(() => [
                w(" Binary file ")
              ]),
              _: 1
            })) : i("", !0)
          ])) : i("", !0)
        ];
      }),
      _: 2
    }, [
      e.data && n.value ? {
        name: "actions",
        fn: p(() => [
          j(N, {
            href: n.value,
            type: o.value.essence
          }, null, 8, ["href", "type"])
        ]),
        key: "0"
      } : void 0
    ]), 1024));
  }
});
export {
  W as default
};
