// @scalar/api-client/dist/views/Request/ResponseSection/ResponseHeaders.vue.js
import { defineComponent as c, openBlock as e, createBlock as a, withCtx as t, createTextVNode as i, createElementBlock as s, Fragment as d, renderList as f, createVNode as o } from "vue";
import u from "../../../components/DataTable/DataTable.vue.js";
import p from "../../../components/DataTable/DataTableRow.vue.js";
import m from "../../../components/DataTable/DataTableText.vue.js";
import _ from "../../../components/ViewLayout/ViewLayoutCollapse.vue.js";
const x = {
  key: 1,
  class: "text-c-3 px-4 text-sm border rounded min-h-12 justify-center flex items-center bg-b-1 mx-1"
}, B = /* @__PURE__ */ c({
  __name: "ResponseHeaders",
  props: {
    headers: {}
  },
  setup(h) {
    return (r, k) => (e(), a(_, {
      defaultOpen: !1,
      itemCount: r.headers.length
    }, {
      title: t(() => [
        i("Headers")
      ]),
      default: t(() => [
        r.headers.length ? (e(), a(u, {
          key: 0,
          columns: ["minmax(auto, min-content)", "minmax(50%, 1fr)"],
          scroll: ""
        }, {
          default: t(() => [
            (e(!0), s(d, null, f(r.headers, (n, l) => (e(), a(p, {
              key: l,
              class: "text-c-1"
            }, {
              default: t(() => [
                o(m, {
                  class: "sticky left-0 z-1 bg-b-1 max-w-48",
                  text: n.name
                }, null, 8, ["text"]),
                o(m, {
                  class: "z-0",
                  text: n.value
                }, null, 8, ["text"])
              ]),
              _: 2
            }, 1024))), 128))
          ]),
          _: 1
        })) : (e(), s("div", x, " No Headers "))
      ]),
      _: 1
    }, 8, ["itemCount"]));
  }
});
export {
  B as default
};
