// @scalar/api-client/dist/views/Request/ResponseSection/ResponseBodyPreview.vue2.js
import { defineComponent as d, ref as i, watch as y, openBlock as s, createElementBlock as t, normalizeClass as n, createElementVNode as a, createBlock as m, withCtx as v, createTextVNode as f } from "vue";
import c from "./ResponseBodyInfo.vue.js";
const k = ["src"], w = ["src", "type"], B = ["src", "type"], E = ["data", "type"], $ = /* @__PURE__ */ d({
  __name: "ResponseBodyPreview",
  props: {
    src: {},
    type: {},
    mode: {},
    alpha: { type: Boolean }
  },
  setup(p) {
    const u = p, r = i(!1);
    return y(
      () => u.src,
      () => r.value = !1
    ), (e, o) => !r.value && e.src ? (s(), t("div", {
      key: 0,
      class: n(["flex justify-center overflow-hidden rounded-b", { "p-2 bg-preview": e.alpha }])
    }, [
      e.mode === "image" ? (s(), t("img", {
        key: 0,
        class: n(["max-w-full", { rounded: e.alpha }]),
        src: e.src,
        onError: o[0] || (o[0] = (l) => r.value = !0)
      }, null, 42, k)) : e.mode === "video" ? (s(), t("video", {
        key: 1,
        autoplay: "",
        controls: "",
        loop: "",
        width: "100%",
        onError: o[1] || (o[1] = (l) => r.value = !0)
      }, [
        a("source", {
          src: e.src,
          type: e.type
        }, null, 8, w)
      ], 32)) : e.mode === "audio" ? (s(), t("audio", {
        key: 2,
        class: "my-12",
        controls: "",
        loop: "",
        onError: o[2] || (o[2] = (l) => r.value = !0)
      }, [
        a("source", {
          src: e.src,
          type: e.type
        }, null, 8, B)
      ], 32)) : (s(), t("object", {
        key: 3,
        class: "w-full aspect-[4/3]",
        data: e.src,
        type: e.type,
        onError: o[3] || (o[3] = (l) => r.value = !0)
      }, null, 40, E))
    ], 2)) : (s(), m(c, { key: 1 }, {
      default: v(() => [
        f("Preview unavailable")
      ]),
      _: 1
    }));
  }
});
export {
  $ as default
};
