// @scalar/api-client/dist/views/Request/ResponseSection/ResponseCookies.vue.js
import { defineComponent as m, openBlock as e, createBlock as o, withCtx as t, createTextVNode as f, createElementBlock as r, Fragment as a, renderList as u, createVNode as l } from "vue";
import _ from "../../../components/DataTable/DataTable.vue.js";
import p from "../../../components/DataTable/DataTableRow.vue.js";
import c from "../../../components/DataTable/DataTableText.vue.js";
import d from "../../../components/ViewLayout/ViewLayoutCollapse.vue.js";
const k = {
  key: 1,
  class: "text-c-3 px-4 text-sm border border-1/2 rounded min-h-12 justify-center flex items-center bg-b-1 mx-1"
}, N = /* @__PURE__ */ m({
  __name: "ResponseCookies",
  props: {
    cookies: {}
  },
  setup(x) {
    return (n, b) => (e(), o(d, { defaultOpen: !1 }, {
      title: t(() => [
        f("Cookies")
      ]),
      default: t(() => [
        (e(), r(a, { key: 0 }, [
          n.cookies.length ? (e(), o(_, {
            key: 0,
            class: "flex-1",
            columns: ["", ""]
          }, {
            default: t(() => [
              (e(!0), r(a, null, u(n.cookies, (s, i) => (e(), o(p, { key: i }, {
                default: t(() => [
                  l(c, {
                    text: s.name
                  }, null, 8, ["text"]),
                  l(c, {
                    text: s.value
                  }, null, 8, ["text"])
                ]),
                _: 2
              }, 1024))), 128))
            ]),
            _: 1
          })) : (e(), r("div", k, " No cookies "))
        ], 64))
      ]),
      _: 1
    }));
  }
});
export {
  N as default
};
