// @scalar/api-client/dist/views/Request/ResponseSection/ResponseCookies.vue.js
import { defineComponent as m, createBlock as n, openBlock as e, withCtx as t, createElementBlock as r, Fragment as a, renderList as f, createVNode as c, createTextVNode as u } from "vue";
import x from "../../../components/DataTable/DataTable.vue.js";
import k from "../../../components/DataTable/DataTableRow.vue.js";
import i from "../../../components/DataTable/DataTableText.vue.js";
import p from "../../../components/ViewLayout/ViewLayoutCollapse.vue.js";
const _ = {
  key: 1,
  class: "text-c-3 bg-b-1 flex min-h-[65px] items-center justify-center border-t px-4 text-sm"
}, B = /* @__PURE__ */ m({
  __name: "ResponseCookies",
  props: {
    cookies: {}
  },
  setup(s) {
    return (d, l) => (e(), n(p, { defaultOpen: !1 }, {
      title: t(() => [...l[0] || (l[0] = [
        u("Cookies", -1)
      ])]),
      default: t(() => [
        (e(), r(a, { key: 0 }, [
          s.cookies.length ? (e(), n(x, {
            key: 0,
            class: "flex-1",
            columns: ["", ""]
          }, {
            default: t(() => [
              (e(!0), r(a, null, f(s.cookies, (o) => (e(), n(k, {
                key: o.name
              }, {
                default: t(() => [
                  c(i, {
                    text: o.name
                  }, null, 8, ["text"]),
                  c(i, {
                    text: o.value
                  }, null, 8, ["text"])
                ]),
                _: 2
              }, 1024))), 128))
            ]),
            _: 1
          })) : (e(), r("div", _, " No cookies "))
        ], 64))
      ]),
      _: 1
    }));
  }
});
export {
  B as default
};
