// @scalar/api-client/dist/views/Collection/components/MarkdownInput.vue2.js
import { defineComponent as h, ref as v, watch as w, nextTick as y, createElementBlock as s, openBlock as n, createElementVNode as o, createBlock as u, createCommentVNode as a, unref as r, withCtx as f, createVNode as d, Fragment as p } from "vue";
import { ScalarButton as b, ScalarIcon as k, ScalarMarkdown as z } from "@scalar/components";
import B from "../../../components/CodeInput/CodeInput.vue.js";
const C = { class: "flex h-full w-full flex-col gap-2 pt-8" }, I = { class: "flex min-h-8 items-center justify-between gap-2 pl-1.5" }, E = { class: "has-[:focus-visible]:bg-b-1 group relative z-1 flex flex-col rounded-lg" }, N = { class: "flex h-full min-h-[calc(1rem*4)] flex-col" }, S = {
  key: 1,
  class: "text-c-3 flex h-full items-center justify-center rounded-lg border p-4"
}, F = /* @__PURE__ */ h({
  __name: "MarkdownInput",
  props: {
    modelValue: {},
    environment: {},
    envVariables: {},
    workspace: {}
  },
  emits: ["update:modelValue"],
  setup(l, { emit: x }) {
    const g = x, t = v("preview"), m = v(null);
    w(t, (c) => {
      c === "edit" && y(() => {
        m.value?.focus();
      });
    });
    const V = () => {
      requestAnimationFrame(() => {
        t.value = "preview";
      });
    };
    return (c, e) => (n(), s("div", C, [
      o("div", I, [
        e[5] || (e[5] = o("h3", { class: "font-bold" }, "Description", -1)),
        t.value === "preview" ? (n(), u(r(b), {
          key: 0,
          class: "text-c-2 hover:text-c-1 flex items-center gap-2",
          size: "sm",
          type: "button",
          variant: "outlined",
          onClick: e[0] || (e[0] = (i) => t.value = "edit")
        }, {
          default: f(() => [
            d(r(k), {
              icon: "Pencil",
              size: "sm",
              thickness: "1.5"
            }),
            e[4] || (e[4] = o("span", null, "Edit", -1))
          ]),
          _: 1
        })) : a("", !0)
      ]),
      o("div", E, [
        o("div", N, [
          t.value === "preview" ? (n(), s(p, { key: 0 }, [
            l.modelValue && l.modelValue.trim().length ? (n(), s(p, { key: 0 }, [
              l.modelValue ? (n(), u(r(z), {
                key: 0,
                class: "h-full flex-1 rounded border border-transparent p-1.5 hover:border-(--scalar-background-3)",
                value: l.modelValue,
                withImages: "",
                onDblclick: e[1] || (e[1] = (i) => t.value = "edit")
              }, null, 8, ["value"])) : a("", !0),
              e[6] || (e[6] = o("div", { class: "brightness-lifted bg-b-1 absolute inset-0 -z-1 hidden rounded group-hover:block group-has-[:focus-visible]:hidden" }, null, -1))
            ], 64)) : (n(), s("div", S, [
              d(r(b), {
                class: "hover:bg-b-2 hover:text-c-1 text-c-2 flex items-center gap-2",
                size: "sm",
                variant: "ghost",
                onClick: e[2] || (e[2] = (i) => t.value = "edit")
              }, {
                default: f(() => [
                  d(r(k), {
                    icon: "Pencil",
                    size: "sm",
                    thickness: "1.5"
                  }),
                  e[7] || (e[7] = o("span", null, "Write a description", -1))
                ]),
                _: 1
              })
            ]))
          ], 64)) : a("", !0),
          t.value === "edit" ? (n(), u(B, {
            key: 1,
            ref_key: "codeInputRef",
            ref: m,
            class: "h-full flex-1 border px-0.5 py-0",
            envVariables: l.envVariables,
            environment: l.environment,
            modelValue: l.modelValue,
            workspace: l.workspace,
            onBlur: V,
            "onUpdate:modelValue": e[3] || (e[3] = (i) => g("update:modelValue", i))
          }, null, 8, ["envVariables", "environment", "modelValue", "workspace"])) : a("", !0)
        ])
      ])
    ]));
  }
});
export {
  F as default
};
