// @scalar/api-reference/dist/components/Anchor/Anchor.vue2.js
import { defineComponent as p, openBlock as a, createElementBlock as c, renderSlot as s, createElementVNode as r, withModifiers as i, unref as u, createTextVNode as n, createVNode as h, withCtx as f, pushScopeId as m, popScopeId as _ } from "vue";
import w from "../ScreenReader.vue.js";
import { useClipboard as C } from "../../hooks/useClipboard.js";
const S = (t) => (m("data-v-d9801d4e"), t = t(), _(), t), b = { class: "label" }, k = { class: "anchor" }, v = /* @__PURE__ */ S(() => /* @__PURE__ */ r("span", null, "​", -1)), $ = /* @__PURE__ */ p({
  __name: "Anchor",
  props: {
    id: {}
  },
  setup(t) {
    const { copyToClipboard: d } = C(), l = (o) => {
      const e = new URL(window.location.href);
      return e.hash = o, e.toString();
    };
    return (o, e) => (a(), c("span", b, [
      s(o.$slots, "default", {}, void 0, !0),
      r("span", k, [
        v,
        r("button", {
          class: "anchor-copy",
          type: "button",
          onClick: e[0] || (e[0] = i((y) => u(d)(l(o.id)), ["stop"]))
        }, [
          n(" # "),
          h(w, null, {
            default: f(() => [
              n('Copy link to "'),
              s(o.$slots, "default", {}, void 0, !0),
              n('"')
            ]),
            _: 3
          })
        ])
      ])
    ]));
  }
});
export {
  $ as default
};
