// @scalar/oas-utils/dist/spec-getters/getRequestFromOperation.js
import { getParametersFromOperation } from './getParametersFromOperation.js';
import { getRequestBodyFromOperation } from './getRequestBodyFromOperation.js';

const getRequestFromOperation = (operation, options, selectedExampleKey) => {
    // Replace all variables of the format {something} with the uppercase variable name without the brackets
    let path = operation.path;
    // {id} -> 123
    const pathParameters = getParametersFromOperation(operation, 'path', false);
    if (pathParameters.length) {
        const pathVariables = path.match(/{(.*?)}/g);
        if (pathVariables) {
            pathVariables.forEach((variable) => {
                const variableName = variable.replace(/{|}/g, '');
                if (pathParameters) {
                    const parameter = pathParameters.find((param) => param.name === variableName);
                    if (parameter?.value) {
                        path = path.replace(variable, parameter.value.toString());
                    }
                }
            });
        }
    }
    // {id} -> __ID__
    if (options?.replaceVariables === true) {
        const pathVariables = path.match(/{(.*?)}/g);
        if (pathVariables) {
            pathVariables.forEach((variable) => {
                const variableName = variable.replace(/{|}/g, '');
                path = path.replace(variable, `__${variableName.toUpperCase()}__`);
            });
        }
    }
    const requestBody = getRequestBodyFromOperation(operation, selectedExampleKey);
    return {
        method: operation.httpVerb.toUpperCase(),
        path,
        headers: [
            ...getParametersFromOperation(operation, 'header', options?.requiredOnly),
            ...(requestBody?.headers ?? []),
        ],
        // @ts-expect-error Sorry, something is off here and I don’t get it.
        postData: requestBody?.postData,
        queryString: getParametersFromOperation(operation, 'query', options?.requiredOnly),
        cookies: getParametersFromOperation(operation, 'cookie', options?.requiredOnly),
    };
};

export { getRequestFromOperation };
