// @scalar/api-reference/dist/components/Content/Tags/components/ClassicLayout.vue2.js
import { defineComponent, createBlock, openBlock, unref, withCtx, renderSlot, createVNode, createTextVNode, toDisplayString } from "vue";
import { ScalarMarkdown } from "@scalar/components";
import SectionContainerAccordion from "../../../Section/SectionContainerAccordion.vue.js";
import SectionHeader from "../../../Section/SectionHeader.vue.js";
import _sfc_main$1 from "../../../Anchor/Anchor.vue.js";
import SectionHeaderTag from "../../../Section/SectionHeaderTag.vue.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "ClassicLayout",
  props: {
    tag: {},
    isCollapsed: { type: Boolean },
    eventBus: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createBlock(unref(SectionContainerAccordion), {
        "aria-label": __props.tag.title,
        class: "tag-section",
        modelValue: !__props.isCollapsed,
        "onUpdate:modelValue": _cache[1] || (_cache[1] = (value) => __props.eventBus?.emit("toggle:nav-item", { id: __props.tag.id, open: value }))
      }, {
        title: withCtx(() => [
          createVNode(unref(SectionHeader), { class: "tag-name" }, {
            default: withCtx(() => [
              createVNode(unref(_sfc_main$1), {
                onCopyAnchorUrl: _cache[0] || (_cache[0] = () => __props.eventBus?.emit("copy-url:nav-item", { id: __props.tag.id }))
              }, {
                default: withCtx(() => [
                  createVNode(unref(SectionHeaderTag), { level: 2 }, {
                    default: withCtx(() => [
                      createTextVNode(toDisplayString(__props.tag.title), 1)
                    ]),
                    _: 1
                  })
                ]),
                _: 1
              })
            ]),
            _: 1
          }),
          createVNode(unref(ScalarMarkdown), {
            class: "tag-description",
            value: __props.tag?.description,
            withImages: ""
          }, null, 8, ["value"])
        ]),
        default: withCtx(() => [
          renderSlot(_ctx.$slots, "default", {}, void 0, true)
        ]),
        _: 3
      }, 8, ["aria-label", "modelValue"]);
    };
  }
});
export {
  _sfc_main as default
};
