// @scalar/api-client/dist/components/Server/ServerVariablesSelect.vue.js
import { defineComponent as f, computed as s, createBlock as v, openBlock as c, unref as a, withCtx as i, createVNode as u, createElementVNode as g, normalizeClass as x, createElementBlock as S, createCommentVNode as V, createTextVNode as b, toDisplayString as h } from "vue";
import { ScalarListbox as C, ScalarButton as k } from "@scalar/components";
import { ScalarIconCaretDown as B } from "@scalar/icons";
const N = {
  key: 0,
  class: "sr-only"
}, D = /* @__PURE__ */ f({
  __name: "ServerVariablesSelect",
  props: {
    enum: {},
    value: {},
    controls: {}
  },
  emits: ["change"],
  setup(t, { emit: m }) {
    const o = t, d = m, n = s(
      () => o.enum.map((e) => ({ id: e, label: e }))
    ), l = s({
      get: () => n.value.find((e) => e.id === o.value),
      set: (e) => d("change", e?.id ?? "")
    });
    return (e, r) => (c(), v(a(C), {
      modelValue: l.value,
      "onUpdate:modelValue": r[0] || (r[0] = (p) => l.value = p),
      options: n.value
    }, {
      default: i(() => [
        u(a(k), {
          "aria-controls": t.controls,
          class: "group/button h-8 gap-1.5 p-1.5 text-base font-normal",
          variant: "ghost"
        }, {
          default: i(() => [
            g("span", {
              class: x({ "text-c-1": t.value })
            }, [
              t.value ? (c(), S("span", N, " Selected: ")) : V("", !0),
              b(" " + h(t.value || "Select value"), 1)
            ], 2),
            u(a(B), {
              weight: "bold",
              class: "mt-0.25 size-3 transition-transform duration-100 group-aria-expanded/button:rotate-180"
            })
          ]),
          _: 1
        }, 8, ["aria-controls"])
      ]),
      _: 1
    }, 8, ["modelValue", "options"]));
  }
});
export {
  D as default
};
