// @scalar/api-client/dist/components/Server/ServerDropdown.vue.js
import { defineComponent as V, computed as u, watch as z, useTemplateRef as E, createBlock as m, openBlock as o, unref as n, withCtx as v, createVNode as c, createElementBlock as a, Fragment as d, createElementVNode as s, createTextVNode as L, toDisplayString as T, createCommentVNode as S, renderList as g } from "vue";
import { ScalarPopover as W, ScalarButton as q, ScalarIcon as h, ScalarFloatingBackdrop as F, ScalarDropdownDivider as I } from "@scalar/components";
import { emitCustomEvent as M } from "@scalar/workspace-store/events";
import { useLayout as O } from "../../hooks/useLayout.js";
import { useWorkspace as P } from "../../store/store.js";
import k from "./ServerDropdownItem.vue.js";
const $ = ["onClick"], j = { class: "flex h-4 w-4 items-center justify-center" }, X = /* @__PURE__ */ V({
  __name: "ServerDropdown",
  props: {
    collection: {},
    operation: {},
    server: {},
    target: {}
  },
  setup(e) {
    const { layout: w } = O(), { servers: f, collectionMutators: y, events: U, serverMutators: C } = P(), p = u(
      () => e.operation?.servers?.map((r) => ({
        id: r,
        label: f[r]?.url ?? "Unknown server"
      }))
    ), x = u(
      () => (
        // Filters out servers already present in the request
        e.collection?.servers?.filter((r) => !e.operation?.servers?.includes(r)).map((r) => ({
          id: r,
          label: f[r]?.url ?? "Unknown server"
        }))
      )
    ), A = u(
      () => p.value?.length && x.value?.length
    );
    z([() => e.collection, () => e.operation], ([r, t]) => {
      if (!r || r.selectedServerUid || t?.selectedServerUid)
        return;
      const l = e.collection.servers?.[0];
      l && y.edit(e.collection.uid, "selectedServerUid", l);
    });
    const D = () => U.commandPalette.emit({
      commandName: "Add Server"
    }), B = u(() => e.server?.url?.endsWith("/") ? e.server.url.slice(0, -1) : e.server?.url || ""), b = (r, t) => {
      if (!e.server)
        return;
      const l = e.server.variables || {};
      l[r] = { ...l[r], default: t }, C.edit(e.server.uid, "variables", l), M(
        N.value?.$el,
        "scalar-update-selected-server-variables",
        { key: r, value: t }
      );
    }, N = E("wrapper-ref");
    return (r, t) => (o(), m(n(W), {
      ref: "wrapper-ref",
      class: "max-h-[inherit] p-0 text-base",
      focus: "",
      offset: 0,
      placement: "bottom-start",
      resize: "",
      target: e.target,
      teleport: `#${e.target}`
    }, {
      popover: v(({ close: l }) => [
        s("div", {
          class: "custom-scroll flex max-h-[inherit] flex-col gap-1 border-t p-1",
          onClick: l
        }, [
          (o(!0), a(d, null, g(p.value, (i) => (o(), m(k, {
            key: i.id,
            collection: e.collection,
            operation: e.operation,
            server: e.server,
            serverOption: i,
            type: "request",
            "onUpdate:variable": b
          }, null, 8, ["collection", "operation", "server", "serverOption"]))), 128)),
          A.value ? (o(), a(d, { key: 0 }, [
            c(n(I)),
            t[2] || (t[2] = s("div", { class: "text-xxs text-c-2 px-2.5 py-1" }, "Collection", -1))
          ], 64)) : S("", !0),
          (o(!0), a(d, null, g(x.value, (i) => (o(), m(k, {
            key: i.id,
            collection: e.collection,
            operation: e.operation,
            server: e.server,
            serverOption: i,
            type: "collection",
            "onUpdate:variable": b
          }, null, 8, ["collection", "operation", "server", "serverOption"]))), 128)),
          n(w) !== "modal" ? (o(), a("button", {
            key: 1,
            class: "text-xxs hover:bg-b-2 flex cursor-pointer items-center gap-1.5 rounded p-1.75",
            type: "button",
            onClick: D
          }, [
            s("div", j, [
              c(n(h), {
                icon: "Add",
                size: "sm"
              })
            ]),
            t[3] || (t[3] = s("span", null, "Add Server", -1))
          ])) : S("", !0)
        ], 8, $)
      ]),
      backdrop: v(() => [
        c(n(F), { class: "-top-(--scalar-address-bar-height) rounded-lg" })
      ]),
      default: v(() => [
        c(n(q), {
          class: "z-context-plus hover:bg-b-2 font-code text-c-2 ml-0.75 h-auto gap-0.75 rounded border px-1.5 text-base whitespace-nowrap",
          variant: "ghost"
        }, {
          default: v(() => [
            e.operation?.selectedServerUid || e.collection.selectedServerUid ? (o(), a(d, { key: 0 }, [
              t[0] || (t[0] = s("span", { class: "sr-only" }, "Server:", -1)),
              L(" " + T(B.value), 1)
            ], 64)) : (o(), a(d, { key: 1 }, [
              t[1] || (t[1] = s("span", { class: "sr-only" }, "Add Server", -1)),
              c(n(h), {
                icon: "Add",
                size: "xs"
              })
            ], 64))
          ]),
          _: 1
        })
      ]),
      _: 1
    }, 8, ["target", "teleport"]));
  }
});
export {
  X as default
};
