// @scalar/workspace-store/dist/mutators/document.js
import { mergeObjects } from "../helpers/merge-object.js";
const toggleSecurity = (document) => {
  if (!document) {
    return;
  }
  document["x-scalar-set-operation-security"] = !document["x-scalar-set-operation-security"];
};
const updateWatchMode = (document, watchMode) => {
  if (!document) {
    return;
  }
  document["x-scalar-watch-mode"] = watchMode;
};
const updateDocumentInfo = (document, payload) => {
  if (!document) {
    return;
  }
  mergeObjects(document.info, payload);
  if (payload.title && document["x-scalar-navigation"]) {
    document["x-scalar-navigation"].title = payload.title;
  }
};
const updateDocumentIcon = (document, icon) => {
  if (!document || !document["x-scalar-navigation"]) {
    return;
  }
  document["x-scalar-icon"] = icon;
  document["x-scalar-navigation"].icon = icon;
};
const createEmptyDocument = async (store, payload) => {
  if (!store) {
    return;
  }
  if (store.workspace.documents[payload.name]) {
    payload.callback?.(false);
    return;
  }
  await store.addDocument({
    name: payload.name,
    document: {
      openapi: "3.1.0",
      info: { title: payload.name, version: "1.0.0" },
      paths: {
        "/": {
          get: {}
        }
      },
      "x-scalar-icon": payload.icon
    }
  });
  payload.callback?.(true);
};
const deleteDocument = (store, payload) => {
  if (!store) {
    return;
  }
  store.deleteDocument(payload.name);
};
const documentMutatorsFactory = ({
  document,
  store
}) => {
  return {
    toggleSecurity: () => toggleSecurity(document),
    updateDocumentInfo: (payload) => updateDocumentInfo(document, payload),
    updateWatchMode: (payload) => updateWatchMode(document, payload),
    updateDocumentIcon: (payload) => updateDocumentIcon(document, payload),
    createEmptyDocument: (payload) => createEmptyDocument(store, payload),
    deleteDocument: (payload) => deleteDocument(store, payload)
  };
};
export {
  createEmptyDocument,
  deleteDocument,
  documentMutatorsFactory,
  toggleSecurity,
  updateDocumentIcon,
  updateDocumentInfo,
  updateWatchMode
};
