// @scalar/workspace-store/dist/mutators/har-to-operation.js
import { getResolvedRef } from "@scalar/workspace-store/helpers/get-resolved-ref";
import { isContentTypeParameterObject } from "../schemas/v3.1/strict/type-guards.js";
const preprocessParameters = (parameters, pathVariables, exampleKey) => {
  parameters.forEach((param) => {
    const resolvedParam = getResolvedRef(param);
    if (isContentTypeParameterObject(resolvedParam)) {
      return;
    }
    setParameterDisabled(getResolvedRef(param), exampleKey, true);
    if (resolvedParam.in === "path") {
      resolvedParam.examples ||= {};
      resolvedParam.examples[exampleKey] = {
        value: pathVariables[resolvedParam.name] ?? "",
        "x-disabled": false
      };
    }
  });
};
const harToOperation = ({
  harRequest,
  exampleKey,
  baseOperation = {},
  pathVariables = {}
}) => {
  if (!baseOperation.parameters) {
    baseOperation.parameters = [];
  }
  preprocessParameters(baseOperation.parameters, pathVariables, exampleKey);
  if (harRequest.queryString && harRequest.queryString.length > 0) {
    for (const queryParam of harRequest.queryString) {
      const param = findOrCreateParameter(baseOperation.parameters, queryParam.name, "query");
      if (!param || isContentTypeParameterObject(param)) {
        continue;
      }
      param.examples ||= {};
      param.examples[exampleKey] = {
        value: queryParam.value,
        "x-disabled": false
      };
    }
  }
  if (harRequest.headers && harRequest.headers.length > 0) {
    for (const header of harRequest.headers) {
      const param = findOrCreateParameter(baseOperation.parameters, header.name, "header");
      if (!param || isContentTypeParameterObject(param)) {
        continue;
      }
      param.examples ||= {};
      param.examples[exampleKey] = {
        value: header.value,
        "x-disabled": false
      };
    }
  }
  if (harRequest.cookies && harRequest.cookies.length > 0) {
    for (const cookie of harRequest.cookies) {
      const param = findOrCreateParameter(baseOperation.parameters, cookie.name, "cookie");
      if (!param || isContentTypeParameterObject(param)) {
        continue;
      }
      param.examples ||= {};
      param.examples[exampleKey] = {
        value: cookie.value,
        "x-disabled": false
      };
    }
  }
  if (harRequest.postData) {
    const { mimeType, text, params } = harRequest.postData;
    if (!baseOperation.requestBody) {
      baseOperation.requestBody = {
        content: {}
      };
    }
    const requestBody = getResolvedRef(baseOperation.requestBody);
    if (!requestBody.content[mimeType]) {
      requestBody.content[mimeType] = {
        schema: {
          type: "object"
        }
      };
    }
    const mediaType = requestBody.content[mimeType];
    if (!mediaType) {
      return baseOperation;
    }
    mediaType.examples ||= {};
    let exampleValue;
    if (params && params.length > 0) {
      exampleValue = [];
      for (const param of params) {
        exampleValue.push({
          name: param.name,
          value: param.value,
          "x-disabled": false
        });
      }
    } else {
      exampleValue = text;
    }
    mediaType.examples[exampleKey] = {
      value: exampleValue,
      "x-disabled": false
    };
    requestBody["x-scalar-selected-content-type"] ||= {};
    requestBody["x-scalar-selected-content-type"][exampleKey] = mimeType;
  }
  return baseOperation;
};
const setParameterDisabled = (param, exampleKey, disabled) => {
  if (isContentTypeParameterObject(param)) {
    return;
  }
  if (!param.examples?.[exampleKey]) {
    return;
  }
  getResolvedRef(param.examples[exampleKey])["x-disabled"] = disabled;
};
const findOrCreateParameter = (parameters, name, inValue) => {
  for (const param of parameters) {
    const resolved = getResolvedRef(param);
    if (isContentTypeParameterObject(resolved)) {
      continue;
    }
    if (resolved.in !== inValue) {
      continue;
    }
    const namesMatch = inValue === "header" ? resolved.name.toLowerCase() === name.toLowerCase() : resolved.name === name;
    if (namesMatch) {
      return resolved;
    }
  }
  const newParam = {
    name,
    in: inValue,
    schema: {
      type: "string"
    }
  };
  parameters.push(newParam);
  return newParam;
};
export {
  harToOperation
};
