// @scalar/workspace-store/dist/mutators/index.js
import { authMutatorsFactory } from "../mutators/auth.js";
import { cookieMutatorsFactory } from "../mutators/cookie.js";
import { documentMutatorsFactory } from "../mutators/document.js";
import { environmentMutatorsFactory } from "../mutators/environment.js";
import { getDocument } from "../mutators/helpers.js";
import { operationMutatorsFactory } from "../mutators/operation.js";
import { serverMutatorsFactory } from "../mutators/server.js";
import { tabsMutatorsFactory } from "../mutators/tabs.js";
import { tagMutatorsFactory } from "../mutators/tag.js";
import { workspaceMutatorsFactory } from "../mutators/workspace.js";
function generateClientMutators(store) {
  const documentMutators = (document) => {
    return {
      auth: authMutatorsFactory({ document }),
      cookie: cookieMutatorsFactory({ collection: document }),
      document: documentMutatorsFactory({ document, store }),
      operation: operationMutatorsFactory({ document, store }),
      server: serverMutatorsFactory({ document }),
      tag: tagMutatorsFactory({ store }),
      environment: environmentMutatorsFactory({ workspace: store?.workspace ?? null, collection: document })
    };
  };
  const workspaceMutators = () => {
    return {
      cookie: cookieMutatorsFactory({ collection: store?.workspace ?? null }),
      tabs: tabsMutatorsFactory({ workspace: store?.workspace ?? null }),
      workspace: workspaceMutatorsFactory({ workspace: store?.workspace ?? null }),
      environment: environmentMutatorsFactory({
        workspace: store?.workspace ?? null,
        collection: store?.workspace ?? null
      })
    };
  };
  return {
    /**
     * Returns mutators for the workspace-level configuration.
     */
    workspace: () => workspaceMutators(),
    /**
     * Returns mutators for the currently active document.
     * Falls back to the first document if no active document is set.
     */
    active: () => documentMutators(store?.workspace.activeDocument ?? null),
    /**
     * Returns mutators for a specific document by name.
     *
     * @param name - The name of the document
     */
    doc: (name) => documentMutators(getDocument(store, name))
  };
}
export {
  generateClientMutators
};
