// @scalar/workspace-store/dist/mutators/server.js
import { findVariables } from "@scalar/helpers/regex/find-variables";
import { coerceValue } from "../schemas/typebox-coerce.js";
import { ServerObjectSchema } from "../schemas/v3.1/strict/openapi-document.js";
const addServer = (document) => {
  if (!document) {
    return void 0;
  }
  const parsed = coerceValue(ServerObjectSchema, {});
  if (!document.servers) {
    document.servers = [];
  }
  document.servers.push(parsed);
  return parsed;
};
const getVariablePositions = (url, variables) => {
  const positions = {};
  for (const varName of variables) {
    const position = url.indexOf(`{${varName}}`);
    if (position !== -1) {
      positions[varName] = position;
    }
  }
  return positions;
};
const syncVariablesForUrlChange = (newUrl, oldUrl, existingVariables) => {
  const oldVariables = findVariables(oldUrl, { includePath: true, includeEnv: false }).filter(
    (v) => v !== void 0
  );
  const newVariables = findVariables(newUrl, { includePath: true, includeEnv: false }).filter(
    (v) => v !== void 0
  );
  const oldPositions = getVariablePositions(oldUrl, oldVariables);
  const newPositions = getVariablePositions(newUrl, newVariables);
  const usedOldVariables = /* @__PURE__ */ new Set();
  const syncedVariables = {};
  for (const newVar of newVariables) {
    if (existingVariables[newVar]) {
      syncedVariables[newVar] = existingVariables[newVar];
      usedOldVariables.add(newVar);
      continue;
    }
    const newVarPosition = newPositions[newVar];
    const oldVarAtPosition = oldVariables.find(
      (oldVar) => oldPositions[oldVar] === newVarPosition && !usedOldVariables.has(oldVar)
    );
    if (oldVarAtPosition && existingVariables[oldVarAtPosition]) {
      syncedVariables[newVar] = existingVariables[oldVarAtPosition];
      usedOldVariables.add(oldVarAtPosition);
      continue;
    }
    syncedVariables[newVar] = { default: "" };
  }
  return syncedVariables;
};
const updateServer = (document, { index, server }) => {
  const oldServer = document?.servers?.[index];
  if (!oldServer) {
    console.error("Server not found at index:", index);
    return void 0;
  }
  const oldUrl = oldServer.url;
  const updatedServer = coerceValue(ServerObjectSchema, { ...oldServer, ...server });
  const hasUrlChanged = oldUrl && oldUrl !== updatedServer.url;
  if (hasUrlChanged) {
    const existingVariables = updatedServer.variables ?? {};
    updatedServer.variables = syncVariablesForUrlChange(updatedServer.url, oldUrl, existingVariables);
    if (document["x-scalar-selected-server"] === oldUrl) {
      document["x-scalar-selected-server"] = updatedServer.url;
    }
  }
  if (!document.servers) {
    document.servers = [updatedServer];
  } else {
    document.servers[index] = updatedServer;
  }
  return updatedServer;
};
const deleteServer = (document, { index }) => {
  if (!document?.servers) {
    return;
  }
  const url = document.servers[index]?.url;
  document.servers.splice(index, 1);
  if (document["x-scalar-selected-server"] === url) {
    document["x-scalar-selected-server"] = document.servers[0]?.url ?? void 0;
  }
};
const updateServerVariables = (document, { index, key, value }) => {
  const variable = document?.servers?.[index]?.variables?.[key];
  if (!variable) {
    console.error("Variable not found", key, index);
    return;
  }
  variable.default = value;
  const url = document?.servers?.[index]?.url;
  if (!url) {
    console.error("URL not found", index);
    return;
  }
  return variable;
};
const updateSelectedServer = (document, { url }) => {
  if (!document) {
    return;
  }
  if (url === "") {
    document["x-scalar-selected-server"] = "";
    return "";
  }
  if (!document.servers?.some((server) => server.url === url)) {
    return;
  }
  document["x-scalar-selected-server"] = document["x-scalar-selected-server"] === url ? "" : url;
  return document["x-scalar-selected-server"];
};
const serverMutatorsFactory = ({ document }) => {
  return {
    addServer: () => addServer(document),
    updateServer: (payload) => updateServer(document, payload),
    deleteServer: (payload) => deleteServer(document, payload),
    updateServerVariables: (payload) => updateServerVariables(document, payload),
    updateSelectedServer: (payload) => updateSelectedServer(document, payload)
  };
};
export {
  addServer,
  deleteServer,
  serverMutatorsFactory,
  updateSelectedServer,
  updateServer,
  updateServerVariables
};
