// @scalar/workspace-store/dist/mutators/tabs.js
import { unpackProxyObject } from "../helpers/unpack-proxy.js";
const TAB_UNPACK_DEPTH = 1;
const hasValidTabs = (workspace) => {
  return workspace !== null && workspace["x-scalar-tabs"] !== void 0;
};
const getUnpackedTabs = (workspace) => {
  return unpackProxyObject(workspace["x-scalar-tabs"], { depth: TAB_UNPACK_DEPTH });
};
const getActiveIndex = (workspace) => {
  return workspace["x-scalar-active-tab"] ?? 0;
};
const updateTabs = (workspace, payload) => {
  if (!workspace) {
    return;
  }
  if (payload["x-scalar-tabs"]) {
    workspace["x-scalar-tabs"] = payload["x-scalar-tabs"];
  }
  if (payload["x-scalar-active-tab"] !== void 0) {
    workspace["x-scalar-active-tab"] = payload["x-scalar-active-tab"];
  }
};
const addTab = (workspace, _payload) => {
  if (!hasValidTabs(workspace)) {
    return false;
  }
  const tabs = getUnpackedTabs(workspace);
  const activeIndex = getActiveIndex(workspace);
  const currentTab = tabs[activeIndex];
  if (!currentTab) {
    return false;
  }
  const newTabs = [...tabs, { ...currentTab }];
  workspace["x-scalar-tabs"] = newTabs;
  workspace["x-scalar-active-tab"] = newTabs.length - 1;
  return true;
};
const closeTab = (workspace, payload) => {
  const getInputIndex = () => {
    if ("event" in payload) {
      return payload.event.code.startsWith("Digit") ? Number.parseInt(payload.event.key, 10) - 1 : Number.NaN;
    }
    return payload.index;
  };
  if (!hasValidTabs(workspace)) {
    return false;
  }
  const index = getInputIndex();
  const filteredTabs = getUnpackedTabs(workspace).filter((_, i) => i !== index);
  if (filteredTabs.length <= 0) {
    return false;
  }
  workspace["x-scalar-tabs"] = filteredTabs;
  if (index >= filteredTabs.length) {
    workspace["x-scalar-active-tab"] = filteredTabs.length - 1;
  }
  return true;
};
const closeOtherTabs = (workspace, payload) => {
  if (!hasValidTabs(workspace)) {
    return false;
  }
  const tabs = getUnpackedTabs(workspace);
  if (tabs.length <= 1) {
    return false;
  }
  workspace["x-scalar-tabs"] = tabs.filter((_, index) => index === payload.index);
  workspace["x-scalar-active-tab"] = 0;
  return true;
};
const navigatePreviousTab = (workspace, _payload) => {
  if (!hasValidTabs(workspace)) {
    return false;
  }
  const activeIndex = getActiveIndex(workspace);
  if (activeIndex <= 0) {
    return false;
  }
  workspace["x-scalar-active-tab"] = activeIndex - 1;
  return true;
};
const navigateNextTab = (workspace, _payload) => {
  if (!hasValidTabs(workspace)) {
    return false;
  }
  const tabs = getUnpackedTabs(workspace);
  const activeIndex = getActiveIndex(workspace);
  if (activeIndex >= tabs.length - 1) {
    return false;
  }
  workspace["x-scalar-active-tab"] = activeIndex + 1;
  return true;
};
const focusTab = (workspace, payload) => {
  if (!hasValidTabs(workspace)) {
    return false;
  }
  const getInputIndex = () => {
    if ("event" in payload) {
      return payload.event.code.startsWith("Digit") ? Number.parseInt(payload.event.key, 10) - 1 : Number.NaN;
    }
    return payload.index;
  };
  const tabs = getUnpackedTabs(workspace);
  const newActiveIndex = getInputIndex();
  if (Number.isNaN(newActiveIndex) || newActiveIndex < 0 || newActiveIndex >= tabs.length) {
    return false;
  }
  workspace["x-scalar-active-tab"] = newActiveIndex;
  return true;
};
const focusLastTab = (workspace, _payload) => {
  if (!hasValidTabs(workspace)) {
    return false;
  }
  const tabs = getUnpackedTabs(workspace);
  if (tabs.length <= 1) {
    return false;
  }
  workspace["x-scalar-active-tab"] = tabs.length - 1;
  return true;
};
const tabsMutatorsFactory = ({ workspace }) => {
  return {
    updateTabs: (payload) => updateTabs(workspace, payload),
    addTab: (payload) => addTab(workspace, payload),
    closeTab: (payload) => closeTab(workspace, payload),
    closeOtherTabs: (payload) => closeOtherTabs(workspace, payload),
    navigatePreviousTab: (payload) => navigatePreviousTab(workspace, payload),
    navigateNextTab: (payload) => navigateNextTab(workspace, payload),
    focusTab: (payload) => focusTab(workspace, payload),
    focusLastTab: (payload) => focusLastTab(workspace, payload)
  };
};
export {
  addTab,
  closeOtherTabs,
  closeTab,
  focusLastTab,
  focusTab,
  navigateNextTab,
  navigatePreviousTab,
  tabsMutatorsFactory,
  updateTabs
};
