// @scalar/use-codemirror/dist/themes/index.js
import { tags as a } from "@lezer/highlight";
import { createTheme as r } from "@uiw/codemirror-themes";
const c = r({
  theme: "light",
  settings: {
    background: "var(--scalar-background-2)",
    foreground: "var(--scalar-color-1)",
    caret: "var(--scalar-color-1)",
    // Selection likely needs a hardcoded color due to it not accepting variables
    selection: "rgba(151, 183, 205, 0.2)",
    selectionMatch: "#e3dcce",
    gutterBackground: "var(--scalar-background-2)",
    gutterForeground: "var(--scalar-color-3)",
    gutterBorder: "transparent",
    lineHighlight: "var(--scalar-background-3)",
    fontFamily: "var(--scalar-font-code)"
  },
  styles: [
    {
      tag: [a.standard(a.tagName), a.tagName],
      color: "var(--scalar-color-purple)"
    },
    {
      tag: [a.comment],
      color: "var(--scalar-color-3)"
    },
    {
      tag: [a.className],
      color: "var(--scalar-color-orange)"
    },
    {
      tag: [a.variableName, a.propertyName, a.attributeName],
      color: "var(--scalar-color-1)"
    },
    {
      tag: [a.operator],
      color: "var(--scalar-color-2)"
    },
    {
      tag: [a.keyword, a.typeName, a.typeOperator],
      color: "var(--scalar-color-green)"
    },
    {
      tag: [a.string],
      color: "var(--scalar-color-blue)"
    },
    {
      tag: [a.bracket, a.regexp, a.meta],
      color: "var(--scalar-color-3)"
    },
    {
      tag: [a.number],
      color: "var(--scalar-color-blue)"
    },
    {
      tag: [a.name, a.quote],
      color: "var(--scalar-color-3)"
    },
    {
      tag: [a.heading],
      color: "var(--scalar-color-3)",
      fontWeight: "bold"
    },
    {
      tag: [a.emphasis],
      color: "var(--scalar-color-3)",
      fontStyle: "italic"
    },
    {
      tag: [a.deleted],
      color: "var(--scalar-color-3)",
      backgroundColor: "transparent"
    },
    {
      tag: [a.atom, a.bool, a.special(a.variableName)],
      color: "var(--scalar-color-3)"
    },
    {
      tag: [a.url, a.escape, a.regexp, a.link],
      color: "var(--scalar-color-1)"
    },
    { tag: a.link, textDecoration: "underline" },
    { tag: a.strikethrough, textDecoration: "line-through" },
    {
      tag: a.invalid,
      color: "var(--scalar-color-3)"
    }
  ]
});
export {
  c as customTheme
};
