// @scalar/api-reference/dist/components/Content/Operation/PathResponses/ExampleResponse.vue2.js
import { defineComponent as D, openBlock as o, createBlock as a, unref as s, createElementBlock as t, Fragment as c, renderList as v, createElementVNode as C, toDisplayString as J, createVNode as L, createCommentVNode as P } from "vue";
import { ScalarCodeBlock as r } from "@scalar/components";
import { prettyPrintJson as V } from "@scalar/oas-utils/helpers";
import { getExampleFromSchema as l } from "@scalar/oas-utils/spec-getters";
const R = {
  key: 0,
  class: "rule"
}, q = { class: "rule-title" }, w = { class: "rule-items" }, z = {
  key: 2,
  class: "empty-state"
}, Q = /* @__PURE__ */ D({
  __name: "ExampleResponse",
  props: {
    response: {}
  },
  setup(G) {
    const F = ["oneOf", "anyOf", "not"], p = (e) => e.reduce((i, m) => ({
      ...i,
      ...m
    }), {});
    return (e, i) => {
      var m, d, g, h, y, u, f, k, b;
      return (m = e.response) != null && m.example ? (o(), a(s(r), {
        key: 0,
        class: "bg-b-2",
        content: s(V)((d = e.response) == null ? void 0 : d.example),
        lang: "json"
      }, null, 8, ["content"])) : (g = e.response) != null && g.schema ? (o(), t(c, { key: 1 }, [
        (o(), t(c, null, v(F, (n) => {
          var S, j, O, B, E, N;
          return o(), t(c, { key: n }, [
            (S = e.response) != null && S.schema[n] && (((j = e.response) == null ? void 0 : j.schema[n].length) > 1 || n === "not") ? (o(), t("div", R, [
              C("div", q, J(n), 1),
              C("ol", w, [
                (o(!0), t(c, null, v((O = e.response) == null ? void 0 : O.schema[n], (_, A) => (o(), t("li", {
                  key: A,
                  class: "rule-item"
                }, [
                  L(s(r), {
                    class: "bg-b-2",
                    content: s(l)(_, {
                      emptyString: "…",
                      mode: "read"
                    }),
                    lang: "json"
                  }, null, 8, ["content"])
                ]))), 128))
              ])
            ])) : (B = e.response) != null && B.schema[n] && ((E = e.response) == null ? void 0 : E.schema[n].length) === 1 ? (o(), a(s(r), {
              key: 1,
              class: "bg-b-2",
              content: s(l)((N = e.response) == null ? void 0 : N.schema[n][0], {
                emptyString: "…",
                mode: "read"
              }),
              lang: "json"
            }, null, 8, ["content"])) : P("", !0)
          ], 64);
        }), 64)),
        (h = e.response) != null && h.schema.allOf ? (o(), a(s(r), {
          key: 0,
          class: "bg-b-2",
          content: p(
            (y = e.response) == null ? void 0 : y.schema.allOf.map(
              (n) => s(l)(n, {
                emptyString: "…",
                mode: "read"
              })
            )
          ),
          lang: "json"
        }, null, 8, ["content"])) : (f = (u = e.response) == null ? void 0 : u.schema.items) != null && f.allOf ? (o(), a(s(r), {
          key: 1,
          class: "bg-b-2",
          content: p(
            (k = e.response) == null ? void 0 : k.schema.items.allOf.map(
              (n) => s(l)(n, {
                emptyString: "…",
                mode: "read"
              })
            )
          ),
          lang: "json"
        }, null, 8, ["content"])) : (o(), a(s(r), {
          key: 2,
          class: "bg-b-2",
          content: s(V)(
            s(l)((b = e.response) == null ? void 0 : b.schema, {
              emptyString: "…",
              mode: "read"
            })
          ),
          lang: "json"
        }, null, 8, ["content"]))
      ], 64)) : (o(), t("div", z, " No Body "));
    };
  }
});
export {
  Q as default
};
