// @scalar/oas-utils/dist/helpers/index.js
import {
  isDefined
} from "@scalar/helpers/array/is-defined";
import {
  json2xml
} from "@scalar/helpers/file/json2xml";
import {
  canMethodHaveBody
} from "@scalar/helpers/http/can-method-have-body";
import {
  REQUEST_METHODS,
  getHttpMethodInfo
} from "@scalar/helpers/http/http-info";
import {
  httpStatusCodes
} from "@scalar/helpers/http/http-status-codes";
import {
  isHttpMethod
} from "@scalar/helpers/http/is-http-method";
import {
  objectKeys
} from "@scalar/helpers/object/object-keys";
import {
  findVariables
} from "@scalar/helpers/regex/find-variables";
import {
  REGEX
} from "@scalar/helpers/regex/regex-helpers";
import {
  replaceVariables
} from "@scalar/helpers/regex/replace-variables";
import {
  camelToTitleWords
} from "@scalar/helpers/string/camel-to-title";
import {
  capitalize
} from "@scalar/helpers/string/capitalize";
import {
  createHash
} from "@scalar/helpers/string/create-hash";
import {
  iterateTitle
} from "@scalar/helpers/string/iterate-title";
import {
  ensureProtocol
} from "@scalar/helpers/url/ensure-protocol";
import {
  isLocalUrl
} from "@scalar/helpers/url/is-local-url";
import {
  isRelativePath
} from "@scalar/helpers/url/is-relative-path";
import {
  isValidUrl
} from "@scalar/helpers/url/is-valid-url";
import {
  makeUrlAbsolute
} from "@scalar/helpers/url/make-url-absolute";
import {
  combineUrlAndPath,
  mergeSearchParams,
  mergeUrls
} from "@scalar/helpers/url/merge-urls";
import { fetchDocument } from "./fetch-document.js";
import { fetchWithProxyFallback } from "./fetch-with-proxy-fallback.js";
import { normalizeMimeType } from "./normalize-mime-type.js";
import { normalizeMimeTypeObject } from "./normalize-mime-type-object.js";
import { getOperationStability, getOperationStabilityColor, isOperationDeprecated } from "./operation-stability.js";
import { formatJsonOrYamlString, isJsonString, json, parseJsonOrYaml, transformToJson, yaml } from "./parse.js";
import { prettyPrintJson, replaceCircularDependencies } from "./pretty-print-json.js";
import { schemaModel } from "./schema-model.js";
import { getServersFromDocument } from "./servers.js";
import { shouldIgnoreEntity } from "./should-ignore-entity.js";
export {
  REGEX,
  REQUEST_METHODS,
  camelToTitleWords,
  canMethodHaveBody,
  capitalize,
  combineUrlAndPath,
  createHash,
  ensureProtocol,
  fetchDocument,
  fetchWithProxyFallback,
  findVariables,
  formatJsonOrYamlString,
  getHttpMethodInfo,
  objectKeys as getObjectKeys,
  getOperationStability,
  getOperationStabilityColor,
  getServersFromDocument,
  httpStatusCodes,
  isDefined,
  isHttpMethod,
  isJsonString,
  isLocalUrl,
  isOperationDeprecated,
  isRelativePath,
  isValidUrl,
  iterateTitle,
  json,
  json2xml,
  makeUrlAbsolute,
  mergeSearchParams,
  mergeUrls,
  normalizeMimeType,
  normalizeMimeTypeObject,
  parseJsonOrYaml,
  prettyPrintJson,
  replaceCircularDependencies,
  replaceVariables,
  schemaModel,
  shouldIgnoreEntity,
  transformToJson,
  yaml
};
