// @scalar/snippetz/dist/libs/http.js
function normalizeRequest(request) {
  return {
    ...request,
    method: (request.method || "GET").toUpperCase()
  };
}
function buildQueryString(queryParams) {
  if (!queryParams?.length) {
    return "";
  }
  const queryPairs = queryParams.map((param) => `${encodeURIComponent(param.name)}=${encodeURIComponent(param.value)}`);
  return `?${queryPairs.join("&")}`;
}
const createSearchParams = (query = []) => {
  const searchParams = new URLSearchParams();
  query.forEach((q) => {
    searchParams.append(q.name, q.value);
  });
  return searchParams;
};
function buildUrl(baseUrl, queryString) {
  return `${baseUrl}${queryString}`;
}
function processHeaders(request) {
  const headers = {};
  if (request.headers) {
    for (const header of request.headers) {
      if (header.value && !/[; ]/.test(header.name)) {
        headers[header.name] = header.value;
      }
    }
  }
  if (request.cookies && request.cookies.length > 0) {
    const cookieString = request.cookies.map((cookie) => `${encodeURIComponent(cookie.name)}=${encodeURIComponent(cookie.value)}`).join("; ");
    headers["Cookie"] = cookieString;
  }
  return headers;
}
export {
  buildQueryString,
  buildUrl,
  createSearchParams,
  normalizeRequest,
  processHeaders
};
