// @scalar/api-client/dist/views/Cookies/Cookies.vue2.js
import { defineComponent as L, computed as V, onMounted as D, openBlock as l, createBlock as y, withCtx as n, createVNode as s, unref as a, createElementVNode as m, createElementBlock as f, Fragment as h, renderList as x, normalizeClass as w, createTextVNode as v, toDisplayString as S, withDirectives as $, vShow as O } from "vue";
import M from "../../components/Sidebar/SidebarButton.vue.js";
import N from "../../components/Sidebar/SidebarList.vue.js";
import B from "../../components/Sidebar/SidebarListElement.vue.js";
import F from "../../components/SubpageHeader.vue.js";
import I from "../../components/ViewLayout/ViewLayout.vue.js";
import R from "../../components/ViewLayout/ViewLayoutContent.vue.js";
import { useWorkspace as A } from "../../store/workspace.js";
import { ScalarIcon as j } from "@scalar/components";
import { createCookie as H } from "@scalar/oas-utils/entities/workspace/cookie";
import { nanoid as P } from "nanoid";
import { useRouter as T } from "vue-router";
import W from "./CookieForm.vue.js";
import q from "./CookieRaw.vue.js";
import G from "../../components/Sidebar/Sidebar.vue.js";
/* empty css                                    */
import { useSidebar as J } from "../../hooks/useSidebar.js";
const K = { class: "flex-1" }, Q = ["onClick"], U = { class: "before:bg-b-3 before:absolute before:left-[calc(1rem_-_1.5px)] before:top-0 before:z-10 before:h-[calc(100%_+_.5px)] last:before:h-full before:w-[.5px] mb-[.5px] last:mb-0 relative" }, X = ["onClick"], Y = { class: "before:bg-b-3 before:absolute before:left-[calc(1.75rem_-_1.5px)] before:top-0 before:z-10 before:h-[calc(100%_+_.5px)] last:before:h-full before:w-[.5px] mb-[.5px] last:mb-0 relative" }, xe = /* @__PURE__ */ L({
  __name: "Cookies",
  setup(Z) {
    const { cookies: p, cookieMutators: g } = A(), { collapsedSidebarFolders: i, toggleSidebarFolder: k } = J(), b = T(), E = () => {
      const t = Object.keys(p).length, e = H({
        uid: P(),
        name: `Cookie ${t}`,
        value: "",
        domain: "example.com",
        path: "/",
        secure: !1,
        httpOnly: !1,
        sameSite: "None"
      });
      g.add(e), b.push(e.uid);
    }, z = (t) => {
      g.delete(t);
      const e = Object.values(p).filter(
        (o) => o.uid !== t
      );
      if (e.length > 1) {
        const o = e[e.length - 1];
        b.push(o.uid);
      } else e.length === 1 && e[0].uid === "default" && b.push("default");
    }, _ = V(() => {
      const t = {};
      return Object.values(p).forEach((e) => {
        e.domain && e.path && (t[e.domain] || (t[e.domain] = {}), t[e.domain][e.path] || (t[e.domain][e.path] = []), t[e.domain][e.path].push(e));
      }), t;
    }), C = (t) => i[t];
    return D(() => {
      const t = Object.keys(_.value), e = Object.entries(_.value).flatMap(
        ([o, r]) => Object.keys(r).map((d) => o + d)
      );
      t.forEach((o) => {
        i[o] = !0;
      }), e.forEach((o) => {
        i[o] = !0;
      });
    }), (t, e) => (l(), y(F, null, {
      default: n(() => [
        s(I, null, {
          default: n(() => [
            s(a(G), { title: "Cookies" }, {
              content: n(() => [
                m("div", K, [
                  s(N, null, {
                    default: n(() => [
                      (l(!0), f(h, null, x(_.value, (o, r) => (l(), f("div", { key: r }, [
                        m("button", {
                          class: "flex font-medium gap-1.5 items-center px-2 py-1.5 text-left text-sm w-full break-words rounded hover:bg-b-2",
                          type: "button",
                          onClick: (d) => a(k)(r)
                        }, [
                          s(a(j), {
                            class: w(["text-c-3", {
                              "rotate-90": a(i)[r]
                            }]),
                            icon: "ChevronRight",
                            size: "sm",
                            thickness: "2.5"
                          }, null, 8, ["class"]),
                          v(" " + S(r), 1)
                        ], 8, Q),
                        $(m("div", U, [
                          (l(!0), f(h, null, x(o, (d, c) => (l(), f("div", { key: c }, [
                            m("button", {
                              class: "flex font-medium gap-1.5 items-center pl-5 pr-2 py-1.5 text-left text-sm w-full break-words rounded hover:bg-b-2",
                              type: "button",
                              onClick: (u) => a(k)(r + c)
                            }, [
                              s(a(j), {
                                class: w(["text-c-3", {
                                  "rotate-90": a(i)[r + c]
                                }]),
                                icon: "ChevronRight",
                                size: "sm",
                                thickness: "2.5"
                              }, null, 8, ["class"]),
                              v(" " + S(c), 1)
                            ], 8, X),
                            $(m("div", Y, [
                              (l(!0), f(h, null, x(d, (u) => (l(), y(B, {
                                key: u.uid,
                                class: "cookie text-xs",
                                variable: { name: u.name, uid: u.uid },
                                warningMessage: "Are you sure you want to delete this cookie?",
                                onDelete: (ee) => z(u.uid)
                              }, null, 8, ["variable", "onDelete"]))), 128))
                            ], 512), [
                              [O, C(r + c)]
                            ])
                          ]))), 128))
                        ], 512), [
                          [O, C(r)]
                        ])
                      ]))), 128))
                    ]),
                    _: 1
                  })
                ])
              ]),
              button: n(() => [
                s(M, { click: E }, {
                  title: n(() => [
                    v("Add Item")
                  ]),
                  _: 1
                })
              ]),
              _: 1
            }),
            s(R, { class: "flex-1" }, {
              default: n(() => [
                s(W),
                s(q)
              ]),
              _: 1
            })
          ]),
          _: 1
        })
      ]),
      _: 1
    }));
  }
});
export {
  xe as default
};
