// @scalar/api-client/dist/views/Cookies/CookieForm.vue.js
import { defineComponent as p, computed as u, openBlock as d, createBlock as m, withCtx as k, createElementVNode as l } from "vue";
import v from "../../components/Form/Form.vue.js";
import { useWorkspace as h } from "../../store/workspace.js";
const _ = { class: "flex items-center pointer-events-none" }, f = /* @__PURE__ */ l("label", {
  class: "absolute w-full h-full top-0 left-0 pointer-events-auto opacity-0 cursor-text",
  for: "cookiename"
}, null, -1), y = ["value"], F = /* @__PURE__ */ p({
  __name: "CookieForm",
  setup(b) {
    const { cookies: r, activeCookieId: o, cookieMutators: c } = h(), s = [
      { label: "Key", key: "key", placeholder: "Username" },
      { label: "Value", key: "value", placeholder: "123" },
      { label: "Domain", key: "domain", placeholder: "scalar.com" },
      { label: "Path", key: "path", placeholder: "/" },
      { label: "Expires", key: "expires", placeholder: "Tomorrow" },
      { label: "Secure", key: "secure", placeholder: "True/False" },
      { label: "HttpOnly", key: "httpOnly", placeholder: "True/False" }
    ], t = u(
      () => r[o.value] || {}
    ), a = (n, e) => {
      o.value && c.edit(o.value, n, e);
    };
    return (n, e) => (d(), m(v, {
      data: t.value,
      onUpdate: a,
      options: s
    }, {
      title: k(() => [
        l("div", _, [
          f,
          l("input", {
            id: "cookiename",
            class: "pl-1 outline-none border-0 text-c-2 rounded pointer-events-auto relative w-full",
            placeholder: "Cookie Name",
            value: t.value.name,
            onInput: e[0] || (e[0] = (i) => a("name", i.target.value))
          }, null, 40, y)
        ])
      ]),
      _: 1
    }, 8, ["data"]));
  }
});
export {
  F as default
};
