// @scalar/workspace-store/dist/navigation/helpers/traverse-webhooks.js
import { isHttpMethod } from "@scalar/helpers/http/is-http-method";
import { objectKeys } from "@scalar/helpers/object/object-keys";
import { getResolvedRef } from "../../helpers/get-resolved-ref.js";
import { isDeprecatedOperation } from "../../navigation/helpers/traverse-paths.js";
import { getTag } from "./get-tag.js";
const createWebhookEntry = ({
  ref,
  method,
  name,
  title,
  generateId,
  parentTag,
  webhook,
  isDeprecated,
  parentId
}) => {
  const id = generateId({
    type: "webhook",
    name,
    method,
    webhook,
    parentTag,
    parentId
  });
  const entry = {
    id,
    title,
    name,
    ref,
    method,
    type: "webhook",
    isDeprecated
  };
  return entry;
};
const traverseWebhooks = ({
  document,
  tagsMap,
  generateId,
  untaggedWebhooksParentId,
  documentId
}) => {
  const untagged = [];
  Object.entries(document.webhooks ?? {}).forEach(([name, pathItemObject]) => {
    const pathKeys = objectKeys(pathItemObject ?? {}).filter((key) => isHttpMethod(key));
    pathKeys.forEach((method) => {
      const _operation = pathItemObject?.[method];
      const operation = getResolvedRef(_operation);
      if (!operation) {
        return;
      }
      if (operation["x-internal"] || operation["x-scalar-ignore"]) {
        return;
      }
      const ref = `#/webhooks/${name}/${method}`;
      if (operation.tags?.length) {
        operation.tags.forEach((tagName) => {
          const { tag, id: tagId } = getTag({ tagsMap, name: tagName, documentId, generateId });
          tagsMap.get(tagName)?.entries.push(
            createWebhookEntry({
              ref,
              method,
              name,
              title: operation.summary ?? name,
              webhook: operation,
              generateId,
              parentTag: { tag, id: tagId },
              parentId: tagId,
              isDeprecated: isDeprecatedOperation(operation)
            })
          );
        });
      } else {
        untagged.push(
          createWebhookEntry({
            ref,
            method,
            name,
            title: operation.summary ?? name,
            generateId,
            isDeprecated: isDeprecatedOperation(operation),
            webhook: operation,
            parentId: untaggedWebhooksParentId
          })
        );
      }
    });
  });
  return untagged;
};
export {
  traverseWebhooks
};
