// @scalar/workspace-store/dist/navigation/helpers/traverse-description.js
import { getHeadingsFromMarkdown, getLowestHeadingLevel } from "../../navigation/helpers/utils.js";
const DEFAULT_INTRODUCTION_SLUG = "introduction";
const traverseDescription = ({
  generateId,
  parentId,
  info
}) => {
  if (!info.description?.trim()) {
    return [];
  }
  const headings = getHeadingsFromMarkdown(info.description);
  const lowestLevel = getLowestHeadingLevel(headings);
  const entries = [];
  let currentParent = null;
  if (info.description && !info.description.trim().startsWith("#")) {
    const heading = {
      depth: 1,
      value: "Introduction",
      slug: DEFAULT_INTRODUCTION_SLUG
    };
    const id = generateId({
      type: "text",
      depth: heading.depth,
      slug: heading.slug,
      parentId,
      info,
      value: heading.value
    });
    const title = heading.value;
    const entry = {
      id,
      title,
      type: "text"
    };
    entries.push(entry);
  }
  for (const heading of headings) {
    if (heading.depth !== lowestLevel && heading.depth !== lowestLevel + 1) {
      continue;
    }
    const entry = {
      id: generateId({
        type: "text",
        depth: heading.depth,
        slug: heading.slug,
        parentId,
        info,
        value: heading.value
      }),
      title: heading.value,
      type: "text"
    };
    if (heading.depth === lowestLevel) {
      entry.children = [];
      entries.push(entry);
      currentParent = entry;
    } else if (currentParent) {
      currentParent.children?.push(entry);
    }
  }
  return entries;
};
export {
  traverseDescription
};
