// @scalar/api-client/dist/layouts/Modal/create-api-client-modal.js
import { createModalRouter as m } from "../../router.js";
import d from "./ApiClientModal.vue.js";
import { createApiClient as c } from "../../libs/create-client.js";
const R = async (t, e = {}, p = !0) => {
  var s, l, a;
  const r = c({
    el: t,
    appComponent: d,
    configuration: e,
    persistData: !1,
    isReadOnly: !0,
    mountOnInitialize: p,
    router: m()
  }), { importSpecFile: y, importSpecFromUrl: o, workspaceMutators: C } = r.store;
  return (s = e.spec) != null && s.url ? await o(
    e.spec.url,
    e.proxyUrl,
    e == null ? void 0 : e.servers
  ) : (l = e.spec) != null && l.content ? await y(
    (a = e.spec) == null ? void 0 : a.content,
    void 0,
    e == null ? void 0 : e.servers
  ) : C.add({
    uid: "default",
    name: "Workspace",
    isReadOnly: !0,
    proxyUrl: "https://proxy.scalar.com"
  }), r;
}, v = (t, e = {}, p = !0) => c({
  el: t,
  appComponent: d,
  configuration: e,
  persistData: !1,
  isReadOnly: !0,
  mountOnInitialize: p,
  router: m()
});
export {
  R as createApiClientModal,
  v as createApiClientModalSync
};
