// @scalar/snippetz/dist/plugins/shell/curl/curl.js
import { escapeSingleQuotes } from "../../../libs/shell.js";
const shellCurl = {
  target: "shell",
  client: "curl",
  title: "Curl",
  generate(request, configuration) {
    const normalizedRequest = {
      method: "GET",
      ...request
    };
    normalizedRequest.method = normalizedRequest.method.toUpperCase();
    const parts = ["curl"];
    const queryString = normalizedRequest.queryString?.length ? "?" + normalizedRequest.queryString.map((param) => {
      const encodedName = encodeURIComponent(param.name);
      const encodedValue = encodeURIComponent(param.value);
      return `${encodedName}=${encodedValue}`;
    }).join("&") : "";
    const url = `${normalizedRequest.url}${queryString}`;
    const hasSpecialChars = /[\s<>[\]{}|\\^%$]/.test(url);
    const urlPart = queryString || hasSpecialChars ? `'${url}'` : url;
    parts[0] = `curl ${urlPart}`;
    if (normalizedRequest.method !== "GET") {
      parts.push(`--request ${normalizedRequest.method}`);
    }
    if (configuration?.auth?.username && configuration?.auth?.password) {
      const authValue = escapeSingleQuotes(`${configuration.auth.username}:${configuration.auth.password}`);
      parts.push(`--user '${authValue}'`);
    }
    if (normalizedRequest.headers?.length) {
      normalizedRequest.headers.forEach((header) => {
        const headerValue = escapeSingleQuotes(`${header.name}: ${header.value}`);
        parts.push(`--header '${headerValue}'`);
      });
      const acceptEncoding = normalizedRequest.headers.find((header) => header.name.toLowerCase() === "accept-encoding");
      if (acceptEncoding && /gzip|deflate/.test(acceptEncoding.value)) {
        parts.push("--compressed");
      }
    }
    if (normalizedRequest.cookies?.length) {
      const cookieString = normalizedRequest.cookies.map((cookie) => {
        const encodedName = encodeURIComponent(cookie.name);
        const encodedValue = encodeURIComponent(cookie.value);
        return `${encodedName}=${encodedValue}`;
      }).join("; ");
      const escapedCookieString = escapeSingleQuotes(cookieString);
      parts.push(`--cookie '${escapedCookieString}'`);
    }
    if (normalizedRequest.postData) {
      if (normalizedRequest.postData.mimeType === "application/json") {
        if (normalizedRequest.postData.text) {
          try {
            const jsonData = JSON.parse(normalizedRequest.postData.text);
            const prettyJson = JSON.stringify(jsonData, null, 2);
            const escapedJson = escapeSingleQuotes(prettyJson);
            parts.push(`--data '${escapedJson}'`);
          } catch {
            const escapedText = escapeSingleQuotes(normalizedRequest.postData.text ?? "");
            parts.push(`--data '${escapedText}'`);
          }
        }
      } else if (normalizedRequest.postData.mimeType === "application/octet-stream") {
        const escapedText = escapeSingleQuotes(normalizedRequest.postData.text ?? "");
        parts.push(`--data-binary '${escapedText}'`);
      } else if (normalizedRequest.postData.mimeType === "application/x-www-form-urlencoded" && normalizedRequest.postData.params) {
        normalizedRequest.postData.params.forEach((param) => {
          const escapedValue = escapeSingleQuotes(param.value ?? "");
          const encodedName = encodeURIComponent(param.name);
          const escapedName = escapeSingleQuotes(encodedName);
          parts.push(`--data-urlencode '${escapedName}=${escapedValue}'`);
        });
      } else if (normalizedRequest.postData.mimeType === "multipart/form-data" && normalizedRequest.postData.params) {
        normalizedRequest.postData.params.forEach((param) => {
          const escapedName = escapeSingleQuotes(param.name);
          if (param.fileName !== void 0) {
            const escapedFileName = escapeSingleQuotes(param.fileName);
            parts.push(`--form '${escapedName}=@${escapedFileName}'`);
          } else {
            const escapedValue = escapeSingleQuotes(param.value ?? "");
            parts.push(`--form '${escapedName}=${escapedValue}'`);
          }
        });
      } else {
        try {
          const jsonData = JSON.parse(normalizedRequest.postData.text ?? "");
          const prettyJson = JSON.stringify(jsonData, null, 2);
          const escapedJson = escapeSingleQuotes(prettyJson);
          parts.push(`--data '${escapedJson}'`);
        } catch {
          const escapedText = escapeSingleQuotes(normalizedRequest.postData.text ?? "");
          parts.push(`--data '${escapedText}'`);
        }
      }
    }
    return parts.join(" \\\n  ");
  }
};
export {
  shellCurl
};
