// @scalar/api-client/dist/components/Search/SearchModal.vue2.js
import { defineComponent as A, ref as i, watch as M, computed as y, openBlock as w, createBlock as B, unref as d, withCtx as r, createElementVNode as h, createVNode as R, createTextVNode as m, toDisplayString as p, createElementBlock as D, Fragment as N, renderList as T, createSlots as W, pushScopeId as L, popScopeId as U } from "vue";
import { useWorkspace as H } from "../../store/workspace.js";
import { ScalarModal as K, ScalarSearchInput as P, ScalarSearchResultList as j, ScalarSearchResultItem as z } from "@scalar/components";
import { useMagicKeys as G, whenever as g } from "@vueuse/core";
import J from "fuse.js";
import { useRouter as O } from "vue-router";
import Q from "../HttpMethod/HttpMethod.vue.js";
const X = (u) => (L("data-v-905c8bd2"), u = u(), U(), u), Y = /* @__PURE__ */ X(() => /* @__PURE__ */ h("div", { class: "ref-search-meta" }, [
  /* @__PURE__ */ h("span", null, "↑↓ Navigate"),
  /* @__PURE__ */ h("span", null, "⏎ Select")
], -1)), re = /* @__PURE__ */ A({
  __name: "SearchModal",
  props: {
    modalState: {}
  },
  setup(u) {
    const c = u, E = O(), { activeWorkspaceRequests: I } = H(), v = G(), l = i([]), f = i([]), o = i(0), s = i(""), V = i(null), S = new J(l.value, {
      keys: ["title", "description", "body"]
    }), F = () => {
      s.value = "", o.value = 0, f.value = [];
    }, _ = (e) => {
      l.value = e.map((a) => ({
        id: a.uid,
        title: a.summary ?? a.method,
        description: a.description ?? "",
        httpVerb: a.method,
        path: a.path
      })), S.setCollection(l.value);
    }, $ = () => {
      o.value = 0, f.value = S.search(s.value);
    };
    M(
      () => c.modalState.open,
      (e) => {
        var a;
        if (!e) {
          l.value.length > 0 && (l.value = [], S.setCollection(l.value));
          return;
        }
        (a = V.value) == null || a.focus(), F(), _(I.value);
      }
    ), M(
      I,
      (e) => {
        _(e);
      },
      { immediate: !0 }
    );
    function b(e) {
      E.push(e.item.id), c.modalState.hide();
    }
    const k = y(
      () => n.value[o.value]
    ), n = y(
      () => s.value.length === 0 ? l.value.map((e) => ({
        item: e
      })) : f.value
    );
    return g(v.enter, () => {
      c.modalState.open && window && b(k.value);
    }), g(v.ArrowDown, () => {
      var e;
      c.modalState.open && window && (o.value < n.value.length - 1 ? o.value++ : o.value = 0, (e = document.getElementById(`search-modal-${k.value.item.id}`)) == null || e.scrollIntoView({
        behavior: "smooth",
        block: "center"
      }));
    }), g(v.ArrowUp, () => {
      var e;
      c.modalState.open && window && (o.value > 0 ? o.value-- : o.value = n.value.length - 1, (e = document.getElementById(`search-modal-${k.value.item.id}`)) == null || e.scrollIntoView({
        behavior: "smooth",
        block: "center"
      }));
    }), (e, a) => (w(), B(d(K), {
      state: e.modalState,
      variant: "search"
    }, {
      default: r(() => [
        h("div", {
          ref_key: "searchModalRef",
          ref: V,
          class: "ref-search-container"
        }, [
          R(d(P), {
            modelValue: s.value,
            "onUpdate:modelValue": a[0] || (a[0] = (t) => s.value = t),
            onInput: $
          }, null, 8, ["modelValue"])
        ], 512),
        R(d(j), {
          class: "ref-search-results custom-scroll",
          noResults: !n.value.length
        }, {
          query: r(() => [
            m(p(s.value), 1)
          ]),
          default: r(() => [
            (w(!0), D(N, null, T(n.value, (t, C) => (w(), B(d(z), {
              id: `#search-modal-${t.item.id}`,
              key: t.refIndex,
              active: o.value === C,
              icon: "Terminal",
              onClick: (x) => b(t),
              onFocus: (x) => o.value = C
            }, W({
              addon: r(() => [
                R(d(Q), {
                  method: t.item.httpVerb ?? "get"
                }, null, 8, ["method"])
              ]),
              default: r(() => [
                m(p(t.item.title) + " ", 1)
              ]),
              _: 2
            }, [
              (t.item.httpVerb || t.item.path) && t.item.path !== t.item.title ? {
                name: "description",
                fn: r(() => [
                  m(p(t.item.path), 1)
                ]),
                key: "0"
              } : t.item.description ? {
                name: "description",
                fn: r(() => [
                  m(p(t.item.description), 1)
                ]),
                key: "1"
              } : void 0
            ]), 1032, ["id", "active", "onClick", "onFocus"]))), 128))
          ]),
          _: 1
        }, 8, ["noResults"]),
        Y
      ]),
      _: 1
    }, 8, ["state"]));
  }
});
export {
  re as default
};
