// @scalar/api-client/dist/components/Search/SearchButton.vue2.js
import { defineComponent as n, openBlock as r, createElementBlock as a, createElementVNode as t, createVNode as i, unref as p, pushScopeId as d, popScopeId as l } from "vue";
import { ScalarIcon as m } from "@scalar/components";
const u = (e) => (d("data-v-2c26b5ad"), e = e(), l(), e), f = { class: "search-button-fade sticky top-0 z-50 px-3 py-2.5 pb-2.5" }, h = { class: "w-[15px] flex items-center justify-center" }, _ = /* @__PURE__ */ u(() => /* @__PURE__ */ t("div", { class: "sidebar-search-input ml-1.5 flex w-full items-center justify-between text-sm font-medium" }, [
  /* @__PURE__ */ t("span", { class: "sidebar-search-placeholder" }, "Search")
], -1)), v = /* @__PURE__ */ n({
  __name: "SearchButton",
  emits: ["openSearchModal"],
  setup(e, { emit: s }) {
    const c = s;
    return (x, o) => (r(), a("div", f, [
      t("button", {
        class: "border-1/2 text-c-2 flex w-full items-center rounded p-1.5",
        type: "button",
        onClick: o[0] || (o[0] = (b) => c("openSearchModal"))
      }, [
        t("div", h, [
          i(p(m), {
            icon: "Search",
            size: "xs",
            thickness: "2.5"
          })
        ]),
        _
      ])
    ]));
  }
});
export {
  v as default
};
