// @scalar/api-client/dist/v2/features/app/helpers/create-api-client-app.js
import { createApp as c } from "vue";
import { createRouter as p, createWebHistory as n, createWebHashHistory as u } from "vue-router";
import l from "../App.vue.js";
/* empty css          */
import { useAppState as d } from "../app-state.js";
import { ROUTES as m } from "./routes.js";
import { useCommandPaletteState as f } from "../../command-palette/hooks/use-command-palette-state.js";
const A = (t) => t === "web" ? p({ history: n(), routes: m }) : p({ history: u(), routes: m }), R = (t, { layout: r = "desktop", plugins: i }) => {
  const o = A(r), a = d(o), s = f(), e = c(l, {
    layout: r,
    plugins: i,
    getAppState: () => a,
    getCommandPaletteState: () => s
  });
  if (e.use(o), !t) {
    console.error(
      "[@scalar/api-client-modal] Could not create the API client.",
      "Invalid HTML element provided.",
      "Read more: https://github.com/scalar/scalar/tree/main/packages/api-client"
    );
    return;
  }
  return e.mount(t), {
    app: e,
    state: a
  };
};
export {
  R as createApiClientApp,
  A as createAppRouter
};
