// @scalar/api-client/dist/v2/features/app/helpers/routes.js
import n from "../../collection/components/Authentication.vue.js";
import e from "../../collection/components/Cookies.vue.js";
import o from "../../collection/components/Environment.vue.js";
import m from "../../collection/components/Overview.vue.js";
import a from "../../collection/components/Servers.vue.js";
import t from "../../collection/components/Settings.vue.js";
import c from "../../collection/DocumentCollection.vue.js";
import p from "../../collection/WorkspaceCollection.vue.js";
import r from "../../operation/Operation.vue.js";
import { workspaceStorage as i } from "../../../helpers/storage.js";
const g = [
  {
    path: "/@:namespace/:workspaceSlug",
    children: [
      {
        path: ":documentSlug",
        children: [
          // Example page
          {
            name: "example",
            path: "path/:pathEncoded/method/:method/example/:exampleName",
            component: r
          },
          // Document Page
          {
            name: "document",
            path: "",
            component: c,
            children: [
              // Redirect to overview
              {
                name: "document.redirect",
                path: "",
                redirect: {
                  name: "document.overview"
                }
              },
              // Document overview
              {
                name: "document.overview",
                path: "overview",
                component: m
              },
              // Document servers
              {
                name: "document.servers",
                path: "servers",
                component: a
              },
              // Document environment
              {
                name: "document.environment",
                path: "environment",
                component: o
              },
              // Document authentication
              {
                name: "document.authentication",
                path: "authentication",
                component: n
              },
              // Document cookies
              {
                name: "document.cookies",
                path: "cookies",
                component: e
              },
              // Document settings
              {
                name: "document.settings",
                path: "settings",
                component: t
              }
            ]
          }
        ]
      },
      // Workspace page
      {
        name: "workspace",
        path: "",
        component: p,
        children: [
          // Workspace environment
          {
            name: "workspace.environment",
            path: "environment",
            component: o
          },
          // Workspace cookies
          {
            name: "workspace.cookies",
            path: "cookies",
            component: e
          },
          // Workspace settings
          {
            name: "workspace.settings",
            path: "settings",
            component: t
          }
        ]
      }
    ]
  },
  {
    path: "/:pathMatch(.*)*",
    redirect: () => i.getLastPath() ?? "/@local/local/drafts/overview"
  }
];
export {
  g as ROUTES
};
