// @scalar/components/dist/components/ScalarDropdown/ScalarDropdown.vue.js
import { defineComponent as p, openBlock as s, createBlock as l, unref as o, withCtx as t, createVNode as a, mergeProps as m, renderSlot as r, createCommentVNode as u, createElementVNode as f } from "vue";
import { Menu as d, MenuItems as c, MenuButton as y } from "@headlessui/vue";
import B from "../ScalarFloating/ScalarFloating.vue.js";
const g = /* @__PURE__ */ f("div", { class: "absolute inset-0 -z-1 rounded bg-b-1 shadow-lg brightness-lifted" }, null, -1), O = /* @__PURE__ */ p({
  inheritAttrs: !1,
  __name: "ScalarDropdown",
  props: {
    placement: {},
    resize: { type: Boolean },
    isOpen: { type: Boolean },
    teleport: { type: [Boolean, String] },
    static: { type: Boolean, default: !1 }
  },
  setup(h) {
    return (e, _) => (s(), l(o(d), null, {
      default: t(({ open: n }) => [
        a(o(B), {
          isOpen: e.static ? !0 : n ?? e.isOpen,
          placement: e.placement ?? "bottom-start",
          resize: e.resize,
          teleport: e.teleport
        }, {
          floating: t(({ width: i }) => [
            a(o(c), m({ class: "relative flex w-56 flex-col p-0.75" }, e.$attrs, {
              static: e.static,
              style: { width: i }
            }), {
              default: t(() => [
                r(e.$slots, "items"),
                g
              ]),
              _: 2
            }, 1040, ["static", "style"])
          ]),
          default: t(() => [
            e.static ? u("", !0) : (s(), l(o(y), {
              key: 0,
              as: "template"
            }, {
              default: t(() => [
                r(e.$slots, "default")
              ]),
              _: 3
            }))
          ]),
          _: 2
        }, 1032, ["isOpen", "placement", "resize", "teleport"])
      ]),
      _: 3
    }));
  }
});
export {
  O as default
};
