// @scalar/components/dist/components/ScalarDropdown/ScalarDropdown.vue.js
import { defineComponent as B, mergeModels as D, ref as b, useModel as g, useId as $, watch as I, computed as K, createBlock as L, openBlock as M, unref as m, mergeProps as R, createSlots as N, withCtx as A, renderSlot as C, createVNode as q, createElementVNode as O, withKeys as a, withModifiers as r, nextTick as E } from "vue";
import { useBindCx as U } from "@scalar/use-hooks/useBindCx";
import { onClickOutside as V } from "@vueuse/core";
import j from "./ScalarDropdownMenu.vue.js";
import { useDropdown as z } from "./useDropdown.js";
import F from "../ScalarFloating/ScalarFloating.vue.js";
const P = ["id", "aria-activedescendant", "aria-labelledby", "onKeydown"], G = {}, Z = /* @__PURE__ */ B({
  ...G,
  inheritAttrs: !1,
  __name: "ScalarDropdown",
  props: /* @__PURE__ */ D({
    placement: {},
    offset: { type: [Number, Object, Function] },
    resize: { type: Boolean },
    target: {},
    middleware: {},
    teleport: { type: [Boolean, String] }
  }, {
    open: { type: Boolean, default: !1 },
    openModifiers: {}
  }),
  emits: ["update:open"],
  setup(k) {
    const l = b(), i = b(), n = g(k, "open"), { active: o } = z(), v = $(), f = b(v), y = $();
    async function x() {
      n.value = !n.value, await E(), n.value && i.value?.focus();
    }
    async function h(e) {
      if (["ArrowDown", "ArrowUp", " ", "Enter"].includes(e.key))
        e.preventDefault();
      else return;
      n.value || (n.value = !0), await E(), i.value?.focus(), ["ArrowDown", " ", "Enter"].includes(e.key) ? c(1) : e.key === "ArrowUp" && c(-1);
    }
    I(
      () => l.value?.targetRef,
      (e, t) => {
        e && (e.id ? f.value = e.id : (f.value = v, e.setAttribute("id", f.value)), e.setAttribute("aria-haspopup", "menu"), e.setAttribute("aria-expanded", `${n.value}`), n.value && e.setAttribute("aria-controls", y), e.addEventListener("click", x), e.addEventListener("keydown", h)), t && t !== e && (t.id === v && t.removeAttribute("id"), t.removeAttribute("aria-controls"), t.removeAttribute("aria-haspopup"), t.removeAttribute("aria-expanded"), t.removeEventListener("click", x), t.removeEventListener("keydown", h));
      },
      { immediate: !0 }
    ), I(
      n,
      (e) => {
        const t = l.value?.targetRef;
        t && (t.setAttribute("aria-expanded", `${e}`), e ? t.setAttribute("aria-controls", y) : t.removeAttribute("aria-controls"));
      },
      { immediate: !0 }
    );
    async function u() {
      l.value?.targetRef?.focus(), n.value = !1;
    }
    function w() {
      if (!o.value || !i.value) return;
      const e = i.value.querySelector(
        `#${o.value}[role="menuitem"]:not([aria-disabled="true"])`
      );
      e && (e.click(), u());
    }
    function c(e) {
      if (!n.value || !i.value) return;
      const t = Array.from(
        i.value.querySelectorAll(
          '[role="menuitem"]:not([aria-disabled="true"])'
        )
      );
      if (t.length === 0) return;
      const d = t.findIndex((p) => p.id === o.value);
      if (d === -1) {
        const p = t[e > 0 ? 0 : t.length - 1];
        p?.id && (o.value = p.id);
        return;
      }
      const s = d + e;
      if (s < 0 || s > t.length - 1)
        return;
      const _ = t[s];
      _?.id && (o.value = _.id);
    }
    V(i, u, {
      ignore: [K(() => l.value?.targetRef)]
    });
    const { cx: S } = U();
    return (e, t) => (M(), L(m(F), R({
      ref_key: "floatingRef",
      ref: l
    }, e.$props, {
      placement: k.placement ?? "bottom-start"
    }), N({
      default: A(() => [
        C(e.$slots, "default", { open: n.value })
      ]),
      _: 2
    }, [
      n.value ? {
        name: "floating",
        fn: A(({ width: d }) => [
          q(j, R({
            style: { width: d }
          }, m(S)("max-h-[inherit] max-w-[inherit]")), {
            menu: A(() => [
              O("div", {
                id: m(y),
                ref_key: "menuRef",
                ref: i,
                "aria-activedescendant": m(o),
                "aria-labelledby": f.value,
                class: "flex flex-col p-0.75 outline-none",
                role: "menu",
                tabindex: "-1",
                onClick: r(u, ["stop"]),
                onKeydown: [
                  t[0] || (t[0] = a(r((s) => c(1), ["prevent", "stop"]), ["down"])),
                  a(r(w, ["prevent", "stop"]), ["enter"]),
                  a(r(u, ["prevent", "stop"]), ["escape"]),
                  a(r(w, ["prevent", "stop"]), ["space"]),
                  a(r(u, ["prevent", "stop"]), ["tab"]),
                  t[1] || (t[1] = a(r((s) => c(-1), ["prevent", "stop"]), ["up"]))
                ]
              }, [
                C(e.$slots, "items", { open: n.value })
              ], 40, P)
            ]),
            _: 3
          }, 16, ["style"])
        ]),
        key: "0"
      } : void 0
    ]), 1040, ["placement"]));
  }
});
export {
  Z as default
};
