// @scalar/components/dist/components/ScalarDropdown/ScalarDropdownButton.vue2.js
import { defineComponent as o, createBlock as i, openBlock as s, resolveDynamicComponent as r, normalizeClass as d, unref as t, withCtx as l, renderSlot as c } from "vue";
import { cva as u, cx as b } from "@scalar/use-hooks/useBindCx";
const m = {}, x = /* @__PURE__ */ o({
  ...m,
  __name: "ScalarDropdownButton",
  props: {
    is: { default: "button" },
    active: { type: Boolean },
    disabled: { type: Boolean }
  },
  setup(e) {
    const a = u({
      base: [
        // Layout
        "block h-8 min-w-0 gap-1.5 rounded px-2.5 py-1.5 text-left",
        // Text / background style
        "truncate  no-underline text-sm text-c-1",
        // Interaction
        "cursor-pointer hover:bg-b-2"
      ],
      variants: {
        disabled: { true: "pointer-events-none text-c-3" },
        active: { true: "" }
      },
      compoundVariants: [{ disabled: !1, active: !0, class: "bg-b-2" }]
    });
    return (n, p) => (s(), i(r(e.is), {
      "aria-disabled": e.disabled,
      class: d(["item", t(b)("scalar-dropdown-item", t(a)({ active: e.active, disabled: e.disabled }))]),
      type: e.is === "button" ? "button" : void 0
    }, {
      default: l(() => [
        c(n.$slots, "default", {}, void 0, !0)
      ]),
      _: 3
    }, 8, ["aria-disabled", "class", "type"]));
  }
});
export {
  x as default
};
