// @scalar/components/dist/components/ScalarDropdown/ScalarDropdownItem.vue.js
import { defineComponent as s, openBlock as i, createBlock as d, unref as t, withCtx as l, createElementVNode as c, normalizeClass as m, renderSlot as p } from "vue";
import { MenuItem as u } from "@headlessui/vue";
import { cva as b, cx as f } from "../../cva.js";
const C = /* @__PURE__ */ s({
  __name: "ScalarDropdownItem",
  props: {
    disabled: { type: Boolean }
  },
  emits: ["click"],
  setup(v) {
    const r = b({
      base: [
        // Layout
        "h-8 min-w-0 items-center gap-1.5 rounded px-2.5 py-1.5 text-left",
        // Text / background style
        "truncate text-sm text-c-1",
        // Interaction
        "cursor-pointer hover:bg-b-2 hover:text-c-1"
      ],
      variants: {
        disabled: { true: "pointer-events-none text-c-3" },
        active: { true: "bg-b-2 text-c-1" }
      }
    });
    return (e, o) => (i(), d(t(u), { disabled: e.disabled }, {
      default: l(({ active: n }) => [
        c("button", {
          class: m(["item", t(f)("scalar-dropdown-item", t(r)({ active: n, disabled: e.disabled }))]),
          type: "button",
          onClick: o[0] || (o[0] = (a) => e.$emit("click", a))
        }, [
          p(e.$slots, "default")
        ], 2)
      ]),
      _: 3
    }, 8, ["disabled"]));
  }
});
export {
  C as default
};
