// @scalar/components/dist/components/ScalarTextField/ScalarTextField.vue2.js
import { defineComponent as S, useAttrs as F, ref as h, onMounted as k, openBlock as i, createElementBlock as v, createElementVNode as s, normalizeClass as u, unref as o, createBlock as z, resolveDynamicComponent as V, mergeProps as M, withKeys as T, normalizeStyle as $, toDisplayString as y, createCommentVNode as O, renderSlot as A } from "vue";
import { useTextareaAutosize as D } from "@vueuse/core";
import { nanoid as E } from "nanoid";
import { cva as K, cx as d } from "../../cva.js";
const N = { class: "scalar-input-container relative" }, P = ["for"], R = { class: "icon-slot cursor-pointer text-c-ghost hover:text-c-1 !empty:flex !empty:w-7 !empty:items-center !empty:pr-3" }, H = /* @__PURE__ */ S({
  __name: "ScalarTextField",
  props: {
    modelValue: {},
    placeholder: {},
    label: {},
    labelShadowColor: { default: "var(--scalar-background-2)" },
    error: { type: Boolean },
    isMultiline: { type: Boolean },
    helperText: {},
    emitOnBlur: { type: Boolean, default: !0 },
    handleFieldSubmit: {},
    handleFieldChange: {},
    disableTrim: { type: Boolean, default: !1 }
  },
  emits: ["submit", "update:modelValue"],
  setup(x, { emit: g }) {
    const t = x, p = g, w = K({
      base: "scalar-input-wrapper relative flex items-center rounded border border-solid border-border",
      variants: {
        focus: {
          true: "scalar-input-wrapper-focus border-c-3 has-actv-btn:border has-actv-btn:border-border"
        },
        error: { true: "scalar-input-wrapper-error border-error" }
      }
    }), c = F(), m = c.id || `id-${E()}`, l = h(), a = h(!1);
    let f;
    if (t.isMultiline) {
      const { triggerResize: e } = D({
        element: l,
        input: t.modelValue
      });
      f = e;
    }
    function B(e) {
      const r = e.target.value;
      t.isMultiline && f(), t.handleFieldChange ? t.handleFieldChange(r) : p("update:modelValue", r);
    }
    function b(e) {
      const r = e.target, n = t.disableTrim || t.isMultiline ? r.value : r.value.trim();
      t.handleFieldSubmit ? t.handleFieldSubmit(n) : p("submit", n);
    }
    function C(e) {
      a.value = !1, t.emitOnBlur && t.modelValue && b(e);
    }
    return k(() => {
      var e;
      Object.prototype.hasOwnProperty.call(c, "autofocus") && ((e = l.value) == null || e.focus());
    }), (e, r) => (i(), v("div", N, [
      s("div", {
        class: u(o(w)({ error: e.error, focus: a.value }))
      }, [
        (i(), z(V(e.isMultiline ? "textarea" : "input"), M({ id: o(m) }, e.$attrs, {
          ref_key: "input",
          ref: l,
          class: [
            "scalar-input",
            o(d)(
              { "min-h-[77px]": e.isMultiline },
              "z-10 w-full resize-none appearance-none border-0 bg-transparent p-3 text-sm text-c-1",
              "outline-none transition-opacity"
            )
          ],
          placeholder: e.placeholder,
          value: e.modelValue,
          onBlur: C,
          onFocus: r[0] || (r[0] = (n) => a.value = !0),
          onInput: B,
          onKeydown: T(b, ["enter"])
        }), null, 16, ["id", "class", "placeholder", "value"])),
        e.label ? (i(), v("label", {
          key: 0,
          class: u(
            o(d)(
              "scalar-input-label pointer-events-none absolute left-0 top-0 mx-2 my-3 px-1 text-sm",
              "z-10 origin-top-left rounded text-c-3 shadow-current transition-transform"
            )
          ),
          for: o(m),
          style: $({
            "box-shadow": `0 0 4px 4px ${e.labelShadowColor}`,
            "background-color": e.labelShadowColor
          })
        }, y(e.label), 15, P)) : O("", !0),
        s("div", R, [
          A(e.$slots, "default", {}, void 0, !0)
        ])
      ], 2),
      s("span", {
        class: u(
          o(d)(
            "helper-text before:font-black mt-1.5 flex items-center text-sm text-red before:rounded-full",
            "before:mr-1.5 before:block before:h-4 before:w-4 before:text-center before:text-xxs before:leading-4",
            "before:text-white empty:hidden"
          )
        )
      }, y(e.helperText), 3)
    ]));
  }
});
export {
  H as default
};
