// @scalar/api-reference/dist/features/toolbar/ApiReferenceToolbarShareRegister.vue.js
import { defineComponent, createElementBlock, openBlock, Fragment, createElementVNode, createVNode, renderList, createBlock, createTextVNode, resolveDynamicComponent, toDisplayString, withCtx } from "vue";
import { ScalarIconGlobeSimple, ScalarIconGitBranch, ScalarIconFileMd, ScalarIconLockSimple, ScalarIconWarningOctagon, ScalarIconBracketsCurly } from "@scalar/icons";
import ApiReferenceToolbarBlurb from "./ApiReferenceToolbarBlurb.vue.js";
import _sfc_main$1 from "./ApiReferenceToolbarRegisterButton.vue.js";
const _hoisted_1 = { class: "text-c-2 grid grid-cols-2 gap-2.5 font-medium" };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "ApiReferenceToolbarShareRegister",
  props: {
    workspace: {}
  },
  setup(__props) {
    const FEATURES = [
      { icon: ScalarIconGlobeSimple, label: "Custom Domains" },
      { icon: ScalarIconGitBranch, label: "GitHub Sync" },
      { icon: ScalarIconFileMd, label: "Markdown/MDX" },
      { icon: ScalarIconLockSimple, label: "Password Protection" },
      { icon: ScalarIconWarningOctagon, label: "Spectral Linting" },
      { icon: ScalarIconBracketsCurly, label: "JSON Schema Support" }
    ];
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(Fragment, null, [
        createElementVNode("ul", _hoisted_1, [
          (openBlock(), createElementBlock(Fragment, null, renderList(FEATURES, (feature) => {
            return createElementVNode("li", {
              key: feature.label,
              class: "flex items-center gap-2"
            }, [
              (openBlock(), createBlock(resolveDynamicComponent(feature.icon), {
                class: "text-c-3 size-3.5",
                weight: "bold"
              })),
              createTextVNode(" " + toDisplayString(feature.label), 1)
            ]);
          }), 64))
        ]),
        createVNode(_sfc_main$1, { workspace: __props.workspace }, {
          default: withCtx(() => [..._cache[0] || (_cache[0] = [
            createTextVNode(" Deploy on Scalar ", -1)
          ])]),
          _: 1
        }, 8, ["workspace"]),
        createVNode(ApiReferenceToolbarBlurb, null, {
          default: withCtx(() => [..._cache[1] || (_cache[1] = [
            createTextVNode(" Uploading documents to the Scalar Registry is a Premium feature. See what else is included in our ", -1),
            createElementVNode("a", {
              href: "https://scalar.com/",
              target: "_blank"
            }, " guides", -1),
            createTextVNode(". ", -1)
          ])]),
          _: 1
        })
      ], 64);
    };
  }
});
export {
  _sfc_main as default
};
