// @scalar/api-client/dist/components/Form/Form.vue.js
import { defineComponent as m, openBlock as t, createBlock as o, withCtx as l, createElementBlock as n, toDisplayString as s, renderSlot as p, createElementVNode as c, Fragment as i, renderList as u, normalizeClass as f, createVNode as _, createTextVNode as k, createCommentVNode as y } from "vue";
import h from "../DataTable/DataTable.vue.js";
import V from "../DataTable/DataTableInput.vue.js";
import b from "../DataTable/DataTableRow.vue.js";
import g from "../ViewLayout/ViewLayoutSection.vue.js";
const C = { key: 0 }, N = { class: "custom-scroll flex flex-1 flex-col gap-1.5 px-5 py-5" }, w = /* @__PURE__ */ m({
  __name: "Form",
  props: {
    title: {},
    options: {},
    data: {},
    onUpdate: { type: Function }
  },
  setup(U) {
    return (e, B) => (t(), o(g, null, {
      title: l(() => [
        e.title ? (t(), n("span", C, s(e.title), 1)) : p(e.$slots, "title", { key: 1 })
      ]),
      default: l(() => [
        c("div", N, [
          Object.keys(e.data).length > 0 ? (t(), o(h, {
            key: 0,
            columns: [""]
          }, {
            default: l(() => [
              (t(!0), n(i, null, u(e.options, (a, r) => (t(), o(b, {
                key: r,
                class: f({ "border-t": r === 0 })
              }, {
                default: l(() => [
                  _(V, {
                    modelValue: String(e.data[a.key] ?? ""),
                    placeholder: a.placeholder,
                    "onUpdate:modelValue": (d) => e.onUpdate(a.key, d)
                  }, {
                    default: l(() => [
                      k(s(a.label), 1)
                    ]),
                    _: 2
                  }, 1032, ["modelValue", "placeholder", "onUpdate:modelValue"])
                ]),
                _: 2
              }, 1032, ["class"]))), 128))
            ]),
            _: 1
          })) : y("", !0)
        ])
      ]),
      _: 3
    }));
  }
});
export {
  w as default
};
