// @scalar/workspace-store/dist/schemas/v3.1/strict/type-guards.js
const isObjectSchema = (schema) => {
  return "type" in schema && (schema.type === "object" || Array.isArray(schema.type) && schema.type.includes("object"));
};
const isArraySchema = (schema) => {
  return "type" in schema && (schema.type === "array" || Array.isArray(schema.type) && schema.type.includes("array"));
};
const isStringSchema = (schema) => {
  return "type" in schema && (schema.type === "string" || Array.isArray(schema.type) && schema.type.includes("string"));
};
const isNumberSchema = (schema) => {
  return "type" in schema && (schema.type === "number" || schema.type === "integer" || Array.isArray(schema.type) && schema.type.includes("number") || Array.isArray(schema.type) && schema.type.includes("integer"));
};
const isSchema = (schema) => schema !== void 0 && "type" in schema;
const isContentTypeParameterObject = (parameter) => {
  return "content" in parameter && parameter.content !== void 0;
};
export {
  isArraySchema,
  isContentTypeParameterObject,
  isNumberSchema,
  isObjectSchema,
  isSchema,
  isStringSchema
};
