// @scalar/api-client/dist/v2/features/collection/DocumentCollection.vue.js
import { defineComponent as g, computed as f, createElementBlock as i, openBlock as u, createElementVNode as t, createVNode as o, createBlock as p, createCommentVNode as h, unref as l, withCtx as s, resolveDynamicComponent as y, mergeProps as k } from "vue";
import { ScalarFormError as w, ScalarButton as m } from "@scalar/components";
import { ScalarIconArrowLeft as b, ScalarIconFloppyDisk as S } from "@scalar/icons";
import { LibraryIcon as _ } from "@scalar/icons/library";
import { RouterView as C } from "vue-router";
import V from "../../../components/IconSelector.vue.js";
import B from "./components/LabelInput.vue.js";
import D from "./components/Tabs.vue.js";
const I = { class: "custom-scroll h-full" }, N = {
  key: 0,
  class: "w-full md:mx-auto md:max-w-[720px]"
}, U = ["aria-label"], z = { class: "flex" }, E = { class: "flex flex-row items-center gap-2" }, L = { class: "group relative ml-1.25" }, $ = { class: "px-1.5 py-8" }, j = {
  key: 1,
  class: "flex w-full flex-1 items-center justify-center"
}, F = {
  name: "DocumentCollection"
}, J = /* @__PURE__ */ g({
  ...F,
  props: {
    documentSlug: {},
    document: {},
    eventBus: {},
    layout: {},
    path: {},
    method: {},
    exampleName: {},
    environment: {},
    securitySchemes: {},
    workspaceStore: {},
    activeWorkspace: {},
    plugins: {}
  },
  setup(a) {
    const n = a, c = f(() => n.document?.info?.title || "Untitled Document"), d = f(
      () => n.document?.["x-scalar-icon"] || "interface-content-folder"
    ), v = () => {
      n.workspaceStore.revertDocumentChanges(n.documentSlug);
    }, x = () => {
      n.workspaceStore.saveDocument(n.documentSlug);
    };
    return (T, e) => (u(), i("div", I, [
      a.document ? (u(), i("div", N, [
        t("div", {
          "aria-label": `title: ${c.value}`,
          class: "mx-auto flex h-fit w-full flex-col gap-2 pt-6 pb-3 md:mx-auto md:max-w-[720px]"
        }, [
          a.document["x-scalar-is-dirty"] ? (u(), p(l(w), {
            key: 0,
            variant: "warning"
          }, {
            default: s(() => [
              e[4] || (e[4] = t("div", null, " You have unsaved changes. Save your work to keep your changes, or undo to revert them. ", -1)),
              e[5] || (e[5] = t("div", { class: "flex-1" }, null, -1)),
              t("div", z, [
                o(l(m), {
                  size: "sm",
                  variant: "ghost",
                  onClick: v
                }, {
                  default: s(() => [
                    o(l(b)),
                    e[2] || (e[2] = t("span", null, "Undo", -1))
                  ]),
                  _: 1
                }),
                o(l(m), {
                  size: "sm",
                  variant: "ghost",
                  onClick: x
                }, {
                  default: s(() => [
                    o(l(S)),
                    e[3] || (e[3] = t("span", null, "Save", -1))
                  ]),
                  _: 1
                })
              ])
            ]),
            _: 1
          })) : h("", !0),
          t("div", E, [
            o(V, {
              modelValue: d.value,
              placement: "bottom-start",
              "onUpdate:modelValue": e[0] || (e[0] = (r) => a.eventBus.emit("document:update:icon", r))
            }, {
              default: s(() => [
                o(l(m), {
                  class: "hover:bg-b-2 aspect-square h-7 w-7 cursor-pointer rounded border border-transparent p-0 hover:border-inherit",
                  variant: "ghost"
                }, {
                  default: s(() => [
                    o(l(_), {
                      class: "text-c-2 size-5",
                      src: d.value,
                      "stroke-width": "2"
                    }, null, 8, ["src"])
                  ]),
                  _: 1
                })
              ]),
              _: 1
            }, 8, ["modelValue"])
          ]),
          t("div", L, [
            o(B, {
              class: "text-xl font-bold",
              inputId: "documentName",
              modelValue: c.value,
              "onUpdate:modelValue": e[1] || (e[1] = (r) => a.eventBus.emit("document:update:info", { title: r }))
            }, null, 8, ["modelValue"])
          ])
        ], 8, U),
        o(D, { type: "document" }),
        t("div", $, [
          o(l(C), null, {
            default: s(({ Component: r }) => [
              (u(), p(y(r), k(n, { collectionType: "document" }), null, 16))
            ]),
            _: 1
          })
        ])
      ])) : (u(), i("div", j, [...e[6] || (e[6] = [
        t("div", { class: "flex h-full flex-col items-center justify-center" }, [
          t("h1", { class: "text-2xl font-bold" }, "Document not found"),
          t("p", { class: "text-gray-500" }, " The document you are looking for does not exist. ")
        ], -1)
      ])]))
    ]));
  }
});
export {
  J as default
};
