// @scalar/api-client/dist/v2/features/collection/WorkspaceCollection.vue.js
import { defineComponent as a, createElementBlock as l, openBlock as r, createElementVNode as e, createVNode as o, unref as c, toDisplayString as i, mergeProps as n } from "vue";
import { ScalarIconGlobe as m } from "@scalar/icons";
import { RouterView as p } from "vue-router";
import d from "./components/Tabs.vue.js";
const u = { class: "custom-scroll h-full" }, _ = { class: "w-full md:mx-auto md:max-w-[720px]" }, f = ["aria-label"], x = { class: "group relative ml-1.25" }, h = { class: "text-c-1 flex h-8 items-center" }, v = { class: "px-1.5 py-8" }, k = {}, B = /* @__PURE__ */ a({
  ...k,
  __name: "WorkspaceCollection",
  props: {
    documentSlug: {},
    document: {},
    eventBus: {},
    layout: {},
    path: {},
    method: {},
    exampleName: {},
    environment: {},
    securitySchemes: {},
    workspaceStore: {},
    activeWorkspace: {},
    plugins: {}
  },
  setup(t) {
    const s = t;
    return (w, b) => (r(), l("div", u, [
      e("div", _, [
        e("div", {
          "aria-label": `title: ${t.activeWorkspace.label}`,
          class: "mx-auto flex h-fit w-full flex-col gap-2 pt-6 pb-3 md:mx-auto md:max-w-[720px]"
        }, [
          o(c(m), { class: "text-c-2 size-6" }),
          e("div", x, [
            e("span", h, i(t.activeWorkspace.label), 1)
          ])
        ], 8, f),
        o(d, { type: "workspace" }),
        e("div", v, [
          o(c(p), n(s, { collectionType: "workspace" }), null, 16)
        ])
      ])
    ]));
  }
});
export {
  B as default
};
