// @scalar/use-hooks/dist/useBindCx/useBindCx.js
import { computed, useAttrs } from "vue";
import { cx } from "./cva.js";
function useBindCx() {
  const attrs = useAttrs();
  const destructured = computed(() => {
    const { class: className, style, ...rest } = attrs;
    return { class: className || "", style, rest };
  });
  function bindCx(...args) {
    return {
      class: cx(...args, destructured.value.class),
      style: destructured.value.style,
      ...destructured.value.rest
    };
  }
  function bindClassAndStyle(...args) {
    return { class: cx(...args, destructured.value.class), style: destructured.value.style };
  }
  return {
    /**
     * Provides a wrapper around the `cx` function that merges the
     * component's class attribute with the provided classes and binds the
     * remaining attributes
     *
     * @example
     * ```html
     * <script setup>
     * ...
     * const { cx } = useBindCx()
     * <\/script>
     * <template>
     *   <div v-bind="cx(...)">...</div>
     * </template>
     * ```
     */
    cx: bindCx,
    /**
     * Provides a wrapper around the `cx` function that merges the
     * component's class attribute with the provided classes and binds the
     * style attribute  but **does not** bind any other attributes.
     *
     * Typically used in conjunction with `otherAttrs` to apply the stylistic
     * attributes to a styled wrapper element, but apply the remaining
     * attributes to an internal semantic element like an `<input>`.
     *
     * @example
     * ```html
     * <script setup>
     * ...
     * const { stylingAttrsCx, otherAttrs } = useBindCx()
     * <\/script>
     * <template>
     *   <!-- Bind the class and style attributes to a wrapper element -->
     *   <div v-bind="stylingAttrsCx(...)">
     *     ...
     *     <!-- Bind the other attributes to a semantic internal element -->
     *     <input v-bind="otherAttrs" />
     *   </div>
     * </template>
     * ```
     */
    stylingAttrsCx: bindClassAndStyle,
    /**
     * The remaining attributes that **are not** the class or style attributes
     * of the component.
     *
     * Typically used in conjunction with `stylingAttrsCx` to apply the stylistic
     * attributes to a styled wrapper element, but apply the remaining
     * attributes to an internal semantic element like an `<input>`.
     *
     * @example
     * ```html
     * <script setup>
     * ...
     * const { stylingAttrsCx, otherAttrs } = useBindCx()
     * <\/script>
     * <template>
     *   <!-- Bind the class and style attributes to a wrapper element -->
     *   <div v-bind="stylingAttrsCx(...)">
     *     ...
     *     <!-- Bind the other attributes to a semantic internal element -->
     *     <input v-bind="otherAttrs" />
     *   </div>
     * </template>
     */
    otherAttrs: computed(() => destructured.value.rest)
  };
}
export {
  useBindCx
};
