// @scalar/api-reference/dist/hooks/useDarkModeState.js
import { ref as u, watch as f } from "vue";
const t = u(!1);
function c(r, n) {
  var i;
  const d = () => {
    var o, a;
    const e = typeof window < "u" ? (o = window.localStorage) == null ? void 0 : o.getItem("isDark") : null;
    return typeof e == "string" ? !!JSON.parse(e) : !!(typeof window < "u" && typeof (window == null ? void 0 : window.matchMedia) == "function" && ((a = window == null ? void 0 : window.matchMedia("(prefers-color-scheme: dark)")) != null && a.matches));
  }, l = () => {
    var e;
    t.value = !t.value, typeof window < "u" && ((e = window == null ? void 0 : window.localStorage) == null || e.setItem("isDark", JSON.stringify(t.value)));
  };
  function s(e) {
    var o;
    t.value = e, typeof window < "u" && ((o = window == null ? void 0 : window.localStorage) == null || o.setItem("isDark", JSON.stringify(t.value)));
  }
  return n ? t.value = n === "dark" : t.value = (typeof window > "u" ? null : JSON.parse(((i = window.localStorage) == null ? void 0 : i.getItem("isDark")) || "null")) ?? r ?? d(), f(
    t,
    (e) => {
      typeof document > "u" || (document.body.classList.toggle("dark-mode", e), document.body.classList.toggle("light-mode", !e));
    },
    { immediate: !0 }
  ), {
    isDark: t,
    toggleDarkMode: l,
    setDarkMode: s
  };
}
export {
  c as useDarkModeState
};
