// @scalar/api-reference/dist/hooks/useNavState.js
import { ssrState as g } from "@scalar/oas-utils/helpers";
import { slug as c } from "github-slugger";
import { ref as d } from "vue";
const r = d(g.hash ?? ""), n = d(), i = d(!1), h = (e) => e.slug ? `description/${e.slug}` : "", u = (e) => {
  var t;
  if (!n.value) return "";
  const o = new RegExp("^" + ((t = n.value) == null ? void 0 : t.basePath) + "/?");
  return decodeURIComponent(e.replace(o, ""));
}, p = (e, o) => {
  if (!e)
    return "webhooks";
  const t = c(e), s = encodeURIComponent(t);
  return `webhook/${o}/${s}`;
}, m = (e) => {
  if (!e)
    return "models";
  const o = c(e);
  return `model/${encodeURIComponent(o)}`;
}, I = (e, o) => `${l(o)}/${e.httpVerb}${e.path}`, l = ({ name: e }) => {
  const o = c(e);
  return `tag/${encodeURIComponent(o)}`;
}, w = (e = r.value) => {
  var a;
  const o = (a = e.match(/(tag\/[^/]+)/)) == null ? void 0 : a[0], t = e.startsWith("model") ? "models" : "", s = e.startsWith("webhook") ? "webhooks" : "";
  return o || t || s;
}, b = () => {
  r.value = n.value ? u(window.location.pathname) : decodeURIComponent(window.location.hash.replace(/^#/, ""));
}, $ = () => ({
  hash: r,
  getWebhookId: p,
  getModelId: m,
  getHeadingId: h,
  getOperationId: I,
  getPathRoutingId: u,
  getSectionId: w,
  getTagId: l,
  isIntersectionEnabled: i,
  pathRouting: n,
  updateHash: b
});
export {
  $ as useNavState
};
