// @scalar/api-reference/dist/hooks/useSidebar.js
import { ssrState as $ } from "@scalar/oas-utils/helpers";
import { ref as m, reactive as J, computed as K, watch as B } from "vue";
import { lazyBus as ee } from "../components/Content/Lazy/lazyBus.js";
import { useNavState as te } from "./useNavState.js";
import { getLowestHeadingLevel as N } from "../helpers/getLowestHeadingLevel.js";
import { openClientFor as D } from "../helpers/openClientFor.js";
import { hasModels as se } from "../helpers/hasModels.js";
import { getModels as F } from "../helpers/getModels.js";
import { hasWebhooks as oe } from "../helpers/hasWebhooks.js";
import { scrollToId as V } from "../helpers/scrollToId.js";
import { getHeadingsFromMarkdown as ie } from "../helpers/getHeadingsFromMarkdown.js";
import { useApiClientStore as re } from "../legacy/stores/useApiClientStore.js";
import { useOpenApiStore as le } from "../legacy/stores/useOpenApiStore.js";
const {
  getHeadingId: z,
  getModelId: P,
  getOperationId: R,
  getSectionId: Q,
  getTagId: U,
  getWebhookId: A,
  hash: O
} = te(), l = m(void 0), S = J({});
function q(t) {
  var s, o;
  return S.tagsSorter === "alpha" ? t.tags = (s = t.tags) == null ? void 0 : s.sort((r, c) => r.name.localeCompare(c.name)) : typeof S.tagsSorter == "function" && (t.tags = (o = t.tags) == null ? void 0 : o.sort(S.tagsSorter)), l.value = t;
}
const X = m(!1), Y = m(!1), f = J(
  $["useSidebarContent-collapsedSidebarItems"] ?? {}
);
function ne(t) {
  f[t] = !f[t];
}
function h(t, s) {
  f[t] = s;
}
const C = m([]);
function ae(t) {
  const s = ie(t), o = N(s);
  return s.filter((r) => (
    // highest level, eg. # Introduction
    r.depth === o || // second highest level, eg. ## Authentication
    r.depth === o + 1
  ));
}
const Z = K(() => {
  var T, I, y, H, M, j, x, E, G, L;
  const { state: t } = re(), s = {}, {
    openApi: { globalSecurity: o }
  } = le(), r = [];
  let c = null;
  C.value.forEach((e) => {
    var i;
    e.depth === N(C.value) ? (c = {
      id: z(e),
      title: e.value,
      show: !t.showApiClient,
      children: []
    }, r.push(c)) : c && ((i = c.children) == null || i.push({
      id: z(e),
      title: e.value,
      show: !t.showApiClient
    }));
  });
  const p = (I = (T = l.value) == null ? void 0 : T.tags) == null ? void 0 : I[0], v = p && ((e) => (e == null ? void 0 : e.length) !== 1 || e[0].name !== "default" || e[0].description !== "")((y = l.value) == null ? void 0 : y.tags) ? (M = (H = l.value) == null ? void 0 : H.tags) == null ? void 0 : M.filter((e) => {
    var i;
    return ((i = e.operations) == null ? void 0 : i.length) > 0;
  }).map((e) => {
    var i;
    return {
      id: U(e),
      title: e.name,
      displayTitle: e["x-displayName"] ?? e.name,
      show: !0,
      children: (i = e.operations) == null ? void 0 : i.map(
        (n) => {
          var d;
          const a = R(n, e), u = n.name ?? n.path;
          return s[a] = u, {
            id: a,
            title: u,
            httpVerb: n.httpVerb,
            deprecated: ((d = n.information) == null ? void 0 : d.deprecated) ?? !1,
            show: !0,
            select: () => {
              t.showApiClient && D(n, o);
            }
          };
        }
      )
    };
  }) : (j = p == null ? void 0 : p.operations) == null ? void 0 : j.map((e) => {
    var a;
    const i = R(e, p), n = e.name ?? e.path;
    return s[i] = n, {
      id: i,
      title: n,
      httpVerb: e.httpVerb,
      deprecated: ((a = e.information) == null ? void 0 : a.deprecated) ?? !1,
      show: !0,
      select: () => {
        t.showApiClient && D(e, o);
      }
    };
  });
  let g = se(l.value) && !X.value ? [
    {
      id: P(),
      title: "Models",
      show: !t.showApiClient,
      children: Object.keys(F(l.value) ?? {}).map(
        (e) => {
          var n;
          const i = P(e);
          return s[i] = e, {
            id: i,
            title: ((n = F(l.value)) == null ? void 0 : n[e]).title ?? e,
            show: !t.showApiClient
          };
        }
      )
    }
  ] : [], w = oe(l.value) ? [
    {
      id: A(),
      title: "Webhooks",
      show: !t.showApiClient,
      children: Object.keys(((x = l.value) == null ? void 0 : x.webhooks) ?? {}).map((e) => {
        var n, a;
        const i = A(e);
        return s[i] = e, Object.keys(
          ((a = (n = l.value) == null ? void 0 : n.webhooks) == null ? void 0 : a[e]) ?? {}
        ).map((u) => {
          var d, b, W;
          return {
            id: A(e, u),
            title: (W = (b = (d = l.value) == null ? void 0 : d.webhooks) == null ? void 0 : b[e][u]) == null ? void 0 : W.name,
            httpVerb: u,
            show: !t.showApiClient
          };
        });
      }).flat()
    }
  ] : [];
  const _ = (E = l.value) != null && E["x-tagGroups"] ? (L = (G = l.value) == null ? void 0 : G["x-tagGroups"]) == null ? void 0 : L.map((e) => {
    var a;
    const i = [];
    return (a = e.tags) == null || a.map((u) => {
      if (u === "models" && g.length > 0)
        i.push(g[0]), g = [];
      else if (u === "webhooks" && w.length > 0)
        i.push(w[0]), w = [];
      else {
        const d = v == null ? void 0 : v.find(
          (b) => b.title === u
        );
        d && i.push(d);
      }
    }), {
      id: e.name,
      title: e.name,
      children: i,
      show: !0,
      isGroup: !0
    };
  }) : void 0, k = [
    ...r,
    ..._ ?? v ?? [],
    ...w,
    ...g
  ];
  return Y.value && k.forEach((e) => {
    h(e.id, !0), e.show = !0;
  }), {
    entries: k,
    titles: s
  };
}), ue = m(!1), de = K(() => {
  var t, s;
  return ((s = (t = Z.value) == null ? void 0 : t.titles) == null ? void 0 : s[O.value]) ?? "";
}), ce = (t) => {
  const s = Q(t);
  if (s !== t)
    if (f[s])
      V(t);
    else {
      const o = ee.on((r) => {
        r.id === t && (V(t), o());
      });
      h(s, !0);
    }
};
function Ie(t) {
  return Object.assign(S, t), t != null && t.parsedSpec && (q(t.parsedSpec), B(
    () => {
      var s, o;
      return (o = (s = l.value) == null ? void 0 : s.tags) == null ? void 0 : o.length;
    },
    () => {
      var s, o;
      if (O.value) {
        const r = Q(O.value);
        r && h(r, !0);
      } else {
        const r = (o = (s = l.value) == null ? void 0 : s.tags) == null ? void 0 : o[0];
        r && h(U(r), !0);
      }
    }
  ), B(
    () => {
      var s, o;
      return (o = (s = l.value) == null ? void 0 : s.info) == null ? void 0 : o.description;
    },
    () => {
      var o, r;
      const s = (r = (o = l.value) == null ? void 0 : o.info) == null ? void 0 : r.description;
      return s ? C.value = ae(s) : C.value = [];
    },
    {
      immediate: !0
    }
  )), {
    breadcrumb: de,
    items: Z,
    isSidebarOpen: ue,
    collapsedSidebarItems: f,
    toggleCollapsedSidebarItem: ne,
    setCollapsedSidebarItem: h,
    hideModels: X,
    setParsedSpec: q,
    defaultOpenAllTags: Y,
    scrollToOperation: ce
  };
}
export {
  ce as scrollToOperation,
  Ie as useSidebar
};
