// @scalar/api-client/dist/libs/event-busses/hot-keys-bus.js
import { isMacOS as f } from "@scalar/use-tooltip";
import { useEventBus as d } from "@vueuse/core";
const m = Symbol(), a = d(m), u = [
  "Escape",
  "ArrowDown",
  "ArrowUp",
  "Enter",
  "F1",
  "F2",
  "F3",
  "F4",
  "F5",
  "F6",
  "F7",
  "F8",
  "F9",
  "F10",
  "F11",
  "F12"
], l = {
  Escape: { event: "closeModal" },
  b: { event: "toggleSidebar", modifiers: ["default"] },
  k: { event: "openCommandPalette", modifiers: ["default"] },
  ArrowUp: { event: "commandPaletteUp" },
  ArrowDown: { event: "commandPaletteDown" },
  Enter: { event: "commandPaletteSelect" },
  t: { event: "addTopNav", modifiers: ["default"] },
  w: { event: "closeTopNav", modifiers: ["default"] },
  ArrowLeft: { event: "navigateTopNavLeft", modifiers: ["default", "Alt"] },
  ArrowRight: { event: "navigateTopNavRight", modifiers: ["default", "Alt"] },
  l: { event: "focusAddressBar", modifiers: ["default"] },
  1: { event: "jumpToTab", modifiers: ["default"] },
  2: { event: "jumpToTab", modifiers: ["default"] },
  3: { event: "jumpToTab", modifiers: ["default"] },
  4: { event: "jumpToTab", modifiers: ["default"] },
  5: { event: "jumpToTab", modifiers: ["default"] },
  6: { event: "jumpToTab", modifiers: ["default"] },
  7: { event: "jumpToTab", modifiers: ["default"] },
  8: { event: "jumpToTab", modifiers: ["default"] },
  9: { event: "jumpToLastTab", modifiers: ["default"] }
}, r = (e) => {
  if (!(e.target instanceof HTMLElement)) return !1;
  const t = e.target;
  return t.tagName === "INPUT" ? !u.includes(e.key) : !!(t.tagName === "TEXTAREA" || t.getAttribute("contenteditable"));
}, c = {
  Alt: "altKey",
  Control: "ctrlKey",
  Shift: "shiftKey",
  Meta: "metaKey"
}, T = (e) => e.map(
  (t) => t === "default" ? f() ? "metaKey" : "ctrlKey" : c[t]
), A = (e, {
  hotKeys: t = l,
  modifiers: n = ["default"]
} = {}) => {
  const i = e.key === " " ? "Space" : e.key, o = t[i];
  o && (i === "Escape" ? a.emit({ [o.event]: e }) : T(o.modifiers || n).every((s) => e[s] === !0) && !r(e) ? a.emit({ [o.event]: e }) : !r(e) && o.modifiers === void 0 && a.emit({ [o.event]: e }));
};
export {
  l as DEFAULT_HOTKEYS,
  A as handleHotKeyDown,
  a as hotKeyBus
};
