// @scalar/components/dist/components/ScalarPopover/ScalarPopover.vue.js
import { defineComponent as c, createBlock as i, openBlock as u, unref as e, withCtx as o, createVNode as r, normalizeProps as d, guardReactiveProps as _, renderSlot as l, mergeProps as P } from "vue";
import { Popover as v, PopoverButton as B, PopoverPanel as y } from "@headlessui/vue";
import { useBindCx as $ } from "@scalar/use-hooks/useBindCx";
import g from "../ScalarFloating/ScalarFloating.vue.js";
import b from "../ScalarFloating/ScalarFloatingBackdrop.vue.js";
const z = /* @__PURE__ */ c({
  inheritAttrs: !1,
  __name: "ScalarPopover",
  props: {
    focus: { type: Boolean },
    placement: {},
    offset: { type: [Number, Object, Function] },
    resize: { type: Boolean },
    target: {},
    middleware: {},
    teleport: { type: [Boolean, String] }
  },
  setup(s) {
    const { cx: p } = $();
    return (t, h) => (u(), i(e(v), { as: "template" }, {
      default: o(({ open: a }) => [
        r(e(g), d(_(t.$props)), {
          floating: o(({ width: f, height: n }) => [
            r(e(y), P({
              focus: s.focus,
              style: { width: f, height: n }
            }, e(p)("relative flex flex-col p-0.75")), {
              default: o(({ close: m }) => [
                l(t.$slots, "popover", {
                  close: () => m(),
                  open: a
                }),
                l(t.$slots, "backdrop", { open: a }, () => [
                  r(e(b))
                ])
              ]),
              _: 2
            }, 1040, ["focus", "style"])
          ]),
          default: o(() => [
            r(e(B), { as: "template" }, {
              default: o(() => [
                l(t.$slots, "default", { open: a })
              ]),
              _: 2
            }, 1024)
          ]),
          _: 2
        }, 1040)
      ]),
      _: 3
    }));
  }
});
export {
  z as default
};
