// @scalar/components/dist/components/ScalarPopover/ScalarPopover.vue.js
import { defineComponent as i, openBlock as f, createBlock as d, unref as o, withCtx as t, createVNode as r, mergeProps as l, renderSlot as s, createElementVNode as m } from "vue";
import { Popover as u, PopoverPanel as c, PopoverButton as _ } from "@headlessui/vue";
import v from "../ScalarFloating/ScalarFloating.vue.js";
const g = /* @__PURE__ */ m("div", { class: "absolute inset-0 -z-1 rounded bg-b-1 shadow-lg brightness-lifted" }, null, -1), b = /* @__PURE__ */ i({
  inheritAttrs: !1,
  __name: "ScalarPopover",
  props: {
    placement: {},
    resize: { type: Boolean },
    isOpen: { type: Boolean },
    teleport: { type: [Boolean, String] }
  },
  setup(h) {
    return (e, B) => (f(), d(o(u), { as: "template" }, {
      default: t(({ open: p }) => [
        r(o(v), l(e.$props, {
          isOpen: p ?? e.isOpen,
          teleport: e.teleport
        }), {
          floating: t(({ width: a, height: n }) => [
            r(o(c), l({
              class: "relative flex flex-col p-0.75",
              style: { width: a, height: n }
            }, e.$attrs), {
              default: t(() => [
                s(e.$slots, "popover"),
                g
              ]),
              _: 2
            }, 1040, ["style"])
          ]),
          default: t(() => [
            r(o(_), { as: "template" }, {
              default: t(() => [
                s(e.$slots, "default")
              ]),
              _: 3
            })
          ]),
          _: 2
        }, 1040, ["isOpen", "teleport"])
      ]),
      _: 3
    }));
  }
});
export {
  b as default
};
