// @scalar/api-reference/dist/features/external-docs/ExternalDocs.vue.js
import { defineComponent, createElementBlock, createCommentVNode, openBlock, createElementVNode, createVNode, unref, toDisplayString } from "vue";
import { ScalarIconBook } from "@scalar/icons";
const _hoisted_1 = {
  key: 0,
  class: "group flex items-center last:border-r-0 xl:border-r xl:first:ml-auto"
};
const _hoisted_2 = ["href"];
const _hoisted_3 = {
  key: 0,
  class: "ml-1 empty:hidden"
};
const _hoisted_4 = {
  key: 1,
  class: "ml-1 empty:hidden"
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "ExternalDocs",
  props: {
    value: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return __props.value ? (openBlock(), createElementBlock("div", _hoisted_1, [
        createElementVNode("a", {
          href: __props.value.url,
          class: "text-c-1 hover:bg-b-2 mr-2 flex min-h-7 min-w-7 items-center rounded-lg border px-2 py-1 no-underline group-last:mr-0 xl:border-none",
          target: "_blank",
          rel: "noopener noreferrer"
        }, [
          createVNode(unref(ScalarIconBook), {
            weight: "bold",
            class: "size-3 text-current"
          }),
          __props.value.description ? (openBlock(), createElementBlock("span", _hoisted_3, toDisplayString(__props.value.description), 1)) : (openBlock(), createElementBlock("span", _hoisted_4, toDisplayString(__props.value.url), 1))
        ], 8, _hoisted_2)
      ])) : createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as default
};
