// @scalar/api-client/dist/components/DataTable/DataTableCheckbox.vue.js
import { defineComponent as a, openBlock as l, createBlock as s, withCtx as n, createElementVNode as t, createVNode as c, unref as d } from "vue";
import { ScalarIcon as i } from "@scalar/components";
import f from "./DataTableCell.vue.js";
const p = ["checked", "disabled"], b = { class: "w-full h-full flex items-center justify-center centered text-border peer-checked:text-c-2 pointer-events-none absolute before:absolute before:opacity-0 group-hover/cell:before:opacity-100 before:border before:rounded before:w-9/12 before:aspect-square before:top-1/2 before:left-1/2 before:transform before:-translate-x-1/2 before:-translate-y-1/2 before:border-[1px]" }, x = /* @__PURE__ */ a({
  __name: "DataTableCheckbox",
  props: {
    modelValue: { type: Boolean },
    disabled: { type: Boolean }
  },
  emits: ["update:modelValue"],
  setup(u) {
    return (e, o) => (l(), s(f, { class: "group/cell relative flex min-w-8" }, {
      default: n(() => [
        t("input", {
          checked: e.modelValue,
          class: "peer absolute inset-0 opacity-0 cursor-pointer",
          disabled: e.disabled,
          tabindex: "-1",
          type: "checkbox",
          onChange: o[0] || (o[0] = (r) => e.$emit("update:modelValue", r.target.checked))
        }, null, 40, p),
        t("div", b, [
          c(d(i), {
            icon: "Checkmark",
            size: "xs",
            thickness: "2.5"
          })
        ])
      ]),
      _: 1
    }));
  }
});
export {
  x as default
};
