// @scalar/api-client/dist/components/DataTable/DataTableInputSelect.vue.js
import { defineComponent as z, computed as A, ref as c, watch as k, nextTick as N, openBlock as n, createElementBlock as m, withDirectives as T, withKeys as j, vModelText as I, createBlock as i, unref as l, withCtx as r, Fragment as E, renderList as K, createElementVNode as p, normalizeClass as R, createVNode as d, createTextVNode as $, toDisplayString as y, createCommentVNode as V } from "vue";
import { ScalarDropdown as F, ScalarDropdownItem as C, ScalarIcon as f, ScalarDropdownDivider as L, ScalarButton as M } from "@scalar/components";
const U = { class: "w-full" }, W = { class: "flex items-center justify-center h-4 w-4" }, q = /* @__PURE__ */ p("span", null, "Add value", -1), O = /* @__PURE__ */ z({
  __name: "DataTableInputSelect",
  props: {
    modelValue: {},
    value: {},
    canAddCustomValue: { type: Boolean, default: !0 }
  },
  emits: ["update:modelValue"],
  setup(_, { emit: g }) {
    const v = _, h = g, S = A(() => v.value ?? []), o = c(v.modelValue.toString()), s = c(!1), a = c(""), x = c(null);
    k(a, (e) => {
      h("update:modelValue", e);
    });
    const w = (e) => {
      o.value = e, h("update:modelValue", e), s.value = !1;
    }, D = () => {
      a.value.trim() && (w(a.value), a.value = "");
    }, B = () => {
      a.value.trim() || (o.value = "", s.value = !1);
    }, b = (e) => o.value === e;
    return k(s, (e) => {
      e && N(() => {
        var u;
        (u = x.value) == null || u.focus();
      });
    }), (e, u) => (n(), m("div", U, [
      s.value ? T((n(), m("input", {
        key: 0,
        ref_key: "inputRef",
        ref: x,
        "onUpdate:modelValue": u[0] || (u[0] = (t) => a.value = t),
        class: "border-none text-c-1 min-w-0 w-full px-2 py-1.5 outline-none",
        placeholder: "Value",
        type: "text",
        onBlur: B,
        onKeyup: j(D, ["enter"])
      }, null, 544)), [
        [I, a.value]
      ]) : (n(), i(l(F), {
        key: 1,
        resize: "",
        value: o.value
      }, {
        items: r(() => [
          (n(!0), m(E, null, K(S.value, (t) => (n(), i(l(C), {
            key: t,
            class: "flex gap-1.5 group/item items-center whitespace-nowrap text-ellipsis overflow-hidden",
            value: t,
            onClick: (G) => w(t)
          }, {
            default: r(() => [
              p("div", {
                class: R([
                  "flex items-center justify-center rounded-full p-[3px] w-4 h-4 group-hover/item:shadow-border",
                  b(t) ? "bg-blue text-b-1" : "text-transparent"
                ])
              }, [
                d(l(f), {
                  class: "size-2.5",
                  icon: "Checkmark",
                  thickness: "3.5"
                })
              ], 2),
              $(" " + y(t), 1)
            ]),
            _: 2
          }, 1032, ["value", "onClick"]))), 128)),
          d(l(L)),
          e.canAddCustomValue ? (n(), i(l(C), {
            key: 0,
            class: "flex items-center gap-1.5",
            onClick: u[1] || (u[1] = (t) => s.value = !0)
          }, {
            default: r(() => [
              p("div", W, [
                d(l(f), {
                  class: "h-2.5",
                  icon: "Add"
                })
              ]),
              q
            ]),
            _: 1
          })) : V("", !0)
        ]),
        default: r(() => [
          d(l(M), {
            class: "gap-1.5 font-normal h-full justify-start px-2 py-1.5",
            fullWidth: "",
            variant: "ghost"
          }, {
            default: r(() => [
              p("span", null, y(o.value || "Select a value"), 1),
              o.value ? V("", !0) : (n(), i(l(f), {
                key: 0,
                icon: "ChevronDown",
                size: "xs"
              }))
            ]),
            _: 1
          })
        ]),
        _: 1
      }, 8, ["value"]))
    ]));
  }
});
export {
  O as default
};
