// @scalar/api-client/dist/components/DataTable/DataTableInput.vue2.js
import { defineComponent as h, ref as d, computed as k, openBlock as o, createBlock as i, normalizeClass as b, withCtx as B, createElementBlock as r, renderSlot as p, createCommentVNode as s, createElementVNode as m, Fragment as C, mergeProps as $, unref as x } from "vue";
import { ScalarIconButton as q } from "@scalar/components";
import A from "./DataTableCell.vue.js";
import E from "./DataTableInputSelect.vue.js";
const F = {
  key: 0,
  class: "text-c-2 flex min-w-[120px] items-center border-r-1/2 pl-2 pr-0"
}, I = { class: "row-1" }, O = ["id", "max", "min", "readOnly", "required", "type", "value"], S = {
  key: 0,
  class: "absolute centered-y right-0 pt-px pr-2 text-xxs text-c-3 bg-b-1 shadow-[-8px_0_4px_var(--scalar-background-1)] opacity-100 duration-150 transition-opacity peer-focus:opacity-0"
}, P = {
  key: 1,
  class: "absolute centered-y right-7 text-orange text-xs"
}, _ = /* @__PURE__ */ h({
  inheritAttrs: !1,
  __name: "DataTableInput",
  props: {
    id: {},
    type: {},
    containerClass: {},
    required: { type: Boolean, default: !1 },
    modelValue: {},
    canAddCustomEnumValue: { type: Boolean, default: !0 },
    readOnly: { type: Boolean, default: !1 },
    enum: {},
    min: {},
    max: {}
  },
  emits: ["update:modelValue", "inputFocus", "inputBlur", "selectVariable"],
  setup(c, { emit: f }) {
    const a = c, n = f, l = d(!0), y = d(""), v = d(!1), w = () => {
      v.value || n("inputBlur");
    }, V = k(
      () => a.type === "password" ? l.value ? "password" : "text" : a.type ?? "text"
    ), g = (e) => {
      const t = e.target;
      y.value = t.value, n("update:modelValue", t.value);
    };
    return (e, t) => (o(), i(A, {
      class: b(["relative row", e.containerClass])
    }, {
      default: B(() => [
        e.$slots.default ? (o(), r("div", F, [
          p(e.$slots, "default", {}, void 0, !0)
        ])) : s("", !0),
        m("div", I, [
          a.enum && a.enum.length ? (o(), i(E, {
            key: 0,
            canAddCustomValue: e.canAddCustomEnumValue,
            modelValue: a.modelValue,
            value: a.enum,
            "onUpdate:modelValue": t[0] || (t[0] = (u) => n("update:modelValue", u))
          }, null, 8, ["canAddCustomValue", "modelValue", "value"])) : (o(), r(C, { key: 1 }, [
            m("input", $(e.$attrs, {
              id: e.id,
              autocomplete: "off",
              class: "border-none text-c-1 min-w-0 w-full peer px-2 py-1.5 outline-none",
              "data-1p-ignore": "",
              max: e.max,
              min: e.min,
              readOnly: e.readOnly,
              required: e.required,
              spellcheck: "false",
              type: V.value,
              value: e.modelValue,
              onBlur: w,
              onFocus: t[1] || (t[1] = (u) => n("inputFocus")),
              onInput: g
            }), null, 16, O),
            e.required ? (o(), r("div", S, " Required ")) : s("", !0)
          ], 64))
        ]),
        e.$slots.warning ? (o(), r("div", P, [
          p(e.$slots, "warning", {}, void 0, !0)
        ])) : s("", !0),
        p(e.$slots, "icon", {}, void 0, !0),
        e.type === "password" ? (o(), i(x(q), {
          key: 2,
          class: "-ml-.5 mr-1 h-6 w-6 self-center p-1.5",
          icon: l.value ? "Show" : "Hide",
          label: l.value ? "Show Password" : "Hide Password",
          onClick: t[2] || (t[2] = (u) => l.value = !l.value)
        }, null, 8, ["icon", "label"])) : s("", !0)
      ]),
      _: 3
    }, 8, ["class"]));
  }
});
export {
  _ as default
};
