// @scalar/components/dist/components/ScalarFloating/ScalarFloating.vue.js
import { defineComponent as w, ref as d, computed as t, openBlock as p, createElementBlock as z, Fragment as S, createElementVNode as r, normalizeClass as b, renderSlot as m, createBlock as k, Teleport as B, normalizeStyle as R, unref as f, createCommentVNode as C } from "vue";
import { getSideAxis as c } from "@floating-ui/utils";
import { useFloating as E, autoUpdate as F, offset as _, flip as $, shift as A } from "@floating-ui/vue";
import { useResizeWithTarget as N } from "./useResizeWithTarget.js";
const O = { class: "scalar-app" }, H = /* @__PURE__ */ w({
  inheritAttrs: !1,
  __name: "ScalarFloating",
  props: {
    placement: {},
    resize: { type: Boolean },
    middleware: {},
    isOpen: { type: Boolean },
    teleport: { type: [Boolean, String] }
  },
  setup(u) {
    const o = u, n = d(null), l = d(null), i = t(
      () => {
        var e, a;
        return (((a = (e = l.value) == null ? void 0 : e.children) == null ? void 0 : a[0]) || l.value) ?? void 0;
      }
    ), s = N(i, {
      enabled: t(() => o.resize)
    }), g = t(
      () => c(o.placement || "bottom") === "y" ? s.width.value : void 0
    ), h = t(
      () => c(o.placement || "bottom") === "x" ? s.height.value : void 0
    ), { floatingStyles: v, middlewareData: y } = E(i, n, {
      placement: t(() => o.placement),
      whileElementsMounted: F,
      middleware: t(() => [
        _(5),
        $(),
        A(),
        ...o.middleware ?? []
      ])
    });
    return (e, a) => (p(), z(S, null, [
      r("div", {
        ref_key: "wrapperRef",
        ref: l,
        class: b({ contents: !!e.$slots.default })
      }, [
        m(e.$slots, "default")
      ], 2),
      e.isOpen ? (p(), k(B, {
        key: 0,
        disabled: !e.teleport,
        to: typeof e.teleport == "string" ? e.teleport : "body"
      }, [
        r("div", O, [
          r("div", {
            ref_key: "floatingRef",
            ref: n,
            class: "relative z-overlay",
            style: R(f(v))
          }, [
            m(e.$slots, "floating", {
              data: f(y),
              height: h.value,
              width: g.value
            })
          ], 4)
        ])
      ], 8, ["disabled", "to"])) : C("", !0)
    ], 64));
  }
});
export {
  H as default
};
