// @scalar/components/dist/components/ScalarFloating/ScalarFloating.vue.js
import { defineComponent as b, ref as d, computed as a, createElementBlock as x, openBlock as s, Fragment as z, createElementVNode as m, createBlock as S, createCommentVNode as $, normalizeClass as k, renderSlot as c, unref as i, withCtx as B, normalizeStyle as E } from "vue";
import { getSideAxis as u } from "@floating-ui/utils";
import { useFloating as R, autoUpdate as C, offset as F, flip as M, shift as H, size as T } from "@floating-ui/vue";
import { useResizeWithTarget as N } from "./useResizeWithTarget.js";
import W from "../ScalarTeleport/ScalarTeleport.vue.js";
const D = /* @__PURE__ */ b({
  inheritAttrs: !1,
  __name: "ScalarFloating",
  props: {
    placement: {},
    offset: { type: [Number, Object, Function], default: 5 },
    resize: { type: Boolean, default: !1 },
    target: {},
    middleware: { default: () => [] },
    teleport: { type: [Boolean, String] }
  },
  setup(e, { expose: g }) {
    const r = d(null), n = d(null), o = a(() => {
      if (typeof window < "u" && n.value) {
        if (typeof e.target == "string") {
          const l = document.getElementById(e.target);
          if (l)
            return l;
          console.warn(`ScalarFloating: Target with id="${e.target}" not found`);
        } else if (e.target instanceof HTMLElement)
          return e.target;
        const t = n.value.children?.[0];
        return t instanceof HTMLElement ? t : n.value;
      }
    }), f = N(o, {
      enabled: a(() => e.resize)
    }), h = a(
      () => u(e.placement ?? "bottom") === "y" ? f.width.value : void 0
    ), y = a(
      () => u(e.placement ?? "bottom") === "x" ? f.height.value : void 0
    ), { floatingStyles: v, middlewareData: w } = R(o, r, {
      placement: a(() => e.placement ?? "bottom"),
      whileElementsMounted: C,
      middleware: a(() => [
        F(e.offset),
        M(),
        H({ padding: 10 }),
        T({
          apply({ availableWidth: t, availableHeight: l, elements: p }) {
            Object.assign(p.floating.style, {
              maxWidth: `${Math.max(0, t) - 20}px`,
              maxHeight: `${Math.max(0, l) - 20}px`
            });
          }
        }),
        ...e.middleware
      ])
    });
    return g({
      /** The resolved target element */
      targetRef: o
    }), (t, l) => (s(), x(z, null, [
      m("div", {
        ref_key: "wrapperRef",
        ref: n,
        class: k({ contents: !!t.$slots.default })
      }, [
        c(t.$slots, "default")
      ], 2),
      t.$slots.floating ? (s(), S(i(W), {
        key: 0,
        disabled: !e.teleport,
        to: typeof e.teleport == "string" ? e.teleport : void 0
      }, {
        default: B(() => [
          m("div", {
            ref_key: "floatingRef",
            ref: r,
            class: "relative z-context",
            style: E(i(v))
          }, [
            c(t.$slots, "floating", {
              data: i(w),
              height: y.value,
              width: h.value
            })
          ], 4)
        ]),
        _: 3
      }, 8, ["disabled", "to"])) : $("", !0)
    ], 64));
  }
});
export {
  D as default
};
