// @scalar/api-client/dist/components/ViewLayout/ViewLayoutCollapse.vue.js
import { defineComponent as c, openBlock as s, createBlock as p, unref as o, withCtx as n, createVNode as i, createElementVNode as r, renderSlot as l, createElementBlock as a, toDisplayString as d, createCommentVNode as u } from "vue";
import { Disclosure as f, DisclosureButton as m, DisclosurePanel as h } from "@headlessui/vue";
import { ScalarIcon as b } from "@scalar/components";
const x = { class: "flex flex-1 items-center gap-1.5" }, _ = {
  key: 0,
  class: "bg-b-2 text-c-2 text-3xs inline-flex h-4 w-4 items-center justify-center rounded-full font-semibold"
}, g = { class: "flex items-center gap-2" }, v = {
  key: 0,
  class: "ui-not-open:invisible -my-1 flex items-center gap-2"
}, D = /* @__PURE__ */ c({
  __name: "ViewLayoutCollapse",
  props: {
    defaultOpen: { type: Boolean, default: !0 },
    itemCount: { default: 0 }
  },
  setup(y) {
    return (e, C) => (s(), p(o(f), {
      as: "div",
      class: "focus-within:bg-b-2 focus-within:text-c-1 text-c-2 rounded request-item ui-not-open:bg-transparent ui-open:pb-1 ui-open:mb-3 ui-not-open:mb-0 ui-not-open:pb-0",
      defaultOpen: e.defaultOpen
    }, {
      default: n(({ open: t }) => [
        i(o(m), { class: "hover:text-c-1 group flex w-full items-center gap-1.5 overflow-hidden py-1.5 text-sm font-medium px-1.5" }, {
          default: n(() => [
            i(o(b), {
              class: "text-c-3 group-hover:text-c-1 ui-open:rotate-90 ui-not-open:rotate-0",
              icon: "ChevronRight",
              size: "sm",
              thickness: "2.5"
            }),
            r("div", x, [
              l(e.$slots, "title", { open: t }),
              !t && e.itemCount ? (s(), a("span", _, d(e.itemCount), 1)) : u("", !0)
            ]),
            r("div", g, [
              e.$slots.actions ? (s(), a("div", v, [
                l(e.$slots, "actions", { open: t })
              ])) : u("", !0)
            ])
          ]),
          _: 2
        }, 1024),
        i(o(h), { class: "rounded-b" }, {
          default: n(() => [
            l(e.$slots, "default", { open: t })
          ]),
          _: 2
        }, 1024)
      ]),
      _: 3
    }, 8, ["defaultOpen"]));
  }
});
export {
  D as default
};
