// @scalar/components/dist/components/ScalarLoading/ScalarLoading.vue2.js
import { defineComponent as d, useCssVars as r, openBlock as i, createElementBlock as l, normalizeClass as o, unref as c, createElementVNode as h, createCommentVNode as n, createStaticVNode as p, reactive as g } from "vue";
import { cx as v } from "../../cva.js";
const m = /* @__PURE__ */ p('<path class="svg-path svg-check-mark" d="m 0 60 l 30 30 l 70 -80" data-v-ece8e3a4></path><path class="svg-path svg-x-mark" d="m 50 50 l 40 -40" data-v-ece8e3a4></path><path class="svg-path svg-x-mark" d="m 50 50 l 40 40" data-v-ece8e3a4></path><path class="svg-path svg-x-mark" d="m 50 50 l -40 -40" data-v-ece8e3a4></path><path class="svg-path svg-x-mark" d="m 50 50 l -40 40" data-v-ece8e3a4></path>', 5), u = {
  key: 0,
  class: "circular-loader"
};
function k() {
  return g({
    isValid: !1,
    isInvalid: !1,
    isLoading: !1,
    startLoading() {
      this.isLoading = !0;
    },
    stopLoading() {
      this.isLoading = !1;
    },
    validate(e = 800, a) {
      this.isInvalid = !1, this.isValid = !0;
      const t = a ? e - 300 : e;
      return new Promise(
        (s) => setTimeout(a ? () => this.clear().then(() => s(!0)) : () => s(!0), t)
      );
    },
    invalidate(e = 1100, a) {
      this.isValid = !1, this.isInvalid = !0;
      const t = a ? e - 300 : e;
      return new Promise(
        (s) => setTimeout(a ? () => this.clear().then(() => s(!0)) : () => s(!0), t)
      );
    },
    clear(e = 300) {
      return this.isValid = !1, this.isInvalid = !1, this.isLoading = !1, new Promise((a) => {
        setTimeout(() => {
          a(!0);
        }, e);
      });
    }
  });
}
const S = /* @__PURE__ */ d({
  __name: "ScalarLoading",
  props: {
    loadingState: {},
    size: { default: "24px" }
  },
  setup(e) {
    return r((a) => ({
      "64dccfe9": a.size
    })), (a, t) => a.loadingState ? (i(), l("div", {
      key: 0,
      class: o(c(v)("loader-wrapper"))
    }, [
      (i(), l("svg", {
        class: o(["svg-loader", {
          "icon-is-valid": a.loadingState.isValid,
          "icon-is-invalid": a.loadingState.isInvalid
        }]),
        viewBox: "0 0 100 100",
        xmlns: "http://www.w3.org/2000/svg",
        "xmlns:xlink": "http://www.w3.org/1999/xlink"
      }, [
        m,
        a.loadingState.isLoading ? (i(), l("g", u, [
          h("circle", {
            class: o(["loader-path", {
              "loader-path-off": a.loadingState.isValid || a.loadingState.isInvalid
            }]),
            cx: "50",
            cy: "50",
            fill: "none",
            r: "20",
            "stroke-width": "2"
          }, null, 2)
        ])) : n("", !0)
      ], 2))
    ], 2)) : n("", !0);
  }
});
export {
  S as default,
  k as useLoadingState
};
