// @scalar/components/dist/components/ScalarLoading/ScalarLoading.vue2.js
import { defineComponent as c, createElementBlock as e, createCommentVNode as n, openBlock as t, normalizeProps as m, mergeProps as v, unref as l, normalizeClass as i, createStaticVNode as p, createElementVNode as d } from "vue";
import { useBindCx as h, cva as f } from "@scalar/use-hooks/useBindCx";
const g = { class: "circular-loader" }, x = {}, w = /* @__PURE__ */ c({
  ...x,
  inheritAttrs: !1,
  __name: "ScalarLoading",
  props: {
    loader: {},
    size: {}
  },
  setup(a) {
    const { cx: r } = h(), o = f({
      variants: {
        size: {
          xs: "size-3",
          sm: "size-3.5",
          md: "size-4",
          lg: "size-5",
          xl: "size-6",
          "2xl": "size-8",
          "3xl": "size-10",
          full: "size-full"
        }
      },
      defaultVariants: {
        size: "full"
      }
    });
    return (z, s) => a.loader ? (t(), e("div", m(v({ key: 0 }, l(r)("loader-wrapper", l(o)({ size: a.size })))), [
      (t(), e("svg", {
        class: i(["svg-loader", {
          "icon-is-valid": a.loader.isValid,
          "icon-is-invalid": a.loader.isInvalid
        }]),
        viewBox: "0 0 100 100",
        xmlns: "http://www.w3.org/2000/svg",
        "xmlns:xlink": "http://www.w3.org/1999/xlink"
      }, [
        s[0] || (s[0] = p('<path class="svg-path svg-check-mark" d="m 0 60 l 30 30 l 70 -80" data-v-27df5cd8></path><path class="svg-path svg-x-mark" d="m 50 50 l 40 -40" data-v-27df5cd8></path><path class="svg-path svg-x-mark" d="m 50 50 l 40 40" data-v-27df5cd8></path><path class="svg-path svg-x-mark" d="m 50 50 l -40 -40" data-v-27df5cd8></path><path class="svg-path svg-x-mark" d="m 50 50 l -40 40" data-v-27df5cd8></path>', 5)),
        d("g", g, [
          d("circle", {
            class: i(["loader-path", {
              "loader-path-off": !a.loader.isLoading
            }]),
            cx: "50",
            cy: "50",
            fill: "none",
            r: "20",
            "stroke-width": "3"
          }, null, 2)
        ])
      ], 2))
    ], 16)) : n("", !0);
  }
});
export {
  w as default
};
