// @scalar/components/dist/components/ScalarLoading/useLoadingState.js
import { reactive as l } from "vue";
function r() {
  return l({
    isValid: !1,
    isInvalid: !1,
    isLoading: !1,
    isActive: !1,
    start() {
      this.isLoading = !0, this.isActive = !0, this.isInvalid = !1, this.isValid = !1;
    },
    validate(t = {}) {
      const { duration: i = 1100, persist: s = !1 } = t;
      this.isLoading = !1, this.isInvalid = !1, this.isValid = !0, this.isActive = !0;
      const e = s ? i : i - 300;
      return new Promise(
        (a) => setTimeout(s ? () => a() : () => this.clear().then(() => a()), e)
      );
    },
    invalidate(t = {}) {
      const { duration: i = 1100, persist: s = !1 } = t ?? {};
      this.isLoading = !1, this.isValid = !1, this.isInvalid = !0, this.isActive = !0;
      const e = s ? i : i - 300;
      return new Promise(
        (a) => setTimeout(s ? () => a() : () => this.clear().then(() => a()), e)
      );
    },
    clear(t = {}) {
      const { duration: i = 300 } = t;
      return this.isValid = !1, this.isInvalid = !1, this.isLoading = !1, new Promise((s) => {
        setTimeout(() => {
          this.isActive = !1, s();
        }, i);
      });
    }
  });
}
export {
  r as useLoadingState
};
