// @scalar/components/dist/components/ScalarCodeBlock/ScalarCodeBlockCopy.vue2.js
import { defineComponent as p, ref as d, createElementBlock as n, openBlock as l, normalizeClass as c, unref as t, createVNode as s, createCommentVNode as m, withCtx as y } from "vue";
import { ScalarIconCopy as C, ScalarIconCheck as h } from "@scalar/icons";
import { cva as v, cx as i } from "@scalar/use-hooks/useBindCx";
import { useClipboard as b } from "@scalar/use-hooks/useClipboard";
import k from "../ScalarButton/ScalarButton.vue.js";
const x = {
  key: 0,
  class: "sr-only",
  role: "alert"
}, T = /* @__PURE__ */ p({
  __name: "ScalarCodeBlockCopy",
  props: {
    content: {},
    controls: {},
    class: {}
  },
  setup(o) {
    const { copyToClipboard: r } = b(), e = d(!1), u = async () => {
      if (e.value)
        return;
      const a = typeof o.content == "string" ? o.content : JSON.stringify(o.content, null, 2);
      await r(a), e.value = !0, setTimeout(() => {
        e.value = !1;
      }, 1200);
    }, f = v({
      base: "absolute top-0 right-0 h-8 w-8 p-0 flex items-center justify-center brightness-lifted bg-inherit rounded focus-visible:opacity-100 -outline-offset-1",
      variants: {
        showCopied: {
          true: "text-c-1",
          false: "text-c-3 hover:text-c-1"
        }
      }
    });
    return (a, g) => (l(), n("div", {
      class: c(
        t(i)(
          "scalar-code-copy opacity-0 group-hover/code-block:opacity-100 sticky flex inset-0 justify-end items-start bg-inherit",
          o.class
        )
      )
    }, [
      s(t(k), {
        "aria-controls": o.controls,
        "aria-label": "Copy",
        class: c(t(i)(t(f)({ showCopied: e.value }))),
        type: "button",
        variant: "ghost",
        onClick: u
      }, {
        default: y(() => [
          s(t(C), {
            class: c(["copy-icon size-4", { copied: e.value }])
          }, null, 8, ["class"]),
          s(t(h), {
            class: c(["check-icon size-4", { visible: e.value }])
          }, null, 8, ["class"])
        ]),
        _: 1
      }, 8, ["aria-controls", "class"]),
      e.value ? (l(), n("div", x, " Copied ")) : m("", !0)
    ], 2));
  }
});
export {
  T as default
};
