// @scalar/components/dist/components/ScalarCodeBlock/ScalarCodeBlock.vue.js
import { defineComponent as s, useId as u, computed as n, createElementBlock as d, openBlock as a, createBlock as m, createCommentVNode as p, createElementVNode as f, unref as o } from "vue";
import { syntaxHighlight as h, standardLanguages as C } from "@scalar/code-highlight";
import { prettyPrintJson as g } from "@scalar/oas-utils/helpers";
import y from "./ScalarCodeBlockCopy.vue.js";
const k = {
  class: "scalar-code-block group/code-block custom-scroll min-h-12 focus-visible:outline",
  tabindex: "0"
}, v = ["id", "innerHTML"], L = /* @__PURE__ */ s({
  __name: "ScalarCodeBlock",
  props: {
    content: {},
    prettyPrintedContent: {},
    lang: { default: "plaintext" },
    lineNumbers: { type: Boolean, default: !1 },
    hideCredentials: {},
    copy: { type: Boolean, default: !0 }
  },
  setup(e) {
    const l = u(), t = n(
      () => e.prettyPrintedContent || g(e.content ?? "")
    ), r = n(() => h(t.value, {
      lang: e.lang.trim(),
      languages: C,
      lineNumbers: e.lineNumbers,
      maskCredentials: e.hideCredentials
    }).slice(5, -6)), c = n(() => t.value !== null && t.value !== "null" && t.value !== "404 Not Found");
    return (i, b) => (a(), d("div", k, [
      e.copy && c.value ? (a(), m(o(y), {
        key: 0,
        content: t.value,
        controls: o(l)
      }, null, 8, ["content", "controls"])) : p("", !0),
      f("pre", {
        id: o(l),
        class: "scalar-codeblock-pre",
        innerHTML: r.value
      }, null, 8, v)
    ]));
  }
});
export {
  L as default
};
