// @scalar/components/dist/components/ScalarCodeBlock/ScalarCodeBlock.vue.js
import { defineComponent as u, computed as s, openBlock as l, createElementBlock as c, createElementVNode as t, unref as o, createVNode as _, createCommentVNode as h } from "vue";
import { syntaxHighlight as C, standardLanguages as f } from "@scalar/code-highlight";
import { prettyPrintJson as r } from "../../helpers/oas-utils.js";
import { useClipboard as b } from "./useClipboard.js";
import g from "../ScalarIcon/ScalarIcon.vue.js";
const y = { class: "scalar-code-block custom-scroll" }, k = { class: "scalar-code-copy" }, v = /* @__PURE__ */ t("span", { class: "sr-only" }, "Copy content", -1), N = { class: "scalar-code-scroll custom-scroll" }, x = ["innerHTML"], M = /* @__PURE__ */ u({
  __name: "ScalarCodeBlock",
  props: {
    content: {},
    lang: { default: "plaintext" },
    lineNumbers: { type: Boolean, default: !1 },
    hideCredentials: {}
  },
  setup(a) {
    const e = a, i = s(() => C(r(e.content), {
      lang: e.lang.trim(),
      languages: f,
      lineNumbers: e.lineNumbers,
      maskCredentials: e.hideCredentials
    }).slice(5, -6)), { copyToClipboard: d } = b(), m = s(() => e.content !== null && e.content !== "null");
    return (p, n) => (l(), c("div", y, [
      t("div", k, [
        m.value ? (l(), c("button", {
          key: 0,
          class: "copy-button",
          type: "button",
          onClick: n[0] || (n[0] = (B) => o(d)(o(r)(e.content)))
        }, [
          v,
          _(o(g), {
            icon: "Clipboard",
            size: "md"
          })
        ])) : h("", !0)
      ]),
      t("div", N, [
        t("pre", {
          class: "scalar-codeblock-pre",
          innerHTML: i.value
        }, null, 8, x)
      ])
    ]));
  }
});
export {
  M as default
};
