// @scalar/api-client/dist/components/AddressBar/AddressBar.vue2.js
import { defineComponent as L, ref as n, watch as O, onMounted as $, onBeforeUnmount as K, unref as l, openBlock as N, createElementBlock as T, createElementVNode as a, createVNode as i, withCtx as y, normalizeClass as B, normalizeStyle as A, createCommentVNode as D, pushScopeId as Q, popScopeId as W } from "vue";
import j from "../CodeInput/CodeInput.vue.js";
import { useWorkspace as F } from "../../store/workspace.js";
import { Listbox as G } from "@headlessui/vue";
import { ScalarButton as J, ScalarIcon as P } from "@scalar/components";
import { REQUEST_METHODS as X } from "@scalar/oas-utils/helpers";
import { isMacOS as Y } from "@scalar/use-tooltip";
import { useMagicKeys as Z, whenever as ee } from "@vueuse/core";
import te from "../HttpMethod/HttpMethod.vue.js";
import le from "./AddressBarHistory.vue.js";
import { executeRequestBus as m } from "../../libs/event-busses/execute-request-bus.js";
import { requestStatusBus as oe } from "../../libs/event-busses/request-status-bus.js";
import { hotKeyBus as k } from "../../libs/event-busses/hot-keys-bus.js";
const v = (f) => (Q("data-v-28ee5506"), f = f(), W(), f), ae = {
  key: 0,
  class: "order-last lg:order-none lg:w-auto w-full"
}, re = { class: "m-auto flex basis-1/2 flex-row items-center" }, se = { class: "pointer-events-none absolute left-0 top-0 z-10 block h-full w-full overflow-hidden" }, ne = { class: "flex gap-1" }, ie = { class: "codemirror-bg-switcher scroll-timeline-x scroll-timeline-x-hidden relative flex w-full" }, ue = /* @__PURE__ */ v(() => /* @__PURE__ */ a("div", { class: "fade-left" }, null, -1)), de = /* @__PURE__ */ v(() => /* @__PURE__ */ a("div", { class: "fade-right" }, null, -1)), fe = /* @__PURE__ */ v(() => /* @__PURE__ */ a("span", { class: "text-xxs relative z-10 lg:flex hidden" }, "Send", -1)), Ee = /* @__PURE__ */ L({
  __name: "AddressBar",
  setup(f) {
    const {
      activeRequest: t,
      activeExample: c,
      isReadOnly: p,
      requestMutators: h,
      requestExampleMutators: S,
      requestsHistory: E
    } = F(), x = n(E.value[0]), b = n(null), g = Z();
    ee(
      Y() ? g.meta_enter : g.ctrl_enter,
      () => m.emit()
    );
    const C = (e) => {
      !t.value || t.value.path === e || h.edit(t.value.uid, "path", e);
    };
    O(
      () => {
        var e;
        return (e = t.value) == null ? void 0 : e.path;
      },
      (e) => {
        t.value && C(e);
      }
    );
    const r = n(100), _ = n(0), s = n(!1), u = n();
    function q() {
      s.value ? r.value -= (r.value - 15) / 60 : r.value -= _.value / 20, r.value <= 0 && (clearInterval(u.value), u.value = void 0, r.value = 100, s.value = !1);
    }
    function R() {
      s.value || (s.value = !0, u.value = setInterval(q, 20));
    }
    function I() {
      _.value = r.value, s.value = !1;
    }
    function V() {
      clearInterval(u.value), u.value = void 0, r.value = 100, s.value = !1;
    }
    oe.on((e) => {
      e === "start" && R(), e === "stop" && I(), e === "abort" && V();
    });
    function M(e) {
      t.value && h.edit(t.value.uid, "method", e);
    }
    function z() {
      if (!t.value) return;
      const { method: e } = t.value;
      return X[e].backgroundColor;
    }
    const U = (e) => {
      c.value && S.edit(c.value.uid, "url", e);
    }, w = (e) => {
      var o;
      e.focusAddressBar && ((o = b.value) == null || o.focus());
    };
    return $(() => k.on(w)), K(() => k.off(w)), (e, o) => l(t) && l(c) ? (N(), T("div", ae, [
      a("div", re, [
        i(l(G), {
          modelValue: x.value,
          "onUpdate:modelValue": o[2] || (o[2] = (d) => x.value = d)
        }, {
          default: y(({ open: d }) => [
            a("div", {
              class: B([
                "addressbar-bg-states text-xxs relative flex w-full lg:min-w-[720px] lg:max-w-[720px] order-last lg:order-none flex-1 flex-row items-stretch rounded-lg border-1/2 p-[3px]",
                { "rounded-b-none": d },
                { "border-transparent": d }
              ])
            }, [
              a("div", se, [
                a("div", {
                  class: B(["bg-mix-transparent bg-mix-amount-90 absolute left-0 top-0 h-full w-full", z()]),
                  style: A({ transform: `translate3d(-${r.value}%,0,0)` })
                }, null, 6)
              ]),
              a("div", ne, [
                i(te, {
                  isEditable: !l(p),
                  isSquare: "",
                  method: l(t).method,
                  onChange: M
                }, null, 8, ["isEditable", "method"])
              ]),
              a("div", ie, [
                ue,
                i(j, {
                  ref_key: "addressBarRef",
                  ref: b,
                  disableCloseBrackets: "",
                  disabled: l(p),
                  disableEnter: "",
                  disableTabIndent: "",
                  emitOnBlur: !1,
                  modelValue: l(c).url,
                  placeholder: "Enter URL to get started",
                  server: "",
                  onSubmit: o[0] || (o[0] = (H) => l(m).emit()),
                  "onUpdate:modelValue": U
                }, null, 8, ["disabled", "modelValue"]),
                de
              ]),
              i(le, { open: d }, null, 8, ["open"]),
              i(l(J), {
                class: "relative h-auto shrink-0 gap-1 overflow-hidden pl-2 pr-2.5 py-1 z-[1] font-bold",
                disabled: s.value,
                onClick: o[1] || (o[1] = (H) => l(m).emit())
              }, {
                default: y(() => [
                  i(l(P), {
                    class: "relative z-10 shrink-0 fill-current",
                    icon: "Play",
                    size: "xs"
                  }),
                  fe
                ]),
                _: 1
              }, 8, ["disabled"])
            ], 2)
          ]),
          _: 1
        }, 8, ["modelValue"])
      ])
    ])) : D("", !0);
  }
});
export {
  Ee as default
};
