// @scalar/api-client/dist/components/AddressBar/AddressBarHistory.vue.js
import { defineComponent as C, computed as w, openBlock as n, createElementBlock as d, Fragment as m, unref as t, createBlock as i, withCtx as p, createVNode as f, createCommentVNode as h, createElementVNode as r, normalizeClass as b, renderList as U, toDisplayString as l } from "vue";
import { formatMs as L } from "../../libs/formatters.js";
import { useWorkspace as q } from "../../store/workspace.js";
import { ListboxButton as y, ListboxOptions as B, ListboxOption as S } from "@headlessui/vue";
import { ScalarIcon as H } from "@scalar/components";
import { httpStatusCodes as P } from "@scalar/oas-utils/helpers";
import { useRouter as R } from "vue-router";
import z from "../HttpMethod/HttpMethod.vue.js";
import { getStatusCodeColor as E } from "./httpStatusCodeColors.js";
const M = { class: "min-w-0" }, N = { class: "min-w-0 truncate text-c-1" }, J = /* @__PURE__ */ C({
  __name: "AddressBarHistory",
  props: {
    open: { type: Boolean }
  },
  setup(V) {
    const { activeRequest: u, requestExampleMutators: x } = q(), g = R(), v = w(() => u.value.history.slice().reverse());
    function _(o) {
      const c = new URL(o), s = new URLSearchParams(c.search).get("scalar_url");
      return s ? new URL(s).href : c.href;
    }
    function k(o) {
      u.value.uid !== o.request.requestUid && g.push(`/request/${o.request.requestUid}`), x.set({ ...o.request });
    }
    return (o, c) => (n(), d(m, null, [
      t(u).history.length ? (n(), i(t(y), {
        key: 0,
        class: "adressbar-history-button mr-1 rounded p-1.5 text-c-3 focus:text-c-1"
      }, {
        default: p(() => [
          f(t(H), {
            icon: "History",
            size: "sm",
            thickness: "2.25"
          })
        ]),
        _: 1
      })) : h("", !0),
      r("div", {
        class: b([
          "absolute left-0 top-[33px] w-full rounded before:pointer-events-none before:absolute before:left-0 before:top-[-33px] before:h-[calc(100%+33px)] before:w-full before:rounded z-50",
          { "before:shadow-lg": o.open }
        ])
      }, [
        f(t(B), { class: "bg-b-1 custom-scroll bg-mix-transparent bg-mix-amount-30 max-h-[300px] rounded-b p-[3px] pt-0 backdrop-blur grid grid-cols-[44px,1fr,repeat(3,auto)] items-center" }, {
          default: p(() => [
            (n(!0), d(m, null, U(v.value, (e, s) => (n(), i(t(S), {
              key: s,
              class: "contents font-code text-sm *:rounded-none first:*:rounded-l last:*:rounded-r *:h-8 *:hover:bg-b-2 *:flex *:items-center *:cursor-pointer *:px-1.5 text-c-2 font-medium",
              value: s,
              onClick: (a) => k(e)
            }, {
              default: p(() => {
                var a;
                return [
                  e.response.config.method ? (n(), i(z, {
                    key: 0,
                    class: "text-[11px]",
                    method: e.response.config.method
                  }, null, 8, ["method"])) : h("", !0),
                  r("div", M, [
                    r("div", N, l(_(e.response.config.url)), 1)
                  ]),
                  r("div", null, l(t(L)(e.response.duration)), 1),
                  r("div", {
                    class: b([t(E)(e.response.status).color])
                  }, l(e.response.status), 3),
                  r("div", null, l((a = t(P)[e.response.status]) == null ? void 0 : a.name), 1)
                ];
              }),
              _: 2
            }, 1032, ["value", "onClick"]))), 128))
          ]),
          _: 1
        })
      ], 2)
    ], 64));
  }
});
export {
  J as default
};
