// @scalar/components/dist/components/ScalarToggle/ScalarToggle.vue.js
import { defineComponent as d, mergeModels as i, useModel as u, createElementBlock as t, openBlock as l, normalizeClass as o, unref as n, createElementVNode as m, createCommentVNode as b, toDisplayString as f } from "vue";
import { cva as p, cx as h } from "@scalar/use-hooks/useBindCx";
const v = ["aria-checked", "aria-disabled"], g = {
  key: 0,
  class: "sr-only"
}, V = /* @__PURE__ */ d({
  __name: "ScalarToggle",
  props: /* @__PURE__ */ i({
    disabled: { type: Boolean },
    label: {}
  }, {
    modelValue: { type: Boolean, default: !1 },
    modelModifiers: {}
  }),
  emits: ["update:modelValue"],
  setup(e) {
    const r = e, a = u(e, "modelValue");
    function s() {
      r.disabled || (a.value = !a.value);
    }
    const c = p({
      base: "relative h-3.5 min-w-6 w-6 cursor-pointer rounded-full bg-b-3 transition-colors duration-300",
      variants: {
        checked: { true: "bg-c-accent" },
        disabled: { true: "cursor-not-allowed opacity-40" }
      }
    });
    return (k, x) => (l(), t("button", {
      "aria-checked": a.value,
      "aria-disabled": e.disabled,
      class: o(n(h)(n(c)({ checked: a.value, disabled: e.disabled }))),
      role: "switch",
      type: "button",
      onClick: s
    }, [
      m("div", {
        class: o(["absolute left-px top-px flex h-3 w-3 items-center justify-center rounded-full bg-b-1 text-c-accent transition-transform duration-300", { "translate-x-2.5": a.value }])
      }, null, 2),
      e.label ? (l(), t("span", g, f(e.label), 1)) : b("", !0)
    ], 10, v));
  }
});
export {
  V as default
};
