// @scalar/components/dist/components/ScalarToggle/ScalarToggleGroup.vue.js
import { defineComponent as d, mergeModels as n, useModel as s, createBlock as u, openBlock as t, unref as c, withCtx as r, createElementBlock as f, Fragment as _, renderList as i, createTextVNode as p, toDisplayString as V } from "vue";
import v from "./ScalarToggleInput.vue.js";
import g from "../ScalarForm/ScalarFormInputGroup.vue.js";
const k = {}, C = /* @__PURE__ */ d({
  ...k,
  __name: "ScalarToggleGroup",
  props: /* @__PURE__ */ n({
    options: { default: () => [] }
  }, {
    modelValue: { default: [] },
    modelModifiers: {}
  }),
  emits: ["update:modelValue"],
  setup(o) {
    const l = s(o, "modelValue");
    return (x, B) => (t(), u(c(g), null, {
      default: r(() => [
        (t(!0), f(_, null, i(o.options, (e) => (t(), u(v, {
          key: e.value,
          modelValue: l.value?.some(({ value: a }) => a === e.value),
          "onUpdate:modelValue": (a) => l.value = a ? [...l.value, e] : l.value.filter(({ value: m }) => m !== e.value)
        }, {
          default: r(() => [
            p(V(e.label), 1)
          ]),
          _: 2
        }, 1032, ["modelValue", "onUpdate:modelValue"]))), 128))
      ]),
      _: 1
    }));
  }
});
export {
  C as default
};
