// @scalar/api-reference/dist/features/ask-agent-button/AskAgentButton.vue2.js
import { defineComponent, ref, createElementBlock, createCommentVNode, unref, openBlock, createVNode, withDirectives, createElementVNode, withKeys, withModifiers, vModelText } from "vue";
import { ScalarIconSparkle, ScalarIconArrowUp } from "@scalar/icons";
import { useAgentContext } from "../../hooks/use-agent.js";
const _hoisted_1 = {
  key: 0,
  class: "agent-button-container flex"
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "AskAgentButton",
  setup(__props) {
    const agentContext = useAgentContext();
    const message = ref("");
    function handleSubmit() {
      agentContext.value?.openAgent(message.value);
      message.value = "";
    }
    return (_ctx, _cache) => {
      return unref(agentContext)?.agentEnabled.value ? (openBlock(), createElementBlock("div", _hoisted_1, [
        createVNode(unref(ScalarIconSparkle), {
          class: "size-3 shrink-0",
          weight: "fill"
        }),
        withDirectives(createElementVNode("input", {
          "onUpdate:modelValue": _cache[0] || (_cache[0] = ($event) => message.value = $event),
          class: "ask-agent-scalar-input",
          placeholder: "Ask AI Agent",
          onKeydown: _cache[1] || (_cache[1] = withKeys(withModifiers(($event) => handleSubmit(), ["stop"]), ["enter"]))
        }, null, 544), [
          [vModelText, message.value]
        ]),
        createElementVNode("button", {
          class: "ask-agent-scalar-send",
          type: "button",
          onClick: _cache[2] || (_cache[2] = ($event) => handleSubmit())
        }, [
          createVNode(unref(ScalarIconArrowUp), {
            class: "size-3",
            weight: "bold"
          })
        ])
      ])) : createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as default
};
