// @scalar/components/dist/components/ScalarSearchResults/ScalarSearchResultItem.vue.js
import { defineComponent as i, computed as d, useAttrs as m, openBlock as o, createElementBlock as r, createElementVNode as t, mergeProps as u, unref as a, renderSlot as s, createBlock as p, createCommentVNode as l } from "vue";
import { cx as f } from "../../cva.js";
import _ from "../ScalarIcon/ScalarIcon.vue.js";
const h = { class: "contents" }, v = { class: "flex h-fit items-center text-sm font-medium text-c-3 group-hover:text-c-1" }, x = /* @__PURE__ */ t("span", null, " ", -1), g = { class: "flex min-w-0 flex-1 flex-col gap-0.75" }, k = { class: "flex items-center gap-1" }, N = { class: "flex-1 truncate text-sm font-medium" }, $ = {
  key: 0,
  class: "text-sm text-c-2"
}, b = { class: "truncate text-sm text-c-2" }, C = /* @__PURE__ */ i({
  inheritAttrs: !1,
  __name: "ScalarSearchResultItem",
  props: {
    icon: {},
    active: { type: Boolean }
  },
  setup(y) {
    const n = d(() => {
      const { class: e, ...c } = m();
      return { className: e || "", rest: c };
    });
    return (e, c) => (o(), r("li", h, [
      t("a", u(n.value.rest, {
        class: a(f)(
          "group flex cursor-pointer gap-2.5 rounded px-3 py-1.5 no-underline hover:bg-b-2",
          {
            "bg-b-2": e.active
          },
          n.value.className
        )
      }), [
        t("div", v, [
          s(e.$slots, "icon", {}, () => [
            e.icon ? (o(), p(a(_), {
              key: 0,
              icon: e.icon,
              size: "sm"
            }, null, 8, ["icon"])) : l("", !0)
          ]),
          x
        ]),
        t("div", g, [
          t("div", k, [
            t("div", N, [
              s(e.$slots, "default")
            ]),
            e.$slots.addon ? (o(), r("div", $, [
              s(e.$slots, "addon")
            ])) : l("", !0)
          ]),
          t("div", b, [
            s(e.$slots, "description")
          ])
        ])
      ], 16)
    ]));
  }
});
export {
  C as default
};
