// @scalar/components/dist/components/ScalarSearchResults/ScalarSearchResultItem.vue.js
import { defineComponent as d, createElementBlock as s, openBlock as o, mergeProps as m, unref as l, createCommentVNode as n, createElementVNode as c, renderSlot as i, createBlock as u } from "vue";
import { useBindCx as f } from "@scalar/use-hooks/useBindCx";
import p from "../ScalarIcon/ScalarIconLegacyAdapter.vue.js";
const x = ["aria-selected"], h = {
  key: 0,
  class: "flex h-fit items-center text-sm font-medium text-c-3 group-hover:text-c-1"
}, b = { class: "flex min-w-0 flex-1 flex-col gap-0.75" }, k = { class: "flex items-center gap-1" }, g = { class: "flex-1 truncate zoomed:!whitespace-normal break-words text-base font-medium" }, v = {
  key: 0,
  class: "text-base text-c-2"
}, y = {
  key: 0,
  class: "truncate zoomed:!whitespace-normal break-words text-base text-c-2"
}, B = /* @__PURE__ */ d({
  inheritAttrs: !1,
  __name: "ScalarSearchResultItem",
  props: {
    icon: { type: [Object, Function] },
    selected: { type: Boolean }
  },
  setup(t) {
    const { cx: a } = f();
    return (e, r) => (o(), s("a", m(
      {
        "aria-selected": t.selected,
        role: "option",
        tabindex: "-1"
      },
      l(a)(
        "group flex cursor-pointer gap-2 rounded px-2 py-1.5 no-underline hover:bg-b-2",
        { "bg-b-2": t.selected }
      )
    ), [
      t.icon ? (o(), s("div", h, [
        i(e.$slots, "icon", {}, () => [
          t.icon ? (o(), u(l(p), {
            key: 0,
            class: "size-4",
            icon: t.icon
          }, null, 8, ["icon"])) : n("", !0)
        ]),
        r[0] || (r[0] = c("span", null, " ", -1))
      ])) : n("", !0),
      c("div", b, [
        c("div", k, [
          c("div", g, [
            i(e.$slots, "default")
          ]),
          e.$slots.addon ? (o(), s("div", v, [
            i(e.$slots, "addon")
          ])) : n("", !0)
        ]),
        e.$slots.description ? (o(), s("div", y, [
          i(e.$slots, "description")
        ])) : n("", !0)
      ])
    ], 16, x));
  }
});
export {
  B as default
};
