// @scalar/components/dist/components/ScalarSidebar/ScalarSidebarNestedItems.vue.js
import { defineComponent as h, mergeModels as f, useModel as T, useTemplateRef as x, ref as z, createElementBlock as m, openBlock as c, normalizeStyle as N, normalizeClass as A, renderSlot as o, createVNode as s, mergeProps as I, unref as i, createSlots as w, withCtx as l, createBlock as M, createCommentVNode as p, Transition as P, normalizeProps as R, guardReactiveProps as V, createTextVNode as q, nextTick as b } from "vue";
import { ScalarIconArrowRight as E, ScalarIconCaretLeft as L } from "@scalar/icons";
import v from "./ScalarSidebarButton.vue.js";
import O from "./ScalarSidebarItems.vue.js";
/* empty css                        */
import j from "./ScalarSidebarSpacer.vue.js";
import { findScrollContainer as k } from "./findScrollContainer.js";
import { useSidebarGroups as F } from "./useSidebarGroups.js";
import { useSidebarNestedItem as G } from "./useSidebarNestedItems.js";
import D from "../ScalarIcon/ScalarIconLegacyAdapter.vue.js";
const H = {
  key: 0,
  class: "absolute inset-0 translate-x-full"
}, J = {}, oe = /* @__PURE__ */ h({
  ...J,
  inheritAttrs: !1,
  __name: "ScalarSidebarNestedItems",
  props: /* @__PURE__ */ f({
    is: {},
    icon: { type: [Object, Function] },
    active: { type: Boolean },
    selected: { type: Boolean },
    disabled: { type: Boolean },
    indent: {},
    controlled: { type: Boolean },
    discrete: { type: Boolean }
  }, {
    open: { type: Boolean, default: !1 },
    openModifiers: {}
  }),
  emits: /* @__PURE__ */ f(["click", "back"], ["update:open"]),
  setup(t, { emit: y }) {
    const d = y, n = T(t, "open");
    G(n);
    const { level: $ } = F({ reset: !0 }), a = x("el"), r = z(0);
    function S() {
      r.value = k(a.value).scrollTop ?? 0, b(
        () => a.value?.querySelector('button:not([aria-expanded="true"])')?.focus({ preventScroll: !0 })
      );
    }
    function C() {
      b(() => {
        a.value?.querySelector("button")?.focus({ preventScroll: !0 }), k(a.value).scrollTop = r.value;
      });
    }
    const B = (e) => {
      d("click", e), t.controlled || (n.value = !0);
    }, g = (e) => {
      d("back", e), t.controlled || (n.value = !1);
    };
    return (e, u) => (c(), m("li", {
      ref_key: "el",
      ref: a,
      class: A(["group/item group/nested-items contents", { "group/nested-items-open": n.value }]),
      style: N({ "--nested-items-offset": `${r.value}px` })
    }, [
      o(e.$slots, "button", {}, () => [
        s(v, I({
          is: "button",
          "aria-expanded": n.value,
          disabled: t.disabled,
          indent: i($),
          selected: t.selected
        }, e.$attrs, { onClick: B }), w({
          aside: l(() => [
            o(e.$slots, "aside", {}, () => [
              s(i(E), { class: "size-4 text-sidebar-c-2" })
            ])
          ]),
          default: l(() => [
            o(e.$slots, "default")
          ]),
          _: 2
        }, [
          t.icon || e.$slots.icon ? {
            name: "icon",
            fn: l(() => [
              o(e.$slots, "icon", {}, () => [
                t.icon ? (c(), M(i(D), {
                  key: 0,
                  icon: t.icon,
                  weight: "bold"
                }, null, 8, ["icon"])) : p("", !0)
              ])
            ]),
            key: "0"
          } : void 0
        ]), 1040, ["aria-expanded", "disabled", "indent", "selected"])
      ]),
      s(P, {
        duration: 300,
        enterActiveClass: "top-(--nested-items-offset)",
        leaveActiveClass: "top-(--nested-items-offset)",
        onEnter: S,
        onLeave: C
      }, {
        default: l(() => [
          n.value ? (c(), m("div", H, [
            s(O, R(V(e.$attrs)), {
              default: l(() => [
                o(e.$slots, "back", {}, () => [
                  s(v, {
                    is: "button",
                    class: "text-sidebar-c-1 font-sidebar-active hover:text-sidebar-c-1",
                    onClick: g
                  }, {
                    icon: l(() => [
                      s(i(L), { class: "size-4 -m-px text-sidebar-c-2" })
                    ]),
                    default: l(() => [
                      o(e.$slots, "back-label", {}, () => [
                        u[0] || (u[0] = q("Back", -1))
                      ])
                    ]),
                    _: 3
                  })
                ]),
                s(j, { class: "h-3" }),
                o(e.$slots, "items")
              ]),
              _: 3
            }, 16)
          ])) : p("", !0)
        ]),
        _: 3
      })
    ], 6));
  }
});
export {
  oe as default
};
