// @scalar/components/dist/components/ScalarSidebar/ScalarSidebarButton.vue.js
import { defineComponent as c, createBlock as d, openBlock as s, resolveDynamicComponent as r, mergeProps as b, unref as i, withCtx as u, renderSlot as a, createElementBlock as f, createCommentVNode as l, createElementVNode as v, createVNode as m } from "vue";
import { cva as h, useBindCx as y } from "@scalar/use-hooks/useBindCx";
import B from "./ScalarSidebarIndent.vue.js";
import x from "../ScalarIcon/ScalarIconLegacyAdapter.vue.js";
const g = {
  key: 0,
  class: "h-[1lh] *:size-4 mr-1"
}, k = { class: "group/button-label flex-1 min-w-0" }, $ = {}, p = /* @__PURE__ */ c({
  ...$,
  inheritAttrs: !1,
  __name: "ScalarSidebarButton",
  props: {
    is: { default: "a" },
    icon: { type: [Object, Function] },
    active: { type: Boolean },
    selected: { type: Boolean },
    disabled: { type: Boolean },
    indent: { default: 0 }
  },
  setup(e) {
    const n = h({
      base: [
        "group/button peer/button flex items-stretch rounded p-2",
        "font-sidebar text-base/5  text-sidebar-c-2 no-underline wrap-break-word"
      ],
      variants: {
        active: { true: "text-sidebar-c-active font-sidebar-active" },
        disabled: { true: "cursor-auto" },
        selected: {
          true: "cursor-auto bg-sidebar-b-active text-sidebar-c-active font-sidebar-active"
        }
      },
      compoundVariants: [
        {
          selected: !1,
          disabled: !1,
          active: !1,
          class: "hover:bg-sidebar-b-hover hover:text-sidebar-c-hover"
        },
        {
          selected: !1,
          disabled: !1,
          active: !0,
          class: "hover:bg-sidebar-b-hover"
        }
      ],
      defaultVariants: { selected: !1, disabled: !1, active: !1 }
    }), { cx: o } = y();
    return (t, C) => (s(), d(r(e.is), b({
      "aria-selected": e.selected,
      type: e.is === "button" ? "button" : void 0
    }, i(o)(i(n)({ selected: e.selected, disabled: e.disabled, active: e.active }))), {
      default: u(() => [
        a(t.$slots, "indent", {}, () => [
          m(B, {
            class: "-my-2",
            disabled: e.disabled,
            indent: e.indent,
            selected: e.selected
          }, null, 8, ["disabled", "indent", "selected"])
        ]),
        e.icon || t.$slots.icon ? (s(), f("div", g, [
          a(t.$slots, "icon", {}, () => [
            e.icon ? (s(), d(i(x), {
              key: 0,
              icon: e.icon
            }, null, 8, ["icon"])) : l("", !0)
          ])
        ])) : l("", !0),
        v("div", k, [
          a(t.$slots, "default")
        ]),
        a(t.$slots, "aside")
      ]),
      _: 3
    }, 16, ["aria-selected", "type"]));
  }
});
export {
  p as default
};
