// @scalar/components/dist/components/ScalarSidebar/ScalarSidebarSearchInput.vue.js
import { defineComponent as u, mergeModels as f, ref as m, useModel as b, onMounted as p, createElementBlock as h, openBlock as l, normalizeProps as v, guardReactiveProps as g, unref as a, createVNode as x, withDirectives as k, createBlock as s, createCommentVNode as y, createElementVNode as S, mergeProps as C, vModelText as M, withModifiers as V } from "vue";
import { ScalarIconMagnifyingGlass as z, ScalarIconX as B } from "@scalar/icons";
import { useBindCx as A } from "@scalar/use-hooks/useBindCx";
import w from "../ScalarLoading/ScalarLoading.vue.js";
import E from "../ScalarIconButton/ScalarIconButton.vue.js";
const I = ["aria-label"], L = /* @__PURE__ */ u({
  inheritAttrs: !1,
  __name: "ScalarSidebarSearchInput",
  props: /* @__PURE__ */ f({
    autofocus: { type: Boolean },
    loader: {},
    label: {}
  }, {
    modelValue: {},
    modelModifiers: {}
  }),
  emits: ["update:modelValue"],
  setup(e) {
    const o = m(null), r = b(e, "modelValue");
    function n() {
      r.value = "", o.value && o.value.focus();
    }
    const { stylingAttrsCx: c, otherAttrs: i } = A();
    return p(() => e.autofocus && o.value?.focus()), (N, t) => (l(), h("label", v(g(
      a(c)(
        "flex items-center rounded border text-base has-[:focus-visible]:bg-sidebar-b-1 has-[:focus-visible]:outline h-8 gap-1 pl-2 pr-1.5",
        "bg-sidebar-b-search border-sidebar-border-search",
        r.value ? "text-sidebar-c-1" : "text-sidebar-c-search"
      )
    )), [
      x(a(z), { class: "text-sidebar-c-search size-4" }),
      k(S("input", C({
        ref_key: "inputRef",
        ref: o
      }, a(i), {
        "onUpdate:modelValue": t[0] || (t[0] = (d) => r.value = d),
        "aria-label": e.label ?? "Enter search query",
        autocapitalize: "off",
        autocomplete: "off",
        autocorrect: "off",
        class: "flex-1 appearance-none rounded border-none bg-transparent outline-none",
        placeholder: "Search...",
        spellcheck: "false",
        type: "search"
      }), null, 16, I), [
        [M, r.value]
      ]),
      e.loader && e.loader.isActive ? (l(), s(a(w), {
        key: 0,
        class: "mr-3 self-center",
        loader: e.loader,
        size: "md"
      }, null, 8, ["loader"])) : r.value ? (l(), s(a(E), {
        key: 1,
        class: "p-0.25 size-4",
        icon: a(B),
        label: "Clear Search",
        weight: "bold",
        onClick: V(n, ["stop", "prevent"])
      }, null, 8, ["icon"])) : y("", !0)
    ], 16));
  }
});
export {
  L as default
};
