// @scalar/components/dist/components/ScalarSidebar/ScalarSidebarGroup.vue.js
import { defineComponent as g, mergeModels as u, useModel as h, createBlock as x, openBlock as s, resolveDynamicComponent as k, normalizeProps as $, guardReactiveProps as B, unref as n, withCtx as a, createElementVNode as C, createElementBlock as c, createCommentVNode as v, renderSlot as o, createVNode as d, normalizeStyle as S, normalizeClass as z, createTextVNode as N, toDisplayString as V, createSlots as M } from "vue";
import { useBindCx as D } from "@scalar/use-hooks/useBindCx";
import E from "./ScalarSidebarButton.vue.js";
import m from "./ScalarSidebarGroupToggle.vue.js";
import G from "./ScalarSidebarIndent.vue.js";
import { useSidebarGroups as O } from "./useSidebarGroups.js";
const P = { class: "group/group-button relative flex flex-col leading-5" }, T = ["aria-expanded"], j = {
  key: 0,
  class: "size-4"
}, w = {
  key: 0,
  class: "group/items flex flex-col gap-px"
}, A = {}, L = /* @__PURE__ */ g({
  ...A,
  inheritAttrs: !1,
  __name: "ScalarSidebarGroup",
  props: /* @__PURE__ */ u({
    is: { default: "li" },
    icon: { type: [Object, Function] },
    active: { type: Boolean },
    selected: { type: Boolean },
    disabled: { type: Boolean },
    indent: {},
    controlled: { type: Boolean },
    discrete: { type: Boolean }
  }, {
    open: { type: Boolean, default: !1 },
    openModifiers: {}
  }),
  emits: /* @__PURE__ */ u(["click", "toggle"], ["update:open"]),
  setup(t, { emit: f }) {
    const r = f, e = h(t, "open"), { level: i } = O({ increment: !0 }), { cx: p } = D(), b = (l) => {
      r("click", l), !t.controlled && !t.discrete && (e.value = !e.value);
    }, y = (l) => {
      r("toggle", l), !t.controlled && t.discrete && (e.value = !e.value);
    };
    return (l, F) => (s(), x(k(t.is), $(B(n(p)("group/item flex flex-col gap-px"))), {
      default: a(() => [
        C("div", P, [
          o(l.$slots, "before", { open: e.value }),
          o(l.$slots, "button", {
            level: n(i),
            open: e.value
          }, () => [
            t.discrete ? (s(), c("button", {
              key: 0,
              type: "button",
              "aria-expanded": e.value,
              class: z([
                "absolute top-[1lh] -translate-y-1/2 p-0.75 rounded left-[calc(4px+var(--scalar-sidebar-indent)*var(--scalar-sidebar-level))]",
                t.selected ? "text-sidebar-c-1  hover:bg-sidebar-b-1 hover:text-sidebar-c-1" : "text-c-3 hover:bg-sidebar-b-hover hover:text-sidebar-c-hover"
              ]),
              style: S({ "--scalar-sidebar-level": n(i) }),
              onClick: y
            }, [
              o(l.$slots, "icon", { open: e.value }, () => [
                d(m, {
                  icon: t.icon,
                  open: e.value
                }, {
                  label: a(() => [
                    N(V(e.value ? "Close" : "Open") + " ", 1),
                    o(l.$slots, "default", { open: e.value })
                  ]),
                  _: 3
                }, 8, ["icon", "open"])
              ])
            ], 14, T)) : v("", !0),
            d(E, {
              is: "button",
              active: t.active,
              "aria-expanded": e.value,
              disabled: t.disabled,
              indent: n(i),
              selected: t.selected,
              onClick: b
            }, M({
              indent: a(() => [
                d(G, {
                  class: "mr-0 -my-2",
                  indent: n(i),
                  selected: t.selected
                }, null, 8, ["indent", "selected"])
              ]),
              icon: a(() => [
                t.discrete ? (s(), c("div", j)) : o(l.$slots, "icon", {
                  key: 1,
                  open: e.value
                }, () => [
                  d(m, {
                    class: "text-c-3",
                    icon: t.icon,
                    open: e.value
                  }, null, 8, ["icon", "open"])
                ])
              ]),
              default: a(() => [
                o(l.$slots, "default", { open: e.value })
              ]),
              _: 2
            }, [
              l.$slots.aside ? {
                name: "aside",
                fn: a(() => [
                  o(l.$slots, "aside", { open: e.value })
                ]),
                key: "0"
              } : void 0
            ]), 1032, ["active", "aria-expanded", "disabled", "indent", "selected"])
          ]),
          o(l.$slots, "after", { open: e.value })
        ]),
        e.value ? (s(), c("ul", w, [
          o(l.$slots, "items", { open: e.value })
        ])) : v("", !0)
      ]),
      _: 3
    }, 16));
  }
});
export {
  L as default
};
