// @scalar/components/dist/components/ScalarSidebar/ScalarSidebarSection.vue.js
import { defineComponent as c, createBlock as d, openBlock as p, resolveDynamicComponent as f, normalizeProps as m, guardReactiveProps as u, unref as e, withCtx as s, createVNode as n, createElementVNode as _, renderSlot as r } from "vue";
import { useBindCx as b } from "@scalar/use-hooks/useBindCx";
import x from "./ScalarSidebarButton.vue.js";
import a from "./ScalarSidebarSpacer.vue.js";
import { useSidebarGroups as g } from "./useSidebarGroups.js";
const h = { class: "flex flex-col gap-px" }, y = {}, N = /* @__PURE__ */ c({
  ...y,
  inheritAttrs: !1,
  __name: "ScalarSidebarSection",
  props: {
    is: { default: "li" },
    icon: { type: [Object, Function] },
    active: { type: Boolean },
    selected: { type: Boolean },
    disabled: { type: Boolean },
    indent: {}
  },
  setup(o) {
    const { level: t } = g({ increment: !1 }), { cx: l } = b();
    return (i, B) => (p(), d(f(o.is), m(u(e(l)("group/item group/sidebar-section contents"))), {
      default: s(() => [
        n(a, {
          class: "group/spacer-before h-3",
          indent: e(t)
        }, null, 8, ["indent"]),
        n(x, {
          is: "div",
          class: "text-sm/5 py-1.75 font-bold",
          disabled: "",
          icon: o.icon,
          indent: e(t)
        }, {
          default: s(() => [
            r(i.$slots, "default")
          ]),
          _: 3
        }, 8, ["icon", "indent"]),
        _("ul", h, [
          r(i.$slots, "items")
        ]),
        n(a, {
          class: "group/spacer-after h-3",
          indent: e(t)
        }, null, 8, ["indent"])
      ]),
      _: 3
    }, 16));
  }
});
export {
  N as default
};
