// @scalar/components/dist/components/ScalarSidebar/ScalarSidebarItem.vue.js
import { defineComponent as a, createElementBlock as d, openBlock as f, normalizeProps as m, guardReactiveProps as p, unref as t, renderSlot as o, createVNode as c, mergeProps as u, createSlots as $, withCtx as n } from "vue";
import { useBindCx as v } from "@scalar/use-hooks/useBindCx";
import y from "./ScalarSidebarButton.vue.js";
import { useSidebarGroups as b } from "./useSidebarGroups.js";
const B = {}, P = /* @__PURE__ */ a({
  ...B,
  inheritAttrs: !1,
  __name: "ScalarSidebarItem",
  props: {
    is: {},
    icon: { type: [Object, Function] },
    active: { type: Boolean },
    selected: { type: Boolean },
    disabled: { type: Boolean },
    indent: {}
  },
  setup(s) {
    const l = s, { level: i } = b(), { cx: r } = v();
    return (e, k) => (f(), d("li", m(p(t(r)("group/item flex flex-col"))), [
      o(e.$slots, "before"),
      o(e.$slots, "button", { level: t(i) }, () => [
        c(y, u(l, {
          indent: s.indent ?? t(i)
        }), $({ _: 2 }, [
          e.$slots.default ? {
            name: "default",
            fn: n(() => [
              o(e.$slots, "default")
            ]),
            key: "0"
          } : void 0,
          e.$slots.icon ? {
            name: "icon",
            fn: n(() => [
              o(e.$slots, "icon")
            ]),
            key: "1"
          } : void 0,
          e.$slots.aside ? {
            name: "aside",
            fn: n(() => [
              o(e.$slots, "aside")
            ]),
            key: "2"
          } : void 0
        ]), 1040, ["indent"])
      ]),
      o(e.$slots, "after")
    ], 16));
  }
});
export {
  P as default
};
