// @scalar/components/dist/components/ScalarMenu/ScalarMenuTeamPicker.vue.js
import { defineComponent as g, computed as x, createBlock as c, openBlock as n, unref as e, withCtx as r, createVNode as a, mergeProps as b, createElementVNode as _, createElementBlock as S, Fragment as w, renderList as v, createTextVNode as C } from "vue";
import { ScalarIconUserSwitch as k, ScalarIconCaretRight as V, ScalarIconPlus as $ } from "@scalar/icons";
import { DropdownMenu as o } from "radix-vue/namespaced";
import I from "./ScalarMenuTeamProfile.vue.js";
import B from "./ScalarMenuLink.vue.js";
import P from "../ScalarDropdown/ScalarDropdownMenu.vue.js";
import f from "../ScalarDropdown/ScalarDropdownButton.vue.js";
import z from "../ScalarListbox/ScalarListboxCheckbox.vue.js";
const y = /* @__PURE__ */ g({
  inheritAttrs: !1,
  __name: "ScalarMenuTeamPicker",
  props: {
    team: {},
    teams: {}
  },
  emits: ["update:team", "add"],
  setup(i, { emit: p }) {
    const d = i, u = p, s = x({
      get: () => d.team?.id,
      set: (m) => u(
        "update:team",
        d.teams.find((t) => t.id === m)
      )
    });
    return (m, t) => (n(), c(e(o).Sub, null, {
      default: r(() => [
        a(e(B), b({
          is: e(o).SubTrigger,
          icon: e(k)
        }, m.$attrs), {
          default: r(() => [
            t[2] || (t[2] = _("div", null, "Change team", -1)),
            a(e(V), {
              class: "ml-auto text-c-2 -mr-0.25 size-3",
              weight: "bold"
            })
          ]),
          _: 1
        }, 16, ["is", "icon"]),
        a(e(o).Portal, null, {
          default: r(() => [
            a(e(o).SubContent, {
              as: e(P),
              class: "max-h-radix-popper z-context-plus",
              sideOffset: 3
            }, {
              default: r(() => [
                a(e(o).RadioGroup, {
                  modelValue: s.value,
                  "onUpdate:modelValue": t[0] || (t[0] = (l) => s.value = l),
                  class: "contents"
                }, {
                  default: r(() => [
                    (n(!0), S(w, null, v(i.teams, (l) => (n(), c(e(o).RadioItem, {
                      key: l.id,
                      as: e(f),
                      class: "group/item flex items-center",
                      value: l.id
                    }, {
                      default: r(() => [
                        a(I, {
                          class: "-ml-0.75 flex-1 min-w-0",
                          label: l.label,
                          src: l.src
                        }, null, 8, ["label", "src"]),
                        a(e(z), {
                          class: "ml-auto",
                          selected: l.id === s.value
                        }, null, 8, ["selected"])
                      ]),
                      _: 2
                    }, 1032, ["as", "value"]))), 128))
                  ]),
                  _: 1
                }, 8, ["modelValue"]),
                a(e(o).Item, {
                  as: e(f),
                  class: "flex items-center",
                  onClick: t[1] || (t[1] = (l) => u("add"))
                }, {
                  default: r(() => [
                    a(e($), {
                      class: "bg-b-3 -ml-0.75 rounded p-1 size-5 text-c-3",
                      weight: "bold"
                    }),
                    t[3] || (t[3] = C(" Create new team ", -1))
                  ]),
                  _: 1
                }, 8, ["as"])
              ]),
              _: 1
            }, 8, ["as"])
          ]),
          _: 1
        })
      ]),
      _: 1
    }));
  }
});
export {
  y as default
};
