// @scalar/components/dist/components/ScalarMenu/ScalarMenuProducts.vue.js
import { defineComponent as a, createElementBlock as f, openBlock as m, normalizeProps as u, guardReactiveProps as h, unref as r, createVNode as n, withCtx as l, createTextVNode as d } from "vue";
import { ScalarIconHouse as p, ScalarIconBook as C, ScalarIconArrowUpRight as k } from "@scalar/icons";
import { useBindCx as b } from "@scalar/use-hooks/useBindCx";
import i from "./ScalarMenuProduct.vue.js";
const I = /* @__PURE__ */ a({
  inheritAttrs: !1,
  __name: "ScalarMenuProducts",
  props: {
    selected: {},
    hrefs: {}
  },
  emits: ["open"],
  setup(o) {
    const { cx: c } = b();
    return (s, e) => (m(), f("div", u(h(r(c)("flex flex-col"))), [
      n(i, {
        href: o.hrefs?.dashboard ?? "https://dashboard.scalar.com",
        icon: r(p),
        selected: o.selected === "dashboard",
        onClick: e[0] || (e[0] = (t) => s.$emit("open", t, "dashboard"))
      }, {
        default: l(() => [...e[3] || (e[3] = [
          d(" Dashboard ", -1)
        ])]),
        _: 1
      }, 8, ["href", "icon", "selected"]),
      n(i, {
        href: o.hrefs?.docs ?? "https://docs.scalar.com",
        icon: r(C),
        selected: o.selected === "docs",
        onClick: e[1] || (e[1] = (t) => s.$emit("open", t, "docs"))
      }, {
        default: l(() => [...e[4] || (e[4] = [
          d(" Docs ", -1)
        ])]),
        _: 1
      }, 8, ["href", "icon", "selected"]),
      n(i, {
        href: o.hrefs?.client ?? "https://client.scalar.com",
        icon: r(k),
        selected: o.selected === "client",
        onClick: e[2] || (e[2] = (t) => s.$emit("open", t, "client"))
      }, {
        default: l(() => [...e[5] || (e[5] = [
          d(" Client ", -1)
        ])]),
        _: 1
      }, 8, ["href", "icon", "selected"])
    ], 16));
  }
});
export {
  I as default
};
