// @scalar/components/dist/components/ScalarMenu/ScalarMenuWorkspacePicker.vue.js
import { defineComponent as k, mergeModels as m, useModel as x, createBlock as u, openBlock as n, unref as e, withCtx as a, createVNode as l, mergeProps as S, createElementVNode as d, createElementBlock as _, Fragment as V, renderList as b, toDisplayString as v, createTextVNode as w } from "vue";
import { ScalarIconSwap as C, ScalarIconCaretRight as I, ScalarIconPlus as M } from "@scalar/icons";
import { DropdownMenu as r } from "radix-vue/namespaced";
import $ from "./ScalarMenuLink.vue.js";
import B from "../ScalarDropdown/ScalarDropdownMenu.vue.js";
import c from "../ScalarDropdown/ScalarDropdownButton.vue.js";
import P from "../ScalarListbox/ScalarListboxCheckbox.vue.js";
const z = { class: "flex h-full items-center gap-1 flex-1 truncate" }, E = /* @__PURE__ */ k({
  inheritAttrs: !1,
  __name: "ScalarMenuWorkspacePicker",
  props: /* @__PURE__ */ m({
    workspaceOptions: {}
  }, {
    modelValue: {},
    modelModifiers: {}
  }),
  emits: /* @__PURE__ */ m(["createWorkspace"], ["update:modelValue"]),
  setup(i, { emit: p }) {
    const f = p, s = x(i, "modelValue");
    return (g, t) => (n(), u(e(r).Sub, null, {
      default: a(() => [
        l($, S({
          is: e(r).SubTrigger,
          icon: e(C)
        }, g.$attrs), {
          default: a(() => [
            t[2] || (t[2] = d("div", null, "Change workspace", -1)),
            l(e(I), {
              class: "ml-auto text-c-2 -mr-0.25 size-3",
              weight: "bold"
            })
          ]),
          _: 1
        }, 16, ["is", "icon"]),
        l(e(r).Portal, null, {
          default: a(() => [
            l(e(r).SubContent, {
              as: e(B),
              class: "max-h-radix-popper z-context-plus",
              sideOffset: 3
            }, {
              default: a(() => [
                l(e(r).RadioGroup, {
                  modelValue: s.value,
                  "onUpdate:modelValue": t[0] || (t[0] = (o) => s.value = o),
                  class: "contents"
                }, {
                  default: a(() => [
                    (n(!0), _(V, null, b(i.workspaceOptions, (o) => (n(), u(e(r).RadioItem, {
                      key: o.id,
                      as: e(c),
                      class: "group/item flex items-center",
                      value: o.id
                    }, {
                      default: a(() => [
                        d("div", z, v(o.label), 1),
                        l(e(P), {
                          class: "ml-auto",
                          selected: o.id === s.value
                        }, null, 8, ["selected"])
                      ]),
                      _: 2
                    }, 1032, ["as", "value"]))), 128))
                  ]),
                  _: 1
                }, 8, ["modelValue"]),
                l(e(r).Item, {
                  as: e(c),
                  class: "flex items-center",
                  onClick: t[1] || (t[1] = (o) => f("createWorkspace"))
                }, {
                  default: a(() => [
                    l(e(M), {
                      class: "bg-b-3 -ml-0.75 rounded p-1 size-5 text-c-3",
                      weight: "bold"
                    }),
                    t[3] || (t[3] = w(" Create workspace ", -1))
                  ]),
                  _: 1
                }, 8, ["as"])
              ]),
              _: 1
            }, 8, ["as"])
          ]),
          _: 1
        })
      ]),
      _: 1
    }));
  }
});
export {
  E as default
};
