// @scalar/components/dist/components/ScalarMenu/ScalarMenuProduct.vue.js
import { defineComponent as a, createBlock as s, openBlock as c, unref as t, mergeProps as i, withCtx as l, renderSlot as d } from "vue";
import { useBindCx as u, cva as f } from "@scalar/use-hooks/useBindCx";
import b from "./ScalarMenuLink.vue.js";
const h = /* @__PURE__ */ a({
  inheritAttrs: !1,
  __name: "ScalarMenuProduct",
  props: {
    is: { default: "a" },
    selected: { type: Boolean },
    icon: { type: [Object, Function] }
  },
  setup(e) {
    const { cx: n } = u(), r = f({
      base: "gap-1.5",
      variants: {
        selected: {
          true: "pointer-events-none bg-b-2 dark:bg-b-3",
          false: "cursor-pointer hover:bg-b-2 dark:hover:bg-b-3"
        }
      }
    });
    return (o, m) => (c(), s(t(b), i({
      is: e.is,
      icon: e.icon,
      strong: "",
      target: "_blank"
    }, t(n)(t(r)({ selected: e.selected }))), {
      default: l(() => [
        d(o.$slots, "default")
      ]),
      _: 3
    }, 16, ["is", "icon"]));
  }
});
export {
  h as default
};
