// @scalar/components/dist/components/ScalarMenu/ScalarMenuLink.vue.js
import { defineComponent as c, createBlock as o, openBlock as t, unref as n, mergeProps as a, withCtx as l, createElementBlock as m, createElementVNode as f, normalizeClass as i, renderSlot as u } from "vue";
import { useBindCx as d } from "@scalar/use-hooks/useBindCx";
import { DropdownMenu as g } from "radix-vue/namespaced";
import x from "../ScalarDropdown/ScalarDropdownButton.vue.js";
import h from "../ScalarIcon/ScalarIconLegacyAdapter.vue.js";
const k = {
  key: 1,
  class: "size-3"
}, S = /* @__PURE__ */ c({
  inheritAttrs: !1,
  __name: "ScalarMenuLink",
  props: {
    is: { default: () => g.Item },
    icon: { type: [Object, Function] },
    strong: { type: Boolean }
  },
  setup(e) {
    const { cx: s } = d();
    return (r, w) => (t(), o(n(x), a(n(s)("flex items-center"), {
      is: e.is,
      as: "a"
    }), {
      default: l(() => [
        e.icon ? (t(), o(n(h), {
          key: 0,
          class: i([
            e.strong ? "text-c-1" : "text-c-2",
            typeof e.icon == "string" ? "size-3" : "size-3.5 -mx-0.25"
          ]),
          icon: e.icon,
          thickness: e.strong ? "2.5" : "2",
          weight: e.strong ? "bold" : "regular"
        }, null, 8, ["class", "icon", "thickness", "weight"])) : (t(), m("div", k)),
        f("div", {
          class: i(["flex items-center flex-1 min-w-0 truncate", e.strong ? "font-medium" : "font-normal"])
        }, [
          u(r.$slots, "default")
        ], 2)
      ]),
      _: 3
    }, 16, ["is"]));
  }
});
export {
  S as default
};
