// @scalar/agent-chat/dist/App.vue.js
import { defineComponent as t, provide as r, createBlock as n, openBlock as c } from "vue";
import o from "./Chat.vue.js";
import { STATE_SYMBOL as s, createState as i } from "./state/state.js";
const u = /* @__PURE__ */ t({
  __name: "App",
  props: {
    registryDocuments: {},
    registryUrl: {},
    dashboardUrl: {},
    baseUrl: {},
    mode: { default: "full" },
    getAccessToken: { type: Function },
    getAgentKey: { type: Function },
    getActiveDocumentJson: { type: Function },
    prefilledMessage: {}
  },
  setup(e) {
    return r(
      s,
      i({
        getActiveDocumentJson: e.getActiveDocumentJson,
        initialRegistryDocuments: e.registryDocuments,
        prefilledMessageRef: e.prefilledMessage,
        registryUrl: e.registryUrl,
        baseUrl: e.baseUrl,
        mode: e.mode,
        getAccessToken: e.getAccessToken,
        getAgentKey: e.getAgentKey,
        dashboardUrl: e.dashboardUrl
      })
    ), (a, l) => (c(), n(o));
  }
});
export {
  u as default
};
