// @scalar/agent-chat/dist/Chat.vue.js
import { defineComponent as f, useTemplateRef as d, ref as g, onMounted as _, onBeforeUnmount as v, createElementBlock as S, openBlock as M, Fragment as h, createElementVNode as C, createVNode as n, unref as D } from "vue";
import { createApiClientModal as w } from "@scalar/api-client/v2/features/modal";
import { useAgentKeyDocuments as y } from "./hooks/use-agent-key-documents.js";
import { useChatScroll as B } from "./hooks/use-chat-scroll.js";
import { useCuratedDocuments as b } from "./hooks/use-curated-documents.js";
import { getTmpDocFromLocalStorage as k } from "./hooks/use-upload-tmp-document.js";
import { useState as x } from "./state/state.js";
import A from "./views/Layout.vue.js";
import E from "./views/Settings/Settings.vue.js";
const F = { ref: "clientModal" }, H = /* @__PURE__ */ f({
  __name: "Chat",
  setup(L) {
    const {
      chat: a,
      prompt: r,
      settingsModal: m,
      eventBus: l,
      workspaceStore: u,
      config: c,
      mode: s,
      addDocument: i
    } = x(), t = d("clientModal"), o = g(null);
    _(async () => {
      const e = k();
      s === "preview" && e && await i({
        namespace: e.namespace,
        slug: e.slug,
        removable: !1
      }), t.value && (o.value = w({
        el: t.value,
        options: c,
        eventBus: l,
        workspaceStore: u
      }));
    }), v(() => {
      o.value?.app.unmount();
    }), B(), y(), b();
    async function p() {
      await a.sendMessage({ text: r.value });
    }
    return (e, N) => (M(), S(h, null, [
      C("div", F, null, 512),
      n(A, { onSubmit: p }),
      n(E, { modalState: D(m) }, null, 8, ["modalState"])
    ], 64));
  }
});
export {
  H as default
};
