// @scalar/api-reference/dist/features/Operation/components/Headers.vue2.js
import { defineComponent, createBlock, openBlock, unref, withCtx, createElementVNode, normalizeClass, createVNode, createElementBlock, Fragment, createTextVNode, renderList } from "vue";
import { Disclosure, DisclosureButton, DisclosurePanel } from "@headlessui/vue";
import { ScalarIcon } from "@scalar/components";
import { getResolvedRef } from "@scalar/workspace-store/helpers/get-resolved-ref";
import _sfc_main$1 from "./Header.vue.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "Headers",
  props: {
    headers: {},
    breadcrumb: {},
    eventBus: {},
    orderRequiredPropertiesFirst: { type: Boolean },
    orderSchemaPropertiesBy: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createBlock(unref(Disclosure), null, {
        default: withCtx(({ open }) => [
          createElementVNode("div", {
            class: normalizeClass(["headers-card headers-card--compact", [{ "headers-card--open": open }]])
          }, [
            createElementVNode("div", {
              class: normalizeClass(["headers-properties", { "headers-properties-open": open }])
            }, [
              createVNode(unref(DisclosureButton), {
                class: "headers-card-title headers-card-title--compact",
                style: {
                  top: `calc(var(--refs-viewport-offset)))`
                }
              }, {
                default: withCtx(() => [
                  createVNode(unref(ScalarIcon), {
                    class: normalizeClass(["headers-card-title-icon", { "headers-card-title-icon--open": open }]),
                    icon: "Add",
                    size: "sm"
                  }, null, 8, ["class"]),
                  open ? (openBlock(), createElementBlock(Fragment, { key: 0 }, [
                    createTextVNode(" Hide Headers ")
                  ], 64)) : (openBlock(), createElementBlock(Fragment, { key: 1 }, [
                    createTextVNode(" Show Headers ")
                  ], 64))
                ]),
                _: 2
              }, 1024),
              createVNode(unref(DisclosurePanel), null, {
                default: withCtx(() => [
                  (openBlock(true), createElementBlock(Fragment, null, renderList(__props.headers, (header, key) => {
                    return openBlock(), createBlock(_sfc_main$1, {
                      key,
                      breadcrumb: __props.breadcrumb ? [...__props.breadcrumb, "headers"] : void 0,
                      eventBus: __props.eventBus,
                      header: unref(getResolvedRef)(header),
                      name: key,
                      orderRequiredPropertiesFirst: __props.orderRequiredPropertiesFirst,
                      orderSchemaPropertiesBy: __props.orderSchemaPropertiesBy
                    }, null, 8, ["breadcrumb", "eventBus", "header", "name", "orderRequiredPropertiesFirst", "orderSchemaPropertiesBy"]);
                  }), 128))
                ]),
                _: 1
              })
            ], 2)
          ], 2)
        ]),
        _: 1
      });
    };
  }
});
export {
  _sfc_main as default
};
