// @scalar/api-client/dist/v2/features/command-palette/components/CommandPaletteImportCurl.vue2.js
import { defineComponent as y, ref as b, computed as i, createBlock as B, openBlock as N, withCtx as s, createVNode as r, createElementVNode as m, unref as l, toDisplayString as d, createTextVNode as D, normalizeClass as I } from "vue";
import { ScalarListbox as $, ScalarButton as K, ScalarIcon as R } from "@scalar/components";
import { useRouter as U } from "vue-router";
import j from "../../../blocks/operation-code-sample/components/HttpMethod.vue.js";
import z from "./CommandActionForm.vue.js";
import E from "./CommandActionInput.vue.js";
import { getOperationFromCurl as k } from "../helpers/get-operation-from-curl.js";
const O = { class: "flex flex-1 flex-col gap-2" }, T = { class: "flex h-9 flex-row items-center gap-2 rounded border p-[3px] text-sm" }, q = { class: "scroll-timeline-x whitespace-nowrap" }, F = { class: "flex items-center gap-2" }, L = {
  name: "CommandPaletteImportCurl"
}, Q = /* @__PURE__ */ y({
  ...L,
  props: {
    workspaceStore: {},
    eventBus: {},
    inputValue: {}
  },
  emits: ["close", "back"],
  setup(a, { emit: w }) {
    const p = w, S = U(), c = b(""), u = i(() => c.value.trim()), { path: f, method: v, operation: V } = k(a.inputValue), x = i(
      () => Object.keys(a.workspaceStore.workspace.documents).map((t) => ({
        id: t,
        label: t
      }))
    ), o = b(
      x.value[0]
    ), h = i(() => !!(!u.value || !o.value || a.workspaceStore.workspace.documents[o.value.id]?.paths?.[f]?.[v])), C = () => {
      const t = o.value;
      if (h.value || !t)
        return;
      const e = k(a.inputValue, u.value);
      a.eventBus.emit("operation:create:operation", {
        documentName: t.id,
        path: e.path,
        method: e.method,
        operation: e.operation,
        exampleKey: u.value,
        callback: (n) => {
          if (n) {
            a.workspaceStore.buildSidebar(t.id);
            const _ = e.path.startsWith("/") ? e.path : `/${e.path}`;
            S.push({
              name: "example",
              params: {
                documentSlug: t.id,
                pathEncoded: encodeURIComponent(_),
                method: e.method,
                exampleName: u.value
              }
            });
          }
        }
      }), p("close");
    }, g = (t) => {
      p("back", t);
    };
    return (t, e) => (N(), B(z, {
      disabled: h.value,
      onSubmit: C
    }, {
      options: s(() => [
        m("div", F, [
          r(l($), {
            modelValue: o.value,
            "onUpdate:modelValue": e[1] || (e[1] = (n) => o.value = n),
            options: x.value
          }, {
            default: s(() => [
              r(l(K), {
                class: "hover:bg-b-2 max-h-8 w-full justify-between gap-1 p-2 text-xs",
                variant: "outlined"
              }, {
                default: s(() => [
                  m("span", {
                    class: I(["whitespace-nowrap", o.value ? "text-c-1" : "text-c-3"])
                  }, d(o.value ? o.value.label : "Select Collection"), 3),
                  r(l(R), {
                    class: "text-c-3",
                    icon: "ChevronDown",
                    size: "md"
                  })
                ]),
                _: 1
              })
            ]),
            _: 1
          }, 8, ["modelValue", "options"])
        ])
      ]),
      submit: s(() => [...e[2] || (e[2] = [
        D("Import Request", -1)
      ])]),
      default: s(() => [
        r(E, {
          modelValue: c.value,
          "onUpdate:modelValue": e[0] || (e[0] = (n) => c.value = n),
          placeholder: "Curl example key (e.g., example-1)",
          onDelete: g
        }, null, 8, ["modelValue"]),
        m("div", O, [
          m("div", T, [
            r(j, {
              class: "border-r-1 px-1",
              method: l(v)
            }, null, 8, ["method"]),
            m("span", q, d(l(V).servers?.[0]?.url || "") + d(l(f)), 1)
          ])
        ])
      ]),
      _: 1
    }, 8, ["disabled"]));
  }
});
export {
  Q as default
};
