// @scalar/api-client/dist/v2/features/command-palette/components/CommandActionInput.vue.js
import { defineComponent as c, ref as m, computed as p, onMounted as f, nextTick as h, withDirectives as b, createElementBlock as g, openBlock as v, mergeProps as w, withKeys as u, vModelText as x } from "vue";
const y = ["placeholder"], E = {
  name: "CommandActionInput"
}, k = /* @__PURE__ */ c({
  ...E,
  inheritAttrs: !1,
  props: {
    modelValue: {},
    placeholder: {},
    autofocus: { type: Boolean, default: !0 }
  },
  emits: ["update:modelValue", "delete"],
  setup(o, { emit: d }) {
    const a = d, r = m(null), n = p({
      get: () => o.modelValue ?? "",
      set: (e) => a("update:modelValue", e)
    });
    f(() => {
      h(() => {
        o.autofocus && r.value?.focus();
      });
    });
    const s = (e) => {
      if (e.shiftKey || !e.target)
        return;
      e.preventDefault();
      const t = e.target, l = new Event("submit", { cancelable: !0 });
      t.form?.dispatchEvent(l);
    }, i = (e) => {
      n.value === "" && (e.preventDefault(), e.stopPropagation(), a("delete", e));
    };
    return (e, t) => b((v(), g("textarea", w({
      id: "command-action-input",
      ref_key: "input",
      ref: r,
      "onUpdate:modelValue": t[0] || (t[0] = (l) => n.value = l),
      class: "min-h-8 w-full flex-1 resize-none border border-transparent py-1.5 pl-8.5 text-sm outline-none focus:border-b-1",
      placeholder: o.placeholder ?? "",
      wrap: "hard"
    }, e.$attrs, {
      onKeydown: [
        u(i, ["delete"]),
        u(s, ["enter"])
      ]
    }), null, 16, y)), [
      [x, n.value]
    ]);
  }
});
export {
  k as default
};
