// @scalar/api-client/dist/v2/features/command-palette/components/CommandPaletteTag.vue.js
import { defineComponent as g, ref as f, computed as i, createBlock as k, openBlock as S, withCtx as l, createVNode as n, createTextVNode as w, unref as c, createElementVNode as V, normalizeClass as C, toDisplayString as B } from "vue";
import { ScalarListbox as D, ScalarButton as N, ScalarIcon as T } from "@scalar/components";
import I from "./CommandActionForm.vue.js";
import h from "./CommandActionInput.vue.js";
const j = {
  name: "CommandPaletteTag"
}, E = /* @__PURE__ */ g({
  ...j,
  props: {
    workspaceStore: {},
    eventBus: {},
    documentId: {}
  },
  emits: ["close", "back"],
  setup(o, { emit: p }) {
    const d = p, u = f(""), m = i(() => u.value.trim()), s = i(
      () => Object.entries(o.workspaceStore.workspace.documents).map(
        ([e, t]) => ({
          id: e,
          label: t.info.title || e
        })
      )
    ), a = f(
      o.documentId ? s.value.find((e) => e.id === o.documentId) : s.value[0] ?? void 0
    ), v = i(() => {
      if (!m.value || !a.value)
        return !0;
      const e = o.workspaceStore.workspace.documents[a.value.id];
      return !!(!e || e.tags?.some((t) => t.name === m.value));
    }), b = () => {
      v.value || !a.value || (o.eventBus.emit("tag:create:tag", {
        name: m.value,
        documentName: a.value.id
      }), d("close"));
    }, x = (e) => {
      d("back", e);
    };
    return (e, t) => (S(), k(I, {
      disabled: v.value,
      onSubmit: b
    }, {
      options: l(() => [
        n(c(D), {
          modelValue: a.value,
          "onUpdate:modelValue": t[1] || (t[1] = (r) => a.value = r),
          options: s.value
        }, {
          default: l(() => [
            n(c(N), {
              class: "hover:bg-b-2 max-h-8 w-fit justify-between gap-1 p-2 text-xs",
              variant: "outlined"
            }, {
              default: l(() => [
                V("span", {
                  class: C(a.value ? "text-c-1" : "text-c-3")
                }, B(a.value ? a.value.label : "Select Collection"), 3),
                n(c(T), {
                  class: "text-c-3",
                  icon: "ChevronDown",
                  size: "md"
                })
              ]),
              _: 1
            })
          ]),
          _: 1
        }, 8, ["modelValue", "options"])
      ]),
      submit: l(() => [...t[2] || (t[2] = [
        w("Create Tag", -1)
      ])]),
      default: l(() => [
        n(h, {
          modelValue: u.value,
          "onUpdate:modelValue": t[0] || (t[0] = (r) => u.value = r),
          label: "Tag Name",
          placeholder: "Tag Name",
          onDelete: x
        }, null, 8, ["modelValue"])
      ]),
      _: 1
    }, 8, ["disabled"]));
  }
});
export {
  E as default
};
