// @scalar/api-client/dist/v2/features/command-palette/components/CommandActionForm.vue.js
import { defineComponent as f, createElementBlock as u, openBlock as c, withModifiers as a, withKeys as p, createElementVNode as s, normalizeProps as x, guardReactiveProps as b, unref as d, renderSlot as l, createVNode as h, withCtx as v, createTextVNode as w } from "vue";
import { useBindCx as B, ScalarButton as C } from "@scalar/components";
const y = { class: "flex gap-2" }, S = { class: "flex max-h-8 flex-1" }, g = {
  name: "CommandActionForm"
}, V = /* @__PURE__ */ f({
  ...g,
  props: {
    loader: {},
    disabled: { type: Boolean, default: !1 }
  },
  emits: ["submit", "cancel", "back"],
  setup(t, { emit: i }) {
    const n = i, { cx: r } = B(), m = () => {
      t.disabled || n("submit");
    };
    return (o, e) => (c(), u("form", {
      class: "flex w-full flex-col gap-3",
      onKeydown: e[0] || (e[0] = p(a(() => {
      }, ["stop"]), ["enter"])),
      onSubmit: a(m, ["prevent", "stop"])
    }, [
      s("div", x(b(d(r)("relative flex min-h-20 flex-col rounded"))), [
        l(o.$slots, "default")
      ], 16),
      s("div", y, [
        s("div", S, [
          l(o.$slots, "options")
        ]),
        h(d(C), {
          class: "max-h-8 px-3 text-xs",
          disabled: t.disabled,
          loader: t.loader,
          type: "submit"
        }, {
          default: v(() => [
            l(o.$slots, "submit", {}, () => [
              e[1] || (e[1] = w("Continue", -1))
            ])
          ]),
          _: 3
        }, 8, ["disabled", "loader"])
      ])
    ], 32));
  }
});
export {
  V as default
};
