// @scalar/api-client/dist/v2/features/command-palette/components/CommandPaletteImport.vue.js
import { defineComponent as z, ref as V, computed as x, watch as J, createBlock as S, openBlock as o, unref as l, withCtx as m, createElementBlock as d, Fragment as p, createElementVNode as C, createCommentVNode as X, createVNode as g, createTextVNode as r, toDisplayString as L } from "vue";
import { useLoadingState as Y, ScalarButton as P, ScalarCodeBlock as G, ScalarIcon as H, ScalarTooltip as K } from "@scalar/components";
import { normalize as Q } from "@scalar/openapi-parser";
import { generateUniqueValue as Z } from "@scalar/workspace-store/helpers/generate-unique-value";
import { useRouter as ee } from "vue-router";
import { getOpenApiDocumentDetails as te } from "../helpers/get-openapi-document-details.js";
import { getOpenApiFromPostman as ae } from "../helpers/get-openapi-from-postman.js";
import { getPostmanDocumentDetails as oe } from "../helpers/get-postman-document-details.js";
import { isPostmanCollection as R } from "../helpers/is-postman-collection.js";
import { isUrl as F } from "../../../helpers/is-url.js";
import { slugify as ne } from "../../../helpers/slugify.js";
import se from "./CommandActionForm.vue.js";
import le from "./CommandActionInput.vue.js";
import re from "./WatchModeToggle.vue.js";
import { useFileDialog as ie } from "../../../../hooks/useFileDialog.js";
const ce = { class: "flex justify-between" }, ue = { class: "flex w-full flex-row items-center justify-between gap-3" }, me = {
  name: "CommandPaletteImport"
}, Ie = /* @__PURE__ */ z({
  ...me,
  props: {
    workspaceStore: {},
    eventBus: {}
  },
  emits: ["close", "back"],
  setup(f, { emit: T }) {
    const k = T, B = 100, b = "document", E = ee(), w = Y(), a = V(""), y = V(!0), i = x(() => F(a.value)), n = x(() => R(a.value) ? oe(a.value) : te(a.value)), U = x(
      () => n.value ? n.value.type : "json"
    ), O = x(() => !a.value.trim());
    J(i, (e) => {
      y.value = e;
    });
    const j = (e) => !Object.keys(f.workspaceStore.workspace.documents).includes(e), h = async (e) => await Z({
      defaultValue: e,
      validation: j,
      maxRetries: B,
      transformation: ne
    }), q = (e) => typeof e.info == "object" && e.info !== null && "title" in e.info && typeof e.info.title == "string" ? e.info.title : b, A = async (e) => {
      if (F(e)) {
        const s = await h(b);
        return s ? { success: await f.workspaceStore.addDocument({
          name: s,
          url: e,
          meta: {
            "x-scalar-watch-mode": y.value
          }
        }), name: s } : (console.error("Failed to generate a unique name"), { success: !1 });
      }
      if (R(e)) {
        const s = ae(e), v = s.info?.title ?? b, D = await h(v);
        return D ? { success: await f.workspaceStore.addDocument({
          name: D,
          document: s
        }), name: D } : (console.error("Failed to generate a unique name"), { success: !1 });
      }
      const t = Q(e), c = q(t), u = await h(c);
      return u ? { success: await f.workspaceStore.addDocument({
        name: u,
        document: t
      }), name: u } : (console.error("Failed to generate a unique name"), { success: !1 });
    }, I = (e) => {
      E.push({
        name: "document.overview",
        params: { documentSlug: e }
      });
    }, { open: M } = ie({
      onChange: (e) => {
        const [t] = e ?? [];
        if (!t)
          return;
        const c = async (N) => {
          const s = N.target?.result, v = await A(s);
          v.success && I(v.name), k("close");
        }, u = new FileReader();
        u.onload = c, u.readAsText(t);
      },
      multiple: !1,
      accept: ".json,.yaml,.yml"
    }), _ = async () => {
      w.start();
      const e = await A(a.value);
      e.success ? (await w.clear(), I(e.name)) : await w.invalidate(), k("close");
    }, W = (e) => {
      if (e.trim().toLowerCase().startsWith("curl"))
        return f.eventBus.emit("ui:open:command-palette", {
          action: "import-curl-command",
          payload: {
            inputValue: e
          }
        });
      a.value = e;
    }, $ = (e) => {
      k("back", e);
    };
    return (e, t) => (o(), S(se, {
      disabled: O.value,
      loader: l(w),
      onSubmit: _
    }, {
      options: m(() => [
        C("div", ue, [
          g(l(P), {
            class: "hover:bg-b-2 relative max-h-8 gap-1.5 p-2 text-xs",
            variant: "outlined",
            onClick: l(M)
          }, {
            default: m(() => [
              t[4] || (t[4] = r(" JSON, or YAML File ", -1)),
              g(l(H), {
                class: "text-c-3",
                icon: "Upload",
                size: "md"
              })
            ]),
            _: 1
          }, 8, ["onClick"]),
          g(l(K), {
            content: i.value ? "Watch mode automatically updates the API client when the OpenAPI URL content changes, ensuring your client remains up-to-date." : "Watch mode is only available for URL imports. When enabled it automatically updates the API client when the OpenAPI URL content changes.",
            placement: "bottom"
          }, {
            default: m(() => [
              g(re, {
                modelValue: y.value,
                "onUpdate:modelValue": t[1] || (t[1] = (c) => y.value = c),
                disabled: !i.value
              }, null, 8, ["modelValue", "disabled"])
            ]),
            _: 1
          }, 8, ["content"])
        ])
      ]),
      submit: m(() => [
        t[5] || (t[5] = r(" Import ", -1)),
        i.value ? (o(), d(p, { key: 0 }, [
          r("from URL")
        ], 64)) : n.value && U.value ? (o(), d(p, { key: 1 }, [
          n.value.title ? (o(), d(p, { key: 0 }, [
            r(' "' + L(n.value.title) + '" ', 1)
          ], 64)) : (o(), d(p, { key: 1 }, [
            r(L(n.value.version), 1)
          ], 64))
        ], 64)) : (o(), d(p, { key: 2 }, [
          r("Collection")
        ], 64))
      ]),
      default: m(() => [
        !n.value || i.value ? (o(), S(le, {
          key: 0,
          modelValue: a.value,
          placeholder: "OpenAPI/Swagger/Postman URL or cURL",
          onDelete: $,
          "onUpdate:modelValue": W
        }, null, 8, ["modelValue"])) : (o(), d(p, { key: 1 }, [
          C("div", ce, [
            t[3] || (t[3] = C("div", { class: "text-c-2 min-h-8 w-full py-2 pl-12 text-center text-xs" }, " Preview ", -1)),
            g(l(P), {
              class: "hover:bg-b-2 relative ml-auto max-h-8 gap-1.5 p-2 text-xs",
              variant: "ghost",
              onClick: t[0] || (t[0] = (c) => a.value = "")
            }, {
              default: m(() => [...t[2] || (t[2] = [
                r(" Clear ", -1)
              ])]),
              _: 1
            })
          ]),
          n.value && !i.value ? (o(), S(l(G), {
            key: 0,
            class: "bg-b-2 mt-1 max-h-[40dvh] rounded border px-2 py-1 text-sm",
            content: a.value,
            copy: !1,
            lang: U.value
          }, null, 8, ["content", "lang"])) : X("", !0)
        ], 64))
      ]),
      _: 1
    }, 8, ["disabled", "loader"]));
  }
});
export {
  Ie as default
};
