// @scalar/api-reference/dist/components/Content/Authentication/Authentication.vue2.js
import { defineComponent as f, computed as v, watch as _, onServerPrefetch as k, useSSRContext as x, unref as n, openBlock as a, createElementBlock as S, createElementVNode as h, createVNode as w, createBlock as A, createCommentVNode as i } from "vue";
import B from "../../../legacy/components/SecurityScheme.vue.js";
import C from "../../../legacy/components/SecuritySchemeSelector.vue.js";
import { sleep as N } from "../../../helpers/sleep.js";
import { hasSecuritySchemes as V } from "../../../helpers/hasSecuritySchemes.js";
import { useAuthenticationStore as E } from "../../../legacy/stores/useAuthenticationStore.js";
const b = { key: 0 }, G = { class: "authentication-header" }, P = { class: "selector" }, R = {
  key: 0,
  class: "authentication-content"
}, H = /* @__PURE__ */ f({
  __name: "Authentication",
  props: {
    parsedSpec: {},
    proxy: {}
  },
  setup(d) {
    const s = d, { authentication: r, setAuthentication: l } = E(), y = v(() => {
      var t, c, o;
      if (!r.preferredSecurityScheme)
        return !1;
      const e = (o = (c = (t = s.parsedSpec) == null ? void 0 : t.components) == null ? void 0 : c.securitySchemes) == null ? void 0 : o[r.preferredSecurityScheme];
      return !!e && "type" in e && !!e.type;
    });
    return _(
      () => {
        var e, t;
        return (t = (e = s.parsedSpec) == null ? void 0 : e.components) == null ? void 0 : t.securitySchemes;
      },
      () => {
        var e, t;
        l({
          securitySchemes: (t = (e = s.parsedSpec) == null ? void 0 : e.components) == null ? void 0 : t.securitySchemes
        });
      },
      { deep: !0, immediate: !0 }
    ), k(async () => {
      const e = x();
      await N(1), e.payload.data["useGlobalStore-authentication"] = r;
    }), (e, t) => {
      var c, o, p, u, m;
      return n(V)(e.parsedSpec) ? (a(), S("div", b, [
        h("div", G, [
          h("div", P, [
            w(C, {
              value: (o = (c = e.parsedSpec) == null ? void 0 : c.components) == null ? void 0 : o.securitySchemes
            }, null, 8, ["value"])
          ])
        ]),
        y.value ? (a(), S("div", R, [
          n(r).preferredSecurityScheme ? (a(), A(B, {
            key: 0,
            proxy: e.proxy,
            value: (m = (u = (p = e.parsedSpec) == null ? void 0 : p.components) == null ? void 0 : u.securitySchemes) == null ? void 0 : m[n(r).preferredSecurityScheme]
          }, null, 8, ["proxy", "value"])) : i("", !0)
        ])) : i("", !0)
      ])) : i("", !0);
    };
  }
});
export {
  H as default
};
