// @scalar/components/dist/components/ScalarForm/ScalarFormInput.vue.js
import { defineComponent as s, createBlock as i, openBlock as a, resolveDynamicComponent as p, mergeProps as c, unref as e, withCtx as d, renderSlot as f } from "vue";
import { useBindCx as l, cva as m } from "@scalar/use-hooks/useBindCx";
import { useFormGroupInput as b } from "./useFormGroups.js";
const x = {}, y = /* @__PURE__ */ s({
  ...x,
  inheritAttrs: !1,
  __name: "ScalarFormInput",
  props: {
    is: { default: "button" }
  },
  setup(t) {
    const { cx: o } = l(), r = b(), n = m({
      base: [
        // Layout
        "bg-b-1.5 flex items-center text-c-2 gap-0.75 px-3 py-2.5 ",
        //Focus
        "outline-offset-[-1px] has-[:focus-visible]:outline"
      ],
      variants: {
        grouped: {
          true: "first:rounded-t-[inherit] last:rounded-b-[inherit]",
          false: "rounded border"
        },
        button: { true: "cursor-pointer hover:bg-b-2" }
      }
    });
    return (u, g) => (a(), i(p(t.is), c({
      type: t.is === "button" ? "button" : void 0
    }, e(o)(e(n)({ button: t.is === "button", grouped: e(r) }))), {
      default: d(() => [
        f(u.$slots, "default")
      ]),
      _: 3
    }, 16, ["type"]));
  }
});
export {
  y as default
};
