// @scalar/components/dist/components/ScalarForm/ScalarFormError.vue.js
import { defineComponent as i, computed as l, createElementBlock as s, openBlock as e, mergeProps as d, unref as a, createBlock as m, renderSlot as u, resolveDynamicComponent as f } from "vue";
import { ScalarIconWarningCircle as v, ScalarIconWarning as g } from "@scalar/icons";
import { cva as b, useBindCx as p } from "@scalar/use-hooks/useBindCx";
const C = /* @__PURE__ */ i({
  inheritAttrs: !1,
  __name: "ScalarFormError",
  props: {
    variant: { default: "error" },
    icon: {}
  },
  setup(r) {
    const n = l(() => r.icon ? r.icon : r.variant === "error" ? v : g), t = b({
      base: "flex items-baseline gap-1.5 rounded border p-2",
      variants: {
        variant: {
          error: "border-c-danger bg-b-danger text-c-danger",
          warning: "border-c-alert bg-b-alert text-c-alert"
        }
      }
    }), { cx: o } = p();
    return (c, x) => (e(), s("div", d({ role: "alert" }, a(o)(a(t)({ variant: r.variant }))), [
      (e(), m(f(n.value), {
        class: "relative top-0.5 shrink-0",
        weight: "bold"
      })),
      u(c.$slots, "default")
    ], 16));
  }
});
export {
  C as default
};
