// @scalar/api-client/dist/v2/features/operation/Operation.vue.js
import { defineComponent as y, computed as n, toValue as o, onMounted as S, createBlock as x, createElementBlock as k, openBlock as c, unref as i, createElementVNode as g } from "vue";
import { getResolvedRef as s } from "@scalar/workspace-store/helpers/get-resolved-ref";
import { mapHiddenClientsConfig as C } from "../modal/helpers/map-hidden-clients-config.js";
import { combineParams as B } from "./helpers/combine-params.js";
import { getSelectedServer as w } from "./helpers/get-selected-server.js";
import { getServers as U } from "../../helpers/get-servers.js";
import p from "../../blocks/operation-block/OperationBlock.vue.js";
const b = {
  key: 1,
  class: "flex h-full w-full items-center justify-center"
}, N = {}, $ = /* @__PURE__ */ y({
  ...N,
  __name: "Operation",
  props: {
    documentSlug: {},
    document: {},
    eventBus: {},
    layout: {},
    path: {},
    method: {},
    exampleName: {},
    environment: {},
    securitySchemes: {},
    workspaceStore: {},
    activeWorkspace: {},
    plugins: {},
    options: {}
  },
  setup(e) {
    const u = n(() => e.path ? s(e.document?.paths?.[e.path]) : null), l = n(() => {
      if (!e.path || !e.method)
        return null;
      const t = s(e.document?.paths?.[e.path]?.[e.method]);
      if (!t)
        return null;
      if (!u.value)
        return t;
      const a = B(
        u.value.parameters,
        t.parameters
      );
      return { ...t, parameters: a };
    }), m = n(() => [
      ...(e.workspaceStore.workspace?.["x-scalar-cookies"] ?? []).map((t) => ({
        ...t,
        location: "workspace"
      })),
      ...(e.document?.["x-scalar-cookies"] ?? []).map((t) => ({
        ...t,
        location: "document"
      }))
    ]), r = n(() => {
      const t = o(e.options)?.servers ?? l.value?.servers ?? u.value?.servers ?? e.document?.servers;
      return U(t, {
        baseServerUrl: o(e.options)?.baseServerURL,
        documentUrl: e.document?.["x-scalar-original-source-url"]
      });
    }), d = n(
      () => w(e.document, r.value)
    );
    S(() => {
      typeof e.document?.["x-scalar-selected-server"] > "u" && r.value?.[0]?.url && e.eventBus.emit("server:update:selected", { url: r.value[0].url });
    });
    const v = n(() => e.document?.["x-scalar-set-operation-security"] ? {
      type: "operation",
      path: e.path ?? "",
      method: e.method ?? "get"
    } : {
      type: "document"
    }), h = n(
      () => C(o(e.options)?.hiddenClients)
    ), f = "2.22.3";
    return (t, a) => e.path && e.method && e.exampleName && l.value ? (c(), x(i(p), {
      key: 0,
      appVersion: i(f),
      authMeta: v.value,
      documentSecurity: e.document?.security ?? [],
      documentSelectedSecurity: e.document?.["x-scalar-selected-security"],
      documentUrl: e.document?.["x-scalar-original-source-url"],
      environment: e.environment,
      eventBus: e.eventBus,
      exampleKey: e.exampleName,
      globalCookies: m.value,
      hideClientButton: o(e.options)?.hideClientButton ?? !1,
      history: [],
      httpClients: h.value,
      layout: e.layout,
      method: e.method,
      operation: l.value,
      path: e.path,
      plugins: e.plugins,
      proxyUrl: e.workspaceStore.workspace["x-scalar-active-proxy"] ?? "",
      securitySchemes: e.securitySchemes,
      selectedClient: e.workspaceStore.workspace["x-scalar-default-client"],
      server: d.value,
      servers: r.value,
      setOperationSecurity: e.document?.["x-scalar-set-operation-security"] ?? !1,
      "onUpdate:servers": a[0] || (a[0] = (V) => e.eventBus.emit("ui:route:page", { name: "document.servers" }))
    }, null, 8, ["appVersion", "authMeta", "documentSecurity", "documentSelectedSecurity", "documentUrl", "environment", "eventBus", "exampleKey", "globalCookies", "hideClientButton", "httpClients", "layout", "method", "operation", "path", "plugins", "proxyUrl", "securitySchemes", "selectedClient", "server", "servers", "setOperationSecurity"])) : (c(), k("div", b, [...a[1] || (a[1] = [
      g("span", { class: "text-c-3" }, "Select an operation to view details", -1)
    ])]));
  }
});
export {
  $ as default
};
