// @scalar/components/dist/components/ScalarTextInput/ScalarTextInput.vue.js
import { defineComponent as y, mergeModels as x, useModel as v, ref as $, onMounted as k, createBlock as w, openBlock as s, unref as i, mergeProps as f, withCtx as _, createElementVNode as d, renderSlot as n, createElementBlock as l, createCommentVNode as a, withDirectives as C, vModelDynamic as g, toDisplayString as V } from "vue";
import { useBindCx as b } from "@scalar/use-hooks/useBindCx";
import B from "../ScalarForm/ScalarFormInput.vue.js";
const M = { class: "flex flex-1 relative" }, A = {
  key: 0,
  class: "select-none whitespace-nowrap text-sm text-transparent"
}, D = ["aria-readonly", "readonly"], S = {
  key: 1,
  class: "absolute inset-0 select-none overflow-hidden whitespace-nowrap text-sm"
}, E = {
  key: 0,
  class: "text-c-2"
}, N = { class: "text-transparent" }, z = {
  key: 1,
  class: "text-c-2"
}, I = {}, j = /* @__PURE__ */ y({
  ...I,
  inheritAttrs: !1,
  __name: "ScalarTextInput",
  props: /* @__PURE__ */ x({
    readonly: { type: Boolean }
  }, {
    modelValue: {},
    modelModifiers: {}
  }),
  emits: ["update:modelValue"],
  setup(t) {
    const r = v(t, "modelValue"), o = $(), { stylingAttrsCx: m, otherAttrs: u } = b();
    k(() => {
      "autofocus" in u.value && o.value?.focus();
    });
    function p() {
      t.readonly ? o.value?.select() : o.value?.focus();
    }
    return (e, c) => (s(), w(i(B), f(
      { is: "div" },
      i(m)(
        "cursor-text text-c-1",
        t.readonly ? "" : "focus-within:bg-b-1"
      ),
      { onClick: p }
    ), {
      default: _(() => [
        d("div", M, [
          e.$slots.prefix ? (s(), l("span", A, [
            n(e.$slots, "prefix")
          ])) : a("", !0),
          C(d("input", f({
            ref_key: "input",
            ref: o,
            "onUpdate:modelValue": c[0] || (c[0] = (h) => r.value = h),
            "aria-readonly": t.readonly || void 0,
            class: "z-1 min-w-0 flex-1 border-none bg-transparent text-sm placeholder:font-[inherit] focus-within:outline-none",
            readonly: t.readonly
          }, i(u)), null, 16, D), [
            [g, r.value]
          ]),
          e.$slots.prefix || e.$slots.suffix ? (s(), l("div", S, [
            e.$slots.prefix ? (s(), l("span", E, [
              n(e.$slots, "prefix")
            ])) : a("", !0),
            d("span", N, V(r.value || e.$attrs.placeholder), 1),
            e.$slots.suffix ? (s(), l("span", z, [
              n(e.$slots, "suffix")
            ])) : a("", !0)
          ])) : a("", !0)
        ]),
        n(e.$slots, "aside")
      ]),
      _: 3
    }, 16));
  }
});
export {
  j as default
};
