// @scalar/sidebar/dist/helpers/create-sidebar-state.js
import { computed as v, toValue as k, ref as c } from "vue";
import { generateReverseIndex as S } from "./generate-reverse-index.js";
const g = (o, t) => {
  const s = v(
    () => S({
      items: k(o),
      nestedKey: t?.key ?? "children"
    })
  ), a = c({}), n = c({}), u = c(null), m = (e) => {
    const f = (r) => {
      r && (a.value[r.id] = !0, "parent" in r && r.parent && f(r.parent));
    };
    t?.hooks?.onBeforeSelect && t.hooks.onBeforeSelect(e), a.value = {}, u.value = e, e !== null && f(s.value.get(e)), t?.hooks?.onAfterSelect && t.hooks.onAfterSelect(e);
  }, p = (e, f) => {
    const r = (l) => {
      l && (n.value[l.id] = !0, "parent" in l && l.parent && r(l.parent));
    };
    t?.hooks?.onBeforeExpand && t.hooks.onBeforeExpand(e), f === !1 ? n.value[e] = !1 : r(s.value.get(e)), t?.hooks?.onAfterExpand && t.hooks.onAfterExpand(e);
  }, x = (e) => n.value[e] ?? !1, d = (e) => a.value[e] ?? !1, h = (e) => s.value.get(e);
  return {
    items: v(() => k(o)),
    index: s,
    selectedItems: a,
    expandedItems: n,
    selectedItem: u,
    setSelected: m,
    setExpanded: p,
    isExpanded: x,
    isSelected: d,
    getEntryById: h,
    reset: () => {
      a.value = {}, n.value = {};
    }
  };
};
export {
  g as createSidebarState
};
