// @scalar/api-reference/dist/legacy/components/CardFormTextInput.vue2.js
import { defineComponent as p, ref as d, computed as i, openBlock as s, createElementBlock as u, createElementVNode as a, renderSlot as c, mergeProps as m, createBlock as f, unref as v, createCommentVNode as k } from "vue";
import { ScalarIconButton as w } from "@scalar/components";
const y = { class: "card-form-input" }, h = ["for"], _ = ["id", "type"], b = /* @__PURE__ */ p({
  inheritAttrs: !1,
  __name: "CardFormTextInput",
  props: {
    id: {},
    type: {}
  },
  setup(l) {
    const o = l, e = d(!0), n = i(
      () => o.type === "password" ? e.value ? "password" : "text" : o.type ?? "text"
    );
    return (t, r) => (s(), u("div", y, [
      a("label", { for: t.id }, [
        c(t.$slots, "default", {}, void 0, !0)
      ], 8, h),
      a("input", m(t.$attrs, {
        id: t.id,
        autocomplete: "off",
        "data-1p-ignore": "",
        spellcheck: "false",
        type: n.value
      }), null, 16, _),
      t.type === "password" ? (s(), f(v(w), {
        key: 0,
        class: "password-mask",
        icon: e.value ? "Show" : "Hide",
        label: e.value ? "Show Password" : "Hide Password",
        onClick: r[0] || (r[0] = (B) => e.value = !e.value)
      }, null, 8, ["icon", "label"])) : k("", !0)
    ]));
  }
});
export {
  b as default
};
