// @scalar/api-reference/dist/legacy/components/SecuritySchemeScopes.vue2.js
import { defineComponent as y, ref as h, computed as x, openBlock as d, createBlock as i, unref as t, withCtx as n, createElementVNode as e, normalizeClass as C, createVNode as p, createTextVNode as _, toDisplayString as r, Teleport as V, normalizeStyle as B, createElementBlock as L, Fragment as O, renderList as I, createCommentVNode as N, pushScopeId as j, popScopeId as z } from "vue";
import { useFloating as E, autoUpdate as F, offset as U, flip as D, shift as T } from "@floating-ui/vue";
import { Listbox as M, ListboxButton as q, ListboxOptions as A, ListboxOption as G } from "@headlessui/vue";
import { ScalarIcon as H } from "@scalar/components";
import J from "./CardFormButton.vue.js";
const K = (o) => (j("data-v-6d6455a4"), o = o(), z(), o), P = { class: "scopes-label" }, Q = /* @__PURE__ */ K(() => /* @__PURE__ */ e("em", null, "/", -1)), R = ["checked"], W = { class: "dropdown-item-title" }, X = { class: "dropdown-item-description" }, oe = /* @__PURE__ */ y({
  __name: "SecuritySchemeScopes",
  props: {
    scopes: {},
    selected: {}
  },
  emits: ["update:selected"],
  setup(o, { emit: w }) {
    const g = o, b = w, m = h(), u = h(), { floatingStyles: v } = E(m, u, {
      placement: "bottom-end",
      whileElementsMounted: F,
      middleware: [U(5), D(), T()]
    }), c = x({
      get: () => g.selected,
      set: (s) => b("update:selected", s)
    });
    return (s, f) => Object.keys(s.scopes ?? {}).length ? (d(), i(t(M), {
      key: 0,
      modelValue: c.value,
      "onUpdate:modelValue": f[0] || (f[0] = (l) => c.value = l),
      multiple: ""
    }, {
      default: n(({ open: l }) => [
        e("div", {
          ref_key: "trigger",
          ref: m,
          class: C(["wrapper", { "wrapper-open": l }])
        }, [
          p(t(q), { as: J }, {
            default: n(() => [
              e("div", P, [
                _(" Scopes " + r(c.value.length), 1),
                Q,
                _(r(Object.entries(s.scopes ?? {}).length) + " ", 1),
                p(t(H), {
                  icon: l ? "ChevronUp" : "ChevronDown",
                  size: "sm"
                }, null, 8, ["icon"])
              ])
            ]),
            _: 2
          }, 1024)
        ], 2),
        (d(), i(V, { to: "body" }, [
          e("div", {
            ref_key: "dropdown",
            ref: u,
            class: "floating scalar-app",
            style: B(t(v))
          }, [
            p(t(A), {
              as: "dl",
              class: "dropdown"
            }, {
              default: n(() => [
                (d(!0), L(O, null, I(Object.entries(s.scopes ?? {}), ([a, S]) => (d(), i(t(G), {
                  key: a,
                  as: "div",
                  class: "dropdown-item",
                  value: a
                }, {
                  default: n(({ selected: k }) => [
                    e("input", {
                      checked: k,
                      class: "dropdown-item-check",
                      tabindex: "-1",
                      type: "checkbox"
                    }, null, 8, R),
                    e("dt", W, r(a), 1),
                    e("dd", X, r(S), 1)
                  ]),
                  _: 2
                }, 1032, ["value"]))), 128))
              ]),
              _: 1
            })
          ], 4)
        ]))
      ]),
      _: 1
    }, 8, ["modelValue"])) : N("", !0);
  }
});
export {
  oe as default
};
