// @scalar/api-reference/dist/legacy/components/SecuritySchemeSelector.vue2.js
import { defineComponent as N, onMounted as O, onServerPrefetch as k, computed as B, openBlock as i, createElementBlock as a, toDisplayString as d, createElementVNode as y, unref as c, createVNode as j, withModifiers as K, Fragment as P, renderList as V, createCommentVNode as $, pushScopeId as D, popScopeId as E } from "vue";
import { ScalarIcon as F } from "@scalar/components";
import { useAuthenticationStore as H } from "../stores/useAuthenticationStore.js";
const M = (r) => (D("data-v-630d0ab4"), r = r(), E(), r), T = {
  key: 0,
  class: "security-scheme-label"
}, z = {
  key: 1,
  class: "security-scheme-selector"
}, q = { class: "security-scheme-label" }, G = ["value"], J = /* @__PURE__ */ M(() => /* @__PURE__ */ y("option", { value: "" }, "No Authentication", -1)), Q = ["value"], X = /* @__PURE__ */ N({
  __name: "SecuritySchemeSelector",
  props: {
    value: {}
  },
  emits: ["input"],
  setup(r, { emit: v }) {
    const l = r, m = v, { authentication: o, setAuthentication: A } = H(), _ = (e) => {
      p(e.target.value);
    };
    O(() => {
      o.preferredSecurityScheme || p(Object.keys(l.value ?? {})[0] ?? null);
    });
    const p = (e) => {
      A({
        preferredSecurityScheme: e
      }), m("input", e);
    };
    k(
      () => p(Object.keys(l.value ?? {})[0] ?? null)
    );
    const w = (e) => !(e != null && e.type), b = (e) => e.type.toLowerCase() === "apikey", C = (e) => e.type === "http" && e.scheme.toLowerCase() === "basic" || e.type.toLowerCase() === "basic", I = (e) => e.type === "http" && e.scheme.toLowerCase() === "bearer", L = (e) => e.type.toLowerCase() === "oauth2", h = (e, t) => `${g(e)} (${t})`, g = (e) => {
      var t, s;
      return w(e) ? "No Authentication" : b(e) ? "API Key" : C(e) ? "Basic Authentication" : I(e) ? "Bearer Authentication" : L(e) ? (t = e.flows) != null && t.implicit ? "OAuth 2.0 Implicit" : (s = e.flows) != null && s.password ? "OAuth 2.0 Password" : "OAuth 2.0" : `${e.type} (not yet supported)`;
    }, n = B(() => Object.keys(l.value ?? {}));
    return (e, t) => {
      var s, f;
      return n.value.length === 1 ? (i(), a("div", T, d(h((s = e.value) == null ? void 0 : s[n.value[0]], n.value[0])), 1)) : n.value.length > 1 ? (i(), a("div", z, [
        y("span", q, d(c(o).preferredSecurityScheme ? h(
          (f = e.value) == null ? void 0 : f[c(o).preferredSecurityScheme],
          c(o).preferredSecurityScheme
        ) : "No Authentication"), 1),
        j(c(F), { icon: "ChevronDown" }),
        y("select", {
          value: c(o).preferredSecurityScheme,
          onClick: t[0] || (t[0] = K(() => {
          }, ["prevent"])),
          onInput: _
        }, [
          J,
          (i(!0), a(P, null, V(n.value, (u) => {
            var S;
            return i(), a("option", {
              key: u,
              value: u ?? null
            }, d(h((S = e.value) == null ? void 0 : S[u], u)), 9, Q);
          }), 128))
        ], 40, G)
      ])) : $("", !0);
    };
  }
});
export {
  X as default
};
