// @scalar/api-reference/dist/legacy/components/SecurityScheme.vue2.js
import { defineComponent as J, computed as Q, openBlock as n, createBlock as w, withCtx as s, unref as u, createElementBlock as P, Fragment as C, createTextVNode as c, toDisplayString as X, createCommentVNode as f, createVNode as i, createElementVNode as W } from "vue";
import { concatenateUrlAndPath as Y, redirectToProxy as Z } from "@scalar/oas-utils/helpers";
import { useToasts as _ } from "@scalar/use-toasts";
import x from "./CardForm.vue.js";
import ee from "./CardFormButton.vue.js";
import b from "./CardFormGroup.vue.js";
import j from "./CardFormRows.vue.js";
import v from "./CardFormTextInput.vue.js";
import K from "./SecuritySchemeScopes.vue.js";
import { getUrlFromServerState as te } from "../helpers/getUrlFromServerState.js";
import { useServerStore as ae } from "../stores/useServerStore.js";
import { useAuthenticationStore as oe } from "../stores/useAuthenticationStore.js";
const fe = /* @__PURE__ */ J({
  __name: "SecurityScheme",
  props: {
    value: {},
    proxy: {}
  },
  setup(se) {
    const { toast: R } = _(), { server: L } = ae(), { authentication: t, setAuthentication: p } = oe(), N = (e) => {
      p({
        apiKey: {
          ...t.apiKey,
          token: e.target.value
        }
      });
    }, V = (e) => {
      p({
        http: {
          ...t.http,
          basic: {
            ...t.http.basic,
            username: e.target.value
          }
        }
      });
    }, D = (e) => {
      p({
        http: {
          ...t.http,
          basic: {
            ...t.http.basic,
            password: e.target.value
          }
        }
      });
    }, E = (e) => {
      p({
        http: {
          ...t.http,
          bearer: {
            ...t.http.bearer,
            token: e.target.value
          }
        }
      });
    }, S = (e) => {
      p({
        oAuth2: {
          ...t.oAuth2,
          clientId: e.target.value
        }
      });
    }, H = (e) => {
      p({
        oAuth2: {
          ...t.oAuth2,
          username: e.target.value
        }
      });
    }, G = (e) => {
      p({
        oAuth2: {
          ...t.oAuth2,
          password: e.target.value
        }
      });
    };
    function M(e) {
      const a = t.oAuth2.scopes.join(" "), l = (Math.random() + 1).toString(36).substring(7), o = new URL(e.authorizationUrl);
      return p({
        oAuth2: { ...t.oAuth2, state: l }
      }), o.searchParams.set("response_type", "token"), o.searchParams.set("client_id", t.oAuth2.clientId), o.searchParams.set("redirect_uri", window.location.href), o.searchParams.set("scope", a), o.searchParams.set("state", l), o.toString();
    }
    function $(e, a) {
      if (typeof e != "string") {
        console.log("tokenUrl is not a string");
        return;
      }
      const l = a != null && a.baseUrl && !e.startsWith("http") ? Y(a == null ? void 0 : a.baseUrl, e) : e, o = new URL(l);
      o.searchParams.set("grant_type", "password"), o.searchParams.set("username", t.oAuth2.username), o.searchParams.set("password", t.oAuth2.password), o.searchParams.set("client_id", t.oAuth2.clientId), o.searchParams.set("scope", t.oAuth2.scopes.join(" ")), fetch(
        a != null && a.proxy ? Z(a == null ? void 0 : a.proxy, o.toString()) : o.toString(),
        {
          method: "POST",
          headers: {
            "Content-Type": "application/x-www-form-urlencoded"
          }
        }
      ).then((r) => {
        if (!r.ok)
          throw new Error(
            "Failed to get an access token. Please check your credentials."
          );
        return r.json();
      }).then((r) => {
        p({
          oAuth2: { ...t.oAuth2, accessToken: r.access_token }
        });
      }).catch((r) => {
        R("Couldn’t retrieve the password grant token", "warn", {
          description: "Open your browser console to get more information."
        }), console.error("[authorizeWithPassword]", r);
      });
    }
    const I = Q({
      get: () => t.oAuth2.scopes,
      set: (e) => p({ oAuth2: { ...t.oAuth2, scopes: e } })
    }), q = (e) => {
      const l = window.open(e, "openAuth2Window", "left=100,top=100,width=800,height=600");
      if (l) {
        const o = setInterval(function() {
          var r;
          try {
            const h = new URLSearchParams(l.location.href).get("access_token");
            if (l.closed || h) {
              clearInterval(o);
              const m = (r = l.location.href.match(/state=([^&]*)/)) == null ? void 0 : r[1];
              h && t.oAuth2.state === m && p({
                oAuth2: { ...t.oAuth2, accessToken: h }
              }), l.close();
            }
          } catch {
          }
        }, 200);
      }
    };
    return (e, a) => {
      var l;
      return e.value && ((l = e.value) != null && l.type) ? (n(), w(x, { key: 0 }, {
        default: s(() => [
          e.value.type === "apiKey" ? (n(), w(v, {
            key: 0,
            id: `security-scheme-${e.value.name}`,
            placeholder: "Token",
            type: "password",
            value: u(t).apiKey.token,
            onInput: N
          }, {
            default: s(() => {
              var o, r, d;
              return [
                e.value.in ? (n(), P(C, { key: 0 }, [
                  c(X(((r = (o = e.value.in) == null ? void 0 : o.charAt(0)) == null ? void 0 : r.toUpperCase()) + ((d = e.value.in) == null ? void 0 : d.slice(1))), 1)
                ], 64)) : f("", !0),
                c(" API ")
              ];
            }),
            _: 1
          }, 8, ["id", "value"])) : e.value.type === "http" || e.value.type === "basic" ? (n(), P(C, { key: 1 }, [
            e.value.type === "basic" || e.value.scheme === "basic" ? (n(), w(b, { key: 0 }, {
              default: s(() => [
                i(v, {
                  id: "http.basic.username",
                  placeholder: "Username",
                  value: u(t).http.basic.username,
                  onInput: V
                }, {
                  default: s(() => [
                    c(" Username ")
                  ]),
                  _: 1
                }, 8, ["value"]),
                i(v, {
                  id: "http.basic.password",
                  placeholder: "Password",
                  type: "password",
                  value: u(t).http.basic.password,
                  onInput: D
                }, {
                  default: s(() => [
                    c(" Password ")
                  ]),
                  _: 1
                }, 8, ["value"])
              ]),
              _: 1
            })) : e.value.type === "http" && e.value.scheme === "bearer" ? (n(), w(v, {
              key: 1,
              id: "http.bearer.token",
              placeholder: "Token",
              type: "password",
              value: u(t).http.bearer.token,
              onInput: E
            }, {
              default: s(() => [
                c(" Bearer Token ")
              ]),
              _: 1
            }, 8, ["value"])) : f("", !0)
          ], 64)) : e.value.type.toLowerCase() === "oauth2" && e.value.flows ? (n(), w(b, { key: 2 }, {
            default: s(() => {
              var o, r;
              return [
                e.value.flows.implicit ? (n(), P(C, { key: 0 }, [
                  u(t).oAuth2.accessToken ? (n(), P(C, { key: 0 }, [
                    i(v, {
                      id: "oAuth2.accessToken",
                      placeholder: "xxxxx",
                      type: "password",
                      value: u(t).oAuth2.accessToken
                    }, {
                      default: s(() => [
                        c(" Access Token ")
                      ]),
                      _: 1
                    }, 8, ["value"]),
                    i(ee, {
                      onClick: a[0] || (a[0] = () => u(p)({
                        oAuth2: {
                          ...u(t).oAuth2,
                          accessToken: "",
                          state: ""
                        }
                      }))
                    }, {
                      default: s(() => [
                        c(" Reset ")
                      ]),
                      _: 1
                    })
                  ], 64)) : f("", !0)
                ], 64)) : f("", !0),
                (r = (o = e.value) == null ? void 0 : o.flows) != null && r.password ? (n(), w(j, { key: 1 }, {
                  default: s(() => [
                    i(b, null, {
                      default: s(() => [
                        i(v, {
                          id: "oAuth2.username",
                          placeholder: "Username",
                          value: u(t).oAuth2.username,
                          onInput: H
                        }, {
                          default: s(() => [
                            c(" Username ")
                          ]),
                          _: 1
                        }, 8, ["value"]),
                        i(v, {
                          id: "oAuth2.password",
                          placeholder: "Password",
                          type: "password",
                          value: u(t).oAuth2.password,
                          onInput: G
                        }, {
                          default: s(() => [
                            c(" Password ")
                          ]),
                          _: 1
                        }, 8, ["value"])
                      ]),
                      _: 1
                    }),
                    i(b, null, {
                      default: s(() => {
                        var d, h, m, A, y, k, T, U, F, B, O;
                        return [
                          i(v, {
                            id: "oAuth2.clientId",
                            placeholder: "12345",
                            type: "text",
                            value: u(t).oAuth2.clientId,
                            onInput: S
                          }, {
                            default: s(() => [
                              c(" Client ID ")
                            ]),
                            _: 1
                          }, 8, ["value"]),
                          e.value !== void 0 && Object.entries(
                            ((m = (h = (d = e.value) == null ? void 0 : d.flows) == null ? void 0 : h.implicit) == null ? void 0 : m.scopes) ?? ((k = (y = (A = e.value) == null ? void 0 : A.flows) == null ? void 0 : y.password) == null ? void 0 : k.scopes) ?? {}
                          ).length > 0 ? (n(), w(K, {
                            key: 0,
                            selected: I.value,
                            "onUpdate:selected": a[1] || (a[1] = (g) => I.value = g),
                            scopes: ((F = (U = (T = e.value) == null ? void 0 : T.flows) == null ? void 0 : U.implicit) == null ? void 0 : F.scopes) ?? ((O = (B = e.value) == null ? void 0 : B.flows) == null ? void 0 : O.password.scopes)
                          }, null, 8, ["selected", "scopes"])) : f("", !0),
                          W("button", {
                            class: "cardform-auth-button",
                            type: "button",
                            onClick: a[2] || (a[2] = () => {
                              var g, z;
                              return $(
                                (z = (g = e.value.flows) == null ? void 0 : g.password) == null ? void 0 : z.tokenUrl,
                                {
                                  baseUrl: u(te)(u(L)),
                                  proxy: e.proxy
                                }
                              );
                            })
                          }, " Authorize ")
                        ];
                      }),
                      _: 1
                    })
                  ]),
                  _: 1
                })) : (n(), w(j, { key: 2 }, {
                  default: s(() => [
                    i(b, null, {
                      default: s(() => {
                        var d, h;
                        return [
                          i(v, {
                            id: "oAuth2.clientId",
                            placeholder: "12345",
                            type: "text",
                            value: u(t).oAuth2.clientId,
                            onInput: S
                          }, {
                            default: s(() => [
                              c(" Client ID ")
                            ]),
                            _: 1
                          }, 8, ["value"]),
                          e.value !== void 0 && Object.entries(
                            ((d = e.value.flows.implicit) == null ? void 0 : d.scopes) ?? e.value.flows.password.scopes
                          ).length > 0 ? (n(), w(K, {
                            key: 0,
                            selected: I.value,
                            "onUpdate:selected": a[3] || (a[3] = (m) => I.value = m),
                            scopes: ((h = e.value.flows.implicit) == null ? void 0 : h.scopes) ?? e.value.flows.password.scopes
                          }, null, 8, ["selected", "scopes"])) : f("", !0),
                          W("button", {
                            class: "cardform-auth-button",
                            type: "button",
                            onClick: a[4] || (a[4] = () => {
                              var m, A, y, k;
                              return q(
                                M(
                                  ((A = (m = e.value) == null ? void 0 : m.flows) == null ? void 0 : A.implicit) ?? ((k = (y = e.value) == null ? void 0 : y.flows) == null ? void 0 : k.password)
                                )
                              );
                            })
                          }, " Authorize ")
                        ];
                      }),
                      _: 1
                    })
                  ]),
                  _: 1
                }))
              ];
            }),
            _: 1
          })) : f("", !0)
        ]),
        _: 1
      })) : f("", !0);
    };
  }
});
export {
  fe as default
};
