// @scalar/api-client/dist/components/ImportCollection/DropEventListener.vue.js
import { defineComponent as g, ref as y, onMounted as x, onBeforeUnmount as h, createBlock as D, openBlock as p, Transition as E, withCtx as T, createElementBlock as _, createCommentVNode as L, createElementVNode as i, createVNode as C, unref as b } from "vue";
import { ScalarIcon as k } from "@scalar/components";
const A = {
  key: 0,
  class: "bg-b-2 fixed right-1/2 bottom-1/2 z-50 h-64 w-64 translate-x-1/2 translate-y-1/2 rounded-xl border transition-opacity duration-200 md:right-10 md:bottom-10 md:translate-x-0 md:translate-y-0"
}, w = { class: "flex h-full flex-col items-center justify-center" }, V = /* @__PURE__ */ g({
  __name: "DropEventListener",
  emits: ["input"],
  setup(B, { emit: v }) {
    const s = v, n = y(!1);
    let a = 0;
    x(() => {
      document.addEventListener("dragenter", f), document.addEventListener("dragleave", c), document.addEventListener("dragover", u), document.addEventListener("drop", l);
    }), h(() => {
      document.removeEventListener("dragenter", f), document.removeEventListener("dragover", u), document.removeEventListener("dragleave", c), document.removeEventListener("drop", l);
    });
    function d(e) {
      return e.dataTransfer?.types.includes("text/html") ?? !1;
    }
    function l(e) {
      if (e.preventDefault(), n.value = !1, a = 0, !d(e) && e.dataTransfer) {
        const t = e.dataTransfer.getData("text").replace(/^blob:/, "");
        if (t)
          s("input", t, null, "drop");
        else if (e.dataTransfer.files.length > 0) {
          const r = e.dataTransfer.files[0];
          if (!r)
            return;
          const m = new FileReader();
          m.onload = (o) => {
            o.target && typeof o.target.result == "string" && s("input", o.target.result, null, "drop");
          }, m.readAsText(r);
        }
      }
    }
    function u(e) {
      e.preventDefault();
    }
    function c(e) {
      e.preventDefault(), a--, a === 0 && (n.value = !1);
    }
    function f(e) {
      if (e.preventDefault(), a++, !d(e)) {
        if (e.dataTransfer) {
          const t = e.dataTransfer.items;
          for (const r of t)
            if (r?.kind === "string" || r?.type?.includes("json") || r?.type?.includes("yml") || r?.type?.includes("yaml")) {
              n.value = !0;
              return;
            }
        }
        n.value = !1;
      }
    }
    return (e, t) => (p(), D(E, {
      enterActiveClass: "transition-opacity duration-200",
      enterFromClass: "opacity-0",
      leaveActiveClass: "transition-opacity duration-200",
      leaveToClass: "opacity-0"
    }, {
      default: T(() => [
        n.value ? (p(), _("div", A, [
          i("div", w, [
            i("div", null, [
              C(b(k), {
                icon: "Download",
                size: "xl",
                thickness: "2"
              })
            ]),
            t[0] || (t[0] = i("div", { class: "text-c-1 m-4 text-center" }, " Drop your OpenAPI document here ", -1))
          ])
        ])) : L("", !0)
      ]),
      _: 1
    }));
  }
});
export {
  V as default
};
