// @scalar/postman-to-openapi/dist/helpers/auth.js
const AUTH_SCHEMES = {
  API_KEY: "apikeyAuth",
  BASIC: "basicAuth",
  BEARER: "bearerAuth",
  OAUTH2: "oauth2Auth"
};
const OAUTH2_DEFAULTS = {
  AUTHORIZE_URL: "/oauth/authorize",
  TOKEN_URL: "/oauth/token"
};
function createApiKeyConfig() {
  return {
    scheme: {
      type: "apiKey",
      name: "api_key",
      in: "header"
    },
    requirement: { [AUTH_SCHEMES.API_KEY]: [] }
  };
}
function createBasicConfig() {
  return {
    scheme: {
      type: "http",
      scheme: "basic"
    },
    requirement: { [AUTH_SCHEMES.BASIC]: [] }
  };
}
function createBearerConfig() {
  return {
    scheme: {
      type: "http",
      scheme: "bearer"
    },
    requirement: { [AUTH_SCHEMES.BEARER]: [] }
  };
}
function createOAuth2Config(auth) {
  if (!auth) {
    return {
      scheme: {},
      requirement: {}
    };
  }
  const oauth2Attrs = auth.oauth2 || [];
  const attrMap = /* @__PURE__ */ new Map();
  oauth2Attrs.forEach((attr) => {
    if (attr.key && attr.value !== void 0) {
      attrMap.set(attr.key, String(attr.value));
    }
  });
  const authUrl = attrMap.get("authUrl") || attrMap.get("authorizationUrl") || OAUTH2_DEFAULTS.AUTHORIZE_URL;
  const tokenUrl = attrMap.get("accessTokenUrl") || attrMap.get("tokenUrl") || OAUTH2_DEFAULTS.TOKEN_URL;
  const scopeValue = attrMap.get("scope") || "";
  const scopes = {};
  if (scopeValue) {
    const scopeList = scopeValue.split(/[,\s]+/).filter(Boolean);
    scopeList.forEach((scope) => {
      scopes[scope] = scope;
    });
  }
  return {
    scheme: {
      type: "oauth2",
      flows: {
        authorizationCode: {
          authorizationUrl: authUrl,
          tokenUrl,
          scopes
        }
      }
    },
    requirement: { [AUTH_SCHEMES.OAUTH2]: Object.keys(scopes) }
  };
}
function createNoAuthConfig() {
  return {
    scheme: {},
    requirement: {}
  };
}
const AUTH_TYPE_HANDLERS = {
  apikey: createApiKeyConfig,
  basic: createBasicConfig,
  bearer: createBearerConfig,
  oauth2: createOAuth2Config,
  noauth: createNoAuthConfig
};
function processAuth(auth) {
  const securitySchemes = {};
  const security = [];
  try {
    const handler = AUTH_TYPE_HANDLERS[auth.type];
    if (!handler) {
      throw new Error(`Unsupported authentication type: ${auth.type}`);
    }
    const { scheme, requirement } = handler(auth);
    if (Object.keys(scheme).length > 0) {
      const schemeKey = `${auth.type}Auth`;
      securitySchemes[schemeKey] = scheme;
      security.push(requirement);
    }
  } catch (error) {
    console.error("Error processing authentication:", error);
    throw error;
  }
  return { securitySchemes, security };
}
export {
  processAuth
};
