// @scalar/postman-to-openapi/dist/helpers/parameters.js
import { inferSchemaType } from "./schemas.js";
function extractParameters(request) {
  const parameters = [];
  const parameterMap = /* @__PURE__ */ new Map();
  if (typeof request === "string" || !request.url) {
    return parameters;
  }
  const url = typeof request.url === "string" ? { raw: request.url } : request.url;
  if (url.query) {
    url.query.forEach((param) => {
      const paramObj = createParameterObject(param, "query");
      if (paramObj.name) {
        parameterMap.set(paramObj.name, paramObj);
      }
    });
  }
  if (url.variable) {
    url.variable.forEach((param) => {
      const paramObj = createParameterObject(param, "path");
      if (paramObj.name) {
        parameterMap.set(paramObj.name, paramObj);
      }
    });
  }
  if (url.path) {
    const pathArray = Array.isArray(url.path) ? url.path : [url.path];
    const extractedVariables = extractPathVariablesFromPathArray(pathArray);
    extractedVariables.forEach((varName) => {
      if (!parameterMap.has(varName)) {
        parameterMap.set(varName, {
          name: varName,
          in: "path",
          required: true,
          schema: {
            type: "string"
          }
        });
      }
    });
  }
  if (request.header && Array.isArray(request.header)) {
    request.header.forEach((header) => {
      const paramObj = createParameterObject(header, "header");
      if (paramObj.name) {
        parameterMap.set(paramObj.name, paramObj);
      }
    });
  }
  return Array.from(parameterMap.values());
}
function extractPathVariablesFromPathArray(pathArray) {
  const variables = [];
  const variableRegex = /{{\s*([\w.-]+)\s*}}/;
  pathArray.forEach((segment) => {
    const segmentString = typeof segment === "string" ? segment : segment.value;
    const match = segmentString.match(variableRegex);
    if (match?.[1]) {
      variables.push(match[1]);
    }
  });
  return variables;
}
function createParameterObject(param, paramIn) {
  const parameter = {
    name: param.key || "",
    in: paramIn,
    description: param.description
  };
  if (paramIn === "path") {
    parameter.required = true;
  } else if (paramIn === "query") {
    const isRequired = param.description?.toLowerCase().includes("[required]") || param.key && param.key.toLowerCase() === "required";
    if (isRequired) {
      parameter.required = true;
      if (parameter.description) {
        parameter.description = parameter.description.replace(/\[required\]/gi, "").trim();
      }
    }
  }
  if (param.value !== void 0) {
    parameter.example = param.value;
    if (paramIn === "path") {
      if (typeof param.value === "number") {
        parameter.schema = inferSchemaType(param.value);
      } else {
        parameter.schema = { type: "string" };
      }
    } else {
      parameter.schema = inferSchemaType(param.value);
    }
  } else {
    parameter.schema = { type: "string" };
  }
  if (param.disabled === true) {
    parameter["x-scalar-disabled"] = true;
  }
  return parameter;
}
export {
  createParameterObject,
  extractParameters
};
