// @scalar/postman-to-openapi/dist/helpers/request-body.js
import { processFormDataSchema } from "./form-data.js";
import { createParameterObject } from "./parameters.js";
function extractRequestBody(body) {
  const requestBody = {
    content: {}
  };
  if (body.mode === "raw") {
    handleRawBody(body, requestBody);
    return requestBody;
  }
  if (body.mode === "formdata" && body.formdata) {
    handleFormDataBody(body.formdata, requestBody);
    return requestBody;
  }
  if (body.mode === "urlencoded" && body.urlencoded) {
    handleUrlEncodedBody(body.urlencoded, requestBody);
    return requestBody;
  }
  return requestBody;
}
function handleRawBody(body, requestBody) {
  const rawBody = body.raw || "";
  const isJsonLanguage = body.options?.raw?.language === "json";
  const hasVariables = /\{\{[\w-]+\}\}/.test(rawBody);
  let jsonBody;
  let isJsonBody = false;
  try {
    jsonBody = JSON.parse(rawBody);
    isJsonBody = true;
  } catch {
  }
  if (isJsonBody) {
    requestBody.content = {
      "application/json": {
        schema: {
          type: "object",
          example: jsonBody
        }
      }
    };
    return;
  }
  if (hasVariables && isJsonLanguage) {
    requestBody.content = {
      "application/json": {
        schema: {
          type: "object",
          description: "Body data set via pre-request script"
        }
      }
    };
    return;
  }
  requestBody.content = {
    "text/plain": {
      schema: {
        type: "string",
        examples: rawBody ? [rawBody] : void 0
      }
    }
  };
}
function handleFormDataBody(formdata, requestBody) {
  requestBody.content = {
    "multipart/form-data": {
      schema: processFormDataSchema(formdata)
    }
  };
}
function handleUrlEncodedBody(urlencoded, requestBody) {
  const schema = {
    type: "object",
    properties: {},
    required: []
  };
  urlencoded.forEach((item) => {
    if (schema.properties) {
      const paramObject = createParameterObject(item, "query");
      const property = {
        type: "string",
        examples: [item.value],
        description: paramObject.description
      };
      if (item.disabled === true) {
        property["x-scalar-disabled"] = true;
      }
      schema.properties[item.key] = property;
      if (paramObject.required) {
        schema.required?.push(item.key);
      }
    }
  });
  requestBody.content = {
    "application/x-www-form-urlencoded": {
      schema
    }
  };
}
export {
  extractRequestBody
};
