// @scalar/postman-to-openapi/dist/helpers/responses.js
import { inferSchemaFromExample } from "./schemas.js";
import { extractStatusCodesFromTests } from "./status-codes.js";
function extractResponses(responses, item) {
  const statusCodes = item ? extractStatusCodesFromTests(item) : [];
  const responseMap = responses.reduce((acc, response) => {
    const statusCode = response.code?.toString() || "default";
    acc[statusCode] = {
      description: response.status || "Successful response",
      headers: extractHeaders(response.header),
      content: {
        "application/json": {
          schema: inferSchemaFromExample(response.body || ""),
          examples: {
            default: tryParseJson(response.body || "")
          }
        }
      }
    };
    return acc;
  }, {});
  statusCodes.forEach((code) => {
    const codeStr = code.toString();
    if (!responseMap[codeStr]) {
      responseMap[codeStr] = {
        description: "Successful response",
        content: {
          "application/json": {}
        }
      };
    }
  });
  if (Object.keys(responseMap).length === 0) {
    return void 0;
  }
  return responseMap;
}
function extractHeaders(headers) {
  if (!headers || typeof headers === "string") {
    return void 0;
  }
  const openapiHeaders = {};
  if (Array.isArray(headers)) {
    headers.forEach((header) => {
      openapiHeaders[header.key] = {
        schema: {
          type: "string",
          examples: [header.value]
        }
      };
    });
  }
  return openapiHeaders;
}
function tryParseJson(jsonString) {
  try {
    return JSON.parse(jsonString);
  } catch (_e) {
    return { rawContent: jsonString };
  }
}
export {
  extractResponses
};
