// @scalar/postman-to-openapi/dist/helpers/servers.js
function createOperationKey(path, method) {
  return `${path}::${method}`;
}
function parseOperationKey(key) {
  const separatorIndex = key.lastIndexOf("::");
  return {
    path: key.substring(0, separatorIndex),
    method: key.substring(separatorIndex + 2)
  };
}
function analyzeServerDistribution(serverUsage, allUniquePaths) {
  const placement = {
    document: [],
    pathItems: /* @__PURE__ */ new Map(),
    operations: /* @__PURE__ */ new Map()
  };
  if (serverUsage.length === 0) {
    return placement;
  }
  const serverMap = /* @__PURE__ */ new Map();
  for (const usage of serverUsage) {
    if (!serverMap.has(usage.serverUrl)) {
      serverMap.set(usage.serverUrl, /* @__PURE__ */ new Set());
    }
    serverMap.get(usage.serverUrl).add(createOperationKey(usage.path, usage.method));
  }
  for (const [serverUrl, operationKeys] of serverMap.entries()) {
    const serverObject = { url: serverUrl };
    const operations = Array.from(operationKeys).map(parseOperationKey);
    const uniquePaths = new Set(operations.map((op) => op.path));
    const pathCount = uniquePaths.size;
    const coversAllPaths = allUniquePaths.size > 0 && uniquePaths.size === allUniquePaths.size && Array.from(uniquePaths).every((path) => allUniquePaths.has(path));
    if (coversAllPaths || pathCount > 1) {
      placement.document.push(serverObject);
    } else if (operations.length > 1) {
      const path = operations[0]?.path;
      if (!path) {
        continue;
      }
      if (!placement.pathItems.has(path)) {
        placement.pathItems.set(path, []);
      }
      placement.pathItems.get(path).push(serverObject);
    } else {
      const operation = operations[0];
      if (!operation) {
        continue;
      }
      if (!placement.operations.has(operation.path)) {
        placement.operations.set(operation.path, /* @__PURE__ */ new Map());
      }
      const methodsMap = placement.operations.get(operation.path);
      if (!methodsMap.has(operation.method)) {
        methodsMap.set(operation.method, []);
      }
      methodsMap.get(operation.method).push(serverObject);
    }
  }
  return placement;
}
export {
  analyzeServerDistribution
};
