// @scalar/postman-to-openapi/dist/helpers/urls.js
import { REGEX } from "@scalar/helpers/regex/regex-helpers";
function parseUrl(urlString) {
  const url = new URL(urlString);
  return {
    protocol: url.protocol,
    hostname: url.hostname,
    port: url.port
  };
}
function getDomainFromUrl(url) {
  const { protocol, hostname, port } = parseUrl(url);
  return `${protocol}//${hostname}${port ? `:${port}` : ""}`;
}
function extractPathFromUrl(url) {
  if (!url) {
    return "/";
  }
  const path = url.replace(/^(?:https?:\/\/)?[^/]+(\/|$)/, "/").split(/[?#]/)[0] ?? "";
  const finalPath = ("/" + path.replace(/\{\{([^{}]{0,1000})\}\}/g, "{$1}").replace(/^\/+/, "")).replace(/\/\/+/g, "/");
  return finalPath;
}
const normalizePath = (path) => path.replace(/:(\w+)/g, "{$1}");
function extractPathParameterNames(path) {
  const params = /* @__PURE__ */ new Set();
  let match;
  while ((match = REGEX.TEMPLATE_VARIABLE.exec(path)) !== null) {
    const param = match[1] || match[2] || match[0].slice(1);
    params.add(param.trim());
  }
  return Array.from(params);
}
function extractServerFromUrl(url) {
  if (!url) {
    return void 0;
  }
  try {
    const protocolMatch = url.match(/^(https?:\/\/)/i);
    const protocol = protocolMatch ? protocolMatch[1] : null;
    const urlMatch = url.match(/^(?:https?:\/\/)?([^/?#]+)/i);
    if (!urlMatch?.[1]) {
      return void 0;
    }
    const hostPart = urlMatch[1];
    const serverUrl = protocol ? `${protocol}${hostPart}`.replace(/\/$/, "") : `https://${hostPart}`.replace(/\/$/, "");
    return serverUrl;
  } catch (error) {
    console.error(`Error extracting server from URL "${url}":`, error);
    return void 0;
  }
}
export {
  extractPathFromUrl,
  extractPathParameterNames,
  extractServerFromUrl,
  getDomainFromUrl,
  normalizePath
};
