// @scalar/api-client/dist/components/Sidebar/SidebarListElement.vue2.js
import { defineComponent as c, resolveComponent as v, openBlock as o, createElementBlock as n, createVNode as t, normalizeClass as l, withModifiers as f, withCtx as g, createElementVNode as r, createCommentVNode as h, toDisplayString as k } from "vue";
import w from "./SidebarListElementActions.vue.js";
import { useRouter as _ } from "vue-router";
const C = {
  key: 0,
  class: "hover:bg-b-3 rounded p-1.5",
  type: "button"
}, y = { class: "empty-variable-name" }, V = /* @__PURE__ */ c({
  __name: "SidebarListElement",
  props: {
    variable: {},
    warningMessage: {}
  },
  emits: ["delete"],
  setup(D, { emit: i }) {
    const s = i, m = _(), p = (e, a) => {
      e.metaKey ? window.open(a, "_blank") : m.push(a);
    }, u = (e) => {
      s("delete", e);
    };
    return (e, a) => {
      const d = v("router-link");
      return o(), n("li", null, [
        t(d, {
          class: l(["h-8 text-c-2 hover:bg-b-2 group relative block flex items-center gap-1 rounded py-1 pr-2 font-medium no-underline", [e.variable.color ? "pl-1" : "pl-2"]]),
          exactActiveClass: "active-link",
          to: `${e.variable.uid}`,
          onClick: a[0] || (a[0] = f((b) => p(b, e.variable.uid), ["prevent"]))
        }, {
          default: g(() => [
            e.variable.color ? (o(), n("button", C, [
              r("div", {
                class: l(["h-2.5 w-2.5 rounded-xl", `bg-${e.variable.color}`])
              }, null, 2)
            ])) : h("", !0),
            r("span", y, k(e.variable.name), 1),
            t(w, {
              variable: { ...e.variable, isDefault: e.variable.isDefault ?? !1 },
              warningMessage: e.warningMessage,
              onDelete: u
            }, null, 8, ["variable", "warningMessage"])
          ]),
          _: 1
        }, 8, ["class", "to"])
      ]);
    };
  }
});
export {
  V as default
};
