// @scalar/api-client/dist/components/Sidebar/Sidebar.vue.js
import { defineComponent as S, ref as f, withDirectives as x, openBlock as p, createElementBlock as v, normalizeClass as h, normalizeStyle as D, unref as r, renderSlot as a, createElementVNode as l, toDisplayString as k, createCommentVNode as L, vShow as _ } from "vue";
import { useWorkspace as B } from "../../store/workspace.js";
const W = {
  key: 0,
  class: "xl:min-h-header py-2.5 flex items-center border-b-1/2 px-4 text-sm"
}, C = { class: "font-medium m-0 text-sm whitespace-nowrap" }, I = /* @__PURE__ */ S({
  __name: "Sidebar",
  props: {
    title: {},
    showSideBar: { type: Boolean, default: !0 }
  },
  setup(g) {
    const b = g, { isReadOnly: i, sidebarWidth: s, setSidebarWidth: o } = B(), n = f(!1), d = f(null), w = (e) => {
      e.preventDefault();
      const m = e.clientX, y = parseInt(
        getComputedStyle(d.value).width || s.value,
        10
      ), c = (E) => {
        n.value = !0, document.body.classList.add("dragging");
        let t = y + E.clientX - m;
        t > 420 && (t = 420 + (t - 420) * 0.2), t < 240 && (t = 240), o(`${t}px`);
      }, u = () => {
        n.value = !1, document.body.classList.remove("dragging"), document.documentElement.removeEventListener("mousemove", c, !1), document.documentElement.removeEventListener("mouseup", u, !1), parseInt(s.value, 10) > 420 ? o("360px") : parseInt(s.value, 10) < 240 && o("240px");
      };
      document.documentElement.addEventListener("mousemove", c, !1), document.documentElement.addEventListener("mouseup", u, !1);
    };
    return (e, m) => x((p(), v("aside", {
      ref_key: "sidebarRef",
      ref: d,
      class: h(["sidebar overflow-hidden relative flex flex-col border-r-1/2 bg-b-1", { dragging: n.value }]),
      style: D({ width: r(s) })
    }, [
      a(e.$slots, "header"),
      !r(i) && e.title ? (p(), v("div", W, [
        l("h2", C, k(e.title), 1)
      ])) : L("", !0),
      l("div", {
        class: h(["custom-scroll sidebar-height", {
          "sidebar-mask": !r(i)
        }])
      }, [
        a(e.$slots, "content")
      ], 2),
      a(e.$slots, "button"),
      l("div", {
        class: "resizer",
        onMousedown: w
      }, null, 32)
    ], 6)), [
      [_, b.showSideBar]
    ]);
  }
});
export {
  I as default
};
