// @scalar/api-client/dist/components/Sidebar/SidebarButton.vue.js
import { defineComponent as n, openBlock as i, createElementBlock as l, createVNode as e, unref as a, withCtx as s, renderSlot as p } from "vue";
import u from "../ScalarHotkey.vue.js";
import { ScalarButton as d } from "@scalar/components";
const m = { class: "fixed relative bottom-[42px] z-10 flex px-2.5 pb-2 pt-0" }, k = /* @__PURE__ */ n({
  __name: "SidebarButton",
  props: {
    click: { type: Function }
  },
  setup(o) {
    const r = o, t = () => {
      r.click();
    };
    return (c, f) => (i(), l("div", m, [
      e(a(d), {
        class: "bg-b-1 text-c-1 hover:bg-b-2 group relative w-full border-1/2 p-1.5 h-auto",
        icon: "Plus",
        variant: "outlined",
        onClick: t
      }, {
        default: s(() => [
          p(c.$slots, "title"),
          e(u, {
            class: "absolute right-2 group-hover:opacity-80 text-c-2 add-item-hotkey",
            hotkey: "K",
            onHotkeyPressed: t
          })
        ]),
        _: 3
      })
    ]));
  }
});
export {
  k as default
};
