// @scalar/api-client/dist/components/Sidebar/SidebarToggle.vue.js
import { defineComponent as s, openBlock as l, createElementBlock as n, createElementVNode as e, normalizeClass as r } from "vue";
const a = {
  class: "size-3.5",
  fill: "none",
  viewBox: "0 0 24 24",
  xmlns: "http://www.w3.org/2000/svg"
}, d = /* @__PURE__ */ e("defs", null, [
  /* @__PURE__ */ e("clipPath", { id: "mask" }, [
    /* @__PURE__ */ e("path", {
      "clip-rule": "evenodd",
      d: "M9 3.2H4c-1.7 0-3 1.3-3 3v11.5c0 1.7 1.3 3 3 3h5V3.2z"
    })
  ])
], -1), i = { "clip-path": "url(#mask)" }, c = /* @__PURE__ */ e("path", {
  d: "M20 20.8H4c-1.7 0-3-1.3-3-3V6.2c0-1.7 1.3-3 3-3h16c1.7 0 3 1.3 3 3v11.5c0 1.7-1.3 3-3 3zM9 3.2v17.5",
  stroke: "currentColor",
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  "stroke-width": "2"
}, null, -1), h = /* @__PURE__ */ s({
  __name: "SidebarToggle",
  props: {
    modelValue: { type: Boolean }
  },
  emits: ["update:modelValue"],
  setup(u) {
    return (t, o) => (l(), n("button", {
      class: "text-c-3 hover:bg-b-2 active:text-c-1 p-2 rounded",
      type: "button",
      onClick: o[0] || (o[0] = (p) => t.$emit("update:modelValue", !t.modelValue))
    }, [
      (l(), n("svg", a, [
        d,
        e("g", i, [
          e("path", {
            class: r(["transition-transform duration-300", t.modelValue ? "translate-x-0" : "-translate-x-1/2"]),
            d: "M1 3.2h8v17.5H1z",
            fill: "currentColor"
          }, null, 2)
        ]),
        c
      ]))
    ]));
  }
});
export {
  h as default
};
