// @scalar/api-client/dist/components/Sidebar/SidebarListElementActions.vue.js
import { defineComponent as g, ref as C, createElementBlock as i, openBlock as o, Fragment as x, createElementVNode as h, createVNode as r, createCommentVNode as s, unref as n, withModifiers as w, withCtx as k, createBlock as M } from "vue";
import { useModal as D, ScalarIcon as m, ScalarModal as N } from "@scalar/components";
import { useClipboard as $ } from "@scalar/use-hooks/useClipboard";
import B from "./Actions/DeleteSidebarListElement.vue.js";
const E = { class: "absolute right-1 flex opacity-0 group-hover:opacity-100" }, R = /* @__PURE__ */ g({
  __name: "SidebarListElementActions",
  props: {
    variable: {},
    warningMessage: {},
    isCopyable: { type: Boolean },
    isDeletable: { type: Boolean },
    isRenameable: { type: Boolean }
  },
  emits: ["delete", "rename"],
  setup(e, { emit: v }) {
    const b = v, a = C({ action: "None", name: "" }), c = D(), { copyToClipboard: f } = $();
    function p(l) {
      a.value = { action: l, name: e.variable.name }, c.show();
    }
    function d() {
      c.hide(), a.value = { action: "None", name: "" };
    }
    function y(l) {
      b("delete", l), d();
    }
    return (l, t) => (o(), i(x, null, [
      h("div", E, [
        e.isCopyable ? (o(), i("button", {
          key: 0,
          class: "text-c-3 hover:bg-b-3 hover:text-c-1 rounded p-[5px]",
          type: "button",
          onClick: t[0] || (t[0] = (u) => n(f)(e.variable.name))
        }, [
          r(n(m), {
            class: "h-3 w-3",
            icon: "Clipboard"
          })
        ])) : s("", !0),
        e.isRenameable ? (o(), i("button", {
          key: 1,
          class: "text-c-3 hover:bg-b-3 hover:text-c-1 rounded p-[5px]",
          type: "button",
          onClick: t[1] || (t[1] = (u) => b("rename", e.variable.uid))
        }, [
          r(n(m), {
            class: "h-3 w-3",
            icon: "Edit"
          })
        ])) : s("", !0),
        !e.variable.isDefault && e.isDeletable ? (o(), i("button", {
          key: 2,
          class: "text-c-3 hover:bg-b-3 hover:text-c-1 rounded p-1",
          type: "button",
          onClick: t[2] || (t[2] = w((u) => p(
            "Delete"
            /* Delete */
          ), ["prevent"]))
        }, [
          r(n(m), {
            class: "h-3.5 w-3.5",
            icon: "Close"
          })
        ])) : s("", !0)
      ]),
      r(n(N), {
        size: "sm",
        state: n(c),
        title: `${a.value.action} ${a.value.name}`
      }, {
        default: k(() => [
          a.value.action === "Delete" ? (o(), M(B, {
            key: 0,
            variableName: a.value.name,
            warningMessage: e.warningMessage,
            onClose: d,
            onDelete: t[3] || (t[3] = (u) => y(e.variable.uid))
          }, null, 8, ["variableName", "warningMessage"])) : s("", !0)
        ]),
        _: 1
      }, 8, ["state", "title"])
    ], 64));
  }
});
export {
  R as default
};
