// @scalar/api-client/dist/components/Sidebar/SidebarListElementActions.vue.js
import { defineComponent as w, ref as M, openBlock as l, createElementBlock as c, Fragment as k, createElementVNode as m, unref as a, createVNode as i, withModifiers as y, createCommentVNode as u, withCtx as N, createBlock as $ } from "vue";
import x from "./Actions/DeleteSidebarListElement.vue.js";
import { useModal as D, ScalarIcon as d, ScalarModal as S } from "@scalar/components";
import { useClipboard as A } from "../../hooks/useClipboard.js";
const B = { class: "absolute flex right-1 opacity-0 group-hover:opacity-100" }, I = /* @__PURE__ */ w({
  __name: "SidebarListElementActions",
  props: {
    variable: {},
    warningMessage: {}
  },
  emits: ["delete"],
  setup(p, { emit: v }) {
    const { variable: b } = p, f = v, t = M({ action: "None", name: "" }), n = D(), { copyToClipboard: g } = A();
    function C(e) {
      t.value = { action: e, name: b.name }, n.show();
    }
    function r() {
      n.hide(), t.value = { action: "None", name: "" };
    }
    function h(e) {
      f("delete", e), r();
    }
    return (e, o) => (l(), c(k, null, [
      m("div", B, [
        m("button", {
          class: "text-c-3 hover:bg-b-3 hover:text-c-1 rounded p-[5px]",
          type: "button",
          onClick: o[0] || (o[0] = (s) => a(g)(e.variable.name))
        }, [
          i(a(d), {
            class: "h-3 w-3",
            icon: "Clipboard"
          })
        ]),
        e.variable.isDefault ? u("", !0) : (l(), c("button", {
          key: 0,
          class: "text-c-3 hover:bg-b-3 hover:text-c-1 rounded p-1",
          type: "button",
          onClick: o[1] || (o[1] = y((s) => C(
            "Delete"
            /* Delete */
          ), ["prevent"]))
        }, [
          i(a(d), {
            class: "h-3.5 w-3.5",
            icon: "Close"
          })
        ]))
      ]),
      i(a(S), {
        size: "sm",
        state: a(n),
        title: `${t.value.action} ${t.value.name}`
      }, {
        default: N(() => [
          t.value.action === "Delete" ? (l(), $(x, {
            key: 0,
            variableName: t.value.name,
            warningMessage: e.warningMessage,
            onClose: r,
            onDelete: o[2] || (o[2] = (s) => h(e.variable.uid))
          }, null, 8, ["variableName", "warningMessage"])) : u("", !0)
        ]),
        _: 1
      }, 8, ["state", "title"])
    ], 64));
  }
});
export {
  I as default
};
