// @scalar/api-client/dist/v2/features/app/components/AppSidebar.vue2.js
import { defineComponent as R, mergeModels as W, computed as k, capitalize as V, useModel as I, ref as F, createElementBlock as c, openBlock as u, createVNode as d, createBlock as M, createCommentVNode as p, unref as s, createSlots as G, withCtx as r, createTextVNode as b, createElementVNode as l, withKeys as w, withModifiers as v, toDisplayString as L, renderSlot as j, normalizeClass as H, Fragment as h, nextTick as U } from "vue";
import { useModal as J, ScalarSidebarItem as O, ScalarIconButton as Y, ScalarButton as $, ScalarModal as Q } from "@scalar/components";
import { isMacOS as X } from "@scalar/helpers/general/is-mac-os";
import { ScalarIconPlus as Z, ScalarIconDotsThree as _, ScalarIconGlobe as ee } from "@scalar/icons";
import { getParentEntry as S } from "@scalar/workspace-store/navigation";
import te from "../../../../assets/rabbit.ascii.js";
import ae from "../../../../assets/rabbitjump.ascii.js";
import E from "../../../../components/ScalarAsciiArt.vue.js";
import oe from "../../../../components/Sidebar/Actions/DeleteSidebarListElement.vue.js";
import ne from "./SidebarItemMenu.vue.js";
import { dragHandleFactory as se } from "../../../helpers/drag-handle-factory.js";
import ie from "../../../components/sidebar/Sidebar.vue.js";
const re = { class: "flex" }, le = {
  key: 0,
  class: "flex items-center"
}, de = {
  key: 0,
  class: "empty-sidebar-item-content overflow-hidden px-2.5 py-2.5"
}, ue = { class: "rabbit-ascii relative m-auto mt-2 h-[68px] w-[60px]" }, me = { class: "gap-1.5 p-2" }, ce = { class: "text-sidebar-c-2 rounded border px-1.25 py-1 text-xs leading-none font-medium uppercase" }, Ce = /* @__PURE__ */ R({
  __name: "AppSidebar",
  props: /* @__PURE__ */ W({
    layout: {},
    sidebarState: {},
    isWorkspaceOpen: { type: Boolean },
    activeWorkspace: {},
    workspaces: {},
    eventBus: {},
    store: {}
  }, {
    isSidebarOpen: { type: Boolean, required: !0 },
    isSidebarOpenModifiers: {},
    sidebarWidth: {
      required: !0,
      default: 288
    },
    sidebarWidthModifiers: {}
  }),
  emits: /* @__PURE__ */ W(["click:workspace", "selectItem", "select:workspace", "create:workspace"], ["update:isSidebarOpen", "update:sidebarWidth"]),
  setup(o, { emit: A }) {
    const f = A, D = k(() => V(o.activeWorkspace.label)), x = I(o, "isSidebarOpen"), B = I(o, "sidebarWidth"), g = k(() => o.sidebarState.items.value.length <= 1), C = k(
      () => se({
        store: o.store,
        sidebarState: o.sidebarState
      })
    ), N = (a, e) => C.value.handleDragEnd(a, e), T = (a, e) => C.value.isDroppable(a, e), i = F(null), y = J(), K = k(() => {
      const a = i.value?.item;
      return a?.type === "document" ? "This cannot be undone. You're about to delete the document and all tags and operations inside it." : `Are you sure you want to delete this ${a?.type ?? "item"}? This action cannot be undone.`;
    }), z = () => {
      const a = i.value?.item;
      if (!a)
        return;
      const e = o.sidebarState.getEntryById(a.id), t = S("document", e), n = S("operation", e);
      if (t) {
        if (a.type === "document")
          o.eventBus.emit("document:delete:document", { name: t.name });
        else if (a.type === "tag")
          o.eventBus.emit("tag:delete:tag", {
            documentName: t.name,
            name: a.name
          });
        else if (a.type === "operation") {
          if (!n)
            return;
          o.eventBus.emit("operation:delete:operation", {
            meta: {
              method: n.method,
              path: n.path
            },
            documentName: t.name
          });
        } else if (a.type === "example") {
          if (!n)
            return;
          o.eventBus.emit("operation:delete:example", {
            meta: {
              method: n.method,
              path: n.path,
              exampleKey: a.name
            },
            documentName: t.name
          });
        }
        y.hide(), i.value = null;
      }
    }, m = async (a, e) => {
      if (i.value?.showMenu)
        return;
      const t = a.currentTarget;
      i.value = { item: e, el: t, showMenu: !0 }, await U();
      const n = a instanceof MouseEvent ? new MouseEvent(a.type, a) : new KeyboardEvent(a.type, a);
      i.value?.el.dispatchEvent(n);
    }, P = () => {
      i.value && (i.value.showMenu = !1);
    }, q = (a) => {
      const e = o.sidebarState.getEntryById(a.id), t = S("document", e), n = S("tag", e);
      o.eventBus.emit("ui:open:command-palette", {
        action: "create-request",
        payload: {
          documentId: t?.id,
          tagId: n?.name
        }
      });
    };
    return (a, e) => (u(), c("div", re, [
      d(s(ie), {
        isSidebarOpen: x.value,
        "onUpdate:isSidebarOpen": e[3] || (e[3] = (t) => x.value = t),
        sidebarWidth: B.value,
        "onUpdate:sidebarWidth": e[4] || (e[4] = (t) => B.value = t),
        activeWorkspace: o.activeWorkspace,
        documents: Object.values(o.store.workspace.documents),
        isDroppable: T,
        layout: o.layout,
        sidebarState: o.sidebarState,
        workspaces: o.workspaces,
        "onCreate:workspace": e[5] || (e[5] = (t) => f("create:workspace")),
        onReorder: e[6] || (e[6] = (t, n) => N(t, n)),
        "onSelect:workspace": e[7] || (e[7] = (t) => f("select:workspace", t)),
        onSelectItem: e[8] || (e[8] = (t) => f("selectItem", t))
      }, G({
        sidebarMenuActions: r(() => [
          j(a.$slots, "sidebarMenuActions", {}, void 0, !0)
        ]),
        workspaceButton: r(() => [
          d(s(O), {
            is: "button",
            active: o.isWorkspaceOpen,
            icon: s(ee),
            onClick: e[0] || (e[0] = (t) => f("click:workspace"))
          }, {
            default: r(() => [
              b(L(D.value), 1)
            ]),
            _: 1
          }, 8, ["active", "icon"])
        ]),
        decorator: r(({ item: t }) => [
          d(s(Y), {
            "aria-expanded": "false",
            "aria-haspopup": "menu",
            icon: s(_),
            label: "More options",
            size: "sm",
            weight: "bold",
            onClick: v((n) => m(n, t), ["stop"]),
            onKeydown: [
              w(v((n) => m(n, t), ["stop"]), ["down"]),
              w(v((n) => m(n, t), ["stop"]), ["enter"]),
              w(v((n) => m(n, t), ["stop"]), ["space"]),
              w(v((n) => m(n, t), ["stop"]), ["up"])
            ]
          }, null, 8, ["icon", "onClick", "onKeydown"])
        ]),
        icon: r(({ item: t }) => [
          t.type === "document" && o.store.workspace.documents[t.name]?.["x-scalar-is-dirty"] === !0 ? (u(), c("div", le, [...e[11] || (e[11] = [
            l("div", { class: "h-2 w-2 rounded-full bg-white" }, null, -1)
          ])])) : p("", !0)
        ]),
        empty: r(({ item: t }) => [
          d(s(O), {
            is: "button",
            onClick: (n) => q(t)
          }, {
            icon: r(() => [
              d(s(Z))
            ]),
            default: r(() => [...e[12] || (e[12] = [
              b("Add operation", -1)
            ])]),
            _: 1
          }, 8, ["onClick"])
        ]),
        _: 2
      }, [
        o.layout !== "modal" ? {
          name: "footer",
          fn: r(() => [
            l("div", {
              class: H({
                "empty-sidebar-item border-t": g.value
              })
            }, [
              g.value ? (u(), c("div", de, [
                l("div", ue, [
                  d(E, {
                    art: s(te),
                    class: "rabbitsit font-bold"
                  }, null, 8, ["art"]),
                  d(E, {
                    art: s(ae),
                    class: "rabbitjump absolute top-0 left-0 font-bold"
                  }, null, 8, ["art"])
                ]),
                e[13] || (e[13] = l("div", { class: "mt-2 mb-2 text-center text-sm text-balance" }, [
                  l("b", { class: "font-medium" }, "Let's Get Started"),
                  l("p", { class: "mt-2 leading-3" }, " Create request, folder, collection or import from OpenAPI/Postman ")
                ], -1))
              ])) : p("", !0),
              l("div", me, [
                g.value ? (u(), M(s($), {
                  key: 0,
                  class: "w-full",
                  size: "sm",
                  onClick: e[1] || (e[1] = (t) => o.eventBus.emit("ui:open:command-palette", {
                    action: "import-from-openapi-swagger-postman-curl",
                    payload: void 0
                  }))
                }, {
                  default: r(() => [...e[14] || (e[14] = [
                    b(" Import Collection ", -1)
                  ])]),
                  _: 1
                })) : p("", !0),
                d(s($), {
                  class: "w-full",
                  hotkey: "K",
                  size: "sm",
                  variant: "outlined",
                  onClick: e[2] || (e[2] = (t) => o.eventBus.emit("ui:open:command-palette"))
                }, {
                  default: r(() => [
                    e[20] || (e[20] = b(" Add Item   ", -1)),
                    l("span", ce, [
                      s(X)() ? (u(), c(h, { key: 0 }, [
                        e[15] || (e[15] = l("span", { class: "sr-only" }, "Command", -1)),
                        e[16] || (e[16] = l("span", { "aria-hidden": "true" }, "⌘", -1))
                      ], 64)) : (u(), c(h, { key: 1 }, [
                        e[17] || (e[17] = l("span", { class: "sr-only" }, "CTRL", -1)),
                        e[18] || (e[18] = l("span", { "aria-hidden": "true" }, "⌃", -1))
                      ], 64)),
                      e[19] || (e[19] = b(" K ", -1))
                    ])
                  ]),
                  _: 1
                })
              ])
            ], 2)
          ]),
          key: "0"
        } : void 0
      ]), 1032, ["isSidebarOpen", "sidebarWidth", "activeWorkspace", "documents", "layout", "sidebarState", "workspaces"]),
      i.value?.showMenu ? (u(), M(ne, {
        key: 0,
        eventBus: o.eventBus,
        item: i.value.item,
        sidebarState: o.sidebarState,
        target: i.value.el,
        onCloseMenu: P,
        onShowDeleteModal: e[9] || (e[9] = (t) => s(y).show())
      }, null, 8, ["eventBus", "item", "sidebarState", "target"])) : p("", !0),
      i.value ? (u(), M(s(Q), {
        key: 1,
        size: "xxs",
        state: s(y),
        title: `Delete ${i.value.item.title}`
      }, {
        default: r(() => [
          d(oe, {
            variableName: i.value.item.title,
            warningMessage: K.value,
            onClose: e[10] || (e[10] = (t) => s(y).hide()),
            onDelete: z
          }, null, 8, ["variableName", "warningMessage"])
        ]),
        _: 1
      }, 8, ["state", "title"])) : p("", !0)
    ]));
  }
});
export {
  Ce as default
};
