// @scalar/api-client/dist/v2/features/app/components/DesktopTab.vue2.js
import { defineComponent as x, computed as k, createBlock as i, openBlock as a, unref as t, withCtx as n, createVNode as o, createElementBlock as v, createCommentVNode as b, createTextVNode as m, resolveDynamicComponent as s, Fragment as I, createElementVNode as f, normalizeClass as g, toDisplayString as D } from "vue";
import { ScalarContextMenu as $, ScalarFloating as B, ScalarDropdownMenu as N, ScalarDropdownButton as u, ScalarHotkey as y, ScalarDropdownDivider as O, ScalarTooltip as U } from "@scalar/components";
import { isMacOS as V } from "@scalar/helpers/general/is-mac-os";
import { ScalarIconPlusSquare as q, ScalarIconLinkSimple as F, ScalarIconXSquare as M, ScalarIconTabs as E, ScalarIconFolderSimple as L, ScalarIconArrowUpRight as P, ScalarIconX as R } from "@scalar/icons";
const H = /* @__PURE__ */ x({
  __name: "DesktopTab",
  props: {
    hotkey: {},
    active: { type: Boolean },
    isSingleTab: { type: Boolean, default: !1 },
    tab: {}
  },
  emits: ["click", "close", "newTab", "copyUrl", "closeOtherTabs"],
  setup(l, { emit: T }) {
    const c = T, S = {
      request: P,
      document: L
    }, C = k(() => V() ? "⌘" : "^"), p = k(
      () => l.hotkey ? `${C.value} ${l.hotkey}` : ""
    ), w = (d) => {
      d.stopPropagation(), c("close");
    };
    return (d, e) => (a(), i(t($), { triggerClass: `overflow-hidden w-full flex-1 select-none
` }, {
      trigger: n(() => [
        (a(), i(s(l.isSingleTab ? "div" : t(U)), {
          content: p.value,
          placement: "bottom"
        }, {
          default: n(() => [
            f("div", {
              class: g(["app-no-drag-region", [
                l.isSingleTab ? "nav-single-tab" : "nav-item",
                { "nav-item__active": l.active && !l.isSingleTab }
              ]]),
              onClick: e[0] || (e[0] = (r) => !l.isSingleTab && c("click"))
            }, [
              f("div", {
                class: g([
                  "flex items-center justify-center gap-1.5",
                  l.isSingleTab ? "custom-scroll h-full w-full whitespace-nowrap" : "nav-item-icon-copy flex-1"
                ])
              }, [
                l.tab.icon && S[l.tab.icon] ? (a(), i(s(S[l.tab.icon]), { key: 0 })) : b("", !0),
                f("span", {
                  class: g([
                    l.isSingleTab ? "" : "custom-scroll nav-item-copy",
                    "text-sm"
                  ])
                }, D(l.tab.title), 3)
              ], 2),
              l.isSingleTab ? b("", !0) : (a(), v("button", {
                key: 0,
                class: "nav-item-close",
                type: "button",
                onClick: w
              }, [
                (a(), i(s(t(R))))
              ]))
            ], 2)
          ]),
          _: 1
        }, 8, ["content"]))
      ]),
      content: n(() => [
        o(t(B), { placement: "right-start" }, {
          floating: n(() => [
            o(t(N), { class: "scalar-app scalar-client" }, {
              default: n(() => [
                o(t(u), {
                  class: "flex items-center gap-1.5",
                  onClick: e[1] || (e[1] = (r) => c("newTab"))
                }, {
                  default: n(() => [
                    (a(), i(s(t(q)))),
                    e[5] || (e[5] = m(" New Tab ", -1)),
                    o(t(y), {
                      class: "bg-b-2 ml-auto",
                      hotkey: "T"
                    })
                  ]),
                  _: 1
                }),
                o(t(u), {
                  class: "flex items-center gap-1.5",
                  onClick: e[2] || (e[2] = (r) => c("copyUrl"))
                }, {
                  default: n(() => [
                    (a(), i(s(t(F)))),
                    e[6] || (e[6] = m(" Copy URL ", -1))
                  ]),
                  _: 1
                }),
                l.isSingleTab ? b("", !0) : (a(), v(I, { key: 0 }, [
                  o(t(O)),
                  o(t(u), {
                    class: "flex items-center gap-1.5",
                    onClick: e[3] || (e[3] = (r) => c("close"))
                  }, {
                    default: n(() => [
                      (a(), i(s(t(M)))),
                      e[7] || (e[7] = m(" Close Tab ", -1)),
                      o(t(y), {
                        class: "bg-b-2 ml-auto",
                        hotkey: "W"
                      })
                    ]),
                    _: 1
                  }),
                  o(t(u), {
                    class: "flex items-center gap-1.5",
                    onClick: e[4] || (e[4] = (r) => c("closeOtherTabs"))
                  }, {
                    default: n(() => [
                      (a(), i(s(t(E)))),
                      e[8] || (e[8] = m(" Close Other Tabs ", -1))
                    ]),
                    _: 1
                  })
                ], 64))
              ]),
              _: 1
            })
          ]),
          _: 1
        })
      ]),
      _: 1
    }));
  }
});
export {
  H as default
};
