// @scalar/api-client/dist/components/CommandPalette/CommandPaletteImport.vue.js
import { defineComponent as y, ref as u, onMounted as S, openBlock as k, createElementBlock as C, createElementVNode as t, withKeys as F, withModifiers as I, createVNode as a, unref as o, withCtx as m, createTextVNode as d } from "vue";
import { useWorkspace as N } from "../../store/workspace.js";
import { ScalarButton as f, ScalarIcon as R } from "@scalar/components";
import { useFileDialog as U } from "../../hooks/useFileDialog.js";
const A = { class: "flex w-full flex-col gap-3" }, B = { class: "gap-3 rounded bg-b-2 focus-within:bg-b-1 focus-within:shadow-border min-h-20 relative" }, K = /* @__PURE__ */ t("label", {
  class: "absolute w-full h-full opacity-0 cursor-text",
  for: "requestimport"
}, null, -1), L = ["onKeydown"], M = { class: "flex gap-2" }, P = { class: "flex flex-1 gap-2 max-h-8" }, W = /* @__PURE__ */ y({
  __name: "CommandPaletteImport",
  emits: ["close"],
  setup(V, { emit: h }) {
    const n = h, { activeWorkspace: _, importSpecFile: x, importSpecFromUrl: g } = N(), r = u(""), { open: w } = U({
      onChange: async (e) => {
        const s = e == null ? void 0 : e[0];
        if (s) {
          const i = new FileReader();
          i.onload = async (v) => {
            var p;
            const b = (p = v.target) == null ? void 0 : p.result;
            x(b, _.value.uid), l(), n("close");
          }, i.readAsText(s);
        }
      },
      multiple: !1,
      accept: ".json,.yaml,.yml"
    }), l = async () => {
      r.value && (await g(r.value), n("close"));
    }, c = u(null);
    return S(() => {
      var e;
      (e = c.value) == null || e.focus();
    }), (e, s) => (k(), C("div", A, [
      t("div", B, [
        K,
        t("input", {
          id: "requestimport",
          ref_key: "importInput",
          ref: c,
          class: "border-transparent outline-none w-full pl-8 text-sm min-h-8 py-1.5",
          label: "Paste Swagger File URL",
          placeholder: "Paste Swagger File URL",
          onKeydown: F(I(l, ["prevent"]), ["enter"])
        }, null, 40, L)
      ]),
      t("div", M, [
        t("div", P, [
          a(o(f), {
            class: "p-2 max-h-8 gap-1 text-xs hover:bg-b-2 relative",
            variant: "outlined",
            onClick: o(w)
          }, {
            default: m(() => [
              d(" JSON, or YAML Files "),
              a(o(R), {
                class: "text-c-3 -rotate-90",
                icon: "ArrowRight",
                size: "sm"
              })
            ]),
            _: 1
          }, 8, ["onClick"])
        ]),
        a(o(f), {
          class: "max-h-8 text-xs p-0 px-3",
          onClick: l
        }, {
          default: m(() => [
            d(" Import Collection ")
          ]),
          _: 1
        })
      ])
    ]));
  }
});
export {
  W as default
};
