// @scalar/api-client/dist/components/CommandPalette/CommandPaletteExample.vue.js
import { defineComponent as $, ref as i, onMounted as f, openBlock as m, createElementBlock as h, createElementVNode as l, withDirectives as B, withKeys as I, withModifiers as R, vModelText as V, createVNode as a, unref as s, withCtx as n, Fragment as q, renderList as K, createBlock as W, createTextVNode as d, toDisplayString as v } from "vue";
import _ from "../HttpMethod/HttpMethod.vue.js";
import { useWorkspace as j } from "../../store/workspace.js";
import { ScalarDropdown as z, ScalarDropdownItem as T, ScalarButton as w, ScalarIcon as F } from "@scalar/components";
import { useRouter as H } from "vue-router";
const L = { class: "flex w-full flex-col gap-3" }, P = { class: "gap-3 rounded bg-b-2 focus-within:bg-b-1 focus-within:shadow-border min-h-20 relative" }, U = /* @__PURE__ */ l("label", {
  class: "absolute w-full h-full opacity-0 cursor-text",
  for: "examplename"
}, null, -1), A = ["onKeydown"], G = { class: "flex gap-2" }, J = { class: "flex flex-1 max-h-8" }, O = { class: "flex items-center gap-2" }, Q = { class: "max-h-40 custom-scroll" }, oe = /* @__PURE__ */ $({
  __name: "CommandPaletteExample",
  props: {
    metaData: {}
  },
  emits: ["close"],
  setup(b, { emit: k }) {
    const y = b, g = k, { push: C } = H(), {
      activeRequest: D,
      activeWorkspace: E,
      activeWorkspaceRequests: S,
      requests: N,
      requestExampleMutators: M
    } = j(), c = i(""), o = i(
      // Ensure we pre-select the correct request
      N[y.metaData ?? ""] ?? D.value
    ), p = (e) => o.value = e, u = i(null);
    f(() => {
      var e;
      return (e = u.value) == null ? void 0 : e.focus();
    });
    const x = () => {
      const e = M.add(
        o.value,
        c.value
      );
      e && (C(
        `/workspace/${E.value.uid}/request/${o.value.uid}/examples/${e.uid}`
      ), g("close"));
    };
    return f(() => {
      var e;
      (e = u.value) == null || e.focus();
    }), (e, r) => (m(), h("div", L, [
      l("div", P, [
        U,
        B(l("input", {
          id: "examplename",
          ref_key: "exampleInput",
          ref: u,
          "onUpdate:modelValue": r[0] || (r[0] = (t) => c.value = t),
          class: "border-transparent outline-none w-full pl-8 text-sm min-h-8 py-1.5",
          label: "Example Name",
          placeholder: "Example Name",
          onKeydown: I(R(x, ["prevent"]), ["enter"])
        }, null, 40, A), [
          [V, c.value]
        ])
      ]),
      l("div", G, [
        l("div", J, [
          a(s(z), {
            placement: "bottom",
            resize: ""
          }, {
            items: n(() => [
              l("div", Q, [
                (m(!0), h(q, null, K(s(S), (t) => (m(), W(s(T), {
                  key: t.uid,
                  class: "flex h-7 w-full items-center justify-between px-1 pr-[26px]",
                  onClick: (X) => p(t)
                }, {
                  default: n(() => [
                    d(v(t.summary) + " ", 1),
                    a(_, {
                      method: t.method
                    }, null, 8, ["method"])
                  ]),
                  _: 2
                }, 1032, ["onClick"]))), 128))
              ])
            ]),
            default: n(() => [
              a(s(w), {
                class: "justify-between p-2 max-h-8 w-full gap-1 text-xs hover:bg-b-2",
                variant: "outlined",
                onClick: r[1] || (r[1] = (t) => p(o.value))
              }, {
                default: n(() => [
                  d(v(o.value.summary) + " ", 1),
                  l("div", O, [
                    a(_, {
                      method: o.value.method
                    }, null, 8, ["method"]),
                    a(s(F), {
                      class: "text-c-3",
                      icon: "ChevronDown",
                      size: "xs"
                    })
                  ])
                ]),
                _: 1
              })
            ]),
            _: 1
          })
        ]),
        a(s(w), {
          class: "max-h-8 text-xs p-0 px-3",
          onClick: x
        }, {
          default: n(() => [
            d(" Create Example ")
          ]),
          _: 1
        })
      ])
    ]));
  }
});
export {
  oe as default
};
