// @scalar/api-client/dist/components/CommandPalette/CommandPaletteCollection.vue.js
import { defineComponent as p, ref as a, onMounted as m, openBlock as f, createElementBlock as h, createElementVNode as e, withDirectives as _, withKeys as x, withModifiers as v, vModelText as w, createVNode as b, unref as C, withCtx as y, createTextVNode as g } from "vue";
import { useWorkspace as k } from "../../store/workspace.js";
import { ScalarButton as N } from "@scalar/components";
const M = { class: "flex w-full flex-col gap-3" }, V = { class: "gap-3 rounded bg-b-2 focus-within:bg-b-1 focus-within:shadow-border min-h-20 relative" }, B = /* @__PURE__ */ e("label", {
  class: "absolute w-full h-full opacity-0 cursor-text",
  for: "collectionanme"
}, null, -1), K = ["onKeydown"], E = { class: "flex" }, I = /* @__PURE__ */ e("div", { class: "flex flex-1 gap-2 max-h-8" }, null, -1), P = /* @__PURE__ */ p({
  __name: "CommandPaletteCollection",
  emits: ["close"],
  setup(S, { emit: c }) {
    const i = c, { activeWorkspace: r, collectionMutators: u } = k(), o = a(""), l = () => {
      u.add(
        {
          spec: {
            openapi: "3.1.0",
            info: {
              title: o.value,
              version: "0.0.1"
            }
          }
        },
        r.value.uid
      ), i("close");
    }, n = a(null);
    return m(() => {
      var t;
      (t = n.value) == null || t.focus();
    }), (t, s) => (f(), h("div", M, [
      e("div", V, [
        B,
        _(e("input", {
          id: "collectionanme",
          ref_key: "collectionInput",
          ref: n,
          "onUpdate:modelValue": s[0] || (s[0] = (d) => o.value = d),
          autocomplete: "off",
          class: "border-transparent outline-none w-full pl-8 text-sm min-h-8 py-1.5",
          "data-form-type": "other",
          "data-lpignore": "true",
          label: "Collection Name",
          placeholder: "Collection Name",
          onKeydown: x(v(l, ["prevent"]), ["enter"])
        }, null, 40, K), [
          [w, o.value]
        ])
      ]),
      e("div", E, [
        I,
        b(C(N), {
          class: "max-h-8 text-xs p-0 px-3",
          onClick: l
        }, {
          default: y(() => [
            g(" Continue ")
          ]),
          _: 1
        })
      ])
    ]));
  }
});
export {
  P as default
};
