// @scalar/api-client/dist/components/CommandPalette/CommandPaletteWorkspace.vue.js
import { defineComponent as d, ref as l, onMounted as m, openBlock as f, createElementBlock as h, createElementVNode as o, withDirectives as w, withKeys as _, withModifiers as x, vModelText as k, createVNode as v, unref as b, withCtx as y, createTextVNode as g } from "vue";
import { useWorkspace as N } from "../../store/workspace.js";
import { ScalarButton as C } from "@scalar/components";
import { useRouter as M } from "vue-router";
const V = { class: "flex w-full flex-col gap-3" }, W = { class: "gap-3 rounded bg-b-2 focus-within:bg-b-1 focus-within:shadow-border min-h-20 relative" }, B = /* @__PURE__ */ o("label", {
  class: "absolute w-full h-full opacity-0 cursor-text",
  for: "workspacename"
}, null, -1), K = ["onKeydown"], E = { class: "flex" }, I = /* @__PURE__ */ o("div", { class: "flex flex-1 gap-2 max-h-8" }, null, -1), U = /* @__PURE__ */ d({
  __name: "CommandPaletteWorkspace",
  emits: ["close"],
  setup(S, { emit: r }) {
    const c = r, { push: i } = M(), { workspaceMutators: u } = N(), t = l(""), s = () => {
      const e = u.add({
        name: t.value
      });
      i(`/workspace/${e.uid}`), c("close");
    }, a = l(null);
    return m(() => {
      var e;
      (e = a.value) == null || e.focus();
    }), (e, n) => (f(), h("div", V, [
      o("div", W, [
        B,
        w(o("input", {
          id: "workspacename",
          ref_key: "workspaceInput",
          ref: a,
          "onUpdate:modelValue": n[0] || (n[0] = (p) => t.value = p),
          autocomplete: "off",
          autofocus: "",
          class: "border-transparent outline-none w-full pl-8 text-sm min-h-8 py-1.5",
          "data-form-type": "other",
          "data-lpignore": "true",
          label: "Workspace Name",
          placeholder: "Workspace Name",
          onKeydown: _(x(s, ["prevent"]), ["enter"])
        }, null, 40, K), [
          [k, t.value]
        ])
      ]),
      o("div", E, [
        I,
        v(b(C), {
          class: "max-h-8 text-xs p-0 px-3",
          onClick: s
        }, {
          default: y(() => [
            g(" Continue ")
          ]),
          _: 1
        })
      ])
    ]));
  }
});
export {
  U as default
};
