// @scalar/api-client/dist/components/CommandPalette/CommandPaletteFolder.vue.js
import { defineComponent as g, ref as r, computed as v, onMounted as y, openBlock as V, createElementBlock as k, createElementVNode as a, withDirectives as N, withKeys as S, withModifiers as F, vModelText as I, createVNode as n, unref as i, withCtx as c, normalizeClass as M, toDisplayString as B, createTextVNode as D } from "vue";
import { useWorkspace as K } from "../../store/workspace.js";
import { ScalarListbox as z, ScalarButton as x, ScalarIcon as E } from "@scalar/components";
const T = { class: "flex w-full flex-col gap-3" }, U = { class: "gap-3 rounded bg-b-2 focus-within:bg-b-1 focus-within:shadow-border min-h-20 relative" }, W = /* @__PURE__ */ a("label", {
  class: "absolute w-full h-full opacity-0 cursor-text",
  for: "foldername"
}, null, -1), j = ["onKeydown"], L = { class: "flex" }, P = { class: "flex flex-1 gap-2 max-h-8" }, J = /* @__PURE__ */ g({
  __name: "CommandPaletteFolder",
  emits: ["close"],
  setup(q, { emit: h }) {
    var p;
    const _ = h, { activeWorkspaceCollections: b, folderMutators: w, activeCollection: C } = K(), s = r(""), d = r(((p = C.value) == null ? void 0 : p.uid) ?? ""), u = v(
      () => b.value.map((e) => {
        var l, o;
        return {
          id: e.uid,
          label: ((o = (l = e.spec) == null ? void 0 : l.info) == null ? void 0 : o.title) ?? ""
        };
      })
    ), t = v({
      get: () => u.value.find(
        ({ id: e }) => e === d.value
      ),
      set: (e) => {
        e != null && e.id && (d.value = e.id);
      }
    }), f = () => {
      s.value && t.value && (w.add(
        {
          name: s.value
        },
        d.value
      ), _("close"));
    }, m = r(null);
    return y(() => {
      var e;
      (e = m.value) == null || e.focus();
    }), (e, l) => (V(), k("div", T, [
      a("div", U, [
        W,
        N(a("input", {
          id: "foldername",
          ref_key: "folderInput",
          ref: m,
          "onUpdate:modelValue": l[0] || (l[0] = (o) => s.value = o),
          autocomplete: "off",
          class: "border-transparent outline-none w-full pl-8 text-sm min-h-8 py-1.5",
          "data-form-type": "other",
          "data-lpignore": "true",
          label: "Folder Name",
          placeholder: "Folder Name",
          onKeydown: S(F(f, ["prevent"]), ["enter"])
        }, null, 40, j), [
          [I, s.value]
        ])
      ]),
      a("div", L, [
        a("div", P, [
          n(i(z), {
            modelValue: t.value,
            "onUpdate:modelValue": l[1] || (l[1] = (o) => t.value = o),
            options: u.value
          }, {
            default: c(() => [
              n(i(x), {
                class: "justify-between p-2 max-h-8 w-full gap-1 text-xs hover:bg-b-2",
                variant: "outlined"
              }, {
                default: c(() => [
                  a("span", {
                    class: M(t.value ? "text-c-1" : "text-c-3")
                  }, B(t.value ? t.value.label : "Select Collection"), 3),
                  n(i(E), {
                    class: "text-c-3",
                    icon: "ChevronDown",
                    size: "xs"
                  })
                ]),
                _: 1
              })
            ]),
            _: 1
          }, 8, ["modelValue", "options"])
        ]),
        n(i(x), {
          class: "max-h-8 text-xs p-0 px-3",
          onClick: f
        }, {
          default: c(() => [
            D(" Create Folder ")
          ]),
          _: 1
        })
      ])
    ]));
  }
});
export {
  J as default
};
