// @scalar/api-client/dist/components/CommandPalette/CommandPaletteImportCurl.vue2.js
import { defineComponent as I, ref as x, computed as B, useTemplateRef as O, createElementBlock as V, openBlock as W, Fragment as L, createElementVNode as r, createVNode as o, withCtx as i, toDisplayString as m, unref as c, normalizeClass as M } from "vue";
import { ScalarListbox as P, ScalarButton as j, ScalarIcon as z } from "@scalar/components";
import { REGEX as F } from "@scalar/oas-utils/helpers";
import { emitCustomEvent as N } from "@scalar/workspace-store/events";
import { useRouter as T } from "vue-router";
import $ from "./CommandActionForm.vue.js";
import A from "../HttpMethod/HttpMethod.vue.js";
import { PathId as f } from "../../routes.js";
import { useActiveEntities as G } from "../../store/active-entities.js";
import { useWorkspace as H } from "../../store/store.js";
const X = { class: "flex h-9 flex-row items-center gap-2 rounded border p-[3px] text-sm" }, J = { class: "flex h-full" }, K = { class: "scroll-timeline-x whitespace-nowrap" }, Q = { class: "flex" }, ie = /* @__PURE__ */ I({
  __name: "CommandPaletteImportCurl",
  props: {
    metaData: {}
  },
  emits: ["close", "back"],
  setup(e, { emit: D }) {
    const w = D, b = H(), { activeWorkspaceCollections: p, activeCollection: S, activeWorkspace: v } = G(), { requestMutators: k, serverMutators: q, servers: R } = b, d = x(""), U = T(), n = B(
      () => p.value.map((a) => ({
        id: a.uid,
        label: a.info?.title ?? "Unititled Collection"
      }))
    ), l = x(
      e.metaData.collectionUid ? n.value.find(
        (a) => a.id === e.metaData.collectionUid
      ) : n.value.find(
        (a) => a.id === S.value?.uid
      )
    );
    function g({ collectionUid: a }) {
      if (!e.metaData.parsedCurl)
        return;
      const t = p.value.find(
        (u) => u.uid === a
      );
      if (!t)
        return;
      const s = t?.info?.title === "Drafts";
      if (!s && e.metaData.parsedCurl.servers) {
        const u = Object.values(R).find(
          (y) => y.url === e.metaData.parsedCurl?.servers?.[0]
        );
        u ? d.value = u.uid : (d.value = q.add(
          { url: e.metaData.parsedCurl.servers[0] ?? "/" },
          t.uid
        ).uid, N(E.value?.$el, "scalar-add-server", {
          server: { url: e.metaData.parsedCurl.servers[0] ?? "/" },
          options: {
            disableOldStoreUpdate: !0
          }
        }));
      }
      const h = k.add(
        {
          summary: s ? e.metaData.parsedCurl?.url?.replace(F.PROTOCOL, "") : e.metaData.parsedCurl?.path,
          path: s ? e.metaData.parsedCurl?.url : e.metaData.parsedCurl?.path,
          method: e.metaData.parsedCurl?.method,
          parameters: e.metaData.parsedCurl?.parameters,
          selectedServerUid: s ? void 0 : d.value,
          requestBody: e.metaData.parsedCurl?.requestBody
        },
        t.uid
      );
      h && v.value?.uid && U.push({
        name: "request",
        params: {
          [f.Workspace]: v.value.uid,
          [f.Collection]: t.uid,
          [f.Request]: h.uid
        }
      }), w("close");
    }
    const C = () => {
      g({
        collectionUid: l.value?.id ?? ""
      });
    }, E = O("wrapper-ref");
    return (a, t) => (W(), V(L, null, [
      t[1] || (t[1] = r("div", { class: "text-c-2 flex-center py-1.5 text-sm" }, "Import cURL", -1)),
      o($, {
        ref: "wrapper-ref",
        class: "mt-1.5 min-h-fit",
        onSubmit: C
      }, {
        options: i(() => [
          r("div", Q, [
            o(c(P), {
              modelValue: l.value,
              "onUpdate:modelValue": t[0] || (t[0] = (s) => l.value = s),
              options: n.value
            }, {
              default: i(() => [
                o(c(j), {
                  class: "hover:bg-b-2 max-h-8 w-full justify-between gap-1 p-2 text-xs",
                  variant: "outlined"
                }, {
                  default: i(() => [
                    r("span", {
                      class: M(["whitespace-nowrap", l.value ? "text-c-1" : "text-c-3"])
                    }, m(l.value ? l.value.label : "Select Collection"), 3),
                    o(c(z), {
                      class: "text-c-3",
                      icon: "ChevronDown",
                      size: "md"
                    })
                  ]),
                  _: 1
                })
              ]),
              _: 1
            }, 8, ["modelValue", "options"])
          ])
        ]),
        submit: i(() => [
          r("span", { onClick: C }, "Import Request")
        ]),
        default: i(() => [
          r("div", X, [
            r("div", J, [
              o(A, {
                isEditable: !1,
                isSquare: "",
                method: e.metaData.parsedCurl?.method || "get"
              }, null, 8, ["method"])
            ]),
            r("span", K, m(e.metaData.parsedCurl?.servers?.[0] || "") + m(e.metaData.parsedCurl?.path || ""), 1)
          ])
        ]),
        _: 1
      }, 512)
    ], 64));
  }
});
export {
  ie as default
};
