// @scalar/api-client/dist/components/CommandPalette/CommandPaletteRequest.vue.js
import { defineComponent as K, ref as d, computed as v, onMounted as T, openBlock as k, createElementBlock as W, createElementVNode as u, withDirectives as $, withKeys as G, withModifiers as H, vModelText as L, createVNode as o, unref as s, withCtx as r, normalizeClass as M, toDisplayString as S, createBlock as O, createCommentVNode as P, createTextVNode as A } from "vue";
import J from "../HttpMethod/HttpMethod.vue.js";
import { useWorkspace as Q } from "../../store/workspace.js";
import { ScalarListbox as U, ScalarButton as p, ScalarIcon as N } from "@scalar/components";
import { useRouter as X } from "vue-router";
const Y = { class: "flex w-full flex-col gap-3" }, Z = { class: "gap-3 rounded bg-b-2 focus-within:bg-b-1 focus-within:shadow-border min-h-20 relative" }, ee = /* @__PURE__ */ u("label", {
  class: "absolute w-full h-full opacity-0 cursor-text",
  for: "requestname"
}, null, -1), te = ["onKeydown"], le = { class: "flex" }, ae = { class: "flex flex-1 gap-2 max-h-8" }, re = /* @__PURE__ */ K({
  __name: "CommandPaletteRequest",
  emits: ["close"],
  setup(oe, { emit: R }) {
    var g, q;
    const I = R, { push: B } = X(), {
      activeCollection: D,
      activeWorkspace: E,
      activeWorkspaceCollections: h,
      requestMutators: j,
      activeRequest: z,
      folders: x
    } = Q(), i = d(""), f = d("GET"), c = d(((g = D.value) == null ? void 0 : g.uid) ?? ""), b = v(
      () => h.value.map((e) => {
        var t;
        return {
          id: e.uid,
          label: ((t = e.spec.info) == null ? void 0 : t.title) ?? "Unititled Collection"
        };
      })
    ), n = v({
      get: () => b.value.find(({ id: e }) => e === c.value),
      set: (e) => {
        e != null && e.id && (c.value = e.id);
      }
    }), m = v(
      () => h.value.flatMap(
        (e) => e.uid === c.value ? e.childUids.flatMap((t) => {
          const l = x[t];
          return l ? [
            {
              id: l.uid,
              label: l.name
            }
          ] : [];
        }) : []
      )
    ), _ = d(
      ((q = Object.values(x).find(
        (e) => {
          var t;
          return e.childUids.includes((t = z.value) == null ? void 0 : t.uid);
        }
      )) == null ? void 0 : q.uid) ?? ""
    ), a = v({
      get: () => m.value.find(({ id: e }) => e === _.value),
      set: (e) => {
        e != null && e.id && (_.value = e.id);
      }
    });
    function F(e) {
      f.value = e;
    }
    const w = () => {
      var l, y, V;
      if (!c.value && !((l = a.value) != null && l.id)) return;
      const e = ((y = a.value) == null ? void 0 : y.id) ?? ((V = n.value) == null ? void 0 : V.id), t = j.add(
        {
          path: "",
          method: f.value.toUpperCase(),
          description: i.value,
          operationId: i.value,
          summary: i.value,
          tags: ["default"]
        },
        e
      );
      B(`/workspace/${E.value.uid}/request/${t.uid}`), I("close");
    }, C = d(null);
    return T(() => {
      var e;
      (e = C.value) == null || e.focus();
    }), (e, t) => (k(), W("div", Y, [
      u("div", Z, [
        ee,
        $(u("input", {
          id: "requestname",
          ref_key: "requestInput",
          ref: C,
          "onUpdate:modelValue": t[0] || (t[0] = (l) => i.value = l),
          autocomplete: "off",
          class: "border-transparent outline-none w-full pl-8 text-sm min-h-8 py-1.5",
          "data-form-type": "other",
          "data-lpignore": "true",
          label: "Request Name",
          placeholder: "Request Name",
          onKeydown: G(H(w, ["prevent"]), ["enter"])
        }, null, 40, te), [
          [L, i.value]
        ])
      ]),
      u("div", le, [
        u("div", ae, [
          o(J, {
            isEditable: !0,
            isSquare: "",
            method: f.value,
            onChange: F
          }, null, 8, ["method"]),
          o(s(U), {
            modelValue: n.value,
            "onUpdate:modelValue": t[1] || (t[1] = (l) => n.value = l),
            options: b.value
          }, {
            default: r(() => [
              o(s(p), {
                class: "justify-between p-2 max-h-8 w-full gap-1 text-xs hover:bg-b-2",
                variant: "outlined"
              }, {
                default: r(() => [
                  u("span", {
                    class: M(n.value ? "text-c-1" : "text-c-3")
                  }, S(n.value ? n.value.label : "Select Collection"), 3),
                  o(s(N), {
                    class: "text-c-3",
                    icon: "ChevronDown",
                    size: "xs"
                  })
                ]),
                _: 1
              })
            ]),
            _: 1
          }, 8, ["modelValue", "options"]),
          m.value.length ? (k(), O(s(U), {
            key: 0,
            modelValue: a.value,
            "onUpdate:modelValue": t[2] || (t[2] = (l) => a.value = l),
            options: m.value
          }, {
            default: r(() => [
              o(s(p), {
                class: "justify-between p-2 max-h-8 w-full gap-1 text-xs hover:bg-b-2",
                variant: "outlined"
              }, {
                default: r(() => [
                  u("span", {
                    class: M(a.value ? "text-c-1" : "text-c-3")
                  }, S(a.value ? a.value.label : "Select Folder"), 3),
                  o(s(N), {
                    class: "text-c-3",
                    icon: "ChevronDown",
                    size: "xs"
                  })
                ]),
                _: 1
              })
            ]),
            _: 1
          }, 8, ["modelValue", "options"])) : P("", !0)
        ]),
        o(s(p), {
          class: "max-h-8 text-xs p-0 px-3",
          onClick: w
        }, {
          default: r(() => [
            A(" Create Request ")
          ]),
          _: 1
        })
      ])
    ]));
  }
});
export {
  re as default
};
