// @scalar/api-client/dist/components/CommandPalette/TheCommandPalette.vue2.js
import { defineComponent as F, ref as i, computed as B, onMounted as T, onBeforeUnmount as U, openBlock as a, createElementBlock as l, Fragment as u, withDirectives as h, createElementVNode as s, vShow as b, unref as f, createVNode as x, vModelText as q, renderList as E, toDisplayString as I, normalizeClass as K, createTextVNode as H, createCommentVNode as G, createBlock as Q, resolveDynamicComponent as j, nextTick as J } from "vue";
import { useModal as O, ScalarIcon as w } from "@scalar/components";
import { useRouter as X } from "vue-router";
import { useWorkspace as Y } from "../../store/workspace.js";
import Z from "./CommandPaletteCollection.vue.js";
import ee from "./CommandPaletteExample.vue.js";
import oe from "./CommandPaletteFolder.vue.js";
import te from "./CommandPaletteImport.vue.js";
import ne from "./CommandPaletteRequest.vue.js";
import ae from "./CommandPaletteWorkspace.vue.js";
import { commandPaletteBus as V } from "../../libs/event-busses/command-palette-bus.js";
import { hotKeyBus as W } from "../../libs/event-busses/hot-keys-bus.js";
const le = { class: "commandmenu" }, se = { class: "bg-b-2 flex items-center rounded mb-2 pl-2 focus-within:bg-b-1 focus-within:shadow-border" }, ce = { for: "commandmenu" }, me = ["onClick"], re = {
  key: 0,
  class: "text-c-3 text-center text-sm p-2 pt-3"
}, ie = {
  name: "TheCommandPalette"
}, ue = {
  "Import Collection": te,
  "Create Request": ne,
  "Create Workspace": ae,
  "Add Folder": oe,
  "Create Collection": Z,
  "Add Example": ee
}, $e = /* @__PURE__ */ F({
  ...ie,
  setup(de) {
    const y = [
      {
        label: "Add to Request Sidebar",
        commands: [
          {
            name: "Create Request",
            icon: "ExternalLink"
          },
          {
            name: "Import Collection",
            icon: "Import"
          },
          {
            name: "Add Folder",
            icon: "Folder"
          },
          {
            name: "Create Collection",
            icon: "Collection"
          },
          {
            name: "Add Example",
            icon: "Example"
          }
        ]
      },
      {
        label: "More Actions",
        commands: [
          {
            name: "Create Workspace",
            icon: "Workspace"
          },
          {
            name: "Add Server",
            icon: "Brackets",
            path: "/servers"
          },
          {
            name: "Add Environment",
            icon: "Server",
            path: "/environment"
          },
          {
            name: "Add Cookie",
            icon: "Cookie",
            path: "/cookies"
          }
        ]
      }
    ], c = O(), { push: z } = X(), { activeWorkspace: M } = Y(), S = i(), m = i(""), r = i(null), d = i(-1), $ = i([]), C = B(
      () => y.reduce((e, n) => {
        const o = n.commands.filter(
          (t) => t.name.toLowerCase().includes(m.value.toLowerCase())
        );
        return [...e, ...o];
      }, [])
    ), p = () => {
      c.hide(), m.value = "", r.value = null, d.value = -1;
    }, A = (e) => {
      "path" in e ? (z(`/workspace/${M.value.uid}${e.path}`), p()) : r.value = e.name;
    }, D = i(), P = ({
      commandName: e,
      metaData: n
    } = {}) => {
      r.value = e ?? null, S.value = n, c.show(), J(() => {
        var o;
        return (o = D.value) == null ? void 0 : o.focus();
      });
    }, R = (e, n) => {
      var v;
      if (!c.open) return;
      n.preventDefault();
      const o = e === "up" ? -1 : 1, t = C.value.length;
      d.value = (d.value + o + t) % t, (v = $.value[d.value]) == null || v.scrollIntoView({
        behavior: "smooth",
        block: "center"
      });
    }, k = B(
      () => C.value[d.value]
    ), N = () => {
      k.value && A(k.value);
    }, g = (e) => {
      c.open && (e.closeModal && p(), e.commandPaletteUp && R("up", e.commandPaletteUp), e.commandPaletteDown && R("down", e.commandPaletteDown), e.commandPaletteSelect && N());
    };
    return T(() => {
      V.on(P), W.on(g);
    }), U(() => {
      V.off(P), W.off(g);
    }), (e, n) => (a(), l(u, null, [
      h(s("div", {
        class: "commandmenu-clickout",
        onClick: n[0] || (n[0] = (o) => p())
      }, null, 512), [
        [b, f(c).open]
      ]),
      h(s("div", le, [
        r.value ? (a(), l(u, { key: 1 }, [
          s("button", {
            class: "absolute p-0.75 hover:bg-b-3 rounded text-c-3 active:text-c-1 m-1.5 z-10",
            type: "button",
            onClick: n[2] || (n[2] = (o) => r.value = null)
          }, [
            x(f(w), {
              icon: "ChevronLeft",
              size: "md",
              thickness: "1.5"
            })
          ]),
          (a(), Q(j(ue[r.value]), {
            metaData: S.value,
            onClose: p
          }, null, 40, ["metaData"]))
        ], 64)) : (a(), l(u, { key: 0 }, [
          s("div", se, [
            s("label", ce, [
              x(f(w), {
                class: "text-c-1 mr-2.5",
                icon: "Search",
                size: "sm",
                thickness: "1.5"
              })
            ]),
            h(s("input", {
              id: "commandmenu",
              ref_key: "commandInputRef",
              ref: D,
              "onUpdate:modelValue": n[1] || (n[1] = (o) => m.value = o),
              autocomplete: "off",
              autofocus: "",
              class: "w-full rounded bg-none border-none py-1.5 text-sm focus:outline-none",
              placeholder: "Search commands...",
              type: "text"
            }, null, 512), [
              [q, m.value]
            ])
          ]),
          (a(), l(u, null, E(y, (o) => (a(), l(u, {
            key: o.label
          }, [
            h(s("div", { class: "text-c-3 font-medium text-xs mt-2" }, I(o.label), 513), [
              [
                b,
                o.commands.filter(
                  (t) => t.name.toLowerCase().includes(m.value.toLowerCase())
                ).length > 0
              ]
            ]),
            (a(!0), l(u, null, E(o.commands.filter(
              (t) => t.name.toLowerCase().includes(m.value.toLowerCase())
            ), (t, v) => {
              var L;
              return a(), l("div", {
                key: t.name,
                ref_for: !0,
                ref: (_) => {
                  _ && ($.value[v] = _);
                },
                class: K(["commandmenu-item text-sm flex items-center py-1.5 px-2 rounded hover:bg-b-2 cursor-pointer", {
                  "bg-b-2": t.name === ((L = k.value) == null ? void 0 : L.name)
                }]),
                onClick: (_) => A(t)
              }, [
                x(f(w), {
                  class: "text-c-1 mr-2.5",
                  icon: t.icon,
                  size: "md",
                  thickness: "1.5"
                }, null, 8, ["icon"]),
                H(" " + I(t.name), 1)
              ], 10, me);
            }), 128))
          ], 64))), 64)),
          C.value.length ? G("", !0) : (a(), l("div", re, " No commands found "))
        ], 64))
      ], 512), [
        [b, f(c).open]
      ])
    ], 64));
  }
});
export {
  ue as PaletteComponents,
  $e as default
};
