// @scalar/api-client/dist/components/CommandPalette/WatchModeToggle.vue.js
import { defineComponent as r, computed as i, createElementBlock as p, openBlock as f, normalizeClass as d, createElementVNode as V, createVNode as n, createTextVNode as g, unref as s } from "vue";
import { ScalarIcon as x, ScalarToggle as b } from "@scalar/components";
const N = /* @__PURE__ */ r({
  __name: "WatchModeToggle",
  props: {
    modelValue: { type: Boolean },
    disabled: { type: Boolean, default: !1 }
  },
  emits: ["update:modelValue"],
  setup(t, { emit: m }) {
    const u = t, l = m, o = i({
      get: () => u.modelValue,
      set: (a) => l("update:modelValue", a)
    });
    return (a, e) => (f(), p("label", {
      class: d(["text-c-2 flex items-center gap-2 rounded p-3 py-1.5 text-sm select-none", t.disabled ? "cursor-default" : "cursor-pointer"]),
      for: "watch-toggle"
    }, [
      V("span", {
        class: d(["text-c-1 flex items-center gap-1 text-xs font-medium", { "text-c-3": !o.value }])
      }, [
        n(s(x), {
          icon: "Watch",
          size: "sm"
        }),
        e[1] || (e[1] = g(" Watch Mode ", -1))
      ], 2),
      n(s(b), {
        id: "watch-toggle",
        disabled: !!t.disabled,
        modelValue: o.value,
        "onUpdate:modelValue": e[0] || (e[0] = (c) => l("update:modelValue", c))
      }, null, 8, ["disabled", "modelValue"])
    ], 2));
  }
});
export {
  N as default
};
