// @scalar/api-reference/dist/features/ExampleRequest/ExamplePicker.vue2.js
import { defineComponent as i, ref as c, watch as s, openBlock as d, createBlock as f, withCtx as x, createTextVNode as V, toDisplayString as _ } from "vue";
import b from "./TextSelect.vue.js";
/* empty css                */
const k = /* @__PURE__ */ i({
  __name: "ExamplePicker",
  props: {
    examples: {}
  },
  emits: ["update:modelValue"],
  setup(p, { emit: r }) {
    const l = p, n = r, t = c(Object.keys(l.examples)[0]);
    function u(e) {
      e && (t.value = e);
    }
    s(
      () => l.examples,
      () => {
        u(Object.keys(l.examples)[0]);
      },
      { immediate: !0 }
    ), s(
      t,
      () => {
        t.value && n("update:modelValue", t.value);
      },
      { immediate: !0 }
    );
    function o(e) {
      if (!e)
        return "Select an example";
      const a = l.examples[e];
      return (a == null ? void 0 : a.summary) ?? e;
    }
    return (e, a) => (d(), f(b, {
      modelValue: t.value,
      "onUpdate:modelValue": a[0] || (a[0] = (m) => t.value = m),
      class: "example-selector",
      options: Object.keys(e.examples).map((m) => ({
        label: o(m),
        value: m
      }))
    }, {
      default: x(() => [
        V(_(o(t.value)), 1)
      ]),
      _: 1
    }, 8, ["modelValue", "options"]));
  }
});
export {
  k as default
};
