// @scalar/api-reference/dist/features/ExampleRequest/ExampleRequest.vue2.js
import { defineComponent as U, ref as _, watch as L, computed as y, inject as I, onServerPrefetch as M, useSSRContext as Y, openBlock as d, createBlock as b, unref as s, withCtx as c, createVNode as p, createElementBlock as h, Fragment as V, createTextVNode as H, toDisplayString as v, createElementVNode as O, renderSlot as j, createCommentVNode as N } from "vue";
import { ScalarCodeBlock as z } from "@scalar/components";
import { createHash as D, ssrState as Q } from "@scalar/oas-utils/helpers";
import { getRequestFromOperation as W } from "@scalar/oas-utils/spec-getters";
import { snippetz as R } from "@scalar/snippetz";
import { asyncComputed as X } from "@vueuse/core";
import { HTTPSnippet as Z } from "httpsnippet-lite";
import G from "./ExamplePicker.vue.js";
import ee from "./TextSelect.vue.js";
/* empty css                */
import { GLOBAL_SECURITY_SYMBOL as te } from "../../helpers/provideSymbols.js";
import oe from "../../components/Card/Card.vue.js";
import re from "../../components/Card/CardHeader.vue.js";
import ae from "../../components/HttpMethod/HttpMethod.vue.js";
import ne from "../../components/Card/CardContent.vue.js";
import { getSecretCredentialsFromAuthentication as se, getRequestFromAuthentication as le } from "../../legacy/helpers/getRequestFromAuthentication.js";
import ie from "../../components/Card/CardFooter.vue.js";
import { useHttpClientStore as ce } from "../../stores/useHttpClientStore.js";
import { getApiClientRequest as pe } from "../../helpers/getApiClientRequest.js";
import { useServerStore as me } from "../../legacy/stores/useServerStore.js";
import { useAuthenticationStore as ue } from "../../legacy/stores/useAuthenticationStore.js";
import { getHarRequest as de } from "../../helpers/getHarRequest.js";
import { getUrlFromServerState as fe } from "../../legacy/helpers/getUrlFromServerState.js";
const ye = { class: "request-header" }, ge = { class: "code-snippet" }, he = {
  key: 0,
  class: "request-card-footer-addon"
}, Ue = /* @__PURE__ */ U({
  __name: "ExampleRequest",
  props: {
    customExamples: {},
    operation: {}
  },
  setup(B) {
    var k;
    const a = B, K = `components-Content-Operation-Example-Request${D(
      a.operation.path + a.operation.httpVerb + a.operation.operationId
    )}`, x = _(), {
      httpClient: n,
      setHttpClient: T,
      availableTargets: $,
      httpTargetTitle: w,
      httpClientTitle: F
    } = ce(), { server: E } = me(), { authentication: g } = ue(), l = _(
      // Default to first custom example
      (k = a.customExamples) != null && k.length ? {
        targetKey: "customExamples",
        clientKey: 0
      } : (
        // Otherwise use the globally selected HTTP client
        {
          targetKey: n.targetKey,
          clientKey: n.clientKey
        }
      )
    );
    L(n, () => {
      l.value = {
        targetKey: n.targetKey,
        clientKey: n.clientKey
      };
    });
    const C = y(
      () => {
        var e, o, t, r;
        return Object.keys(
          ((r = (t = (o = (e = a.operation.information) == null ? void 0 : e.requestBody) == null ? void 0 : o.content) == null ? void 0 : t["application/json"]) == null ? void 0 : r.examples) ?? {}
        ).length > 1;
      }
    ), i = I(te);
    async function S() {
      var r, m, u;
      if (l.value.targetKey === "customExamples")
        return ((m = (r = a.customExamples) == null ? void 0 : r[l.value.clientKey]) == null ? void 0 : m.source) ?? "";
      const e = de(
        {
          url: fe(E)
        },
        W(
          a.operation,
          {
            replaceVariables: !0
          },
          x.value
        ),
        le(
          g,
          ((u = a.operation.information) == null ? void 0 : u.security) ?? (i == null ? void 0 : i())
        )
      ), o = n.clientKey === "undici" || n.clientKey === "fetch" || n.clientKey === "ofetch" ? n.clientKey : null, t = n.targetKey.replace("javascript", "js");
      if (o && R().hasPlugin(t, o) && (t === "node" || t === "js"))
        return R().print(t, o, e) ?? "";
      try {
        return await new Z(e).convert(
          n.targetKey,
          n.clientKey
        );
      } catch (f) {
        return console.error("[ExampleRequest]", f), "";
      }
    }
    const q = X(async () => {
      try {
        return await S();
      } catch (e) {
        return console.error("[generateSnippet]", e), "";
      }
    }, Q[K] ?? "");
    M(async () => {
      const e = Y();
      e.payload.data[K] = await S();
    }), y(() => pe({
      serverState: E,
      authenticationState: g,
      operation: a.operation,
      globalSecurity: i == null ? void 0 : i()
    }));
    const A = y(() => {
      var o, t, r;
      const e = (
        // Specified language
        ((o = l.value) == null ? void 0 : o.targetKey) === "customExamples" ? ((r = (t = a.customExamples) == null ? void 0 : t[l.value.clientKey]) == null ? void 0 : r.lang) ?? "plaintext" : (
          // Or language for the globally selected HTTP client
          n.targetKey
        )
      );
      return e === "shell" && q.value.includes("curl") ? "curl" : e === "Objective-C" ? "objc" : e;
    }), J = y(() => {
      var o;
      const e = $.value.map((t) => ({
        value: t.key,
        label: t.title,
        options: t.clients.map((r) => ({
          value: JSON.stringify({
            targetKey: t.key,
            clientKey: r.key
          }),
          label: r.title
        }))
      }));
      return (o = a.customExamples) != null && o.length && e.unshift({
        value: "customExamples",
        label: "Examples",
        options: a.customExamples.map((t, r) => ({
          value: JSON.stringify({
            targetKey: "customExamples",
            clientKey: r
          }),
          label: t.label ?? t.lang ?? `Example #${r + 1}`
        }))
      }), e;
    });
    function P(e) {
      const o = JSON.parse(e);
      o.targetKey === "customExamples" ? l.value = o : T(o);
    }
    return (e, o) => (d(), b(s(oe), { class: "dark-mode" }, {
      default: c(() => [
        p(s(re), { muted: "" }, {
          actions: c(() => [
            p(ee, {
              class: "request-client-picker",
              modelValue: JSON.stringify(l.value),
              options: J.value,
              "onUpdate:modelValue": P
            }, {
              default: c(() => {
                var t;
                return [
                  l.value.targetKey === "customExamples" ? (d(), h(V, { key: 0 }, [
                    H(v(((t = a.customExamples) == null ? void 0 : t[l.value.clientKey].label) ?? "Example"), 1)
                  ], 64)) : (d(), h(V, { key: 1 }, [
                    H(v(s(w)) + " " + v(s(F)), 1)
                  ], 64))
                ];
              }),
              _: 1
            }, 8, ["modelValue", "options"])
          ]),
          default: c(() => [
            O("div", ye, [
              p(s(ae), {
                as: "span",
                class: "request-method",
                method: e.operation.httpVerb
              }, null, 8, ["method"]),
              j(e.$slots, "header", {}, void 0, !0)
            ])
          ]),
          _: 3
        }),
        p(s(ne), {
          borderless: "",
          class: "request-editor-section custom-scroll",
          frameless: ""
        }, {
          default: c(() => [
            O("div", ge, [
              p(s(z), {
                class: "bg-b-2",
                content: s(q),
                hideCredentials: s(se)(s(g)),
                lang: A.value,
                lineNumbers: ""
              }, null, 8, ["content", "hideCredentials", "lang"])
            ])
          ]),
          _: 1
        }),
        C.value || e.$slots.footer ? (d(), b(s(ie), {
          key: 0,
          class: "request-card-footer",
          contrast: ""
        }, {
          default: c(() => {
            var t, r, m, u;
            return [
              C.value ? (d(), h("div", he, [
                p(G, {
                  class: "request-example-selector",
                  examples: ((u = (m = (r = (t = e.operation.information) == null ? void 0 : t.requestBody) == null ? void 0 : r.content) == null ? void 0 : m["application/json"]) == null ? void 0 : u.examples) ?? [],
                  "onUpdate:modelValue": o[0] || (o[0] = (f) => x.value = f)
                }, null, 8, ["examples"])
              ])) : N("", !0),
              j(e.$slots, "footer", {}, void 0, !0)
            ];
          }),
          _: 3
        })) : N("", !0)
      ]),
      _: 3
    }));
  }
});
export {
  Ue as default
};
