// @scalar/api-reference/dist/features/ExampleRequest/TextSelect.vue.js
import { defineComponent as d, openBlock as l, createElementBlock as t, normalizeClass as i, createElementVNode as u, renderSlot as m, Fragment as s, renderList as r, toDisplayString as p } from "vue";
const v = ["value"], _ = ["label"], c = ["value"], g = ["value"], f = /* @__PURE__ */ d({
  __name: "TextSelect",
  props: {
    modelValue: {},
    options: {}
  },
  emits: ["update:modelValue"],
  setup(k) {
    return (a, n) => (l(), t("div", {
      class: i(["text-select", a.options.length === 1 ? "text-select--single-option" : ""])
    }, [
      u("span", null, [
        m(a.$slots, "default")
      ]),
      u("select", {
        value: a.modelValue,
        onInput: n[0] || (n[0] = (e) => a.$emit("update:modelValue", e.target.value))
      }, [
        (l(!0), t(s, null, r(a.options, (e) => (l(), t(s, {
          key: e.value
        }, [
          e.options ? (l(), t("optgroup", {
            key: 0,
            label: e.label
          }, [
            (l(!0), t(s, null, r(e.options, (o) => (l(), t("option", {
              key: o.value,
              value: o.value
            }, p(o.label), 9, c))), 128))
          ], 8, _)) : (l(), t("option", {
            key: e.value,
            value: e.value
          }, p(e.label), 9, g))
        ], 64))), 128))
      ], 40, v)
    ], 2));
  }
});
export {
  f as default
};
