// @scalar/components/dist/components/ScalarTooltip/ScalarTooltip.vue.js
import { defineComponent as n, openBlock as c, createBlock as d, unref as l, withCtx as t, createVNode as s, normalizeClass as i, renderSlot as o } from "vue";
import { TooltipProvider as f, TooltipRoot as p, TooltipTrigger as u, TooltipPortal as g, TooltipContent as k } from "radix-vue";
const C = /* @__PURE__ */ n({
  __name: "ScalarTooltip",
  props: {
    click: {},
    delay: {},
    skipDelay: { default: 1e3 },
    align: { default: "center" },
    side: { default: "top" },
    sideOffset: {},
    class: {},
    triggerClass: {},
    resize: { type: Boolean }
  },
  emits: ["click"],
  setup(r) {
    const e = r;
    return (a, m) => (c(), d(l(f), {
      delayDuration: e.delay,
      skipDelayDuration: e.skipDelay
    }, {
      default: t(() => [
        s(l(p), null, {
          default: t(() => [
            s(l(u), {
              class: i(["flex items-center justify-center", [e.resize ? "w-full" : "", e.triggerClass]]),
              onClick: e.click
            }, {
              default: t(() => [
                o(a.$slots, "trigger")
              ]),
              _: 3
            }, 8, ["class", "onClick"]),
            s(l(g), { to: ".scalar-client" }, {
              default: t(() => [
                s(l(k), {
                  align: e.align,
                  class: i(["scalar-app", e.class]),
                  side: e.side,
                  sideOffset: e.sideOffset
                }, {
                  default: t(() => [
                    o(a.$slots, "content")
                  ]),
                  _: 3
                }, 8, ["align", "class", "side", "sideOffset"])
              ]),
              _: 3
            })
          ]),
          _: 3
        })
      ]),
      _: 3
    }, 8, ["delayDuration", "skipDelayDuration"]));
  }
});
export {
  C as default
};
