// @scalar/api-reference/dist/helpers/getApiClientRequest.js
import { getRequestFromOperation as n, getParametersFromOperation as d } from "@scalar/oas-utils/spec-getters";
import { getHarRequest as f } from "./getHarRequest.js";
import { getRequestFromAuthentication as c } from "../legacy/helpers/getRequestFromAuthentication.js";
import { getUrlFromServerState as s } from "../legacy/helpers/getUrlFromServerState.js";
function O({
  serverState: t,
  authenticationState: o,
  operation: e,
  globalSecurity: l
}) {
  var u, a;
  const r = f(
    {
      url: s(t)
    },
    n(e, { requiredOnly: !1 }),
    // Only generate authentication parameters if an authentication state is passed.
    o ? c(
      o,
      ((u = e.information) == null ? void 0 : u.security) ?? l ?? []
    ) : {}
  ), p = n(e, {
    requiredOnly: !1
  }), q = d(e, "path", !1);
  return {
    id: e.operationId,
    name: e.name,
    type: r.method,
    path: p.path ?? "",
    variables: q,
    cookies: i(r.cookies),
    query: r.queryString.map((m) => ({ ...m, enabled: m.required ?? !0 })),
    headers: i(r.headers),
    url: s(t) ?? "",
    body: (a = r.postData) == null ? void 0 : a.text
  };
}
function i(t) {
  return (t ?? []).map((o) => ({ ...o, enabled: !0 }));
}
export {
  O as getApiClientRequest
};
