// @scalar/api-client/dist/store/request-example.js
import { LS_KEYS as s } from "@scalar/helpers/object/local-storage";
import { iterateTitle as c } from "@scalar/helpers/string/iterate-title";
import { createExampleFromRequest as x } from "@scalar/oas-utils/entities/spec";
import { mutationFactory as u } from "@scalar/object-utils/mutator-record";
import { reactive as i } from "vue";
function L(a) {
  const t = i({}), m = u(
    t,
    i({}),
    a && s.REQUEST_EXAMPLE
  );
  return {
    requestExamples: t,
    requestExampleMutators: m
  };
}
function T({
  requestExamples: a,
  requestExampleMutators: t,
  requestMutators: m,
  requests: d
}) {
  return {
    addRequestExample: (e, o) => {
      const n = o ?? c(
        (e.summary ?? "Example") + " #1",
        (p) => e.examples.some((l) => a[l]?.name === p)
      ), r = x(e, n);
      return t.add(r), m.edit(e.uid, "examples", [...e.examples, r.uid]), r;
    },
    deleteRequestExample: (e) => {
      e.requestUid && (m.edit(
        e.requestUid,
        "examples",
        d[e.requestUid]?.examples.filter((o) => o !== e.uid) || []
      ), t.delete(e.uid));
    }
  };
}
export {
  L as createStoreRequestExamples,
  T as extendedExampleDataFactory
};
