// @scalar/api-client/dist/store/workspace.js
import { PathId as f, fallbackMissingParams as xe } from "../router.js";
import { useModal as je } from "@scalar/components";
import { createWorkspace as Ce } from "@scalar/oas-utils/entities/workspace";
import { createCollection as We } from "@scalar/oas-utils/entities/workspace/collection";
import { createCookie as ge } from "@scalar/oas-utils/entities/workspace/cookie";
import { createEnvironment as Fe } from "@scalar/oas-utils/entities/workspace/environment";
import { createFolder as Ie } from "@scalar/oas-utils/entities/workspace/folder";
import { createSecurityScheme as Ke, securitySchemeApiKeyIn as Ae } from "@scalar/oas-utils/entities/workspace/security";
import { createServer as Pe } from "@scalar/oas-utils/entities/workspace/server";
import { createRequest as Te, createRequestExample as Ne, createRequestExampleParameter as qe } from "@scalar/oas-utils/entities/workspace/spec";
import { iterateTitle as _e, fetchSpecFromUrl as Le } from "@scalar/oas-utils/helpers";
import { getRequestBodyFromOperation as Be } from "@scalar/oas-utils/spec-getters";
import { importSpecToWorkspace as De } from "@scalar/oas-utils/transforms";
import { mutationFactory as E, LS_KEYS as b } from "@scalar/object-utils/mutator-record";
import { computed as u, reactive as l, ref as Ve, inject as $e, toRaw as Je } from "vue";
const lt = (H, y = !0) => {
  const W = u(() => {
    const e = {
      [f.Collection]: "default",
      [f.Environment]: "default",
      [f.Request]: "default",
      [f.Examples]: "default",
      [f.Schema]: "default",
      [f.Cookies]: "default",
      [f.Servers]: "default",
      [f.Workspace]: "default"
    }, t = H.currentRoute.value;
    return t && Object.values(f).forEach((s) => {
      t.params[s] && (e[s] = t.params[s]);
    }), e;
  }), m = l({}), h = E(
    m,
    l({}),
    y && b.REQUEST
  ), q = (e, t, s) => {
    const i = Te(e), c = Q(i, s);
    return i.childUids.push(c.uid), h.add(i), t && (d[t] ? p.edit(t, "childUids", [
      ...d[t].childUids,
      i.uid
    ]) : r[t] && R.edit(t, "childUids", [
      ...r[t].childUids,
      i.uid
    ])), i;
  }, _ = (e, t) => {
    e.childUids.forEach((s) => F.delete(s)), t && d[t] ? p.edit(
      t,
      "childUids",
      d[t].childUids.filter((s) => s !== e.uid)
    ) : t && r[t] && R.edit(
      t,
      "childUids",
      r[t].childUids.filter((s) => s !== e.uid)
    ), h.delete(e.uid);
  }, U = u(() => {
    var i, c;
    const e = W.value[f.Request], t = (c = (i = Z.value) == null ? void 0 : i[0]) == null ? void 0 : c.uid, s = m[e] ?? m[t];
    return xe(f.Request, s), s;
  }), L = (e, t = []) => {
    const s = Object.values(d).find(
      (c) => c.childUids.includes(e)
    );
    if (s) return [...t, s.uid];
    const i = Object.values(r).find(
      ({ childUids: c }) => c.includes(e)
    );
    return i ? L(i.uid, [...t, i.uid]) : t;
  }, g = l({}), F = E(
    g,
    l({}),
    y && b.REQUEST_EXAMPLE
  ), K = (e) => {
    var t, s, i, c, n, v, k;
    return qe({
      key: e.name,
      value: e.schema && "default" in e.schema ? e.schema.default : e.schema && "examples" in e.schema && e.schema.examples.length > 0 ? e.schema.examples[0] : "",
      description: e.description,
      required: e.required,
      enabled: !!e.required,
      enum: (t = e.schema) == null ? void 0 : t.enum,
      type: (s = e.schema) == null ? void 0 : s.type,
      format: (i = e.schema) == null ? void 0 : i.format,
      minimum: (c = e.schema) == null ? void 0 : c.minimum,
      maximum: (n = e.schema) == null ? void 0 : n.maximum,
      default: (v = e.schema) == null ? void 0 : v.default,
      nullable: (k = e.schema) == null ? void 0 : k.nullable
    });
  }, Q = (e, t, s) => {
    var k;
    const i = {
      path: Object.values(e.parameters.path).map(K),
      query: Object.values(e.parameters.query).map(K),
      headers: Object.values(e.parameters.headers).map(
        K
      ),
      cookies: Object.values(e.parameters.cookies).map(
        K
      )
    }, c = {
      activeBody: "raw",
      raw: {
        encoding: "json",
        value: ""
      }
    };
    if (e.requestBody) {
      const S = Be({
        httpVerb: e.method,
        path: e.path,
        information: {
          requestBody: e.requestBody
        }
      });
      ((k = S == null ? void 0 : S.postData) == null ? void 0 : k.mimeType) === "application/json" && (i.headers.push({
        key: "Content-Type",
        value: "application/json",
        enabled: !0
      }), c.activeBody = "raw", c.raw.value = S.postData.text);
    }
    const n = s ?? _e(
      (e.summary ?? "Example") + " #1",
      (S) => e.childUids.some((o) => S === g[o].name)
    ), v = Ne({
      url: t != null && t.url ? `{{${t == null ? void 0 : t.url}}}${e.path}` : e.path,
      requestUid: e.uid,
      parameters: i,
      name: n,
      body: c
    });
    return F.add(v), v;
  }, ae = (e, t) => {
    const s = Q(
      e,
      V.value[0],
      t
    );
    return h.edit(e.uid, "childUids", [
      ...e.childUids,
      s.uid
    ]), s;
  }, de = (e) => {
    h.edit(
      e.requestUid,
      "childUids",
      m[e.requestUid].childUids.filter(
        (t) => t !== e.uid
      )
    ), F.delete(e.uid);
  }, le = u(
    () => {
      var e;
      return g[W.value[f.Examples]] ?? g[((e = U.value) == null ? void 0 : e.childUids[0]) ?? ""];
    }
  ), re = u(() => Object.values(m).flatMap((e) => e.history).filter((e) => e.request && e.response).sort((e, t) => t.timestamp - e.timestamp)), B = l({
    default: Fe({
      uid: "default",
      name: "Global Environment",
      color: "blue",
      raw: JSON.stringify({ exampleKey: "exampleValue" }, null, 2),
      parsed: [],
      isDefault: !0
    })
  }), G = E(
    B,
    l({}),
    y && b.ENVIRONMENT
  ), ne = (e) => {
    if (e === "default") {
      console.warn("Default environment cannot be deleted.");
      return;
    }
    G.delete(e);
  }, ue = u(() => {
    const e = V.value.map((s) => ({
      key: s.url,
      value: s.url
    })), t = Object.values(B).map((s) => {
      try {
        return {
          _scalarEnvId: s.uid,
          ...JSON.parse(s.raw)
        };
      } catch {
        return null;
      }
    }).filter((s) => s).flatMap(
      (s) => Object.entries(s).flatMap(([i, c]) => i !== "_scalarEnvId" ? [{ _scalarEnvId: s._scalarEnvId, key: i, value: c }] : [])
    );
    return [...e, ...t];
  }), X = l({
    default: ge({
      uid: "default",
      name: "Cookie",
      value: "",
      domain: "",
      path: "/",
      secure: !1,
      httpOnly: !1,
      sameSite: "None"
    })
  }), fe = E(
    X,
    l({}),
    y && b.COOKIE
  ), me = u(
    () => W.value[f.Cookies]
  ), w = l({}), O = E(
    w,
    l({}),
    y && b.WORKSPACE
  ), he = (e = {}) => {
    const t = Ce(e);
    O.add(t);
    const s = $(
      {
        spec: {
          info: {
            title: "Drafts"
          }
        }
      },
      t.uid
    );
    return q({ summary: "My First Request" }, s.uid), t;
  }, pe = (e) => {
    if (Object.keys(w).length <= 1) {
      console.warn("The last workspace cannot be deleted.");
      return;
    }
    O.delete(e);
  }, ve = (e, t) => {
    w[e] ? O.edit(e, "name", t) : console.warn(`Workspace with uid ${e} not found.`);
  }, x = u(
    () => w[W.value[f.Workspace]] ?? w[Object.keys(w)[0]]
  ), D = u(
    () => {
      var e;
      return (e = x.value) == null ? void 0 : e.collectionUids.map((t) => d[t]).sort((t, s) => {
        var i, c, n, v;
        return ((c = (i = t.spec) == null ? void 0 : i.info) == null ? void 0 : c.title) === "Drafts" ? 1 : ((v = (n = s.spec) == null ? void 0 : n.info) == null ? void 0 : v.title) === "Drafts" ? -1 : 0;
      });
    }
  ), V = u(
    () => {
      var e;
      return (e = D.value) == null ? void 0 : e.flatMap(
        (t) => t.spec.serverUids.map((s) => T[s])
      );
    }
  ), z = (e) => m[e] ?? r[e].childUids.flatMap((t) => z(t)), Z = u(
    () => {
      var e;
      return (e = D.value) == null ? void 0 : e.flatMap(
        (t) => t.childUids.flatMap((s) => z(s))
      );
    }
  ), ye = u(() => {
    var e;
    return ((e = x.value) == null ? void 0 : e.isReadOnly) ?? !1;
  }), d = l({}), p = E(
    d,
    l({}),
    y && b.COLLECTION
  ), $ = (e, t) => {
    const s = We(e);
    return O.edit(t, "collectionUids", [
      ...w[t].collectionUids,
      s.uid
    ]), p.add(s), s;
  }, Se = (e) => {
    var t, s, i;
    if (x.value) {
      if (((i = (s = (t = d[e.uid]) == null ? void 0 : t.spec) == null ? void 0 : s.info) == null ? void 0 : i.title) === "Drafts") {
        console.warn("The drafts collection cannot be deleted");
        return;
      }
      if (Object.values(d).length === 1) {
        console.warn("You must have at least one collection");
        return;
      }
      e.childUids.forEach((c) => {
        m[c] ? _(m[c]) : r[c] && J(r[c]);
      }), O.edit(
        x.value.uid,
        "collectionUids",
        x.value.collectionUids.filter(
          (c) => c !== e.uid
        )
      ), p.delete(e.uid);
    }
  }, A = u(() => {
    const e = Object.values(d)[0];
    if (!U.value) return e;
    const t = L(U.value.uid);
    if (!t.length) return null;
    const s = t[t.length - 1];
    return d[s] ?? e;
  }), Ee = u(
    () => A.value && T[A.value.selectedServerUid]
  ), r = l({}), R = E(
    r,
    l({}),
    y && b.FOLDER
  ), ee = (e, t) => {
    const s = Ie(e);
    if (t)
      if (d[t])
        p.edit(t, "childUids", [
          ...d[t].childUids,
          s.uid
        ]);
      else if (r[t])
        R.edit(t, "childUids", [
          ...r[t].childUids,
          s.uid
        ]);
      else {
        console.error("Could not find folder's parent ID");
        return;
      }
    R.add(s);
  }, J = (e, t) => {
    e.childUids.forEach((s) => {
      m[s] ? _(m[s]) : r[s] && J(r[s]);
    }), t && d[t] ? p.edit(
      t,
      "childUids",
      d[t].childUids.filter((s) => s !== e.uid)
    ) : t && r[t] && R.edit(
      t,
      "childUids",
      r[t].childUids.filter((s) => s !== e.uid)
    ), R.delete(e.uid);
  }, Y = l({}), P = E(
    Y,
    l({}),
    y && b.SECURITY_SCHEME
  ), be = u(
    () => {
      var e;
      return ((e = U.value) == null ? void 0 : e.selectedSecuritySchemeUids.map(
        (t) => Y[t]
      )) ?? [];
    }
  ), we = u(
    () => {
      var e, t;
      return ((e = U.value) == null ? void 0 : e.security) ?? ((t = A.value) == null ? void 0 : t.spec.security) ?? [];
    }
  ), I = (e, t, s, i = !1) => {
    const c = Ke(e);
    P.add(c), t && e.nameKey && p.edit(
      t,
      `securitySchemeDict.${e.nameKey}`,
      c.uid
    ), s && (h.edit(s.uid, "securitySchemeUids", [
      ...s.securitySchemeUids,
      c.uid
    ]), i && h.edit(s.uid, "selectedSecuritySchemeUids", [
      ...s.selectedSecuritySchemeUids,
      c.uid
    ]));
  }, Re = (e, t) => {
    t && (h.edit(
      t.uid,
      "securitySchemeUids",
      t.securitySchemeUids.filter((s) => s !== e.uid)
    ), h.edit(
      t.uid,
      "selectedSecuritySchemeUids",
      t.selectedSecuritySchemeUids.filter((s) => s !== e.uid)
    )), P.delete(e.uid);
  }, T = l({}), N = E(
    T,
    l({}),
    y && b.SERVER
  ), te = (e, t) => {
    const s = Pe(e);
    t && p.edit(t, "spec.serverUids", [
      ...d[t].spec.serverUids,
      s.uid
    ]), N.add(s);
  }, ke = (e, t) => {
    p.edit(
      t,
      "spec.serverUids",
      d[t].spec.serverUids.filter(
        (s) => s !== e
      )
    ), N.delete(e);
  }, se = async (e, t = "default", s) => {
    var S;
    const i = Je(e), c = await De(i, s);
    c.requests.forEach(
      (o) => q(o, void 0, c.servers[0])
    );
    const n = $(c.collection, t);
    c.folders.forEach((o) => ee(o)), c.servers.forEach((o) => te(o));
    const v = Object.entries(
      (((S = c.components) == null ? void 0 : S.securitySchemes) || c.securityDefinitions) ?? {}
    ), k = (o, a) => {
      var C;
      const M = (C = Object.keys(o[0])) == null ? void 0 : C[0];
      if (!M) return;
      const j = n.securitySchemeDict[M];
      h.edit(a.uid, "selectedSecuritySchemeUids", [j]);
    };
    v.forEach(([o, a]) => {
      if (a && "flows" in a) {
        const { flows: M, ...j } = a;
        if (M) {
          const C = Object.entries(M), [ie, oe] = C[0];
          ie && oe && I(
            {
              ...j,
              type: "oauth2",
              flow: { ...oe, type: ie },
              nameKey: o
            },
            n.uid
          );
        }
      } else a.type === "basic" ? I(
        { ...a, type: "http", scheme: "basic", nameKey: o },
        n.uid
      ) : a.type === "http" ? I(
        {
          ...a,
          type: "http",
          scheme: a.scheme === "bearer" ? "bearer" : "basic",
          nameKey: o
        },
        n.uid
      ) : a.type === "apiKey" && I(
        {
          ...a,
          type: "apiKey",
          in: Ae.includes(
            a.in
          ) ? a.in : "header",
          nameKey: o
        },
        n.uid
      );
    }), v.length && Object.values(m).forEach((o) => {
      var M, j;
      const a = ((M = o.security) == null ? void 0 : M.filter((C) => JSON.stringify(C) !== "{}")) ?? [];
      a != null && a.length ? k(a, o) : typeof o.security > "u" && ((j = n.spec.security) != null && j.length) && k(n.spec.security, o);
    });
  };
  async function Me(e, t, s) {
    try {
      const i = await Le(e, t);
      await se(i, void 0, s);
    } catch (i) {
      console.error("Failed to fetch spec from URL:", i);
    }
  }
  const ce = Ve(localStorage.getItem("sidebarWidth") || "280px"), Oe = (e) => {
    ce.value = e, localStorage.setItem("sidebarWidth", e);
  }, Ue = je();
  return {
    // ---------------------------------------------------------------------------
    // STATE
    workspaces: w,
    collections: d,
    cookies: X,
    environments: B,
    folders: r,
    requestExamples: g,
    requests: m,
    servers: T,
    securitySchemes: Y,
    activeCollection: A,
    activeCookieId: me,
    activeExample: le,
    activeRequest: U,
    activeRouterParams: W,
    activeSecurityRequirements: we,
    activeSecuritySchemes: be,
    activeServer: Ee,
    activeWorkspace: x,
    activeWorkspaceCollections: D,
    activeWorkspaceServers: V,
    activeParsedEnvironments: ue,
    activeWorkspaceRequests: Z,
    modalState: Ue,
    isReadOnly: ye,
    router: H,
    sidebarWidth: ce,
    setSidebarWidth: Oe,
    // ---------------------------------------------------------------------------
    // METHODS
    findRequestFolders: L,
    importSpecFile: se,
    importSpecFromUrl: Me,
    cookieMutators: fe,
    collectionMutators: {
      ...p,
      rawAdd: p.add,
      add: $,
      delete: Se
    },
    environmentMutators: {
      ...G,
      delete: ne
    },
    folderMutators: {
      ...R,
      rawAdd: R.add,
      add: ee,
      delete: J
    },
    requestMutators: {
      ...h,
      rawAdd: h.add,
      add: q,
      delete: _
    },
    requestExampleMutators: {
      ...F,
      rawAdd: F.add,
      add: ae,
      delete: de
    },
    requestsHistory: re,
    securitySchemeMutators: {
      ...P,
      rawAdd: P.add,
      add: I,
      delete: Re
    },
    serverMutators: {
      ...N,
      rawAdd: N.add,
      add: te,
      delete: ke
    },
    workspaceMutators: {
      ...O,
      rawAdd: O.add,
      add: he,
      delete: pe,
      rename: ve
    }
  };
}, rt = () => $e("workspace");
export {
  lt as createWorkspaceStore,
  rt as useWorkspace
};
