// @scalar/api-reference/dist/components/Content/Schema/SchemaEnumPropertyItem.vue2.js
import { defineComponent, createElementBlock, openBlock, createElementVNode, createCommentVNode, createVNode, unref } from "vue";
import { ScalarWrappingText, ScalarMarkdown } from "@scalar/components";
const _hoisted_1 = { class: "property-enum-value" };
const _hoisted_2 = { class: "property-enum-value-content" };
const _hoisted_3 = { class: "property-enum-value-label" };
const _hoisted_4 = {
  key: 0,
  class: "property-enum-value-description"
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "SchemaEnumPropertyItem",
  props: {
    label: {},
    description: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("li", _hoisted_1, [
        createElementVNode("div", _hoisted_2, [
          createElementVNode("span", _hoisted_3, [
            createVNode(unref(ScalarWrappingText), {
              text: __props.label,
              preset: "property"
            }, null, 8, ["text"])
          ]),
          __props.description ? (openBlock(), createElementBlock("span", _hoisted_4, [
            createVNode(unref(ScalarMarkdown), { value: __props.description }, null, 8, ["value"])
          ])) : createCommentVNode("", true)
        ])
      ]);
    };
  }
});
export {
  _sfc_main as default
};
