// @scalar/api-reference/dist/components/Content/Schema/SchemaPropertyHeading.vue2.js
import { defineComponent as A, openBlock as a, createElementBlock as i, renderSlot as M, createCommentVNode as l, Fragment as m, createTextVNode as u, toDisplayString as t, createVNode as y, unref as L, withCtx as r, createBlock as o, renderList as P } from "vue";
import n from "./SchemaPropertyDetail.vue.js";
import N from "../../Badge/Badge.vue.js";
const S = { class: "property-heading" }, V = {
  key: 0,
  class: "property-name"
}, C = {
  key: 1,
  class: "property-additional"
}, $ = {
  key: 2,
  class: "property-deprecated"
}, b = {
  key: 3,
  class: "property-required"
}, D = {
  key: 4,
  class: "property-const"
}, j = {
  key: 5,
  class: "property-details"
}, E = {
  key: 6,
  class: "property-write-only"
}, F = {
  key: 7,
  class: "property-read-only"
}, K = /* @__PURE__ */ A({
  __name: "SchemaPropertyHeading",
  props: {
    value: {},
    enum: { type: Boolean },
    required: { type: Boolean, default: !1 },
    additional: { type: Boolean }
  },
  setup(H) {
    const O = ["oneOf", "anyOf", "allOf", "not"], w = (e) => Array.isArray(e == null ? void 0 : e.default) && e.default.length === 1 ? e.default[0] : e == null ? void 0 : e.default;
    return (e, T) => {
      var v, f, k, h, g, B, q;
      return a(), i("div", S, [
        e.$slots.name ? (a(), i("div", V, [
          M(e.$slots, "name", {}, void 0, !0)
        ])) : l("", !0),
        e.additional ? (a(), i("div", C, [
          (v = e.value) != null && v["x-additionalPropertiesName"] ? (a(), i(m, { key: 0 }, [
            u(t(e.value["x-additionalPropertiesName"]), 1)
          ], 64)) : (a(), i(m, { key: 1 }, [
            u("additional properties")
          ], 64))
        ])) : l("", !0),
        (f = e.value) != null && f.deprecated ? (a(), i("div", $, [
          y(L(N), null, {
            default: r(() => [
              u("deprecated")
            ]),
            _: 1
          })
        ])) : l("", !0),
        e.required ? (a(), i("div", b, " required ")) : l("", !0),
        (k = e.value) != null && k.const || (h = e.value) != null && h.enum && e.value.enum.length === 1 ? (a(), i("div", D, [
          y(n, { truncate: "" }, {
            prefix: r(() => [
              u("const:")
            ]),
            default: r(() => [
              u(" " + t(e.value.const ?? e.value.enum[0]), 1)
            ]),
            _: 1
          })
        ])) : (g = e.value) != null && g.type ? (a(), i("div", j, [
          y(n, null, {
            default: r(() => {
              var d, s, p;
              return [
                (s = (d = e.value) == null ? void 0 : d.items) != null && s.type ? (a(), i(m, { key: 0 }, [
                  u(t(e.value.type) + " " + t(e.value.items.type) + "[] ", 1)
                ], 64)) : (a(), i(m, { key: 1 }, [
                  u(t(Array.isArray(e.value.type) ? e.value.type.join(" | ") : e.value.type) + " " + t((p = e.value) != null && p.nullable ? " | nullable" : ""), 1)
                ], 64)),
                e.value.minItems || e.value.maxItems ? (a(), i(m, { key: 2 }, [
                  u(t(e.value.minItems) + "…" + t(e.value.maxItems), 1)
                ], 64)) : l("", !0)
              ];
            }),
            _: 1
          }),
          e.value.minLength ? (a(), o(n, { key: 0 }, {
            prefix: r(() => [
              u("min:")
            ]),
            default: r(() => [
              u(" " + t(e.value.minLength), 1)
            ]),
            _: 1
          })) : l("", !0),
          e.value.maxLength ? (a(), o(n, { key: 1 }, {
            prefix: r(() => [
              u("max:")
            ]),
            default: r(() => [
              u(" " + t(e.value.maxLength), 1)
            ]),
            _: 1
          })) : l("", !0),
          e.value.uniqueItems ? (a(), o(n, { key: 2 }, {
            default: r(() => [
              u(" unique! ")
            ]),
            _: 1
          })) : l("", !0),
          e.value.format ? (a(), o(n, { key: 3 }, {
            default: r(() => [
              u(t(e.value.format), 1)
            ]),
            _: 1
          })) : l("", !0),
          e.value.minimum !== void 0 && e.value.exclusiveMinimum ? (a(), o(n, { key: 4 }, {
            prefix: r(() => [
              u("greater than:")
            ]),
            default: r(() => [
              u(" " + t(e.value.minimum), 1)
            ]),
            _: 1
          })) : l("", !0),
          e.value.minimum !== void 0 && !e.value.exclusiveMinimum ? (a(), o(n, { key: 5 }, {
            prefix: r(() => [
              u("min:")
            ]),
            default: r(() => [
              u(" " + t(e.value.minimum), 1)
            ]),
            _: 1
          })) : l("", !0),
          e.value.maximum !== void 0 && e.value.exclusiveMaximum ? (a(), o(n, { key: 6 }, {
            prefix: r(() => [
              u("less than:")
            ]),
            default: r(() => [
              u(" " + t(e.value.maximum), 1)
            ]),
            _: 1
          })) : l("", !0),
          e.value.maximum !== void 0 && !e.value.exclusiveMaximum ? (a(), o(n, { key: 7 }, {
            prefix: r(() => [
              u("max:")
            ]),
            default: r(() => [
              u(" " + t(e.value.maximum), 1)
            ]),
            _: 1
          })) : l("", !0),
          e.value.pattern ? (a(), o(n, {
            key: 8,
            code: "",
            truncate: ""
          }, {
            default: r(() => [
              u(t(e.value.pattern), 1)
            ]),
            _: 1
          })) : l("", !0),
          e.$props.enum ? (a(), o(n, { key: 9 }, {
            default: r(() => [
              u("enum")
            ]),
            _: 1
          })) : l("", !0),
          e.value.default ? (a(), o(n, {
            key: 10,
            truncate: ""
          }, {
            prefix: r(() => [
              u("default:")
            ]),
            default: r(() => [
              u(" " + t(w(e.value)), 1)
            ]),
            _: 1
          })) : l("", !0)
        ])) : l("", !0),
        (B = e.value) != null && B.writeOnly ? (a(), i("div", E, " write-only ")) : (q = e.value) != null && q.readOnly ? (a(), i("div", F, " read-only ")) : l("", !0),
        (a(!0), i(m, null, P(O.filter((d) => {
          var s, p, I;
          return ((s = e.value) == null ? void 0 : s[d]) || ((I = (p = e.value) == null ? void 0 : p.items) == null ? void 0 : I[d]);
        }), (d) => (a(), o(L(N), { key: d }, {
          default: r(() => [
            u(t(d), 1)
          ]),
          _: 2
        }, 1024))), 128))
      ]);
    };
  }
});
export {
  K as default
};
