// @scalar/use-tooltip/dist/useTooltip.js
import f from "tippy.js";
import { shallowRef as u, ref as m, watchEffect as h } from "vue";
import { isMacOS as y } from "./isMacOS.js";
function S(r = {}) {
  const t = u(null), e = m(null);
  return h(() => {
    var o;
    (o = e.value) == null || o.destroy(), t.value && r.content && (e.value = f(t.value, {
      allowHTML: !0,
      theme: "app-tooltip",
      arrow: !1,
      delay: 400,
      duration: [100, 200],
      offset: [0, 5],
      placement: "top",
      ...r
    }));
  }), t;
}
function l(r, ...t) {
  let e = "";
  return r.forEach((o, i) => {
    e += o + (t[i] || "");
  }), e;
}
function x(r, ...t) {
  let e = "";
  return r.forEach((o, i) => {
    e += o + (t[i] || "");
  }), e;
}
function w(r, t) {
  const e = (n) => n.split("+").map((a) => a === "mod" ? y() ? "command" : "ctrl" : a).join("+"), i = ((n) => e(n).split("+").map((a) => {
    const s = {
      escape: "ESC",
      command: "⌘",
      shift: "⇧",
      ctrl: "⌃",
      alt: "⌥"
    };
    return a in s ? s[a] : a.charAt(0).toUpperCase() + a.slice(1);
  }))(r), c = x`
    border: 1px solid var(--background-2);
    padding: 2px;
    display: inline-block;
    background: rgba(255, 255, 255, 0.2);
    border-radius: 2px;
    min-width: 20px;
    text-align: center;
  `, p = (n) => l`<span style="${c}">${n}</span>`, d = t ? l`<span style="margin: 0 6px 0 3px">${t}</span>` : "";
  return l`
    <div style="display: flex; align-items: center">
      ${d}
      <div style="display: flex; gap: 3px">
        ${i.map((n) => p(n)).join("")}
      </div>
    </div>
  `;
}
export {
  w as keyboardShortcutTooltip,
  S as useTooltip
};
