// @scalar/api-client/dist/v2/blocks/request-block/components/RequestBody.vue2.js
import { defineComponent as F, computed as d, watch as q, createBlock as p, openBlock as u, unref as t, withCtx as i, createVNode as r, createElementVNode as c, toDisplayString as v, createElementBlock as y, Fragment as O, createTextVNode as T } from "vue";
import { ScalarListbox as S, ScalarButton as x, ScalarIcon as k } from "@scalar/components";
import { unpackProxyObject as B } from "@scalar/workspace-store/helpers/unpack-proxy";
import D from "./RequestBodyForm.vue.js";
import { getFileName as V } from "../helpers/files.js";
import { getExampleFromBody as U } from "../helpers/get-request-body-example.js";
import E from "../../../components/layout/CollapsibleSection.vue.js";
import L from "../../../components/data-table/DataTable.vue.js";
import M from "../../../components/data-table/DataTableHeader.vue.js";
import $ from "../../../components/data-table/DataTableRow.vue.js";
import z from "../../../components/code-input/CodeInput.vue.js";
import { useFileDialog as K } from "../../../../hooks/useFileDialog.js";
const R = {
  key: 0,
  class: "text-c-3 flex min-h-10 w-full items-center justify-center border-t p-2 text-sm"
}, I = {
  key: 1,
  class: "flex items-center justify-center overflow-hidden border-t p-1.5"
}, J = { class: "text-c-2 w-full max-w-full overflow-hidden rounded border px-1.5 py-1 text-xs whitespace-nowrap" }, ne = /* @__PURE__ */ F({
  __name: "RequestBody",
  props: {
    requestBody: {},
    exampleKey: {},
    title: {},
    environment: {}
  },
  emits: ["update:contentType", "update:value", "update:formValue"],
  setup(l, { emit: C }) {
    const s = C, N = {
      "application/json": "json",
      "application/xml": "xml",
      "application/yaml": "yaml"
    }, b = {
      "multipart/form-data": "Multipart Form",
      "application/x-www-form-urlencoded": "Form URL Encoded",
      "application/octet-stream": "Binary File",
      "application/json": "JSON",
      "application/xml": "XML",
      "application/yaml": "YAML",
      "application/edn": "EDN",
      other: "Other",
      none: "None"
    }, n = d(
      () => l.requestBody?.["x-scalar-selected-content-type"]?.[l.exampleKey] ?? Object.keys(l.requestBody?.content ?? {})[0] ?? "none"
    );
    q(
      () => l.requestBody?.["x-scalar-selected-content-type"]?.[l.exampleKey],
      (o) => {
        o || s("update:contentType", { value: n.value });
      },
      { immediate: !0 }
    );
    const f = Object.entries(b).map(([o, e]) => ({
      id: o,
      label: e
    })), g = d({
      get: () => f.find(
        (e) => e.id === n.value
      ) ?? f.at(-1),
      set: (o) => {
        s("update:contentType", { value: o.id });
      }
    });
    function h(o) {
      const { open: e } = K({
        onChange: (a) => {
          const w = a?.[0];
          w && o(w);
        },
        multiple: !1,
        accept: "*/*"
      });
      e();
    }
    const m = d(
      () => l.requestBody && U(l.requestBody, n.value, l.exampleKey)
    ), j = d(() => {
      if (!m.value)
        return "";
      const o = m.value.value;
      return typeof o == "string" ? o : JSON.stringify(o, null, 2);
    });
    return (o, e) => (u(), p(t(E), null, {
      title: i(() => [
        T(v(l.title), 1)
      ]),
      default: i(() => [
        r(t(L), {
          columns: [""],
          presentational: ""
        }, {
          default: i(() => [
            r(t(M), { class: "relative col-span-full flex h-8 cursor-pointer items-center justify-between border-r-0 !p-0" }, {
              default: i(() => [
                r(t(S), {
                  modelValue: g.value,
                  "onUpdate:modelValue": e[0] || (e[0] = (a) => g.value = a),
                  options: t(f),
                  teleport: ""
                }, {
                  default: i(() => [
                    r(t(x), {
                      class: "text-c-2 hover:text-c-1 flex h-full w-fit gap-1.5 px-3 font-normal",
                      fullWidth: "",
                      variant: "ghost"
                    }, {
                      default: i(() => [
                        c("span", null, v(b[n.value] ?? n.value), 1),
                        r(t(k), {
                          icon: "ChevronDown",
                          size: "md"
                        })
                      ]),
                      _: 1
                    })
                  ]),
                  _: 1
                }, 8, ["modelValue", "options"])
              ]),
              _: 1
            }),
            r(t($), null, {
              default: i(() => [
                n.value === "none" ? (u(), y("div", R, [...e[5] || (e[5] = [
                  c("span", null, "No Body", -1)
                ])])) : n.value === "application/octet-stream" ? (u(), y("div", I, [
                  t(V)(t(B)(m.value?.value)) !== void 0 ? (u(), y(O, { key: 0 }, [
                    c("span", J, v(t(V)(t(B)(m.value?.value))), 1),
                    r(t(x), {
                      class: "bg-b-2 hover:bg-b-3 text-c-2 ml-1 border-0 shadow-none",
                      size: "sm",
                      variant: "outlined",
                      onClick: e[1] || (e[1] = (a) => s("update:value", {
                        payload: void 0,
                        contentType: n.value
                      }))
                    }, {
                      default: i(() => [...e[6] || (e[6] = [
                        T(" Delete ", -1)
                      ])]),
                      _: 1
                    })
                  ], 64)) : (u(), p(t(x), {
                    key: 1,
                    class: "bg-b-2 hover:bg-b-3 text-c-2 border-0 shadow-none",
                    size: "sm",
                    variant: "outlined",
                    onClick: e[2] || (e[2] = () => h(
                      (a) => s("update:value", {
                        payload: a,
                        contentType: n.value
                      })
                    ))
                  }, {
                    default: i(() => [
                      e[7] || (e[7] = c("span", null, "Select File", -1)),
                      r(t(k), {
                        class: "ml-1",
                        icon: "Upload",
                        size: "xs",
                        thickness: "2.5"
                      })
                    ]),
                    _: 1
                  }))
                ])) : n.value === "multipart/form-data" || n.value === "application/x-www-form-urlencoded" ? (u(), p(D, {
                  key: 2,
                  environment: l.environment,
                  example: m.value,
                  selectedContentType: n.value,
                  "onUpdate:formValue": e[3] || (e[3] = (a) => s("update:formValue", {
                    payload: a,
                    contentType: n.value
                  }))
                }, null, 8, ["environment", "example", "selectedContentType"])) : (u(), p(t(z), {
                  key: 3,
                  class: "border-t px-3",
                  content: "",
                  environment: l.environment,
                  language: N[n.value] ?? "plaintext",
                  lineNumbers: "",
                  lint: "",
                  modelValue: j.value,
                  "onUpdate:modelValue": e[4] || (e[4] = (a) => s("update:value", {
                    payload: a,
                    contentType: n.value
                  }))
                }, null, 8, ["environment", "language", "modelValue"]))
              ]),
              _: 1
            })
          ]),
          _: 1
        })
      ]),
      _: 1
    }));
  }
});
export {
  ne as default
};
