// @scalar/api-client/dist/v2/blocks/request-block/components/RequestTableRow.vue.js
import { defineComponent as q, ref as x, watch as y, computed as o, createBlock as m, openBlock as d, unref as t, normalizeClass as E, withCtx as s, createVNode as u, createCommentVNode as f, createElementBlock as R, Fragment as N, createElementVNode as h, toDisplayString as P } from "vue";
import { ScalarButton as C, ScalarIconButton as j, ScalarIcon as G } from "@scalar/components";
import { ScalarIconTrash as K, ScalarIconGlobe as O } from "@scalar/icons";
import { unpackProxyObject as B } from "@scalar/workspace-store/helpers/unpack-proxy";
import { getFileName as W } from "../helpers/files.js";
import { validateParameter as A } from "../helpers/validate-parameter.js";
import D from "./RequestTableTooltip.vue.js";
import H from "../../../components/data-table/DataTableRow.vue.js";
import J from "../../../components/data-table/DataTableCheckbox.vue.js";
import g from "../../../components/data-table/DataTableCell.vue.js";
import F from "../../../components/code-input/CodeInput.vue.js";
const L = { class: "text-c-2 filemask flex w-full max-w-[100%] items-center justify-center overflow-hidden p-1" }, M = {
  key: 1,
  class: "p-0.5"
}, ne = /* @__PURE__ */ q({
  __name: "RequestTableRow",
  props: {
    data: {},
    hasCheckboxDisabled: { type: Boolean },
    invalidParams: {},
    label: {},
    environment: {},
    showUploadButton: { type: Boolean }
  },
  emits: ["upsertRow", "deleteRow", "uploadFile", "removeFile", "navigate"],
  setup(e, { emit: S }) {
    const r = S, c = x(e.data.name ?? ""), n = x(B(e.data.value) ?? ""), v = x(e.data.isDisabled ?? !1);
    y(
      () => e.data.name,
      (i) => c.value = i ?? ""
    ), y(
      () => e.data.value,
      (i) => n.value = B(i) ?? ""
    ), y(
      () => e.data.isDisabled,
      (i) => v.value = i ?? !1
    );
    const k = o(() => n.value instanceof File), w = o(
      () => (k.value ? W(n.value) : n.value) ?? ""
    ), z = o(() => e.data.schema?.default), I = o(() => e.data.schema?.enum), T = o(
      () => e.data.schema && "minimum" in e.data.schema ? e.data.schema.minimum : void 0
    ), $ = o(
      () => e.data.schema && "maximum" in e.data.schema ? e.data.schema.maximum : void 0
    ), U = o(
      () => e.data.schema && "type" in e.data.schema ? e.data.schema.type : void 0
    ), V = o(
      () => A(e.data.schema, n.value)
    ), b = (i) => {
      i.name !== void 0 && (c.value = i.name), i.value !== void 0 && (n.value = i.value), v.value = i.isDisabled ?? !1, r("upsertRow", {
        name: c.value,
        value: n.value,
        isDisabled: v.value
      });
    };
    return (i, a) => (d(), m(t(H), {
      id: e.data.name,
      class: E({
        alert: V.value.ok === !1,
        error: V.value.ok === !1 && e.invalidParams?.has(e.data.name)
      })
    }, {
      default: s(() => [
        u(t(J), {
          class: "!border-r",
          disabled: e.hasCheckboxDisabled ?? !1,
          modelValue: !v.value,
          "onUpdate:modelValue": a[0] || (a[0] = (l) => b({ isDisabled: !l }))
        }, null, 8, ["disabled", "modelValue"]),
        u(t(g), null, {
          default: s(() => [
            u(t(F), {
              "aria-label": `${e.label} Key`,
              disableCloseBrackets: "",
              disabled: e.data.isReadonly,
              disableEnter: "",
              disableTabIndent: "",
              environment: e.environment,
              lineWrapping: "",
              modelValue: c.value,
              placeholder: "Key",
              required: !!e.data.isRequired,
              onSelectVariable: a[1] || (a[1] = (l) => b({ name: l })),
              "onUpdate:modelValue": a[2] || (a[2] = (l) => b({ name: l }))
            }, null, 8, ["aria-label", "disabled", "environment", "modelValue", "required"])
          ]),
          _: 1
        }),
        u(t(g), null, {
          default: s(() => [
            u(t(F), {
              "aria-label": `${e.label} Value`,
              class: "pr-6 group-hover:pr-10 group-has-[.cm-focused]:pr-10",
              default: z.value,
              disableCloseBrackets: "",
              disabled: e.data.isReadonly,
              disableEnter: "",
              disableTabIndent: "",
              enum: I.value ?? [],
              environment: e.environment,
              examples: e.data.schema?.examples?.map((l) => String(l)) ?? [],
              linethrough: e.data.isOverridden,
              lineWrapping: "",
              max: $.value,
              min: T.value,
              modelValue: w.value,
              placeholder: "Value",
              type: U.value,
              "onUpdate:modelValue": a[5] || (a[5] = (l) => b({ value: l }))
            }, {
              icon: s(() => [
                (e.data.name || n.value) && !e.data.isRequired && e.data.isReadonly !== !0 ? (d(), m(t(C), {
                  key: 0,
                  class: "text-c-2 hover:text-c-1 hover:bg-b-2 z-context -mr-0.5 hidden h-fit rounded p-1 group-hover:flex group-has-[.cm-focused]:flex",
                  size: "sm",
                  variant: "ghost",
                  onClick: a[3] || (a[3] = (l) => r("deleteRow"))
                }, {
                  default: s(() => [
                    u(t(K), { class: "size-3.5" })
                  ]),
                  _: 1
                })) : f("", !0),
                e.data.globalRoute !== void 0 ? (d(), m(t(j), {
                  key: 1,
                  class: "text-c-2 hover:text-c-1 hover:bg-b-2 z-context -mr-0.5 h-fit",
                  icon: t(O),
                  label: "Global cookies are shared across the whole workspace. Click to navigate.",
                  size: "xs",
                  tooltip: "top",
                  variant: "ghost",
                  onClick: a[4] || (a[4] = (l) => r("navigate", e.data.globalRoute))
                }, null, 8, ["icon"])) : f("", !0),
                e.data.isReadonly ? (d(), m(D, {
                  key: 2,
                  description: "This is a readonly property and you can not modify it! If you want to change it you have to override it or disable it using the checkbox",
                  value: null
                })) : e.data.schema ? (d(), m(D, {
                  key: 3,
                  description: e.data.description,
                  schema: e.data.schema,
                  value: n.value
                }, null, 8, ["description", "schema", "value"])) : f("", !0)
              ]),
              _: 1
            }, 8, ["aria-label", "default", "disabled", "enum", "environment", "examples", "linethrough", "max", "min", "modelValue", "type"])
          ]),
          _: 1
        }),
        e.showUploadButton ? (d(), m(t(g), {
          key: 0,
          class: "group/upload flex items-center justify-center whitespace-nowrap"
        }, {
          default: s(() => [
            k.value ? (d(), R(N, { key: 0 }, [
              h("div", L, [
                h("span", null, P(w.value), 1)
              ]),
              h("button", {
                class: "bg-b-2 centered-x centered-y absolute hidden w-[calc(100%_-_8px)] rounded p-0.5 text-center text-xs font-medium group-hover/upload:block",
                type: "button",
                onClick: a[6] || (a[6] = (l) => r("removeFile"))
              }, " Delete ")
            ], 64)) : (d(), R("div", M, [
              u(t(C), {
                class: "bg-b-2 hover:bg-b-3 text-c-2 h-fit border-0 py-px shadow-none",
                size: "sm",
                variant: "outlined",
                onClick: a[7] || (a[7] = (l) => r("uploadFile"))
              }, {
                default: s(() => [
                  a[8] || (a[8] = h("span", null, "Select File", -1)),
                  u(t(G), {
                    class: "ml-1",
                    icon: "Upload",
                    size: "xs",
                    thickness: "2.5"
                  })
                ]),
                _: 1
              })
            ]))
          ]),
          _: 1
        })) : f("", !0)
      ]),
      _: 1
    }, 8, ["id", "class"]));
  }
});
export {
  ne as default
};
