// @scalar/api-client/dist/router.js
import { createRouter as t, createWebHistory as p, createWebHashHistory as n, createMemoryHistory as l } from "vue-router";
var i = /* @__PURE__ */ ((e) => (e.Request = "request", e.Examples = "examples", e.Cookies = "cookies", e.Collection = "collection", e.Schema = "schema", e.Environment = "environment", e.Servers = "servers", e.Workspace = "workspace", e))(i || {});
const s = [
  {
    path: "",
    redirect: (e) => `${e.fullPath.replace(/\/$/, "")}/request/default`
  },
  {
    path: "request",
    redirect: (e) => `${e.fullPath.replace(/\/$/, "")}/default`
  },
  {
    name: "request",
    path: "request/:request",
    component: () => import("./views/Request/Request.vue.js")
  },
  {
    name: "examples",
    path: "request/:request/examples/:examples",
    component: () => import("./views/Request/Request.vue.js")
  }
], m = [
  {
    path: "/",
    redirect: "/workspace/default/request/default"
  },
  {
    path: "/workspace",
    redirect: "/workspace/default/request/default"
  },
  {
    path: "/workspace/:workspace",
    children: s
  }
], c = [
  {
    path: "/",
    redirect: "/workspace/default/request/default"
  },
  {
    path: "/workspace",
    redirect: "/workspace/default/request/default"
  },
  {
    path: "/workspace/:workspace",
    children: [
      ...s,
      // {
      //   path: 'collection',
      //   redirect: (to) =>
      //     `${to.fullPath.replace(/\/$/, '')}/default`,
      // },
      // {
      //   name: PathId.Collection,
      //   path: `collection/:${PathId.Collection}`,
      //   component: () => import('./views/Collection/Collection.vue'),
      //   children: [
      //     // Nested collection request
      //     {
      //       path: `request/${PathId.Request}`,
      //       component: () => import('./views/Request/Request.vue.js'),
      //     },
      //   ],
      // },
      /** Components will map to each section of the spec components object */
      // {
      //   path: 'components',
      //   redirect: '/components/schemas/default',
      //   children: [
      //     {
      //       path: `schemas/:${PathId.Schema}`,
      //       component: () => import('./views/Components/Schemas/Schemas.vue'),
      //     },
      //   ],
      // },
      {
        path: "environment",
        redirect: (e) => `${e.fullPath.replace(/\/$/, "")}/default`
      },
      {
        name: "environment",
        path: "environment/:environment",
        component: () => import("./views/Environment/Environment.vue.js")
      },
      {
        path: "cookies",
        redirect: (e) => `${e.fullPath.replace(/\/$/, "")}/default`
      },
      {
        name: "cookies",
        path: "cookies/:cookies",
        component: () => import("./views/Cookies/Cookies.vue.js")
      },
      {
        path: "servers",
        redirect: (e) => `${e.fullPath.replace(/\/$/, "")}/default`
      },
      {
        name: "servers",
        path: "servers/:servers",
        component: () => import("./views/Servers/Servers.vue.js")
      }
    ]
  }
], r = t({
  history: p(),
  routes: c
}), f = () => t({
  history: n(),
  routes: c
}), d = () => t({
  history: l(),
  routes: m
});
function v(e, u) {
  var a, o;
  r.currentRoute.value && // If the item is missing then we know the UID is no longer in use and redirect to the default
  !u && ((a = r.currentRoute.value) != null && a.params[e]) && ((o = r.currentRoute.value) == null ? void 0 : o.params[e]) !== "default" && // We only redirect if the key is missing for the matching route
  r.currentRoute.value.path.includes(e) && r.push({
    params: {
      ...r.currentRoute.value.params,
      [e]: "default"
    }
  });
}
export {
  i as PathId,
  d as createModalRouter,
  f as createWebHashRouter,
  v as fallbackMissingParams,
  m as modalRoutes,
  r as router
};
