// @scalar/api-client/dist/components/HttpMethod/HttpMethod.vue2.js
import { defineComponent as S, computed as s, openBlock as p, createBlock as g, unref as t, withCtx as y, createElementVNode as l, normalizeClass as r, toDisplayString as c, createElementBlock as q } from "vue";
import { ScalarListbox as w } from "@scalar/components";
import { getRequest as k, REQUEST_METHODS as x } from "@scalar/oas-utils/helpers";
import { cva as B, cx as m } from "cva";
const O = /* @__PURE__ */ S({
  __name: "HttpMethod",
  props: {
    isSquare: { type: Boolean, default: !1 },
    method: {},
    isEditable: { type: Boolean, default: !1 }
  },
  emits: ["change"],
  setup(f, { emit: b }) {
    const i = f, h = b, o = s(() => k(i.method)), n = Object.entries(x).map(
      ([e, { short: a }]) => ({
        id: e,
        label: e.charAt(0) + e.toLowerCase().slice(1)
      })
    ), d = s({
      get: () => n.find(({ id: e }) => e === i.method),
      set: (e) => (e == null ? void 0 : e.id) && h("change", e.id)
    }), u = B({
      base: "text-center font-code text-3xs justify-center items-center flex",
      variants: {
        isSquare: {
          true: "px-2.5 whitespace-nowrap font-bold border-r h-fit m-auto",
          false: "rounded-full"
        },
        isEditable: {
          true: "px-0 http-bg-gradient rounded-md border-1/2 border-r-1/2",
          false: "cusor-pointer"
        }
      }
    }), v = s(() => o.value.short);
    return (e, a) => e.isEditable ? (p(), g(t(w), {
      key: 0,
      modelValue: d.value,
      "onUpdate:modelValue": a[0] || (a[0] = (E) => d.value = E),
      options: t(n)
    }, {
      default: y(() => [
        l("div", {
          class: r(["h-full", { "pointer-events-none": !e.isEditable }])
        }, [
          l("button", {
            class: r(["relative h-full", t(m)(t(u)({ isSquare: e.isSquare, isEditable: e.isEditable }), o.value.color)]),
            type: "button"
          }, [
            l("span", null, c(v.value), 1)
          ], 2)
        ], 2)
      ]),
      _: 1
    }, 8, ["modelValue", "options"])) : (p(), q("div", {
      key: 1,
      class: r(["relative gap-1 whitespace-nowrap", t(m)(t(u)({ isSquare: e.isSquare, isEditable: e.isEditable }), o.value.color)]),
      type: "button"
    }, c(o.value.short), 3));
  }
});
export {
  O as default
};
