// @scalar/agent-chat/dist/hooks/use-upload-tmp-document.js
import { ref as d } from "vue";
import { z as o } from "zod/mini";
import { useState as g } from "../state/state.js";
const u = "scalar-tmp-doc";
function f({ namespace: t, slug: e }) {
  localStorage.setItem(u, JSON.stringify({ namespace: t, slug: e }));
}
function h() {
  const t = localStorage.getItem(u);
  if (t)
    return o.object({
      namespace: o.string(),
      slug: o.string()
    }).parse(JSON.parse(t));
}
function T() {
  const t = g(), e = d();
  function c(n) {
    const a = `${t.baseUrl}${n}`;
    if (a.startsWith("/")) return a;
    const s = new URLSearchParams({ scalar_url: a.toString() });
    return new URL(`https://proxy.scalar.com/?${s}`);
  }
  async function l(n, a = !1) {
    try {
      e.value = { type: "uploading" };
      const s = await fetch(c(`/core/share/upload/apis${a ? "?source=agent" : ""}`), {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({ document: n })
      });
      if (!s.ok) {
        e.value = {
          type: "error",
          error: "Failed to upload document."
        };
        return;
      }
      const p = await s.json(), { success: i, data: r } = o.object({ url: o.string(), namespace: o.string(), slug: o.string() }).safeParse(p);
      if (!i) {
        e.value = {
          type: "error",
          error: "Failed to process document."
        };
        return;
      }
      e.value = {
        type: "processing"
      };
      const m = await fetch(
        c(`/vector/registry/embeddings/${r.namespace}/${r.slug}`),
        {
          method: "GET"
        }
      );
      if (f({
        namespace: r.namespace,
        slug: r.slug
      }), await t.addDocument({
        namespace: r.namespace,
        slug: r.slug,
        removable: !1
      }), !m.ok) {
        e.value = {
          type: "error",
          error: "Failed to embed document."
        };
        return;
      }
      return e.value = { type: "done" }, t.uploadedTmpDocumentUrl.value = r.url, setTimeout(() => {
        e.value = void 0;
      }, 2e3), r;
    } catch {
      e.value = { type: "error", error: "Failed to upload document." };
      return;
    }
  }
  return {
    uploadTempDocument: l,
    uploadState: e
  };
}
export {
  h as getTmpDocFromLocalStorage,
  T as useUploadTmpDocument
};
