// @scalar/api-reference/dist/features/BaseUrl/BaseUrl.vue.js
import { defineComponent as b, ref as S, watch as c, openBlock as U, createBlock as g, unref as u } from "vue";
import j from "./ServerForm.vue.js";
import { getServers as E } from "./utils/getServers.js";
import { createEmptySpecification as O } from "../../helpers/createEmptySpecification.js";
import { useServerStore as _ } from "../../legacy/stores/useServerStore.js";
const w = /* @__PURE__ */ b({
  __name: "BaseUrl",
  props: {
    specification: {},
    defaultServerUrl: {},
    servers: {}
  },
  setup(f) {
    const a = f, { server: i, setServer: o } = _(), s = S(0);
    c(
      s,
      () => o({
        selectedServer: s.value
      }),
      {
        immediate: !0
      }
    ), c(
      () => a.specification,
      () => {
        var r;
        const t = (
          // Use the specification
          a.servers === void 0 ? a.specification : (
            // Or create an empty one with the specified servers list
            O({
              servers: a.servers
            })
          )
        ), e = E(t, {
          defaultServerUrl: a.defaultServerUrl
        });
        o({
          servers: e,
          variables: {
            // Set the initial values for the variables
            ...m(
              ((r = e[s.value]) == null ? void 0 : r.variables) ?? {}
            ),
            // Don’t overwrite existing values, but filter out non-existing variables
            ...p(
              i.variables,
              e[s.value]
            )
          }
        });
      },
      { deep: !0, immediate: !0 }
    );
    function m(t) {
      return Object.fromEntries(
        Object.entries(t ?? {}).map(([e, r]) => {
          var n, l, v;
          return [
            e,
            // 1) Default
            ((n = r.default) == null ? void 0 : n.toString()) ?? // 2) First enum value
            ((v = (l = r.enum) == null ? void 0 : l[0]) == null ? void 0 : v.toString()) ?? // 3) Empty string
            ""
          ];
        })
      );
    }
    function p(t, e) {
      return Object.fromEntries(
        Object.entries(t).filter(
          ([r]) => r in (e.variables ?? {})
        )
      );
    }
    function d(t, e) {
      o({
        variables: {
          ...i.variables,
          [t]: e
        }
      });
    }
    return (t, e) => (U(), g(j, {
      selected: s.value,
      servers: u(i).servers,
      variables: u(i).variables,
      "onUpdate:selected": e[0] || (e[0] = (r) => {
        s.value = r;
      }),
      "onUpdate:variable": d
    }, null, 8, ["selected", "servers", "variables"]));
  }
});
export {
  w as default
};
