// @scalar/api-reference/dist/features/BaseUrl/ServerForm.vue2.js
import { defineComponent as S, toRef as I, computed as k, openBlock as c, createElementBlock as u, createElementVNode as t, createVNode as o, withCtx as w, unref as C, createCommentVNode as p, pushScopeId as N, popScopeId as U } from "vue";
import { ScalarMarkdown as V } from "@scalar/components";
import B from "./ServerUrl.vue.js";
import R from "./ServerUrlSelect.vue.js";
import g from "./ServerVariablesForm.vue.js";
const y = (r) => (N("data-v-da7210d5"), r = r(), U(), r), E = { key: 0 }, F = /* @__PURE__ */ y(() => /* @__PURE__ */ t("span", { class: "server-form-title" }, "Base URL", -1)), L = { class: "server-form" }, M = { class: "server-form-container" }, $ = { class: "server-item" }, j = {
  key: 0,
  muted: ""
}, q = { class: "description" }, H = /* @__PURE__ */ S({
  __name: "ServerForm",
  props: {
    selected: { default: 0 },
    servers: {},
    variables: {}
  },
  emits: ["update:selected", "update:variable"],
  setup(r, { emit: m }) {
    const i = r, f = m, l = I(Number(i.selected)), _ = (e) => {
      const s = parseInt(e, 10);
      f("update:selected", s), l.value = s;
    }, a = k(() => {
      var e;
      return (e = i.servers) == null ? void 0 : e[l.value];
    });
    return (e, s) => {
      var d, v, n;
      return (d = e.servers) != null && d.length ? (c(), u("div", E, [
        F,
        t("div", L, [
          t("div", M, [
            t("div", $, [
              o(R, {
                options: e.servers,
                value: l.value,
                onChange: _
              }, {
                default: w(() => [
                  o(B, {
                    server: a.value,
                    variables: e.variables
                  }, null, 8, ["server", "variables"])
                ]),
                _: 1
              }, 8, ["options", "value"])
            ]),
            o(g, {
              values: e.variables,
              variables: (v = a.value) == null ? void 0 : v.variables,
              "onUpdate:variable": s[0] || (s[0] = (b, h) => e.$emit("update:variable", b, h))
            }, null, 8, ["values", "variables"])
          ])
        ]),
        (n = a.value) != null && n.description ? (c(), u("div", j, [
          t("div", q, [
            o(C(V), {
              value: a.value.description
            }, null, 8, ["value"])
          ])
        ])) : p("", !0)
      ])) : p("", !0);
    };
  }
});
export {
  H as default
};
