// @scalar/api-reference/dist/features/BaseUrl/ServerUrlSelect.vue2.js
import { defineComponent as v, computed as a, openBlock as r, createElementBlock as f, createVNode as c, unref as l, withCtx as i, normalizeClass as h, createElementVNode as _, renderSlot as g, createBlock as S, createCommentVNode as V } from "vue";
import { ScalarListbox as C, ScalarButton as k, ScalarIcon as x } from "@scalar/components";
const B = { class: "w-full" }, w = { class: "custom-scroll" }, b = /* @__PURE__ */ v({
  __name: "ServerUrlSelect",
  props: {
    options: {},
    value: {}
  },
  emits: ["change"],
  setup(u, { emit: d }) {
    const n = u, m = d, o = a(
      () => n.options.map((e, t) => ({
        id: t.toString(),
        label: e.url ?? ""
      }))
    ), s = a({
      get: () => {
        var e;
        return (e = o.value) == null ? void 0 : e.find((t) => t.id === n.value.toString());
      },
      set: (e) => m("change", (e == null ? void 0 : e.id) ?? "")
    });
    return (e, t) => (r(), f("div", B, [
      c(l(C), {
        modelValue: s.value,
        "onUpdate:modelValue": t[0] || (t[0] = (p) => s.value = p),
        options: o.value,
        resize: ""
      }, {
        default: i(() => [
          c(l(k), {
            class: h(["url-select", { "pointer-events-none": o.value.length <= 1 }]),
            fullWidth: "",
            variant: "ghost"
          }, {
            default: i(() => [
              _("span", w, [
                g(e.$slots, "default", {}, void 0, !0)
              ]),
              o.value.length > 1 ? (r(), S(l(x), {
                key: 0,
                icon: "ChevronDown",
                size: "xs"
              })) : V("", !0)
            ]),
            _: 3
          }, 8, ["class"])
        ]),
        _: 3
      }, 8, ["modelValue", "options"])
    ]));
  }
});
export {
  b as default
};
