// @scalar/api-reference/dist/features/BaseUrl/ServerVariablesForm.vue2.js
import { defineComponent as h, openBlock as t, createElementBlock as o, Fragment as k, renderList as y, createElementVNode as m, toDisplayString as V, createBlock as p, createCommentVNode as C } from "vue";
import S from "./ServerVariablesSelect.vue.js";
import _ from "./ServerVariablesTextbox.vue.js";
const B = {
  key: 0,
  class: "variable-container"
}, j = ["for"], O = /* @__PURE__ */ h({
  __name: "ServerVariablesForm",
  props: {
    variables: {},
    values: {}
  },
  emits: ["update:variable"],
  setup(d, { emit: f }) {
    const i = d, g = f;
    function n(a, r) {
      g("update:variable", a, r);
    }
    const s = (a) => {
      var r, e;
      return (((r = i.values) == null ? void 0 : r[a]) ?? ((e = i.variables) == null ? void 0 : e[a].default) ?? "").toString();
    };
    return (a, r) => a.variables && Object.keys(a.variables ?? {}).length ? (t(), o("div", B, [
      (t(!0), o(k, null, y(Object.keys(a.variables), (e) => {
        var u, v, b, c;
        return t(), o("div", {
          key: e,
          class: "variable-container-item"
        }, [
          m("label", {
            class: "variable-label",
            for: `variable-${e}`
          }, [
            m("code", null, V(e), 1)
          ], 8, j),
          (v = (u = a.variables) == null ? void 0 : u[e].enum) != null && v.length ? (t(), p(S, {
            key: 0,
            enum: ((c = (b = a.variables[e]) == null ? void 0 : b.enum) == null ? void 0 : c.map((l) => `${l}`)) ?? [],
            value: s(e),
            onChange: (l) => n(e, l)
          }, null, 8, ["enum", "value", "onChange"])) : (t(), p(_, {
            key: 1,
            value: s(e),
            onChange: (l) => n(e, l)
          }, null, 8, ["value", "onChange"]))
        ]);
      }), 128))
    ])) : C("", !0);
  }
});
export {
  O as default
};
