// @scalar/api-reference/dist/features/BaseUrl/ServerVariablesSelect.vue2.js
import { defineComponent as p, computed as i, openBlock as v, createElementBlock as f, createVNode as t, unref as a, withCtx as r, createElementVNode as _, toDisplayString as h } from "vue";
import { ScalarListbox as S, ScalarButton as V, ScalarIcon as g } from "@scalar/components";
const b = { class: "w-full" }, B = /* @__PURE__ */ p({
  __name: "ServerVariablesSelect",
  props: {
    enum: {},
    value: {}
  },
  emits: ["change"],
  setup(u, { emit: c }) {
    const l = u, d = c, o = i(
      () => l.enum.map((e) => ({ id: e, label: e }))
    ), n = i({
      get: () => o.value.find((e) => e.id === l.value),
      set: (e) => d("change", (e == null ? void 0 : e.id) ?? "")
    });
    return (e, s) => (v(), f("div", b, [
      t(a(S), {
        modelValue: n.value,
        "onUpdate:modelValue": s[0] || (s[0] = (m) => n.value = m),
        options: o.value
      }, {
        default: r(() => [
          t(a(V), {
            class: "variable-select",
            fullWidth: "",
            variant: "ghost"
          }, {
            default: r(() => [
              _("span", null, h(e.value), 1),
              t(a(g), {
                icon: "ChevronDown",
                size: "xs"
              })
            ]),
            _: 1
          })
        ]),
        _: 1
      }, 8, ["modelValue", "options"])
    ]));
  }
});
export {
  B as default
};
