// @scalar/api-reference/dist/components/Content/Introduction/Description.vue2.js
import { defineComponent as y, computed as S, openBlock as o, createElementBlock as a, Fragment as u, renderList as C, createBlock as c, withCtx as b, createVNode as B, unref as d, createCommentVNode as H } from "vue";
import { splitContent as I, getHeadings as N } from "@scalar/code-highlight/markdown";
import { ScalarMarkdown as m } from "@scalar/components";
import p from "github-slugger";
import x from "../../IntersectionObserver.vue.js";
import { useNavState as E } from "../../../hooks/useNavState.js";
import { joinWithSlash as L } from "../../../helpers/joinWithSlash.js";
const P = {
  key: 0,
  class: "introduction-description"
}, T = /* @__PURE__ */ y({
  __name: "Description",
  props: {
    value: {}
  },
  setup(h) {
    const i = h, f = S(() => {
      if (!i.value)
        return [];
      const e = new p();
      return I(i.value).map((t) => {
        const [r] = N(t);
        return {
          id: r ? l({
            ...r,
            slug: e.slug(r.value)
          }) : void 0,
          content: t
        };
      });
    }), { getHeadingId: l, hash: g, isIntersectionEnabled: v, pathRouting: s } = E();
    function w(e = "") {
      if (!v.value) return;
      const n = new URL(window.location.href);
      s.value ? n.pathname = L(s.value.basePath, e) : n.hash = e, g.value = e, window.history.replaceState({}, "", n);
    }
    const _ = new p(), k = (e) => (e.data = {
      hProperties: {
        id: l({
          depth: e.depth,
          value: e.children[0].value,
          slug: _.slug(e.children[0].value)
        })
      }
    }, e);
    return (e, n) => e.value ? (o(), a("div", P, [
      (o(!0), a(u, null, C(f.value, (t, r) => (o(), a(u, { key: r }, [
        t.id ? (o(), c(x, {
          key: 0,
          id: t.id,
          class: "introduction-description-heading",
          onIntersecting: () => w(t.id)
        }, {
          default: b(() => [
            B(d(m), {
              transform: k,
              transformType: "heading",
              value: t.content
            }, null, 8, ["value"])
          ]),
          _: 2
        }, 1032, ["id", "onIntersecting"])) : (o(), c(d(m), {
          key: 1,
          value: t.content,
          withImages: ""
        }, null, 8, ["value"]))
      ], 64))), 128))
    ])) : H("", !0);
  }
});
export {
  T as default
};
