// @scalar/api-reference/dist/components/Content/Introduction/Introduction.vue2.js
import { defineComponent as C, computed as p, openBlock as n, createBlock as r, unref as o, withCtx as t, createVNode as i, createElementVNode as f, createTextVNode as s, toDisplayString as a, createCommentVNode as d, renderSlot as c } from "vue";
import h from "github-slugger";
import k from "./Description.vue.js";
import y from "./DownloadSpec.vue.js";
import V from "../../Section/SectionContainer.vue.js";
import w from "../../Section/Section.vue.js";
import N from "../../Section/SectionContent.vue.js";
import T from "../../Section/SectionColumns.vue.js";
import m from "../../Section/SectionColumn.vue.js";
import g from "../../Badge/Badge.vue.js";
import B from "../../Section/SectionHeader.vue.js";
const D = { class: "badges" }, $ = { class: "sticky-cards" }, K = /* @__PURE__ */ C({
  __name: "Introduction",
  props: {
    info: {},
    parsedSpec: {}
  },
  setup(S) {
    const l = S, v = new h(), u = p(() => l.parsedSpec.openapi ?? l.parsedSpec.swagger ?? ""), _ = p(() => v.slug(l.info.title ?? ""));
    return (e, b) => (n(), r(o(V), null, {
      default: t(() => [
        i(o(w), { class: "introduction-section" }, {
          default: t(() => [
            i(o(N), {
              loading: !e.info.description && !e.info.title
            }, {
              default: t(() => [
                i(o(T), null, {
                  default: t(() => [
                    i(o(m), null, {
                      default: t(() => [
                        f("div", D, [
                          e.info.version ? (n(), r(o(g), { key: 0 }, {
                            default: t(() => [
                              s(a(e.info.version), 1)
                            ]),
                            _: 1
                          })) : d("", !0),
                          u.value ? (n(), r(o(g), { key: 1 }, {
                            default: t(() => [
                              s(" OAS " + a(u.value), 1)
                            ]),
                            _: 1
                          })) : d("", !0)
                        ]),
                        i(o(B), {
                          level: 1,
                          loading: !e.info.title,
                          tight: ""
                        }, {
                          default: t(() => [
                            s(a(e.info.title), 1)
                          ]),
                          _: 1
                        }, 8, ["loading"]),
                        i(y, { specTitle: _.value }, null, 8, ["specTitle"]),
                        i(k, {
                          value: e.info.description
                        }, null, 8, ["value"])
                      ]),
                      _: 1
                    }),
                    e.$slots.aside ? (n(), r(o(m), { key: 0 }, {
                      default: t(() => [
                        f("div", $, [
                          c(e.$slots, "aside", {}, void 0, !0)
                        ])
                      ]),
                      _: 3
                    })) : d("", !0)
                  ]),
                  _: 3
                })
              ]),
              _: 3
            }, 8, ["loading"]),
            c(e.$slots, "after", {}, void 0, !0)
          ]),
          _: 3
        })
      ]),
      _: 3
    }));
  }
});
export {
  K as default
};
