// @scalar/api-reference/dist/components/Content/Models/components/ModernLayout.vue.js
import { defineComponent, createBlock, openBlock, unref, withCtx, createVNode } from "vue";
import { ScalarErrorBoundary } from "@scalar/components";
import CompactSection from "../../../Section/CompactSection.vue.js";
import Schema from "../../Schema/Schema.vue.js";
import SectionHeaderTag from "../../../Section/SectionHeaderTag.vue.js";
import SchemaHeading from "../../Schema/SchemaHeading.vue.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "ModernLayout",
  props: {
    id: {},
    name: {},
    schema: {},
    isCollapsed: { type: Boolean },
    eventBus: {},
    options: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createBlock(unref(CompactSection), {
        id: __props.id,
        key: __props.name,
        label: __props.name,
        modelValue: !__props.isCollapsed,
        onCopyAnchorUrl: _cache[0] || (_cache[0] = () => __props.eventBus?.emit("copy-url:nav-item", { id: __props.id })),
        "onUpdate:modelValue": _cache[1] || (_cache[1] = (value) => __props.eventBus?.emit("toggle:nav-item", { id: __props.id, open: value }))
      }, {
        heading: withCtx(() => [
          createVNode(unref(SectionHeaderTag), { level: 3 }, {
            default: withCtx(() => [
              createVNode(unref(SchemaHeading), {
                name: __props.schema.title ?? __props.name,
                value: __props.schema
              }, null, 8, ["name", "value"])
            ]),
            _: 1
          })
        ]),
        default: withCtx(() => [
          createVNode(unref(ScalarErrorBoundary), null, {
            default: withCtx(() => [
              createVNode(unref(Schema), {
                eventBus: __props.eventBus,
                hideHeading: "",
                hideModelNames: "",
                level: 1,
                noncollapsible: "",
                options: __props.options,
                schema: __props.schema
              }, null, 8, ["eventBus", "options", "schema"])
            ]),
            _: 1
          })
        ]),
        _: 1
      }, 8, ["id", "label", "modelValue"]);
    };
  }
});
export {
  _sfc_main as default
};
