// @scalar/sidebar/dist/components/ScalarSidebar.vue.js
import { defineComponent as p, useSlots as S, createBlock as f, openBlock as s, unref as i, normalizeStyle as b, withCtx as a, renderSlot as l, createVNode as g, createElementVNode as D, createElementBlock as E, Fragment as $, renderList as k, createSlots as I, mergeProps as d } from "vue";
import { ScalarSidebar as v, ScalarSidebarItems as x } from "@scalar/components";
import { filterItems as F } from "../helpers/filter-items.js";
import B from "./SidebarItem.vue.js";
const w = /* @__PURE__ */ p({
  __name: "ScalarSidebar",
  props: {
    layout: {},
    items: {},
    isSelected: { type: Function },
    isExpanded: { type: Function },
    options: {},
    indent: { default: 20 },
    isDraggable: {},
    isDroppable: { type: Function }
  },
  emits: ["reorder", "selectItem"],
  setup(e, { emit: u }) {
    const m = u, n = S(), y = (t, r) => {
      m("reorder", t, r);
    };
    return (t, r) => (s(), f(i(v), {
      class: "flex min-h-0 flex-col",
      style: b({
        "--scalar-sidebar-indent": e.indent + "px"
      })
    }, {
      default: a(() => [
        l(t.$slots, "header"),
        l(t.$slots, "default", {}, () => [
          g(i(x), { class: "custom-scroll pt-0" }, {
            default: a(() => [
              l(t.$slots, "before"),
              (s(!0), E($, null, k(i(F)(e.layout, e.items), (c) => (s(), f(B, {
                key: c.id,
                isDraggable: e.isDraggable ?? e.layout === "client",
                isDroppable: e.isDroppable,
                isExpanded: e.isExpanded,
                isSelected: e.isSelected,
                item: c,
                layout: e.layout,
                options: e.options,
                onOnDragEnd: y,
                onSelectItem: r[0] || (r[0] = (o) => m("selectItem", o))
              }, I({ _: 2 }, [
                n.decorator ? {
                  name: "decorator",
                  fn: a((o) => [
                    l(t.$slots, "decorator", d({ ref_for: !0 }, o))
                  ]),
                  key: "0"
                } : void 0,
                n.empty ? {
                  name: "empty",
                  fn: a((o) => [
                    l(t.$slots, "empty", d({ ref_for: !0 }, o))
                  ]),
                  key: "1"
                } : void 0,
                n.icon ? {
                  name: "icon",
                  fn: a((o) => [
                    l(t.$slots, "icon", d({ ref_for: !0 }, o))
                  ]),
                  key: "2"
                } : void 0
              ]), 1032, ["isDraggable", "isDroppable", "isExpanded", "isSelected", "item", "layout", "options"]))), 128))
            ]),
            _: 3
          }),
          r[1] || (r[1] = D("div", { class: "flex-1" }, null, -1))
        ]),
        l(t.$slots, "footer")
      ]),
      _: 3
    }, 8, ["style"]));
  }
});
export {
  w as default
};
