// @scalar/sidebar/dist/components/SidebarItem.vue.js
import { defineComponent as N, useSlots as V, resolveComponent as H, createBlock as d, unref as n, openBlock as l, mergeProps as c, toHandlers as v, withCtx as i, createTextVNode as L, toDisplayString as O, createElementBlock as f, Fragment as I, renderList as $, createSlots as b, renderSlot as r, createVNode as s, createCommentVNode as D, normalizeClass as w } from "vue";
import { ScalarSidebarSection as p, ScalarSidebarGroup as z, ScalarSidebarGroupToggle as A, ScalarSidebarItem as P } from "@scalar/components";
import { LibraryIcon as j } from "@scalar/icons/library";
import C from "./SidebarItemDecorator.vue.js";
import { filterItems as x } from "../helpers/filter-items.js";
import { hasChildren as q } from "../helpers/has-children.js";
import { isSidebarFolder as J } from "../helpers/is-sidebar-folder.js";
import { useDraggable as K } from "../hooks/use-draggable.js";
import F from "./SidebarHttpBadge.vue.js";
import g from "./SidebarItemLabel.vue.js";
const M = {
  key: 0,
  class: "line-through"
}, Q = {
  key: 0,
  class: "line-through"
}, oe = /* @__PURE__ */ N({
  __name: "SidebarItem",
  props: {
    item: {},
    layout: {},
    isSelected: { type: Function },
    isExpanded: { type: Function },
    options: {},
    isDraggable: {},
    isDroppable: { type: Function }
  },
  emits: ["selectItem", "onDragEnd"],
  setup(e, { emit: G }) {
    const y = G, a = V(), B = (t) => "isGroup" in t && t.isGroup, E = (t) => ("isDeprecated" in t && t.isDeprecated) ?? !1, S = (t, m) => {
      y("onDragEnd", t, m);
    }, { draggableAttrs: h, draggableEvents: k } = K({
      id: e.item.id,
      isDraggable: e.isDraggable,
      isDroppable: e.isDroppable,
      onDragEnd: S
    });
    return (t, m) => {
      const T = H("SidebarItem", !0);
      return n(q)(e.item) && B(e.item) ? (l(), d(n(p), c({ key: 0 }, n(h), v(n(k))), {
        items: i(() => [
          (l(!0), f(I, null, $(n(x)(e.layout, e.item.children), (u) => (l(), d(T, {
            key: u.id,
            isDraggable: e.isDraggable,
            isDroppable: e.isDroppable,
            isExpanded: e.isExpanded,
            isSelected: e.isSelected,
            item: u,
            layout: e.layout,
            options: e.options,
            onOnDragEnd: S,
            onSelectItem: m[0] || (m[0] = (o) => y("selectItem", o))
          }, b({ _: 2 }, [
            a.decorator ? {
              name: "decorator",
              fn: i((o) => [
                r(t.$slots, "decorator", c({ ref_for: !0 }, o))
              ]),
              key: "0"
            } : void 0,
            a.empty ? {
              name: "empty",
              fn: i((o) => [
                r(t.$slots, "empty", c({ ref_for: !0 }, o))
              ]),
              key: "1"
            } : void 0,
            a.icon ? {
              name: "icon",
              fn: i((o) => [
                r(t.$slots, "icon", c({ ref_for: !0 }, o))
              ]),
              key: "2"
            } : void 0
          ]), 1032, ["isDraggable", "isDroppable", "isExpanded", "isSelected", "item", "layout", "options"]))), 128))
        ]),
        default: i(() => [
          L(O(e.item.title) + " ", 1)
        ]),
        _: 3
      }, 16)) : n(J)(e.layout, e.item, a.empty !== void 0) ? (l(), d(n(z), c({
        key: 1,
        active: e.isSelected(e.item.id),
        class: "relative",
        controlled: "",
        open: e.isExpanded(e.item.id)
      }, n(h), v(n(k)), {
        onClick: m[2] || (m[2] = () => y("selectItem", e.item.id))
      }), b({
        items: i(() => [
          (l(!0), f(I, null, $(n(x)(e.layout, e.item.children ?? []), (u) => (l(), d(T, {
            key: u.id,
            isDraggable: e.isDraggable,
            isDroppable: e.isDroppable,
            isExpanded: e.isExpanded,
            isSelected: e.isSelected,
            item: u,
            layout: e.layout,
            options: e.options,
            parentIds: [],
            onOnDragEnd: S,
            onSelectItem: m[1] || (m[1] = (o) => y("selectItem", o))
          }, b({ _: 2 }, [
            a.decorator ? {
              name: "decorator",
              fn: i((o) => [
                r(t.$slots, "decorator", c({ ref_for: !0 }, o))
              ]),
              key: "0"
            } : void 0,
            a.empty ? {
              name: "empty",
              fn: i((o) => [
                r(t.$slots, "empty", c({ ref_for: !0 }, o))
              ]),
              key: "1"
            } : void 0,
            a.icon ? {
              name: "icon",
              fn: i((o) => [
                r(t.$slots, "icon", c({ ref_for: !0 }, o))
              ]),
              key: "2"
            } : void 0
          ]), 1032, ["isDraggable", "isDroppable", "isExpanded", "isSelected", "item", "layout", "options"]))), 128)),
          a.empty && (e.item.children?.length ?? 0) === 0 ? r(t.$slots, "empty", {
            key: 0,
            item: e.item
          }) : D("", !0)
        ]),
        default: i(() => [
          E(e.item) ? (l(), f("span", M, [
            s(g, {
              item: e.item,
              operationTitleSource: e.options?.operationTitleSource
            }, null, 8, ["item", "operationTitleSource"])
          ])) : (l(), d(g, {
            key: 1,
            item: e.item,
            operationTitleSource: e.options?.operationTitleSource
          }, null, 8, ["item", "operationTitleSource"]))
        ]),
        _: 2
      }, [
        e.item.type === "document" ? {
          name: "icon",
          fn: i(({ open: u }) => [
            r(t.$slots, "icon", {
              item: e.item,
              open: u
            }, () => [
              s(n(j), {
                class: "text-c-3 block group-hover/group-button:hidden",
                src: "icon" in e.item && e.item.icon || "interface-content-folder"
              }, null, 8, ["src"]),
              s(n(A), {
                class: "text-c-3 hidden group-hover/group-button:flex",
                open: u
              }, null, 8, ["open"])
            ])
          ]),
          key: "0"
        } : void 0,
        "method" in e.item ? {
          name: "aside",
          fn: i(() => [
            "method" in e.item ? (l(), d(F, {
              key: 0,
              active: e.isSelected(e.item.id),
              class: w(["ml-2 h-4 self-start", {
                // Hide the badge when we're showing the decorator
                "group-hover/button:opacity-0 group-focus-visible/button:opacity-0 group-has-[~*_[aria-expanded=true]]/button:opacity-0 group-has-[~*:focus-within]/button:opacity-0 group-has-[~*:hover]/button:opacity-0": a.decorator
              }]),
              method: e.item.method,
              webhook: e.item.type === "webhook"
            }, null, 8, ["active", "class", "method", "webhook"])) : D("", !0)
          ]),
          key: "1"
        } : void 0,
        a.decorator ? {
          name: "after",
          fn: i(() => [
            s(C, null, {
              default: i(() => [
                r(t.$slots, "decorator", { item: e.item })
              ]),
              _: 3
            })
          ]),
          key: "2"
        } : void 0
      ]), 1040, ["active", "open"])) : (l(), d(n(P), c({
        key: 2,
        is: "button"
      }, n(h), {
        class: "relative",
        selected: e.isSelected(e.item.id)
      }, v(n(k)), {
        onClick: m[3] || (m[3] = () => y("selectItem", e.item.id))
      }), b({
        default: i(() => [
          E(e.item) ? (l(), f("span", Q, [
            s(g, {
              item: e.item,
              operationTitleSource: e.options?.operationTitleSource
            }, null, 8, ["item", "operationTitleSource"])
          ])) : (l(), d(g, {
            key: 1,
            item: e.item,
            operationTitleSource: e.options?.operationTitleSource
          }, null, 8, ["item", "operationTitleSource"]))
        ]),
        _: 2
      }, [
        a.icon ? {
          name: "icon",
          fn: i(() => [
            r(t.$slots, "icon", {
              item: e.item,
              open: !0
            })
          ]),
          key: "0"
        } : void 0,
        "method" in e.item ? {
          name: "aside",
          fn: i(() => [
            "method" in e.item ? (l(), d(F, {
              key: 0,
              active: e.isSelected(e.item.id),
              class: w(["ml-2 h-4 self-start", {
                // Hide the badge when we're showing the decorator
                "group-hover/button:opacity-0 group-focus-visible/button:opacity-0 group-has-[~*_[aria-expanded=true]]/button:opacity-0 group-has-[~*:focus-within]/button:opacity-0 group-has-[~*:hover]/button:opacity-0": a.decorator
              }]),
              method: e.item.method,
              webhook: e.item.type === "webhook"
            }, null, 8, ["active", "class", "method", "webhook"])) : D("", !0)
          ]),
          key: "1"
        } : void 0,
        a.decorator ? {
          name: "after",
          fn: i(() => [
            s(C, null, {
              default: i(() => [
                r(t.$slots, "decorator", { item: e.item })
              ]),
              _: 3
            })
          ]),
          key: "2"
        } : void 0
      ]), 1040, ["selected"]));
    };
  }
});
export {
  oe as default
};
