// @scalar/sidebar/dist/components/SidebarHttpBadge.vue2.js
import { defineComponent as n, createBlock as t, openBlock as r, normalizeClass as d, withCtx as m, createElementVNode as s, renderSlot as i, createCommentVNode as c, unref as a, normalizeStyle as h } from "vue";
import { getHttpMethodInfo as f } from "@scalar/helpers/http/http-info";
import { ScalarIconWebhooksLogo as p } from "@scalar/icons";
import y from "./HttpMethod.vue.js";
const w = /* @__PURE__ */ n({
  __name: "SidebarHttpBadge",
  props: {
    method: {},
    active: { type: Boolean },
    webhook: { type: Boolean }
  },
  setup(e) {
    return (l, o) => (r(), t(y, {
      class: d([
        "sidebar-heading-type",
        `sidebar-heading-type--${e.method.toLowerCase()}`,
        { "sidebar-heading-type-active": e.active }
      ]),
      method: e.method,
      property: "--method-color",
      short: ""
    }, {
      default: m(() => [
        o[0] || (o[0] = s("span", { class: "sr-only" }, "HTTP Method: ", -1)),
        i(l.$slots, "default", {}, () => [
          e.webhook ? (r(), t(a(p), {
            key: 0,
            style: h({
              color: a(f)(e.method).colorVar
            }),
            weight: "bold"
          }, null, 8, ["style"])) : c("", !0)
        ], !0)
      ]),
      _: 3
    }, 8, ["class", "method"]));
  }
});
export {
  w as default
};
