// @scalar/agent-chat/dist/views/PromptForm.vue2.js
import { defineComponent as F, useTemplateRef as H, computed as g, watch as A, createElementBlock as i, openBlock as l, createBlock as u, createCommentVNode as r, createElementVNode as n, withDirectives as C, unref as e, withModifiers as N, vModelText as K, createVNode as v, withCtx as j, Fragment as S, renderList as w, createTextVNode as T, toDisplayString as q, vShow as J } from "vue";
import { ScalarIconButton as _, ScalarLoading as X } from "@scalar/components";
import { ScalarIconPlus as O, ScalarIconX as Z, ScalarIconLockSimple as G, ScalarIconCheck as Q, ScalarIconArrowUp as W } from "@scalar/icons";
import Y from "../components/ApprovalSection.vue.js";
import ee from "../components/ErrorMessage.vue.js";
import te from "../components/FreeMessagesInfoSection.vue.js";
import oe from "../components/PaymentSection.vue.js";
import se from "../components/SearchPopover.vue.js";
import le from "../components/UploadSection.vue.js";
import { AgentErrorCodes as P } from "../entities/error/constants.js";
import { MAX_PROMPT_SIZE as ne } from "../entities/prompt/constants.js";
import { useChatApprovals as re } from "../hooks/use-chat-approvals.js";
import { useChatError as ae } from "../hooks/use-chat-error.js";
import { useChatPendingClientToolParts as ie } from "../hooks/use-chat-pending-client-tool-parts.js";
import { useUploadTmpDocument as ue } from "../hooks/use-upload-tmp-document.js";
import { useState as ce } from "../state/state.js";
const pe = { class: "actionContainer" }, me = ["disabled"], de = { class: "inputActionsContainer" }, ge = { class: "inputActionsLeft" }, ve = {
  class: "addAPIButton",
  type: "button"
}, fe = ["src"], he = ["onClick"], ye = { class: "inputActionsRight" }, be = { class: "sendCheckboxContinue" }, ke = {
  key: 0,
  class: "relative flex items-center gap-1.5"
}, Ae = {
  class: "termsAgree",
  for: "agentTermsAgree"
}, Ce = { class: "addMoreContext" }, Se = { class: "ml-auto flex items-center gap-1" }, we = ["onClick"], Te = ["alt", "src"], Ke = /* @__PURE__ */ F({
  __name: "PromptForm",
  emits: ["submit"],
  setup(_e, { expose: x, emit: I }) {
    const D = I;
    x({ focusPrompt: U });
    const c = H("agentPrompt"), t = ce(), E = g(() => t.prompt.value.trim().length > 0), M = g(
      () => t.prompt.value.trim().length > ne
    ), L = g(
      () => t.chat.messages.length > 1 && !t.getAgentKey?.() && d?.value?.code !== P.LIMIT_REACHED
    );
    A(t.prompt, () => {
      if (c?.value) {
        if (!t.prompt.value.length) {
          c.value.style.height = "0px";
          return;
        }
        c.value.style.height = "auto", c.value.style.height = c.value.scrollHeight + "px";
      }
    });
    function R(a) {
      t.loading.value || a.key === "Enter" && !a.shiftKey && (a.preventDefault(), b(), window.scrollTo(0, document.body.scrollHeight));
    }
    function U() {
      c.value?.focus();
    }
    A(
      () => t.chat.status,
      (a) => {
        (a === "ready" || a === "error") && c.value?.focus();
      }
    );
    const { respondToToolCalls: p, approvalRequestedParts: h } = re(), { pendingClientToolParts: z } = ie(), m = ue();
    function B() {
      t.terms.accept(), t.mode === "preview" && t.getActiveDocumentJson && m.uploadTempDocument(t.getActiveDocumentJson(), !0);
    }
    const y = g(() => {
      const a = M.value, s = !E.value, o = h.value.length > 0, f = z.value.length > 0, k = t.mode === "preview", $ = k && !t.terms.accepted.value, V = k && !!m.uploadState.value;
      return a || s || o || f || $ || V;
    });
    function b() {
      y.value || D("submit");
    }
    const d = ae();
    return (a, s) => (l(), i("div", pe, [
      e(m).uploadState.value ? (l(), u(le, {
        key: 0,
        uploadState: e(m).uploadState.value
      }, null, 8, ["uploadState"])) : r("", !0),
      e(d) ? (l(), u(ee, {
        key: 1,
        error: e(d)
      }, null, 8, ["error"])) : r("", !0),
      e(h).length ? (l(), u(Y, {
        key: 2,
        onApprove: s[0] || (s[0] = (o) => e(p)(!0)),
        onReject: s[1] || (s[1] = (o) => e(p)(!1))
      })) : r("", !0),
      e(d)?.code === e(P).LIMIT_REACHED ? (l(), u(oe, {
        key: 3,
        onApprove: s[2] || (s[2] = (o) => e(p)(!0)),
        onReject: s[3] || (s[3] = (o) => e(p)(!1))
      })) : r("", !0),
      L.value ? (l(), u(te, { key: 4 })) : r("", !0),
      n("form", {
        class: "promptForm",
        onSubmit: N(b, ["prevent"])
      }, [
        s[7] || (s[7] = n("label", {
          class: "agentLabel",
          for: "agentTextarea"
        }, " Type a Request To get Started ", -1)),
        C(n("textarea", {
          id: "agentTextarea",
          ref: "agentPrompt",
          "onUpdate:modelValue": s[4] || (s[4] = (o) => e(t).prompt.value = o),
          class: "prompt custom-scroll",
          disabled: e(t).loading.value,
          name: "prompt",
          placeholder: "Ask me anything...",
          onKeydown: R
        }, null, 40, me), [
          [K, e(t).prompt.value]
        ]),
        n("div", de, [
          n("div", ge, [
            v(se, null, {
              default: j(() => [
                n("button", ve, [
                  v(e(O), {
                    class: "size-4",
                    weight: "bold"
                  })
                ])
              ]),
              _: 1
            }),
            (l(!0), i(S, null, w(e(t).registryDocuments.value, (o) => (l(), i("div", {
              key: o.id,
              class: "apiPill"
            }, [
              o.logoUrl ? (l(), i("img", {
                key: 0,
                class: "apiPillLogo",
                src: o.logoUrl
              }, null, 8, fe)) : r("", !0),
              T(" " + q(o.title) + " ", 1),
              o.removable ? (l(), i("button", {
                key: 1,
                class: "apiPillRemove",
                type: "button",
                onClick: (f) => e(t).removeDocument(o)
              }, [
                v(e(Z), {
                  class: "size-4",
                  weight: "bold"
                })
              ], 8, he)) : r("", !0)
            ]))), 128))
          ]),
          n("div", ye, [
            e(t).loading.value ? r("", !0) : (l(), u(e(_), {
              key: 0,
              class: "settingsButton h-7 w-7 p-1.5",
              icon: e(G),
              label: "Scalar",
              size: "md",
              weight: "bold",
              onClick: s[5] || (s[5] = (o) => e(t).settingsModal.show())
            }, null, 8, ["icon"])),
            n("div", be, [
              !e(t).terms.accepted.value && e(t).mode === "preview" ? (l(), i("div", ke, [
                n("input", {
                  id: "agentTermsAgree",
                  class: "sr-only",
                  type: "checkbox",
                  onChange: B
                }, null, 32),
                n("label", Ae, [
                  v(e(Q), {
                    class: "termsAgreeIcon",
                    weight: "bold"
                  }),
                  s[6] || (s[6] = T(" Agree to Terms & Conditions ", -1))
                ])
              ])) : r("", !0),
              e(t).loading.value ? (l(), u(e(X), {
                key: 2,
                class: "loader h-7 w-7",
                loader: {
                  isLoading: e(t).loading.value,
                  isValid: !1,
                  clear: async () => {
                  },
                  invalidate: async () => {
                  },
                  isInvalid: !1,
                  isActive: !1,
                  validate: async () => {
                  },
                  start: () => {
                  }
                },
                size: "2xl"
              }, null, 8, ["loader"])) : (l(), u(e(_), {
                key: 1,
                class: "sendButton h-7 w-7 p-1.5",
                disabled: y.value,
                icon: e(W),
                label: "Scalar",
                size: "md",
                type: "submit",
                weight: "bold"
              }, null, 8, ["disabled", "icon"]))
            ])
          ])
        ])
      ], 32),
      C(n("div", Ce, [
        s[8] || (s[8] = n("span", null, "Add context from dozens of API's", -1)),
        n("div", Se, [
          (l(!0), i(S, null, w(e(t).curatedDocuments.value, (o) => (l(), i("button", {
            key: o.id,
            class: "addAPIContext",
            type: "button",
            onClick: (f) => e(t).addDocument(o)
          }, [
            o.logoUrl ? (l(), i("img", {
              key: 0,
              alt: o.title,
              class: "size-4",
              src: o.logoUrl
            }, null, 8, Te)) : r("", !0)
          ], 8, we))), 128))
        ])
      ], 512), [
        [J, e(t).chat.messages.length <= 1]
      ])
    ]));
  }
});
export {
  Ke as default
};
