// @scalar/agent-chat/dist/views/Start.vue2.js
import { defineComponent as p, createElementBlock as r, openBlock as s, createVNode as a, createElementVNode as t, createTextVNode as o, unref as d, Fragment as n } from "vue";
import u from "../components/Logo.vue.js";
import { useState as g } from "../state/state.js";
import c from "./PromptForm.vue.js";
const y = { class: "startContainer" }, f = { class: "disclaimerText" }, L = /* @__PURE__ */ p({
  __name: "Start",
  emits: ["submit"],
  setup(k, { emit: i }) {
    const m = i, { mode: l } = g();
    return (S, e) => (s(), r("div", y, [
      a(u, { class: "agentLogo" }),
      e[5] || (e[5] = t("h1", { class: "heading" }, "How can I help you today?", -1)),
      a(c, {
        ref: "promptFormField",
        onSubmit: e[0] || (e[0] = (b) => m("submit"))
      }, null, 512),
      t("p", f, [
        d(l) === "preview" ? (s(), r(n, { key: 0 }, [
          o(" By messaging Agent Scalar your OpenAPI Document will be Temporarily uploaded to Scalar's Servers. You must agree to our ")
        ], 64)) : (s(), r(n, { key: 1 }, [
          o(" By messaging Agent Scalar you agree to our ")
        ], 64)),
        e[1] || (e[1] = t("a", {
          class: "disclaimerLink",
          href: "https://scalar.com/legal/terms-and-conditions",
          target: "_blank"
        }, "Terms ", -1)),
        e[2] || (e[2] = o(" and ", -1)),
        e[3] || (e[3] = t("a", {
          class: "disclaimerLink",
          href: "https://scalar.com/legal/privacy-policy",
          target: "_blank"
        }, " Privacy Policy", -1)),
        e[4] || (e[4] = o(". ", -1))
      ])
    ]));
  }
});
export {
  L as default
};
