// @scalar/api-client/dist/components/Sidebar/Actions/SidebarListElementForm.vue2.js
import { defineComponent as c, openBlock as m, createElementBlock as p, withModifiers as b, renderSlot as f, createElementVNode as n, createVNode as r, unref as s, createTextVNode as l, toDisplayString as d } from "vue";
import { ScalarIcon as i } from "@scalar/components";
const g = ["error"], S = /* @__PURE__ */ c({
  __name: "SidebarListElementForm",
  props: {
    danger: { type: Boolean },
    label: {}
  },
  emits: ["cancel", "submit"],
  setup(x, { emit: u }) {
    const o = u;
    return (t, e) => (m(), p("form", {
      class: "text-base",
      onSubmit: e[1] || (e[1] = b((a) => o("submit"), ["prevent"]))
    }, [
      f(t.$slots, "default", {}, void 0, !0),
      n("div", null, [
        n("button", {
          class: "w-full text-left focus:bg-b-2 hover:bg-b-2 rounded gap-1.5 px-2.5 py-1.5 focus:outline-none flex items-center cursor-pointer",
          type: "button",
          onClick: e[0] || (e[0] = (a) => o("cancel"))
        }, [
          r(s(i), {
            class: "inline-flex",
            icon: "Close",
            size: "sm",
            thickness: "1.75"
          }),
          l(" Cancel ")
        ]),
        n("button", {
          class: "w-full text-left focus:bg-b-2 hover:bg-b-2 rounded gap-1.5 px-2.5 py-1.5 focus:outline-none flex items-center cursor-pointer delete-warning-button",
          error: t.danger,
          type: "submit"
        }, [
          r(s(i), {
            class: "inline-flex",
            icon: "Trash",
            size: "sm",
            thickness: "1.5"
          }),
          l(" " + d(t.label ?? "Submit"), 1)
        ], 8, g)
      ])
    ], 32));
  }
});
export {
  S as default
};
