// @scalar/components/dist/components/ScalarCheckboxInput/ScalarCheckboxRadioGroup.vue.js
import { defineComponent as n, mergeModels as c, useModel as s, useId as i, createBlock as t, openBlock as l, unref as u, withCtx as d, createElementBlock as f, Fragment as _, renderList as p, createTextVNode as v, toDisplayString as V } from "vue";
import k from "./ScalarCheckboxInput.vue.js";
import x from "../ScalarForm/ScalarFormInputGroup.vue.js";
const g = {}, b = /* @__PURE__ */ n({
  ...g,
  __name: "ScalarCheckboxRadioGroup",
  props: /* @__PURE__ */ c({
    options: { default: () => [] }
  }, {
    modelValue: {},
    modelModifiers: {}
  }),
  emits: ["update:modelValue"],
  setup(a) {
    const o = s(a, "modelValue"), m = i();
    return (h, y) => (l(), t(u(x), null, {
      default: d(() => [
        (l(!0), f(_, null, p(a.options, (e) => (l(), t(k, {
          key: e.value,
          modelValue: o.value?.value === e.value,
          name: u(m),
          type: "radio",
          value: e.value,
          "onUpdate:modelValue": (r) => o.value = r ? e : void 0
        }, {
          default: d(() => [
            v(V(e.label), 1)
          ]),
          _: 2
        }, 1032, ["modelValue", "name", "value", "onUpdate:modelValue"]))), 128))
      ]),
      _: 1
    }));
  }
});
export {
  b as default
};
