// @scalar/components/dist/components/ScalarCheckboxInput/ScalarCheckboxInput.vue.js
import { defineComponent as c, mergeModels as u, useModel as m, createBlock as p, openBlock as f, unref as o, mergeProps as s, withCtx as _, createVNode as y, createElementVNode as r, withDirectives as h, renderSlot as x, vModelDynamic as v } from "vue";
import { useBindCx as V } from "@scalar/use-hooks/useBindCx";
import b from "./ScalarCheckbox.vue.js";
import g from "../ScalarForm/ScalarFormInput.vue.js";
const k = { class: "flex-1 text-left min-w-0 truncate" }, C = ["type"], B = {}, D = /* @__PURE__ */ c({
  ...B,
  inheritAttrs: !1,
  __name: "ScalarCheckboxInput",
  props: /* @__PURE__ */ u({
    type: { default: "checkbox" }
  }, {
    modelValue: { type: Boolean },
    modelModifiers: {}
  }),
  emits: ["update:modelValue"],
  setup(t) {
    const e = m(t, "modelValue"), { stylingAttrsCx: n, otherAttrs: a } = V();
    return (i, l) => (f(), p(o(g), s(
      { is: "label" },
      o(n)("cursor-pointer gap-2 hover:bg-b-2", { "text-c-1": e.value })
    ), {
      default: _(() => [
        y(b, {
          class: "shrink-0",
          selected: e.value,
          type: t.type
        }, null, 8, ["selected", "type"]),
        r("div", k, [
          x(i.$slots, "default")
        ]),
        h(r("input", s({
          "onUpdate:modelValue": l[0] || (l[0] = (d) => e.value = d),
          class: "sr-only",
          type: t.type
        }, o(a)), null, 16, C), [
          [v, e.value]
        ])
      ]),
      _: 3
    }, 16));
  }
});
export {
  D as default
};
