// @scalar/components/dist/components/ScalarCheckboxInput/ScalarCheckboxGroup.vue.js
import { defineComponent as s, mergeModels as c, useModel as f, useId as _, createBlock as o, openBlock as u, unref as m, withCtx as r, createElementBlock as i, Fragment as v, renderList as p, createTextVNode as V, toDisplayString as k } from "vue";
import x from "./ScalarCheckboxInput.vue.js";
import g from "../ScalarForm/ScalarFormInputGroup.vue.js";
const h = {}, S = /* @__PURE__ */ s({
  ...h,
  __name: "ScalarCheckboxGroup",
  props: /* @__PURE__ */ c({
    options: { default: () => [] }
  }, {
    modelValue: { default: [] },
    modelModifiers: {}
  }),
  emits: ["update:modelValue"],
  setup(t) {
    const l = f(t, "modelValue"), d = _();
    return (B, C) => (u(), o(m(g), null, {
      default: r(() => [
        (u(!0), i(v, null, p(t.options, (e) => (u(), o(x, {
          key: e.value,
          modelValue: l.value?.some(({ value: a }) => a === e.value),
          name: m(d),
          value: e.value,
          "onUpdate:modelValue": (a) => l.value = a ? [...l.value, e] : l.value.filter(({ value: n }) => n !== e.value)
        }, {
          default: r(() => [
            V(k(e.label), 1)
          ]),
          _: 2
        }, 1032, ["modelValue", "name", "value", "onUpdate:modelValue"]))), 128))
      ]),
      _: 1
    }));
  }
});
export {
  S as default
};
