// @scalar/api-client/dist/v2/features/settings/CollectionSettings.vue.js
import { defineComponent as M, createElementBlock as y, openBlock as c, createVNode as n, withCtx as s, createElementVNode as o, unref as r, normalizeClass as a, createTextVNode as l, createBlock as f, createCommentVNode as p, Fragment as C, renderList as w, toDisplayString as U, normalizeStyle as g } from "vue";
import { cva as $, ScalarButton as m, cx as x, ScalarIcon as b } from "@scalar/components";
import { themeLabels as T } from "@scalar/themes";
import B from "../../../components/ImportCollection/IntegrationLogo.vue.js";
import L from "./components/Appearance.vue.js";
import k from "./components/Section.vue.js";
import { getThemeColors as h } from "./helpers/get-theme-colors.js";
const N = { class: "flex flex-col gap-4" }, R = { class: "flex flex-col gap-2" }, V = { class: "flex flex-col gap-2" }, A = { class: "grid grid-cols-2 gap-2" }, E = { class: "flex items-center gap-2" }, F = { class: "flex items-center gap-1" }, O = { class: "grid grid-cols-2 gap-2" }, D = { class: "flex items-center gap-2" }, W = { class: "flex items-center gap-1" }, q = { class: "size-7 rounded-xl" }, v = "https://proxy.scalar.com", I = /* @__PURE__ */ M({
  __name: "CollectionSettings",
  props: {
    activeProxyUrl: {},
    activeThemeId: {},
    colorMode: {}
  },
  emits: ["update:proxyUrl", "update:themeId", "update:colorMode"],
  setup(i, { emit: S }) {
    const d = S, P = [
      "default",
      "alternate",
      "purple",
      "solarized",
      "saturn",
      "kepler"
    ], z = ["elysiajs", "fastify"], u = $({
      base: "w-full shadow-none text-c-1 justify-start pl-2 gap-2 border",
      variants: {
        active: {
          true: "bg-primary text-c-1 hover:bg-inherit",
          false: "bg-b-1 hover:bg-b-2"
        }
      }
    });
    return (G, e) => (c(), y("div", N, [
      n(k, null, {
        title: s(() => [...e[3] || (e[3] = [
          l(" CORS Proxy ", -1)
        ])]),
        description: s(() => [...e[4] || (e[4] = [
          l(" Browsers block cross-origin requests for security. We provide a public proxy to ", -1),
          o("a", {
            class: "hover:text-c-1 underline-offset-2",
            href: "https://en.wikipedia.org/wiki/Cross-origin_resource_sharing",
            target: "_blank"
          }, " bypass CORS issues ", -1),
          l(" . Check the ", -1),
          o("a", {
            class: "hover:text-c-1 underline-offset-2",
            href: "https://github.com/scalar/scalar/tree/main/projects/proxy-scalar-com",
            target: "_blank"
          }, " source code on GitHub ", -1),
          l(" . ", -1)
        ])]),
        default: s(() => [
          o("div", R, [
            n(r(m), {
              class: a(
                r(x)(
                  r(u)({
                    active: i.activeProxyUrl === v
                  })
                )
              ),
              onClick: e[0] || (e[0] = (t) => d("update:proxyUrl", v))
            }, {
              default: s(() => [
                o("div", {
                  class: a(["flex h-5 w-5 items-center justify-center rounded-full border-[1.5px] p-1", {
                    "bg-c-accent text-b-1 border-transparent": i.activeProxyUrl === v
                  }])
                }, [
                  i.activeProxyUrl === v ? (c(), f(r(b), {
                    key: 0,
                    icon: "Checkmark",
                    size: "xs",
                    thickness: "3.5"
                  })) : p("", !0)
                ], 2),
                e[5] || (e[5] = l(" Use proxy.scalar.com (default) ", -1))
              ]),
              _: 1
            }, 8, ["class"]),
            n(r(m), {
              class: a(r(x)(r(u)({ active: !i.activeProxyUrl }))),
              onClick: e[1] || (e[1] = (t) => d("update:proxyUrl", null))
            }, {
              default: s(() => [
                o("div", {
                  class: a([
                    "flex h-5 w-5 items-center justify-center rounded-full border-[1.5px] p-1",
                    !i.activeProxyUrl && "bg-c-accent text-b-1 border-transparent"
                  ])
                }, [
                  i.activeProxyUrl ? p("", !0) : (c(), f(r(b), {
                    key: 0,
                    icon: "Checkmark",
                    size: "xs",
                    thickness: "3.5"
                  }))
                ], 2),
                e[6] || (e[6] = l(" Skip the proxy ", -1))
              ]),
              _: 1
            }, 8, ["class"])
          ])
        ]),
        _: 1
      }),
      n(k, null, {
        title: s(() => [...e[7] || (e[7] = [
          l(" Themes ", -1)
        ])]),
        description: s(() => [...e[8] || (e[8] = [
          l(" We've got a whole rainbow of themes for you to play with: ", -1)
        ])]),
        default: s(() => [
          o("div", V, [
            o("div", A, [
              (c(), y(C, null, w(P, (t) => n(r(m), {
                key: t,
                class: a(
                  r(x)(
                    r(u)({
                      active: i.activeThemeId === t
                    })
                  )
                ),
                onClick: (j) => d("update:themeId", t)
              }, {
                default: s(() => [
                  o("div", E, [
                    o("div", {
                      class: a(["flex h-5 w-5 items-center justify-center rounded-full border-[1.5px] p-1", {
                        "bg-c-accent text-b-1 border-transparent": i.activeThemeId === t
                      }])
                    }, [
                      i.activeThemeId === t ? (c(), f(r(b), {
                        key: 0,
                        icon: "Checkmark",
                        size: "xs",
                        thickness: "3.5"
                      })) : p("", !0)
                    ], 2),
                    l(" " + U(r(T)[t]), 1)
                  ]),
                  o("div", F, [
                    o("span", {
                      class: "border-c-3 -mr-3 inline-block h-5 w-5 rounded-full",
                      style: g({
                        backgroundColor: r(h)(t).light
                      })
                    }, null, 4),
                    o("span", {
                      class: "border-c-3 -mr-3 inline-block h-5 w-5 rounded-full",
                      style: g({
                        backgroundColor: r(h)(t).dark
                      })
                    }, null, 4),
                    o("span", {
                      class: "border-c-3 inline-block h-5 w-5 rounded-full",
                      style: g({
                        backgroundColor: r(h)(t).accent
                      })
                    }, null, 4)
                  ])
                ]),
                _: 2
              }, 1032, ["class", "onClick"])), 64))
            ])
          ])
        ]),
        _: 1
      }),
      n(k, null, {
        title: s(() => [...e[9] || (e[9] = [
          l(" Framework Themes ", -1)
        ])]),
        description: s(() => [...e[10] || (e[10] = [
          l(" Are you a real fan? Show your support by using your favorite framework's theme! ", -1)
        ])]),
        default: s(() => [
          o("div", O, [
            (c(), y(C, null, w(z, (t) => n(r(m), {
              key: t,
              class: a(
                r(x)(
                  r(u)({
                    active: i.activeThemeId === t
                  })
                )
              ),
              onClick: (j) => d("update:themeId", t)
            }, {
              default: s(() => [
                o("div", D, [
                  o("div", {
                    class: a(["flex h-5 w-5 items-center justify-center rounded-full border-[1.5px] p-1", {
                      "bg-c-accent text-b-1 border-transparent": i.activeThemeId === t
                    }])
                  }, [
                    i.activeThemeId === t ? (c(), f(r(b), {
                      key: 0,
                      icon: "Checkmark",
                      size: "xs",
                      thickness: "3.5"
                    })) : p("", !0)
                  ], 2),
                  l(" " + U(r(T)[t]), 1)
                ]),
                o("div", W, [
                  o("div", q, [
                    n(B, { integration: t }, null, 8, ["integration"])
                  ])
                ])
              ]),
              _: 2
            }, 1032, ["class", "onClick"])), 64))
          ])
        ]),
        _: 1
      }),
      n(k, null, {
        title: s(() => [...e[11] || (e[11] = [
          l(" Appearance ", -1)
        ])]),
        description: s(() => [...e[12] || (e[12] = [
          l(" Choose between light, dark, or system-based appearance for your workspace. ", -1)
        ])]),
        default: s(() => [
          n(L, {
            colorMode: i.colorMode,
            "onUpdate:colorMode": e[2] || (e[2] = (t) => d("update:colorMode", t))
          }, null, 8, ["colorMode"])
        ]),
        _: 1
      })
    ]));
  }
});
export {
  I as default
};
