// @scalar/api-client/dist/v2/features/settings/DocumentSettings.vue.js
import { defineComponent as p, createElementBlock as d, openBlock as a, Fragment as i, createElementVNode as e, createVNode as n, unref as o, createTextVNode as u, toDisplayString as h, withCtx as m } from "vue";
import { useModal as v, ScalarToggle as w, ScalarIcon as f, ScalarButton as D, ScalarModal as y } from "@scalar/components";
import k from "../../../components/Sidebar/Actions/DeleteSidebarListElement.vue.js";
const M = { class: "flex flex-col gap-12" }, U = { class: "flex flex-col gap-2" }, C = { class: "bg-b-2 rounded-lg border text-sm" }, N = { class: "bg-b-1 flex items-center justify-between gap-4 rounded-t-lg p-3" }, S = { class: "text-c-1 flex items-center overflow-x-auto border-t py-1.5 whitespace-nowrap" }, B = { class: "flex items-center" }, V = ["href"], T = { class: "flex flex-col gap-4" }, z = { class: "flex items-center justify-between rounded-lg border p-3 text-sm" }, O = /* @__PURE__ */ p({
  __name: "DocumentSettings",
  props: {
    documentUrl: {},
    watchMode: { type: Boolean },
    title: {},
    isDraftDocument: { type: Boolean, default: !1 }
  },
  emits: ["delete:document", "update:watchMode"],
  setup(l, { emit: x }) {
    const r = x, s = v(), g = () => {
      l.isDraftDocument || s.show();
    }, b = () => {
      r("delete:document"), s.hide();
    };
    return (A, t) => (a(), d(i, null, [
      e("div", M, [
        e("div", U, [
          t[5] || (t[5] = e("div", { class: "flex h-8 items-center" }, [
            e("h3", { class: "font-bold" }, "Features")
          ], -1)),
          e("div", C, [
            e("div", N, [
              t[2] || (t[2] = e("div", null, [
                e("h4", null, "Watch Mode"),
                e("p", { class: "text-c-2 mt-1" }, " When enabled, the OpenAPI document will be polled for changes. The collection will be updated automatically. ")
              ], -1)),
              n(o(w), {
                class: "w-4",
                disabled: !l.documentUrl,
                modelValue: l.watchMode ?? !1,
                "onUpdate:modelValue": t[0] || (t[0] = (c) => r("update:watchMode", c))
              }, null, 8, ["disabled", "modelValue"])
            ]),
            e("div", S, [
              e("div", B, [
                l.documentUrl ? (a(), d(i, { key: 0 }, [
                  t[3] || (t[3] = e("span", { class: "bg-b-2 sticky left-0 pr-2 pl-3" }, "Source", -1)),
                  e("a", {
                    class: "text-c-2 group rounded pr-3 no-underline hover:underline",
                    href: l.documentUrl,
                    target: "_blank"
                  }, [
                    u(h(l.documentUrl) + " ", 1),
                    n(o(f), {
                      class: "ml-1 hidden w-2.5 group-hover:inline",
                      icon: "ExternalLink"
                    })
                  ], 8, V)
                ], 64)) : (a(), d(i, { key: 1 }, [
                  n(o(f), {
                    class: "text-c-2 mr-2 ml-3 w-4",
                    icon: "NotAllowed",
                    size: "sm"
                  }),
                  t[4] || (t[4] = e("span", { class: "text-c-2 pr-3" }, " No URL configured. Try importing an OpenAPI document from an URL. ", -1))
                ], 64))
              ])
            ])
          ])
        ]),
        e("div", T, [
          t[8] || (t[8] = e("h3", { class: "font-bold" }, "Danger Zone", -1)),
          e("div", z, [
            t[7] || (t[7] = e("div", null, [
              e("h4", null, "Delete Collection"),
              e("p", { class: "text-c-2 mt-1" }, " Be careful, my friend. Once deleted, there is no way to recover the collection. ")
            ], -1)),
            n(o(D), {
              disabled: l.isDraftDocument,
              size: "sm",
              variant: "danger",
              onClick: g
            }, {
              default: m(() => [...t[6] || (t[6] = [
                u(" Delete Collection ", -1)
              ])]),
              _: 1
            }, 8, ["disabled"])
          ])
        ])
      ]),
      n(o(y), {
        size: "xxs",
        state: o(s),
        title: `Delete ${l.title}`
      }, {
        default: m(() => [
          n(k, {
            variableName: l.title ?? "",
            warningMessage: "This action cannot be undone.",
            onClose: t[1] || (t[1] = (c) => o(s).hide()),
            onDelete: b
          }, null, 8, ["variableName"])
        ]),
        _: 1
      }, 8, ["state", "title"])
    ], 64));
  }
});
export {
  O as default
};
