// @scalar/api-client/dist/v2/components/forms/ConfirmationForm.vue.js
import { defineComponent as u, createElementBlock as d, openBlock as p, withModifiers as c, renderSlot as v, createElementVNode as b, createVNode as a, unref as i, withCtx as o, createTextVNode as l, toDisplayString as x } from "vue";
import { ScalarButton as r } from "@scalar/components";
const C = { class: "flex justify-between gap-10" }, g = /* @__PURE__ */ u({
  __name: "ConfirmationForm",
  props: {
    label: { default: "Save" },
    variant: { default: "solid" }
  },
  emits: ["cancel", "submit"],
  setup(t, { emit: s }) {
    const n = s;
    return (m, e) => (p(), d("form", {
      class: "flex flex-col gap-4 text-base",
      onSubmit: e[1] || (e[1] = c((f) => n("submit"), ["prevent"]))
    }, [
      v(m.$slots, "default"),
      b("div", C, [
        a(i(r), {
          size: "sm",
          type: "button",
          variant: "outlined",
          onClick: e[0] || (e[0] = (f) => n("cancel"))
        }, {
          default: o(() => [...e[2] || (e[2] = [
            l(" Cancel ", -1)
          ])]),
          _: 1
        }),
        a(i(r), {
          size: "sm",
          type: "submit",
          variant: t.variant
        }, {
          default: o(() => [
            l(x(t.label), 1)
          ]),
          _: 1
        }, 8, ["variant"])
      ])
    ], 32));
  }
});
export {
  g as default
};
