// @scalar/api-client/dist/layouts/App/ApiClientApp.vue.js
import { defineComponent as _, onMounted as i, watchEffect as v, onBeforeMount as y, onBeforeUnmount as h, computed as k, openBlock as l, createElementBlock as d, Fragment as g, createElementVNode as p, createVNode as a, unref as n, withCtx as w, createBlock as f, KeepAlive as S, resolveDynamicComponent as x, createCommentVNode as T } from "vue";
import C from "../../components/SideNav/SideNav.vue.js";
import L from "../../components/TopNav/TopNav.vue.js";
import { useWorkspace as B } from "../../store/workspace.js";
import { addScalarClassesToHeadless as E } from "@scalar/components";
import { LS_KEYS as K } from "@scalar/object-utils/mutator-record";
import { getThemeStyles as M } from "@scalar/themes";
import { ScalarToasts as W } from "@scalar/use-toasts";
import { RouterView as A } from "vue-router";
import { loadAllResources as D } from "../../libs/local-storage.js";
import N from "../../components/CommandPalette/TheCommandPalette.vue.js";
import { useDarkModeState as b } from "../../hooks/useDarkModeState.js";
import { handleHotKeyDown as H } from "../../libs/event-busses/hot-keys-bus.js";
const V = ["innerHTML"], F = {
  key: 0,
  class: "flex min-h-0 flex-1 z-0"
}, O = { class: "flex flex-1 flex-col min-w-0" }, ee = /* @__PURE__ */ _({
  __name: "ApiClientApp",
  setup(P) {
    i(() => {
      v(() => {
        document.body.classList.toggle("dark-mode", c.value), document.body.classList.toggle("light-mode", !c.value);
      });
    });
    const { isDark: c } = b(), r = B();
    y(async () => {
      if (localStorage.getItem(K.WORKSPACE)) {
        const o = {};
        let s = 0, t = 0, e = "";
        for (e in localStorage)
          Object.prototype.hasOwnProperty.call(localStorage, e) && (t = (localStorage[e].length + e.length) * 2, s += t, o[e] = (t / 1024).toFixed(2) + " KB");
        o.Total = (s / 1024).toFixed(2) + " KB", console.table(o), D(r);
      } else
        r.workspaceMutators.add({
          uid: "default",
          name: "Workspace",
          proxyUrl: "https://proxy.scalar.com"
        });
      E();
    });
    const m = (o) => H(o);
    i(() => window.addEventListener("keydown", m)), h(() => window.removeEventListener("keydown", m));
    const u = k(
      () => r.activeWorkspace.value && `<style>
  ${M(r.activeWorkspace.value.themeId, {
        fonts: !0
      })}</style>`
    );
    return (o, s) => {
      var t;
      return l(), d(g, null, [
        p("div", { innerHTML: u.value }, null, 8, V),
        a(L),
        (t = n(r).activeWorkspace.value) != null && t.uid ? (l(), d("main", F, [
          a(C),
          a(n(N)),
          p("div", O, [
            a(n(A), null, {
              default: w(({ Component: e }) => [
                (l(), f(S, null, [
                  (l(), f(x(e)))
                ], 1024))
              ]),
              _: 1
            })
          ])
        ])) : T("", !0),
        a(n(W))
      ], 64);
    };
  }
});
export {
  ee as default
};
