// @scalar/api-reference/dist/components/Layouts/ModernLayout.vue2.js
import { defineComponent as k, useSlots as M, watch as l, openBlock as i, createBlock as p, normalizeClass as b, createSlots as h, withCtx as n, unref as d, isRef as y, createCommentVNode as u, createElementVNode as D, createVNode as v, renderList as w, renderSlot as C, normalizeProps as B, guardReactiveProps as N } from "vue";
import { useMediaQuery as $ } from "@vueuse/core";
import H from "../ApiReferenceLayout.vue.js";
import K from "../MobileHeader.vue.js";
import L from "../../features/Search/SearchButton.vue.js";
import R from "../DarkModeToggle/DarkModeToggle.vue.js";
import { useSidebar as T } from "../../hooks/useSidebar.js";
import { useNavState as V } from "../../hooks/useNavState.js";
const z = { class: "scalar-api-references-standalone-search" }, F = /* @__PURE__ */ k({
  __name: "ModernLayout",
  props: {
    configuration: {},
    parsedSpec: {},
    rawSpec: {},
    isDark: { type: Boolean }
  },
  emits: ["toggleDarkMode", "updateContent"],
  setup(c) {
    const s = c, f = M(), m = $("(max-width: 1000px)"), { isSidebarOpen: a } = T();
    l(m, (e, o) => {
      e && !o && (a.value = !1);
    });
    const { hash: g } = V();
    return l(g, (e, o) => {
      e && e !== o && (a.value = !1);
    }), (e, o) => (i(), p(H, {
      class: b({
        "scalar-api-references-standalone-mobile": e.configuration.showSidebar
      }),
      configuration: e.configuration,
      parsedSpec: e.parsedSpec,
      rawSpec: e.rawSpec,
      onUpdateContent: o[2] || (o[2] = (r) => e.$emit("updateContent", r))
    }, h({
      header: n(() => [
        s.configuration.showSidebar ? (i(), p(K, {
          key: 0,
          open: d(a),
          "onUpdate:open": o[0] || (o[0] = (r) => y(a) ? a.value = r : null)
        }, null, 8, ["open"])) : u("", !0)
      ]),
      "sidebar-start": n(({ spec: r }) => {
        var t;
        return [
          D("div", z, [
            v(d(L), {
              searchHotKey: (t = s.configuration) == null ? void 0 : t.searchHotKey,
              spec: r
            }, null, 8, ["searchHotKey", "spec"])
          ])
        ];
      }),
      "sidebar-end": n(() => [
        s.configuration.hideDarkModeToggle ? u("", !0) : (i(), p(d(R), {
          key: 0,
          isDarkMode: e.isDark,
          onToggleDarkMode: o[1] || (o[1] = (r) => e.$emit("toggleDarkMode"))
        }, null, 8, ["isDarkMode"]))
      ]),
      _: 2
    }, [
      w(f, (r, t) => ({
        name: t,
        fn: n((S) => [
          C(e.$slots, t, B(N(S || {})), void 0, !0)
        ])
      }))
    ]), 1032, ["class", "configuration", "parsedSpec", "rawSpec"]));
  }
});
export {
  F as default
};
