// @scalar/snippetz/dist/plugins/csharp/httpclient/httpclient.js
import { encode } from "js-base64";
import { createSearchParams } from "../../../libs/http.js";
const csharpHttpclient = {
  target: "csharp",
  client: "httpclient",
  title: "HttpClient",
  generate(request, configuration) {
    const normalizedRequest = {
      method: "GET",
      url: "",
      ...request
    };
    normalizedRequest.method = normalizedRequest.method.toUpperCase();
    const searchParams = createSearchParams(normalizedRequest.queryString);
    const queryString = searchParams.size ? `?${searchParams.toString()}` : "";
    const url = `${normalizedRequest.url}${queryString}`;
    const lines = [];
    lines.push("using var client = new HttpClient();");
    lines.push("");
    const httpMethod = getHttpMethod(normalizedRequest.method);
    lines.push(`var request = new HttpRequestMessage(${httpMethod}, "${url}");`);
    addHeadersAndAuth(lines, normalizedRequest, configuration);
    addBodyContent(lines, normalizedRequest);
    lines.push("");
    lines.push("using var response = await client.SendAsync(request);");
    return lines.join("\n");
  }
};
function getHttpMethod(method) {
  switch (method) {
    case "GET":
      return "HttpMethod.Get";
    case "POST":
      return "HttpMethod.Post";
    case "PUT":
      return "HttpMethod.Put";
    case "DELETE":
      return "HttpMethod.Delete";
    case "PATCH":
      return "HttpMethod.Patch";
    case "HEAD":
      return "HttpMethod.Head";
    case "OPTIONS":
      return "HttpMethod.Options";
    default:
      return `new HttpMethod("${method}")`;
  }
}
function addHeadersAndAuth(lines, request, configuration) {
  const headers = request.headers || [];
  const cookies = request.cookies || [];
  const authHeader = headers.find((h) => h.name.toLowerCase() === "authorization");
  if (authHeader) {
    const [scheme, parameter] = authHeader.value.split(" ", 2);
    if (scheme && parameter) {
      lines.push(`request.Headers.Authorization = new AuthenticationHeaderValue("${scheme}", "${parameter}");`);
    }
  } else if (configuration?.auth?.username && configuration?.auth?.password) {
    const credentials = encode(`${configuration.auth.username}:${configuration.auth.password}`);
    lines.push(`request.Headers.Authorization = new AuthenticationHeaderValue("Basic", "${credentials}");`);
  }
  const processedHeaders = /* @__PURE__ */ new Map();
  for (const header of headers) {
    const name = header.name;
    const value = header.value;
    if (name.toLowerCase() === "authorization") {
      continue;
    }
    processedHeaders.set(name, value);
  }
  for (const [name, value] of processedHeaders) {
    if (name.toLowerCase() === "accept" && isMediaType(value)) {
      lines.push(`request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("${value}"));`);
    } else if (name.toLowerCase() === "content-type" && request.postData) {
      continue;
    } else {
      lines.push(`request.Headers.TryAddWithoutValidation("${name}", "${value}");`);
    }
  }
  if (cookies.length > 0) {
    const cookieString = cookies.map((cookie) => `${cookie.name}=${cookie.value}`).join("; ");
    lines.push(`request.Headers.TryAddWithoutValidation("Cookie", "${cookieString}");`);
  }
}
function addBodyContent(lines, request) {
  if (!request.postData) {
    return;
  }
  const { mimeType, text, params } = request.postData;
  if (mimeType === "application/json" && text) {
    try {
      const jsonData = JSON.parse(text);
      const prettyJson = JSON.stringify(jsonData, null, 2);
      const rawStringLiteral = createRawStringLiteral(prettyJson);
      lines.push("request.Content = new StringContent(");
      lines.push(`${rawStringLiteral},`);
      lines.push('System.Text.Encoding.UTF8, "application/json");');
    } catch {
      const rawStringLiteral = createRawStringLiteral(text);
      lines.push("request.Content = new StringContent(");
      lines.push(`${rawStringLiteral},`);
      lines.push('System.Text.Encoding.UTF8, "application/json");');
    }
  } else if (mimeType === "application/x-www-form-urlencoded" && params) {
    const fieldNames = params.map((p) => p.name);
    const hasDuplicates = fieldNames.length !== new Set(fieldNames).size;
    if (hasDuplicates) {
      lines.push("var formParams = new List<KeyValuePair<string, string>>");
      lines.push("{");
      for (const param of params) {
        lines.push(`  new("${param.name}", "${param.value}"),`);
      }
      lines.push("};");
      lines.push("request.Content = new FormUrlEncodedContent(formParams);");
    } else {
      lines.push("var formParams = new Dictionary<string, string>");
      lines.push("{");
      for (const param of params) {
        lines.push(`  ["${param.name}"] = "${param.value}",`);
      }
      lines.push("};");
      lines.push("request.Content = new FormUrlEncodedContent(formParams);");
    }
  } else if (mimeType === "multipart/form-data" && params) {
    lines.push("var content = new MultipartFormDataContent();");
    for (const param of params) {
      if (param.fileName !== void 0) {
        lines.push(
          `content.Add(new StreamContent(File.OpenRead("${param.fileName}")), "${param.name}", "${param.fileName}");`
        );
      } else {
        lines.push(`content.Add(new StringContent("${param.value}"), "${param.name}");`);
      }
    }
    lines.push("request.Content = content;");
  } else if (mimeType === "application/octet-stream" && text) {
    lines.push(
      'var content = new ByteArrayContent(System.Text.Encoding.UTF8.GetBytes("' + text.replace(/"/g, '\\"') + '"));'
    );
    lines.push('content.Headers.ContentType = new MediaTypeHeaderValue("application/octet-stream");');
    lines.push("request.Content = content;");
  } else if (text) {
    const rawStringLiteral = createRawStringLiteral(text);
    lines.push("request.Content = new StringContent(");
    lines.push(`${rawStringLiteral},`);
    lines.push(`System.Text.Encoding.UTF8, "${mimeType}");`);
  }
}
function createRawStringLiteral(text) {
  let quoteCount = 3;
  while (text.includes('"'.repeat(quoteCount))) {
    quoteCount++;
  }
  const quotes = '"'.repeat(quoteCount);
  return `${quotes}
${text}
${quotes}`;
}
function isMediaType(value) {
  return /^[a-zA-Z0-9][a-zA-Z0-9!#$&\-\^_]*\/[a-zA-Z0-9][a-zA-Z0-9!#$&\-\^_]*(\s*;\s*[a-zA-Z0-9][a-zA-Z0-9!#$&\-\^_]*=.*)?$/.test(
    value
  );
}
export {
  csharpHttpclient
};
